// @lk/ui-library/src/Component/V1/Footer/FooterAccordion/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";
interface Props {
    expand?: boolean
    column?: boolean
    isMaxHeight?: boolean
    isRTL?: boolean
    contentBgColor: string;
    contentColor: string;
    contentPadding: string;
	isHeaderStyle?:boolean
}

export const AccordionWrapper = styled.div<{ borderBottomStyle:boolean, borderTopStyle: boolean, styleFont?: TypographyENUM, isRTL: boolean, backgroundColor: string, color: string , isHeaderStyle?:boolean}>`
	border-top: ${(props) => props.borderTopStyle === true ? "1px solid var(--light-grey) !important" : "none"};
	text-align: left;
	font-family: ${props => props.isHeaderStyle?"none":`${TypographyENUM.lkSansRegular}`};
	font-size: var(--fs-14);
	font-weight:${props => props.isHeaderStyle?"none":"bold"};
	font:${props => props.isHeaderStyle ? "500 14px/20px Roboto,Arial,Helvetica,sans-serif" : "none"};
	letter-spacing: 1.3px;
	color: var(--darker-grey);
	border: 0px solid var(--transparent);
    background-color: ${props => props.backgroundColor};
    border-bottom: ${(props) => props.borderBottomStyle === true ? props.isHeaderStyle?"1px solid rgba(255,255,255,.2)" :"1px solid var(--light-grey) !important" : "none"};
`;

export const TitleBox = styled.div<{isHeaderStyle?:boolean,stylePadding: string, expand: boolean, isRTL: boolean }>`
	cursor: var(--pointer);
	padding: ${(props) => props.stylePadding ? props.stylePadding : "var(--pd-15) var(--pd-10)"};
	background-color: ${(props) =>
		props.expand === true ? props.isHeaderStyle? "#18cfa8" :"var(--turquoise-blue-dark)" : "var(--grey-100)"};
	border: 1px solid ${(props) =>
		props.expand === true ? "var(--turquoise-blue-dark)" : "var(--grey-100)"};
	border-radius: ${(props) =>
		props.isHeaderStyle === true ? "none" : "10px 10px 10px 10px"};
	display: flex;
	justify-content: var(--space-between);
	height: auto;
	color: ${(props) =>
		props.expand === true ? "var(--text-color)" : "var(--text-color-black)"};
	font-weight: ${(props) =>
		props.isHeaderStyle === true ? "none" : "var(--fw-md)"};
	align-items: center;
	flex-direction: ${props => props.isRTL ? "row-reverse" : "row"};
`;

export const TitleWrapper = styled.div`
	display: flex;
	flex-direction: row;
	width: 100%;
`;

export const Title = styled.span<{isHeaderStyle?:boolean,
    isRTL: boolean, color: string, fontSize: number, fontWeight: number, width?: string
}>`
	width: ${props => props.width ? props.width : "calc(100% - 20px)"};
	display: ${props => props.isRTL ? "table" : "flex"};
	text-align: ${props => props.isRTL ? "end" : "initial"};
	align-items: center;
	font-size: ${props => props.fontSize ? props.fontSize : ""}px;
	letter-spacing: 1.3px;
    color: ${props => props.color ? props.color : ""};
	font-weight: ${(props) =>
		props.isHeaderStyle? "none" :props.fontWeight ? props.fontWeight : ""};
`;

export const Content = styled.div<Props>`
	padding: ${ props => props.contentPadding ? props.contentPadding : "0px var(--pd-15) var(--pd-15) var(--pd-15)"};
	background:  ${ props => props.contentBgColor ? props.contentBgColor : "#fff" };
	border-radius: ${(props) =>
		props.isHeaderStyle === true ? "none" : "0px 0px var(--pd-10) var(--pd-10)"};
	transition: all 0.5s ease-in-out;
    color: ${props => props.contentColor ? props.contentColor : "#fff"};
	/* max-height: ${(props) =>
		props.isMaxHeight === true ? "250px" : ""}; */
	/* overflow: hidden; */
	border-bottom:${props => props.isHeaderStyle ? "2px solid #18cfa8" : "none"};
	font:${props => props.isHeaderStyle ? "500 14px/20px Roboto,Arial,Helvetica,sans-serif" : "none"};
	/* width */
	::-webkit-scrollbar {
		width: 5px;
	}

	/* Track */
	::-webkit-scrollbar-track {
		border-radius: 0;
	}

	/* Handle */
	::-webkit-scrollbar-thumb {
		background: var(--whitish-gray);
		border-radius: 5px;
	}

	/* Handle on hover */
	::-webkit-scrollbar-thumb:hover {
		background: var(--black);
	}
`;

export const FlexCol = styled.div`
	display: flex;
	flex-direction: column;
`;

export const Flex = styled.div`
	display: flex;
`;

export const UpArrow = styled.span`
	border: 5px solid var(--transparent);
	border-top: 5px solid #00bac6;
	margin-top: -10px;
	float: right;
`;

export const DownArrow = styled.span`
	border: 5px solid var(--transparent);
	border-bottom: 5px solid #00bac6;
	margin-top: -15px;
	float: right;
`;

export const Label = styled.label`
	cursor: var(--pointer);
	display: flex;
    align-items: center;
`;

export const Icon = styled.i<{deviceSpecificIssue?: boolean}>`
    margin-right: 20px;
	${props => props.deviceSpecificIssue ? "margin-right: 0" : ""};

	color: #000042;
`;

export const BtnText = styled.span<{isRTL: boolean, deviceSpecificIssue?: boolean}>`
	position: absolute;
	right: ${ props => props.isRTL ? 0: -15}px;
	left: ${ props => props.isRTL ? 0: ""}px;
	top: -10px;
	font-weight: 700;
	font-size: 14px;
	cursor: pointer;
	text-transform: uppercase;

	${props => props.deviceSpecificIssue ? "right: 0" : ""};
`;

export const BtnWrapper = styled.div`
	/* width: 200px; */
	position: relative;
`;

export const Image = styled.img`
	max-width: 100%;
`;

export const StatusIcon = styled.div`
	width: 30px;
	height: 30px;
	margin-right: 12px;
`;
