// @lk/ui-library/src/Component/V1/PDP/Review/ReviewCard/ReviewCard.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import { ReviewCard } from "./ReviewCard.component";

export default {
	title:"Components/V1/PDP/Review/ReviewCard",
	component: ReviewCard,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		}),
		title: generateArgTypes({
			type: "string",
			required: true,
			description: "Title of Review",
			category: "Component",
			defaultValue: "",
		}),
		desc: generateArgTypes({
			type: "string",
			required: true,
			description: "Description of Review",
			category: "Component",
			defaultValue: "",
		}),
		userName: generateArgTypes({
			type: "string",
			required: true,
			description: "User Name of Review",
			category: "Component",
			defaultValue: "",
		}),
		date: generateArgTypes({
			type: "string",
			required: true,
			description: "Date of Review",
			category: "Component",
			defaultValue: "2022-07-24T13:48:57Z",
		}),
		rating: generateArgTypes({
			type: "number",
			required: true,
			description: "Rating given for Review",
			category: "Component",
			defaultValue: "0",
		}),
		images: generateArgTypes({
			type: "array",
			required: true,
			description: "Images of Review",
			category: "Component",
			defaultValue: "",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show review right to left",
			category: "Status",
			defaultValue: "false",
		}),
	}
} as ComponentMeta<typeof ReviewCard>;

const Template: ComponentStory<typeof ReviewCard> = args => <ReviewCard {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	reviewId:"387621028",
	reviewTitle: "OSM AND SUPERB Quality",
	reviewDetail: "Highly Recommended , Its Best Choice For Modern Peoples",
	reviewee: "ershad A.",
	reviewDate: "2022-07-24T13:48:57Z",
	noOfStars: 5,
	images: [],
	font:TypographyENUM.defaultBook,
};
