// @lk/ui-library/src/Component/V1/BuyOnCallWidget/BuyOnCallWidget.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import BuyOnCallWidget from "./index";

export default {
	title: "Components/v1/BuyOnCallWidget",
	component: BuyOnCallWidget,
	argTypes: {
		...generateGeneralArgs(),
		index: generateArgTypes({
			type: "number",
			required: false,
			description: "Index of product image",
			category: "Component",
			defaultValue: 0,
		}),
		imageSrc: generateArgTypes({
			type: "string",
			required: true,
			description: "Image source",
			category: "Component",
			defaultValue: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "Function to be called on click",
			category: "Component",
			defaultValue: () => null,
		}),
	}
} as ComponentMeta<typeof BuyOnCallWidget>;

const Template: ComponentStory<typeof BuyOnCallWidget> = (args) => <BuyOnCallWidget {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	category: true,
	buyWithCallConfig: {
		buyonchat: true,
		buyonchat_link: "https://api.whatsapp.com/send?phone=918929853854&text=Hi Lenskart, help me get started!",
		contact_lens: {
			tel: "8470007367"
		},
		cta: {
			icon: "https://static.lenskart.com/media/desktop/img/09-jan-20/call_icon.png",
			iconGreen: "https://static.lenskart.com/media/mobile/images/call_icon_green.png",
			isShown: true,
			text: "Help?",
			whatsappIcon: "https://static1.lenskart.com/media/mobile/images/whatsapp-smallicon.png",
			whatsappIconGreen: "https://static.lenskart.com/media/mobile/images/whatsapp-smallicon-green.png"
		},
		description: "Tap to whatsapp our eyewear expert",
		enable_new_flow: true,
		enabled: true,
		eyeglasses: {
			tel: "18001020767"
		},
		floating_image: "https://static1.lenskart.com/media/desktop/img/buyonchat1.png",
		icon: "https://static1.lenskart.com/media/mobile/images/whatsapp-gogreen.png",
		pdpbuyonchat_icon: "https://static1.lenskart.com/media/mobile/images/small-whatsapp.png",
		pdpbuyonchat_text: "Buy on Chat",
		sunglasses: {
			tel: "18001020767"
		},
		title: "Buy on Chat"
	}
};
