// @lk/ui-library/src/Component/V1/BuyOnCallWidget/styles.tsx
import styled from "styled-components";

export const BuyWithCall = styled.div`
 a{
	border: none;
    margin: 0px;
	padding: 8px 0 0 0;
    display: flex;
    justify-content: space-between;
 }
 div{
	padding: 0px;
    font-family: LKSans-Regular;
    font-size: 12px;
    line-height: 18px;
    color: rgb(102, 102, 142);
 }
 h3{
	font-size: 16px;
    line-height: 24px;
    font-family: LKSerif-Book;
    color: rgb(0, 0, 66);
 }
`;

export const LinkProvider = styled.a`
    border-top: 1px solid #d8d8d8;
    display: flex;
    align-items: center;
    text-align: left;
    margin-top: 10px;
    padding: 15px 15px 5px 15px;
`;

export const HeadingWithDescription = styled.div`
padding: 0px 16px;
`;

export const Heading = styled.h3`
margin-bottom: 5px;
font-size: var(--fs-17);
color: var(--black);
`;

export const Description = styled.div<{ textColor: boolean }>`
    font-size: var(--fs-14);
    color: ${ props => props.textColor ? "var(--black)" : "var(--dark-grey)" };
`;

export const ImageSpan = styled.span`
    padding-left: 15px;
	img{
		width:28px;
	}
`;

export const Image = styled.img``;
