// @lk/ui-library/src/Component/V1/HomePage/WearTrendWithCarousel/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const WearTrendWithCarouselWrapper = styled.div<{font:TypographyENUM}>`
padding:var(--pd-20);
background-color:var(--white);
.image-slider {
    .slick-next:before, .slick-prev:before {
		display: block;
		color: var(--black);
		font-size:var(--fs-40);
		position:absolute;
		bottom:0;
	}
    .slick-prev:before {
		content: '❮';
		left: var(--spacing-lg);
	}
    .slick-next:before {
		content: '❯';
		right: var(--spacing-lg);
	}
  }
`;

export const Image = styled.img`
	width: 100%;
`;

export const ImageWrapper = styled.div`
display:flex;
flex-direction:column;
gap:var(--sapcing-sm);
justify-content:center;
align-items:center;
margin:var(--spacing-lg)

`;
export const ImageText = styled.p`
font-size: var(--fs-18);
font-weight:var(--fw-medium);
font-family: 'Futura-Medium';
 `;


//  export const WearTrendTextWrapper = styled.div`
//  display:flex;
//  flex-direction:column;
// `;

// export const WearTrendText = styled.p`
// font-size:var(--fs-52);
// margin:0;
// font-family:Rajdhani-Medium;
// font-weight:var(--fs-bold)
// `;
// export const WearTrendSubText = styled.p`
// font-family: Futura-Medium;
// font-size: var(--fs-16);
// font-weight: var(--fw-regular);
// `;


// export const CarouselWrapper = styled.div`
// .image-slider {
//    .slick-next:before, .slick-prev:before {
// 	   display: block;
// 	   color: var(--black);
// 	   font-size:var(--fs-40);
// 	   position:absolute;
// 	   bottom:0;
//    }
//    .slick-prev:before {
// 	   content: '❮';
// 	   left: var(--spacing-lg);
//    }
//    .slick-next:before {
// 	   content: '❯';
// 	   right: var(--spacing-lg);
//    }
//  }
// `;
