// @lk/ui-library/src/Component/V1/HomePage/WearTrendWithCarousel/WearTrendWithCarousel.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import WearTrendWithCarousel from ".";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/V1/HomePage/WearTrendWithCarousel",
	component: WearTrendWithCarousel,
	argTypes: {
		...generateGeneralArgs(),
		title:generateArgTypes({
			type: "string",
			required: true,
			description: "title",
			category: "Component",
		}),
		onViewRangeClick:generateArgTypes({
			type: "function",
			required: true,
			description: "function called on view range button click",
			category: "Event",
		}),
	}
} as ComponentMeta<typeof WearTrendWithCarousel>;

const Template: ComponentStory<typeof WearTrendWithCarousel> = (args) => <WearTrendWithCarousel {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "banner-section",
	images:[
		{
			text:"Round",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static1.lenskart.com/media/desktop/img/Sep21/image179.png"
		},
		{
			text:"Cat-Eye",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static1.lenskart.com/media/desktop/img/Sep21/cateeye.jpg"
		},
		{
			text:"Clubmaster",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static1.lenskart.com/media/desktop/img/Sep21/clubmaster.jpg"
		},
		{
			text:"Transparent",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static1.lenskart.com/media/desktop/img/Sep21/trans.jpg"
		},
		{
			text:"Blent Edit",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static1.lenskart.com/media/desktop/img/Sep21/blend.jpg"
		},
		{
			text:"Air Clip On",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static1.lenskart.com/media/desktop/img/Sep21/clipon.jpg"
		},
		{
			text:"Air Flex",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static1.lenskart.com/media/desktop/img/Sep21/airflex.jpg"
		},
		{
			text:"Retro Aviator",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static1.lenskart.com/media/desktop/img/Sep21/aviator.jpg"
		},

	],
	title:"EYEGLASSES",
	font:TypographyENUM.defaultBook,
	dataLocale:{
		"VIEW_RANGE": "VIEW RANGE",
	},
	onViewRangeClick:()=>console.log("view range clicked")
};
