// @lk/ui-library/src/Component/V1/PDP/Coating/Coating.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const CoatingContainer = styled.div`
	position: relative;
	font-family: var(--font-default-book);
	font-synthesis: none;
	width: 90%;
`;
export const CoatingFirstRow = styled.div`
    font-style: normal;
    font-weight: 700;
    font-size: 21px;
    color: #414b56;
	display: flex;
	justify-content: center;
`;
export const Icon = styled.span`
	display: flex;
	align-items: center;
	margin-right: 7px;
`;
export const BackPressSection = styled.div`
	font-family: ${TypographyENUM.rajdhaniRegular};
	position: absolute;
    font-weight: 700;
    font-size: 15px;
    color: #00bac6;
	left: 0;
	top: 10px;
	cursor: pointer;
	letter-spacing: 1px;
	display: flex;
`;
export const OptionalText = styled.div`
	display: inline-block;
    height: 38px;
    width: 97px;
    padding: 8px 14px;
	font-size: 18px;
    margin-bottom: 10px;
    background: -webkit-gradient(linear,left bottom,left top,from(rgba(255,255,255,.9)),to(rgba(255,255,255,.9))),#FF9800;
    background: linear-gradient(0deg,rgba(255,255,255,.9),rgba(255,255,255,.9)),#FF9800;
    border-radius: 16px;
	color: #FFB400;
`;
export const CoatingSelectSection = styled.div`
	margin: 0 auto;
	border: 1px solid #eee;
    border-radius: 12px;
    overflow: hidden;
	position: relative;
    max-width: 36vw;
	cursor: pointer;
`;
export const SkipCoating = styled.div`
	text-align: -webkit-center;
	padding-bottom: 10px;
	padding-top: 10px;
    font-weight: 700;
	cursor: pointer;
    font-size: 18px;
    color: #00bac6;
	// font-family: var(--font-default-book);
	font-family: "Rajdhani-Regular";
	text-transform: uppercase;
`;
export const CoatingImage = styled.img`
margin: 0;
    padding: 0;
	max-width: 100%;
`;
export const CoatingImageCheck = styled.img`
	margin: 0;
    padding: 0;
	max-width: 100%;
	position: absolute;
    top: 15px;
    right: 15px;
`;
export const CoatingBottomTextContainer = styled.div`
padding: 15px;
font-size: 14px;
margin: 0;
    padding: 0;
`;
export const CoatingTitle = styled.div`
	color: #414b56;
	font-size: 14px;
	font-family: Roboto;
	line-height: 143%;
	letter-spacing: .15px;
	cursor: pointer;
	font-weight: 500;
	font-family: var(--font-default-book);
	font-synthesis: none;
	margin-left: 10px;

`;
export const CoatingPrice = styled.div`
	margin-left: 10px;
	font-family: var(--font-default-heavy);
`;
