// @lk/ui-library/src/Component/V1/PDP/Coating/Coating.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import Coating from "./Coating.component";

export default {
	title: "Components/v1/Coating",
	component: Coating,
	argTypes: {
		...generateGeneralArgs(),
		coatingImg: generateArgTypes({
			type: "string",
			required: true,
			description: "Coating Image",
			category: "Component",
			defaultValue: "https://static.lenskart.com/images/cust_mailer/18-Oct-18/photochromatic-lens.jpg",
		}),
		gobackText: generateArgTypes({
			type: "string",
			required: false,
			description: "Text shown on back button",
			category: "Component",
			defaultValue: "LENS PACKAGE",
		}),
		onBackPress: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for back press",
			category: "Events",
		}),
		coatingTitle: generateArgTypes({
			type: "string",
			required: true,
			description: "Coating Image title",
			category: "Component",
			defaultValue: "Photo-Chromatic Coating (For Power below 6)",
		}),
		coatingPrice: generateArgTypes({
			type: "string",
			required: true,
			description: "Coating Price",
			category: "Component",
			defaultValue: "₹1000",
		}),
		skipCoatingAction: generateArgTypes({
			type: "function",
			required: true,
			description: "onClick handler for skipping coating",
			category: "Events",
		}),
		onSelectCoating: generateArgTypes({
			type: "function",
			required: true,
			description: "onClick handler for selecting coating",
			category: "Events",
		}),
		coatingSelected: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to show/hide Check mark",
			category: "Component",
			defaultValue: "false",
		}),
	},
} as ComponentMeta<typeof Coating>;

const Template: ComponentStory<typeof Coating> = (args) => <Coating {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	coatingImg: "https://static.lenskart.com/images/cust_mailer/18-Oct-18/photochromatic-lens.jpg",
	gobackText: "LENS PACKAGE",
	onBackPress: () => null,
	coatingTitle: "Photo-Chromatic Coating (For Power below 6)",
	coatingPrice: "₹1000",
	skipCoatingAction: () => null,
	onSelectCoating: () => null,
	coatingSelected: false
};
