// @lk/ui-library/src/Component/Common/NeedHelpCta/NeedHelpCta.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import NeedHelpCta from ".";
import { generateArgTypes } from "../../../StorybookHelper";

export default {
	title: "Components/common/NeedHelpCta",
	component: NeedHelpCta,
	argTypes: {
		isRendered: generateArgTypes({
			type: "boolean",
			category: "Component",
			description: "Whether to render the component or not.",
			defaultValue: true,
			required: true,
		}),
		isBuyOnChat: generateArgTypes({
			type: "boolean",
			category: "Component",
			description: "Whether to enable buy on chat option on the component or not.",
			defaultValue: true,
			required: true,
		})
	}
} as ComponentMeta<typeof NeedHelpCta>;

const Template: ComponentStory<typeof NeedHelpCta> = (args) => <NeedHelpCta {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	localeData: { NEED_HELP: "Need help", WHATSAPP_CHAT_URL: "https://api.whatsapp.com/send/?phone=918929853854&text=" },
	whatsappChatMsg:  "Hi Lenskart, help me get started! I am on <pageName> page of mobile site. I need help for product ID: <pid-no>".replace("<pageName>", "Payment").replace("<pid-no>", "147098"),
	phoneNumber: "18001020767",
	onClick: () => {
		console.log("Clicked!");
	}
};
