// @lk/ui-library/src/Component/Common/Breadcrumbs/Breadcrumbs.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../StorybookHelper";
import Breadcrumbs from "./Breadcrumbs.component";
import { breadcrumbData } from "./mockValue";
import { ComponentSizeENUM } from "../../../Types/general";

export default {
	title: "Components/common/Breadcrumbs",
	component: Breadcrumbs,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		productID: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		contactLensPowerFromUrl: generateArgTypes({
			type: "string",
			required: false,
			description: "Redirect Link for Contact lens",
			category: "Component",
			defaultValue: "",
		}),
		helplineNo: generateArgTypes({
			type: "string",
			required: false,
			description: "Helpline line no. that will render ",
			category: "Component",
			defaultValue: "1800-111-111",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "right to left content",
			category: "general",
		}),
		isMobileView: generateArgTypes({
			type: "boolean",
			required: false,
			description: "change the view of breadcrum according to mobile, and hide phone number",
			category: "general",
		}),
		componentSize: generateArgTypes({
			type: "string",
			required: false,
			description: "Size of the component",
			category: "General",
			defaultValue: ComponentSizeENUM.small,
			control: "radio",
			options: ComponentSizeENUM,
		}),
	},
} as ComponentMeta<typeof Breadcrumbs>;

const Template: ComponentStory<typeof Breadcrumbs> = (args) => (
	<Breadcrumbs {...args} />
);
export const Overview = Template.bind({});
Overview.args = {
	id: "Breadcrumbs",
	productID: "147654",
	contactLensPowerFromUrl: "",
	helplineNo: "1800-111-111",
	dataLocale: {
		PROBLEM_IN_PLACING_ORDER: "Problem in placing order ? Give a missed call",
	},
	breadcrumbData: breadcrumbData,
	isRTL: false,
	componentSize: ComponentSizeENUM.medium
};

export const MobileView = Template.bind({});
MobileView.args = {
	id: "Breadcrumbs",
	productID: "147654",
	contactLensPowerFromUrl: "",
	helplineNo: "1800-111-111",
	dataLocale: {
		PROBLEM_IN_PLACING_ORDER: "Problem in placing order ? Give a missed call",
	},
	breadcrumbData: breadcrumbData,
	isRTL: false,
	// isMobileView: true,
	componentSize: ComponentSizeENUM.small
};
