// @lk/ui-library/src/Component/V1/MyOrderMobile/OrderTrackingMobile/OrderTrackingMobile.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const Container = styled.div`
	font-family: ${TypographyENUM.lkSansRegular};
	color: #333333;
	padding: 15px 20px;
`;

export const Heading = styled.div`
	font-size: 14px;
	font-family: ${TypographyENUM.lkSansBold};
	margin: 0 0 25px;
`;

export const Tracking = styled.div`
	padding-left: 10px;
	position: relative;
`;

export const CurrentIcon = styled.div`
	position: absolute;
	left: 0;
	-webkit-transform: translateX(-50%);
    transform: translateX(-50%);
	span {
		display: inline-block;
		margin-top: -7px;
    	width: 30px;
    	height: 30px;
	}

	span img {
		width: 100%;
		height: 100%;
	}
`;

export const Checkpoint = styled.div`
	position: absolute;
	left: 0;
	-webkit-transform: translateX(-50%);
    transform: translateX(-50%);
	span {
		display: inline-block;
		border: 1px solid #dddddd;
		width: 10px;
		height: 10px;
		margin-left: -2px;
		border-radius: 50%;
		background-color: #ffffff;
	}
`;

export const StageData = styled.div`
	padding-bottom: 13px;
	font-size: 12px;
	border-bottom: 1px solid #eeeeee;
`;

export const Stage = styled.div`
	z-index: 10;
	border-left: 2px solid #eeeeee;
	padding-bottom: 13px;
	padding-left: 25px;
	position: relative;
	&:first-child {
		${Checkpoint} {
			margin-top: -7px;
		}
	}
	&:last-child {
		border-left-color: transparent;
    	padding-bottom: 0;
		${Checkpoint} {
			margin-top: -7px;
		}
		${StageData} {
			border-bottom: none;
		}
	}
`;

export const StageTitle = styled.div`
	display: flex;
`;

export const StageName = styled.div<{ current: boolean }>`
	flex-basis: 64%;
	color: ${props => props.current ? "#FAA61A" : "" };
	span {
		text-align: left;
		font-family: ${TypographyENUM.lkSansBold};
	}
`;

export const StageDate = styled.div`
	flex-basis: 20%;
	span {
		text-align: right;
		color: #666666;
	}
`;

export const StageSubText = styled.div`
	font-size: 12px;
	margin-top: 5px;
	color: #666666;
`;

export const Dropdown = styled.div<{ showDescription: boolean }>`
	width: 10%;
	text-align: right;
	div {
		font-family: ${TypographyENUM.lkSansBold};
		display: inline-block;
		transition: transform 0.5s ease-in-out;
		transform: ${props => props.showDescription ? "rotate(180deg)" : "rotate(0deg)"};
	}
`;

export const DescriptionContainer = styled.div<{ showDescription: boolean }>`
	width: 100%;
	overflow: hidden;
	padding-top: ${props => props.showDescription ? "10px" : "0px"};
	height: ${props => props.showDescription ? "210px" : "0px"};
	transition: height 0.5s ease-in-out, padding 0.5s ease-in-out;
`;

export const Description = styled.div`
	width: 100%;
`;

export const Text = styled.div`
	color: #666666;
`;

export const Media = styled.div`
	padding-top: 10px;
	z-index: 99999;
	img {
		width: 100%;
	}
`;

export const TrackingArrow = styled.div`
background-image: url(https://static.lenskart.com/media/Assets/Images/down-arrow.svg);
width: 100%;
height: 100%;
background-repeat: no-repeat;
`;
