// @lk/ui-library/src/Component/V1/Header/MainHeader/SecondRow/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const SecondRowContainer = styled.div<{isExchangeFlow?: boolean}>`
	display: flex;
	justify-content: var(--space-between);
	width: 100%;
	align-items: center;
	padding: ${props => props.isExchangeFlow ? "0 10px" : "0 5px"};
	@media screen and (max-width:767px){
	  border-bottom:1px solid #0000001f;
	}
`;
export const ImageContainer = styled.div<{isRTL?: boolean}>`
	display: flex;
	justify-content: var(--space-between);
	align-items: center;
	flex-basis: 23.6%;
	min-width: 300px;
	align-items: center;
    max-height: 50px;
	margin-right: 0px;
	a{
		img{
			cursor:pointer;
		}
	}
	/* overflow: hidden; */
	@media (max-width: 1054px) {
		flex-basis: 15%;
	}
`;

export const MsiteImageContainer = styled.div<{isRTL?: boolean}>`
	display: flex;
	align-items: center;
	align-items: center;
    max-height: 50px;
	margin-right: ${props => !props.isRTL ? "5px" : ""};
	/* overflow: hidden; */
	img[alt="ph-no"]{
		width: ${props => props.isRTL ? "95px" : ""};
	}
`;

export const ExchangeFlowTitleContainer = styled.div`
	display: flex;
	flex-direction: row;
	font-family: var(--font-lksans-regular);
	h1{
		font-size: 14px;
		margin: 5px;
    	text-transform: uppercase;
    	color: #333333;
		font-weight: normal;
	}
`;

export const CircularBack = styled.div`
	background-color: #18cfa8;
	color: #fff;
	border-radius: 50%;
	width: 25px;
	height: 25px;
	text-align: center;
	align-items: center;
	display: flex;
	justify-content: center;
`;

export const HeaderImage = styled.img<{isRTL?:boolean}>`
	height: 100%;
	display: block;
	height: 57px;
	${props => props.isRTL ? "margin-right: -10px;" : "margin-left: -10px;"};
    margin-top: 3px;
	@media screen and (max-width:767px){
		margin-left: 0px;
		margin-top: 0px;
	}
`;
export const PhoneNumberImage = styled(HeaderImage)`
	height: 33px;
	width: 125px;
	transform: translateX(30px);
	margin-right: -30px;
	margin-left:0px;
	@media screen and (max-width:1680px){
		transform: translateX(0px);
		margin-right: -17px;
	}
	@media screen and (max-width:1099px){
		transform: translateX(15px);
		margin-right: 0;
		width: 110px;
        height: auto;
	}
	@media screen and (max-width:676px){
		transform: translateX(10px);
		width:auto;
		margin-right:0px;
		margin-left: -11px;
	}
`;
export const ActionLinksContainer = styled.div<{showActionLink: boolean}>`
	display: flex;
	align-items: center;
	width: 414px;
    justify-content: space-between;
    // padding-left: 10px;
	margin-left: 10px;
	margin-right: 10px;
	visibility: ${(props) => props.showActionLink ? "" : "hidden"};

	@media (max-width: 1520px) {
		width: 390px;
	}
`;

export const SpanButton = styled.span`
	margin: auto 3px;
`;

export const ActionLink = styled.div<{styleLoading: boolean}>`
	font-size: 12px;
	line-height: 20px;
	letter-spacing: -.02em;
	color: ${(props) => props.styleLoading ? "#EEE" : "var(--dark-blue-100)"};
	font-family: ${TypographyENUM.lkSansRegular};
    white-space: nowrap;
	display: flex;
    padding: 0;
	position: relative;
	// padding-left: 30px;
	cursor: ${(props) => props.styleLoading ? "no-drop" : "pointer"};
	align-items: center;
	&.lanswitcher{
		margin-right:3px;
	}
`;

export const SignInDropBox = styled.div`
	display: block;
	position: absolute;
	width: 170px;
	top: 17px;
	left: -38px;
	background: #fff;
	z-index: 21;
	-webkit-box-shadow: 0 3px 6px rgb(0 0 0 / 16%), 0 3px 6px rgb(0 0 0 / 23%);
	box-shadow: 0 3px 6px rgb(0 0 0 / 16%), 0 3px 6px rgb(0 0 0 / 23%);
	margin: 0;
    padding: 0;
	list-style: none;
    font-family: ${TypographyENUM.lkSansRegular};
`;

export const SignInDropActions = styled.a`
	display: block;
	padding: 10px;
	font-size: 12px;
    line-height: 20px;
    letter-spacing: -0.02em;
    color: var(--dark-blue-100);
    font-family: ${TypographyENUM.lkSansRegular};
	text-decoration: none;
	&:hover {
		background-color: #fafafa;
	}
`;

export const CircleFont = styled.div<{isRTL: boolean}>`
	font-size: 12px;
    color: #fff;
    position: absolute;
    background: #329c92;
    width: 18px;
    height: 18px;
    top: -9px;
	${props => props.isRTL ? "right: -7px" : "left: 14px"};
    border-radius: 50%;
    line-height: 18px;
    text-align: center;
	letter-spacing: -0.02em;
	font-family: ${TypographyENUM.lkSansRegular};
`;

export const IconImage = styled.div<{styleLoading: boolean,isRTL: boolean}>`
    svg{
		height: 20px;
		width: 20px;
     }
	margin-right: 13px;
	${props => props.isRTL ? "margin-left: 5px;" : "margin-left: 2px;"};
	filter: ${({ styleLoading }) => styleLoading ? "opacity(0.2)" : "none"};
`;


// Msite Styles

export const HeaderWrapper = styled.div`
background-color: #fff;
position: fixed;
z-index: 100;
width: 100%;
box-shadow: 0 1px 1px 0 rgb(148 150 159 / 30%);
display:flex;
align-items: center;
`;


export const HeaderMenu = styled.div`
background-color: transparent;
    transition-duration: 0.35s;
    transition-property: background-color;
    transition-timing-function: ease;
`;

export const HeaderIcon = styled.span<{
	isSelected:boolean
}>`
font-size: 17px;
padding: 18px 5px 14px;
color: ${props => props.isSelected ? "#fff" : "#27394e"};
margin:${props => props.isSelected ? "0px" : "0px"};
background-color:${props => props.isSelected ? "#27394e" : "none"};
display:flex;
margin-top: -2px;
width:33px;
svg{
	width: 20px;
    height: 20px;
}
&.mobile-menu{
	margin-left:-5px;
	svg{
		fill:${props => props.isSelected ? "#fff" : "#27394e"};
		width: auto;
        height: auto;
		margin:auto;
	}
	+ a{
		margin-left:0px;
	}
	@media screen and (max-width:320px){
		+ a + a{
			display:none
		}
	}
}

`;

export const HeaderLogo = styled.div`
height: 18px;
display:flex;
align-items: center;
`;

export const HeaderRightWrapper = styled.div`
display: flex;
align-items:center;
gap: 1px;
`;
export const HeaderRightItem = styled.div`
display: inline-block;
position: relative;
`;

export const MsiteCircleFont = styled.div<{isRTL: boolean}>`
position: absolute;
top: 7px;
// left: -5px;
color: #fff;
background: #fd6051;
font-weight: 500;
border-radius: 50%;
font-size: 11px;
display: inline-block;
width: 19px;
height: 19px;
text-align: center;
line-height: 19px;
font-family: ${TypographyENUM.lkSansRegular};
`;

export const Icon = styled.img`
	margin-bottom: 5px;
	&[alt="qr-scanner"]{
		margin-right: 10px;
		margin-bottom: 0px;
	}
`;
