// @lk/ui-library/src/Component/V1/SliderCarousel/SliderCarousel.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import SliderCarousel from "./index";

export default {
	title: "Components/v1/SliderCarousel",
	component: SliderCarousel,
	argTypes: {
		...generateGeneralArgs(),
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		sliderImages: generateArgTypes({
			type: "array",
			required: true,
			description: "images for slider",
			category: "General",
		}),
		autoPlay: generateArgTypes({
			type: "boolean",
			required: false,
			description: "whether to autoplay images",
			category: "General",
		}),
		duration: generateArgTypes({
			type: "number",
			required: false,
			description: "time between autoscroll of images",
			category: "General",
		})
	},
} as ComponentMeta<typeof SliderCarousel>;

const Template: ComponentStory<typeof SliderCarousel> = args => <SliderCarousel {...args} />;
export const Overview = Template.bind({});
Overview.args = {
	id: "slider-carousel-main",
	sliderImages: [
		{ id: 0, imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/transparent-full-rim-rectangle-vincent-chase-classic-acetate-vc-e14999-c3-eyeglasses_g_0036_12_07_2022.jpg" },
		{ id: 1, imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/transparent-full-rim-rectangle-vincent-chase-classic-acetate-vc-e14999-c3-eyeglasses_29_july_vc_m_f115539_204093_01_aug.jpg" },
		{ id: 2, imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/transparent-full-rim-rectangle-vincent-chase-classic-acetate-vc-e14999-c3-eyeglasses_22_july_female110313_27_july.jpg" }
	],
	autoPlay: true,
	duration: 2
};
