// @lk/ui-library/src/Component/V1/SliderCarousel/styles.tsx
import styled from "styled-components";

export const SliderMain = styled.div`
	width: 100%;
	box-sizing: border-box;
    margin: 0;
    padding: 0;

`;

export const SliderHolder = styled.div`
	touch-action: pan-y;
    -webkit-tap-highlight-color: transparent;
	position: relative;
    display: block;
	margin: 0;
    padding: 0;
`;
export const SlidesContainer = styled.div`
	overflow: hidden;
    margin: 0;
    padding: 0;
	position: relative;
    display: block;
`;
export const SliderList = styled.div<{activeIndex: number; isRTL: boolean}>`
	top: 0;
    left: 0;
	position: relative;
    display: block;
	box-sizing: border-box;
	white-space: nowrap;
    transition: transform 0.4s;
    margin: 0;
    padding: 0;
	transform: translateX(${props => props.activeIndex ? (props.isRTL ? 100 : -100) * props.activeIndex : 0}%);
`;
export const Slide = styled.div`
	outline: none;
	display: inline-block;
    height: 100%;
    min-height: 1px;
	box-sizing: border-box;
    margin: 0;
    padding: 0;
	width: 100%;
`;
export const SlideInline = styled.div`
	width: 100%;
    display: inline-block;
`;
export const SlideImg = styled.img`
	width: 100%;
	display: block;
	margin: 0 auto;
	vertical-align: middle;
	border: 0;
`;
export const SlideControllerList = styled.ul`
	position: absolute;
    bottom: 0;
    display: block;
    width: 100%;
    padding: 0;
    list-style: none;
    text-align: center;
`;
export const SlideList = styled.li`
	position: relative;
    display: inline-block;
    width: 20px;
    height: 20px;
    margin: 0 5px;
    padding: 0;
    cursor: var(--pointer);
`;
export const SlideNavigateBtn = styled.button<{active: boolean;}>`
	font-size: 0;
    line-height: 0;
    display: block;
    width: 20px;
    height: 20px;
    padding: 5px;
    cursor: var(--pointer);
    color: transparent;
    border: 0;
    outline: none;
    background: 0 0;
	&:before{
		font-family: slick;
		line-height: 20px;
		position: absolute;
		top: 0px;
		left: 0px;
		width: 20px;
		height: 20px;
		content: "•";
		text-align: center;
		opacity: ${props => props.active ? 1 : "0.25"};
		color: ${props => props.active ? "#289f95" : "rgb(0, 0, 0)"};
		font-size: 20px;
	}
	&:hover{
		&:before {
			opacity: 1;
			color: #289f95;
		}
	}
`;
