// @lk/ui-library/src/Component/V1/Auth/SignIn/SignIn.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import { SignInType, TypographyENUM } from "../../../../Types/general";
import { SignInComponent } from "./SignIn.component";

export default {
	title:"Components/V1/SignIn",
	isMobileView: true,
	component: SignInComponent,
	argTypes: {
		...generateGeneralArgs(),
		onSignIn: generateArgTypes({
			type: "function",
			required: false,
			description: "onSignIn handler for sign in form",
			category: "Events",
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			defaultValue:{
				SIGN_IN: "Sign In",
				ENTER_ONE_TIME_PASSWORD_SENT_TO: "",
				RESEND_OTP: "",
				PROCEED: "",
				FORGOT_PASSWORD: "",
				NEW_MEMBER: "",
				CREATE_ACCOUNT: ""
			},
			control:"object"
		}),
		signInStatus: generateArgTypes({
			type: "object",
			required: true,
			description: "signIn info data",
			category: "Component",
			defaultValue:{
				type: "email",
				showLogin: true,
				showOTP: false,
				sendOTP: false,
				showPassword: false,
				isLoading: false,
				isError: false,
				errorMessage: null,
				otpSent: false
			},
			control:"object"
		}),
		onClose: generateArgTypes({
			type: "function",
			required: false,
			description: "onSignIn handler for sign in form",
			category: "Events",
		}),
		moveToSignUp: generateArgTypes({
			type: "function",
			required: false,
			description: "onSignIn handler for sign in form",
			category: "Events",
		}),
		onProceed: generateArgTypes({
			type: "function",
			required: false,
			description: "onProceed handler for sign in form",
			category: "Events",
		}),
		countryCode: generateArgTypes({
			type: "string",
			required: true,
			description: "country code",
			category: "Component",
			defaultValue: "+91",
		}),
		signInImgLink: generateArgTypes({
			type: "string",
			required: true,
			description: "Sign In Image Link",
			category: "Component",
			defaultValue: "https://static1.lenskart.com/media/desktop/img/July22/homec_03.jpg",
		}),
		loaderImageLink: generateArgTypes({
			type: "string",
			required: true,
			description: "Loader Image Link",
			category: "Component",
			defaultValue: "//static.lenskart.com/skin/frontend/base/default/images/loader2.gif",
		}),
		showOTPView: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show OTP view",
			category: "Component",
			defaultValue: false,
		}),
		isLoginSuccess: generateArgTypes({
			type: "boolean",
			required: true,
			description: "if login is successful",
			category: "Component",
			defaultValue: false,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show to right to left",
			category: "Status",
			defaultValue: false,
		}),
	}
} as ComponentMeta<typeof SignInComponent>;

const Template: ComponentStory<typeof SignInComponent> = args => <SignInComponent {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"signin-form",
	dataLocale:{
		SIGN_IN: "Sign In",
		ENTER_ONE_TIME_PASSWORD_SENT_TO: "",
		RESEND_OTP: "",
		PROCEED: "",
		FORGOT_PASSWORD: "",
		NEW_MEMBER: "",
		CREATE_ACCOUNT: "",
		OtpTime:30,
	},
	font:TypographyENUM.defaultBook,
	onSignIn: () => console.log("api called and user signed in"),
	onProceed: () => console.log("api called and otp sent"),
	countryCode:"+91",
	signInStatus:{
		type: SignInType.EMAIL,
		showLogin: true,
		showOTP: false,
		sendOTP: false,
		showPassword: false,
		isLoading: false,
		isError: false,
		errorMessage: null,
		otpSent: false
	},
	onClose: () => null,
	moveToSignUp: () => null,
	isRTL: false,
	signInImgLink: "https://static1.lenskart.com/media/desktop/img/July22/homec_03.jpg",
	loaderImageLink: "//static.lenskart.com/skin/frontend/base/default/images/loader2.gif",
	isMobileView: true
};
