// @lk/ui-library/src/Component/V1/Auth/SignIn/styles.ts
import styled, { css } from "styled-components";
import { devices } from "../../../../helper";
import { TypographyENUM } from "../../../../Types/general";

export const InlineLoader = styled.img`
    display: block;
    width: 80px;
    margin: 0 auto;
`;
export const InfoText = styled.div`
	margin-bottom: 24px;
	font-family: var(--font-lksans-hairline);
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 20px;
	letter-spacing: -0.02em;
	color: var(--text);
	@media screen and (min-width:768px) and (max-width:1439px){
		margin-bottom: 0px;
		font-size: 12px;
		margin-top: -10px;
		button{
			font-size: 12px;
		}
	}
`;

export const CheckoutSigninContainer = styled.div<{ isMobileView?: boolean, isRTL?: boolean }>`
	width: 100%;
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	padding: 0px 0px 10px;
	margin-top:-10px;
	@media screen and (max-width:767px){
		& .otp-sent{
			margin:-13px 0 -5px 0;
		}
	}
	&.enterAndEditField{
		gap:10px;
		margin-top:0px;
		.spacer5{
			height:5px;
		}
		.spacer10{
			height:10px;
		}
		.spacer15{
			height:15px;
		}
		@media screen and (max-width:767px){
			gap:0px;
		}
		@media screen and (max-width:767px){
		.spacer5,.spacer10,.spacer15{
			height:20px;
		}
		.spacer5{
			height:10px;
		}
	}
	}
	@media ${devices.laptop} {

		${props => !props.isRTL && !props.isMobileView &&
		`max-width: 420px};
			`}

		${props => props.isRTL &&
		`max-width: 450px;
			`}
	}
`;
export const FlexCenter = styled.div`
	display: flex;
	align-items: center;
`;

export const WhatsAppText = styled.span`
	font-family: ${TypographyENUM.lkSansRegular};
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 18px;
    letter-spacing: -0.02em;
    color: #333368;
	margin-right: 5px;
	@media screen and (min-width:768px) and (max-width:1439px){
		font-size: 10px;
	}
`;
export const Verify = styled.button`
width: 72px;
font-family: ${TypographyENUM.lkSansBold};
font-size: 12px;
border-radius: 100px;
cursor: pointer;
background-color: #cccccc;
	margin: 12px;
	margin-left: 0;
	padding: 4px;
	margin-bottom: 20px;
`;

export const Captcha = styled.div<{ isRTL?: boolean, isCompact?: boolean }>`
margin: ${props => props.isRTL ? "10px 0" : "10px auto"};
${props => props.isRTL ? "margin-right: 0;" : "margin-left: 0;"}
margin-bottom: ${props => props.isCompact ? "10px" : "20px"};
`;

export const SigninInfoField = styled.p`
	margin-bottom: 20px;
    font-family: ${TypographyENUM.lkSansRegular};
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    letter-spacing: -0.02em;
    color: #333368;
`;

export const ErrorMessage = styled.div`
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 12px;
    letter-spacing: -0.02em;
    color: var(--error);
    margin: 18px auto;
    height: 12px;
    width: 100%;
`;
export const SuccessMessage = styled.div`
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 12px;
	letter-spacing: -0.02em;
	color: #319731;
	margin: 18px auto;
	height: 12px;
	width: 100%;
`;

export const ForgotPasswordContainer = styled.div<{ isSignin?: boolean; msiteForgotPassword?: boolean }>`
	${props => props.isSignin ? `background-color: #fff;
	border-radius: 12px;
	padding: 30px 48px 18px;` : ""}
	@media screen and (min-width:768px) and (max-width:1439px){
		${props => props.isSignin ? `
	padding: 2px 48px 18px;` : ""}
	}

	${props => props.msiteForgotPassword && css`
		background: #f3fbef;
		overflow-y: auto;
		min-height: calc(100vh - 50px);
		padding:0px;

	`}
`;


export const ForgotPasswordContainerMobile = styled.div`
	height: 100vh;
	background: #F3FBEF;
`;

export const CheckoutSigninContainerMobileContent = styled.div`
	padding: 20px 20px 0;
`;

export const ButtonContainer = styled.div`
	width: 100%;
	padding: 0 16px;
`;

export const CheckoutSigninContainerMobile = styled.div`
	display: flex;
	flex-direction: column;

	Button{
		font-family: ${TypographyENUM.lkSansRegular};
		font-style: normal;
		font-weight: 700;
		font-size: 16px;
		line-height: 24px;
		letter-spacing: -0.02em;
		color: #000042;
	}
`;

export const CheckoutSigninHeader = styled.div`
	font-family: ${TypographyENUM.lkSerifNormal};
	font-style: normal;
	font-weight: 400;
	font-size: 24px;
	line-height: 36px;
	letter-spacing: -0.02em;
	color: #333368;
	margin-bottom:30px;
`;


export const MobileBottomButtonSection = styled.div`
	position: fixed;
	bottom: 16px;
  width: 100%;
  left: 0px;
  padding: 0 16px;

	Button{
		font-size:16px;
		font-family: ${TypographyENUM.lkSansRegular};
		font-weight: 700;
		letter-spacing: -0.02em;
	}
`;
export const ButtonWrapper = styled.div`
	width: 100%;
	margin-bottom: 24px;
	& #create-account-button{
		font-size:16px;
		height:50px;
	}
`;

export const SignInContainerHeader = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-between;
`;

export const CrossContainer = styled.div<{ isRTL?: boolean }>`
top: 22px;
right: ${props => props.isRTL ? "" : "26px"};
left: ${props => props.isRTL ? "26px" : ""};
cursor:pointer;
position: absolute;
`;

export const NewOTP = styled.div`
font-family: ${TypographyENUM.lkSansRegular};
font-size: 14px;
`;

export const Image = styled.img<{ isMobilView?: boolean }>`
    width: 100%;
    height: ${props => props.isMobilView ? "48vh" : "100%"};
    max-height: ${props => props.isMobilView ? "48vh" : ""};
    display: block;
    max-height: ${props => props.isMobilView ? "48vh" : ""};
`;

export const PrivacyPolicyText = styled.div<{ cartSignUp?: boolean }>`
	font-size: ${props => props.cartSignUp ? "12px" : "10px"};

	${props => props.cartSignUp ? "margin-bottom: 15px; margin-top: 20px;" : ""}
	@media screen and (max-width:767px){
		${props => props.cartSignUp ? "margin-bottom: 15px; margin-top: 0px;" : ""}
	}
`;

export const ForgotPasswordOuter = styled.div<{ negativeMargin?: boolean }>`
	${props => props.negativeMargin ? "margin: -15px;" : ""};
`;

export const ResendOtpText = styled.span`
	font-family: ${TypographyENUM.lkSansRegular};
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 18px;
    letter-spacing: -.02em;
    color: #333368;
	b {
		font-weight: 700;
		font-family: ${TypographyENUM.lkSansBold};
	}
`;

export const PhoneCodeDropdownWrapper = styled.div<{top: string}>`
	position: relative;
	top: ${props => props.top}px;
	input{
		width:25px;
	}
`;

export const LoggedInPresalesContainer = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	gap: 10px;
`;

export const InfoMessage = styled.div`
	font-family: var(--font-lkserif-normal);
    font-style: normal;
    font-weight: 400;
    font-size: 18px;
    line-height: 24px;
    letter-spacing: -0.02em;
    color: var(--text);
`;


export const ForgotPasswordSignInWrapper = styled.div`
margin: 20px;
`;
