// @lk/ui-library/src/Component/V1/Auth/components/SelectField.tsx
import React, { useEffect } from "react";
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

const CheckBox = styled.div<{isRTL: boolean}>`
	display: inline-block;
	${props => props.isRTL ? "margin-left: 8px" : "margin-right: 8px"};
	height: 16px;
	width: 16px;
	border-radius: 4px;
	border: 1px solid var(--dark-blue-100);
	background: #F5F5FF;
	cursor: pointer;
	position: relative;
	overflow: hidden;
	&#whatsapp-updates{
	   height: 18px;
	   width: 18px;
	}
`;
const MainContainer = styled.span<{isMobileView:boolean, isCheckout:boolean}>`
    display: inline-block;
    font-family: ${TypographyENUM.lkSansRegular};
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    letter-spacing: -0.02em;
    color: var(--text);
    margin-bottom: 20px;
	margin-top:10px;
	margin-top: ${props => props.isCheckout ? "0" : "10px"};
    display: flex;
	@media screen and (max-width:767px){
		margin-top:0px;
	}
	@media screen and (min-width:768px) and (max-width:1439px){
		margin-bottom: 10px;
		margin-top:5px;
	}
	@media screen and (max-width: 1300px) and (max-height: 900px) and (orientation: landscape) {
		margin-top:0px;
    }
    align-items: center;

		label{
			font-size: ${props => props.isMobileView ? "12px" : "14px"}
		}
`;
const CheckMark = styled.div`
    display: block;
    border-width: 0 1px 1px 0;
    border-color: var(--dark-blue-100);
    border-style: solid;
    transform: rotate(45deg);
    width: 4px;
    height: 9px;
    position: absolute;
    top: 2px;
    left: 5px;
`;

const SelectField = ({
	setGetsAppUpdate, children, isRTL = false, isMobileView=false,isCheckout=false, isPreChecked = false, selected
}: { children?: React.ReactNode, setGetsAppUpdate: (props?: any) => void | undefined, isRTL?: boolean, isMobileView?: boolean, isCheckout?:boolean, isPreChecked?: boolean; selected?: boolean }) => {
	const [showCheckMark, setShowCheckMark] = React.useState(isPreChecked ? true : false);
	// const getUpdateOnWhatsapp = (event: React.MouseEvent<HTMLElement>) => {
	const getUpdateOnWhatsapp = () => {
		setShowCheckMark(showcheck => !showcheck);

		if(typeof setGetsAppUpdate === "function") setGetsAppUpdate();
		// action("get update on whatsapp")(event, { getUpdateOnWhatsapp:event.target.checked });
	};

	useEffect(() => {
		if (selected) setShowCheckMark(selected);
	}, [selected]);

	return (
		<MainContainer isMobileView={isMobileView} isCheckout={isCheckout} >
			<CheckBox
				tabIndex={0}
				onKeyUp={(e) => {
					if (e.key === "Enter") getUpdateOnWhatsapp();
				}}
				id="whatsapp-updates"
				onClick={getUpdateOnWhatsapp}
				isRTL={isRTL}
			>
				{showCheckMark && <CheckMark />}
			</CheckBox>
			<label>{children}</label>
		</MainContainer>
	);
};

export { SelectField };
