// @lk/ui-library/src/Component/V1/AddressTextField/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const FieldInput = styled.input<{ isRTL: boolean, fontSize?: string; rightPadding?: boolean}>`
	width: 100%;
	height: 100%;
	border: none;
	outline: none;
	color: #000042;
	font-family: ${TypographyENUM.lkSansRegular};
	background: transparent;
	${props => props.isRTL ? "padding-left: 15px;" : "padding-right: 15px;" };
	${({ fontSize }) => fontSize ? `font-size: ${fontSize}` : "font-size:14px"};
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}

	${props => props.isRTL && props.rightPadding ? "direction: rtl; padding-right: 15px;" : ""}
`;
export const LeftPositioned = styled.div<{isRTL?: boolean}>`
	left: 0;
	width: fit-content;
	height: 100%;
	display: flex;
    align-items: center;
`;

export const CrossSection = styled.div<{isRTL: boolean, shiftCross?: boolean}>`
	position: absolute;
	${props => props.isRTL ? "left: 10px" :"right: 10px"};
	cursor: pointer;
	height: 12px;
	width: 12px;
	display: flex;
    align-items: center;
    justify-content: center;
    top: 40%;

		${props => props.shiftCross && props.isRTL ? "right: 10px; left: auto;" : ""}
`;
