// @lk/ui-library/src/Component/V1/CheckoutMobile/LkCash/LkCash.styles.tsx
import styled, { css } from "styled-components";

export const CommonSection = styled.div<{mergedWithLKCash?: boolean}>`
  display: flex;
  flex-direction: row;
  padding: 16px;
  background: #FFFFFF;
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
  border-radius: 12px;

  display:flex;
  align-items:center;
  justify-content:space-between;

  ${props => props.mergedWithLKCash && css`
    border-top-left-radius: 0;
    border-top-right-radius: 0;
    box-shadow: none;
  `}
`;

export const FirstSection = styled.div<{isRTL: boolean}>`
  #coupon_code {
	font-family: var(--font-lksans-bold);
    font-style: normal;
    font-weight: 500;
    font-size: 12px;
    line-height: 20px;
    letter-spacing: -0.02em;
    color: #27394e;
  }

  #coupon_status {
    vertical-align: middle;
    margin-top: -2px;
    background: #B1EAEB;
    width: 50px;
    height: 15px;
    font-size: 8px;
    border-radius: 100px;
    display: flex;
    justify-content: center;
    align-items: center;
    margin-left: ${props => props.isRTL ? 0 : 5}px;
    margin-right: ${props => props.isRTL ? 5 : 0}px;
    display: inline-flex;
    font-weight: 500;
    color: #0A5D5B;
    text-transform: capitalize;
  }

  div{
    span:first-child {
      margin-right: 4px;
    }
	font-family: var(--font-lksans-regular);
    font-style: normal;
    font-weight: 400;
    font-size: 12px;
    line-height: 20px;
    letter-spacing: -0.02em;
    color: rgba(60,60,60,.54);
  }
`;

export const SecondSection = styled.div<{isRTL: boolean}>`
  div{
	font-family: var(--font-lksans-regular);
    font-style: normal;
    font-weight: 700;
    font-size: 12px;
    line-height: 18px;
    display: flex;
    align-items: center;
    letter-spacing: -0.02em;
    text-transform: capitalize;
    color: #000042;
  }
  img {
    transform: ${props => props.isRTL ? "rotate(180deg)": "rotate(0deg)"};
  }
`;
