// @lk/ui-library/src/Component/V1/CheckoutMobile/Atome/PayLater.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { PayLater } from "./PayLater.component";

export default{
	title: "Components/v1/CheckoutMobile/PayLater",
	component: PayLater,
	argTypes: {
		methodData: generateArgTypes({
			category: "General",
			required: true,
			type: "object",
			description: "Contains the payment method data",
		})
	}
} as ComponentMeta<typeof PayLater>;

const Template: ComponentStory<typeof PayLater> = (args) => <PayLater {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	localeData : {
		"ATOME": "Atome"
	},
	groupData: { methods:[
		{
			logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png",
			key: "ATOME",
			label: "ATOME",
			offerText: "Flat 50% off"
		},
		{
			logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png",
			key: "HOOLAH",
			label: "HOOLAH"
		}
	] },
	renderPrimerUI: (key) => console.log({ key }),
	setPayLaterSelected: () => null
};
