// @lk/ui-library/src/Component/V1/CheckoutMobile/Atome/PayLater.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const PayLaterWrapper = styled.div`
  box-shadow: 0px 0px 12px rgba(0, 0, 66, .06);
  margin-bottom: 24px;
  border-radius: var(--border-radius-xs);
  background: var(--white);
`;

export const RadioTabWrapper = styled.div`
    padding: var(--pd-16);
    font-family: ${TypographyENUM.lkSansRegular};
    display: flex;
    align-items: center;
    gap: 12px;
    position:relative;
    &:before{
      content:"";
      position:absolute;
      left:10px;
      right:10px;
      border:1px dashed #E2E2EE;
      bottom:0px;
    }
    &:last-child:before{
      display:none;
    }
    `;

export const NameContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 4px;
`;

export const Name = styled.div`
  font-weight: 400;
  font-size: var(--fs-14);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--text);
`;

export const OfferText = styled.div`
  font-family: ${TypographyENUM.lkSansMedium};
  font-size: var(--fs-12);
  color: #006400;
`;

export const ModalHeader = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-size: var(--fs-16);
  line-height: 24px;
  letter-spacing: -0.02em;
  padding-bottom: var(--pd-8);
`;

export const ModalContent = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-size: var(--fs-14);
  line-height: 20px;
  letter-spacing: -0.02em;
  padding-bottom: var(--pd-24);
  padding-top: var(--pd-20);
  font-weight: 500;
`;

export const ModalText = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-size: var(--fs-12);
  line-height: 20px;
  letter-spacing: -0.02em;
  padding-bottom: var(--pd-20);
`;
