// @lk/ui-library/src/Component/V1/CheckoutMobile/TopSection/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const TopSectionWrapper = styled.div<{disableTopPadding?: boolean, isRTL?:boolean}>`
  padding: var(--pd-32) var(--pd-16) var(--pd-12);
  background: var(--warm-grey-25);
  border-radius: 0px;
  display: flex;
  align-items: center;
  justify-content: var(--space-between);
  width: 100%;
  &.paymentHead{
	position:sticky;
	top:51px;
	z-index:1;
	padding-bottom:40px;
	#need-help-cta{
		padding-left: ${props => props.isRTL ? "0" : ""};
	}
  }
  @media screen and (max-width:767px){
	padding: var(--pd-20) var(--pd-16) var(--pd-12);
  }

  ${props => props.disableTopPadding ? "padding-top: 0" : ""};
`;

export const Title = styled.div<{ font?: TypographyENUM, fontSmall: boolean, isAddressPage?:boolean; }>`
  font-family: ${TypographyENUM.lkSerifNormal};
  font-weight: var(--fw-regular);
  font-size: var(--fs-20);
  line-height: 32px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  ${props => props.isAddressPage ? "font-size : 14px;font-family: var(--font-lksans-bold); " : "" };
`;
