// @lk/ui-library/src/Component/V1/CheckoutMobile/TopSection/TopSection.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { TopSection } from "./TopSection.component";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";
import NeedHelpCta from "../../../Common/NeedHelpCta";

export default {
	title: "Components/v1/CheckoutMobile/TopSection",
	component: TopSection,
	argTypes: {
		heading: generateArgTypes({
			type: "string",
			required: true,
			description: "To set the heading in the top section on checkout pages",
			category: "Component",
			defaultValue: "Title"
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "To set the font family of the heading",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "radio",
			options: TypographyENUM,
		}),
		showChildren: generateArgTypes({
			type: "boolean",
			required: false,
			description: "To handle whether to show the children on the right side of the component or not.",
			category: "Component",
			defaultValue: false,
		})
	}
} as ComponentMeta<typeof TopSection>;

const Template: ComponentStory<typeof TopSection> = (args) => <TopSection {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	heading: "Select Address",
	font: TypographyENUM.lkSerifNormal,
	showChildren: true,
	children: <NeedHelpCta
		localeData={{ NEED_HELP: "Need help", WHATSAPP_CHAT_URL: "https://api.whatsapp.com/send/?phone=918929853854&text=" }}
		isRendered={true}
		isBuyOnChat={true}
		whatsappChatMsg={"Hi Lenskart, help me get started! I am on <pageName> page of mobile site. I need help for product ID: <pid-no>".replace("<pageName>", "Payment").replace("<pid-no>", "147098")}
		phoneNumber={"18001020767"}
		onClick={() => {
			console.log("Clicked!");
		}}
	/>
};
