// @lk/ui-library/src/Component/Common/SecureCardModal/SecureCardModalStyles.tsx
import styled from "styled-components";

export const OverLay = styled.div`
  height: 100vh;
  overflow-y: hidden;
  position: fixed;
  top: 0px;
  left: 0px;
  right: 0px;
  bottom: 0px;
  background-color: rgba(0, 0, 0, 0.8);
  z-index: 999;

  .modalWrapper {
    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
    width: 600px;
    height: auto;
    position: fixed;
    background: white;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    padding: 10px;
    border-radius: 10px;
  }
`;
export const Heading = styled.div`
  font-size: 24px;
  font-weight: 500;
  color: rgb(51, 51, 51);
  line-height: 32px;
  font-style: normal;
  text-align: center;
  margin: 8px 0px;
`;
export const SubText = styled.div`
  font-weight: 500;
  font-size: 16px;
  line-height: 24px;
  font-style: normal;
  max-width: 400px;
  margin: 8px 0px;
`;
export const ListPoints = styled.li`
  max-width: 400px;
  font-size: 16px;
  font-weight: 500;
  line-height: 24px;
  list-style-type: disc !important;
`;
export const CloseBtn = styled.button`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
    line-height: 24px;
    letter-spacing: 0.4px;
    text-transform: uppercase;
    color: rgb(0, 185, 198);
    margin: 30px 0px;
    padding: 6px 48px;
    border-width: 1px;
    border-style: solid;
    border-color: rgba(0, 185, 198, 0.5);
    border-image: initial;
    border-radius: 4px;
    background: none;
	cursor:pointer;
`;
export const CardImg = styled.div`
  svg {
    width: 200px;
    height: 142.857px;
  }
`;
