// @lk/ui-library/src/Component/Common/SecureCardModal/SecureCardModal.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../StorybookHelper";
import SecureCardModal from "./index";

export default {
	title: "Components/common/SecureCardModal",
	component: SecureCardModal,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		onClose: generateArgTypes({
			type: "function",
			required: true,
			description: "Function to close the modal",
			category: "Component",
			defaultValue: () => null,
		}),
		isVisible: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to show hide the modal",
			category: "Component",
			defaultValue: true,
		}),
	}
} as ComponentMeta<typeof SecureCardModal>;

const Template: ComponentStory<typeof SecureCardModal> = (args) => <SecureCardModal {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "SecureCardModal",
	isVisible: true,
	closeModal: () => console.log("Closed"),
	localeData:{
		SECURE_YOUR_CARD: "Secure your card with RBI guidelines",
		RBI_GUIDELINES:"As per RBI guidelines we need to add an additional layer to secure your card. Agreeing to this means:",
		RBI_GUIDELINE_1:"Securing this card and avoiding unauthorised use.",
		RBI_GUIDELINE_2:"Avoiding entering card details everytime you transact on Lenskart using this card",
		OKAY_GOT_IT:"OKAY GOT IT"
	}
};
