// @lk/ui-library/src/Component/V1/Cart/PriceBreakup/PriceBreakup.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import PriceBreakup from "./index";
import { TypographyENUM } from "../../../../Types/general";
// import { cartAmount } from "./mockValue";
import { CartAmountType } from "./PriceBreakup.types";

export default {
	title: "Components/v1/Cart/PriceBreakup",
	component: PriceBreakup,
	argTypes: {
		...generateGeneralArgs(),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		})
	},
} as ComponentMeta<typeof PriceBreakup>;

const Template: ComponentStory<typeof PriceBreakup> = (args) => <PriceBreakup {...args} />;
// export enum CartAmountType {
// 	POSITIVE = "positive",
// 	NEGATIVE = "negative",
// 	FREE = "free",
// 	TOTAL = "total",
//   }
export const Overview = Template.bind({});
Overview.args = {
	id: "price-breakup",
	width: "100",
	priceData: [
		{ label: "Total Gold VIP price", amount: 999, type: CartAmountType.POSITIVE },
		{ label: "Tax", amount: 119.88, type: CartAmountType.POSITIVE },
		{ label: "Convenience Fees", amount: 0, type: CartAmountType.FREE },
		{ amount: 1118.88, label: "Total payable", type: CartAmountType.TOTAL }
	],
	dataLocale: {
		ITEM_TOTAL: "Item Total",
		LK_CASH: "LK Cash",
		TOTAL_BEFORE_TAX: "Total before tax",
		TAXES: "Taxes",
		CONVENICNCE_FEE: "Convenience Fee",
		TOTAL_PAYABLE: "Total Payable"
	},
	font: TypographyENUM.defaultBook,
	// cartAmount: cartAmount
};
