// @lk/ui-library/src/Component/V1/Cart/PriceBreakup/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const PriceBreakupWrapper = styled.div <{ width: any } >`
	box-sizing: border-box;
	padding: var(--pd-16);
	border: 1px solid var(--light-purple-10);
	background-color: var(--white);
	border-radius: var(--border-radius-xs);
	display: flex;
	flex-flow: column;
	gap: 6px;
`;

export const FlexWrapper = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  justify-content: var(--space-between);
`;
export const InfoIcon = styled.img`
	width: var(--grid-space-desktop);
`;

export const BreakUp = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: var(--space-between);
`;

export const HR = styled.hr`
	width: 100%;
	margin: 0;
	border-bottom: 1px dashed var(--light-purple-10);
	border-top: none;
	border-right: none;
	border-left: none;
	overflow: hidden;
	height: 5px;
`;


export const Text = styled.div<{isDiscount?: boolean, isBold?: boolean, isTotal?: boolean, styleFont: TypographyENUM}>`
	font-family:  ${(props) => (props.isBold ? "var(--font-lksans-bold)" : "var(--font-lksans-regular)")};
	font-size: ${(props) => (props.isTotal ? "var(--fs-16)" : "var(--fs-14)")};
	color:  ${(props) => (props.isDiscount ? "var(--vivid-green-100)" : "var(--dark-blue-100)")};
`;
