// @lk/ui-library/src/Component/V1/MyAccount/Mobile/CustomerReview/productReview.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const ReviewInfoWrapper = styled.div<{styleFont?:TypographyENUM}>`
	flex-grow:1;
	display:flex;
	flex-direction:column;
	gap:var(--pd-10);
	font-family:${props => props.styleFont};
	background:#fff;
	padding:10px;
	& #ratings{
		margin-bottom:20px;
		svg{
			width:20px;
			height:20px;
		}
	}
`;

export const InputFieldWrapper = styled.div`
	margin-top:10px;
`;

export const FieldWrapper = styled.div`
	margin-bottom: 8px;
	margin-top: 30px;
`;

export const AddressInputField = styled.div`
    margin-bottom: 30px;
       input,textarea{
		border-bottom: 1px solid #99a0a9;
		font-size:var(--fs-16);
		box-shadow:none;
		color:#424242;
		padding:4px 0;
		&:focus{
			outline: none;
            border-bottom: 1px solid #18cfa8;
		}
		user-select: initial;
		&::before, &::after {
    		user-select: initial;
		}
	}
`;

export const ReviewWrapper = styled.div`
    border-top: 10px solid #FBF9F7;
    margin: 5px -10px -10px;
    padding: 8px 10px 0;
`;
export const ReviewProductInfo = styled.div`
 display:flex;
 justify-content:space-between;
 padding: 8px;
 border:1px solid #eaeff4;
 border-radius:4px;
 align-items:center;
`;
export const ReviewProductImage = styled.div`
   max-width:120px;
   img{
	 max-width:100%;
   }
`;

export const ReviewProductDes = styled.div`
 padding-left:10px;
`;

export const ReviewBtnWrapper = styled.div`
 position:fixed;
 left:0px;
 bottom:0;
 width: 100%;
 padding: 8px;
 font-weight: var(--fw-bold);
`;

export const ProductTexTitle = styled.strong`
    font-size:var(--fs-14);
	font-weight: var(--fw-medium);
	margin-bottom:6px;
	display:block;
	color: #333;
	font: 12px/1.2857 "Roboto",sans-serif,"Arial","Helvetica",sans-serif;
`;

export const ProductTextDes = styled.p`
	font-size:var(--fs-12);
	color: #666;
	font: 12px/1.2857 "Roboto",sans-serif,"Arial","Helvetica",sans-serif;
	font-weight: var(--fw-light);
`;

export const ShareFeedbackWrapper = styled.div`
	text-transform:uppercase;
	border-bottom: 1px solid #eaeff4;
	font-size:var(--fs-14);
	font-weight: var(--fw-medium);
	text-align:center;
	padding:var(--pd-10) var(--pd-10) var(--pd-15) var(--pd-10);
	font: 12px/1.2857 "Roboto",sans-serif,"Arial","Helvetica",sans-serif;
`;

export const ReviewSubTitle = styled.h3`
   color: #18cfa8;
   margin:30px 0 15px;
   font-size:var(--fs-13);
	font-weight: var(--fw-medium);
	&.margin-top10{
		margin-top:10px;
	}
`;

export const ValidateError = styled.div`
	font-size: 10px;
	margin-top: 4px;
	color: #E34934;
`;
