// @lk/ui-library/src/Component/V1/MyAccount/Mobile/CustomerReview/productReview.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import ProductReview from "./productReview.component";

export default {
	title:"Components/V1/MyAccount/Mobile/ProductReview",
	component: ProductReview,
	argTypes:{
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "radio",
			options: TypographyENUM,
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			defaultValue: {
				WRITE_A_REVIEW: "Write A Review",
				ENTER_EMAIL_ID: "Enter Email Id",
				YOUR_NAME: "Your Name",
				TITLE: "Title",
				REVIEW: "Review",

			},
			control:"object"
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left ",
			category: "Status",
			defaultValue: "false",
		})
	}
} as ComponentMeta<typeof ProductReview>;

const Template: ComponentStory<typeof ProductReview> = args => <ProductReview {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"product-review-info",
	font:TypographyENUM.defaultBook,
	dataLocale: {
		TITLE: "Title",
		REVIEW: "Review",
		EMAIL:"Email",
		YOUR_NAME: "Your Name",
		RATING: "Rating",
		PERSONAL_DATA:"Personal Data",
		POST:"POST",
		SHARE_YOUR_FEEDBACK:"Share Your Feedback",
	},
	productImage:"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//l/i/sky-blue-gunmetal-full-rim-rectangle-lenskart-air-signia-la-e14573-c2-eyeglasses_g_4977_6_13_22.jpg",
	productTitle:"Lenskart Air",
	productDec:"Sky Blue Full Rim Rectangle Eyeglasses",
	fullName: "test Test",
	email: "test@gmail.com",
	submitFunction: (reviewData, emailValue) => console.log(reviewData, emailValue),
};
