// @lk/ui-library/src/Component/V1/Checkout/AddAddressFormSa/components/AddressInputField/styles.tsx
import styled from "styled-components";
// import { TypographyENUM } from "../../../../Types/general";
import { TypographyENUM } from "../../../../../../Types/general";

export const FieldContainer = styled.div<{fullWidth?: boolean, line?: number}>`
	display: flex;
	flex-direction: row;
	align-items: flex-start;
	// padding: 6px 0px;
	padding: 16px;
	gap: 8px;
	width: ${props => props.fullWidth ? "100%" : "431px"};
	height: ${props => props.line ? `${props.line * 56}px` : "56px"};
	background: #FFFFFF;
	border: 1px solid #E2E2EE;
	box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
	border-radius: 12px;
	flex: none;
	align-self: stretch;
	flex-grow: 0;
	position: relative;

	&:invalid[data-focused="true"]{
		border: 1px solid #ec3331;
	}

	&:invalid[data-focused="true"] ~ span{
		display: block;
	}
`;

export const FieldInput = styled.input`
	width: 100%;
	height: 100%;
	border: none;
	outline: none;
	background: transparent;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;
export const LeftPositioned = styled.div`
	// position: absolute;
	left: 0;
	// width: 48px;
	width: fit-content;
	// top: 25%;
`;

export const Box = styled.div<{font: TypographyENUM}>`
    font-family: ${props => props.font};
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	// width: fit-content;
	width: 100%;
`;

export const Label = styled.label`
	// font-family: 'Lenskart Sans';
	font-style: normal;
	font-weight: 400;
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: #333368;
	margin-bottom: 8px;
`;

export const SubText = styled.div<{isError?: boolean}>`
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 20px;
	letter-spacing: -0.02em;
	color: ${props => props.isError ? "#ec3331" : "#66668E"};
	margin-top: 8px;
`;
export const SubTextSpan = styled.span`
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 20px;
	letter-spacing: -0.02em;
	color: #ec3331;
	margin-top: 8px;

	// font-size: 12px;
    // padding: 3px;
    // color: red;
    display: none;
`;

export const CrossSection = styled.div`
	position: absolute;
	right: 10px;
	cursor: pointer;
	height: 12px;
	width: 12px;
	display: flex;
    align-items: center;
    justify-content: center;
    top: 40%;
`;
