// @lk/ui-library/src/Component/V1/PDP/Mobile/ReviewProgressBar/ReviewMobileProgressBar.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { ComponentSizeENUM, TypographyENUM } from "../../../../../Types/general";
import ReviewMobileProgressBar from "./ReviewProgressBar.component";

export default {
	title: "Components/V1/PDP/mobile/ReviewMobileProgressBarTypes",
	component: ReviewMobileProgressBar,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		componentSize: generateArgTypes({
			type: "string",
			required: false,
			description: "Size of the component",
			category: "General",
			defaultValue: ComponentSizeENUM.small,
			control: "radio",
			options: ComponentSizeENUM,
		}),
		isReadOnly: generateArgTypes({
			type: "boolean",
			required: true,
			description: "read only or not",
			category: "Component",
			defaultValue: "false",
		}),
		onmouseClick: generateArgTypes({
			type: "function",
			required: false,
			description: "on Mouse click handler for rating",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show rating right to left",
			category: "",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		})
	}
} as ComponentMeta<typeof ReviewMobileProgressBar>;

const Template: ComponentStory<typeof ReviewMobileProgressBar> = args => <ReviewMobileProgressBar {...args}/>;
export const Overview = Template.bind({});
Overview.args = {
	id:"star-rating",
	rating:4,
	maxRating:5,
	percentage:"50",
	stars:"5",
	componentSize: ComponentSizeENUM.medium,
	isReadOnly: true,
	numOfRatings: 101
};
