// @lk/ui-library/src/Component/V1/PDP/Mobile/ReviewProgressBar/ReviewMobileProgressBar.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const ReviewMobileProgressBar = styled.div<{font: TypographyENUM}>`
	display: flex;
	flex-direction: row;
	gap: 10px;
	font-family: ${props => props.font};
	align-items:center;
`;

export const ProgressBarRatings = styled.div`
  display:flex;
  gap:5px;
  align-items:center;
`;
export const StarCount = styled.div`
 width:10px;
`;

export const ProgressBarPercentage = styled.div`
	width: 70px;
	display: flex;
	justify-content: center;
`;
export const ReviewProcessBar = styled.div`
	float: left;
	width: 0%;
	height: 100%;
	font-size: 12px;
	line-height: 20px;
	color: #fff;
	text-align: center;
	background-color: #28c5be;
	-webkit-box-shadow: inset 0 -1px 0 rgba(0,0,0,.15);
	box-shadow: inset 0 -1px 0 rgba(0,0,0,.15);
	-webkit-transition: width .6s ease;
	-o-transition: width .6s ease;
	transition: width .6s ease;
`;

export const ReviewProcessBarWrapper = styled.div`
    height: 15px;
	overflow: hidden;
	width:100%;
	background-color: #f5f5f5;
	border-radius: 8px;
	-webkit-box-shadow: inset 0 1px 2px rgba(0,0,0,.1);
	box-shadow: inset 0 1px 2px rgba(0,0,0,.1)
`;
