// @lk/ui-library/src/Component/V1/Checkout/AddAddressForm/AddAddressForm.styles.tsx
import styled from "styled-components";
import { ThemeENUM, TypographyENUM } from "../../../../Types/general";

export const ReviewFormWrapper = styled.div<{styleFont:TypographyENUM}>`
	font-family:${props => props.styleFont};
	padding: var(--pd-20);
	// padding: 10px;
	// border: 1px solid rgba(60, 60, 60, .54);
    // border-radius: 8px;
`;

export const Icon = styled.div`
font-size: 22px;
`;

export const InvisibleButton = styled.button<{btnVisible: boolean, theme?: ThemeENUM}>`
	display: ${props => props.btnVisible ? "inline-block" : "none" };
	background-color: ${props => props.theme !== ThemeENUM.secondary ? "#11DAAC" : "var(--white)"};
    color: #000042;
	font-family: ${TypographyENUM.lkSansBold};
	/* font-weight: var(--fw-bold); */
	width: 50%;
	font-size: var(--fs-16);
	line-height: var(--fs-24);
	letter-spacing: -0.02em;
	border: 0;
	border: ${props => props.theme === ThemeENUM.primary ? "0" : "1px solid #11DAAC"};
	border-radius: 100px;
	padding: 16px 24px;
	cursor: pointer;
	filter: drop-shadow(0px 1px 2px rgba(16, 24, 40, 0.05));
	&:hover {
			box-shadow: 0 3px 16px 0 rgba(0, 0, 0, 0.1);
		}

    /* margin-bottom: 0;
    text-align: center;
    white-space: nowrap;
    vertical-align: middle;
    touch-action: manipulation;
    cursor: pointer;
    background-image: none;
    border: 1px solid transparent;
    padding: 6px 12px;
    font-size: 14px;
    line-height: 1.42857143;
    border-radius: 4px;
    border-color: #ccc;
	margin-bottom: 10px; */
`;

export const ButtonsContainer = styled.div<{btnVisible?: boolean}>`
	display: ${props => props.btnVisible ? "flex" : "none"};
	flex-direction: row;
	align-items: center;
	justify-content: center;
	gap: 20px;
`;

export const FormButtonWrapper = styled.div`
	position: relative;
    justify-content: left;
    left: -10px;
    top: -10px;
	button{
		border-radius: 8px 0 8px 0;
	}
`;

export const Form=styled.form`
	display:flex;
	flex-direction:column;
	gap:var(--spacing-lg);
`;

export const InputWrapper = styled.div`
	display:flex;
	gap:var(--spacing-md);
	/* flex-wrap: wrap; */
`;

export const InputField = styled.input<{width:string}>`
	width: ${props => props.width};
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const TextAreaField = styled.textarea<{width:string}>`
	width: ${props => props.width};
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const FormActionWrapper = styled.div`
	display:flex;
	align-items:center;
	justify-content:right;
`;

export const FormInputError = styled.small`
	color:var(--error);
`;

export const LandmarkTextarea = styled.textarea<{font: TypographyENUM}>`
	width: 100%;
	height: 100%;
	border: none;
	outline: none;
	background: transparent;
	font-family:${props => props.font};
	resize: none;
	margin-top: 8px;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const AddressLabel = styled.label`
	font-style: normal;
	font-weight: 400;
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-100);
	margin-bottom: 16px;
`;
