// @lk/ui-library/src/Component/V1/Checkout/AddAddressForm/short-config.ts
export const shortAddressForm = [
	{
		rowType: "normal", // radio, normal, dropdown, swatch, ~ addressInfo
		data: [
			{
				placeholder: "First Name*",
				label: "",
				apiKey: "firstName",
				errorMessage: "First Name should be 3-16 characters and shouldn't include any special character!",
				required: true,
				validation: "name",
				disabled: false,
				visible: true,
				// fieldType: "firstName", // "radio", "select"
				fieldType: "normal",
				subText: "",
				defaultValue: "",
			},
			{
				placeholder: "Last Name*",
				label: "",
				apiKey: "lastName",
				validation: "name",
				errorMessage: "Last Name should be 3-16 characters and shouldn't include any special character!",
				required: true,
				disabled: false,
				visible: true,
				// fieldType: "lastName", // "radio", "select"
				fieldType: "normal",
				subText: "",
				defaultValue: "",
			}
		]
	},
	{
		rowType: "null",
		data: [
			{
				label: "Select Gender",
				apiKey: "gender",
				fieldType: "gender", // "radio", "select",
				validation: "required",
				options: [{ key: "Male", value: "male" }, { key: "Female", value: "female" }], // "display", "value"
				subText: "",
				required: true,
				disabled: false,
				visible: true,
				errorMessage: "Please select your gender",
				defaultValue: "male",
			}
		]
	},
	{
		rowType: "normal",
		data: [
			{
				required: true,
				disabled: false,
				visible: true,
				validation: "phone",
				label: "",
				placeholder: "Phone Number*",
				apiKey: "phone",
				errorMessage: "Please enter a valid phone number!",
				fieldType: "phone", // "radio", "select"
				subText: "",
				defaultValue: "",
				phoneCode: {
					// options: [{ key: "+91", value: "+91", maxLength: 10 }, { key: "+1", value: "+1", maxLength: 8 }],
					defaultValue: "+91",
					apiKey: "phoneCode",
					required: true,
					disabled: false,
					visible: true,
					validation: "required",
					type: "dropdown"
				}
			},
			{
				required: true,
				disabled: false,
				visible: true,
				validation: "email",
				placeholder: "Email Address*",
				errorMessage: "It should be a valid email address!",
				subText: "",
				label: "",
				apiKey: "email",
				fieldType: "email",
				defaultValue: "",
			}
		]
	},
	{
		rowType: "normal",
		data: [
			{
				placeholder: "Address*",
				errorMessage: "Address is required",
				required: true,
				disabled: false,
				visible: true,
				validation: "required",
				label: "",
				apiKey: "addressline1",
				// fieldType: "address",
				fieldType: "normal",
				defaultValue: "",
				subText: "House / Flat Number"
			}
		]
	},
	{
		rowType: "normal",
		data: [
			{
				maxLength: 6,
				required: true,
				disabled: false,
				visible: true,
				validation: "postcode",
				placeholder: "Zip/Postal Code*",
				errorMessage: "Please enter a correct postal code",
				label: "",
				apiKey: "postcode",
				// fieldType: "postcode",
				fieldType: "normal",
				options: [],
				defaultValue: "",
				subText: "Please enter a postal code (10710)"
			},
			{
				disabled: false,
				required: true,
				visible: true,
				validation: "required",
				placeholder: "City*",
				errorMessage: "Please enter a city",
				label: "",
				apiKey: "city",
				// fieldType: "city",
				fieldType: "normal",
				options: [],
				defaultValue: "",
				subText: ""
			},
		]
	},
	{
		rowType: "normal",
		data: [
			{
				placeholder: "Select State*",
				errorMessage: "Please select a state",
				required: true,
				disabled: false,
				visible: true,
				validation: "required",
				label: "",
				apiKey: "state",
				fieldType: "select-state",
				options: [],
				defaultValue: "",
				subText: ""
			},
			{
				required: true,
				disabled: false,
				visible: true,
				validation: "required",
				placeholder: "Select Country*",
				errorMessage: "Please select a country",
				label: "",
				apiKey: "country",
				fieldType: "select-country",
				options: [],
				defaultValue: "",
				subText: ""
			},
		]
	},
	{
		rowType: "normal",
		data: [
			{
				// placeholder: "Select State*",
				errorMessage: "",
				required: false,
				disabled: false,
				visible: true,
				validation: "required",
				label: "",
				apiKey: "addressLabel",
				fieldType: "address-swatch",
				options: [],
				defaultValue: "",
				subText: ""
			}
		]
	}
];
export const phoneCodeMapping = [{ code: "+91", value: "+91", maxLength: 10 }, { code: "+1", value: "+1", maxLength: 8 }];
// const addressConfig = [
// 	{
// 		props: {
// 			id: 1,
// 			name: "username",
// 			type: "text",
// 			placeholder: "Username",
// 			errorMessage:
// 				"Username should be 3-16 characters and shouldn't include any special character!",
// 			label: "Username",
// 			pattern: "^[A-Za-z0-9]{3,16}$",
// 			required: true
// 		},
// 		fieldType: "normal", // "radio", "select"
// 		options: [{ display: "", value: "" }], // "display", "value"
// 		subText: "",
// 		// error: "",
// 		// validationFn: () => null
// 	}
// ];
