// @lk/ui-library/src/Component/V1/Checkout/AddAddressForm/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const SelectGender = styled.h3<{marginBottom?: boolean}>`
	font-family: ${TypographyENUM.lkSansRegular};
	font-style: normal;
	font-weight: 400;
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: #000042;

	${props => props.marginBottom ? "margin-bottom: 24px" : ""}
`;

export const GenderText = styled.label`
	font-family: ${TypographyENUM.lkSansRegular};
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 150%;
	letter-spacing: 0.15px;
	color: var(--dark-blue-100);
`;

export const ErrorField = styled.div`
    color: #d90502;
	margin-top: 8px;
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 20px;
	letter-spacing: -0.02em;
`;

export const EmailIconWrapper = styled.span`
	display: flex;
	svg{height:100%};
`;

export const GenderWrapper = styled.div<{noGap?: boolean}>`
	display: flex;
 	flex-direction: column;
  	${props => props.noGap ? "" : "gap: 24.3px;"};
`;

export const EmptyComponent = styled.div`
	width:100%;
	visibility:hidden;
`;

export const PowerCheckout = styled.label`
	background: #F7F1DE;
	border: 1px solid #FFE092;
	border-radius: 4px;
    padding: 16px;
    margin-top: 10px;
	font-size: 14px;
	line-height: 20px;
	letter-spacing: -0.02em;
	color: #000042;
	display:flex;
	align-items: center;
	font-family: ${TypographyENUM.lkSansBold};
	svg{
		margin-right:12px;
		path{
		 stroke: #000042;
		}
	}
`;
