// @lk/ui-library/src/Component/V1/MyAccount/SavedCards/SavedCards.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import SvaedCard from "./index";

export default {
	title: "Components/V1/MyAccount/SvaedCards",
	component: SvaedCard,
	argTypes: {
		...generateGeneralArgs(),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left onKeepMEPosted",
			category: "Staus",
		}),
	}
} as ComponentMeta<typeof SvaedCard>;

const Template: ComponentStory<typeof SvaedCard> = (args) => <SvaedCard {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		CARD_EXPIRES_IN: "Card expires in",
		YOUR_SAVED_CARDS: "Your Saved Cards",
		YOU_DONT_HAVE_SAVED_CARDS: "You dont have any saved cards",
	},
	savedCardList: [{
		cardBrand: "MAST",
		cardMode: "CC",
		cardToken: "2c4eb1c6ecef679acb3a9",
		cardType: "CC",
		expired: false,
		expiryMonth: "10",
		expiryYear: "2025",
		nameOnCard: "DUMMY",
		number: "XXXXXXXXXXXX0008",
		storeCard: true
	},
	{
		cardBrand: "MAST",
		cardMode: "CC",
		cardToken: "2c4eb1c6ecef679acb3a9",
		cardType: "CC",
		expired: false,
		expiryMonth: "10",
		expiryYear: "2025",
		nameOnCard: "DUMMY",
		number: "XXXXXXXXXXXX0008",
		storeCard: true
	}
	],
	onDeleteCard: () => console.log("on Delete saved card"),
};
