// @lk/ui-library/src/Component/V1/MyAccount/SavedCards/styles.tsx
import styled from "styled-components";

export const SavedCardWrapper = styled.div``;

export const Heading = styled.div`
    font-family: "FuturaStd-Book";
	font-synthesis: none;
    padding-left: 15px;
    font-weight: 400;
    font-size: 30px;
    margin: 20px 0px 10px 0px;
`;

export const CardWrapper = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
    margin-top: 10px;
    width: 40%;
    border: 1px solid #ccc;
    border-radius: 5px;
    padding: 15px;
    height: 80px;
`;

export const CardImageConatiner = styled.div`
    padding-left: 10px;
`;

export const Image = styled.img``;

export const CardImage = styled.div`
    background: url(https://static.lenskart.com/media/desktop/img/savedcardimage.png) no-repeat scroll -175px -4px transparent;
    height: 29px;
    width: 43px;
    top: 20%;
`;

export const CardDetail = styled.div`
    display: flex;
    flex-direction: column;
    width: 100%;
    margin-left: 10px;
`;

export const CardNameDeleteIconWrapper = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding-bottom: 5px;
`;

export const CardName = styled.div`
    font-weight: 500;
    color: #333333;
    font-family: 'FuturaStd-Book';
	font-synthesis: none;
    font-size: 13px;
`;

export const DeleteIcon = styled.div``;

export const CardNumberExpiry = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    color: #999999;
    font-family: 'FuturaStd-Book';
	font-synthesis: none;
    font-size: 12px;
`;

export const CardNumber = styled.div``;

export const CardExpiry = styled.div``;

export const AlertMessage = styled.div<{isError?: boolean;}>`
    color: ${props => props.isError ? "#a94442" : "#8a6d3b"};
    background-color: ${props => props.isError ? "#f2dede" : "#fcf8e3"};
    border-color: ${props => props.isError ? "#ebccd1" : "#faebcc"};
    ${props => props.isError ? `
        font-size: 13px;
    ` : ""}
    padding: 15px;
    margin-bottom: 20px;
    border: 1px solid transparent;
    border-radius: 4px;
    width: 100%;
`;
