// @lk/ui-library/src/Component/V1/CheckoutMobile/Cards/Cards.types.tsx
import { DataType } from "../../../../Types/general";

export interface SavedCard{
  logoImageUrl?:string;
  number: number;
  cardBrand: string;
  cardToken: string;
}
export interface CardsTypes {
  savedCards: SavedCard[];
  dataLocale: ConfigDataType;
  methodData: any;
  savedCardsHandler: (e: any,ev: any)=>void;
  getSubmitFunction:(e: any)=>(...args: any)=>void;
  borderVisible?: boolean
  setSelectedPaymentMethod: (methodKey: string) => void;
  selectedPaymentMethod: string;
  renderPrimerUI: (methodToShow: string) => void;
  primerToken: string;
  isContactLensConsentEnabled: boolean;
  payLaterAllowed: boolean;
  isPrimerActive: boolean;
  isContactLensCheckboxChecked: boolean;
	setIsContactLensCheckboxChecked: (arg0: boolean)=> void;
  currencyCode: string;
	totalAmount: string;
  setSectedPrimerCreditCard: (arg0: boolean)=> void;
  showPrimerCardBottomSheet: boolean;
  configData: DataType
  abandonedLeadsFunction: () => void
}

export interface ConfigDataType {
	[name: string]: string
}
