// @lk/ui-library/src/Component/V1/CheckoutMobile/Cards/Cards.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const CardsSection = styled.div<{borderVisible?: boolean}>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: flex-start;
  padding: ${props => props.borderVisible ? "0 16px 12px 16px" : "" };
  background: ${props => props.borderVisible ? "var(--white)" : "" };
  box-shadow: ${props => props.borderVisible ? "0px 0px 12px rgb(0 0 66 / 6%)" : "" };
  border-radius: ${props => props.borderVisible ? "var(--fs-12)" : "" };
`;

export const CardSheetSection = styled.div`
  padding: var(--fs-16);
`;
export const PayLaterButtonWrapper = styled.div`
	display: flex;
	justify-content: right;
  flex-direction: column;
  margin-top: 20px;
  gap: 10px
`;

export const CardSheetHeader = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-16);
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  margin-bottom:20px;
`;

export const CardSmallInput = styled.div`
  span{
    display: flex;
    justify-content:space-between;
    align-items:center;
    gap:20px;
  }
`;

export const GuideLinesWrapper = styled.div`
  display: flex;
  gap: 2px;
  position:relative;
`;

export const Tick = styled.span``;
export const Box = styled.span``;

export const CheckBox = styled.span`
position:relative;
input {
  display: none;
}
label {
${Tick} {
  width: 16px;
  height: 16px;
  border-radius: 4px;
  content: '';
  position: absolute;
  transition: all 0.3s ease;
  transition-property: transform, border-color, -webkit-transform;
  top: -1px;
  left: -1px;
}

${Box} {
  width: 16px;
  height: 16px;
  border-radius: 4px;
  border: 1px solid #000042;
  position: absolute;
  top: 2px;
}
}

input[type='checkbox']:checked + label ${Tick} {
  width: 4px;
  height: 9px;
  border: solid #fff;
  border-width: 0 1px 1px 0;
  -webkit-transform: rotate(45deg);
  transform: rotate(45deg);
  margin: 5px 7px;
  z-index: 1;
  border-radius: 0px;
  position: absolute;
  top: -1px;
}

input[type='checkbox']:checked + label ${Box} {
  background: #000042;
  position: absolute;
  transform: scale(1);
  content: '';
}
`;


export const SavedSectionOuter = styled.div`
position: relative;
width:100%;
`;

export const SavedSectionOpen = styled.div<{valCheck:boolean}>`
  display: ${props => props.valCheck ? "block" : "none"};
  transition: all 0.1s cubic-bezier(0, 1, 0, 1);
  padding: 20px 0 10px 0;
  position:relative;
`;
export const InputError = styled.div<{hasError: boolean}>`
  display: ${props => props.hasError ? "block" : "none"};
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #E34934;
  margin-top:2px;
  padding-left: 10px;
  position:absolute;
  top: 58px;
`;

export const InputErrorForCVV = styled.div<{hasError: boolean}>`
  display: ${props => props.hasError ? "block" : "none"};
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #E34934;
  margin-top:2px;
  padding-left: 10px;
  position:absolute;
  top: 76px;
`;

export const InputWrapper = styled.div`
  display:flex;
  flex-direction:column;
  width:100%;
  padding-bottom:25px;
  position:relative;
`;

export const ATMCard = styled.div<{ borderVisible?: boolean; isBottomsheetOpen?: boolean; }>`
width:100%;
padding-top: ${props => props.borderVisible ? "var(--fs-16)" : "" };
  li{
    i{
    img{
      width: 46px;
      height: 32px;
    }
  }
    display:flex;
    align-items:center;
    justify-content:space-between;

    i:nth-child(2){
      margin-right:auto;
      margin-left:15px;
    //   margin-top: -6px;
    }
  }

  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 14px;
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--text);

  li {
    span {
      img {
        transition: 0.3s all ease;
        transform: rotate(${({ isBottomsheetOpen }) => isBottomsheetOpen ? "90deg" : "0deg"});
      }
    }
  }
`;

export const GuideLinesHeader = styled.div`
  display:flex;
  align-items:center;
  gap:10px;
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-16);
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  margin-bottom:20px;
  margin-right:30px;
`;

export const VerifyPayButton = styled.div<{isDisabled?:boolean}>`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  // padding: 8px 16px;
  height: 40px;
  background: var(--vivid-green-100);
  box-shadow: 0px 1px 2px rgba(0, 0, 66, 0.1);
  border-radius: 100px;
  width:40%;
  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  font-weight: var(--fw-bold);
  font-size: 14px;
  line-height: var(--fs-24);
  display: flex;
  align-items: center;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  @media screen and (max-width:767px){
	// font-size: 12px;
  }
  ${props => props.isDisabled ? "pointerEvents:none; opacity:0.3" : ""};
`;

export const GuideLines = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  margin-bottom:20px;
  // text-align:center;
  margin-left:20px;
`;

export const GuildeLinesContent = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--text);
  position:relative;

  ul{
    margin-bottom:15px;
    li{
      margin-left:20px;
      &:first-child{
        margin-top:20px;
      }
    }
  }
`;

export const GuideLinesSection=styled.div`
  padding:20px 16px;
`;

export const LearnMore = styled.span`
  text-decoration: underline;
  font-weight: var(--fw-bold);
  font-style: normal;
  font-size: 12px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  padding-left: 3px;
  padding-top: 2px;
`;

export const SavedSectionInner = styled.div`
  display:flex;
  gap:10px;
  align-items:center;
`;

export const SavedCards = styled.div`
  width:100%;
  li{
	i:first-child{
	background: rgb(255, 255, 255);
	border: 1px solid rgb(226, 226, 238);
	border-radius: 4px;
	display: flex;
    align-items: center;
    justify-content: center;
	width:58px;
	height:40px;
    img{
      width: 46px;
      height: auto;
    }
}
    padding-top: var(--fs-16);
    display:flex;
    align-items:center;
    justify-content:space-between;
    i:nth-child(2){
      margin-right:auto;
      margin-left:15px;
      margin-top: -3px;

    }
  }
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 14px;
  line-height: 20px;
  color: var(--text);
  letter-spacing: 0.22em;
`;

export const DottedLine = styled.hr`
  border: 1px dashed #E2E2EE;
  width: 100%;
  margin-top:var(--fs-16);
`;

export const PrimerCardForm = styled.div`
  .inputPrimer {
    font-weight: 400;
    font-size: 14px;
    line-height: 24px;
    letter-spacing: -0.02em;
    color: #66668E;
    border: 1px solid #A2A2B8;
    border-radius: 8px;
    height: 56px;
    // padding: 0 10px;
    text-transform: capitalize;
    width: 100%;
    display: block;
    border-bottom: 1px solid #99a0a9;
    background: transparent;
    display: flex;
    flex-direction: column;
    margin-bottom: 20px;
    iframe {
      margin: 0px;
    }
  }
  .cardHolderInput {
    color: #242a2f !important;
    font-family: sans-serif !important;
    font-weight: 400 !important;
    font-size: 16px !important;
    padding: 0 10px !important;
    margin-bottom: 5px !important;
    user-select: auto !important;
    &:focus-visible {
      outline: transparent;
    }
  }

  .error {
    color: var(--red);
    font-size: var(--fs-12);
    font-family: ${TypographyENUM.lkSansRegular};

    &.margin-top-error {
      margin-top: 4px;
    }
  }
`;

export const NameInputContainer = styled.div`
  position: relative;
  min-height: 18px;

  .error {
    position: absolute;
    bottom: 0;
    left: 0;
  }
`;
