// @lk/ui-library/src/Component/V1/CheckoutMobile/Cards/Cards.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import Cards from "./Cards.component";

export default{
	title: "Components/v1/CheckoutMobile/Cards",
	component: Cards,
	savedCards: generateArgTypes({
		type: "string",
		required: true,
		description: "Array of objects of Saved Cards",
		category: "Component",
	}),
} as ComponentMeta<typeof Cards>;

const Template: ComponentStory<typeof Cards> = (args) => <Cards {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	savedCards: [ 
		{ logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
		// { logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", number: 1928737782, cardToken: "123", cardBrand: "Saved card photo" },
	],
	dataLocale : {
		"RBI_GUIDELINES" : "As per RBI guidelines we need to add an additional layer to secure your card. Agreeing to this means:",
		"RBI_GUIDELINE_1" : "Securing this card and avoiding unauthorised use.",
		"RBI_GUIDELINE_2" : "Avoiding entering card details everytime you transact on Lenskart using this card",
		"SECURE_YOUR_CARD" : "Secure your card with RBI guidelines",
		"SECURE_THIS_CARD" : "Secure this card as per RBI guidelines",
		"ADD_NEW_CARD" : "Add new card",
		"ADD_DEBIT_CREDIT" : "Add Credit/ Debit/ ATM cards",
		"LEARN_MORE" : "Learn more",
		"ENTER_CVV_CODE": "Enter your CVV code",
		"CARD_NUMBER":"Card number",
		"EXPIRY":"Expiry",
		"CVV":"CVV",
		"FULL_NAME":"Full Name",
		"VERIFY_PAY":"Verify and pay",
		"ADD_CARD_AND_PAY":"Add card and pay",
		"OK_GOT_IT":"Ok, got it"
	},
	methodData: {
		logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png",
		key: "PAYMENT_CARD"
	}
};
