// @lk/ui-library/src/Component/V1/MyAccount/MyOrder/AddPower/AddPower.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import AddPower from ".";

export default {
	title: "Components/V1/MyAccount/AddPower",
	component: AddPower
} as ComponentMeta<typeof AddPower>;

const Template: ComponentStory<typeof AddPower> = (args) => <AddPower {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		LOGOUT: "LogOut",
		GET_ORDER_UPDATES_ON_WHATSAPP: "Get Order Updates on WhatsApp",
		ORDER_PAGE: "Orders Page",
		ORDER_ID: "Order ID",
		DATE: "Date",
		TOTAL_PRICE: "Total Price",
		NEED_HELP: "Need Help",
		ORDER_DETAIL: "Order_Detail",
		PAYMENT_PENDING: "Payment Pending",
		COMPLETE_YPUR_PAYMENT_TO_PROCESS_ORDER: "Complete your payment to process order",
		CURRENCY_SYMBOL: "₹",
		PAY_NOW: "Pay Now",
		CONFIRM_ORDER: "Confirm Order",
		EARLY: "Early",
		ENTER_PRESCRIPTION_MANAULLY: "Enter Prescription Manually",
		USE_SAVED_PRESCRIPTION: "Use Saved Prescription",
		UPLOAD_PRESCRIPTION: "Upload Prescription",
		REQUEST_CALL_BACK: "Request A Call Back",
		SEND_VIA_EMAIL: "Send It Via Email"
	},
	//formatDate: (createdAt: string, pattern:string, year:string) => console.log(createdAt, pattern, year),
};
