// @lk/ui-library/src/Component/V1/MyAccount/MyOrder/AddPower/styles.tsx
import styled from "styled-components";

export const AddPowerContainer = styled.div`
    margin-top: 20px;
    font-family: 'Conv_FuturaStd-Light';
    font-size: 17px;
`;

export const Anchor = styled.a``;

export const PrescriptionManually =styled.div`
    margin: 15px auto 8px;
    width: 123px;
    height: 119px;
    background-position: 7px -2px;
    background-size: 90%;
    background: url(https://static.lenskart.com/media/wysiwyg/sccess/ic-manual.svg) no-repeat;
`;

export const PrescriptionManuallyList = styled.li`
    min-height: 188px;
    background-color: #fff;
    float: left;
    width: 31.9%;
    text-align: center;
    padding: 0px;
    list-style: none;
    margin-right: 2.09%;
    border-radius: 8px;
    box-shadow: 1px 0 0 0 #d9dedf, 0 1px 0 0 #d9dedf, 1px 1px 0 0 #d9dedf, 1px 0 0 0 #d9dedf inset, 0 1px 0 0 #d9dedf inset;
    &:nth-child(-n+3) {
        margin-top: 0px;
    };
    &:nth-child(3) {
        margin-right: 0;
    };
    cursor: var(--pointer);
`;

export const PrescriptionText = styled.p`
    font-size: 17px;
    line-height: 24px;
    font-family: "Conv_FuturaStd-Light";
    text-transform: capitalize;
    color: #445962;
    letter-spacing: 1.3px;
    margin: 10px 0 3px;
`;

export const SavedPrescriptionlist = styled.li`
    min-height: 188px;
    background-color: #fff;
    float: left;
    width: 31.9%;
    text-align: center;
    padding: 0px;
    list-style: none;
    margin-right: 2.09%;
    border-radius: 8px;
    box-shadow: 1px 0 0 0 #d9dedf, 0 1px 0 0 #d9dedf, 1px 1px 0 0 #d9dedf, 1px 0 0 0 #d9dedf inset, 0 1px 0 0 #d9dedf inset;
    &:nth-child(-n+3) {
        margin-top: 0px;
    };
    &:nth-child(3) {
        margin-right: 0;
    };
    cursor: var(--pointer);
`;

export const SavedPrescription = styled.div`
    margin: 15px auto 8px;
    width: 123px;
    height: 119px;
    background-position: 7px -2px;
    background-size: 90%;
    background: url(https://static.lenskart.com/media/wysiwyg/sccess/ic-saved.svg) no-repeat;
`;

export const UploadPrescriptionList = styled.li`
    min-height: 188px;
    background-color: #fff;
    float: left;
    width: 31.9%;
    text-align: center;
    padding: 0px;
    list-style: none;
    margin-right: 2.09%;
    border-radius: 8px;
    box-shadow: 1px 0 0 0 #d9dedf, 0 1px 0 0 #d9dedf, 1px 1px 0 0 #d9dedf, 1px 0 0 0 #d9dedf inset, 0 1px 0 0 #d9dedf inset;
    &:nth-child(-n+3) {
        margin-top: 0px;
    };
    &:nth-child(3) {
        margin-right: 0;
    };
    cursor: var(--pointer);
`;

export const UploadPrescription = styled.div`
    margin: 15px auto 8px;
    width: 123px;
    height: 119px;
    background-position: 7px -2px;
    background-size: 90%;
    background: url(https://static.lenskart.com/media/wysiwyg/sccess/ic-upload.svg) no-repeat;
`;

export const CallPrescriptionList = styled.div`
    min-height: 188px;
    background-color: #fff;
    float: left;
    width: 31.9%;
    text-align: center;
    padding: 0px;
    list-style: none;
    margin-right: 2.09%;
    border-radius: 8px;
    box-shadow: 1px 0 0 0 #d9dedf, 0 1px 0 0 #d9dedf, 1px 1px 0 0 #d9dedf, 1px 0 0 0 #d9dedf inset, 0 1px 0 0 #d9dedf inset;
    &:nth-child(-n+3) {
        margin-top: 0px;
    };
    &:nth-child(3) {
        margin-right: 0;
    };
    margin-top: 20px;
    cursor: var(--pointer);
`;

export const RequestCallBack = styled.div`
    margin: 15px auto 8px;
    width: 123px;
    height: 119px;
    background-position: 7px -2px;
    background-size: 90% !important;
    background: url(https://static.lenskart.com/media/desktop/img/Call.png) no-repeat;
`;

export const EamilPrescriptionList = styled.li`
    min-height: 188px;
    background-color: #fff;
    float: left;
    width: 31.9%;
    text-align: center;
    padding: 0px;
    list-style: none;
    margin-right: 2.09%;
    border-radius: 8px;
    box-shadow: 1px 0 0 0 #d9dedf, 0 1px 0 0 #d9dedf, 1px 1px 0 0 #d9dedf, 1px 0 0 0 #d9dedf inset, 0 1px 0 0 #d9dedf inset;
    &:nth-child(-n+3) {
        margin-top: 0px;
    };
    &:nth-child(3) {
        margin-right: 0;
    };
    margin-top: 20px;
    cursor: var(--pointer);
`;

export const SendViaEmail = styled.div`
    margin: 15px auto 8px;
    width: 123px;
    height: 119px;
    background-position: 7px -2px;
    background-size: 90%;
    background: url(https://static.lenskart.com/media/wysiwyg/sccess/Email.svg) no-repeat;
`;

export const UL = styled.ul``;
