// @lk/ui-library/src/Component/V1/MyAccount/ManageNotification/ManageNotification.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import ManageNotification from "./index";

export default {
	title: "Components/V1/MyAccount/ManageNotification",
	component: ManageNotification,
	argTypes: {
		...generateGeneralArgs(),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left onKeepMEPosted",
			category: "Staus",
		}),
	}
} as ComponentMeta<typeof ManageNotification>;

const Template: ComponentStory<typeof ManageNotification> = (args) => <ManageNotification {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		MANAGE_NOTIFICATIONS: "Manage Notifications",
		WHATSAPP_NOTIFICATION: "Whatsapp Notification",
		SMS_NOTIFICATION: "SMS Notification",
		EMAIL_NOTIFICATION: "Email Notification",
		PUSH_NOTIFICATION: "Push Notification",
	},
	onSaveNotification: () => console.log("on save notification"),
	notificationData: {
		whatsapp: true,
		sms: true,
		pushNotification: true,
		email: true
	},
	mobileView: true,
};
