// @lk/ui-library/src/Component/V1/MyAccount/ManageNotification/styles.tsx
import styled from "styled-components";

export const NotificationWrapper = styled.div<{ mobileView?: boolean }>`
	${props => props.mobileView ? "font-family: var(--font-lksans-regular); color: #808080;" : ""};
`;

export const NotificationTypeWrapper = styled.div<{ mobileView?: boolean }>`
    border-bottom: 1px solid #E8E8E8;
	padding: ${props => props.mobileView ? "20px 15px 20px 15px" : "10px"};
    margin-right: ${props => props.mobileView ? "" : "15%"};
`;

export const NotificatioTextImageWrapper = styled.div`
    display: flex;
	align-items: center;
`;

export const NotificationLogo = styled.div`
    width: 44px;
    margin-right: 3px;
	vertical-align: middle;
    max-width: 100%;
    height: 100%;
	font-size: 25px;
	font-family: var(--font-lksans-regular);
`;

export const IconWrapper = styled.div`
`;

export const Label = styled.label`
    display: flex;
    justify-content: space-between;
    cursor: pointer;
	align-items: center;
`;

export const Notificationtext = styled.div`
font-weight: 400;
font-size: 14px;
`;

export const Span = styled.span`
    font-weight: 400;
    font-size: 14px;
    top: 4px;
`;

export const SwitchContainer = styled.div`
    padding-left: 5px;
    cursor: pointer;
`;

export const ToggleButton = styled.div`
    border-right-width: 1px;
    border: 1px solid white;
    border-color: white;
    >.handle {
        top: 4px;
    }
    @include toggle-switch-mixin(20px, #8e9bad, #45C476);
    .active {
        background-color: rgb(69, 196, 118);
        transition: background-color 0.25s ease 0s;
    }
`;

export const CmnToggleLabelWrapper = styled.label <{checked: boolean}>`
	display: block;
	position: relative;
	cursor: var(--pointer);
	outline: none;
	user-select: none;
	padding: 2px;
	width: 40px;
	height: 20px;
	background-color: var(--white);
	border-radius: 20px;
	&:before {
		display: block;
		position: absolute;
		top: 1px;
		left: 1px;
		bottom: 1px;
		content: "";
		right: 1px;
		background-color: ${(props) => (!props.checked ? "#8e9bad" : "#45C476")};
		border-radius: var(--border-radius-sm);
		border-radius: white;
		transition: background 0.4s;
		&:checked {
			background-color: #8e9bad;
		}
	}
	&:after {
		display: block;
		position: absolute;
		top: 2.5px;
		left: ${(props) => (!props.checked ? "3px" : "22px")};
		/* left: ${(props) => (props.checked ? "22px" : "inherit")}; */
		bottom: 0px;
		content: "";
		width: 15px;
		height: 15px;
		background-color: var(--white);
		border-radius: 100%;
		box-shadow: 0 2px 5px var(--light-shadow);
		transition: left 0.25s;
		&:checked {
			margin-left: 44px;
		}
	}
`;

export const ButtonWrapper = styled.div<{ mobileView?: boolean }>`
    padding: ${props => props.mobileView ? "0 2%" : "8px 22px"};
    margin-top: ${props => props.mobileView ? "45vh" : "3%"};
    margin-left: ${props => props.mobileView ? "" : "65%"};
    width: 100%;
	button {
		box-shadow: 0px 3px 1px -2px rgb(0 0 0 / 20%), 0px 2px 2px rgb(0 0 0 / 14%), 0px 1px 5px rgb(0 0 0 / 12%);
	}
`;

export const Heading = styled.h2`
	font-family: 'LKFuturaStd-Medium';
	font-weight: 400;
	margin-top: 20px;
	margin-bottom: 10px;
	font-size: 25px;
`;
