// @lk/ui-library/src/Component/V1/NewStoreCredit/InputField/InputField.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const InputContainer = styled.div`
	position: relative;
`;

export const InputTextField = styled.input<{ animate:boolean }>`
	box-sizing: border-box;
	display: flex;
	flex-direction: row;
	align-items: center;
	gap: 12px;
	width: 411.68px;
	height: 56px;
	font-family: ${TypographyENUM.lkSansRegular};
	color: #000042;
	line-height: 24px;
	border: 1px solid #A2A2B8;
	border-radius: 8px;
	background-color: #FFFFFF;
	padding: 16px;
	padding-bottom: ${props => props.animate ? "8px" : "" };
	outline: none;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const PlaceholderText = styled.span<{animate:boolean}>`
	position: absolute;
	font-family: ${TypographyENUM.lkSansRegular};
	font-size: ${ props => props.animate ? "10px" : "14px" };
	line-height: ${ props => props.animate ? "16px" : "24px" };
	transform: ${ props => props.animate ? "translateY(-8px)" : ""};
	color: #66668E;
	letter-spacing: -0.02em;
	top: 16px;
	left: 16px;
	transition: 0.2s font-size ease-out, 0.2s transform ease-out;
`;
