// @lk/ui-library/src/Component/V1/LensQty/styles.tsx
import styled from "styled-components";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";
import { fontSize } from "../../Common/Input/helper";

export const QtyContainer = styled.div<{ isRTL?: boolean }>`
	display:flex;
	align-items:center;
	gap:10px;
	justify-content: center;
`;

export const QtyText = styled.div`
	font-size:13px;
	color: rgb(51,51,51);
`;

export const InputField = styled.input<{ isRTL?: boolean }>`
	width: 35px;
	text-align: center;
	border: none;
	padding:0 2px;
	height:24px;
	font-size:13px;
	background:#fff;
	border: 1px solid rgb(118,118,118);
	display: block;
	outline: none;
	color: rgb(51,51,51);
	direction: ${props => props.isRTL ? "rtl" : ""};
	&::placeholder{
		color:var(--placeholder);
	}
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const QtnWrapper = styled.div<{ styleSize: ComponentSizeENUM, styleFont: TypographyENUM }>`
	font-size: var(${(props) => fontSize[props.styleSize]});
	font-family:${props => props.styleFont};
	width: 100%;
	border-top: 1px solid #ccc;
    padding: 10px 0;
`;
