// @lk/ui-library/src/Component/V1/PrescriptionModal/EnterPd/EnterPd.styles.tsx
import styled, { css } from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const RootHeader = styled.div<{ isMobile: boolean, isRTL?: boolean }>`
  ${(props) => !props.isMobile && "margin-left: 30px;"}
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 24px;
  line-height: 36px;
  /* identical to box height, or 150% */

  /* Text/Dark/Main */

  margin-right: ${(props) => props.isRTL ? "30px" : ""};

  color: #000042;
`;

export const Helper = styled.div`
  margin: 36px 40px;
  display: flex;
  flex-direction: column;
  gap: 20px;
  /* justify-content: space-between; */
`;

export const HelperRow = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

export const UpperRoot = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: 20px;
`;

export const CardRoot = styled.div<{ selected: boolean; isMobile: boolean }>`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 16px 10px;
  gap: 4px;
  width: ${(props) => (props.isMobile ? "101px" : "140px")};
  height: 92px;
  background: #ffffff;
  border: 1px solid #cecedf;
  border-radius: 12px;
  cursor: pointer;
    ${(props) =>
		props.selected && "background: #f5f5ff;border: 1px solid #000042;"};
`;

export const PdContent = styled.span`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 700;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #000042;
  text-align: center;
  margin-top: 4px;
`;

export const Img = styled.img``;

export const LowerRoot = styled.div<{ isMobile: boolean }>`
  display: flex;
  width: ${(props) => (props.isMobile ? "100%" : "500px")};
  ${(props) => !props.isMobile && "height: 300px;"}
  ${(props) => props.isMobile && "flex-direction: column; align-items: center;"}
  justify-content: space-around;
  margin-top: 25px;
  flex-wrap: wrap;
  ${css`
    input::-webkit-outer-spin-button,
    input::-webkit-inner-spin-button {
      -webkit-appearance: none;
      margin: 0;
    }

    /* Firefox */
    input[type="number"] {
      -moz-appearance: textfield;
    }
  `}
`;
export const LowerContainer = styled.div<{lower?: boolean}>`
    ${props => props.lower && "margin-bottom: 120px;"}
`;
export const ImgContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const ImgCard = styled.img`
  border-radius: 10px;
  width: 238px;
`;
export const ImgText = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  text-align: center;
  color: #333368;
  margin-top: 10px;
`;

export const Input = styled.input<{ error: boolean }>`
  padding: 20px 16px 0px 16px;
  background: ${(props) => (props.error ? "#FFEBEA" : "#ffffff")};
  border: ${(props) =>
		props.error ? "1px solid #E34934" : "1px solid #cecedf"};
  border-radius: 8px;
  width: 100%;
  height: 56px;
  font-family: "LenskartSans-Regular";
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #000042;
  user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const ErrorHead = styled.div<{ error: boolean, isRTL: boolean }>`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 10px;
  line-height: 16px;
  color: ${(props) => (props.error ? "#E34934" : "#66668e")};
  position: relative;
  top: 26px;

  ${props => props.isRTL ? "right: 15px" : "left: 15px"};
`;

export const PdDiv = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`;

export const ErrorSign = styled.div<{isRTL: boolean}>`
  position: absolute;
  top: 33px;

  ${props => props.isRTL ? "left: 18px" : "right: 18px"};
`;

export const Field = styled.div`
  position: relative;
`;

export const Root = styled.div`
  display: flex;
  justify-content: center;
`;

export const CareRoot = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  margin-top: 100px;
  gap: 10px;
`;

export const PdMessage = styled.div`
  font-family: "LenskartSans-Regular";
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  text-align: center;
  letter-spacing: -0.02em;
  color: #333368;
`;

export const A = styled.a`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 700;
  font-size: 12px;
  line-height: 20px;

  display: flex;
  align-items: center;
  text-decoration-line: underline;

  /* Text/Dark/Main */

  color: #000042;
  cursor: pointer;
`;

export const BodyCover = styled.div`
  display: flex;
  justify-content: center;
`;

export const Body = styled.div<{ isMobile: boolean }>`
  width: ${(props) => (props.isMobile ? "100%" : "500px")};
  ${(props) => props.isMobile && "padding: 0 20px;"}
`;

export const Step = styled.div`
  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  /* font-weight: 700; */
  font-size: 14px;
  line-height: 24px;
  margin: 24px 0;

  /* Text/Dark/Main */

  color: #000042;
`;

export const ErrorText = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 20px;
  color: #e34934;
  margin-left: 10px;
  margin-top: 10px;
`;

export const ButtonBox = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-around;
`;

export const PdUpdateText = styled.div`
  position: absolute;
  background: white;
  width: 60%;
  right: 140px;
  border: 1px solid black;
  padding: 5px;
  border-radius: 10px;
  font-size: 12px;
  font-family: ${TypographyENUM.lkSansRegular};
  color: #27394e;
`;

export const Icon = styled.div`
  padding: 3px;
  background-color: black;
  border-radius: 50%;
  position: relative;
  left: 0px;
  top: 17px;
  color: white;
  cursor: pointer;
`;

export const PdInfoWrapper = styled.div`
`;

export const MobileHeadContainer = styled.div`
  padding-top: 110px;
  margin-top: 0;
  display: flex;
  align-items: center;
`;

export const ImgContain = styled.div<{isMobile: boolean}>`
  padding: 4px;
  width: 100%;
  margin: 30px;
  ${props => props.isMobile && "margin:0; margin-bottom:30px;"}
`;

export const UploadText = styled.span`
    font-size: 14px;
    color: rgb(153, 153, 153);
`;

export const UploadBox = styled.div`
    margin: 3px;
    margin-top: 10px;
    margin-bottom: 100px;
    padding: 10px;
    font-size: 13px;
    height: 81px;
    display: flex;
    align-items: center;
    background: #fafafa;
    border-radius: 4px;
`;

export const ImgSavedBox = styled.div`
    width: 100%;
    height: 37px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 10px;
    margin-left: 8px;
    margin-right: 8px;
    background: white;
    border-radius: 4px;
    border: 1px solid #dedede;
`;
