// @lk/ui-library/src/Component/V1/ProductImage/ProductImage.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import ProductImage from "./index";

export default {
	title: "Components/v1/ProductImage",
	component: ProductImage,
	argTypes: {
		...generateGeneralArgs(),
		index: generateArgTypes({
			type: "number",
			required: false,
			description: "Index of product image",
			category: "Component",
			defaultValue: 0,
		}),
		imageSrc: generateArgTypes({
			type: "string",
			required: true,
			description: "Image source",
			category: "Component",
			defaultValue: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "Function to be called on click",
			category: "Component",
			defaultValue: () => null,
		}),
	}
} as ComponentMeta<typeof ProductImage>;

const Template: ComponentStory<typeof ProductImage> = (args) => <ProductImage {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	index: 0,
	imageSrc: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
	onClick: () => null,
};
