// @lk/ui-library/src/Component/V1/OldPayment/StoreCredit/storeCredit.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { StoreCredit } from "./index";

export default {
	title:"Components/V1/Payment/StoreCredit",
	component: StoreCredit,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: false,
			description: "Unique Id",
			category: "General",
			defaultValue: "near-by-store"
		})
	}
} as ComponentMeta<typeof StoreCredit>;

const Template: ComponentStory<typeof StoreCredit> = args => <StoreCredit {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"store-credit-card",
	applyCode: () => console.log("Apply Code function cal"),
	cartData: {},
	createOrderPayment: () => console.log("Create Order Payment function call"),
	currencyMark: { INR: "₹", SGD: "$", AED: "AED", SAR: "SAR", USD: "$" },
	dataLocale: {
		APPLY: "APPLY",
		APPLY_STORE_CREDIT: "Apply Store Credit",
		CHECKOUT: "CHECKOUT",
		ENTER_AMOUNT: "Enter Amount",
		ENTER_CODE: "Enter Code",
		PLACE_ORDER: "PLACE ORDER",
		PLEASE_PAY_REMAINING_AMOUNT: "Please pay remaining amount using other payment option.",
		REMOVE: "remove"
	},
	frameProduct: false,
	loadingPlaceOrder: undefined,
	payZero: null,
	paymentCTA: { mainText: "Place order", subText: "(Submit Power in the Next Step)" },
	scMessage: null
};
