// @lk/ui-library/src/Component/V1/OldPayment/StoreCredit/storeCredit.styles.tsx
import styled from "styled-components";

export const FormWrapper = styled.div`
	display: flex;
    gap: var(--pd-10);
`;
export const InputBox = styled.input`
	min-width: 135px;
    height: var(--pd-40);
    margin-right: 0;
    /* width: 100%; */
    background: #eceeef;
    padding: var(--pd-12) 0 var(--pd-10);
    border: 1px solid var(--white);
    text-align: center;
    border-radius: var(--border-radius-xxxs);
    box-shadow: inset 0 1px 1px rgb(0 0 0 / 8%);
    transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s,-webkit-box-shadow ease-in-out 0.15s;
    font-size: var(--fs-14);
	font-family: var(--font-default-book);
	font-synthesis: none;
    line-height: 2px;
    color: var(--dark-gray);
	:focus{
		border-color: #66afe9;
    outline: 0;
    -webkit-box-shadow: inset 0 1px 1px rgb(0 0 0 / 8%), 0 0 8px rgb(102 175 233 / 60%);
    box-shadow: inset 0 1px 1px rgb(0 0 0 / 8%), 0 0 8px rgb(102 175 233 / 60%);
	}
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const StoreCreditButton = styled.button`
	color: var(--white);
    background: var(--light-gray);
    font-size: var(--fs-13);
    border: none;
    padding: var(--pd-10) var(--pd-20);
    letter-spacing: 1px;
    border-radius: var(--border-radius-xxxs);
    font-weight: var(--fw-bold);
    text-transform: var(--uppercase);
    height: var(--pd-40);
    min-width: 125px;
    pointer-events: none;
	font-family: var(--font-default-book);
	font-synthesis: none;
`;

export const StoreInfo = styled.div`
	padding: 0px var(--pd-20);
	display: flex;
	flex-flow: column;
	justify-content: center;
	height: 80px;
`;

export const StoreInfoText = styled.span`
	font-weight: var(--fw-medium);
	color: var(--gray);
	font-size: var(--fs-13);
`;

export const StoreInfoLink = styled.div`
	color: var(--irish-blue);
	cursor: var(--pointer);
	font-size: var(--fs-13);
	padding-top: var(--pd-5);
`;
