// @lk/ui-library/src/Component/V1/StudioFlow/UserDetailsForm/UserDetailsForm.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const FormWrapper = styled.form``;

export const FieldGroup = styled.div`
  display: flex;
  align-items: flex-start;
  gap: 33px;
  padding-bottom: var(--pd-32);
`;

export const ButtonWrapper = styled.div`
  max-width: 361px;
  padding-top: 32px;
  display: none;
`;

export const NoteHeadingWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: flex-start;
  gap: 4px;
`;

export const Heading = styled.div`
  font-family: ${TypographyENUM.lkSansBold};
  /* font-weight: 700; */
  font-size: var(--fs-14);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--brownish-yellow);
`;

export const NoteContent = styled.div`
  padding-top: var(--pd-16);
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: var(--fs-16);
  line-height: 24px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);

  strong {
    font-family: ${TypographyENUM.lkSansBold};
    /* font-weight: 700; */
  }
`;

export const NoteFooter = styled.div`
  padding-top: var(--pd-16);
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: var(--fs-12);
  line-height: 24px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;
