// @lk/ui-library/src/Component/V1/StudioFlow/MsiteSearch/MsiteSearch.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const SearchWrapper = styled.div`
  padding: var(--pd-24) var(--pd-16) 0;
`;

export const HeaderWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Header = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-16);
  line-height: 24px;
  letter-spacing: -0.02em;
  color: var(--text);
`;

export const IconContainer = styled.div`
  height: 32px;
  width: 32px;
`;

export const SearchWindow = styled.div`
  position: fixed;
  inset: 0;
  background: var(--white);
  padding-top: var(--pd-64);
  overflow: scroll;
  padding-bottom: var(--pd-150);
`;

export const SearchInputWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  border: 1px solid var(--light-purple-10);
  border-radius: var(--border-radius-xs);
  height: 56px;
  margin-left: var(--pd-16);
  margin-right: var(--pd-16);
  margin-bottom: var(--pd-14);
`;

export const LeftIcon = styled.div`
  padding-left: var(--pd-16);
  margin-top: 4px;
`;

export const InputField = styled.input`
  width: 100%;
  outline: none;
  border: none;
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-14);
  line-height: 24px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
  padding: 0 var(--pd-12) 0;
  user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const RightIcon = styled.div`
  padding-right: var(--pd-16);
  margin-top: 4px;
`;

export const StoreRow = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 76px;
  border-bottom: 1px dashed var(--purple-haze);
  padding-left: var(--pd-16);
  padding-right: var(--pd-16);
`;

export const StoreDetails = styled.div``;

export const StoreName = styled.div`
  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  font-weight: 700;
  font-size: var(--fs-12);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;

export const Details = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-12);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--text);
`;
