// @lk/ui-library/src/Component/NewHtoFlow/FAQs/Accordion.component.tsx
import React, { useState } from "react";
import * as Icons from "../../../Icons/index";
import {
	AccordionWrapper,
	TitleBox,
	TitleWrapper,
	Title,
	Content,
	Icon,
	Image,
	StatusIcon
} from "./Accordion.styles";

import { ReactNode } from "react";
import { TypographyENUM } from "../../../Types/general";

interface AccordionTypes {
	children: ReactNode
	title: string
	isMaxHeight: boolean
	showBorderTop: boolean
	canCollapse: boolean
	isComponent: boolean
	font: TypographyENUM
	isRTL: boolean
    btnText: string;
	icon: string;
}

const Accordion = ({
	children,
	title,
	isMaxHeight,
	showBorderTop,
	canCollapse,
	isComponent,
	font,
	isRTL = false,
	icon
}: AccordionTypes) => {
	const [expand, setExpand] = useState<boolean>(false);
	const handleExpand = () => {
		setExpand(!expand);
	};
	return (
		<AccordionWrapper borderTopStyle={showBorderTop} styleFont={font} isRTL={isRTL}>
			<TitleBox styleComponent={isComponent} onClick={() => handleExpand()} expand={expand} isRTL={isRTL}>
				<TitleWrapper>
					{
						icon && <StatusIcon><Image src={icon} alt={"icon"} /></StatusIcon>
					}
					<Title isRTL={isRTL}>{title}</Title>
				</TitleWrapper>
				{canCollapse && <Icon>{!expand ? <Icons.DownArrow stroke="#000042" /> : <Icons.UpArrow stroke="#000042" />}</Icon>}
			</TitleBox>
			{expand && <Content isMaxHeight={isMaxHeight}>{children}</Content>}
		</AccordionWrapper>
	);
};

export default Accordion;
