// @lk/ui-library/src/Component/V1/Cart/CartCard/NewCoupon/NewCouponCard.styles.tsx
import styled from "styled-components";

export const Container = styled.div<{ borderColor?: string }>`
	display: flex;
	flex-direction: column;
	background: var(--white);

	border: 1px solid #11DAAC;
	border-color: ${props => props.borderColor || "var(--purple-haze)"};
	box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
	border-radius: var(--border-radius-xs);
`;

export const TopSection = styled.div<{ backgroundColor?: string }>`
	display: flex;
	flex-direction: column;
	background-color: ${props => props.backgroundColor || ""};
	padding	: 24px 24px 12px 24px;
	border-radius: var(--border-radius-xs) var(--border-radius-xs) 0 0;
	gap: 10px;
`;


export const BottomSection = styled.div`
	display: flex;
	flex-direction: column;
	gap: 10px;
	padding	: 12px 24px 24px 24px;
`;

export const Row = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
`;

export const SubItems = styled.div`
	display: flex;
	flex-direction: row;
	gap: 5px;
`;

export const ApplyCoupon = styled.div<{disable?: boolean}>`
display: flex;
flex-direction: row;
gap: 5px;
font-family: var(--font-lksans-bold);
font-style: normal;
font-weight: 700;
font-size: 12px;
line-height: 16px;
text-align: right;
text-decoration-line: underline;
color: var(--dark-blue-100);
pointer-events: ${props => props.disable ? "none" : "auto"};
cursor: ${props => props.disable ? "auto" : "pointer"};
/* svg path {
    stroke: var(--dark-blue-100);
  } */
`;

export const BrandImageWrapper = styled.div<{ brandImage?: string }>`
	padding: 4px;
	width: 42px;
	height: 32px !important;
	background-position: center;
	background-image: ${(props) => props.brandImage ? `url(${props.brandImage})` : null};
	background-repeat: no-repeat;
	background-size: contain;
	border: 1px solid var(--light-purple-10);
	border-radius: var(--border-radius-xxxs);
`;

export const Heading = styled.div`
	font-family: var(--font-lksans-bold);
	font-style: normal;
	font-weight: 700;
	font-size: 14px;
	line-height: 20px;
	color: var(--dark-blue-100);
`;

export const DescriptionText = styled.div`
	font-family: var(--font-lksans-regular);
	font-style: normal;
	font-weight: 500;
	font-size: 12px;
	line-height: 16px;
	color: #474786;
`;

export const Subdescription = styled.div<{ subDescriptionColorCode?: string }>`
	font-family: var(--font-lksans-regular);
	font-style: normal;
	font-weight: 500;
	font-size: 12px;
	line-height: 16px;
	color: ${props => props.subDescriptionColorCode || "#99382B"};
`;

export const TermsAndCondition = styled.div`
    font-family: var(--font-lksans-bold);
	font-style: normal;
	font-weight: 700;
	letter-spacing: -0.02em;
	font-size: 12px;
	line-height: 16px;
	text-decoration-line: underline;
	color: var(--dark-blue-100);
`;
