// @lk/ui-library/src/Component/Common/Dropdown(common)/Dropdown.types.tsx
import { ComponentSizeENUM, DataType, TypographyENUM } from "../../../Types/general";
// import React from "react";

export interface InputType {
	placeholder: string;
	type: string;
	componentSize?: ComponentSizeENUM;
	width: string;
	id: string;
	disabled?: boolean;
	// readonly?: boolean;
	error?: boolean;
	label:string;
	font:TypographyENUM;
	// onValueChange?: (value: React.ChangeEvent<HTMLInputElement>) => void;
	// includeButton?:boolean;
	// buttonText?:string;
	// buttonPosition?:positionENUM;
	// buttonOnClick?:() => void;
	// will add functionalities for below parameters later
	// includeIcon?:boolean;
	// iconText?:string;
	// iconPosition?:positionENUM;
	// iconOnClick?:() => void;
	countryCodePrefix?:string;
	// onKeyUpChange?: (value: React.ChangeEvent<HTMLInputElement>) => void;
	// suffix?:string;
	options: DataType[];
	optionsText: string;
	optionValue: string;
}

export enum positionENUM{
	right="right",
	left="left",
	// top="top",
}

export enum typeENUM {
	text = "text",
	password = "password",
	email = "email",
	number = "number",
}

export const inputs = {
	type: typeENUM,
};
