// @lk/ui-library/src/Component/Common/Dropdown(common)/Dropdown.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Dropdown from "./Dropdown.component";
import React from "react";
import { inputs, positionENUM, typeENUM } from "./Dropdown.types";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";

export default {
	title: "Components/Common/Dropdown(Common)",
	component: Dropdown,
	argTypes: {
		...generateGeneralArgs(),
		title: generateArgTypes({
			type: "string",
			required: true,
			description: "Value that will be rendered",
			category: "Component",
			defaultValue: "Hello World",
		}),
		placeholder: generateArgTypes({
			type: "string",
			required: false,
			description: "Placeholder that will be rendered",
			category: "Component",
			defaultValue: "Lorem Ipsum placeholder",
		}),
		label: generateArgTypes({
			type: "string",
			required: false,
			description: "Label that will be rendered",
			category: "Component",
			defaultValue: "",
		}),
		type: generateArgTypes({
			type: "string",
			required: true,
			description: "Type of input will be rendered",
			category: "Component",
			defaultValue: typeENUM.text,
			control: "radio",
			options: typeENUM,
		}),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		}),
		disabled: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to enable/disable the input",
			category: "Component",
			defaultValue: "false",
		}),
		readonly: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to set readOnly condition in input",
			category: "Component",
			defaultValue: "false",
		}),
		error: generateArgTypes({
			type: "boolean",
			required: false,
			description: "Boolean to set error in input",
			category: "Component",
			defaultValue: "false",
		}),
		onChange: generateArgTypes({
			type: "function",
			required: false,
			description: "onChange handler for input",
			category: "Events",
		}),
	},
} as ComponentMeta<typeof Dropdown>;

const Template: ComponentStory<typeof Dropdown> = (args) => <Dropdown {...args}/>;

export const Fields = Template.bind({});
Fields.args = {
	label:"First Name",
	componentSize: ComponentSizeENUM.large,
	disabled: false,
	error: false,
	readonly: false,
	width: "50%",
	type: inputs.type.text,
	includeButton: true,
	buttonText: "CHANGE",
	buttonPosition: positionENUM.right,
	buttonOnClick: () => console.log("button clicked"),
	countryCodePrefix:"+91",
	font:TypographyENUM.defaultBook,
	options: ["Andaman and Nicobar Islands", "Jharkhand", "Bihar"],
	optionsText: "",
	optionValue: ""
};
