// @lk/ui-library/src/Component/V1/PackageRadioGroup/mockValue.ts
export const prescriptionType = [{
	"id": "single_vision",
	"title": "Single Vision",
	"isPackageAvailable": true,
	imageUrl: "https://static.lenskart.com/media/desktop/img/pdp/single_vision.png",
	subText: ""
},
{
	"id": "zero_power",
	"title": "Zero Power",
	"isPackageAvailable": true,
	imageUrl: "https://static.lenskart.com/media/desktop/img/pdp/zero_power.png",
	subText: "Block 98% of harmful rays<br>(Anti-glare and blue-cut options)"
}
];

export const packageTextConfigData = {
	"single_vision": {
		"title": "Powered Specs_sv",
		"subtitle": "For distance or near vision <br>(Thin, anti-glare, blue-cut options)"
	},
	"zero_power": {
		"title": "Phone/PC Zero Power Eyeglasses",
		"subtitle": "Block 98% of harmful rays<br>(Anti-glare and blue-cut options)"
	},
	"bifocal": {
		"title": "Bifocal Progressive",
		"subtitle": "Bifocal and Progressives<br>(For two powers in same lenses)"
	},
	"tinted_sv": {
		"title": "Power Sunglasses",
		"subtitle": "For Distance or Near Vision<br>(Green, Grey, Brown)"
	},
	"frame_only": {
		"title": "Frame Only",
		"subtitle": "Buy Only Frame"
	},
	"without_power": {
		"title": "Without Power",
		"subtitle": "Regular Sunglasses"
	},
	"sunglasses": {
		"title": "With Power",
		"subtitle": "Prescription Sunglasses"
	}
};

export const selectedPowerTypeData = {
	id: "single_vision",
	isDefault: true,
	isPackageAvailable: true,
	title: "Single Vision"
};
