// @lk/ui-library/src/Component/V1/PackageRadioGroup/PackageRadioGroup.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../StorybookHelper";
import { ComponentSizeENUM } from "../../../Types/general";
import PackagesRadioGroup from "./index";
import { prescriptionType, selectedPowerTypeData } from "./mockValue";

export default {
	title: "Components/v1/PackagesRadioGroup",
	component: PackagesRadioGroup,
	argTypes: {
		...generateGeneralArgs(),
		onChange: generateArgTypes({
			type: "function",
			required: true,
			description: "onClick handler for button",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show right to left",
			category: "Status",
		}),
	},
} as ComponentMeta<typeof PackagesRadioGroup>;

const Template: ComponentStory<typeof PackagesRadioGroup> = (args) => <PackagesRadioGroup {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "radio-button",
	componentSize: ComponentSizeENUM.small,
	prescriptionTypeData: prescriptionType,
	selectedId: selectedPowerTypeData.id,
	onChange: (id) => { console.log(`${id} clicked`);}
};
