// @lk/ui-library/src/Component/V1/PackageRadioGroup/styles.tsx
import styled from "styled-components";
import { ComponentSizeENUM } from "../../../Types/general";
import { fontSize, spacing, TitlefontSize } from "./helper";


export const ButtonWrapper = styled.button<{
	styleSize: ComponentSizeENUM
	isBorder: boolean
	width: string
}>`
	cursor: ${(props) => (props.disabled ? "not-allowed" : "pointer")};
	border-radius: var(--border-radius-xs);
	border: ${(props) => (props.isBorder ? "2px solid var(--irish-blue)" : "1px solid #e5e8e9")};
	background: var(--white);
	padding: var(${(props) => spacing[props.styleSize]});
	font-size: var(${(props) => fontSize[props.styleSize]});
	width: ${(props) => (props.width ? props.width : null)}%;
	display: flex;
	justify-content: center;
`;

export const FlexWrapper = styled.div`
	display: flex;
	align-items: center;
	width: 100%;
	justify-content: var(--space-between);
`;
export const LeftWrapper = styled.div`
	display: flex;
	align-items: center;
`;
export const RightWrapper = styled.div`
	display: flex;
	align-items: center;
`;
export const IconWrapper = styled.span`
	margin: 5px;
`;

export const PowerText = styled.div`
	text-align: left;
	margin-left: var(--pd-10) ;
`;

export const Wrapper = styled.div`
	display: flex;
	flex-direction: column;
	gap: var(--pd-10);
	width: 80%;
`;

export const Title = styled.div<{
	styleSize: ComponentSizeENUM
}>`
	font-size: var(${(props) => TitlefontSize[props.styleSize]});
	font-weight: var(--fw-semi-bold);
`;
