// @lk/ui-library/src/Component/V1/CheckoutMobile/AddAddressForm/AddAddressForm.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const ReviewFormWrapper = styled.div<{styleFont:TypographyENUM}>`
	font-family:${props => props.styleFont};
	// padding: 10px;
	// border: 1px solid rgba(60, 60, 60, .54);
    // border-radius: 8px;
`;

export const InvisibleButton = styled.button<{btnVisible: boolean}>`
	display: ${props => props.btnVisible ? "inline-block" : "none" };
	background-color: #00b9c6;
  color: #fff;
	width: 50%;
  margin-bottom: 0;
  font-weight: 400;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid transparent;
  padding: 6px 12px;
  font-size: 14px;
  line-height: 1.42857143;
  border-radius: 4px;
  border-color: #ccc;
	margin-bottom: 10px;
`;

export const FormButtonWrapper = styled.div`
	position: relative;
  justify-content: left;
  left: -10px;
  top: -10px;
	button{
		border-radius: 8px 0 8px 0;
	}
`;

export const Form=styled.form`
	display:flex;
	flex-direction:column;
	gap:var(--spacing-md);
`;

export const InputWrapper = styled.div`
	display:flex;
	gap:var(--spacing-md);
	flex-wrap: wrap;
	align-items: center;
`;

export const RadioContainer = styled.div`
	display: flex;
	align-items: center;
	gap: var(--spacing-md);
	min-width: 96px;

`;

export const InputField = styled.input<{width:string}>`
	width: ${props => props.width};
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const TextAreaField = styled.textarea<{width:string}>`
	width: ${props => props.width};
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const FormActionWrapper = styled.div`
	display:flex;
	align-items:center;
	justify-content:right;
`;

export const FormInputError = styled.small`
	color:var(--error);
`;

export const FloatingInputError = styled.div<{font: TypographyENUM}>`
	color:var(--error);
	font-size: var(--fs-10);
	display: block;
	line-height: 2;
	font-family:${props => props.font};
`;

export const SubTextUnitNumber = styled.p`
	color:"#66668E";
	margin:0;
	padding-top:var(--pd-5);
	font-size:0.8em;
`;

export const FloatingInputWrapper = styled.div`
	display:flex;
	flex-direction:column;
	position: relative;
	margin-bottom:5px;
`;

export const LandmarkTextarea = styled.textarea<{font: TypographyENUM}>`
	width: 100%;
	height: 100%;
	border: none;
	outline: none;
	background: transparent;
	font-family:${props => props.font};
	margin-top: 8px;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const AddressLabel = styled.label`
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-100);
	margin-bottom: 16px;
	font-family: ${TypographyENUM.lkSansRegular};
`;

export const RadioWrapper = styled.div`
	position: relative;

`;


export const RadioButton = styled.div`
	width: 20px;
	height: 20px;
	background: #FFFFFF;
	border: 1px solid #5C5C86;
	border-radius: 10px;

	&::before {
		// top: 68px;
		// left: 14px;
		border: 1px solid #a2a2b8;
		width: 20px;
		height: 20px;
		border-radius: 10px;
		content: '';
		position: absolute;
		transition: all 0.3s ease;
		transition-property: transform, border-color, -webkit-transform;
	}

	&::after {
		// top: 68px;
		// left: 14px;
		width: 20px;
		height: 20px;
		border-radius: 10px;
		background: #000042;
		position: absolute;
	}
`;

export const RadioLabel = styled.div<{font: TypographyENUM}>`
	font-family:${props => props.font};
	font-style: normal;
	font-weight: 400;
	font-size: 12px;
	line-height: 14px;
	letter-spacing: -0.02em;
	color: #000042;
	margin-bottom: 20px;
    margin-top: -2px;
`;

export const SubHeading = styled.div<{font: TypographyENUM}>`
    font-family: ${TypographyENUM.lkSansBold};
	font-style: normal;
	font-weight: var(--fw-regular);
	font-size: var(--fs-14);
	line-height: 24px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-100);
	margin-bottom: -4px;
    margin-top: 8px
`;
