// @lk/ui-library/src/Component/V1/CheckoutMobile/AddAddressForm/Radio.tsx
import React from "react";
import styled from "styled-components";

const RadioContainer = styled.div<{isSelected: boolean;disabled?: boolean;}>`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 20px;
    height: 20px;
    cursor: ${props => props.disabled ? "not-allowed" : "pointer"};
    background: ${props => props.isSelected ? "#D0FBF1" : "#FFFFFF"};
    border-radius: 10px;
    border: 1px solid #5C5C86;
    background: ${props => props.isSelected ? "#000042" : ""};
    transform: scale(1);
    content: '';
`;
const RadioInnerCircle = styled.div<{isSelected: boolean;disabled?: boolean;}>`
    display: ${props => props.isSelected ? "block" : "none"};
    width: 8px;
    height: 8px;
    border-radius: 50%;
    width: 4px;
    height: 9px;
    border: ${props => props.isSelected ? "solid #fff" : ""};
    border-width: 0 1px 1px 0;
    -webkit-transform: rotate(45deg);
    transform: rotate(45deg);
    margin: 5px 7px;
    z-index: 1;
    border-radius: 0px;
`;

const Radio = ({
	isSelected = false,
	onClick = () => null,
	disabled
}: {
        isSelected?: boolean,
        onClick?: () => void,
        disabled?: boolean
}) => {
    
	return (
		<RadioContainer isSelected={isSelected} onClick={onClick} disabled={disabled}>
			<RadioInnerCircle className="radioBtn" isSelected={isSelected} disabled={disabled} />
		</RadioContainer>
	);
};

export { Radio };
