// @lk/ui-library/src/Component/Common/Spinner/styles.tsx
import styled, { keyframes } from "styled-components";

export const SpinnerWrapper = styled.div<{fullPage?:boolean, padding?: string,backgroundColor?:string}>`
	padding	: ${props => !props.fullPage ? props.padding || "20px 10px" : ""};
	position: ${props => !props.fullPage ? "relative" : "fixed"};
	left: ${props => props.fullPage ? 0 : ""};
	right: ${props => props.fullPage ? 0 : ""};
	top: ${props => props.fullPage ? 0 : ""};
	bottom: ${props => props.fullPage ? 0 : ""};
	z-index: ${props => props.fullPage ? 1000 : 3};
	background: ${props => props.fullPage ? "rgba(255,255,255,0.9)" : ""};
	display: ${props => props.fullPage ? "flex" : ""};
	align-items: ${props => props.fullPage ? "center" : ""};
	justify-content: ${props => props.fullPage ? "center" : ""};
	${props => props.backgroundColor ? `background:${props.backgroundColor}` : ""};
`;

export const spinnerAnimation = keyframes`
	100%{
		transform: rotate(360deg);
	}
`;

export const SpinnerLoader = styled.div<{isAlgoliaSearch?:boolean}>`
	height: 20px;
  width: 20px;
  animation: ${spinnerAnimation} 0.8s infinite linear;
  border: ${props => props.isAlgoliaSearch ? "1.5px solid #000042" : "2px solid #18cfa8"};
  border-right-color: var(--transparent);
  border-radius: var(--border-radius-sm);
  will-change: transform;
  margin: 0 auto;
`;
