// @lk/ui-library/src/Component/V1/Accordion/Accordion.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../StorybookHelper";
import Accordions from "./index";
import { filters } from "./mockValue";
import { TypographyENUM } from "../../../Types/general";

export default {
	title: "Components/v1/Accordion",
	component: Accordions,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean Text left to right",
			category: "Component",
			defaultValue: "false",
		}),
		filters: generateArgTypes({
			type: "array",
			required: true,
			description: "Filters Data",
			category: "Component",
			control: "array"
		})
	},
} as ComponentMeta<typeof Accordions>;

const Template: ComponentStory<typeof Accordions> = (args) => (
	<Accordions {...args} />
);
export const Overview = Template.bind({});
Overview.args = {
	id: "AccordionGroup",
	filters: filters,
	font: TypographyENUM.defaultBook,
	isRTL: false
};
