// @lk/ui-library/src/Component/V1/PackageScreenNew/PackageItemGroup/PackageItemGroup.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { prescriptionType, selectedPowerTypeData } from "../../PackageRadioGroup/mockValue";
import PackageItemGroup from "./PackageItemGroup";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";

export default {
	title: "Components/v1/PackageScreenNew/PackageItemGroup",
	component: PackageItemGroup,
	argTypes: {
		...generateGeneralArgs(),
		onChange: generateArgTypes({
			type: "function",
			required: true,
			description: "onClick handler for button",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show right to left",
			category: "Status",
		}),
	},
} as ComponentMeta<typeof PackageItemGroup>;

const Template: ComponentStory<typeof PackageItemGroup> = (args) => <PackageItemGroup {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "radio-button",
	prescriptionTypeData: prescriptionType,
	selectedId: selectedPowerTypeData.id,
	onChange: (id) => { console.log(`${id} clicked`);},
	descriptiveText: "Not sure what to select?",
	helplineNumber:  "180011111",
	localeData:{
		"CALL": "Call"
	}
};
