// @lk/ui-library/src/Component/V1/Authorization/Authorization.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateGeneralArgs } from "../../../StorybookHelper";
import { TypographyENUM } from "../../../Types/general";
import Authorization from "./index";

export default {
	title:"Components/V1/Authorization",
	component: Authorization,
	argTypes: {
		...generateGeneralArgs()
	}
} as ComponentMeta<typeof Authorization>;

const Template: ComponentStory<typeof Authorization> = args => <Authorization {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"signup-form",
	dataLocale:{
		"REGISTER": "Register",
		"FIRST_NAME": "First Name",
		"PASSWORD": "Password",
		"LAST_NAME": "Last Name",
		"EMAIL_ADDRESS": "Email Address",
		"MOBILE": "Mobile",
		"REFERRAL_CODE": "Referral code",
		"T_C_APPLY": "T&C Apply",
		"OPTIONAL": "Optional",
		"HAVE_REFERRAL": "Have a referral code",
		"BANNER": "banner",
		"MOBILE_EMAIL_TEXT": "Mobile / Email",
		"PROCEED": "Proceed",
	},
	font:TypographyENUM.defaultBook,
	signInStatus:{
		type: null,
		showOTP: false,
		showPassword: false,
		signInStatus: false,
		isLoading: false,
		isError: false,
		errorMessage: null
	},
	onSignIn:() => null,
	onSignUp:() => null,
};
