// @lk/ui-library/src/Component/V1/Cart/Coupon/Coupon.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import Coupon from "./index";
import { TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/v1/Cart/Coupon",
	component: Coupon,
	argTypes: {
		...generateGeneralArgs(),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of box",
			category: "Component",
			defaultValue: "100",
		}),
		brandImage: generateArgTypes({
			type: "string",
			required: true,
			description: "To set the image of offer provided by bank or organisation",
			category: "Component",
			defaultValue: "",
		}),
		headText: generateArgTypes({
			type: "string",
			required: true,
			description: "To set head text of any offer",
			category: "Component",
			defaultValue: "FLAT100",
		}),
		subText: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "Save ₹100 with FLAT100",
		}),
		descText: generateArgTypes({
			type: "string",
			required: true,
			description: "To set description of any offer",
			category: "Component",
			defaultValue: "Valid on orders with items worth ₹3000 or more",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for Apply Coupon",
			category: "Events",
		}),
	},
} as ComponentMeta<typeof Coupon>;

const Template: ComponentStory<typeof Coupon> = (args) => <Coupon {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "Coupon",
	width: "100",
	brandImage: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png",
	headText: "FLAT100",
	subText: "Save ₹100 with FLAT100",
	descText: "Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more",
	onClick: () => {console.log("Apply Coupon Clicked");},
	font: TypographyENUM.defaultBook,
};
