// @lk/ui-library/src/Component/V1/Cart/Coupon/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const CouponWrapper = styled.div <{ width: number, styledFont: TypographyENUM, mobileView: boolean } >`
	box-sizing: border-box;
	display: flex;
	flex-direction: column;
	/* padding: var(--pd-20) var(--pd-20); */
	padding: 24px;
	/* gap: var(--spacing-sm); */
	gap: 12px;
	gap: ${(props) => props.mobileView ? "0" : "12px"};
	background: var(--white);
	border: 1px solid var(--purple-haze);
	border-radius: var(--border-radius-xs);
	font-family: ${(props) => props.styledFont};
	color: var(--dark-blue-100);
	box-shadow: 0px 1px 3px rgba(16, 24, 40, 0.1), 0px 1px 2px rgba(16, 24, 40, 0.06);
`;

export const FlexWrapper = styled.div<{isGap?: boolean, isBold?:boolean, isDescText?: boolean, isSubText?: boolean}>`
	display: flex;
	letter-spacing: -0.02em;
	line-height: 20px;
	align-items: center;
	width: 100%;
	font-size: ${props => props.isSubText ? "var(--fs-16)" : "var(--fs-12)"};
	// gap:  ${(props) => props.isGap ? "5px" : null };
	justify-content: space-between;
	font-family:  ${(props) => props.isBold ? "var(--font-lksans-bold)" : "var(--font-lksans-regular)" };
  	color: ${(props) => props.isDescText ? "var(--dark-blue-50)" : "var(--dark-blue-100)"};
`;

export const FlexColWrapper = styled.div`
	display: flex;
	flex-direction:column ;
	align-items: start;
	width: 100%;
	justify-content: var(--space-between);
	/* gap: var(--spacing-sm); */
	gap: 12px;
`;

export const SubText = styled.div<{styleFont: TypographyENUM}>`
	font-weight: var(--fw-regular);
	font-size: var(--fs-14);
	color: var(--dark-blue-75);
	font-family: ${(props) => props.styleFont};
`;

export const BrandImageWrapper = styled.div<{brandImage?: string}>`
	padding: 4px;
	width: 42px;
	height: 32px !important;
	background-position: center;
	background-image: ${(props) => props.brandImage ? `url(${props.brandImage})` : null };
	background-repeat: no-repeat;
	background-size: contain;
	border: 1px solid var(--light-purple-10);
	border-radius: var(--border-radius-xxxs);
`;

export const ApplyButton = styled.span`
	font-size: var(--fs-14);
	font-family: var(--font-lksans-bold);
	color: var(--vivid-green-100) !important;
	/* margin-right: var(--spacing-md); */
	cursor: var(--pointer);
`;

export const ReadMoreButton = styled.span`
	font-size: var(--fs-14);
	font-family: var(--font-lksans-bold);
	color: var(--dark-blue-100);
	text-decoration: underline;
	cursor: var(--pointer);
`;

export const DashedHR = styled.div<{mobileView:boolean}>`
	background-image: linear-gradient(to right, #cecedf 70%, rgba(255,255,255,0) 10%);
    background-position: bottom;
    background-size: 5px 2px;
    background-repeat: repeat-x;
	height: 1px;

	margin:  ${(props) => props.mobileView ? "10px 0" : "0 0" };
`;

export const ApplyButtonNew = styled.div`
	font-family: 'LenskartSans-Regular';
	font-style: normal;
	font-weight: 700;
	font-size: 12px;
	line-height: 20px;
	letter-spacing: -0.02em;
	text-decoration-line: underline;
	color: #000042;
`;

export const CouponHeader = styled.div<{ brandImage?: string }>`
	font-family: 'LenskartSans-Regular';
	font-style: normal;
	font-weight: 700;
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: #000042;

	margin-right:auto;
	margin-left:${props => props.brandImage ? "10px" : ""};
`;

export const FlexWrapperNew = styled.div`
	font-weight: 700;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #000042;
  font-family: 'LenskartSans-Bold';
`;

export const FlexWrapperNewText= styled.div`
	font-family: 'LenskartSans-Regular';
	font-style: normal;
	font-weight: 400;
	font-size: 10px;
	line-height: 16px;
	letter-spacing: -0.02em;
	color: #737397;

	margin-top:5px;
`;

export const TermsCondition = styled.div`
	font-family: 'LenskartSans-Regular';
	font-style: normal;
	font-weight: 700;
	font-size: 10px;
	line-height: 16px;
	letter-spacing: -0.02em;
	text-decoration-line: underline;
	color: #333368;
	margin-top:10px;
`;
