// @lk/ui-library/src/Component/V1/NewPayment/components/subcomponents/PayLater.tsx

import React, { useState } from "react";
import styled from "styled-components";
import { Cross } from "../../../../../Icons";
import { DataType, kindENUM, ThemeENUM, TypographyENUM } from "../../../../../Types/general";
import Button from "../../../Button";

const PayNowBtn = styled.button<{font: TypographyENUM}>`
    margin-top: 20px;
	padding: 12px 20px;
	width: 250px;
	height: 48px;
	border-radius: 100px;
	background: #fff;
	color: #000042;
	border:1px solid #000042;
	cursor: pointer;
	font-family: ${props => props.font};
	font-style: normal;
	font-weight: 700;
	font-size: 16px;
	line-height: 24px;
	display: flex;
	align-items: center;
	justify-content: center;
	letter-spacing: -0.02em;
	outline: none;
	&:disabled,
	&[disabled] {
		color: #9999B3;
		background: #DBDBEA;
		border:1px solid transparent;
	};
	span {
		display: inline-flex;
		margin-right: 10px;
		svg {
			width: 6px;
    		height: 12px;
			path{
				stroke: #000042;
			}
		}
	};
	.disable {
		pointer-events: none;
	}
`;

export const PayLaterModal = styled.div`
	position: relative;
    padding: 15px;
    height: 100%;
	max-height: 506px;
	box-shadow: 0 5px 15px rgba(0,0,0,.5);
	background-color: #fff;
	width: 900px;
	margin: auto;
	top: 15%;
`;

export const CloseButton = styled.div`
	display: flex;
	justify-content: flex-end;
	width: 100%;
	margin-top: -12px;
	margin-bottom: 16px;
	margin-right: 10px;
	cursor:pointer;
`;

export const PayLaterContainer = styled.div`
	display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    margin: 20px 0px;
`;

export const PayLaterHeader = styled.div`
	font-weight: 400;
    font-size: 24px;
    line-height: 160%;
    letter-spacing: 0.15px;
	margin-top:20px;
`;

export const PayLaterDescription = styled.div`
	font-weight: 400;
    font-size: 16px;
    line-height: 143%;
    text-align: center;
    letter-spacing: 0.15px;
    margin: 10px 20px;
    width: 640px;
    height: 46px;
`;

export const PayLaterSubText = styled.div`
	font-weight: 400;
    font-size: 16px;
    letter-spacing: 0.15px;
    margin-top: 30px;
`;

export const PayLaterButtonContainer = styled.div`
	display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    margin: 20px 0px;
`;

export const PayLaterButtonPrimary = styled.div`
	color: #fff;
    background: var(--vivid-green-100);
    font-size: 14px;
    display: block;
    border: none;
    padding: 12px 20px;
    letter-spacing: 2px;
    border-radius: 5px;
    min-width: 290px;
    font-weight: 700;
    text-align: center;
    cursor: pointer;
`;

export const PayLaterButtonSecondary = styled.div`
	font-weight: 500;
    font-size: 16px;
    line-height: 26px;
    letter-spacing: 0.47px;
    text-transform: uppercase;
    color:var(--vivid-green-100);
    margin-top: 10px;
    cursor: pointer;
`;
export const ModalWrapper = styled.div`
	position: fixed;
    margin: 0 auto;
    top: 0;
    left: 0;
    padding-top: 10px;
    z-index: 12000 !important;
    overflow-x: hidden;
    overflow-y: hidden;
    height: 100vh;
	width: 100vw;
	background:rgba(0,0,0,0.5);
`;


export interface PayLaterType {
    dataLocale: DataType;
    handlePayLater: ()=>void;
    handlePayNow?: ()=>void;
	isPrimerOff?: boolean;
	isContactLensConsentEnabled?: boolean;
	isContactLensCheckboxChecked?: boolean;
}

// handlePayNow will be passed from jus base component for other conditions close this modal on click
const PayLater = ({ dataLocale, handlePayLater, handlePayNow, isPrimerOff=false, isContactLensConsentEnabled, isContactLensCheckboxChecked } : PayLaterType) => {
	const [ isPayLaterModalActive, setIsPayLaterModalActive]=useState(false);
	const handlePayLaterClick = ()=>{
		handlePayLater();
	};
	const handlePayNowClick = ()=>{
		if(isPrimerOff && handlePayNow){
			handlePayNow();
			return;
		}
		setIsPayLaterModalActive(false);
	};
	return <>
		{isPrimerOff? <Button id="pay-later" font={TypographyENUM.lkSansBold} kind={kindENUM.border}
			onClick={()=>setIsPayLaterModalActive(true)}
			text={dataLocale.PAY_LATER}
			theme={ThemeENUM.secondary}
			fontSize={20}
			width="100"
			disabled={isContactLensConsentEnabled && !isContactLensCheckboxChecked}
		/> :<PayNowBtn
			onClick={()=>setIsPayLaterModalActive(true)}
			font={TypographyENUM.lkSansBold}
			disabled={isContactLensConsentEnabled && !isContactLensCheckboxChecked}
		>
			<span>{dataLocale.PAY_LATER}</span>

		</PayNowBtn>}
		{isPayLaterModalActive &&
		<ModalWrapper>
			<PayLaterModal>
				<PayLaterContainer>
					<CloseButton >
						<Cross onClick={()=>setIsPayLaterModalActive(false)} />
					</CloseButton>
					<img alt="img" src="https://static.lenskart.com/media/mobile/images/card.svg"></img>
					<PayLaterHeader>
						Not Ready to Pay Yet?
					</PayLaterHeader>
					<PayLaterDescription>
						Visit the showroom first and complete payment after your consultation.
						<div>Note: Your frame selection is not guaranteed until payment is made</div>
					</PayLaterDescription>
					<PayLaterSubText>
						Do you still wish to pay later?
					</PayLaterSubText>
					<PayLaterButtonContainer>
						<PayLaterButtonPrimary onClick={handlePayLaterClick}>Yes, continue</PayLaterButtonPrimary>
						<PayLaterButtonSecondary onClick={handlePayNowClick}>No, I&apos;ll pay for my order now</PayLaterButtonSecondary>
					</PayLaterButtonContainer>
				</PayLaterContainer>
			</PayLaterModal>
		</ModalWrapper>}
	</>;
};

export { PayLater };
