// @lk/ui-library/src/Component/V1/NewPayment/components/subcomponents/QRCodeForm.tsx
import React from "react";
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

const BankFormContainer = styled.div<{isOpen?: boolean}>`
    padding-left: 36px;
    ${props => !props.isOpen && `
        display: none;
    `}
`;
const BankForm = styled.div`
    display: flex;
    flex-direction: column;
    /* width: 444px; */
	> p {
		font-family: ${TypographyENUM.lkSansRegular};
		font-style: normal;
		font-weight: 400;
		font-size: 12px;
		line-height: 14px;
		letter-spacing: -0.02em;
		color: #000042;
	}
`;
const PayNowBtn = styled.button<{font: TypographyENUM}>`
	padding: 12px 20px;
	width: 100%;
	height: 48px;
	border-radius: 100px;
	background: #11DAAC;
	color: #000042;
	cursor: pointer;
	font-family: ${TypographyENUM.lkSansBold};
	font-style: normal;
	font-weight: 700;
	max-width: 233px;
	font-size: 16px;
	line-height: 24px;
	display: flex;
	align-items: center;
	justify-content: center;
	letter-spacing: -0.02em;
	outline: none;
	border: none;
	position: absolute;
    left: 50%;
    top: 50%;
    -webkit-transform: translate(-50%,-50%);
    transform: translate(-50%,-50%);
    margin: 0!important;
	&:disabled,
	&[disabled] {
		color: #9999B3;
		background: #DBDBEA;
	}
	span {
		display: inline-block;
		margin-right: 10px;
	}
`;

const QRCodeContainer = styled.div`
	position: relative;
	display: flex;
	align-items: center;
	justify-content: center;
`;
const QrImage = styled.div`
	display: block;
    width: 173px;
    margin: 20px;
    height: 200px;
    background: url(https://static.lenskart.com/media/desktop/img/QRBackground.png) no-repeat 50% 50%;
`;
const QRCodeForm = ({ isOpen, onSubmit, paymentMethod, gatewayId }: {isOpen: boolean, onSubmit: (...args: any) => void, paymentMethod: string, gatewayId: string}) => {
	return (
		<BankFormContainer isOpen={isOpen}>
			<BankForm>
				<p>Please click on Generate QR Code & Place Order</p>
				<QRCodeContainer>
					<QrImage />
					<PayNowBtn 
						onClick={() => onSubmit(paymentMethod, gatewayId)} 
						font={TypographyENUM.lkSansMedium}>Generate QR Code</PayNowBtn>
				</QRCodeContainer>
			</BankForm>
		</BankFormContainer>
	);
};

export { QRCodeForm };
