// @lk/ui-library/src/Component/V1/NewPayment/components/subcomponents/SelectField.tsx
import React from "react";
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

const CheckBox = styled.div<{disabled?: boolean;}>`
	display: inline-block;
	margin-right: 8px;
	height: 16px;
	width: 16px;
	border-radius: 4px;
	border: 1px solid var(--dark-blue-100);
	background: #F5F5FF;
	cursor: ${props => props.disabled ? "not-allowed" : "pointer"};
	position: relative;
	overflow: hidden;
`;
const MainContainer = styled.span`
    display: inline-block;
    font-family: ${TypographyENUM.lkSansRegular};
    font-style: normal;
    font-weight: 400;
    font-size: 14px;
    letter-spacing: -0.02em;
    color: var(--text);
    // margin-bottom: 24px;
    display: flex;
    align-items: center;
`;
const CheckMark = styled.div`
    display: block;
    border-width: 0 1px 1px 0;
    border-color: var(--dark-blue-100);
    border-style: solid;
    transform: rotate(45deg);
    width: 4px;
    height: 9px;
    position: absolute;
    top: 2px;
    left: 5px;
`;

const SelectField = ({ onClick, selected, disabled }: { onClick?: (e: any) => void, selected?: boolean, disabled?: boolean }) => {
	// const [showCheckMark, setShowCheckMark] = React.useState(!!defaultSelected);
	// const getUpdateOnWhatsapp = (event: React.MouseEvent<HTMLElement>) => {
	/*const onClickHandler = (e: any) => {
		setShowCheckMark(showcheck => !showcheck);
		onClick(e);
	};*/
	return (
		<MainContainer>
			<CheckBox onClick={onClick} disabled={disabled}>
				{selected && <CheckMark />}
			</CheckBox>
		</MainContainer>
	);
};

export { SelectField };
