// @lk/ui-library/src/Component/V1/MyOrderMobile/ReturnItemCard/ReturnItemCard.styles.tsx
import styled from "styled-components";

export const Container = styled.div<{ exchangeOrRefund?: boolean, border: boolean }>`
	max-width: 100%;
	border: ${props => props.border ? "1px solid #ddd" : "none"};
	margin: ${props => props.exchangeOrRefund ? "" : "8px"};
	padding: 10px;
	border-radius: 4px;
	font-family: var(--font-lksans-regular);
	background-color: #fff;
`;

export const ItemCard = styled.div<{ showDetails: boolean }>`
	display: ${props => props.showDetails ? "" : "grid"};
	grid-template-columns: ${props => props.showDetails ? "" : "42% 58%"};
`;

export const Media = styled.div`
	padding: 0 10px;
	img {
		max-width: 100%;
	}
`;

export const DetailsGrid = styled.div<{ showDetails: boolean }>`
	display: grid;
	grid-template-columns: ${props => props.showDetails ? "67% 33%" : ""};
	div {
		display: flex;
		&:first-child{
			align-items: flex-start;
			flex-direction: column;
		}
		&:last-child {
			align-items: center;
			justify-content: ${props => props.showDetails ? "flex-end" : "flex-start"};
		}
	}
`;

export const DataHeader = styled.div<{ showDetails: boolean }>`
	margin-bottom: ${props => props.showDetails ? "" : "5px"};
	div {
		&:first-child {
			color: #999999;
			font-size: 11px;
			letter-spacing: ${props => props.showDetails ? "" : "1px"};
			margin-bottom: ${props => props.showDetails ? "4px" : ""};
		}

		&:last-child {
			color: #333333;
			font-size: 13px;
		}
	}
`;

export const ProductFeatures = styled.div<{ showDetails: boolean }>`
	color: ${props => props.showDetails ? "#666666" : "#999999"};
	font-size: 11px;
	margin-top: ${props => props.showDetails ? "" : "5px"};
	text-transform: capitalize;
`;

export const AdditionalFeatures = styled.div`
	color: #999999;
	font-size: 11px;
	margin-top: 5px;
`;

export const ProductPrice = styled.div<{ showDetails: boolean, displayTable: boolean }>`
	color: #27394E;
	font-family: var(--font-lksans-bold);
	margin-top: ${props => props.showDetails ? "" : "5px;"};
	font-size: 15px;
	text-align: end;
    align-items: flex-end;
	div {
		font-size: 11px;
		color: #18CFA8;
		margin-top: 6px;
		display: flex;
		flex-direction: row;

		svg {
			margin-left: 5px;
			transform: ${props => props.displayTable ? "rotate(180deg)" : ""};
			transition: transform 0.5s ease-in-out;
		}
	}
`;

export const DetailsButton = styled.button<{ showDetails: boolean }>`
	font-family: var(--font-lksans-bold);
	border: none;
	background: transparent;
	color: #18CFA8;
	font-size: 11px;
	${props => props.showDetails ? "width: 100%; border-top: 1px solid #eee; padding-top: 10px; margin-top: 10px;" : ""};
`;

export const CardHeading = styled.h3`
	color: #333;
	font-size: 14px;
	font-weight: 700;
	line-height: 16px;
	letter-spacing: -.3px;
    -webkit-tap-highlight-color: transparent;
	margin: 0;
	padding: 10px 10px 0 10px;
`;
