// @lk/ui-library/src/Component/V1/Wishlist/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const WishlistWrapper = styled.div<{ show: boolean; }>`
    width: 18%;
    position: fixed;
    background: var(--white);
    left: 40%;
    bottom: -500px;
    border-radius: 4px 4px 0 0;
    transition: 0.7s all;
    z-index: 100;
    font-family: ${TypographyENUM.defaultBook};
	font-synthesis: none;

    ${({ show }) => show && `
        bottom: 0px;
        z-index: 999;
        animation: slide-up 0.7s ease;
    `}

    @keyframes slide-up {
        0% {
            bottom: -500px;
        }
        100% {
        bottom: 0;
        }
    }
`;

export const CloseWrapper = styled.div`
    color: var(--white);
    font-weight: 600;
    font-size: var(--fs-15);
    text-transform: var(--uppercase);
`;

export const Header = styled.div`
    background: var(--dark-gray);
    display:flex;
    justify-content:space-between;
    padding: 10px;
    text-align: center;
    cursor: var(--pointer);
    border-radius: 4px 4px 0 0;
`;

export const HeaderText = styled.div`
    color: var(--white);
    font-weight: 600;
    font-size: var(--fs-15);
    text-transform: var(--uppercase);
    display: flex;
    justify-content: center;
    width:100%;
	span{
		margin-left:5px;
	}
`;

export const NoProductsInfo = styled.p`
    padding: 40px 20px;
    text-align: center;
    margin: 0;
    box-shadow: 0 0 3px var(--medium-shadow);
    font-size: 13px;
    line-height: 1.42857;
    color: #333;
`;

export const ProductsContainer = styled.div`
    position: relative;
    overflow: auto;
    box-shadow: 0 0 3px var(--medium-shadow);
    max-height: 225px;
    overflow-y: auto;
`;

export const BrandName = styled.h5`
    margin-top: 0px;
    margin-bottom: 2px;
    font-family: 'LKFuturaStd-Medium';
    font-weight: 400!important;
`;

export const CrossIcon = styled.a`
    display: none;
`;

export const ItemContainer = styled.div<{isRTL: boolean}>`
    padding: 5px;
    position: relative;
    border-bottom:1px solid var(--light-gray);

    &:hover {
        background: var(--gold-10);
        cursor: var(--pointer);

        ${CrossIcon} {
            background: var(--blashish-gray);
            padding: 0px 5px 2px 5px;
            font-size: 12px;
            color: #fff;
            text-shadow: none;
            display: flex;
            align-items: center;
            justify-content: center;
            position: absolute;
            top: 5px;
		     ${props => props.isRTL ? "left: 5px;" : "right: 5px;"};
        }

        ${BrandName} {
            color: var(--blashish-gray);
        }
    }
`;

export const ProductContainer = styled.a`
    width: 100%;
    display: flex;
    align-items: center;
    gap: var(--pd-10);
`;

export const ProductDetailsContainer = styled.div`
    display: inline-block;
    width: 70%;
`;

export const ItemPrice = styled.span`
    color: var(--black);

    &.ml-5 {
        margin-left: 5px;
    }
`;

export const ItemImgContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    width: 30%;
    line-height: 35px;
`;

export const ProductImg = styled.img`
    max-width: 100%!important;
    width: auto\9;
    height: auto;
    vertical-align: middle;
    border: 0;
    -ms-interpolation-mode: bicubic;
`;

export const OutOfStockContainer = styled.div`
    margin-left:10px;
    color: var(--red);
`;

export const ClearButtonContainer = styled.div`
    background: var(--light-gray);
    overflow: hidden;
    padding: 15px;
    cursor: var(--pointer);
    box-shadow: 0 0 3px var(--medium-shadow);
    display: flex;
    align-items: center;
    justify-content: center;
`;

export const ClearButton = styled.a`
    background: var(--blashish-gray);
    padding: 3px 10px;
    font-size: var(--fs-14);
    color: var(--white);
    float: left;
    text-shadow: none;
    border-radius: 5px;
    text-transform: var(--uppercase);

    &:hover {
        text-decoration: underline;
    }
`;
