// @lk/ui-library/src/Component/V1/Wishlist/Wishlist.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import { CurrencySymbol, CurrencyType } from "../../../Types/general";
import Wishlist from "./index";

export default {
	title: "Components/V1/Wishlist",
	component: Wishlist,
	argTypes: {
		show: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show or hide the wishlist",
			category: "Component",
			defaultValue: true,
		}),
		setShow: generateArgTypes({
			type: "function",
			required: true,
			description: "Function to handle the value of show property",
			category: "Events"
		}),
		productList: generateArgTypes({
			type: "array",
			required: true,
			description: "List of products",
			category: "Component",
			defaultValue: [{
				id: "1",
				pid: 1,
				productUrl: "",
				productImageUrl: "",
				brandName: "",
				prices: [
					{ price: "" }
				]
			}],
		}),
		isLoading: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show or hide the loader",
			category: "Component",
			defaultValue: false,
		}),
		deleteProductFromWishlist: generateArgTypes({
			type: "function",
			required: true,
			description: "Delete product from wishlist",
			category: "Component",
			defaultValue: () => null,
		}),
		clearWishlist: generateArgTypes({
			type: "function",
			required: true,
			description: "Clear wishlist",
			category: "Component",
			defaultValue: () => null,
		}),
	}
} as ComponentMeta<typeof Wishlist>;

const Template: ComponentStory<typeof Wishlist> = (args) => <Wishlist {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	show: true,
	productList: [
		{
			id: "1",
			pid: 1,
			productUrl: "https://www.lenskart.com/vincent-chase-vc-e13784-c2-eyeglasses.html",
			productImageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
			brandName: "Lenskart",
			prices: {
				basePrice: 1999,
				lkPrice: 999,
				currency: CurrencyType.INR,
				symbol: CurrencySymbol.INR
			}
		},
		{
			id: "2",
			pid: 2,
			productUrl: "https://www.lenskart.com/vincent-chase-vc-e13784-c2-eyeglasses.html",
			productImageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
			brandName: "Lenskart",
			prices: {
				basePrice: 1999,
				lkPrice: 999,
				currency: CurrencyType.INR,
				symbol: CurrencySymbol.INR
			}
		},
		{
			id: "3",
			pid: 3,
			productUrl: "https://www.lenskart.com/vincent-chase-vc-e13784-c2-eyeglasses.html",
			productImageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
			brandName: "Lenskart",
			prices: {
				basePrice: 1999,
				lkPrice: 999,
				currency: CurrencyType.INR,
				symbol: CurrencySymbol.INR
			}
		},
		{
			id: "4",
			pid: 4,
			productUrl: "https://www.lenskart.com/vincent-chase-vc-e13784-c2-eyeglasses.html",
			productImageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
			brandName: "Lenskart",
			prices: {
				basePrice: 1999,
				lkPrice: 999,
				currency: CurrencyType.INR,
				symbol: CurrencySymbol.INR
			}
		},
		{
			id: "5",
			pid: 5,
			productUrl: "https://www.lenskart.com/vincent-chase-vc-e13784-c2-eyeglasses.html",
			productImageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
			brandName: "Lenskart",
			prices: {
				basePrice: 1999,
				lkPrice: 999,
				currency: CurrencyType.INR,
				symbol: CurrencySymbol.INR
			}
		},
		{
			id: "6",
			pid: 6,
			productUrl: "https://www.lenskart.com/vincent-chase-vc-e13784-c2-eyeglasses.html",
			productImageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
			brandName: "Lenskart",
			prices: {
				basePrice: 1999,
				lkPrice: 999,
				currency: CurrencyType.INR,
				symbol: CurrencySymbol.INR
			}
		}
	],
	dataLocale: {
		PRODUCTS: "Products",
		YOU_HAVE_NOT_SELECTED_ANY_PRODUCT: "You have not selected any products to compare.",
		PLEASE_ADD_PRODUCTS_OF_YOUR_CHOICE: "Please add products of your choice and view here.",
		CLEAR_LIST: "Clear list",
		CURRENCY_CODE: "CURRENCY_CODE"
	},
	setShow: () => {console.log("Show/jide wishlist");},
	deleteProductFromWishlist: (id:any) => {console.log("Delete specific item from wishlist" + id);},
	clearWishlist: () => {console.log("ClearAll Wishlist");},
};
