// @lk/ui-library/src/Component/V1/Checkout/styles.tsx
import styled, { css } from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const OuterContainer = styled.div`
  align-items: center;
`;

export const Select = styled.select`
  width: 100%;
  padding: 10px;
  border: none;
  outline: none;
`;

export const SelectOuter = styled.div`
  width: 400px;
  padding: 10px;
  border: 1px solid #c3c3c3;
  border-radius: 4px;
`;

export const ProductId = styled.div`
  color: #999999;
  font-size: 12px;
  line-height: 17.5px;
`;

export const BrandName = styled.div`
  font-weight: 500;
  font-size: 16px;
  margin-bottom: 7px;
  margin-top: 4px;
`;
export const TextTransform = styled.span`
  text-transform: capitalize;
`;
export const Price = styled.div`
  font-size: 14px;
  margin-top: 10px;
  color: #999999;
`;
export const DarkColorPrice = styled.div`
  color: #333;
  font-size: 14px;
  margin-top: 10px;
`;

export const Header = styled.div`
  margin-left: 12px;
  text-align: center;
`;
export const Texth4 = styled.h4`
  text-align: left;
  font-size: 20px;
  margin-top: 10px;
  margin-bottom: 10px;
  font-weight: 500;
`;

export const Body = styled.div`
  margin-left: 12px;
`;
export const ImgOuter = styled.div`
  margin-left: 12px;
  margin-top: 12px;
  display: flex;
  img {
    margin-top: 15px;
  }
`;

export const SelectTop = styled.div`
  display: flex;
  margin-left: 20px;
  justify-content: space-between;
  img {
    display: block;
    flex: 1;
  }
`;
export const Divmr = styled.div`
  margin-right: 5px;
`;
export const CardFooter = styled.div<{leftMargin?: boolean;}>`
  text-align: left;
  color: #999999;
  font-size: 14px;
  margin-top: 15px;
  ${props => props.leftMargin && css`margin-left: 20px;`}
`;
export const ButtonOut = styled.div`
  width: 100%;
  display: flex;
  justify-content: end;
`;

export const ButtonOutMargin = styled.div`
  margin-top: 20px;
  width: 100%;
  display: flex;
  justify-content: end;
`;

/////////////////////exchange double

export const Outer = styled.div`
  margin-top: 12px;
  display: flex;
  justify-content: space-between;
  // border: 1px solid #dddddd;
  border-radius: 10px;
  margin-left: 12px;
  margin-right: 12px;
  // padding: 10px;
`;

export const DoubleLeft = styled.div`
  padding-top: 15px;
  padding-bottom: 15px;
  margin-right: 10px;
  margin-left: 10px;
`;

///////////////////////////////////////////////////

export const LeftOuter = styled.div`
  width: 70vw;
  margin-right: 5px;
`;

export const LeftOuterMost = styled.div`
  border: 1px solid #dddddd;
  border-radius: 8px;
  margin: 10px;
`;

export const LeftRoot = styled.div`
  margin-bottom: 10px;
`;
export const HeaderLeft = styled.div<{ background: boolean }>`
  background-color: ${(props) =>props.background ? "rgba(0, 185, 198, 0.08)" : "none"};
  color: ${(props) => (props.background ? "balck" : "#999999")};
  padding: 15px;
  display: flex;
  justify-content: space-between;
  line-height: 1.43;
`;
export const HeaderText = styled.span`
  font-weight: 600;
  font-size: 15px;
`;
export const Divider = styled.div`
  display: flex;
  justify-content: space-between;
  margin-bottom: 10px;
`;
export const LeftDiv = styled.div`
  text-align: center;
  flex: 1;
  margin-left: 5px;
  margin-right: 5px;
  padding: 5px;
  margin-top: 15px;
`;

export const RightDiv = styled.div`
  margin-top: 15px;
  margin-right: 10px;
  margin-left: 25px;
  padding: 5px;
`;

export const InputOuter = styled.div`
  margin: 10px auto;
`;

///////////////////////////

export const Root = styled.div`
  width: 30vw;
  height: fit-content;
  margin-left: 5px;
  margin-top: 10px;
  padding: 5px;
  border: 1px solid #dddddd;
  border-radius: 8px;
`;

///////////////////ReturnProductDetails

export const Heading = styled.div<{ exchangewith: boolean }>`
  border: 1px solid #ebebeb;
  border-radius: 10px;
  padding: 3px;
  text-align: center;
  font-size: 13px;
  color: ${(props) => (props.exchangewith ? "green" : "")};
`;

export const RPDImgOuter = styled.div`
  margin: 12px auto;
`;
export const HeaderOuter = styled.div`
  display: inline-block;
`;
export const LKDivider = styled.div`
  width: 1px;
  background: #ebebeb;
  margin-top: 0px;
  margin-bottom: 0px;
  margin-right: 10px;
`;
export const DividerImg = styled.div`
  //   position: relative;
  //   left: -4%;
  display: flex;
  -webkit-box-pack: center;
  justify-content: center;
  -webkit-box-align: center;
  align-items: center;
`;
export const CardImg = styled.img`
  color: #fff;
  background: #00b9c6;
  border-radius: 100%;
`;

//////////Return add

export const ReturnBody = styled.div`
  padding: 9px;
`;

export const Addressbox = styled.div`
  border: 1px solid rgba(60, 60, 60, 0.54);
  padding: 13px;
  border-radius: 8px;
  margin-left: 30px;
  font-size: 12px;
`;

export const FirstRow = styled.div`
  display: flex;
  justify-content: space-between;
  color: rgba(60, 60, 60, 0.54);
  div:nth-of-type(2) {
    color: rgb(0, 185, 198);
    font-weight: 500;
    font-size: 14px;
    cursor: pointer;
  }
`;

//////////////Return mode////////////

export const BorderBottom = styled.div<{ isBottom: boolean }>`
  border-bottom: ${(props) => (props.isBottom ? "1px solid #ebebeb" : "")};
  padding-bottom: 10px;
`;

export const OuterInput = styled.div<{ disable: boolean }>`
  text-color: ${(props) => (props.disable ? "#999999" : "")};
  display: flex;
  align-items: center;
`;
export const Icon = styled.img<{ disable: boolean }>`
  margin: 8px;
  margin-left: 25px;
  max-width: 100%;
  cursor: ${(props) => (props.disable ? "no-drop" : "pointer")};
`;
export const Info = styled.div<{ disable: boolean }>`
  width: 100%;
  margin-top: 15px;
  cursor: ${(props) => (props.disable ? "no-drop" : "pointer")};
`;
export const ReturnType = styled.span`
  font-weight: 600;
  font-size: 14px;
  margin-left: 40px;
`;
export const Badge = styled.span`
  border-radius: 16px;
  color: #3b873e;
  background-color: #fff;
  border: 1px solid rgba(76, 175, 80, 0.5);
  font-family: Roboto;
  font-size: 12px;
  line-height: 18px;
  letter-spacing: 0.16px;
  padding: 0 6px;
  margin-left: 15px;
`;

export const List = styled.li`
  color: #999999;
  font-size: 12px;
  list-style-type: none;
  margin-left: 40px;
`;

//////////////////////////AddressTab

export const AddressTabWrapper = styled.div<{ styleFont?: TypographyENUM }>`
  font-family: ${(props) => props.styleFont};
  box-sizing: border-box;
  display: flex;
  align-items: center;
  padding: var(--pd-15) var(--pd-15);
  gap: var(--spacing-lg);
  background: var(--white);
`;

export const PersonName = styled.div`
  display: flex;
  width: 100%;
  gap: var(--pd-5);
  font-family: var(--font-default-medium);
  font-size: var(--fs-14);
  color: var(--black);
`;

export const LightText = styled.div`
  font-size: var(--fs-13);
  font-weight: var(--fw-regular);
  color: var(--gray);
  gap: var(--pd-5);
  display: flex;
`;

export const HR = styled.hr`
  border: none;
  height: 1px;
  margin: 0;
  background-color: rgba(0, 0, 0, 0.12);
`;

export const RadioInput = styled.input`
  cursor: pointer;
  user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const AddressBox = styled.div<{
  styledFont: TypographyENUM;
}>`
  width: 100%;
  padding: 16px;
  // cursor: pointer;
  transition: all 0.2s ease-in-out;
  margin-bottom: 16px;
  border-top: 1px solid rgba(0, 0, 0, 0.12);
  font-size: 12px;
  font-family: ${(props) => props.styledFont};
`;
export const AddressRow = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 16px;
  // transition: all 0.2s ease-in-out;
`;
export const AddressColumn = styled.div`
  display: flex;
  flex-direction: column;
  color: rgba(60, 60, 60, 0.54);
`;
export const AddressColumnFlex = styled.div`
  display: flex;
  flex-direction: column;
  color: rgba(60, 60, 60, 0.54);
  flex-direction: row;
  color: #999999;
`;

export const AddressHead = styled.h3<{ font?: TypographyENUM }>`
  font-weight: 700;
  line-height: 20px;
  ${(props) => props.font && `font-family: ${props.font};`}
  color: var(--black);
  letter-spacing: -0.02em;
`;
export const ButtonGroup = styled.div`
  display: flex;
  align-items: center;
  button {
    margin-left: 25px;
  }
`;
export const DottedRow = styled.div`
  border: 1px dashed #cecedf;
  height: 0px;
  margin-bottom: 16px;
`;
export const Checkbox = styled.div<{ isActive?: boolean }>`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0px;
  width: 24px;
  height: 24px;
  background-color: ${(props) =>props.isActive ? "var(--dark-blue-100)" : "transparent"};
  border-radius: 32px;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.38);
  &:after {
    display: ${(props) => (props.isActive ? "block" : "none")};
    content: "";
    width: 4px;
    height: 10px;
    border-width: 0 1px 1px 0;
    border-color: #fff;
    border-style: solid;
    transform: rotate(45deg) translate(-1px, -1px);
  }
`;
export const AddressButton = styled.button<{ font?: TypographyENUM }>`
  ${(props) => props.font && `font-family: ${props.font};`}
  border: none;
  outline: none;
  background-color: transparent;
  padding: 0;
  cursor: pointer;
`;
export const MainInfo = styled.div``;
export const OrderPageContainer = styled.div`
  min-height: 580px;
  max-width: 1188px;
  margin: 15px auto 0;
  padding: 0;
`;
export const AddressData = styled.div<{ font?: TypographyENUM }>`
  min-width: 0;
  ${(props) => props.font && `font-family: ${props.font};`}
  transition: all 0.2s ease-in-out;
`;
export const DeliveryEstimate = styled.div<{ font?: TypographyENUM }>`
  ${(props) => props.font && `font-family: ${props.font};`}
`;
export const ButtonOuter = styled.div`
  margin-top: 15px;
`;

//////////////////////Sidebar/////////////

export const Sidebar = styled.div<{ disabled: boolean }>`
  display: ${(props) => (props.disabled ? "block" : "none")};
  height: 100%;
  right: 0;
  width: 50%;
  // left: 0;
  background-color: #fff;
  // opacity: 0.6;
  position: fixed;
  z-index: 1000;
  top: 0;
  overflow: auto;
  border-left: 1px solid #ccc;
`;
export const BackdropContainer = styled.div<{ disabled: boolean }>`
  display: ${(props) => (props.disabled ? "block" : "none")};
  height: 100%;
  position: fixed;
  right: 0;
  width: 100%;
  background-color: #000;
  opacity: 0.6;
  z-index: 200;
  top: 0;
  overflow: auto;
  border-left: 1px solid #ccc;
`;

//////////////changebutton

export const ChangeButton = styled.span`
  color: #329c92;
  cursor: pointer;
  font-size: 12px;
`;

export const SummaryOuter = styled.div`
  margin-top: -15px;
  padding: 15px;
  margin-left: 15px;
  padding-top: 0;
  font-size: 13px;
`;
export const Lightcolor = styled.span`
  color: #999999;
`;
export const Div = styled.div`
`;
export const DivMar5 = styled.div`
  margin-top: 5px;
`;
