// @lk/ui-library/src/Component/Mobile/Cart/CartProduct/Subcomponents/PowerStatusMessage.tsx
import React, { memo } from "react";
import { DataType } from "../../../../../Types/general";
// import "./PowerStatusMessage.scss";
import { CartViewBasicMsg, CartViewMoreInfo, CartViewMsgContainer, CartViewPrimaryMsg, CartViewSecondaryMsg } from "./styles";
interface PowerStatusMsgType {
    message: string;
    isMobileView: boolean, 
    showUploadPrescModal?: () => void, 
    dataLocale: DataType, 
    powerValue: boolean
}
function PowerStatusMessage({ message, isMobileView, showUploadPrescModal, dataLocale, powerValue }: PowerStatusMsgType) {
	return (
		<>
			<CartViewMsgContainer className="cartView-msg-container">
				<CartViewPrimaryMsg className="cartView-primary-msg">
					{!powerValue ? (
						<>
							<CartViewBasicMsg className="cartView-basicMsg">{message}</CartViewBasicMsg>
							{typeof showUploadPrescModal === "function" && isMobileView ? (
								<>
				&#x2022;
									<CartViewMoreInfo className="cartView-moreInfo" onClick={showUploadPrescModal}>
										{dataLocale?.VIEW_PRESCRIPTION}
									</CartViewMoreInfo>
								</>
							) : null}
						</>
					) : (
						<>
							<span>
								{!isMobileView
									? dataLocale?.POWER_PRESCRIPTION_UPLOADED
									: dataLocale?.PRESCRIPTION_UPLOADED}
							</span>
							{typeof showUploadPrescModal === "function" && isMobileView && (
								<>
									{" "}
				&#x2022;
									<CartViewMoreInfo className="cartView-moreInfo" onClick={showUploadPrescModal}>
										{dataLocale?.VIEW_PRESCRIPTION}
									</CartViewMoreInfo>
								</>
							)}

							{typeof showUploadPrescModal === "function" && !isMobileView && (
								<>
									{" "}
				&#x2022;
									<CartViewMoreInfo className="cartView-moreInfo" onClick={showUploadPrescModal}>
										{dataLocale?.VIEW_DETAILS}
									</CartViewMoreInfo>
								</>
							)}
						</>
					)}
				</CartViewPrimaryMsg>

				<CartViewSecondaryMsg>
					{dataLocale?.NO_EXTRA_CHARGES_FOR_HIGH_POWER || "No extra charge for high power"}
				</CartViewSecondaryMsg>
			</CartViewMsgContainer>
		</>
	);
}

export default memo(PowerStatusMessage);
