// @lk/ui-library/src/Component/Mobile/Cart/CartProduct/Subcomponents/styles.tsx
import { TypographyENUM } from "../../../../../Types/general";
import styled, { css } from "styled-components";

export const PowerStatusMessage = styled.div`
    background-color: #eee;
    width: 95%;

`;
export const StatusIcon = styled.div`
width: 30px;
height: 30px;
`;
export const Icon = styled.div`
width: 30px;
height: 30px;
border-radius: 50%;
`;
export const Width100 = css`
width: 100%;
`;
export const Padding07 = css`
padding: 0px 7px;
`;
export const Padding08 = css`
padding: 0px 8px;
`;
export const NewStyleDetails = styled.div`
    color: #66668E;
    font-size: 14px;
    line-height: 20px;
    letter-spacing: -0.02em;
    font-family: ${TypographyENUM.lkSansRegular};
    text-transform: capitalize;
`;
export const CartViewMsgContainer = styled.div`
display: flex;
flex-direction: row;
align-items: center;
justify-content: space-between;
gap: 20px;
flex: 1;
`;
export const CartViewPrimaryMsg = styled.div`
font-style: normal;
font-size: 12px;
line-height: 20px;
letter-spacing: -0.02em;
color: #000042;
`;
export const CartViewBasicMsg = styled.span`
  font-family: ${TypographyENUM.lkSansRegular};
  font-weight: 400;
  margin-right: 3px;
`;
export const CartViewMoreInfo = styled.div`
  font-family: ${TypographyENUM.lkSansBold};
  /* font-weight: 700; */
  text-decoration: underline;
  cursor: pointer;
  // margin-left: 10px;
`;
export const CartViewSecondaryMsg = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 10px;
  line-height: 14px;
  text-align: right;
  letter-spacing: -0.02em;
  color: #333368;
`;
export const NameStrip = styled.div`
  background: #eee;
  margin-left: -15px;
  margin-right: -15px;
  margin-left: 0;
  margin-right: 0;
  border-bottom: none;
  padding-left: 0px;
  text-transform: capitalize;
  font-size: 14px;
  padding-bottom: 10px;
  padding-top: 10px;
`;
export const GridSpan = styled.span`
  font-weight: 700;
  display: inline-block;
  padding-bottom: 5px;
  margin-left: 15px;
  display:flex;
  justify-content: space-between;
  align-items: center;
  margin-right: 15px;
`;
export const PowerAccordionTitle = styled.div<{expanded?: boolean}>`
  color: #00BAC6;
  &:before {
    content: "Show Power";
    color: map-get($colorpallete, color_link_blue);
    margin-right: 5px;
    font-size: 12px;
  }
  span {
    font-size: 12px;
    content: '\e81e';
  }
  ${props => props.expanded ? `
    &:before {
      content: "Hide Power";
    }
    span:before {
      transform: rotate(-180deg);
    }
  ` : ""}
`;

////////// PowerGrid ///////
export const NewHeaderContainer = styled.div`
  margin: 10px 0;
`;

export const NewHeaderContent = styled.div`
display: flex;
flex-direction: row;
align-items: center;
justify-content: space-between;
font-size: 12px;
line-height: 18px;
letter-spacing: -0.02em;
color: #000042;
font-style: normal;
`;

export const HeaderTextPrimary = styled.span`
  font-family: ${TypographyENUM.lkSansRegular};
  font-weight: 400;
`;
export const HeaderTextSecondary = styled.span`
  font-family: ${TypographyENUM.lkSansBold};
  /* font-weight: 700; */
  text-decoration: underline;
`;

export const NewHeaderIcon = styled.span`
border: solid #000042;
border-width: 0 2px 2px 0;
display: inline-block;
padding: 3px;
margin-left: 5px;
`;

export const IconArrow = styled.span<{isActive?:boolean;}>`
  ${props => props.isActive ? `
    transform: rotate(-135deg);
    -webkit-transform: rotate(-135deg);
  ` : `
    transform: rotate(45deg);
    -webkit-transform: rotate(45deg);
  `}
  border: solid #000042;
  border-width: 0 2px 2px 0;
  display: inline-block;
  padding: 3px;
  margin-left: 5px;
`;

export const HeaderTextFix = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-weight: 400;
  font-size: 10px;
  line-height: 14px;
  color: #333368;
`;

export const CartViewBodyContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
  background: #ffffff;
  border: 1px solid #e2e2ee;
  border-radius: 8px;

  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  /* font-weight: 700; */
  font-size: 12px;
  line-height: 18px;
  letter-spacing: -0.02em;
  color: #000042;
`;

export const CartViewHeader = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  gap: 10px;
  background: #f5f5ff;
  padding: 12px;
  text-transform: uppercase;
`;

export const CartViewTable = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
  flex: 1;
  padding: 12px;
`;

export const TableRow = styled.div`
  display: flex;
  flex-direction: row;
  gap: 10px;
  align-items: center;
  justify-content: space-between;
`;

export const TableRowItem = styled.div`
  min-width: 30px;
  text-align: center;
`;

////////// PFUViewDetail ///////
export const PFUBottomBorder = styled.div`
border-bottom: 1px solid lightgrey;
margin-left: 15px;
margin-right: 15px;
	`;
export const AddPowerStatus = styled.div`
  background-color: #EEEEEE;
  background-color: #eee;
  width: 95%;
  width: 100%;
  justify-content: center;
  padding: 10px 0;
  position: relative;
  margin: auto;
  display: flex;
  justify-content: space-between;
  padding: 10px 10px 0 10px;
  margin: auto;
  border-radius: 4px;
  margin-bottom: 10px;
	`;
export const PFUStatusMessage = styled.div`
	font-size: 12px;
  margin-bottom: 10px;
	`;
export const BlueTextLink = styled.span`
	font-size: 12px;
  color: #00BAC6;
  text-transform: uppercase;
  cursor: pointer;
  font-weight: 400;
  margin-left: 10px;
	`;
export const PFUBottomHeadContainer = styled.div`
  height: 34px;
  display: flex;
  align-items: end;
  margin-bottom: -6px;
	`;
export const CrossButton = styled.span`
    /* font-weight: 400; */
    font-size: 14px;
    line-height: 24px;
    letter-spacing: -0.02em;
    color: #000042;
    margin-bottom: 15px;
    font-family: ${TypographyENUM.lkSansBold};
    font-size: 14px;
    padding: 5px;
    position: absolute;
    right: 20px;
        z-index: 100;
        top: 20px;
	`;
export const PFUHomePage = styled.div`
  width: 100%;
  height: 100%;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  display: flex;
`;

export const PFUDescription = styled.div`
  text-align: center;
  font-size: 17px;
  font-weight: 700;
  margin-top: 10px;
`;
