// @lk/ui-library/src/Component/Mobile/Cart/CartProduct/Subcomponents/ViewUploadImage.tsx
import React, { FC } from "react";
import styled from "styled-components";
// import config from '../../../../config';
// import PinchZoomComponent from "./PinchToZoomComp";

interface Props {
	showUploadPrescModal: () => void;
	imageFileName: string;
	altText: string;
}

const Wrapper = styled.section`
  position: fixed;
  top: 0;
  left: 0;
  z-index: 102;
  transition: all 1.5s ease-in-out;
  width: 100%;
  bottom: 0;
  background-color: grey;
  .pinch-zoom-container {
    display: flex;
    min-height: 100vh;
    max-height: 100vh;
    align-items: center;
    justify-content: center;
  }
`;

const CloseButton = styled.span`
    position: absolute;
    right: 4px;
    top: 4px;
    z-index: 1;
    color: white;
    padding-left: 10px;
    padding-right: 10px;
    padding-top: 5px;
    padding-bottom: 5px;
    font-size: 20px;
  `;

const PinchZoomComponentImg = styled.div`
	display: flex;
	align-items: center;
	justify-content: center;
	height: 100vh;
	padding:20px;
	img{
		max-width:100%;
	}
`;

// const PinchZoomContainer = styled.div`
//   display: flex;
//     min-height: 100vh;
//     max-height: 100vh;
//     align-items: center;
//     justify-content: center;
//   `;

// const PZoomBody = styled.div`
//   transform: scale(1, 1) translate(0, 0);
//     width: 100%;
//     height: 100vh;
//     padding: 0 20px;
//     display: flex;
//     align-items: center;

//     visibility: visible;
//     transform-origin: 0% 0%;
//     /* position: absolute; */
//     transform: scale(1, 1) translate(0px, 0px);

//     img {
//       width: 100%;
//     }
//   `;

const ViewUploadImage: FC<Props> = ({ showUploadPrescModal, imageFileName, altText }) => {
	return (
		<Wrapper data-testid="viewUploadImage">
			<CloseButton
				onClick={showUploadPrescModal}
			>
				&#10005;
			</CloseButton>
			{/* <PinchZoomContainer> */}
			<div>
				<PinchZoomComponentImg /*options={{ draggableUnzoomed: false }}*/>
					{/* <PZoomBody> */}
					<img
						alt={altText}
						src={`${process.env.NEXT_PUBLIC_API_URL}/v2/utility/customer/prescriptions/download/${imageFileName}`}
					/>
					{/* </PZoomBody> */}
				</PinchZoomComponentImg>
			</div>
			{/* </PinchZoomContainer> */}
		</Wrapper>
	);
};

export default ViewUploadImage;
