// @lk/ui-library/src/Component/V1/MyOrderMobile/ReturnItemCard/PowerTable/ReturnPowerTable.styles.tsx
import styled from "styled-components";

export const Container = styled.div<{ displayTable: boolean }>`
	background-color: #EEEEEE;
	border-radius: 5px;
	font-family: var(--font-lksans-regular);
	background-color: transparent;
	margin-top: ${props => props.displayTable ? "8px" : "0px"};
	padding: ${props => props.displayTable ? "8px" : "0px 8px"};
	padding-bottom: 0;
	overflow: hidden;
	height: ${props => props.displayTable ? "130px" : "0px"};
	transition: height ease-in-out 0.5s, padding ease-in-out 0.5s, margin-top ease-in-out 0.5s;
`;

export const TableContainer = styled.div`
	font-size: 12px;
	width: 100%;
`;

export const Table = styled.div`
	display: table;
	width: 100%;
`;

export const HeadRow = styled.div`
	text-transform: capitalize;
	width: 100%;
	display: table-row;
	background-color: #f2f2f2;
	color: #424242;
	font-family: var(--font-lksans-bold);
`;

export const Row = styled.div`
	display: table-row;
	width: 100%;
	color: #666666;
`;

export const PowerCell = styled.div`
	display: table-cell;
	width: 60%;
	padding: 4px 10px;
`;

export const Cell = styled.div`
	display: table-cell;
	width: 20%;
	text-align: center;
`;

export const BoldCell = styled.div`
	/* font-family: var(--font-lksans-bold); */
	text-align: center;
	display: table-cell;
`;
