// @lk/ui-library/src/Component/V1/CheckoutMobile/UPI/UPI.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../../StorybookHelper";
import React from "react";
import UPI from "./UPI.component";

export default{
	title: "Components/v1/CheckoutMobile/UPI",
	component: UPI,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: false,
			description: "String used as unique identifier",
			category: "General"
		}),
		groups: generateArgTypes({
			type: "string",
			required: true,
			description: "Array of objects of UPIs",
			category: "Component",
		}),
	}
} as ComponentMeta<typeof UPI>;

const Template: ComponentStory<typeof UPI> = (args) => <UPI {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	groups: [ 
		{ label: "Google Pay", logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", warning: "Experiencing low success rates", key: "upi_payu", code: "PU:UPI" },
		{ label: "Google Pay", logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", warning: "", key: "upi_payu", code: "PU:UPI" },
		{ label: "Google Pay", logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", warning: "", key: "upi_payu", code: "PU:UPI" },
		{ label: "Google Pay", logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", warning: "Experiencing low success rates", key: "upi_payu", code: "PU:UPI" },
		{ label: "Google Pay", logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png", warning: "Experiencing low success rates", key: "upi_payu", code: "PU:UPI" }
	],
	hasError: false,
	errorMessage: "This UPI ID doesn’t exist, please try again",
	dataLocale: {
		"MORE": "more",
		"ADD_NEW_ID": "Add new UPI ID",
		"UPI": "Upi",
		"YOUR_UPI_ID": "Your UPI ID",
		"VERIFT_AND_PAY": "Verify & Pay",
		"GOOGLE_PAY": "Google Pay",
		"MORE_UPI_APPS": "More UPI Apps",
		"INTENT_FLOW": "Intent Flow",
		"PAY_VIA_UPI_APPS": "Pay via UPI Apps",
		"PROCEED_TO_GET_INSTALLED_UPI_APPS": "Proceed to get a list of all installed UPI Apps",
		"PROCEED_TO_PAY": "Proceed to pay",
		"CHOOSE_OTHER_PAYMENT_OPTIONS": "Choose other payment options",
		"SELECT_APP_FROM_BELOW_LIST": "Please select an app from below list to make the payment.",
		"NOTE": "Note",
		"DONT_CLOSE_TILL_TRANSACTION_COMPLETE": "Do not close this screen until the transaction is complete",
		"CANCEL_PAYMENT": "Cancel payment",
		"AWAITING_PAYMENT": "Awaiting payment",
		"APPROVE_BEFORE_TIMER_RUNS_OUT": "Please approve the payment on your UPI app before the timer runs out",
		"MINS_REMAINING": "Mins remaining",
		"DONT_CLOSE_UNTIL_TRANSACTION_COMPLETE": "Do not close this screen until the transaction is complete",
		"YES_CANCEL":"Yes, cancel",
		"NO":"No",
		"PAYMENT_UNSUCCESSFUL_RETRY_WITH_OTHER_METHOD":"Your payment was unsuccessful, retry with another method.",
		"UNSUCCESSFUL_DESCRIPTION":"Any amount debited from your account, will be credited back within 2-3 working days",
		"RETRY_PAYMENT": "Retry with other payment methods",
		"SURE_YOU_WANT_TO_CANCEL": "Are you sure you want to cancel payment?",
		"CANCELLATION_TEXT": "Cancellation will be successful only if you are yet to complete this payment on your UPI app"
	},
	// eslint-disable-next-line @typescript-eslint/no-unused-vars, react/display-name
	getSubmitFunction: (e) => (_e,_ev) => null,
	paymentData: {
		"order": {
			"id": 1930505304,
			"legacyEntityId": 51224786,
			"cartId": 501604918,
			"storeId": 1,
			"customerId": 51636757,
			"customerEmail": "7985334941@lenskartomni.com",
			"customerWalletId": "3050000012050495",
			"type": "NORMAL",
			"childOrderIds": [],
			"quantity": 1,
			"itemCount": 1,
			"status": {
				"status": "PENDING",
				"state": "NEW",
				"returnable": false,
				"cancellable": false,
				"comment": "Order placed using payuwallet",
				"timestamp": 1676480561748,
				"trackingStatus": "PAYMENT_NOT_INITIATED",
				"cancellableStatus": "NON_CANCELLABLE",
				"returnableStatus": "NON_RETURNABLE",
				"orderTrackingStatusCheckpoint": "ORDER_PLACED"
			},
			"statusHistory": [
				{
					"status": "PENDING",
					"state": "NEW",
					"returnable": false,
					"cancellable": false,
					"comment": "Order placed using payuwallet",
					"timestamp": 1676480561748,
					"trackingStatus": "PAYMENT_NOT_INITIATED",
					"cancellableStatus": "NON_CANCELLABLE",
					"returnableStatus": "NON_RETURNABLE",
					"orderTrackingStatusCheckpoint": "ORDER_PLACED"
				}
			],
			"items": [
				{
					"id": 590520,
					"productId": 147098,
					"productType": "Eyeglasses",
					"relatedItems": [],
					"sku": "eye:lenskart-air-la-e13852-c2-eyeglasses",
					"quantity": 1,
					"availableQuantity": 359867,
					"quantityReturnable": 0,
					"powerRequired": "POWER_NOT_REQUIRED",
					"lensCategory": "EYEGLASS",
					"lensType": "FRAME_ONLY",
					"prescription": {
						"showPd": false
					},
					"options": [],
					"catalogPrices": [
						{
							"currencyCode": "INR",
							"value": 1700,
							"name": "Market Price"
						},
						{
							"currencyCode": "INR",
							"value": 1700,
							"name": "Lenskart Price"
						}
					],
					"price": {
						"currencyCode": "INR",
						"value": 1700,
						"name": "Gold VIP Price"
					},
					"framePrice": {
						"currencyCode": "INR",
						"value": 1700
					},
					"freeFrame": false,
					"gender": "Unisex",
					"productTypeValue": "Eyeglasses",
					"amount": {
						"currencyCode": "INR",
						"discounts": [],
						"totalDiscount": 0,
						"taxes": [
							{
								"name": "VAT",
								"amount": 306
							}
						],
						"totalTax": 306,
						"shipping": 0,
						"subTotal": 1700,
						"total": 2006
					},
					"appliedRules": [],
					"coupons": [],
					"inventory": {
						"deliveryStoreId": 0,
						"posStoreInventoryId": 0
					},
					"brandName": "Lenskart Air",
					"brandName_en": "Lenskart Air",
					"name": "Gunmetal Grey Half Rim Rectangle Lenskart Air Classic LA E13852-C2 Eyeglasses",
					"name_en": "Gunmetal Grey Half Rim Rectangle Lenskart Air Classic LA E13852-C2 Eyeglasses",
					"modelName": "LA E13852",
					"productUrl": "https://www.lenskart.com/lenskart-air-la-e13852-c2-eyeglasses.html",
					"thumbnail": "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e13852-c2-eyeglasses_G_2552.jpg",
					"classificationId": "11355",
					"frameColour": "Gunmetal",
					"frameColour_en": "Gunmetal",
					"status": {
						"state": "NEW",
						"status": "PENDING",
						"returnableStatus": "NON_RETURNABLE",
						"timestamp": 1676480561748,
						"returnable": false
					},
					"statusHistory": [
						{
							"state": "NEW",
							"status": "PENDING",
							"returnableStatus": "NON_RETURNABLE",
							"timestamp": 1676480561748,
							"returnable": false
						}
					],
					"createdAt": 1676480561748,
					"updatedAt": 1676480561748,
					"frameType": "half_rim",
					"classification": "eyeframe",
					"frameShape": "Rectangle",
					"frameSize": "Medium",
					"frameWidth": "136 mm",
					"subcategory": "Eyeglasses-Brands",
					"shipToStoreRequired": false,
					"isLocalFittingRequired": false,
					"productDeliveryType": "B2B",
					"reviews": [],
					"itemTracking": {
						"itemTrackingStatus": "ORDER_PLACED",
						"itemTrackingStatusCheckpoint": "ORDER_PLACED",
						"status": "ORDER_PLACED",
						"statusLabel": "Order Created",
						"histories": [
							{
								"status": "ORDER_PLACED",
								"statusLabel": "Order Created",
								"createdAt": 1676480561748,
								"updatedAt": 1676480561748,
								"current": true,
								"isVisible": true,
								"iconUrl": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/OrderCreated.png",
								"subTitle": "Payment Pending"
							},
							{
								"status": "IN_WAREHOUSE",
								"statusLabel": "Preparing for Dispatch",
								"current": false,
								"isVisible": true,
								"iconUrl": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Processing.png",
								"description": "Check out how your product is prepared",
								"descriptionLink": "MR-ZnDoZx1A",
								"descriptionImageLink": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Group_17.png"
							},
							{
								"status": "IN_TRANSIT",
								"statusLabel": "On the Way",
								"current": false,
								"isVisible": true,
								"iconUrl": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/OnTheWay.png"
							},
							{
								"status": "DELIVERED_TO_CUSTOMER",
								"statusLabel": "Delivered",
								"current": false,
								"isVisible": true,
								"iconUrl": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/DeliveredToStore.png"
							},
							{
								"status": "CANCELLED",
								"statusLabel": "Cancelled",
								"current": false,
								"isVisible": false,
								"iconUrl": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Cancelled.png"
							},
							{
								"status": "RETURN_TO_ORIGIN",
								"statusLabel": "Returned to Warehouse",
								"current": false,
								"isVisible": false,
								"iconUrl": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png"
							},
							{
								"status": "RETURN_REQUESTED",
								"statusLabel": "Return Requested",
								"current": false,
								"isVisible": false,
								"iconUrl": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png"
							},
							{
								"status": "RETURN_RESHIPPED",
								"statusLabel": "Returned Product Shipped Back",
								"current": false,
								"isVisible": false,
								"iconUrl": "https://static1.lenskart.com/media/desktop/img/Sep20/ProductShippedBacktoCustomer.png"
							},
							{
								"status": "RETURNED",
								"statusLabel": "Returned",
								"current": false,
								"isVisible": false,
								"iconUrl": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png"
							},
							{
								"status": "RETURN_CANCELLED",
								"statusLabel": "Return Cancelled ",
								"current": false,
								"isVisible": false,
								"iconUrl": "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/RetuenCancelled.png"
							},
							{
								"status": "RETURN_REFUNDED",
								"statusLabel": "Refund Completed",
								"current": false,
								"isVisible": false,
								"iconUrl": "https://static1.lenskart.com/media/desktop/img/Sep20/Refund.png"
							},
							{
								"status": "RETURN_RECEIVED",
								"statusLabel": "Return Received",
								"current": false,
								"isVisible": false,
								"iconUrl": "https://static1.lenskart.com/media/desktop/img/Sep20/ReturnReceived.png"
							},
							{
								"status": "EXCHANGE_CREATED",
								"statusLabel": "Exchange Order Created",
								"current": false,
								"isVisible": false,
								"iconUrl": "https://static1.lenskart.com/media/desktop/img/Sep20/ExchangeOrderCreated.png"
							}
						]
					},
					"image": "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e13852-c2-eyeglasses_G_2551.jpg",
					"isSplitOrderItem": true,
					"isBogoItem": false,
					"merchantId": "DEALSKART",
					"sellerLabel": "Sold by Dealskart Online Services Pvt Ltd.",
					"jit": true,
					"flags": {
						"canUpdatePrescription": false,
						"showTrackingHistory": true
					},
					"deliveryType": "FR0_WAREHOUSE_TO_CUSTOMER",
					"prescriptionView": {
						"showPd": false,
						"pdConfigAvailable": false
					},
					"hubCode": "LKST607",
					"tierMembershipName": "",
					"hsnCode": "90031900",
					"activeInventories": [
						{
							"id": "62fb6f0e5042464143b1a630",
							"productId": 147098,
							"facilityCode": "LKST607",
							"quantity": 359867,
							"priority": 1
						},
						{
							"productId": 147098,
							"storeId": "LKH03",
							"facilityCode": "LKH03",
							"quantity": 7784,
							"priority": 3
						}
					],
					"isExpressDelivery": false,
					"threeOrFree": false,
					"originalDeliveryDate": 1676937600000
				}
			],
			"appliedRules": [],
			"amount": {
				"currencyCode": "INR",
				"discounts": [],
				"totalDiscount": 0,
				"taxes": [
					{
						"name": "VAT",
						"amount": 306
					}
				],
				"totalTax": 306,
				"shipping": 0,
				"subTotal": 1700,
				"total": 2006
			},
			"payments": {
				"paymentList": [
					{
						"method": "payuwallet",
						"gateway": "PU:UPI"
					}
				],
				"totalPrepaid": 2006,
				"totalCOD": 0
			},
			"billingAddress": {
				"id": 41010602,
				"firstName": "sa",
				"lastName": "sa",
				"phone": "7985334941",
				"phoneCode": "+91",
				"email": "satyam.dua@lenskart.in",
				"addressType": "billing",
				"addressline1": "sa",
				"addressline2": "sa",
				"city": "Gurgaon",
				"floor": 0,
				"liftAvailable": false,
				"state": "Haryana",
				"postcode": "122004",
				"country": "IN",
				"gender": "male"
			},
			"shippingAddress": {
				"id": 41010601,
				"firstName": "sa",
				"lastName": "sa",
				"phone": "7985334941",
				"phoneCode": "+91",
				"email": "satyam.dua@lenskart.in",
				"addressType": "shipping",
				"addressline1": "sa",
				"addressline2": "sa",
				"city": "Gurgaon",
				"floor": 0,
				"liftAvailable": false,
				"state": "Haryana",
				"postcode": "122004",
				"country": "IN",
				"gender": "male"
			},
			"otpState": "OTP_NOT_REQUIRED",
			"guest": false,
			"offer3OrFree": "00000000",
			"createdAt": 1676480561748,
			"updatedAt": 1676480561748,
			"deliveryDate": 1676937600000,
			"dispatchDate": 1676505600000,
			"etaGenerated": true,
			"trackingDetails": [
				{
					"createdTime": 1676480561748,
					"updatedTime": 1676480561748,
					"trackStatus": "ORDER_PLACED",
					"status": "ORDER_PLACED",
					"details": [
						{
							"time": "15 Feb 10:32 PM",
							"trackStatus": "ORDER_PLACED"
						}
					]
				}
			],
			"deliveryOption": "STANDARD",
			"version": 0,
			"isLoyalty": false,
			"facilityCode": "0",
			"isDualCompanyEnabled": true,
			"isExchangeEnabled": false,
			"isBulkOrder": false,
			"shipToStoreRequired": false,
			"isLocalFittingRequired": false,
			"isPaymentCaptured": false,
			"transactionId": "",
			"merchantId": "DEALSKART",
			"lkCountry": "IN",
			"metadata": {
				"apiClient": "MOBILESITE"
			},
			"isMarketPlace": false,
			"customerPhone": "7985334941",
			"phoneCode": "+91",
			"isVirtualOrder": false,
			"flags": {
				"itemGroupingEnabled": true,
				"ORDER_PLACED_NOTIFICATION_SENT": true,
				"canUpdateAddress": true
			},
			"studioFlow": false,
			"isDigitalCart": false,
			"gokwikRtoDetails": {
				"requestId": "a352d2a2-f287-441e-9739-ee842f69ceb9",
				"riskFlag": "High Risk",
				"reason": "Address Incomplete",
				"score": 1
			},
			"payLater": false
		},
		"payment": {
			"paymentId": "1930505304upi70440834",
			"orderId": "1930505304",
			"amount": {
				"currencyCode": "INR",
				"price": 2006,
				"discounts": []
			},
			"actionInfo": {
				"action": "DEEPLINK",
				"redirectUrl": "upi://pay?pa=lenskart.payu@hdfcbank&pn=DealsKart Online Services Private Limited&tr=16804952994&am=2006.0",
				"requestParams": {
					"country": "IN",
					"s2s_device_info": "",
					"amount": "2006.00",
					"curl": "https://api-gateway.juno.preprod.lenskart.com/juno-payment-redirect/v2/payment-response/PU/1930505304upi70440834/cancelled",
					"firstname": "sa",
					"txn_s2s_flow": "2",
					"city": "Gurgaon",
					"address1": "sa",
					"surl": "https://api-gateway.juno.preprod.lenskart.com/juno-payment-redirect/v2/payment-response/PU/1930505304upi70440834/success",
					"udf2": "mobilesite",
					"lastname": "sa",
					"zipcode": "122004",
					"user_credentials": "uQXimM:7985334941@lenskartomni.com",
					"phone": "7985334941",
					"pg": "UPI",
					"furl": "https://api-gateway.juno.preprod.lenskart.com/juno-payment-redirect/v2/payment-response/PU/1930505304upi70440834/failure",
					"state": "Haryana",
					"productinfo": "Product Information",
					"s2s_client_ip": "",
					"bankcode": "INTENT",
					"email": "satyam.dua@lenskart.in",
					"hash": "2c868a622cfba5fdf0223c0203e06c62236459f0af71e1bc97c2ae2247bc88b230f429c07bb6bc894ac9e11d6e3d10c91020b217ad2baad327cb18e9b3e834c8",
					"key": "uQXimM",
					"txnid": "1930505304upi70440834"
				}
			},
			"upiFlowType": "INTENT",
			"payuPaymentMode": "INTENT"
		}
	},
	configData: {
		CART_UPI_CONFIG: "{ \"responseRedirection\": \"5000\", \"BottomSheetErrMsgDelay\": \"2000\", \"timerLoaderDelay\": \"300\", \"upiPaymentStatusRequest\": \"5000\", \"INTENT\": { \"android\": \"ON\", \"ios\": \"OFF\" }, \"COLLECT\": { \"android\": \"ON\", \"ios\": \"ON\" }, \"gPay\": { \"android\": \"ON\", \"ios\": \"ON\" } }"
	}
};
