// @lk/ui-library/src/Component/V1/CheckoutMobile/UPI/UPI.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const UPISection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  padding: var(--fs-16);
  background: var(--white);
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
  border-radius: 12px;
`;

export const UPISubSection = styled.div`
  text-align: center;
//   width: 25%;
     display: flex;
     gap: 15px;
	 align-items: center;
	 margin-bottom:5px;
  i{
	display:flex;
    img{
      max-width: 58px;
      height: 40px;
      display: inline-block;
    }
  }
`;

export const UPILabel = styled.span`
  font-family: ${TypographyENUM.lkSansRegular};
  font-weight: var(--fw-regular);
  font-size: 14px;
  line-height: 18px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
  display: block;
  padding-top: 0px;
`;

export const AddUpiSection = styled.div`
  padding: 16px 20px;
`;

export const UPIMore = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-bold);
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  text-decoration-line: underline;
  color: var(--dark-blue-100);
  padding-top:10px;
`;

export const UPISectionFirst = styled.div`
  display:flex;
  flex-direction:column;
  justify-content: flex-start;
  gap: 4px;
`;

export const BorderLine = styled.div`
  border: 1px dashed #E2E2EE;
  margin-top: 10px;
`;

export const AddUpiHeader = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-16);
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;

export const ErrorMessage = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #E34934;
  padding:10px;
`;

export const InputField = styled.div`
  margin:25px 0;
`;

export const AddNewId = styled.div`
width:100%;
padding-top: var(--fs-16);
  li{
    i{
    img{
      width: 58px;
      height: 40px;
    }
  }

    display:flex;
    align-items:center;
    justify-content:space-between;

    i:nth-child(2){
      font-style: normal;
      margin-right:auto;
      margin-left:15px;
    //   margin-top: -6px;
    }
  }

  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 14px;
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--text);
`;

export const UPIMoreSection = styled.div`
h2{
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size:  var(--fs-16);
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  margin-bottom:25px;
}
width:100%;
padding:  var(--fs-16);
  li{
    i{
    img{
      width: 46px;
      height: 32px;
    }
    padding: 16px 0;

    div{
      font-weight: var(--fw-regular);
      font-size: 12px;
      line-height: 20px;
      letter-spacing: -0.02em;
      color: #E34934;
    }
  }

    display:flex;
    align-items:center;
    justify-content:space-between;

    i:nth-child(2){
      margin-right:auto;
      margin-left:15px;
    //   margin-top: -6px;
    }
  }

  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 14px;
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--text);
`;

export const IntentLogoContainer = styled.div`
  padding-top: 25px;
  display: flex;
  align-items: center;
  justify-content: center;

  .intent-upi-icon {
    width: 108px;
    height: 30px;
  }
`;

export const IntentBottomSheetHeading = styled.div`
  text-align: center;
  font-size: var(--fs-16);
  font-family: var(--font-lksans-regular);
  font-weight: 700;
  margin-top: 20px;
  margin-bottom: 10px;
  `;

export const IntentBottomSheetSubHeading = styled.div`
  text-align: center;
  font-size: var(--fs-14);
  font-family: var(--font-lksans-regular);
  font-weight: 400;
  margin-bottom: 15px;
  line-height: 20px;
  `;

export const ProceedButtonContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  text-align: center;

  button {
    color: var(--white);
    font-size: var(--fs-15);
    margin-bottom: 20px;
    font-weight: 500;
    line-height: 40px;
    padding: 0;
    min-width: 236px;
  }
`;

export const ChooseOtherPaymentContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;

  button {
    width: fit-content;
    font-family: var(--font-lksans-bold);
    text-transform: uppercase;
    text-decoration: none;
    color: var(--topaz);
    font-size: var(--fs-12);
    /* font-weight: 500; */
  }
`;
