// @lk/ui-library/src/Component/Common/RatingStar/RatingStar.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../StorybookHelper";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";
import { RatingStar } from "./RatingStar.component";
import SvgAddedToCart from "../../../Icons/AddedToCart";

export default {
	title: "Components/Common/RatingStar",
	component: RatingStar,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		componentSize: generateArgTypes({
			type: "string",
			required: false,
			description: "Size of the component",
			category: "General",
			defaultValue: ComponentSizeENUM.small,
			control: "radio",
			options: ComponentSizeENUM,
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "radio",
			options: TypographyENUM,
		}),
		rating: generateArgTypes({
			type: "number",
			required: true,
			description: "Rating count",
			category: "Component",
		}),
		backgroundColor: generateArgTypes({
			type: "string",
			required: true,
			description: "background color",
			category: "Component",
		}),
		numberOfRatings: generateArgTypes({
			type: "number",
			required: true,
			description: "number of ratings",
			category: "Component",
		}),
		primaryTextColor: generateArgTypes({
			type: "string",
			required: true,
			description: "primary text color",
			category: "Component",
		}),
		secondaryTextColor: generateArgTypes({
			type: "string",
			required: true,
			description: "secondary text color",
			category: "Component",
		}),
		starColor: generateArgTypes({
			type: "string",
			required: true,
			description: "star icon color",
			category: "Component",
		}),
		icon: generateArgTypes({
			type: "string",
			required: false,
			description: "Rating Icon",
			category: "Component",
			control: "select",
			defaultValue: "starFilled",
			options: SvgAddedToCart,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show rating right to left",
			category: "",
		}),
	},
} as ComponentMeta<typeof RatingStar>;

const Template: ComponentStory<typeof RatingStar> = (args) => (
	<RatingStar {...args} />
);

export const Overview = Template.bind({});
Overview.args = {
	id: "numbered-star",
	rating: 4.5,
	componentSize: ComponentSizeENUM.medium,
	icon: SvgAddedToCart,
	font: TypographyENUM.serif,
	backgroundColor:"var(--warm-grey-25)",
	numberOfRatings:3,
	primaryTextColor:"var(--dark-blue-100)",
	starColor:"var(--turquoise)",
	secondaryTextColor:"var(--dark-blue-75)",
	isRTL: false,
};
