// @lk/ui-library/src/Component/Common/Try On Button/styles.tsx
import styled from "styled-components";

export const TryOnWrapper = styled.button<{
	width:string
	height:string
	borderColor?:string
	backgroundColor?: string
}>`
	border: 1px solid #dbdbdb;
	border-color: ${props => props.borderColor || "#dbdbdb"};
	border-radius: var(--border-radius-xxxs);
	padding: 0px var(--pd-25) 0px var(--pd-12);
	display: flex;
	align-items: center;
	justify-content:center;
	font-size: var(--fs-13);
	position:relative;
	cursor: var(--pointer);
	font-weight:600;
	background-color:var(--transparent);
	width: ${(props)=> props.width};
	height: ${(props)=> props.height};
	min-width: 120px;
	background-color: ${props =>props.backgroundColor};
	font-family: 'LKFuturaStd-Medium';
	span{
	   margin: auto;
       width: 100%;
	}
	div{
		margin-left: auto;
	}
	@media screen and (max-width:767px){
		padding: 0px 0 0px var(--pd-12);
		width: 135px;
		height: 28px;
		min-width: 100px;
		font-size: var(--fs-12);
		font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
		font-weight: 400;
        letter-spacing: 1px;
	}
`;

export const TryOnImage = styled.img<{
	imagePosition:string,
	isRTL: boolean
}>`
	height: 120%;
	// position: absolute;
	// right: ${props => props.isRTL ? "" : "30px"};
	// left: ${props => props.isRTL ? "230px" : ""};
	//bottom:0;
	margin-top: -2px;
	@media screen and (max-width:767px){
		width:40px;
		margin-top: -10px;
	}
`;
