// @lk/ui-library/src/Component/V1/PDP/Mobile/ReviewCard/ReviewCardMobile.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const ReviewCardWrapper = styled.div<{font:TypographyENUM}>`
	display: flex;
	flex-direction: column;
	gap: 10px;
	font-family: ${props => props.font};
`;

export const ReviewCardTopContent = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	gap: 10px;
	& .review-date{
		margin-left:auto;
		font-size:12px;
	}
`;

export const ReviewCardBottomContent = styled.div`
	display: flex;
	flex-direction: column;
	gap: 10px;
`;

export const NameTag = styled.div`
	width: 45px;
	height: 45px;
	border-radius: var(--border-radius-xl);
	background-color: #82b5f4;
	color: var(--white);
	display: flex;
	align-items: center;
	justify-content: center;
`;

export const SideContent = styled.div`
	display: flex;
	flex-direction: column;
	gap: 5px;
`;

export const Text = styled.div<{weight?:string, color?:string,font?:string,primaryFont: TypographyENUM,
	secondaryFont: TypographyENUM}>`
	font-weight: ${props => props.weight === "primary" ? "var(--fw-bold)" : props.weight === "secondary" ? "var(--fw-medium)" : "var(--fw-light)" };
	color: ${props => props.color === "primary" ? "#2f84ed" : "#6B6D76"};
	font-family: ${props => props.font === "primary" ? props.primaryFont : props.font === "secondary" ? props.secondaryFont : "inherit" };
`;
