// @lk/ui-library/src/Component/V1/PDP/Mobile/ReviewCard/ReviewCardMobile.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { ComponentSizeENUM, TypographyENUM } from "../../../../../Types/general";
import ReviewCardMobile from "./ReviewCardMobile.component";

export default {
	title: "Components/V1/PDP/mobile/ReviewCardMobile",
	component: ReviewCardMobile,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		componentSize: generateArgTypes({
			type: "string",
			required: false,
			description: "Size of the component",
			category: "General",
			defaultValue: ComponentSizeENUM.small,
			control: "radio",
			options: ComponentSizeENUM,
		}),
		isReadOnly: generateArgTypes({
			type: "boolean",
			required: true,
			description: "read only or not",
			category: "General",
			defaultValue: "false",
		}),
		onmouseClick: generateArgTypes({
			type: "function",
			required: false,
			description: "on Mouse click handler for rating",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show rating right to left",
			category: "General",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,

		})
	}
} as ComponentMeta<typeof ReviewCardMobile>;

const Template: ComponentStory<typeof ReviewCardMobile> = args => <ReviewCardMobile {...args}/>;
export const Overview = Template.bind({});
Overview.args = {
	id:"star-rating",
	rating:4,
	maxRating:5,
	componentSize: ComponentSizeENUM.medium,
	isReadOnly: true,
	reviewTitle: "This Product is awesome ...!",
	reviewDetail: "this product is good, size may vary!"
};
