// @lk/ui-library/src/Component/V1/PDP/Mobile/ProductDetails/ProductDetailsMobile.styles.tsx
import styled from "styled-components";

export const ProductDetailsMobileWrapper = styled.div`
	display: flex;
	flex-direction: column;
	gap: 5px;
`;

export const ExchangePriceWrapper = styled.div`

`;

export const StrikeText = styled.div`

`;

export const ExchangePrice = styled.div`

`;

export const ExchangeInfoText = styled.div`
	border: 1px solid #18cfa8;
    padding: 2px;
    background: rgba(24,207,168,.1);
	color: #18cfa8;
	font-size: 9px;
	border-radius: 4px;
	font-weight: 500;
	width: fit-content;
`;

export const ProductTitleRow = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
`;

export const ProductSecondaryRow = styled.div`
	display: flex;
	flex-direction: column;
	gap: 5px;
`;

export const ProductDetailsRow = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
`;

export const LeftSection = styled.div`

`;

export const RightSection = styled.div`
`;
export const ProductDetailItem = styled.div<{gap?:boolean}>`
	display: flex;
	align-items: center;
	${({ gap })=> gap ? "gap: 4px":""}
`;

export const SizeWidthWrapper = styled.span`
	background:#bcf7e3;
	border-radius: 25px;
	display:flex;
	font-family: roboto slab, sans-serif, arial, helvetica, sans-serif;
	align-items:center;
	padding:3px 7px;
	margin:-5px 4px -5px 4px;
	color:#000042;
	gap:4px;
	span{
		display:flex;
		path{
         stroke:#27394e;
		}
	}
`;

export const InlineWrapper = styled.div`
	display: inline-flex;
	&:empty{
		display:none;
	}
	& #product-rating{
		margin-top:27px;
	}
`;

export const ButtonsContainer = styled.div`
	display: flex;
	flex-direction: column;
	gap: 10px;
`;

export const TextH1 = styled.h1<{type?:string, color?: string}>`
	font-size: ${props => props.type === "primary" ? "14px" : props.type === "secondary" ? "14px" : "10px"};
	color: ${props => props.color || "#27394e "};
	font-family: roboto slab, sans-serif, arial, helvetica, sans-serif;
	font-weight: ${props => props.type === "primary" ? "500" : "300"};
`;

export const Text = styled.h2<{type?:string, color?: string}>`
	font-size: ${props => props.type === "primary" ? "18px" : props.type === "secondary" ? "12px" : "12px"};
	color: ${props => props.color || "#27394e"};
	font-family: roboto, sans-serif, arial, helvetica, sans-serif;
	font-weight: ${props => props.type === "primary" ? "500" : "400"};
	&.boughtText{
		font-size: ${props => props.type === "primary" ? "18px" : props.type === "secondary" ? "15px" : "12px"};
	}
`;

export const Img = styled.img`
	height: 22px;
`;
