// @lk/ui-library/src/Component/V1/StudioFlow/MsiteUserDetailsForm/MsiteUserDetailsForm.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";
import { FieldGroup } from "../UserDetailsForm/UserDetailsForm.styles";

export const FieldContainer = styled.div`
  padding-bottom: var(--pd-12);
`;

export const MsiteFieldGroup = styled(FieldGroup)`
  padding-top: var(--pd-16);
`;

export const CLCheckboxContainer = styled.div`
  display: flex;
  align-items: center;
  gap: 16px;
`;

export const CLCheckboxIcon = styled.div<{ isChecked?: boolean; }>`
  width: 16px;
  height: 16px;
  border-radius: 4px;
  border: 1px solid var(--gray);

  svg {
    transform: translate(-1px, -1px);
    transition: 0.3s ease all;
    opacity: ${({ isChecked }) => isChecked ? 1 : 0};
  }
`;

export const CLCheckboxInfo = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-12);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;

export const TAndC = styled.a`
  font-family: ${TypographyENUM.lkSansBold};
  text-decoration: underline;
`;
