// @lk/ui-library/src/Component/V1/StudioFlow/MsiteUserDetailsForm/MsiteUserDetailsForm.stories.tsx
import { action } from "@storybook/addon-actions";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { FormData } from "../UserDetailsForm/UserDetailsForm.types";
import { MsiteUserDetailsForm } from "./MsiteUserDetailsForm.component";

export default {
	title: "Components/v1/StudioFlow/Mobile/MsiteUserDetailsForm",
	component: MsiteUserDetailsForm,
	argTypes: {
		
	},
} as ComponentMeta<typeof MsiteUserDetailsForm>;

const Template: ComponentStory<typeof MsiteUserDetailsForm> = (args) => <MsiteUserDetailsForm {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	submitCallback: (formData: FormData) => action("submitCallback: ")({ formData }),
	initialFormState: {
		firstName: "",
		lastName: "",
		email: "",
		gender: "male",
		phone: "",
		phoneCode: "+65",
		contactLensChecked: false
	},
	isRTL: false,
	isCLProduct: true,
	tNCLink: "https://lenskart.com/",
	localeData: {
		CL_CHECK_TEXT_1: "By proceeding, you agree to our",
		TERMS_AND_CONDITIONS: "Terms & Conditions",
		CL_CHECK_TEXT_2: "for contact lens purchases"
	}
};
