// @lk/ui-library/src/Component/V1/Checkout/AddressTab/AddressTab.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import { AddressTab } from "./index";
import { TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/v1/Checkout/AddressTab",
	component: AddressTab,
	argTypes: {
		...generateGeneralArgs(),
		addressline1: generateArgTypes({
			type: "string",
			required: true,
			description: "To set head text of any offer",
			category: "Component",
			defaultValue: "Sanket Apartments",
		}),
		addressline2: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "614, 3rd Main Road",
		}),
		city: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "Bangalore",
		}),
		country: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "IN",
		}),
		firstName: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "Sourabh",
		}),
		lastName: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "Kumar",
		}),
		landmark: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "Passport Office",
		}),
		phone: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "9110086006",
		}),
		postcode: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "560095",
		}),
		state: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "Karnatka",
		}),
		checked: generateArgTypes({
			type: "boolean",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "false",
		}),
		key: generateArgTypes({
			type: "number",
			required: true,
			description: "To set key of address pane",
			category: "Component",
			defaultValue: 1,
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for Apply Coupon",
			category: "Events",
		}),
	},
} as ComponentMeta<typeof AddressTab>;

const Template: ComponentStory<typeof AddressTab> = (args) => <AddressTab {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: 10961471,
	addressline1: "Sanket Apartment",
	addressline2: "3rd Main Road",
	city: "Bangalore",
	country: "IN",
	firstName: "Sourabh",
	landmark: "Koramangla Passport Office",
	lastName: "Kumar",
	phone: "9999999999",
	postcode: "560035",
	state: "Karnatka",
	selectedAddressIndex: { id: 10961471 },
	checked: true,
	onChange: () => {console.log("Radio Button Clicked");},
	font: TypographyENUM.defaultBook,
};
