// @lk/ui-library/src/Component/V1/Checkout/AddressTab/styles.tsx
import styled from "styled-components";
import { ThemeENUM, TypographyENUM } from "../../../../Types/general";

export const AddressTabWrapper = styled.div <{ styleFont?: TypographyENUM } >`
  font-family: ${(props) => props.styleFont};
  box-sizing: border-box;
  display: flex;
  align-items: center;
  padding: var(--pd-15) var(--pd-15);
  gap: var(--spacing-lg);
  background: var(--white);
`;

export const PersonName = styled.div`
  display: flex;
  width: 100%;
  gap: var(--pd-5);
  font-family: var(--font-default-medium);
  font-size: var(--fs-14);
  color: var(--black);
`;

export const LightText = styled.div`
font-size: var(--fs-13);
font-weight: var(--fw-regular);
color: var(--gray);
gap: var(--pd-5);
display: flex;
`;

export const HR = styled.hr`
    border: none;
    height: 1px;
    margin: 0;
    background-color: rgba(0, 0, 0, .12);
`;

export const RadioInput = styled.input`
	cursor: pointer;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

//////////////////////////////////////////////////////

export const AddressBox = styled.div<{styledFont: TypographyENUM, selected: boolean;}>`
  // display: flex;
  // flex-direction: column;
  width: 100%;
  padding: 16px;
  background-color: ${props => props.selected ? "#f5f5ff" : "#fff"};
//   cursor: pointer;
  transition: all 0.2s ease-in-out;
  margin-bottom: 16px;
  &:hover {
    background-color: #F5F5FF;
  }
  border: ${ props => props.selected ? "1px solid #737397" : ""};
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
  border-radius: 12px;
  font-family: ${(props) => props.styledFont};
  word-wrap: break-word;
`;
export const AddressRow = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 16px;
  // transition: all 0.2s ease-in-out;
`;
export const AddressHead = styled.h3<{font?: TypographyENUM}>`
font-weight: 700;
font-size: 16px;
line-height: 24px;
font-family: ${TypographyENUM.lkSansBold};
color: var(--dark-blue-100);
letter-spacing: -0.02em;
margin-bottom: 4px;
`;
export const ButtonGroup = styled.div`
  display: flex;
  align-items: center;
  button {
    margin-left: 25px;
  }
`;
export const DottedRow = styled.div`
  border: 1px dashed #CECEDF;
  height: 0px;
  margin-bottom: 16px;
`;
export const Checkbox = styled.div<{isActive?: boolean}>`
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0px;
  cursor: pointer;
  width: 24px;
  height: 24px;
  background-color: ${props => props.isActive ? "var(--dark-blue-100)" : "transparent"};
  border-radius: 32px;
  border-width: 1px;
  border-style: solid;
  border-color: rgba(0, 0, 0, 0.38);
  &:after {
    display: ${props => props.isActive ? "block" : "none"};
    content: "";
    width: 4px;
    height: 10px;
    border-width: 0 1px 1px 0;
    border-color: #fff;
    border-style: solid;
    transform: rotate(45deg) translate(-1px, -1px);
  }
`;
export const AddressButton = styled.button<{font?: TypographyENUM, theme?: ThemeENUM}>`
  font-family: ${TypographyENUM.lkSansBold};
  font-size: 14px;
  font-weight: 700;
  line-height: 20px;
  border: none;
  outline: none;
  background-color: transparent;
  padding: 0;
  cursor: pointer;
  color: ${props => props.theme === ThemeENUM.secondary ? "#E34934" : "var(--dark-blue-100)"};
  text-decoration: underline;
`;
export const MainInfo = styled.div``;
export const AddressData = styled.div<{font?: TypographyENUM}>`
  min-width: 0;
  ${props => props.font && `font-family: ${props.font};`}
  // overflow: hidden;
  // white-space: nowrap;
  // text-overflow: ellipsis;
  transition: all 0.2s ease-in-out;
  // &:hover {
  //   transition: all 0.2s ease-in-out;
  //   overflow: visible;
  //   white-space: normal;
  // }
`;
export const DeliveryEstimate = styled.div<{font?: TypographyENUM}>`
${props => props.font && `font-family: ${props.font};`}
`;

export const DeletePopupContainer = styled.div`
	padding: 24px;
    gap: 32px;
    background: #FFFFFF;
    box-shadow: 0px 0px 8px rgb(0 0 0 / 10%);
    width: 507px;
    border-radius: 12px;
`;
export const ConfirmationText = styled.h4`
	font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    /* font-weight: 700; */
    font-size: 16px;
    line-height: 24px;
    letter-spacing: -0.02em;
    color: #333368;
    margin-bottom: 32px;
`;
export const ConfirmBtnRow = styled.div`
	display: flex;
  justify-content: space-between;
  gap:16px;
`;
export const PopupDeleteBtn = styled.button`
	cursor: pointer;
    padding: 12px 20px;
    /* width: 114px; */
    height: 48px;
	border: 1.5px solid #000042;
    border-radius: 100px;
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    /* font-weight: 700; */
    font-size: 16px;
    letter-spacing: -0.02em;
	margin-right: 8px;
	width: 100%;
  background: #000042;
  color: #E2E2EE;
`;
export const PopupCancelBtn = styled.button`
	cursor: pointer;
  border: 1.5px solid #737397;
    outline: none;
    padding: 12px 20px;
	width: 100%;
    /* width: 221.5px; */
    height: 48px;
    background: rgba(0, 0, 66, 0.01);
    border-radius: 100px;
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    /* font-weight: 700; */
    font-size: 16px;
    letter-spacing: -0.02em;
    color: #000042;
	margin-left: 8px;
`;

export const AddressFirstSection = styled.div`
  display:flex;
  align-items:center;
  gap:10px;
`;


export const ConfirmationHeader = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 24px;
  line-height: 36px;
  letter-spacing: -0.02em;
  color: #333368;
  margin-bottom: 15px;
`;
