// @lk/ui-library/src/Types/general.tsx
export enum ComponentSizeENUM {
	small = "small",
	medium = "medium",
	large = "large",
	extraLarge = "extraLarge"
}
export interface DataType {
	[name: string]: string
}
export enum TypographyENUM {
	sans = "var(--font-sans)",
	serif = "var(--font-serif)",
	lkSansBold = "var(--font-lksans-bold)",
	lkSansHairline = "var(--font-lksans-hairline)",
	lkSansMedium = "var(--font-lksans-medium)",
	lkSansRegular = "var(--font-lksans-regular)",
	defaultHeavy = "var(--font-default-heavy)",
	defaultMedium = "var(--font-default-medium)",
	defaultLight = "var(--font-default-light)",
	defaultBook="var(--font-default-book)",
	rajdhaniRegular = "var(--font-rajdhani-regular)",
	lkSerifNormal = "var(--font-lkserif-normal)"
}

export enum ThemeENUM {
	primary = "primary",
	secondary = "secondary",
	msitePrimary = "msitePrimary",
	disableSignInColor = "disableSignInColor",
	signInPrimary = "signInPrimary",
	newSecondary = "newSecondary"
}

export enum CurrencySymbol {
	INR="₹",
	SGD="$",
	AED="AED",
	USD="$",
	SA="SAR"
}

export enum CurrencyType {
	INR="INR",
	SGD="SGD",
	AED="AED"
}

export interface Price {
    symbol: CurrencySymbol,
	currency: CurrencyType,
	basePrice: number,
	lkPrice: number
}

//alert colors
export enum AlertColorsENUM {
	golden = "#F7F1DE",
	blue = "#E0FFFD",
	grey = "#EEEEF5"
}

export enum kindENUM {
	background = "background",
	border = "border",
	tertiary = "tertiary",
}

export enum TextTypeENUM {
	password="password",
    text="text",
    number="number"
}

export enum SignInType {
    EMAIL = "email",
    PHONE = "phone"
}
export interface SignInStatusType {
    type: SignInType,
    showLogin: boolean,
    showOTP: boolean,
    sendOTP: boolean,
    showPassword: boolean,
    isLoading: boolean,
    isError: boolean,
    errorMessage: string | null,
    otpSent: boolean
}

export enum DeviceTypes {
	MOBILE = "mobilesite",
	TAB = "tab",
	DESKTOP = "desktop",
  }
export type JSON = {
	[key: string]: [] | JSON | string | boolean | number | null;
};
