// @lk/ui-library/src/Component/V1/Cart/CartCard/PrescriptionModal/PrescriptionModal.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const TableWrapper = styled.div`
  max-height: 600px;
  margin-bottom: 20px;
  overflow: scroll;
`;

export const Table = styled.table`
  max-width:100%;
	background-color:transparent;
	border-collapse:collapse;
	border-spacing:0;
  width:100%;
	margin-bottom:20px;
  max-height: 600px;
  overflow: scroll;

  tbody>tr:nth-child(odd)>td, tbody>tr:nth-child(odd)>th {
    background-color:#f4f4f4;
  }

  tbody tr:hover>td, tbody tr:hover>th {
    background-color:#ecf0f1;
	transition-property:background-color .2s linear 0s;
	-moz-transition:background-color .2s linear 0s;
	-webkit-transition:background-color .2s linear 0s;
	-o-transition: background-color .2s linear 0s;
  }
`;

export const THead = styled.thead`
  position: sticky;
  top: 0;
  background-color: white;
`;

export const TH = styled.th`
	line-height:20px;
	vertical-align:top;
  font-weight:bold;
  vertical-align:bottom;
  font-family: ${TypographyENUM.lkSansBold};
  font-size: 16px;
  text-align: center;
  border: 1px solid rgba(0,0,0,0.3);
  padding: 10px 20px;
`;

export const TBody = styled.tbody`
  +tbody {
    border-top:2px solid #cccccc;
  }
`;

export const TD = styled.td`
	line-height:20px;
	text-align:left;
	vertical-align:top;
  font-family: ${TypographyENUM.lkSansRegular};
`;

export const UL = styled.ul`
  max-height: 500px;
  overflow: scroll;
`;

export const LI = styled.li`
  padding: 10px 20px;
  border: 1px solid rgba(0,0,0,0.1);
  list-style: none;
  text-align: center;
`;
