// @lk/ui-library/src/Component/V1/MyOrderMobile/ReturnRefundMobilePage/SelectStore/SelectStore.styles.tsx
import styled from "styled-components";

export const Container = styled.div`
	background-color: #ffffff;
	margin: 10px 0 0;
	padding: 10px 10px 70px;
	font-family: var(--font-lksans-regular);
`;

export const Title = styled.div`
	font-family: var(--font-lksans-regular);
	font-size: 13px;
	color: #333333;
	&:last-child {
		color: #27394E;
		margin-top: 12px;
		font-family: var(--font-lksans-bold);
	}
`;

export const Sub = styled.div`
	color: #999999;
	padding: 5px 0 10px;
	font-size: 11px;
`;

export const StoreInput = styled.form`
	margin-bottom: 10px;
`;

export const StoreListContainer = styled.div`
    border: 1px solid #eaeff4;
    border-radius: 5px;
	margin-top: 10px;
`;

export const StoreCard = styled.div`
	padding: 10px;
	display: flex;
	/* justify-content: space-between; */
	input {
		align-self: flex-start;
		margin-top: 3px;
	}
	`;

export const Label = styled.label`
	padding-left: 10px;
	width: 100%;
`;

export const StoreData = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: flex-start;
`;

export const Left = styled.div`
	color: #424242;
	font-size: 11px;
	font-family: var(--font-lksans-regular);
`;

export const StoreName = styled.div`
	font-size: 13px;
	font-family: var(--font-lksans-bold);
`;

export const StoreAddress = styled.div`
	color: #666666;
	margin-top: 4px;
`;

export const StoreTiming = styled.div`
	margin-top: 10px;
	strong {
		font-family: var(--font-lksans-bold);
	}
`;

export const Right = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	color: #424242;
	margin-left: 15px;
`;

export const Icon = styled.div`
	height: 40px;
	width: 40px;
	img {
		width: 100%;
		height: 100%;
	}
`;

export const DirectionTitle = styled.div`
	font-size: 11px;
	font-family: var(--font-lksans-bold);
	margin-top: 5px;
`;

export const Distance = styled.div`
	font-size: 11px;
	font-family: var(--font-lksans-regular);
	margin-left: 5px;
`;

export const Loading = styled.div`
	display: flex;
	align-items: center;
	justify-content: center;
`;
