// @lk/ui-library/src/Component/V1/MyAccount/AccountInformation/AccountInformation.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import AccountInformation from "./index";

export default {
	title: "Components/V1/MyAccount/AccountInformation",
	component: AccountInformation
} as ComponentMeta<typeof AccountInformation>;

const Template: ComponentStory<typeof AccountInformation> = (args) => <AccountInformation {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	firstName: "test",
	lastName: "test",
	gender: "Male",
	currentPassword: "",
	newPassword: "",
	customerEmail: "lenskart@gmail.com",
	confirmNewPassword: "",
	dataLocale: {
		FIRSTNAME: "First Name",
		LASTNAME: "Last Name",
		GENDER: "Gender",
		EMAIL: "Email",
		ERRORMESSAGE: "This is required",
		CHANGE_PASSWORD: "Change Password",
		CURRENT_PASSWORD: "Current Password",
		NEW_PASSOWRD: "New Password",
		CONFIRM_NEW_PASSWORD: "Confirm New Passowrd",
		EDIT_ACCOUNT_INFORMATION: "Edit Account Information",
		ACCOUNT_INFORMATION: "Account Information",
		MATCH_PASSWORD: "New Password does  not match."
	},
	labelPosition: "left",
	options: [
		{
			value: "",
			key: "Select"
		},
		{
			value: "male",
			key: "Male"
		},
		{
			value: "female",
			key: "Female"
		}
	]
};
