// @lk/ui-library/src/Component/V1/MyAccount/AccountInformation/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";
import { positionENUM } from "./AccountInformation.types";

export const FirstNameWrapper = styled.div<{isRTL?: boolean;}>`
    width: 30%;
    ${props => props.isRTL ? "margin-left: 15px;" : ""}
`;

export const LastNameWrapper = styled.div`
    width: 30%;
    margin-left: 10px;
`;

export const InputField = styled.input<{error:boolean, disabled:boolean, prefixText:boolean, isRTL?: boolean, width: string}>`
    width: ${props => props.width ? props.width : ""}%;
    border: 1px solid ${props => props.error ? "#a94442" : "var(--serene-gray)"};
    border-radius: 4px;
    z-index: 42;
    background: #fff;
    padding: 6px;
    margin: 0;
    position: relative;
    letter-spacing: 1px;
    box-sizing: border-box;
    font-size: var(--fs-14);
    line-height: 20px;
    letter-spacing: -.02em;
    color: #555;
    font-family: ${TypographyENUM.lkSansRegular};
    outline: none;
    margin: ${props => props.isRTL ? "10px 0px 10px 10px" : "10px 10px 10px 0px"};
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const InputLabel= styled.label<{prefixText:boolean, labelPosition?: positionENUM, error: boolean }>`
	font-size: 13px;
	font-weight:700;
	margin-bottom: 0.4em;
	${props => props.labelPosition}:${props => props.prefixText ? "2em" : "0"};
    color: ${props => props.error ? "#a94442" : "#333"};
	transition: 200ms cubic-bezier(0, 0, 0, 1) 0ms;
    display: inline-block;
    margin-bottom: 5px;
    min-width: 160px;
	font-family: "FuturaStd-Medium";
	cursor:pointer;
`;

export const NameWrapper = styled.div`
    display: flex;
    width: 70%;
`;

export const GenderWrapper = styled.div``;

export const DropdownLabel = styled.label<{error: boolean}>`
	font-family: inherit;
	font-size: 13px;
	letter-spacing: 1.5px;
	color: ${props => props.error ? "#a94442" : "#333"};
	line-height: 30px;
    display: inline-block;
    max-width: 100%;
    margin-bottom: 5px;
    font-weight: 700;
	font-family: "FuturaStd-Medium";
`;

export const SelectBox = styled.select<{error: boolean}>`
	border-radius: 0;
	padding: 6px 12px;
	// background: var(--white) url(//static.lenskart.com/media/desktop/img/select-bg.png)
	// 	no-repeat scroll right center !important;
	cursor: var(--pointer);
	border: 1px solid ${props => props.error ? "#a94442" : "var(--light-grey)"};
	font-family: inherit;
	font-size: 1em;
	line-height: 1em;
    display: block;
    width: 21%;
    border-radius: 5px;
	font-family: "FuturaStd-Medium";
	color:#333;
`;

export const Option = styled.option`
	font-family: inherit;
	font-size: 1em;
	line-height: 1em;
`;

export const ChangePassowrdBox = styled.div`
    margin-bottom:10px;
	margin-top:3px;
`;

export const PassowrdWrapper = styled.div`
    margin-left: 20px;
    width: 100%;
    margin-bottom:25px;
`;

export const CurrentPassowrdWrapper = styled.div`
    margin-bottom: 15px;
    width: 100%;
    display: flex;
    align-items: center;
`;

export const PasswordInputWrapper = styled.div`
    margin-left: 10%;
    width: 80%;
    display: inline-block;
`;

export const NewPassowrdWrapper = styled.div`
    margin-top: 10px;
    margin-bottom: 5px;
    width: 100%;
    display: flex;
    align-items: center;
`;

export const ConfirmPassowrdWrapper = styled.div`
    margin-top: 10px;
    margin-bottom: 5px;
    width: 100%;
    display: flex;
    align-items: center;
`;

export const ErrorMessage = styled.p`
	color:var(--error);
	margin:0;
	padding-top: var(--pd-5);
	font-size:0.8em;
`;

export const HeaderWrapper = styled.div`

`;

export const Heading = styled.h2`
    margin-top: 20px;
    margin-bottom: 10px;
    font-family: "FuturaStd-Medium";
    font-weight: 400;
    font-size: 30px;
`;

export const SubHeading = styled.h3`
    font-family: "FuturaStd-Medium";
    font-weight: 400;
    margin-top: 20px;
    margin-bottom: 10px;
    font-size: 24px;
`;

export const ButtonWrapper = styled.div`
    margin-top: 2%;
    width: 100%;
`;

export const EmailWrapper = styled.div`
    width: 21%;
    margin: 15px 0px 10px 0px;
`;

export const EmailInputField = styled.input<{disabled:boolean, prefixText:boolean, isRTL?: boolean, width: string}>`
    width: ${props => props.width ? props.width : ""}%;
    border: 1px solid var(--serene-gray);
    border-radius: 4px;
    z-index: 42;
    background-color: #eee;
    opacity: 1;
    padding: 6px;
    margin: 0;
    position: relative;
    letter-spacing: 1px;
    box-sizing: border-box;
    font-size: var(--fs-14);
    line-height: 20px;
    letter-spacing: -.02em;
    color: #555;
    font-family: ${TypographyENUM.lkSansRegular};
    outline: none;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const Border = styled.div<{id: string}>`
    border: 1px dashed rgb(204, 204, 204);
`;

export const Toast = styled.div<{state:string}>`
    width: 100%;
    color: ${props => props.state==="error"?"#721c24":"#155724"};
    border: 1px solid transparent;
    background-color: ${props => props.state==="error"?"#f8d7da":"#d4edda"};
    padding: 0.75rem 1.25rem;
    border-color: ${props => props.state==="error"?"#f5c6cb":"#c3e6cb"};
    margin-bottom: 1rem;
    border-radius: 0.25rem;
`;

export const ToastMessageTitle = styled.strong`
    text-transform: capitalize;
`;
