// @lk/ui-library/src/Component/V1/PrescriptionModal/SpPanel/SPPanel.component.tsx
import React from "react";
import * as Icons from "../../../../Icons/index";
import { DataType } from "../../../../Types/general";
import { Relative, Root, Tick, UpperBody, Header, Body, TableConatiner, TableTitle, TableContent, Date } from "./SPPanel.styles";

interface SpPanelTypes {
    isMobile: boolean;
    info: DataType,
    selected: string,
    onTabClick: (props: DataType) => void;
    prescription: DataType,
    getDate: (props: string) => void;
    POWER_NAME_CONFIG :DataType
    localeData: DataType
	isRTL: boolean
	powerType?: string,
	precheckout?: boolean,
	productData?: any,
}

const SPPanel = ({
	isMobile,
	info,
	selected,
	onTabClick,
	// prescription,
	getDate,
	POWER_NAME_CONFIG,
	localeData,
	isRTL,
	// powerType,
	// precheckout,
	// productData,
}: SpPanelTypes) => {
	return (
		<Relative isMobile={isMobile}>
			<Root isRTL={isRTL} isMobile = {isMobile} selected={selected === info.id} onClick={() => onTabClick(info)}>
				{selected === info.id && (
					<Tick isRTL={isRTL}>
						<Icons.Steppericon />
					</Tick>
				)}
				<UpperBody selected={selected === info.id}>
					<Header>{info.userName}</Header>
					<Date>
						<>
							{(info?.powerType !== "SUNGLASSES" && info?.powerType !== "CONTACT_LENS")
								? info?.powerType === "SINGLE_VISION"
									? localeData.SINGLE_VISION
									: localeData.BIFOCAL_PROGRESSIVE
								: ""}
							{info.powerType !== "SUNGLASSES" && info?.powerType !== "CONTACT_LENS" && " •"} {localeData.CREATED_ON}{" "}
							{getDate(info.recordedAt)}
						</>
					</Date>
				</UpperBody>
				<Body selected={selected === info.id}>
					<TableConatiner>
						<TableTitle>{localeData.POWER}</TableTitle>
						<TableTitle>{localeData.LEFT}</TableTitle>
						<TableTitle>{localeData.RIGHT}</TableTitle>

						{Object.keys(info.left).map((key: any, keyIndex) => {
							if (
								(info.left[key] === "" && info.right[key] === "") ||
                key === "boxes"
							) {
								return null;
							}
							return (
								<React.Fragment key={keyIndex}>
									<TableContent>{POWER_NAME_CONFIG[key]}</TableContent>
									<TableContent>{info.left[key]}</TableContent>
									<TableContent>{info.right[key]}</TableContent>
								</React.Fragment>
							);
						})}
					</TableConatiner>
				</Body>
			</Root>
		</Relative>
	);
};

export { SPPanel };
