// @lk/ui-library/src/Component/V1/PrescriptionModal/SpPanel/SPPanel.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const Root = styled.div<{isMobile: boolean, selected: boolean, isRTL: boolean }>`
  ${props => !props.isMobile && "width: 500px;"}
  box-shadow: 0px 0px 12px rgb(0 0 66 / 6%);
  border-radius: 12px;
  border: none;
  background: #ffffff;
  border: ${(props) => (props.selected ? "1px solid #737397" : "")};
  margin: 8px;

  &:before {
    ${props => props.isRTL ? "left: 25px" : "right: 25px"};
    border: 1px solid #a2a2b8;
    width: 20px;
    height: 20px;
    border-radius: 50%;
    top: 24px;
    content: "";
    position: absolute;
    transition: all 0.3s ease;
  }
`;

export const Tick = styled.span<{isRTL: boolean}>`
  position: absolute;
  top: 24px;
  font-size: 20px;
  ${props => props.isRTL ? "left: 25px" : "right: 25px"};
  color: #000042;
`;

export const Header = styled.div`
  font: 12px/1.2857 "Roboto", sans-serif, "Arial", "Helvetica", sans-serif;
  font-weight: 700;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #000042;
`;

export const Date = styled.div`
  font: 12px/1.2857 "Roboto", sans-serif, "Arial", "Helvetica", sans-serif;
  font-weight: 200;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #66668e;
  margin-top: 5px;
`;

export const Relative = styled.div<{isMobile: boolean}>`
  position: relative;
  cursor: pointer;
  ${props => props.isMobile && "width: 100%;"}
  -webkit-tap-highlight-color: transparent;
`;

export const Body = styled.div<{ selected: boolean }>`
  height: auto;
  max-height: 9999px;
  padding: 16px;
  transition: all 0.5s cubic-bezier(1, 0, 1, 0);
  padding: ${(props) => (props.selected ? "16px" : "")};
  display: ${(props) => (props.selected ? "" : "none")};
`;

export const TableConatiner = styled.div`
  display: grid;
  text-align: center;
  row-gap: 14px;
  grid-template-columns: repeat(3, 1fr);
`;

export const TableTitle = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-weight: 900;
  font-size: 12px;
  line-height: 20px;
  text-align: center;
  color: #000042;
`;

export const UpperBody = styled.div<{ selected: boolean }>`
  padding: 16px;
  border-radius: 12px 12px 0 0;
  ${(props) => props.selected && "background: #F5F5FF"};
`;

export const TableContent = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-weight: 400;
  font-size: 12px;
  line-height: 20px;
  text-align: center;
  letter-spacing: -0.02em;
  color: #000042;
`;
