// @lk/ui-library/src/Component/V1/PDP/Mobile/PackageScreenCard/PackageScreenCard.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import PackageScreenCard from "./PackageScreenCard.component";
import { CurrencySymbol, CurrencyType } from "./PackageScreenCard.type";

export default {
	title:"Components/V1/PDP/Mobile/PackageScreenCard",
	component: PackageScreenCard,
	argTypes:{
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "radio",
			options: TypographyENUM,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left ",
			category: "Status",
			defaultValue: "false",
		}),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		}),
		selected: generateArgTypes({
			type: "boolean",
			required: false,
			description: "show checkbox",
			category: "Component",
			defaultValue: true,
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for button",
			category: "Events",
		}),
	}
} as ComponentMeta<typeof PackageScreenCard>;

const Template: ComponentStory<typeof PackageScreenCard> = args => <PackageScreenCard {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	width: "100",
	onClick: () => { console.log("Clicked");},
	font:TypographyENUM.defaultBook,
	id:"573eb04755206af4266f2820",
	// pkgImageLink: "https://static1.lenskart.com/media/desktop/img/Aug22/progressive@2x.png",
	pkgImageLink: "",
	PlayIcon:"https://static.lenskart.com/media/desktop/img/DesignStudioIcons/PlayCircleFilledGreen.svg",
	title:"Wide Corridor",
	labelBgColor:"var(--ghost-white)",
	labelColor:"var(--dark-blue-75)",
	selected:false,
	// tags: ["Maximum eye protection", "Sunlight Protection"],
	tags: [""],
	// tags: ["Maximum eye protection"],
	showOfferRibbon: true,
	offerColorCode: "lightblue",
	offerTextHasIcon: true,
	offerText2: "https://static.lenskart.com/media/desktop/img/DesignStudioIcons/Check.svg",
	// offerTextHasIcon: false,
	// offerText2: "60% add on off",
	specifications:[
		{
			groupType: "TINT",
			groupTitle: "",
			title: "Wide Corridor Lenses",
			subTitle: "",
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		},
		{
			groupType: "UV_BLOCK",
			groupTitle: "",
			title: "Thinnest Lens As Per Your Power",
			subTitle: "",
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		},
		{
			groupType: "WARRANTY",
			groupTitle: "",
			title: "Near To Far Progression",
			subTitle: "",
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		},
		{
			groupType: "MATERIAL",
			groupTitle: "",
			title: "Anti-Glare Lens",
			subTitle: "",
			hoverUrl: null,
			isEnabled: false,
			boldText: false
		},
		{
			groupType: "THICKNESS",
			groupTitle: "",
			title: "Scratch Resistant",
			subTitle: "",
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		},
		{
			groupType: "THICKNESS",
			groupTitle: "",
			title: "Water & Dust Repellent",
			subTitle: "",
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		},
		{
			groupType: "THICKNESS",
			groupTitle: "",
			title: "UV 400 Protection",
			subTitle: "",
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		},
		{
			groupType: "THICKNESS",
			groupTitle: "",
			title: "1 Year Warranty",
			subTitle: "",
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		}
	],
	offerText1:"Buy 1 Get 1 Free applicable",
	showTaxText:" +tax",
	priceFree: "",
	price:{
		currency: CurrencyType.INR,
		basePrice: 1100,
		lkPrice: 0,
		symbol: CurrencySymbol.INR,
		itemExchangePrice: 3700
	},
	productPrices:{
		currency: CurrencyType.INR,
		basePrice: 1100,
		lkPrice: 0,
		symbol: CurrencySymbol.INR,
		itemExchangePrice: 3700
	},
	deviceType:"mobilesite",
	isExchangeFlow: true
};
