// @lk/ui-library/src/Component/V1/StudioFlow/BookAppointment/Mobile/styles.tsx
import styled from "styled-components";

export const ParentContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 20px;
  padding-top: 20px;
  background-color: #f5f5f5;
  height: 100%;
  overflow-y: scroll;
`;

export const Heading = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #000042;
  align-self: flex-start;
  margin: 0 10px;
  text-transform: capitalize;
`;

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
`;
export const StoreNameContainer = styled.div`
  padding: 0 20px;
  display: flex;
  flex-direction: row;
  align-items: center;
`;
export const StoreName = styled.div<{ alert?: boolean }>`
  font-family: var(--font-lksans-bold);
  font-style: normal;
  font-weight: 700;
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: ${(props) => (props.alert ? "red" : "var(--dark-blue-100)")};
  display: flex;
  gap: 10px;
`;

export const StoreIcon = styled.img``;

export const StoreAddress = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
  width: 200px;
`;

export const BookingDetails = styled.div`
  background: #ffffff;
  box-shadow: 0px 4px 12px rgba(243, 233, 226, 0.9);
  border-radius: 12px;
  display: flex;
  flex-direction: column;
  gap: 20px;
  padding: 30px;
  width: 100%;
`;

export const Section = styled.div`
  display: flex;
  flex-direction: column;
  gap: 20px;
`;

export const Row = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
`;

export const RowItem = styled.div`
  display: flex;
  gap: 20px;
  align-items: center;
`;

export const StoreDateSelector = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 0 20px;
  width: 100%;
`;

export const DottedLine = styled.div`
  width: 100%;
  height: 0px;
  border: 1px dashed #e2e2ee;
`;

export const DateHeading = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;

export const DatesWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
  overflow: hidden;
  width: 100%;
`;

export const DateSubHeading = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
`;

export const DateContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  gap: 10px;
  overflow: scroll;
  //* to hide scroll bars
  ::-webkit-scrollbar {
    width: 0px;
    background: none;
  }
  ::-webkit-scrollbar-track {
    width: 0;
    background: none;
  }
`;

export const StoreTimeSelector = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 0 20px;
  gap: 10px;
  width: 100%;
  &#store-time-selector{
	padding-bottom:110px;
  }
`;

export const TimeWrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

export const TimeContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 15px;
`;

export const CTAWrapper = styled.div`
  position: fixed;
  bottom: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  background: #ffffff;
  box-shadow: 0px -2px 18px -8px rgba(0, 0, 0, 0.18);
  border-radius: 12px 12px 0px 0px;
  padding: 16px;
`;

export const BookAppointmentCTAContainer = styled.div<{
  enabled?: boolean;
  bookSlotLoading?: boolean;
}>`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  align-self: center;
  gap: 10px;
  padding: 16px 24px;
  width: 100%;
  height: 48px;
  background: ${(props) =>
		props.enabled && !props.bookSlotLoading
			? "var(--dark-blue-100)"
			: "#BDBDD3"};
  border-radius: 100px;
  color: var(--white);
  cursor: ${(props) => (props.enabled ? "pointer" : "auto")};
  pointer-events: ${(props) => (!props.enabled ? "none" : "auto")};
  svg path {
    stroke: var(--white);
  }
`;

export const PayNowCTAContainer = styled.div<{ primary?: boolean }>`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  align-self: center;
  gap: 10px;
  padding: 16px 24px;
  width: 100%;
  height: 48px;
  background: ${(props) => (props.primary ? "#11DAAC" : "white")};
  border-radius: 100px;
  color: var(--dark-blue-100);
  border: ${(props) => (!props.primary ? "1.5px solid #737397" : "")};
`;

export const BookAppointmentCTA = styled.div<{ primary?: boolean }>`
  font-family: var(--font-lksans-bold);
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  display: flex;
  align-items: center;
  letter-spacing: -0.02em;
`;

export const AnimatedCheckContainer = styled.div`
  display: flex;
  padding: 20px;
  justify-content: center;
`;
export const ConfirmHeading = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #000042;
  margin: 0 10px;
`;
export const ConfirmAppointmentScreen = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 20px;
  padding: 0 20px;
`;

export const SubHeading = styled.div<{ fontSize?: string }>`
  color: var(--dark-blue-75);
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: var(--fw-medium);
  font-size: ${(props) => props?.fontSize || "var(--fs-12)"};
  line-height: 20px;
  letter-spacing: -0.02em;
`;

export const NoSlotsContainer = styled.div<{ borderColor?: string, background?: string }>`
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 8px 12px;
  gap: 12px;
  background: ${props => props.background || "#fff3d9"};
  border: 1px solid;
  border-color: ${props => props.borderColor || "#e9ae15"};
  border-radius: 8px;
`;

export const ErrorContainer = styled.div`
  padding: 0 20px;
`;
