// @lk/ui-library/src/Component/V1/PrescriptionModal/EnterPowerManually/EnterPowerManually.styles.tsx
import styled, { css } from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const Root = styled.div`
  padding: 20px 16px;
  background: #ffffff;
  box-shadow: 0px 0px 12px rgb(0 0 66 / 6%);
  border-radius: 12px;
  margin-bottom: 15px;
  position: relative;
`;

export const OrderTitle = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;

  color: #66668e;
`;

export const Contain = styled.div`
  padding-bottom: 28px;
`;

export const Label = styled.label`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  color: #000042;
`;

export const Div1Css = css`
    margin-bottom: 20px;
    display: flex;

    input {
      display: none;
    }
    label {
      margin-left: 23px;

      &::before {
        left: 14px;
        top: 24px;
        border: 1px solid #a2a2b8;
        width: 16px;
        height: 16px;
        // border-radius: 50%;
        border-radius: 4px;
        content: '';
        position: absolute;
        transition: all 0.3s ease;
        transition-property: transform, border-color, -webkit-transform;
      }

      &::after {
        left: 14px;
        top: 24px;
        width: 16px;
        height: 16px;
        // border-radius: 50%;
        border-radius: 4px;
        background: #000042;
        position: absolute;
      }
    }
    input[type='checkbox']:checked + label::before {
        width: 4px;
        height: 9px;
        border: solid #fff;
        border-width: 0 1px 1px 0;
        -webkit-transform: rotate(45deg);
        transform: rotate(45deg);
        margin: 5px 7px;
        z-index: 1;
        border-radius: 0px;
        position: absolute;
        left: 13px;
        top: 22px;
      }
      input[type='checkbox']:checked + label::after {
        background: #000042;
        position: absolute;
        transform: scale(1);
        content: '';
        left: 0;
        left: 14px;
        top: 24px;
      }
  }
  `;

export const Div2Css = css`
  margin-bottom: 20px;
  input {
    display: none;
  }
  display: flex;
  label {
    margin-left: 23px;

    &::before {
      top: 68px;
      left: 14px;
      border: 1px solid #a2a2b8;
      width: 16px;
      height: 16px;
      // border-radius: 50%;
      border-radius: 4px;
      content: "";
      position: absolute;
      transition: all 0.3s ease;
      transition-property: transform, border-color, -webkit-transform;
    }

    &::after {
      top: 68px;
      left: 14px;
      width: 16px;
      height: 16px;
      // border-radius: 50%;
      border-radius: 4px;
      background: #000042;
      position: absolute;
    }
  }
  input[type="checkbox"]:checked + label::before {
    width: 4px;
    height: 9px;
    border: solid #fff;
    border-width: 0 1px 1px 0;
    -webkit-transform: rotate(45deg);
    transform: rotate(45deg);
    margin: 5px 7px;
    z-index: 1;
    border-radius: 0px;
    position: absolute;
    top: 66px;
    left: 13px;
  }
  input[type="checkbox"]:checked + label::after {
    background: #000042;
    position: absolute;
    transform: scale(1);
    content: "";
    top: 68px;
    left: 14px;
  }
`;

export const Div1 = styled.div`
  ${Div1Css};
`;

export const Div2 = styled.div`
  ${Div2Css};
`;

export const RootHeader = styled.div<{ isMobile: boolean,isRTL?:boolean }>`
  ${(props) => !props.isMobile && "margin-left: 30px;"}
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 24px;
  line-height: 36px;
  /* identical to box height, or 150% */

  /* Text/Dark/Main */

  margin-right: ${(props) => props.isRTL ? "30px" : ""};

  color: #000042;
  @media screen and (max-width:767px){
	font-size: 20px;
  }
`;

export const Helper = styled.div`
  margin: 36px 40px;
  display: flex;
  justify-content: space-between;
`;

export const PhoneNumberText = styled.div`
  margin: 20px 0 0px;
  display: flex;
  justify-content: space-between;
  align-items:center;
  justify-content:center;
  a{
	color: #358d84;
  }
`;
export const EnterPowerManuallyWrap = styled.div`
  margin-bottom: 40px;
`;

export const PowerType = styled.div`
  display: grid;
  grid-template-columns: 1fr 2fr 2fr;
  grid-gap: 12px;
  -webkit-box-align: center;
  align-items: center;
`;

export const PowerHeader = styled.div<{ align: boolean; isTitle: boolean }>`
  /* font-weight: 900; */
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #333368;
  margin-bottom: 15px;
  font-family: ${(props) =>
		props.isTitle
			? `${TypographyENUM.lkSansBold}`
			: `${TypographyENUM.lkSansRegular}`};
  ${(props) => props.align && "text-align: center"};
`;

export const Input = styled.select<{isRTL:boolean}>`
  appearance: none;
  ${css`
    background-image: url(https://static.lenskart.com/media/Assets/Images/down-arrow.svg);
  `}
  background-repeat: no-repeat;
  background-position: right 1rem center;
  background-size: 1.2em;

  background-repeat: no-repeat;
  background-position-x: 85%;
  background-position-y: 17px;
  color: #66668e;
  font-family: ${TypographyENUM.lkSansRegular};

  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  display: flex;
  align-items: center;
  color: #66668e;
  height: 48px;
  width: 100%;
  padding: 0 40px 0 16px;

  border: 1px solid #cecedf;

  margin-bottom: 15px px;
  border-radius: 8px;

  text-align: ${(props) => props.isRTL ? "center" : ""}
`;

export const AxisInput = styled.input`
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0px 16px;
  isolation: isolate;
  height: 48px;
  background: #ffffff;
  border: 1px solid #cecedf;
  border-radius: 8px;
  width: 100%;
  text-align: center;
  margin-bottom: 15px;
  font-family: "LenskartSans-Regular";
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #66668e;
  user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const Title = styled.div`
  font-weight: 400;
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #333368;
  font-family: "LenskartSans-Regular";
`;

export const InputCode = styled.span<{isRTL: boolean}>`
    position: absolute;
    top: 37px;
    z-index: 2;
    left: 12px;
    font-weight: 400;
    font-size: 14px;
    direction: ltr;
    display: flex;
`;

export const InputContainer = styled.div`
  position: relative;
  font-family: "LenskartSans-Regular";
  color: #000042;

    input::-webkit-outer-spin-button,
    input::-webkit-inner-spin-button {
      -webkit-appearance: none;
      margin: 0;
    }

    /* Firefox */
    input[type="number"] {
      -moz-appearance: textfield;
    }
`;

export const InputText = styled.input<{phoneNumber: string, paddingLeft?: boolean}>`
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #333368;
  padding: ${props => props.phoneNumber ? "22px 16px 6px 45px" : "22px 16px 6px 16px"};
  ${props => props.paddingLeft && "padding-left: 70px"};
  width: 100%;
  isolation: isolate;
  background: #ffffff;
  border: 1px solid #cecedf;
  border-radius: 8px;
  margin-top: 12px;
  user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}

  ${css`
    &:focus ~ label,
    &:not(:placeholder-shown) + label {
      position: absolute;
      font-weight: 400;
      font-size: 10px;
      line-height: 16px;
      letter-spacing: -0.02em;
      color: #66668e;
      left: 15px;
      top: 19px;
    }
  `}
`;

export const LabelText = styled.label<{isRTL: boolean}>`
  position: absolute;
  font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #66668e;
  ${props => props.isRTL ? "right: 15px" : "left: 15px"};
  top: 27px;
  -webkit-transition: top 0.3s;
  transition: top 0.3s;
  pointer-events: none;
`;

export const LowerBodyContainer = styled.div`
  display: flex;
  justify-content: center;
`;

export const LowerBody = styled.div`
  width: 500px;
`;

export const ErrorSpan = styled.span`
  font-size: 12px;
  margin-top: 4px;
  color: #d54c44;
  font-family: ${TypographyENUM.lkSansRegular};
  margin: 1px;
`;

export const PDWarning = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 20px;

  color: #333368;
`;

export const Body = styled.div<{ isMobile: boolean }>`
  ${(props) =>
		props.isMobile && "padding: 88px 0; margin-top: 0; margin: 0 20px;"}
`;

export const Break = styled.span`
    display: block;
    margin: 0 6px;
    transform: scaleY(2.5);
`;

export const BreakNew = styled.div`
  position: absolute;
  transform: scaleY(2.2);
  top: -4px;
  left: 48px;
`;

export const ErrorText = styled.span`
  color: #e34934;
  font-size: 12px;
  margin-top: 4px;
  font-family: ${TypographyENUM.lkSansRegular};
`;
