// @lk/ui-library/src/Component/V1/PackageRadio/PackageRadio.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../StorybookHelper";
import { ComponentSizeENUM } from "../../../Types/general";
import PackageRadio from "./index";

export default {
	title: "Components/v1/PackageRadio",
	component: PackageRadio,
	argTypes: {
		...generateGeneralArgs(),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		}),
		showIcon: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to show only icon inside button",
			category: "Icon",
			defaultValue: "false",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for button",
			category: "Events",
		}),
		isSelected: generateArgTypes({
			type: "boolean",
			required: false,
			description: "Whether the Radio is selected or not",
			category: "Events",
		}),
		title: generateArgTypes({
			type: "string",
			required: false,
			description: "title of radio",
			category: "Component",
		}),
		packageDesc: generateArgTypes({
			type: "string",
			required: false,
			description: "The description of PackageRadio",
			category: "Component",
		}),
		imageUrl: generateArgTypes({
			type: "string",
			required: false,
			description: "Image of PackageRadio",
			category: "Component",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: false,
			description: "show to right to left",
			category: "Status",
		})
	},
} as ComponentMeta<typeof PackageRadio>;

const Template: ComponentStory<typeof PackageRadio> = (args) => <PackageRadio {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "single_vision",
	width: "100",
	componentSize: ComponentSizeENUM.small,
	isSelected: true,
	onClick: () => { console.log("Clicked");},
	title: "Package",
	packageDesc: "Package Description",
	imageUrl: "https://static.lenskart.com/media/desktop/img/pdp/single_vision.png"
};
