// @lk/ui-library/src/Component/V1/PackageRadio/styles.tsx
import styled from "styled-components";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";
import { fontSize, spacing, TitlefontSize } from "./helper";


export const ButtonWrapper = styled.button<{
	styleSize: ComponentSizeENUM
	isBorder: boolean
	width?: string | number
}>`
	cursor: ${(props) => (props.disabled ? "not-allowed" : "pointer")};
	border-radius: var(--border-radius-xs);
	border: ${(props) => (props.isBorder ? "2px solid var(--irish-blue)" : "1px solid #e5e8e9")};
	background: var(--white);
	padding: var(${(props) => spacing[props.styleSize]});
	font-size: var(${(props) => fontSize[props.styleSize]});
	width: ${(props) => (props.width ? props.width : null)}%;
	display: flex;
	justify-content: center;
`;

export const FlexWrapper = styled.div<{isRTL: boolean}>`
	display: flex;
	align-items: center;
	width: 100%;
	justify-content: var(--space-between);
	margin: 3px;
`;
export const LeftWrapper = styled.div`
	display: flex;
	align-items: center;
`;
export const RightWrapper = styled.div`
	display: flex;
	align-items: center;
`;
export const IconWrapper = styled.span`
	margin: 5px;
`;

export const PowerText = styled.div<{isRTL: boolean, styleFont: TypographyENUM}>`
	font-family: ${props => props.styleFont};
	text-align: ${ props => props.isRTL ? "right": "left"};
	margin-left: ${props => props.isRTL ? "" : "var(--pd-20)"};
	margin-right: ${props => props.isRTL ? "var(--pd-20)" : ""};
`;

export const Wrapper = styled.div`
	display: flex;
	flex-direction: column ;
	gap: var(--pd-10);
`;

export const Title = styled.div<{
	styleSize: ComponentSizeENUM
}>`
	letter-spacing: 1px;
	line-height: 1.5;
	font-size: var(${(props) => TitlefontSize[props.styleSize]});
	font-weight: var(--fw-semi-bold);
`;

export const Description = styled.div`
	letter-spacing: 1px;
	font-size: 13px;
	color: #6b7378;
	line-height: 1.4;
`;
