// @lk/ui-library/src/Component/Common/Confetti/ConfettiStyles.tsx
/* eslint-disable no-constant-condition */
import styled from "styled-components";

// const bang = keyframes`
//    	from {
//           transform: translate3d(0, 0, 0);
//           opacity: 1;
//     }
// `;

// const animation = css`${bang}`;

function template(i:any) {
	return `
        &:nth-child(${i + 1}) {
		  	transform: translate3d(${Math.floor((Math.random() * 190) - 100)}px, ${Math.floor((Math.random() * 200) - 100)}px, 0) rotate(${Math.floor((Math.random() * 360))}deg);
          	background: rgb(${Math.floor(Math.random() * 256)} , ${Math.floor(Math.random() * 256)}, ${Math.floor(Math.random() * 256)});
			animation-duration: 2s
			opacity: 0;
         }
      `;
}

function getAnimations() {
	let str = "";
	for (let index = 0; index < 50; index += 1) {
		str += template(index);
	}
	return str;
}
export const Item = styled.i`
		position: absolute;
        display: block;
        left: 50%;
        top: 0;
        width: 5px;
        height: 10px;
        background: red;
        opacity: 0;
	${getAnimations()}
`;


export const ConfettiWrapper = styled.div`
`;
