// @lk/ui-library/src/Component/NewHtoFlow/HtoSignIn/HtoSignIn.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import SignIn from ".";
import { SignInType } from "../../../Types/general";

export default {
	title: "NewHtoFlow/HtoSignIn",
	component: SignIn,
	argTypes: {}
} as ComponentMeta<typeof SignIn>;

const Template: ComponentStory<typeof SignIn> = (args) => <SignIn {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		SIGN_UP: "Signup",
		LOG_IN: "Login",
		ENTER_VALID_10_DIGIT_NUMBER: "Enter a valid 10-digit mobile number",
		SEND_ME_TEXT_WHATSAPP: "send me text updates on WhatsApp",
		ENTER_OTP: "Enter Otp",
		OTP_SENT_TO: "Otp sent to",
		CHANGE: "Change",
		CONTINUE: "Continue",
		LOGIN_VIA_WHATSAPP: "Login via WhatsApp",
		OTP_VERIFIED: "OTP Verified",
		RESEND_OTP: "Resend OTP", 
		RESEND_OTP_IN: "Resend OTP in",
		OTP_HAS_BEEN_SENT_SUCCESSFULLY: "OTP has been sent successfully"
	},
	supportMultipleCountries: false,
	incCountryCodeFont: true,
	loaderImageLink: "//static.lenskart.com/skin/frontend/base/default/images/loader2.gif",
	countryCode: "+91",
	signInStatus: {
		isLoading: false,
		showLogin: false,
		type: SignInType.EMAIL,
		showOTP: false,
		sendOTP: false,
		showPassword: false,
		isError: false,
		errorMessage:  null,
		otpSent: false
	}
};
