// @lk/ui-library/src/Component/NewHtoFlow/HtoSignIn/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";


export const SignInWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 10px;
    padding: 0px 16px;
`;

export const SignInPageDetailsContainer = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    align-items: space-between;  
`;

export const LoginMobileWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    gap: 10px;
`;

export const LogInSignUpContainer = styled.div`
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-16);
    line-height: 20px;
    color: var(--dark-blue-100);
`;

export const MobileFiledWithLabelWhatsappWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 0px;
    gap: 24px;
`;

export const MobileFiledWithLabelWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 0px;
    gap: 16px;
`;

export const Label = styled.div`
    display: flex;
    font-family: ${TypographyENUM.lkSansMedium};
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
    line-height: 20px;
    color: #66668E;
`;

export const InputFiledWrapper = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 4px;
    gap: 8px;
    font-family: ${TypographyENUM.lkSansMedium};
    font-style: normal;
    font-weight: var(--fw-medium);
    font-size: var(--fs-14);
    line-height: 20px;
    color: #000042;
`;

export const PhoneCodeDropdownWrapper = styled.div<{top: string}>`
	position: relative;
	top: ${props => props.top}px;
`;

export const WhatsAppCheckboxWithLabel = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
`;

export const InputCheckbox = styled.input`
    width: 16px;
    height: 16px;
    border-radius: 4px;
    border: 1px solid var(--dark-blue-100);
    &:checked {
        accent-color: var(--dark-blue-100);
    }
`;

export const WhatsAppText = styled.div`
    margin-left: 10px;
    font-family: ${TypographyENUM.lkSansMedium};
    font-style: normal;
    font-weight: var(--fw-medium);
    font-size: var(--fs-14);
    line-height: 20px;
    color: var(--text);
`;

export const ButtonWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    padding: 24px 24px 32px;
    gap: 16px;
    position: absolute;
    width: 100%;
    //height: 168px;
    left: 0px;
    bottom: 0px;
    background: #FFFFFF;
    box-shadow: 0px -4px 20px rgba(0, 0, 66, 0.1);
    border-radius: 12px 12px 0px 0px;
`;

export const LoginViaWhatsApp = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 14px 20px;
    gap: 8px;
    width: 312px;
    height: 48px;
    border: 1.5px solid #000042;
    border-radius: 12px;
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-16);
    line-height: 20px;
    color: var(--dark-blue-100);
`;

export const ContinueButton = styled.div<{disabled?: boolean}>`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    padding: 14px 20px;
    gap: 8px;
    width: 312px;
    height: 48px;
    background: var(--dark-blue-100);
    border-radius: 12px;
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-16);
    line-height: 20px;
    color: #FFFFFF;
    ${props => props.disabled ? `
        cursor: none;
        background: var(--dark-gray);
        pointer-events: none;
    ` : ""}
`;

export const Change = styled.span`
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-12);
    line-height: 16px;
    display: flex;
    align-items: center;
    color: var(--dark-blue-100);
`;

export const InlineLoader = styled.img`
    display: block;
    width: 80px;
    margin: 0 auto;
`;

export const OtpInputField = styled.input<{isError?: boolean, isSuccess?: boolean}>`
    width: 56px;
    height: 56px;
    background: #FFFFFF;
    border: 1px solid ${ props => props.isError ? "#FFA499" : props.isSuccess ? "#60B527" : "#CECEDF" };
    box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
    border-radius: 8px;
    font-weight: var(--fw-medium);
    font-size: var(--fs-16);
    line-height: 24px;
    text-align: center;
    color: ${ props => props.isError || props.isSuccess ? "var(--dark-blue)" : "#BDBDD3" };    
`;

export const Error = styled.div`
    color: red;
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: 12px;
    line-height: 16px;
    letter-spacing: -0.02em;
    color: #ff0000;
`;

export const OtpVerified = styled.div`
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-12);
    line-height: 16px;
    display: flex;
    align-items: center;
    color: var(--dark-blue-100);
    margin-left: 5px;
`;

export const EditBtn = styled.button<{ disabled?: boolean, fullWidth?: boolean }>`
    font-weight: var(--fw-bold);
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-size: var(--fs-14);
    line-height: 20px;
    letter-spacing: -0.02em;
    border: none;
    background: transparent;
    color: #0FBD95;
	${props => props.fullWidth && `
		width: 100%;
	`}
`;

export const Timer = styled.span`
    color: #0FBD95;
    font-weight: var(--fw-bold);
    font-family: ${TypographyENUM.lkSansBold};
`;

export const ResendOtpTimer = styled.div`
    font-family: ${TypographyENUM.lkSansMedium};
    font-style: normal;
    font-weight: var(--fw-medium);
    font-size: var(--fs-14);
    line-height: 20px;
    color: #66668E;
`;

export const IconWrapper =styled.div`
    display: flex;
    color: var(--dark-blue-100);
`;
