// @lk/ui-library/src/Component/V1/StudioFlow/MsiteFloatingWindow/MsiteFloatingWindow.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { MsiteFloatingWindow } from "./MsiteFloatingWindow.component";

export default {
	title: "Components/v1/StudioFlow/Mobile/MsiteFloatingWindow",
	component: MsiteFloatingWindow,
	argTypes: {
		onClick: generateArgTypes({
			type: "function",
			category: "General",
			required: true,
			description: "Click Event handler of the Radio Card"
		}),
		disabled: generateArgTypes({
			type: "boolean",
			category: "General",
			required: true,
			description: "Whether the button is disabled or not"
		})
	},
} as ComponentMeta<typeof MsiteFloatingWindow>;

const Template: ComponentStory<typeof MsiteFloatingWindow> = (args) => <MsiteFloatingWindow {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	localeData: {
		SELECT_STORE: "Select store",
		STUDIOFLOW_INFO: "Store eye test & fitting required for dual vision specs. (Available for free at Lenskart stores)"
	},
	disabled: false
};
