// @lk/ui-library/src/Component/V1/StudioFlow/MsiteFloatingWindow/MsiteFloatingWindow.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const FloatingWindowWrapper = styled.div`
  position: fixed;
  bottom: 0;
  left: 0;
  right: 0;
  padding: var(--pd-16) var(--pd-16) var(--pd-32);
  background: var(--white);
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
  border-radius: var(--border-radius-xs) var(--border-radius-xs) 0px 0px;
`;

export const InfoContainer = styled.div`
  display: flex;
  align-items: center;
  gap: 8px;
  padding-right: var(--pd-24);
  padding-bottom: var(--pd-12);
`;

export const InfoIconContainer = styled.div``;

export const Info = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-10);
  line-height: 16px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;

export const ButtonContainer = styled.div``;

export const Button = styled.button`
  border: none;
  width: 100%;
  font-family: ${TypographyENUM.lkSansBold};
  background: var(--dark-blue-100);
  border-radius: var(--border-radius-xxl);
  padding: var(--pd-12) var(--pd-20);
  color: var(--white);
  text-align: center;
  transition: 0.3s all ease;

  &:disabled {
    background: var(--ghost-white);
    color: var(--gray);
  }
`;
