// @lk/ui-library/src/Component/Common/CollapsibleSideBar/CollapsibleSidebar.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../StorybookHelper";
import CollapsibleSidebar from "./CollapsibleSidebar.component";
import Coupon from "../../V1/Cart/Coupon";
import { TypographyENUM } from "../../../Types/general";
import { action } from "@storybook/addon-actions";

export default {
	title: "Components/common/CollapsibleSidebar",
	component: CollapsibleSidebar,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		overLay: generateArgTypes({
			type: "boolean",
			required: true,
			description: "String used as unique identifier Note - If overLay is false then xPosition should be equal to width!",
			category: "General",
			defaultValue: true
		}),
		width: generateArgTypes({
			type: "number",
			required: true,
			description: "To set width of Sidebar",
			category: "General",
			defaultValue: 430
		}),
		xPosition: generateArgTypes({
			type: "number",
			required: true,
			description: "To set position on X-axis",
			category: "General",
			defaultValue: 0
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "View contetn right to left",
			category: "General",
		}),
		onClose: generateArgTypes({
			type: "function",
			required: true,
			description: "Collapse Sidebar",
			category: "General",
		}),
	},
} as ComponentMeta<typeof CollapsibleSidebar>;

const Template: ComponentStory<typeof CollapsibleSidebar> = (args) => <CollapsibleSidebar {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "collapsible",
	height: 100,
	overLay: true,
	width: 496,
	xPosition: 0,
	title: "Apply Coupon",
	isRTL: false,
	children:
	<Coupon id = "Coupon"
		width = "100"
		brandImage = "https://upload.wikimedia.org/wikipedia/commons/thumb/4/42/Paytm_logo.png/640px-Paytm_logo.png"
		headText = "FLAT100"
		subText = "Save ₹100 with FLAT100"
		descText = "Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more, Valid on orders with items worth ₹3000 or more"
		onClick = {() => {
			action("onClick")("Apply Coupon Clicked");
		}}
		font= {TypographyENUM.defaultBook}
	/>,
	// onClose: () => {console.log("Close clicked");}
};
