// @lk/ui-library/src/Component/V1/CheckoutMobile/MorePaymentOptions/Common/CommonSection.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const SectionWrapper = styled.div<{borderVisible?: boolean}>`
  display:flex;
  justify-content:space-between;
  align-items:center;

  img:first-child{
    width: 58px;
    height: 40px;
  }
  padding: ${props => props.borderVisible ? "var(--fs-16)" : "" };
  background: ${props => props.borderVisible ? "#FFFFFF" : "" };
  box-shadow: ${props => props.borderVisible ? "0px 0px 12px rgb(0 0 66 / 6%)" : "" };
  border-radius: ${props => props.borderVisible ? "var(--fs-12)" : "" };
  @media screen and (max-width:767px){
	&.sectionPayWrapper{
		margin-bottom:24px;
	}
	&.sectionPayWrapper + h2{
		margin-top:0px;
	}
  }
`;

export const Section = styled.div<{isImage: boolean}>`
  display:flex;
  flex-direction:column;
  margin-right:auto;
  margin-left:${props => props.isImage ? "10px" : "0"};
`;

export const SectionName = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #333368;
`;

export const SectionContent = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #66668E;
`;
