// @lk/ui-library/src/Component/V1/Checkout/AddAddressFormSa/AddAddressForm.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";
import { AddAddressFormSa } from "./index";

export default {
	title: "Components/V1/Checkout/AddAddressFormSa1",
	component: AddAddressFormSa,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			defaultValue: {
				HOW_WOULD_YOU_RATE_PRODUCT: "How would you rate the product?",
				WRITE_A_REVIEW: "Write A Review",
				ENTER_EMAIL_ID: "Enter Email Id",
				YOUR_NAME: "Your Name",
				TITLE: "Title",
				REVIEW: "Review",
				POST_REVIEW: "POST REVIEW",
				THIS_FIELD_IS_REQUIRED: "This Field is Required",

			},
			control:"object"
		}),
		onSubmit: generateArgTypes({
			type: "function",
			required: true,
			description: "event on form submission",
			category: "Event",
		}),
	}
} as ComponentMeta<typeof AddAddressFormSa>;

const Template: ComponentStory<typeof AddAddressFormSa> = args => <AddAddressFormSa {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "review-form",
	dataLocale: {
		FIRST_NAME: "First Name",
		LAST_NAME: "Last Name",
		MOBILE:"Mobile",
		EMAIL:"Email",
		ADDRESS_LINE_1: "Address Line 1",
		ADDRESS_LINE_2: "Address Line 2",
		ZIP_CODE: "Zip/Postal Code",
		CITY_DISTRICT: "City/District",
		COUNTRY: "Country",
		STATE_PROVINCE:"State/Province",
		CONTINUE: "CONTINUE",
		VIEW_SAVED_ADDRESS: "VIEW SAVED ADDRESS"
	},
	font:TypographyENUM.defaultBook,
	onSubmit: () => console.log("submit"),
	//onViewSaved: () => console.log("View Saved address"),
	//onPincodeChanged: () => console.log("Pincode changed"),
	countryList: {
		options: [{ country_name: "United Arab Emirates", country_code: "AE" }, { country_name: "Botswana", country_code: "BW" }, { country_name: "India", country_code: "IN" }],
		optionsText: "country_name",
		optionValue: "country_code"
	},
	stateList: {
		options: ["Andaman and Nicobar Islands", "Jharkhand", "Bihar"],
		optionsText: "",
		optionValue: ""
	}
};
