// @lk/ui-library/src/Component/V1/Checkout/AddAddressFormSa/config.ts
export const addressForm = [
	[
		{
			props: {
				id: 1,
				name: "firstName",
				type: "text",
				placeholder: "First Name*",
				errorMessage:
					"First Name should be 3-16 characters and shouldn't include any special character!",
				pattern: "^[A-Za-z0-9]{3,16}$",
				required: true
			},
			label: "",
			apiKey: "firstName",
			fieldType: "normal", // "radio", "select"
			options: [], // "display", "value"
			subText: "",
			defaultValue: "",
		},
		{
			props: {
				id: 2,
				name: "lastName",
				type: "text",
				placeholder: "Last Name*",
				errorMessage:
					"Last Name should be 3-16 characters and shouldn't include any special character!",
				pattern: "^[A-Za-z0-9]{3,16}$",
				required: true
			},
			label: "",
			apiKey: "lastName",
			fieldType: "normal", // "radio", "select"
			options: [], // "display", "value"
			subText: "",
			defaultValue: "",
		}
	],
	[
		{
			props: {
				id: 11,
				// name: "lastName",
				// type: "text",
				// placeholder: "Last Name*",
				errorMessage:
					"Please select your gender",
				// pattern: "^[A-Za-z0-9]{3,16}$",
				required: true
			},
			label: "Select Gender",
			apiKey: "gender",
			fieldType: "radio", // "radio", "select"
			options: [{ key: "Male", value: "male" }, { key: "Female", value: "female" }], // "display", "value"
			subText: "",
			defaultValue: "male",
		}
	],
	[
		{
			props: {
				id: 3,
				name: "phone",
				type: "text",
				placeholder: "Phone Number*",
				errorMessage:
					"Last Name should be 3-16 characters and shouldn't include any special character!",
				pattern: "^[A-Za-z0-9]{3,16}$",
				required: true,
				maxLength: 10
			},
			label: "",
			apiKey: "phone",
			fieldType: "phone", // "radio", "select"
			options: [], // "display", "value"
			subText: "",
			defaultValue: "",
			children: {
				props: {
					id: 4,
					name: "phoneCode",
					type: "text",
					placeholder: "",
					required: true
				},
				fieldType: "phoneCode", // "radio", "select"
				options: [{ key: "+91", value: "+91" }, { key: "+1", value: "+1" }], // "display", "value",
				defaultValue: "+91",
			}
		},
		{
			props: {
				id: 5,
				name: "email",
				type: "email",
				placeholder: "Email Address*",
				errorMessage: "It should be a valid email address!",
				pattern: "",
				required: true,
			},
			label: "",
			apiKey: "email",
			fieldType: "email",
			options: [],
			defaultValue: "",
		}
	],
	[
		{
			props: {
				id: 6,
				name: "addressline1",
				type: "text",
				placeholder: "Address*",
				errorMessage: "Address is required",
				pattern: "",
				required: true,
			},
			label: "",
			apiKey: "addressline1",
			fieldType: "normal",
			options: [],
			defaultValue: "",
			subText: "House / Flat Number"
		}
	],
	[
		{
			props: {
				id: 7,
				name: "postcode",
				type: "text",
				placeholder: "Zip/Postal Code*",
				errorMessage:
					"Please enter a correct postal code",
				pattern: "^(?=.*[0-9])(?=.*[a-zA-Z])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{8,20}$",
				required: true,
				maxLength: 6
			},
			label: "",
			apiKey: "postcode",
			fieldType: "number",
			options: [],
			defaultValue: "",
			subText: "Please enter a postal code (10710)"
		},
		{
			props: {
				id: 8,
				name: "city",
				type: "text",
				placeholder: "City*",
				errorMessage:
					"Please enter a city",
				pattern: "",
				required: true,
				disabled: true
			},
			label: "",
			apiKey: "city",
			fieldType: "normal",
			options: [],
			defaultValue: "",
			subText: ""
		},
	],
	[
		{
			props: {
				id: 9,
				name: "state",
				type: "text",
				placeholder: "Select State*",
				errorMessage:
					"Please select a state",
				pattern: "",
				required: true,
				disabled: true
			},
			label: "",
			apiKey: "state",
			fieldType: "select-state",
			options: [],
			defaultValue: "",
			subText: ""
		},
		{
			props: {
				id: 10,
				name: "country",
				type: "text",
				placeholder: "Select Country*",
				errorMessage:
					"Please select a country",
				pattern: "",
				required: true,
			},
			label: "",
			apiKey: "country",
			fieldType: "select-country",
			options: [],
			defaultValue: "",
			subText: ""
		},
	]
];

// const addressConfig = [
// 	{
// 		props: {
// 			id: 1,
// 			name: "username",
// 			type: "text",
// 			placeholder: "Username",
// 			errorMessage:
// 				"Username should be 3-16 characters and shouldn't include any special character!",
// 			label: "Username",
// 			pattern: "^[A-Za-z0-9]{3,16}$",
// 			required: true
// 		},
// 		fieldType: "normal", // "radio", "select"
// 		options: [{ display: "", value: "" }], // "display", "value"
// 		subText: "",
// 		// error: "",
// 		// validationFn: () => null
// 	}
// ];
