// @lk/ui-library/src/Component/Common/Input/Input.types.tsx
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";
import React from "react";
export interface InputType {
	placeholder?: string
	type: string
	componentSize?: ComponentSizeENUM
	width: string
	id?: string
	disabled?: boolean
	readonly?: boolean
	hasError?: boolean
	errorMessage?:string;
	label:string
	font:TypographyENUM
	onValueChange?: (value: React.ChangeEvent<HTMLInputElement>) => void
	includeButton?:boolean
	buttonText?:string
	buttonPosition?:positionENUM
	labelPosition?:positionENUM
	buttonOnClick?:() => void
	// will add functionalities for below parameters later
	// includeIcon?:boolean
	// iconText?:string
	// iconPosition?:positionENUM
	// iconOnClick?:() => void
	countryCodePrefix?:string,
	onKeyUpChange?: (value: React.ChangeEvent<HTMLInputElement>) => void
	isRTL?: boolean
	dir?: string
	labelFontSize?: string
	inputType?: string;
	onBlur?: (e: React.FocusEvent<HTMLInputElement>) => void;
	onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
	// suffix?:string
}

export enum positionENUM{
	right="right",
	left="left",
	// top="top",
}

export enum typeENUM {
	text = "text",
	password = "password",
	email = "email",
	number = "number",
	dir = "auto"
}

export const inputs = {
	type: typeENUM,
};
