// @lk/ui-library/src/Component/V1/MyAccount/MyOrder/TrackingTimeLine/styles.tsx
import styled from "styled-components";

export const TrackingDataContainer = styled.div`
    margin: 25px 0px 20px 0px;
    display: inline-block;
    width: 100%;
    text-align: center;
`;

export const StatusInfo = styled.div`
    display: flex;
    height: 35px;
    justify-content: center;
    align-items: center;
    position: relative;
`;

export const SatatusIcon = styled.div`
    position: relative;
    z-index: 3;
`;

export const TrackingIcon = styled.div``;

export const RoundIcon = styled.div`
    width: 10px;
    height: 10px;
    border: 1px solid #ddd;
    z-index: 3;
    border-radius: 50%;
    background-color: #ffffff;
`;

export const Image = styled.img`
    max-height: 35px;
    max-width: 35px;
    width: 100%;
    display: block;
    vertical-align: middle;
`;

export const ProcessBar = styled.span`
    border: 1px solid #d8d8d8;
    width: 100%;
    height: 2px;
    position: absolute;
    left: -50%;
    top: 15px;
    z-index: 2;
`;

export const StatusLabel = styled.div<{current: boolean}>`
    font-weight: 700;
    font-family: "FuturaStd-Medium";
    font-size: 12px;
    margin-top: 5px;
    color: ${props => props.current ? "#faa61a": "#666"};
`;

export const Date = styled.div`
    color: #999999;
    font-size: 12px;
    line-height: 20px;
`;

export const DeliveryWrapper = styled.div`
    color: #999999;
    font-size: 12px;
    line-height: 20px;
`;

export const WatchVideo = styled.div`
    line-height: 20px;
    cursor: pointer;
    color: #00bac6;
    font-size: 12px;
    display: inline-block;
`;

export const ShowToolTip = styled.div`
    line-height: 20px;
    color: #999999;
    font-size: 12px;
`;

export const ViewDetails = styled.div`
    line-height: 20px;
    color: #999999;
    font-size: 12px;
`;

export const WatchVideoImage = styled.img`
    max-width: 14px;
    display: inline !important;
    margin-left: 5px;
`;

export const TrackingTimeLineWrapper = styled.div`
    padding: 0px 40px;
    .slider-main {
            padding: 0 10px;
    };
    .slick-arrow {
        box-shadow: 0px 0px 16px rgba(0, 0, 0, .08);
        width: 30px;
        height: 30px;
        border-radius:50%;
        top: 40px;
        background: #fff !important;
        text-align: center;
        z-index: 20;
    };
    // .slick-slide {
    //     min-height: 0px;
    // };
    .slick-next:before {
        color: #666; 
    };
    .slick-prev:before {
        color: #666; 
    };
    .slick-disabled {
        display: none !important;
    };
`;
