// @lk/ui-library/src/Component/V1/MyAccount/MyOrder/TrackingTimeLine/TrackingTimeLine.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import TrackingTimeLine from ".";

export default {
	title: "Components/V1/MyAccount/MyOrder/TrackingTimeLine",
	component: TrackingTimeLine
} as ComponentMeta<typeof TrackingTimeLine>;

const Template: ComponentStory<typeof TrackingTimeLine> = (args) => <TrackingTimeLine {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		ARRIVING: "Arriving",
		RUNNING_LATE: "Running Late",
		WAS_EXPECTED: "Was Expected",
		NOW: "Now",
		WATCH_VIDEO: "Watch Video",
		TRACKING_ICON: "Tracking Icon",
		VIDEO_PLAY_ICON: "Video Play Icon",
		VIEW_DETAILS: "View Details",
	},
	orderId: 1930493941,
	itemId: 573995,
	revisedEta: undefined,
	deliveryDate: 1672012800000,
	itemTracking: {
		itemTrackingStatus: "ORDER_PLACED",
		itemTrackingStatusCheckpoint: "ORDER_PLACED",
		expectedDeliveryDate: 1669852800000,
		status: "IN_WAREHOUSE",
		statusLabel: "Preparing for Dispatch",
		histories: [
			{
				status: "ORDER_PLACED",
				statusLabel: "Order Created",
				createdAt: 1669305765311,
				updatedAt: 1669305765311,
				current: false,
				isVisible: true,
				iconUrl:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/OrderCreated.png",
				isEarly: false
			},
			{
				status: "IN_WAREHOUSE",
				statusLabel: "Preparing for Dispatch",
				createdAt: 1669325637000,
				updatedAt: 1669325637000,
				current: true,
				isVisible: true,
				iconUrl:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Processing.png",
				description: "Check out how your product is prepared",
				descriptionLink: "MR-ZnDoZx1A",
				descriptionImageLink:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Group_17.png",
				isEarly: false
			},
			{
				status: "IN_TRANSIT",
				statusLabel: "On the Way",
				current: false,
				isVisible: true,
				iconUrl:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/OnTheWay.png",
				isEarly: false
			},
			{
				status: "DELIVERED_TO_CUSTOMER",
				statusLabel: "Delivered",
				current: false,
				isVisible: true,
				iconUrl:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/DeliveredToStore.png",
				isEarly: false
			},
			{
				status: "CANCELLED",
				statusLabel: "Cancelled",
				current: false,
				isVisible: false,
				iconUrl:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Cancelled.png",
				isEarly: false
			},
			{
				status: "RETURN_TO_ORIGIN",
				statusLabel: "Returned to Warehouse",
				current: false,
				isVisible: false,
				iconUrl:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png",
				isEarly: false
			},
			{
				status: "RETURN_REQUESTED",
				statusLabel: "Return Requested",
				current: false,
				isVisible: false,
				iconUrl:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png",
				isEarly: false
			},
			{
				status: "RETURN_RESHIPPED",
				statusLabel: "Returned Product Shipped Back",
				current: false,
				isVisible: false,
				iconUrl:
                    "https://static1.lenskart.com/media/desktop/img/Sep20/ProductShippedBacktoCustomer.png",
				isEarly: false
			},
			{
				status: "RETURNED",
				statusLabel: "Returned",
				current: false,
				isVisible: false,
				iconUrl:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png",
				isEarly: false
			},
			{
				status: "RETURN_CANCELLED",
				statusLabel: "Return Cancelled ",
				current: false,
				isVisible: false,
				iconUrl:
                    "https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/RetuenCancelled.png",
				isEarly: false
			},
			{
				status: "RETURN_REFUNDED",
				statusLabel: "Refund Completed",
				current: false,
				isVisible: false,
				iconUrl: "https://static1.lenskart.com/media/desktop/img/Sep20/Refund.png",
				isEarly: false
			},
			{
				status: "RETURN_RECEIVED",
				statusLabel: "Return Received",
				current: false,
				isVisible: false,
				iconUrl: "https://static1.lenskart.com/media/desktop/img/Sep20/ReturnReceived.png",
				isEarly: false
			},
			{
				status: "EXCHANGE_CREATED",
				statusLabel: "Exchange Order Created",
				current: false,
				isVisible: false,
				iconUrl:
                    "https://static1.lenskart.com/media/desktop/img/Sep20/ExchangeOrderCreated.png",
				isEarly: false
			}
		]
	}
};
