// @lk/ui-library/src/Component/V1/PrescriptionModal/DeleteItem.tsx
/* eslint-disable @typescript-eslint/no-empty-function */
import React from "react";
import styled from "styled-components";
import { DeviceTypes, TypographyENUM } from "../../../Types/general";
import { CartClose, WrapperText, DelHandling, CartPopUp, PopUpConfirm, PopUpWishlist, Wrapper } from "../Cart/CartCard/styles";

const DeletePopUp = styled.div<{isMobile: boolean}>`
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  background: #ffffff;
  font-family: ${TypographyENUM.lkSansRegular};
  letter-spacing: -0.02em;
  color: #000042;
  height: 30vh;
  border-radius: 16px 16px 0 0;
  z-index: 10;
  transition: position ease 500ms;
  top: 50%;
  right: ${props => props.isMobile ? "27%" : "50%"};
  transform: translate( ${props => props.isMobile ? "20%" : "50%"},-50%);
  display: flex;
    height: 200px;
    flex-direction: column;
    border-radius: 12px;
    justify-content: center;
    align-items: center;
    gap: 24px;
    isolation: isolate;
    ${props => !props.isMobile && "position: absolute;"} 
	box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;

`;

const PopUpWrapper = styled.div<{isMobile: boolean}>`
 ${props => props.isMobile && "transform: scale(.75);"}
`;

interface DeleteItemTypes{
    setShowDeleteModal: (props: boolean) => void,
    removeImage: () => void
	deviceType: string;
}

const DeleteItem = (props: DeleteItemTypes) => {
	const {
		setShowDeleteModal,
		removeImage,
		deviceType
		// currencySymbol,
	} = props;

	const handleCloseModal = () => {
		setShowDeleteModal(false);
	};

	return (
		<>
			<DeletePopUp isMobile={deviceType === DeviceTypes.MOBILE}>
				<CartClose onClick={()=>handleCloseModal()}>
					<img
						alt="close"
						src="https://static.lenskart.com/media/desktop/img/DesignStudioIcons/CrossBlue.svg"
					/>
				</CartClose>
				<Wrapper>
					<WrapperText>
						<DelHandling>
          Delete the image?
						</DelHandling>
					</WrapperText>
				</Wrapper>
				<PopUpWrapper isMobile={deviceType === DeviceTypes.MOBILE}>
					<CartPopUp>
						<PopUpConfirm onClick={() => {removeImage(); handleCloseModal();}}>
            Yes, remove
						</PopUpConfirm>
					
						<PopUpWishlist onClick={() => {
							handleCloseModal();
						}}>
              Cancel
						</PopUpWishlist>
					
					</CartPopUp>
				</PopUpWrapper>
			</DeletePopUp>
		</>
	);
};

export default DeleteItem;
