// @lk/ui-library/src/Component/V1/PrescriptionModal/EnterMyPowerMenuOptionsNew.tsx

import React from "react";
import styled from "styled-components";
import { DataType, TypographyENUM } from "../../../Types/general";

const PanelItems = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 16px 5px 16px 16px;
  cursor: pointer;
`;

const Title = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  color: #000042;
  line-height: 20px;
  font-size: 14px;
  font-weight: 400;
  width: 296px;
`;
const Img = styled.img<{isRTL:boolean}>`
transform: ${(props) => props.isRTL ? "rotate(180deg)" : ""}
`;

interface OptionTypes {options: DataType, localeData: DataType, ChangePage: (props: any) => void, helpFunction: () => void, isRTL?: boolean}

export default function EnterMyPowerMenuOptionsNew({
	options,
	localeData,
	ChangePage,
	helpFunction,
	isRTL=false
}: OptionTypes) {
	const { name } = options;

	return (
		<PanelItems
			key={name}
			onClick={() => {
				if(name === "WHATSAPP_SUPPORT"){
					helpFunction();
					return;
				}
				ChangePage(name);
			}}
		>
			<Title>{localeData[name]}</Title>

			<div>
				<Img
					src="https://static1.lenskart.com/media/desktop/img/DesignStudioIcons/RightArrow.svg"
					alt="right arrow"
					isRTL={isRTL}
				/>
			</div>
		</PanelItems>
	);
}
