// @lk/ui-library/src/Component/NewHtoFlow/NeedHelp/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const NeedHelpWrapper = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
`;

export const NeedHelpText = styled.div`
    font-family: ${TypographyENUM.lkSansMedium};
    font-style: normal;
    font-weight: var(--fw-medium);
    font-size: var(--fs-14);
    line-height: 20px;
    color: var(--dark-blue-100);
`;

export const ChatWithUSText = styled.a`
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-12);
    line-height: 16px;
    color: var(--dark-blue-100);
    border-bottom: 1px dashed var(--dark-blue-100);
    text-decoration: none;
`;

export const IconWrapper = styled.div`
    font-size: var(--fs-13);
    color: var(--dark-blue-100);
    transform: rotate(270deg);
    margin-left: 10px;
    font-family: ${TypographyENUM.lkSansBold};
    font-weight: var(--fw-bold);
`;

export const ChatWithUsIconWrapper = styled.div`
    display: flex;
    align-items: center;
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold)
    font-size: var(--fs-12);
    line-height: 16px;
    color: var(--dark-blue-100);
`;
