// @lk/ui-library/src/Component/V1/CheckoutMobile/BankOffers/BankOffers.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const OffersCard = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: var(--fs-16);
  width: 100%;
  background: var(--white);
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
  border-radius: var(--fs-12);
`;

export const OffersCardSection = styled.div`
  padding: var(--fs-20);
`;

export const OffersCardHeader = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 14px;
  line-height: var(--border-radius-sm);
  letter-spacing: -0.02em;
  color: var(--text);
  margin-bottom: 5px;
`;

export const OffersCardContent = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 10px;
  line-height: var(--fs-16);
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
`;

export const OffersShowMore = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-bold);
  font-size: var(--fs-12);
  line-height: var(--fs-20);
  display: flex;
  align-items: center;
  letter-spacing: -0.02em;
  text-decoration-line: underline;
  color: var(--dark-blue-100);
  cursor: pointer;
  margin-top: 5px;
`;

export const OffersSheetContent = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-12);
  line-height: var(--fs-20);;
  letter-spacing: -0.02em;
  color: var(--text);

  ul{
    padding-left: var(--fs-20);;
  }

  li{
    margin-bottom: var(--fs-20);;
  }
`;

export const OffersSheetHeader = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-16);
  line-height: var(--border-radius-sm);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  margin-bottom: var(--fs-20);;
`;
