// @lk/ui-library/src/Component/V1/Button/helper.ts
export const colors = {
	primary: {
		default: "--vivid-green-100",
		hover: "--white",
		text: "--dark-blue-100",
		disabledText: "--dark-blue-100",
		disabledBgColor: "--whitish-grey",
		pressedText: "--dark-blue-100",
		pressedBgColor: "--whitish-grey",
		hoverColor: "--vivid-green-100",
	},
	secondary: {
		default: "--dark-blue-100",
		hover: "--white",
		text: "--white",
		disabledText: "--white",
		disabledBgColor: "--dark-blue-50",
		pressedText: "--white",
		pressedBgColor: "--dark-blue-50",
		hoverColor: "--dark-blue-100",
	},
	newSecondary: {
		default: "--dark-blue-100",
		hover: "--white",
		text: "--white",
		disabledText: "--dark-blue-10",
		disabledBgColor: "--ghost-white",
		pressedText: "--white",
		pressedBgColor: "--dark-blue-50",
		hoverColor: "--dark-blue-100",
	},
	msitePrimary: {
		default: "--white",
		hover: "--white",
		text: "--text",
		disabledText: "--white",
		disabledBgColor: "--white",
		pressedText: "--text",
		pressedBgColor: "--white",
		hoverColor: "--text"
	},
	disableSignInColor: {
		default: "--vivid-green-100",
		hover: "--vivid-green-100",
		text: "--dark-blue-100",
		disabledText: "--dark-blue-100",
		disabledBgColor: "--whitish-grey",
		pressedText: "--dark-blue-100",
		pressedBgColor: "--vivid-green-100",
		hoverColor: "--dark-blue-100"
	},
	signInPrimary: {
		default: "--acetate-green-25",
		hover: "--white",
		text: "--text",
		disabledText: "--white",
		disabledBgColor: "--white",
		pressedText: "--text",
		pressedBgColor: "--white",
		hoverColor: "--text"
	},
};

export const fontSize = {
	small: "--fs-12",
	medium: "--fs-16",
	large: "--fs-20",
	extraLarge: "--fs-24"
};

export const spacing = {
	small: "--fs-12",
	medium: "--fs-16",
	large: "--fs-20",
	extraLarge: "--fs-24"
};

export const borderRadius = {
	small: "--border-radius-sm",
	medium: "--border-radius-md",
	large: "--border-radius-lg",
	extraLarge: "--border-radius-lg"
};
