// @lk/ui-library/src/Component/NewHtoFlow/HtoHomeHeader/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const Image = styled.img`
  width: 100%;
`;

export const HeaderContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  font-family: ${TypographyENUM.lkSansRegular};
  padding: 6px 12px;
  background: var(--white);
`;

export const PhoneIconNumberWrapper = styled.a`
    flex-direction: row;
    display: flex;
    align-items: center;
    text-decoration: none;
`;

export const LenskartBlueLogoWrapper = styled.div`

`;

export const LenskartBlueLogo = styled.img`
  width: 112px;
`;

export const IconWrapper = styled.div`
  color: var(--dark-blue-100); 
  font-family: ${TypographyENUM.lkSansMedium};
  font-weight: var(--fw-medium);
`;

export const TelePhoneNumber = styled.span`
    margin-left: 10px;
    font-style: normal;
    font-weight: var(--fw-medium);
    font-size: var(--fs-16);
    line-height: 24px;
    text-decoration: none;
    text-align: right;
    color: var(--dark-blue-100); 
    font-family: ${TypographyENUM.lkSansMedium}
`;
