// @lk/ui-library/src/Component/V1/AdvancedDropdown/MenuItem/styles.tsx
import styled from "styled-components";

export const MenuItRipple = styled.span`
	top: 0;
	left: 0;
	right: 0;
	bottom: 0;
	z-index: 0;
	overflow: hidden;
	position: absolute;
	border-radius: inherit;
	pointer-events: none;
	white-space: nowrap;
	letter-spacing: 0.00938em;
	user-select: none;
`;
// .menuit-ripple {
// 	top: 0;
// 	left: 0;
// 	right: 0;
// 	bottom: 0;
// 	z-index: 0;
// 	overflow: hidden;
// 	position: absolute;
// 	border-radius: inherit;
// 	pointer-events: none;
// 	white-space: nowrap;
// 	letter-spacing: 0.00938em;
// 	user-select: none;
//   }
export const MenuItListItem = styled.button`
	min-height: auto;
	width: auto;
	overflow: hidden;
	font-size: 14px;
	box-sizing: border-box;
	min-height: 48px;
	font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
	font-weight: 400;
	line-height: 1.5;
	padding-top: 6px;
	white-space: nowrap;
	letter-spacing: 0.00938em;
	padding-bottom: 6px;
	transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
	padding-left: 16px;
	padding-right: 16px;
	width: 100%;
	display: flex;
	position: relative;
	box-sizing: border-box;
	text-align: left;
	align-items: center;
	justify-content: flex-start;
	text-decoration: none;
	color: inherit;
	border: 0;
	cursor: pointer;
	margin: 0;
	outline: 0;
	position: relative;
	align-items: center;
	user-select: none;
	border-radius: 0;
	vertical-align: middle;
	-moz-appearance: none;
	text-decoration: none;
	background-color: transparent;
	-webkit-appearance: none;
	-webkit-tap-highlight-color: transparent;
	list-style-type: none;
	@media (max-width: 600px) {
		min-height: auto;
		font-size: 1rem;
	}
	&:disabled {
		background-color: #fff !important;
		color: #d3d3d3;
	}
	&:hover,
	&:focus {
		text-decoration: none;
		background-color: rgba(0, 0, 0, 0.04);
	}
`;
//   .menuit-list-item {
// 	min-height: auto;
// 	width: auto;
// 	overflow: hidden;
// 	font-size: 14px;
// 	box-sizing: border-box;
// 	min-height: 48px;
// 	font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
// 	font-weight: 400;
// 	line-height: 1.5;
// 	padding-top: 6px;
// 	white-space: nowrap;
// 	letter-spacing: 0.00938em;
// 	padding-bottom: 6px;
// 	transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
// 	padding-left: 16px;
// 	padding-right: 16px;
// 	width: 100%;
// 	display: flex;
// 	position: relative;
// 	box-sizing: border-box;
// 	text-align: left;
// 	align-items: center;
// 	// padding-top: 8px;
// 	// padding-bottom: 8px;
// 	justify-content: flex-start;
// 	text-decoration: none;
// 	color: inherit;
// 	border: 0;
// 	cursor: pointer;
// 	margin: 0;
// 	// display: inline-flex;
// 	outline: 0;
// 	// padding: 0;
// 	position: relative;
// 	align-items: center;
// 	user-select: none;
// 	border-radius: 0;
// 	vertical-align: middle;
// 	-moz-appearance: none;
// 	// justify-content: center;
// 	text-decoration: none;
// 	background-color: transparent;
// 	-webkit-appearance: none;
// 	-webkit-tap-highlight-color: transparent;
// 	list-style-type: none;
// 	@media (max-width: 600px) {
// 	  min-height: auto;
// 	  font-size: 1rem;
// 	}
// 	&:disabled {
// 	  background-color: #fff !important;
// 	  color: #d3d3d3;
// 	}
// 	&:hover,
// 	&:focus {
// 	  text-decoration: none;
// 	  background-color: rgba(0, 0, 0, 0.04);
// 	}
//   }
