// @lk/ui-library/src/Component/Common/GoogleLocationInput/GoogleLocationInput.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import GoogleLocationInput from "../GoogleLocationInput";

export default {
	title: "Components/Common/GoogleLocationInput",
	component: GoogleLocationInput,
	argTypes: {
		callback: generateArgTypes({
			type: "function",
			required: true,
			category: "Events",
			description: "Callback to access the value of input field",
			defaultValue: (value: string) => console.log(value),
		}),
		label: generateArgTypes({
			type: "string",
			required: false,
			category: "Components",
			description: "Label of the input field",
			defaultValue: "Label",
		})
	}
} as ComponentMeta<typeof GoogleLocationInput>;

const Template: ComponentStory<typeof GoogleLocationInput> = (args) => <GoogleLocationInput {...args} />;

export const Overview = Template.bind({
});

Overview.args = {
	label: "Label",
	callback: (value: string) => console.log({ callback: value }),
	configData: {
		GOOGLE_MAP_CONFIG: "{\"libraries\":[\"places\"],\"preventGoogleFontsLoading\":true,\"types\":[\"geocode\"],\"fields\":[\"address_components\",\"formatted_address\",\"geometry\"],\"restrictions\":{\"country\":[\"in\"]}}",
		GOOGLE_MAP_KEY: "AIzaSyAgBBoD8hH6dQpesH3FjcVeKgEeAqezDu4"
	}
};
