// @lk/ui-library/src/Component/V1/Cart/ApplyCoupon/ApplyCoupon.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import ApplyCoupon from "./index";
import { TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/v1/Cart/ApplyCoupon",
	component: ApplyCoupon,
	argTypes: {
		...generateGeneralArgs(),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of box",
			category: "Component",
			defaultValue: "100",
		}),
		headText: generateArgTypes({
			type: "string",
			required: true,
			description: "To set head text of any offer",
			category: "Component",
			defaultValue: "Apply Coupon",
		}),
		subText: generateArgTypes({
			type: "string",
			required: true,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "Save ₹100 with FLAT100",
		}),
		isApplied: generateArgTypes({
			type: "boolean",
			required: true,
			description: "To set whether coupon is applied or not",
			category: "Component",
			defaultValue: false,
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for Apply Coupon",
			category: "Events",
		}),
		onRemoveClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for Remove Applied Coupon",
			category: "Events",
		}),
	},
} as ComponentMeta<typeof ApplyCoupon>;

const Template: ComponentStory<typeof ApplyCoupon> = (args) => <ApplyCoupon {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "ApplyCoupon",
	width: "100",
	headText: "Apply Coupon",
	subText:"Save ₹100 with FLAT100",
	isApplied: false,
	onClick: () => {console.log("Apply Coupon Clicked");},
	onRemoveClick: () => {console.log("Remove Coupon Clicked");},
	font: TypographyENUM.defaultBook,
};
