// @lk/ui-library/src/Component/V1/Cart/ApplyCoupon/styles.tsx
import styled, { keyframes } from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

const shine = keyframes`
  100% {
    left: 125%;
  }
`;

export const Root = styled.div<{showAnimation: boolean; isMobileView?: boolean;}>`
${props => props.isMobileView ? "width: 100%;" : ""}
	position: relative;
	overflow: hidden;
	border-radius: 12px;
  box-shadow: rgb(99 99 99 / 20%) 0px 2px 8px 0px;

	&::before {
    background: -webkit-gradient(linear, left top, right top, from(rgba(211, 211, 211, 0)), to(rgba(211, 211, 211, .3)));
		content: "";
		display: ${({ showAnimation }) => showAnimation ? "block" : "none"};
		height: 100%;
		left: -75%;
		position: absolute;
		top: 0;
		transform: skewX(-25deg);
		width: 50%;
		// z-index: 2;
		animation: ${shine} 2s infinite;
	}
`;

export const ApplyCouponWrapper = styled.div <{ width: number } >`
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: var(--pd-16);
  gap: var(--spacing-lg);
  background: var(--white);
  // border: 1px solid var(--purple-haze);
  border-radius: var(--border-radius-xs);
`;

export const FlexWrapper = styled.div<{isMobileView?: boolean;}>`
  display: flex;
  align-items: center;
  width: 100%;
  justify-content: var(--space-between);
`;

export const FlexColWrapper = styled.div`
  /* display: flex; */
  flex-direction:column ;
  align-items: start;
  width: 100%;
  justify-content: var(--space-between);
  gap: var(--spacing-sm);
`;

export const LeftWrapper = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  justify-content: var(--space-between);
`;

export const RightWrapper = styled.div`
  display: flex;
  align-items: center;
  width: 30%;
  justify-content: flex-end;
  max-width: 100px;
  padding-right:10px;
  @media screen and (max-width:767px){
	max-width: 65px;
    padding-right: 0;
  }
`;

export const IconWrapperButton = styled.div<{isRTL: boolean}>`
  display: flex;
  align-items: center;
  padding: 4px var(--pd-8);
  border: 1px solid var(--dark-blue-50);
  border-radius: 100px;
  cursor: var(--pointer);
  ${props => props.isRTL ? "transform: rotate(180deg)" : ""};
`;

export const SubText = styled.div<{styleFont: TypographyENUM, isApplied: boolean, isRTL?: boolean}>`
  font-weight: var(--fw-regular);
  font-size: var(--fs-14);
  /* color: ${(props) => props.isApplied ? "var(--turquoise)" : "var(--dark-blue-75)"}; */
  color: var(--dark-blue-75);
  font-family: ${(props) => props.styleFont};
  /* span {
	${props => props.isRTL ? "display: flex; flex-direction: row-reverse; gap: 5px; align-items: center;" : ""};
  } */
  @media screen and (max-width:375px){
	font-size: var(--fs-13);
  }
`;

export const HeadText = styled.div<{styleFont?: TypographyENUM}>`
  /* font-weight: var(--fw-bold); */
  font-size: var(--fs-14);
  color: var(--dark-blue-100);
  font-family: ${({ styleFont }) => styleFont ? `${styleFont}`: "var(--font-lksans-bold)"};
`;
