// @lk/ui-library/src/Component/V1/PDP/NearStore/nearStore.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";
import { NearStore } from "./nearStore.component";

export default {
	title:"Components/V1/PDP/NearStore",
	component: NearStore,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: false,
			description: "Unique Id",
			category: "General",
			defaultValue: "near-by-store"
		}),
		font: generateArgTypes({
			type: "string",
			required: false,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.defaultBook,
			control: "select",
			options: TypographyENUM,
		}),
		storeImg: generateArgTypes({
			type: "string",
			required: false,
			description: "Image fo the store the be displayed",
			category: "Component",
			defaultValue: "https://static.lenskart.com/media/desktop/img/pdp/visit_store.png",
		}),
		storeInfo: generateArgTypes({
			type: "string",
			required: false,
			description: "Info Text of the store",
			category: "Component",
			defaultValue: "Please Share Your Location To See Nearby Stores",
		}),
		onClickStoreLocator: generateArgTypes({
			type: "function",
			required: true,
			description: "onClick handler for near by store",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show Store Right to left",
			category: "Status",
		}),
	}
} as ComponentMeta<typeof NearStore>;

const Template: ComponentStory<typeof NearStore> = args => <NearStore {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"store-info-card",
	storeImg: "https://static.lenskart.com/media/desktop/img/pdp/visit_store.png",
	onClickStoreLocator: () => null,
	storeInfo: "Please Share Your Location To See Nearby Stores",
	font:TypographyENUM.defaultBook
};
