// @lk/ui-library/src/Component/V1/MyAccount/Mobile/ExchangeCurrentStatus/ExchangeCurrentStatus.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import { ExchangeCurrentStatus } from "./ExchangeCurrentStatus.component";
import { Cross, Tick } from "../../../../../Icons";

export default {
	title: "Components/V1/MyAccount/Mobile/ExchangeCurrentStatus",
	component: ExchangeCurrentStatus,
	argTypes: {
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.lkSansRegular,
			control: "radio",
			options: TypographyENUM,
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			control: "object"
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left ",
			category: "Status",
			defaultValue: "false",
		})
	}
} as ComponentMeta<typeof ExchangeCurrentStatus>;

const Template: ComponentStory<typeof ExchangeCurrentStatus> = args => <ExchangeCurrentStatus {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	id: "exchange-return-details",
	font: TypographyENUM.lkSansRegular,
	status: [
		{
			"id": 1,
			"title": "Pickup Scheduled",
			"date": "7 Feb",
			"icon":<Tick />
		},
		{
			"id": 2,
			"title": "Pickup Cancelled",
			"date": "7 Feb",
			"icon":<Cross />,
		},
		{
			"id": 4,
			"title": "Store Visit Scheduled",
			"date": "7 Feb",
			"icon":<Tick />
		},
		{
			"id": 4,
			"title": "Pickup Cancelled",
			"date": "7 Feb",
			"icon":<Cross />,
		}
	],
	dataLocale: {
		PRODUCT_STATUS: "Current Status",
	},
};
