// @lk/ui-library/src/Component/V1/MyAccount/Mobile/ExchangeCurrentStatus/ExchangeCurrentStatus.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const ExchangeInfoWrapper = styled.div<{styleFont:TypographyENUM}>`
	flex-grow:1;
	display:flex;
	flex-direction:column;
	gap:var(--pd-10);
	font-family:${props => props.styleFont};
`;

export const ProductReturnWrapper = styled.div`
	background:#fff;
	padding:10px 0;
`;

export const ExchangeProductDes = styled.ul`
   padding-left:0px;
   margin:0 15px;
`;
export const StatusLI = styled.li`
   display:flex;
   justify-content:space-between;
   margin:0px 0px 10px 25px;
   padding:10px 0 10px 0px;
   border-bottom:1px solid #ddd;
   position:relative;
   &:last-child{
	border-bottom:none;
	margin-bottom:0px;
	padding-bottom:0px;
	i{
		&:before{
			display:none;
		}
	}
   }
   strong{
	font-size:var(--fs-12);
	/* font-weight: var(--fw-bold); */
	font-family: var(--font-lksans-bold);
   }
   span{
	font-size:var(--fs-12);
	font-weight: var(--fw-normal);
	font-family: var(--font-lksans-normal);
   }
   i{
	position:absolute;
	left:-25px;
	top:9px;
	border-radius:50%;
	width:18px;
	height:18px;
	padding:3px;
	background:#28c7bf;
	display:flex;
	align-items:center;
	justify-content:center;
	color:#fff;
	&:before{
		content: "";
		height: 41px;
		background: #28c7bf;
		left: 8px;
		width: 2px;
		top: 100%;
		position: absolute;
	}
	svg{
		width:10px;
		height:10px;
	}
   }
`;


export const ReturnTitle = styled.div`
	font-size:var(--fs-14);
	/* font-weight: var(--fw-bold); */
	text-align:left;
	padding:0 var(--pd-10) var(--pd-15) var(--pd-10);
	font-family: var(--font-lksans-bold);
`;
