// @lk/ui-library/src/Component/Common/MoneySaveModal/MoneyModal.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../StorybookHelper";
import MoneyModal from "./index";

export default {
	title: "Components/common/MoneyModal",
	component: MoneyModal,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		onClose: generateArgTypes({
			type: "function",
			required: true,
			description: "Function to close the modal",
			category: "Component",
			defaultValue: () => null,
		}),
		isVisible: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to show hide the modal",
			category: "Component",
			defaultValue: true,
		}),
		totalDiscount: generateArgTypes({
			type: "string",
			required: true,
			description: "To show totalDiscount",
			category: "Component",
			defaultValue: 200,
		}),
	}
} as ComponentMeta<typeof MoneyModal>;

const Template: ComponentStory<typeof MoneyModal> = (args) => <MoneyModal {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "MoneyModal",
	isVisible: true,
	totalDiscount: 200,
	closeModal: () => {console.log("Closed");}
};
