// @lk/ui-library/src/Component/V1/Cart/CartCard/CartCard.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import CartCard from "./index";
import { ComponentSizeENUM, TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/v1/Cart/CartCard",
	component: CartCard,
	argTypes: {
		...generateGeneralArgs(),
	},
} as ComponentMeta<typeof CartCard>;

const Template: ComponentStory<typeof CartCard> = (args) => <CartCard {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "cart-card",
	onAddClick: ()=> console.log("Item Add Clicked"),
	onDeleteClick: ()=> console.log("Item Delete Clicked"),
	componentSize: ComponentSizeENUM.large,
	dataLocale: {
		REMOVE: "Remove",
		DUPLICATE: "Duplicate",
		REPEAT: "REPEAT",
		GOLD_TEXT: "Buy 1 Get 1 Free on over 5000+ items, applicable everywhere",
		FINAL_PRICE: "Final Price",
		EXTRA_DISCOUNT: "Extra Discount"
	},
	font: TypographyENUM.lkSansBold,
	itemName: "Gunmetal Black Green Solid Full Rim Aviator John Jacobs JJ Tints JJ S13803 C2 Sunglasses",
	itemImg: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//j/o/john-jacobs-jj-s13803-c2-sunglasses_sunglasses_g_5404.jpg",
	itemModel: "JJ S13803",
	itemClassification: "sunglasses",
	itemQty: 1,
	showStrikeOffPrice: true,
	itemPriceObj:{
		itemName: "Matte Black Gunmetal Grey Transparent Full Rim Round Vincent Chase SLEEK STEEL VC E13784-C2 Eyeglasses",
		itemPrice: 999,
		showStrikeOffPrice: true,
		strikeOffPrice: 2000,
	},
	strikeOffPrice: 5000,
	itemPrice: 4000,
	isItemFree: true,
	showRibbon: true,
	ribbonMessage: "Free",
	showOfferMessage: false,
	offerMessage: "",
	price: {
		value: 999
	},
	itemDiscountedPrice: {
		value: 899
	},
};
