// @lk/ui-library/src/Component/V1/Cart/CartCard/styles.tsx
import styled from "styled-components";
import { devices } from "../../../../helper";
import { ComponentSizeENUM, TypographyENUM } from "../../../../Types/general";

export const CartCardWrapper = styled.div<{
  styledFont: TypographyENUM;
  styledSize: ComponentSizeENUM;
}>`
  display: flex;
  flex-direction: row;
  background: var(--white);
  border-radius: var(--border-radius-xs);
  /* padding: 0 16px 0 16px; */
  padding: var(--pd-30);
  padding-bottom: 0;
  font-family: ${(props) => props.styledFont};
  box-shadow: 0px 1px 4px rgb(0 0 0 / 8%);
  transition: box-shadow 0.2s ease-in-out;
  gap: 16px;
  /* @media ${devices.laptop} {
    flex-direction: row;
    padding: var(--pd-30) var(--pd-30) 0px var(--pd-30);
  } */
`;

export const DontKnowPower = styled.div`
  padding-bottom: 10px;
`;

export const ItemImageWrapper = styled.a`
  display: flex;
  flex-flow: column;
  justify-content: center;
  /* margin-bottom: 16px; */
  margin-top: auto;
  margin-bottom: auto;
  min-width: 236px;
  flex-basis: 46%;
  align-items: center;
  /* @media ${devices.laptop} {
    margin-bottom: auto;
  } */
`;

export const ItemImage = styled.img<{ styleIsGold: boolean; isRTL: boolean }>`
  ${(props) => (props.styleIsGold ? "width: 200px" : "")};
  height: 138px;
  padding-bottom: 15px;
  @media ${devices.laptop} {
    ${props => props.isRTL ? "margin-left: 20px" : "margin-right: 20px"};
    height: 113px;
  }
`;

export const ItemDetailSection = styled.div`
  display: flex;
  flex-flow: column;
  width: 100%;
  gap: 16px;
`;

export const ItemTitle = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-weight: 400;
  font-size: var(--fs-14);
  line-height: 24px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  text-transform: capitalize;

  @media ${devices.laptop} {
    font-size: var(--fs-16);
  }
`;

export const ItemPrice = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-size: var(--fs-14);
  color: var(--dark-blue-100);
`;

export const StrickOffText = styled.span<{isRTL: boolean}>`
  color: var(--dark-blue-10);
  font-weight: var(--fw-regular);
  font-family: var(--font-lksans-regular);
  ${props => props.isRTL ? "margin-left: var(--fs-6)" : "margin-right: var(--fs-6)"};
  font-size: var(--fs-14);
  text-decoration: line-through;
`;

export const FreeText = styled.span`
  font-family: var(--font-lksans-bold);
  /* font-weight: var(--fw-bold); */
  font-size: var(--fs-14);
  color: #0fbd95;
  letter-spacing: -0.02em;
  line-height: 20px;
  display: inherit;
`;

export const OfferText = styled.div<{isRTL: boolean}>`
  padding: 8px;
  height: 40px;
  background: linear-gradient(${props => props.isRTL ? "270deg" : "90deg"}, #f7f1de 0%, rgba(247, 241, 222, 0) 100%);
  display: flex;
  align-items: center;
  justify-content: center;
  color: #333368;
  letter-spacing: -0.02em;
  font-size: 12px;
  line-height: 18px;
  /* font-weight: 700; */
  font-family: ${TypographyENUM.lkSansBold};
  margin-right: auto;
  margin-left: -16px;

  @media ${devices.laptop} {
    border-radius: ${props => props.isRTL ? "0px 8px 0px 0px" : "8px 0px 0px 0px"};
    ${props => props.isRTL ? "margin-right: 0" : "margin-left: 0"};
  }
`;

export const HR = styled.hr`
  width: 100%;
  border: 1px dashed var(--purple-haze);
`;

export const ItemActions = styled.div`
  display: flex;
  flex-flow: column;
  gap: 20px;
  margin-bottom: 24px;
`;

export const Action = styled.span<{ rightSpacing: boolean; leftSpacing?: boolean; }>`
  font-size: var(--fs-14);
  text-decoration: underline;
  line-height: 24px;
  cursor: pointer;
   & {
    ${(props) => props.rightSpacing ? "margin-right: 16px;" : "margin-left: 16px;"}
    ${(props) => props.leftSpacing ? "margin-left: 16px;" : ""}
  	}
  color: var(--dark-blue-100);
`;

export const Pipe = styled.span`
  height: 20px;
  border-left: 1px solid var(--light-purple-10);
  display: none;
  @media ${devices.laptop} {
    display: block;
  }
`;

export const RowSection = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  flex-shrink: 0;
  width: 100%;
  gap: 20px;
  @media ${devices.laptop} {
    gap: 50px;
  }
`;

export const PriceWrapper = styled.div`
  display: flex;
  margin-top: 3px;
  @media ${devices.laptop} {
    margin-top: 0;
  }
`;

export const SubHeading = styled.div`
  color: var(--dark-blue-100);
  font-size: var(--fs-14);
  line-height: 20px;
  letter-spacing: -0.02em;
  font-family: var(--font-lksans-regular);
  text-transform: capitalize;
`;

export const FinalPriceText = styled.div`
  font-size: var(--fs-14);
  font-family: var(--font-lksans-bold);
  color: var(--dark-blue-100);
`;

export const FreeAmountText = styled.span`
color: #9999b3;
                    margin-right: 6px;
`;
export const FreePriceText = styled.span`
color:#0fbd95 ;
`;

export const RobinWrapper = styled.div`
  width: 115px;
  height: 150px;
  overflow: hidden;
  position: absolute;
  top: 1px;
  left: 0px !important;
  right: 0px;
  transform: rotate(359deg);
`;

export const RobinSpan = styled.span<{isRTL: boolean}>`
  position: absolute;
  display: block;
  width: 225px;
  background-color: #0fbd95;
  color: #fff;
  font-family: ${TypographyENUM.lkSansBold};
  text-shadow: 0 1px 1px rgb(0 0 0 / 20%);
  text-transform: uppercase;
  text-align: center;
  right: -27px;
  top: 25px;
  transform: rotate(-45deg);
  font-size: 12px;

  @media ${devices.laptop} {
    right: -88px;
    top: 30px;
    transform: ${props => props.isRTL ? "rotate(45deg)" : "rotate(320deg)"};
    left: -70px;
    width: 247px;
    display: flex;
    justify-content: center;
    overflow: hidden;
	font-size: 14px;
  }
`;

export const CartRobinWarp = styled.div`
  position: relative;
  width: 100%;
  border-radius: 12px;
  cursor: pointer;
  -webkit-box-shadow: 0px 1px 4px rgb(0 0 0 / 10%);
  box-shadow: 0px 1px 4px rgb(0 0 0 / 10%);
`;

export const CardContactTitle = styled.span`
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #000042;
  text-transform: capitalize;
  font-synthesis: none;
`;

export const CardContentTextWarpper = styled.div`
	border-radius: 8px;
	padding: 12px;
	background-color: #eee;
`;

export const CardContentText = styled.span`
color: #333;;
font-size: 14px;
line-height: 20px;
letter-spacing: -0.02em;
text-transform: capitalize;
`;

export const PowerRoot = styled.div<{mobileView: boolean}>`
    ${props => props.mobileView ? "border-bottom: 1px solid #ddd" : "border: 1px solid #ddd"};
    border-top: none;
    border-radius: 8px;
  `;
export const Header = styled.div`
  font-size: 12px;
  font-weight: 700;
  padding: 5px;
  padding-left: 15px;
  background: #eee;
  text-transform: capitalize;
`;
export const Table = styled.div`
    font-size: 12px;
    padding: 5px 15px 10px 15px;
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 20px;
	  justify-content: space-between;
`;
export const TableMobile = styled.div`
    font-size: 12px;
    padding: 5px 15px 10px 15px;
    display: flex;
    flex-direction: column;
    gap: 5px;
`;
export const TableStructure = styled.div`
    display: flex;
    flex-direction: column;
    gap: 5px;
    margin-top: 15px;
    text-align: right;
	align-items: center;
`;
export const TableStructureLeft = styled.div`
  display: flex;
  flex-direction: column;
  gap: 5px;
  margin-top: 15px;
`;

export const DeletePopUp = styled.div`
  position: fixed;
  left: 0;
  right: 0;
  bottom: 0;
  width: 100%;
  background: #ffffff;
  font-family: ${TypographyENUM.lkSansRegular};
  letter-spacing: -0.02em;
  color: #000042;
  /* height: 30vh; */
  border-radius: 16px 16px 0 0;
  z-index: 10;
  transition: position ease 500ms;
  padding-bottom: 10px;

  @media ${devices.laptop} {
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
    -ms-flex-direction: column;
    height: 100%;
    flex-direction: column;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    border-radius: 12px;
    justify-content: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    gap: 24px;
    isolation: isolate;
    position: absolute;
  }
`;

export const CartClose = styled.span`
  color: #000042;
  position: absolute;
  top: 5px;
  right: 10px;
  font-size: 16px;
  cursor: pointer;
  padding: 4px;
  font-weight: 700;
`;

export const DelHandling = styled.div`
  font-weight: 700;
  font-size: 14px;
  line-height: 24px;
  text-transform: capitalize;

  @media ${devices.laptop} {
    font-size: 16px;
  }
`;

export const DelText = styled.div`
  font-weight: 400;
  font-size: 12px;
  line-height: 20px;

  @media ${devices.laptop} {
    font-size: 14px;
  }
`;

export const CartPopUp = styled.div`
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  gap: 20px;
  justify-content: center;
  padding: 10px;
  @media ${devices.laptop} {
    padding: 0;
    justify-content: space-between;
  }
`;

export const PopUpConfirm = styled.div`
  display: flex;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: rgba(0, 0, 66, 0.01);
  color: #000042;
  border: 1px solid #737397;
  border-radius: 100px;
  padding: 4px 8px;
  font-weight: 700;
  width: 150px;
  cursor: pointer;

  &:hover{
	transition: transform ease 300ms;
	transform: translateY(-15%);
  }
`;

export const PopUpWishlist = styled.div`
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #000042;
  color: #e2e2ee;
  border: 1px solid #737397;
  border-radius: 100px;
  padding: 4px 8px;
  font-weight: 700;
  width: 150px;
  cursor: pointer;
  &:hover{
	transition: transform ease 300ms;
	transform: translateY(-15%);
  }
`;

export const PowerTable = styled.div`
//   max-width: 330px;
  width: fit-content;
  font-family: ${TypographyENUM.lkSansRegular};
`;

export const Buttons = styled.div`
  display: flex;
  justify-content: space-between;
  @media ${devices.laptop} {
    display: flex;
    justify-content: flex-start;
  }
`;

export const Wrapper = styled.div`
  padding: 10px;
  padding-top: 50px;
  display: flex;
  justify-content: space-around;
  align-items: center;
  @media ${devices.laptop} {
    padding: 0;
	text-align: center;
  }
`;

export const WrapperText = styled.div`
  padding: 10px;
`;

export const WrapperImg = styled.img`
  width: 30vw;
  height: fit-content;
  @media ${devices.laptop} {
    display: none;
  }
`;

export const OpaqueBackDrop = styled.div`
  @media ${devices.laptopMaxWidth} {
    background-color: rgba(33, 33, 33, 0.75);
    pointer-events: auto;
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 8;
  }
`;

export const TableStructureWrapper = styled.div`
	display: flex;
	flex-direction: row;
	gap: 30px;
`;

export const PowerText = styled.span`
	text-align: right;
`;

export const LeftSpan = styled.span`
  font-weight: 700;
`;
export const LeftSpanMobile = styled.span<{ header:string}>`
  font-weight: 700;
  float: left;
  text-align: right;
  font-family:${(props) => props.header === "true" ? TypographyENUM.lkSansBold : "inherit"};;
`;
export const RightSpan = styled.div`
  font-weight: 700;
`;
export const RightSpanMobile = styled.div<{ margin: string }>`
  margin-left: ${(props) => props.margin === "left" ? "20px" : "inherit"};
  width: ${(props) => props.margin === "left" ? "10vw" : "inherit"};
  font-weight: 700;
  float: right;
  text-align: right;
  font-family:${TypographyENUM.lkSansBold}
`;

export const PowerLabelsSpan = styled.span`
font-weight: 700;
`;

export const LoyaltyDelete = styled.div`
	color: #e34934;
	font-family: ${TypographyENUM.lkSansBold};
	/* font-weight: 700; */
	line-height: 20px;
	font-size: 14px;
	letter-spacing: -0.02em;
`;

export const ViewPres = styled.span<{padding: number}>`
    color: #00bac6;
    text-transform: uppercase;
    padding: ${props => props.padding}px;
    font-weight: 900;
    font-family: ${TypographyENUM.defaultBook};
    font-size: 12px;
`;

export const Flex = styled.div`
  display: flex;
`;

export const PdText = styled.span<{color: string, padding?: boolean}>`
color: ${props => props.color};
font-weight: 900;
font-family: ${TypographyENUM.defaultBook};
font-size: 12px;
${props => props.padding && "padding:10px;"}
line-height: 1.5em;
`;

export const DivTemp =styled.div`
padding: 15px;
`;
