// @lk/ui-library/src/Component/V1/MyOrderMobile/OrderListMobile/OrderListMobile.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const Container = styled.div`
	width: 100%;
	background-color: #fafbfc;
	padding: 10px 15px;
	margin-top: 10px;
	margin-bottom: 10px;
`;

export const OrderHeader = styled.div`
	div {
		&:first-child {
			display: flex;
			justify-content: space-between;
			padding-bottom: 12px;
			padding-top:8px;
		}
	}
`;

export const Span = styled.span`
	color: rgba(60, 60, 60, 0.54);
	font-size: 13px;
	font-family: var(--font-lksans-regular);
`;

export const Strong = styled.strong`
	color: rgb(51, 51, 51);
	font-family: var(--font-lksans-bold);
`;

export const OrderTotal = styled.div`
	border-top: 1px solid rgb(221, 221, 221);
	padding-top: 10px;
`;

export const ButtonNeedHelpOrderDetails = styled.div`
	display: flex;
	justify-content: var(--space-between);
	margin-top: 15px;
	width: 100%;
	font-family: var(--font-lksans-regular);
`;

export const ButtonNeedhelp = styled.button`
	font-family: var(--font-lksans-regular);
	letter-spacing: 0.4px;
	flex: 1;
	border-radius: 4px;
	background-color: #ffffff;
	font-size: var(--fs-13);
	padding: 12px;
	cursor: var(--pointer);
	-webkit-box-shadow: 0px 1px 4px rgb(0 0 0 / 20%);
    box-shadow: 0px 1px 4px rgb(0 0 0 / 20%);
	border: none;
	color: #424242;
`;

export const ButtonOrderDetails = styled.button<{noLeftMargin?: boolean}>`
	letter-spacing: 0.4px;
	font-family: var(--font-lksans-regular);
	flex: 1;
	border-radius: 4px;
	background-color: #ffffff;
	font-size: var(--fs-12);
	padding: 12px;
	${props => props.noLeftMargin ? "" : "margin-left:15px;"};
	cursor: var(--pointer);
	-webkit-box-shadow: 0px 1px 4px rgb(0 0 0 / 20%);
    box-shadow: 0px 1px 4px rgb(0 0 0 / 20%);
	border: none;
	color: #424242;
`;


export const ButtonContent = styled.div<{
  styledFont: TypographyENUM;
  isRTL?: boolean;
  fontSize?: string;
}>`
  font-family: ${(props) => props.styledFont};
  font-weight: 700;
  font-size: ${(props) => (props.fontSize ? props.fontSize : "16px")};
  color: var(--dark-blue-100);
  align-items: center;
  display: flex;
  gap: 10px;
  svg {
    transform: ${(props) => (props.isRTL ? "scaleX(-1)" : "")};
  }
`;

export const ButtonOuter = styled.div`
	margin-bottom: 30px;
`;
