// @lk/ui-library/src/Component/V1/MyOrderMobile/OrderListMobile/OrderListMobile.types.tsx
import { DataType } from "../../../../Types/general";

export interface OrderListingType {
    orderList: any;
    dataLocale: any;
    redisCommonData: any;
    onClickReturnExchange: (itemId: string) => void;
    formatDate: (createdAt:string, pattern:string, year:string) => string;
    onAddPowerClick?: ({ order, item } : {order: {[key: string]: unknown}, item: {[key: string]: unknown}}) => void;
    onPdClick?: ({ order, item } : {order: {[key: string]: unknown}, item: {[key: string]: unknown}}) => void;
    editPowerHandler?: ({ order, item } : {order: {[key: string]: unknown}, item: {[key: string]: unknown}}) => void;
    orderStatusRedis: any;
    getPendingStatus: (pendingOrderStatus: string,
        powerRequired: string,
        dataLocale: any,
        createdAt: number,
        isSeamless: boolean) => any;
    filterTrackingStatus: (item:any, early:string) => any;
    onNeedHelp: (params:any) => void
    confirmOrder: (params:any) => void;
    handlePayNow: (params:any) => void;
    returnEligibilityDetails: any;
    returnReason: any;
    handleManuallprescription: (id:string | number, powerType: string, item:any) => void;
    handleSavedprescription: (item:any, order:any) => void;
    handleUploadprescription: (item:any) => void;
    handleCallBack: (params:any) => void;
    handleSendEmail: (params:any) => void;
    handleOpenPrescriptionImageModal: (imageFileName: string) => void;
    onOrderDetailClick: (order:any) => void;
    configData: DataType;
    returnConfig: {[key: string]: any};
    removeDomainName: (a:string, b:string, c:string) => string;
    subdirectoryPath: string
    country: string,
    handleStudioFlowAppointmentClick: () => void
    addPowerCtaClick?: (val:boolean) => void;
    addPowerCtaGA?: (a:string,b:string,c:any,d:string) => void
    // returnSelection: {[key: string]: any};
}
