// @lk/ui-library/src/Component/V1/MyOrderMobile/ReturnRefundMobilePage/ExchangeBottomSheet/ExchangeBottomSheet.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import ExchangeBottomSheet from "./ExchangeBottomSheet.component";
import { action } from "@storybook/addon-actions";

export default {
	title: "Components/v1/MyOrder/ExchangeBottomSheet",
	component: ExchangeBottomSheet,
	args: {
	},
} as ComponentMeta<typeof ExchangeBottomSheet>;

const Template: ComponentStory<typeof ExchangeBottomSheet> = (args) => <ExchangeBottomSheet {...args}/>;

export const Overview = Template.bind({});
Overview.args = {
	show: true,
	exchangeOptions: [
		{
			imgURL: "https://static.lenskart.com/media/mobile/universal/images/returnExchange/Exchange_Old_SPSL.svg",
			value: "same-frame-same-lens"
		},
		{
			imgURL: "https://static.lenskart.com/media/mobile/universal/images/returnExchange/Exchange_Old_SPDL.svg",
			value: "same-frame-change-lens"
		},
		{
			imgURL: "https://static.lenskart.com/media/mobile/universal/images/returnExchange/Exchange_New_Product.svg",
			value: "change-frame-change-lens"
		}
	],
	item: {
		brandName: "Vincent Chase",
		itemURL: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e11259-c1-eyeglasses_vincent-chase-vc-e11259-c1-eyeglasses_g_0232_4.jpg",
		frameFeatures: ["Black", "Round", "Full Rim"],
		lensFeatures: ["Hydrophobic Anti-Glare"],
	},
	continueHandler: (value: string) => action("continue pressed, value")(value),
	closeHandler: () => action("close")("bottom sheet"),
};
