// @lk/ui-library/src/Component/V1/MyOrderMobile/ReturnRefundMobilePage/ExchangeBottomSheet/ExchangeBottomSheet.styles.tsx
import styled from "styled-components";

export const Container = styled.div`
	max-height: 90vh;
	padding: 10px 10px 70px;
	margin: 5px 0 0;
	line-height: 18px;
`;

export const ExchangeTitle = styled.div`
	font-family: var(--font-lksans-medium);
	margin: 0 0 5px;
	padding: 0 0 10px;
	font-size: 14px;
`;

export const ItemCard = styled.div`
	display: flex;
	border: 1px solid #e2e2e2;
    padding-top: 5px;
    padding-bottom: 5px;
    margin-bottom: 20px;
`;

export const ItemImage = styled.div`
	padding: 0 10px;
	flex-basis: 30%;
	max-width: 143px;
	img {
		width: 100%;
	}
`;

export const ItemInfo = styled.div`
	color: #999999;
	font-family: var(--font-lksans-regular);
`;

export const ItemBrand = styled.div`
	font-size: 13px;
	color: #333333;
	margin-bottom: 5px;
`;

export const Frame = styled.div`
	font-size: 11px;
`;

export const Lens = styled.div`
	font-size: 11px;
`;

export const RadioTitle = styled.div`
	font-family: var(--font-lksans-medium);
	padding: 0 0 10px;
	font-size: 14px;
	line-height: 16px;
`;

export const RadioContainer = styled.div`
	border: 1px solid #ddd;
	border-radius: 4px;
	font-family: var(--font-lksans-regular);
	background-color: #fff;
`;

export const ButtonContainer = styled.div`
	background: #ffffff;
    padding: 8px;
    width: 100%;
    position: fixed;
	border: 1px solid #eaeff4;
	bottom: 0;
	left: 0;
    width: 100vw;
    z-index: 3;
`;

export const Button = styled.div<{ exchangeValue: string | undefined }>`
	text-align: center;
    text-transform: uppercase;
    border-radius: 4px;
    letter-spacing: 1px;
	padding: 10px 0;
	background-color: ${props => props.exchangeValue ? "#18cfa8":"#DDDDDD"};
	font-family: var(--font-lksans-regular);
	color: #ffffff;
	line-height: 16px;
	div{
		&:first-child {
			font-size: 15px;
		}
		&:last-child {
			font-size: 11px;
			text-transform: none;
		}
	}
`;
