// @lk/ui-library/src/Component/V1/CheckoutMobile/StoreCredit/StoreCredit.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const StoreCreditSection = styled.div`
  display: flex;
  flex-direction: column;
  padding: var(--fs-16);
  background: var(--white);
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
  border-radius: 12px;
  margin: 24px 0;
`;

export const StoreCreditInnerHead = styled.div`
  display: flex;
  flex-direction: flex;
  justify-content:space-between;
  align-items: center;
`;

export const Header = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 14px;
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--text);
`;

export const CreditBottomSheet = styled.div`
  padding: var(--fs-16);
`;

export const CreditBottomSheetHeader = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-16);
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);

  margin-bottom:20px;
`;

export const InputWrapper = styled.div`
  margin-bottom:15px;
`;

export const ErrorMessage = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-12);
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  color: #E34934;
`;

export const AppliedCreditsBar = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-12);
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  color: #333368;
  display: flex;
  justify-content: space-between;
  padding-top: 8px;
`;

export const AppliedCreditsCode = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-12);
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  display: flex;
  justify-content: space-between;
  color: #333368;
  width: 65%;
`;

export const AppliedCreditsCodeMain = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-12);
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  color: #333368;
`;

export const AppliedCreditsPrice = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-bold);
  font-size: var(--fs-14);
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  color: #333368;
`;

export const AppliedCreditsRemoveButton = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-bold);
  font-size: var(--fs-12);
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  color: #333368;
  text-transform: capitalize;
  text-decoration: underline;
  margin-left: auto;
`;

export const AppliedCreditsRemoveButtonMain = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-bold);
  font-size: var(--fs-12);
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  color: #333368;
  text-transform: capitalize;
  text-decoration: underline;
`;

export const AppliedCreditsInfo = styled.div`
  font-weight: 400;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #333368;
  font-size: var(--fs-10);
`;

export const AppliedCreditWrapper = styled.div`
margin:8px 0px;
`;

export const ImgWrapper = styled.div<{show: boolean}>`
  visibility: ${props => props.show ? "visible" : "hidden"};
`;
