// @lk/ui-library/src/Component/Common/PhoneCaptureModal/PhoneCaptureModal.styles.tsx
import styled from "styled-components";
import { DeviceTypes, TypographyENUM } from "../../../Types/general";

export const InputField = styled.input<{
    error: boolean;
    disabled: boolean;
    isRTL?: boolean;
    deviceType?:string
  }>`
  ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "border-radius: 4px":"border-radius: 0px 50px 50px 0px"};
  ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "border: 1px solid #00000077":"border: 1px solid var(--white)"};
  ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "background: none":"background: var(--white)"};
  ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "width: 100%":"width: 80%"};
  ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "var(--dark-grey)":"color: var(--grey)"};
  cursor: ${(props) => (props.disabled ? "not-allowed" : "auto")};
  text-align: ${(props) => (props.isRTL ? "right" : "")};
  direction: ${(props) => (props.isRTL ? "rtl" : "")};
  font-size: 15px;
  padding: 5px;
  display: block;
  outline: none;
    &::placeholder {
      color: var(--placeholder);
    }
    user-select: initial;
    &::before,
    &::after {
      user-select: initial;
    }
  `;

export const CloseButton = styled.div<{deviceType?:string}>`
    position: absolute;
    top: 15px;
    right: 10px;
    display: flex;
    font-size: 20px;
    flex-direction: column;
    width: 20px;
    height: 20px;
    cursor: pointer;
    .left,
    .right {
      height: 2px;
      width: 100%;
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "background: var(--black)":"background: var(--white)"};
    }
    .left {
      transform: rotate(45deg) translateY(1.3px);
    }
    .right {
      transform: rotate(-45deg) translateY(-1.2px);
    }
  `;

export const DiscountContainer = styled.div<{deviceType?:string,offerUnlocked?:boolean}>`
    text-align: center;
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "width: 115%":"width: 90%"};
    ${({ deviceType,offerUnlocked })=> deviceType !== DeviceTypes.DESKTOP && offerUnlocked ? "width:100%":""};
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "font-weight: 500":"font-weight: 600"};
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "font-size: 36px":"font-size: 14px"};
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "letter-spacing: 0.15px":"letter-spacing: 0.15px"};
    color: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "":"var(--white)"};
    text-transform: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "":"uppercase"};
    margin-bottom: ${({ deviceType, offerUnlocked })=> deviceType=== DeviceTypes.DESKTOP && offerUnlocked ? "20px":""};
    ${({ deviceType })=> deviceType !== DeviceTypes.DESKTOP ? "margin-bottom:10px":""};
    font-family: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP?TypographyENUM.defaultBook:TypographyENUM.defaultHeavy};
  `;

export const Container = styled.div<{deviceType?:string}>`
    width: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "400px":"100%"};
    // ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "height: 300px":""};
    padding-top: 40px;
	padding-bottom:20px;
    margin: auto;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    font-family: ${TypographyENUM.defaultBook};
  `;

export const InputContainer = styled.div<{deviceType?:string}>`
    display: flex;
    margin: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "20px":"10px"} 0px;
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "width: 90%":"width: 82%"};
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "":"justify-content: center"};
    // margin-top: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "10px":"20px"};
  `;

export const PhoneCode = styled.span<{deviceType?:string}>`
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "border: 1px solid #00000077":""};
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "":"background: var(--white)"};
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "":"color: var(--dark-grey)"};
    font-size: 15px;
    margin-right: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "5px":"0px"};
    border-radius: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "4px":"50px 0px 0px 50px"};
    display: flex;
    justify-content: center;
    align-items: center;
    padding: 5px;
    border-right: 1px solid #00000077;
    `;

export const UnlockOfferContainer = styled.div`
margin-bottom: 20px;
`;

export const UnlockButton = styled.button<{deviceType?:string,disabled:boolean}>`
    color: #ffffff;
    border-radius: 4px;
    cursor: pointer;
    background-color: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "#00b0bf":"#18cfa8"};
    border: 1px solid ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "#00b0bf":"#18cfa8"};
    ${({ disabled })=> disabled? "pointer-event: none;background-color: var(--dark-grey);border:1px solid var(--dark-grey)":""};
    width: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "90%":"75%"};
    padding: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "10px 20px":"5px 10px"};
    font-size: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "1.5em":"16px"};
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "":" margin-top: 1px"};
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "":"margin-bottom: 15px"};
  `;

export const ErrorIndicator = styled.p<{deviceType?:string}>`
  text-align: left;
  font-size: 13px;
  letter-spacing: 0.11px;
  margin-bottom: 20px;
  margin-left: -45px;
  width: 55%;
  color: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "#f30f0f":"#f29a9a"};
  margin-top: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "-20px":"-5px"};
  margin-bottom: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "20px":"5px"};
  margin-left: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "-40px":"0px"};

`;

export const TermAndPP = styled.div<{deviceType?:string}>`
    text-align: center;
    -webkit-font-smoothing: antialiased;
     ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "letter-spacing: 0.11px":""};
     ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "":"width: 95%"};
    font-size: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "13px":"10px"};
    color: ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "#33333377":"var(--white)"};
    ${({ deviceType })=> deviceType=== DeviceTypes.DESKTOP ? "":"font-weight: 700"};
	padding-bottom:30px;
  `;

export const MobileContainerPhoneCapture = styled.div`
height: 225px;
position: fixed;
width: 100%;
bottom: 0;
background-color: rgba(33, 33, 33, 0.85);
z-index: 9999999999 !important;
`;

export const BackdropContainer = styled.div`
  background: rgba(33, 33, 33, 0.7);
  width: 100%;
  height: 100%;
  position: fixed;
  z-index: 9999999999 !important;
  top: 0;
`;
