// @lk/ui-library/src/Component/V1/MyOrderMobile/ReturnRefundMobilePage/ReturnRefundRadio/ReturnRefundRadio.styles.tsx
import styled from "styled-components";

export const Container = styled.div`
	border: 1px solid #ddd;
	border-radius: 4px;
	font-family: var(--font-lksans-regular);
	background-color: #fff;
`;

export const Radio = styled.div`
	display: flex;
	border-bottom: 1px solid #ddd;
	&:last-child {
		border: none;
	}
	padding: 10px;
	gap: 10px;
	line-height: 1.5;
	input {
		display: inline-block;
		margin-top: 5px;
		align-self: flex-start;
	}
`;

export const Label = styled.label`
	display: flex;
	gap: 15px;
	align-items: flex-start;
`;

export const Icon = styled.div`
	width: 36px;
	flex-basis: 20%;
	margin-top: 2px;
	img {
		width: 100%;
	}
`;

export const Data = styled.div<{fade?: boolean;}>`
${props => props.fade ? `
	opacity: 0.4;
` : ""}
`;

export const DataHeader = styled.div`
	span {
		font-family: var(--font-lksans-bold);
		font-size: 13px;
	}
	display: flex;
	align-items: center;
	gap: 20px;
`;

export const Tag = styled.div`
	border: 1px solid #18cfa8;
	color: #18cfa8;
	text-transform: uppercase;
    padding: 2px 3px 1px 3px;
    background: rgba(24, 207, 168, .1);
	font-size: 7px;
	font-family: var(--font-lksans-bold);
`;

export const DataBody = styled.div`
	font-size: 11px;
	color: #999999;
`;

export const List = styled.ul`
	padding-left: 15px;
`;

export const DivError = styled.div<{ error: boolean }>`
  cursor: ${(props) => (props.error ? "" : "pointer")};
`;
export const DivFlex = styled.div`
  display: flex;
  align-items: center;
`;
export const ErrorSpan = styled.span`
color: #f00;
`;
