// @lk/ui-library/src/Component/V1/PDP/CheckDeliveryOption/CheckDeliveryOption.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";
import { CheckDeliveryOption } from "./CheckDeliveryOption.component";

export default {
	title: "Components/V1/PDP/CheckDeliveryOption",
	component: CheckDeliveryOption,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			defaultValue:{
				ENTER_PIN_CODE: "Enter Pin Code",
				CHECK: "CHECK",
				DELIVERY_OPTIONS_TO: "Delivery Options To",
				STANDARD_DELIVERY: "Standard Delivery",
				FREE: "Free",
				GET_IT_DELIVERED_BY: "Get it delivered by",
				CHANGE: "Change",
				PLEASE_ENTER_VALID_PINCODE: "Please Enter Valid Pincode",
			},
			control:"object"
		}),
		deliveryDate: generateArgTypes({
			type: "number",
			required: true,
			description: "delivery date",
			category: "Component",
		}),
		deliveryOption: generateArgTypes({
			type: "string",
			required: true,
			description: "delivery option",
			category: "Component",
		}),
		dispatchDate: generateArgTypes({
			type: "number",
			required: true,
			description: "dispatch date",
			category: "Component",
		}),
		shippingCharges: generateArgTypes({
			type: "number",
			required: true,
			description: "shipping charges",
			category: "Component",
		}),
		showEstimateDelivery: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show estimate delivery or not",
			category: "Component",
		}),
		isLoading: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show estimate delivery or not",
			category: "Component",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean Text left to right",
			category: "Status",
			defaultValue: "false",
		}),

	}
} as ComponentMeta<typeof CheckDeliveryOption>;

const Template: ComponentStory<typeof CheckDeliveryOption> = (args) => <CheckDeliveryOption {...args} />;

export const checkPincodeView = Template.bind({});
checkPincodeView.args = {
	id: "check-delivery-option",
	dataLocale: {
		ENTER_PIN_CODE: "Enter Pin Code",
		CHECK: "CHECK",
		DELIVERY_OPTIONS_TO: "Delivery Options To",
		STANDARD_DELIVERY: "Standard Delivery",
		FREE: "Free",
		GET_IT_DELIVERED_BY: "Get it delivered by",
		CHANGE: "Change",
		PLEASE_ENTER_VALID_PINCODE: "Please Enter Valid Pincode",
	},
	dispatchDate: null,
	shippingCharges: null,
	font:TypographyENUM.defaultBook,
	showEstimateDelivery:false,
	isLoading:false
};

export const estimateDeliveryView = Template.bind({});

estimateDeliveryView.args = {
	...checkPincodeView,
	showEstimateDelivery:true,
	deliveryDate: 1661126400000,
	deliveryOption: "STANDARD",
};

export const loadingState = Template.bind({});

loadingState.args = {
	...checkPincodeView,
	isLoading:true
};
