// @lk/ui-library/src/Component/V1/PDP/CheckDeliveryOption/CheckDeliveryOption.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const CheckDeliveryContainer = styled.div<{styleFont:TypographyENUM}>`
	padding-top: 12px;
	background-color:var(--white);
	width:100%;
	font-size:var(--fs-13);
	font-family:${props => props.styleFont};
`;

export const DeliveryOptionsContainer=styled.div`
	display:flex;
	flex-direction:column;
	gap:var(--spacing-md);
	padding:var(--pd-10) 0;
	color:var(--light-black);
`;

export const DeliveryOptionTextWrapper=styled.div<{isRTL: boolean}>`
	display:flex;
	flex-wrap:wrap;
	justify-content:var(--space-between);
	align-items:center;
	flex-direction: ${props => props.isRTL ? "row-reverse": "row"}
`;

export const DeliveryOptionText=styled.div`
	display:flex;
	flex-direction:column;
	gap:var(--spaing-sm);
`;

export const MainText=styled.p`
	margin:0;
`;

export const SubText=styled.p`
	margin:0;
	color: var(--gray);
`;

export const ChangeButton=styled.button`
	background:none;
	color:var(--turquoise);
	border:none;
	text-trasform:uppercase;
	cursor: var(--pointer);
	font-size:var(--fs-12);
`;

export const StandardDeliveryWrapper=styled.div<{isRTL: boolean}>`
	display:flex;
	flex-wrap:wrap;
	align-items:center;
	gap:var(--spacing-md);
	flex-direction: ${props => props.isRTL ? "row-reverse": "row"}
`;

export const DeliveryLogo=styled.img`
	min-width: 35px;
`;

export const StandardDeliveryText=styled.div<{isRTL: boolean}>`
	flex-grow: ${props => props.isRTL ? 0 : 1};
`;

export const ErrorMessage = styled.p`
	color:var(--error);
	margin:0;
	font-size:var(--fs-13);
`;

export const InputWrapper = styled.div`
	display:flex;
	flex-direction:column;
	gap:var(--pd-5);
	#input-pincode{
		input{
			border:1px solid var(--serene-gray);
			padding: 5px 10px;
			height: 40px;
            border-radius: 4px;
			font-size: 13px;
		}
		label{
			padding: 8px 3px;
			top: 6px;
			background: #fff;
			left: 8px;
		}
		button{
			bottom:10px;
			right:10px;
		}
	}
`;

export const LoaderWrapper = styled.div`
	height:min-content;
	position:relative;

	.no-overlay{
		display:none;
	}

	.loader{
		position:static;
		height:50px; // wapper height
	}

	img{
		position: static;
		transform: perspective(1px) translateY(-25%);
		height:100px; //image height
	}
`;
