// @lk/ui-library/src/Component/NewHtoFlow/ReviewsAndComment/ReviewsAndComment.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import ReviewsAndComment from "./index";


export default {
	title: "NewHtoFlow/ReviewsAndComment",
	component: ReviewsAndComment,
	argTypes: {}
} as ComponentMeta<typeof ReviewsAndComment>;

const Template: ComponentStory<typeof ReviewsAndComment> = (args) => <ReviewsAndComment {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		LOAD_MORE: "Load More",
		LOAD_LESS: "Load Less"
	},
	commentsInfo: [
		{
			image: "https://static.lenskart.com/media/desktop/img/HTO/ReviewPersonOne.svg",
			rating: "4",
			name: "Jagdish yadav",
			comment: "An absolutely excellent place to pick up glasses. This space boasts of 2 floor with a massive range of frames to pick from. The place is also very reasonably priced."
		},
		{
			image: "https://static.lenskart.com/media/desktop/img/HTO/ReviewPersonOne.svg",
			rating: "4",
			name: "Jagdish yadav",
			comment: "An absolutely excellent place to pick up glasses. This space boasts of 2 floor with a massive range of frames to pick from. The place is also very reasonably priced."
		}
	]
};
