// @lk/ui-library/src/Component/NewHtoFlow/ReviewsAndComment/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const ReviewsAndCommentWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 16px;
    gap: 16px;
    background: #FBF9F7;
    border-radius: 12px;
`;

export const RatingNameImageWrapper = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
    gap: 8px;

`;

export const ImageWrap = styled.div`
`;


export const RatingNameWrap = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin-left: 10px;
    gap: 4px;
`;

export const RatingName = styled.div`
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-10);
    line-height: 12px;
    color: var(--dark-blue-100);
`;

export const RatingComment = styled.div`
    font-family: ${TypographyENUM.lkSansMedium};
    font-style: normal;
    font-weight: var(--fw-medium);
    font-size: var(--fs-12);
    line-height: 16px;
    color: var(--dark-blue-100);
`;

export const ReviewsAndCommentContainer = styled.div`
    display: flex;
    flex-direction: column;
    gap: 12px;
    background: #FFFFFF;
`;

export const LoadMore = styled.div`
    font-family: ${TypographyENUM.lkSansMedium};
    font-style: normal;
    font-weight: var(--fw-medium);
    font-size: var(--fs-12);
    line-height: 16px;
    color: var(--dark-blue-100);
    padding: 10px 0px 0px;
`;
