// @lk/ui-library/src/Component/V1/StudioFlow/SearchBox/SearchBox.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { SearchBox } from "./SearchBox.component";
import { action } from "@storybook/addon-actions";

export default {
	title: "Components/v1/StudioFlow/SearchBox",
	component: SearchBox,
	argTypes: {
		onSearch: generateArgTypes({
			type: "function",
			category: "Action",
			required: true,
			description: "Handler which triggers the search event",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			category: "General",
			required: true,
			description: "Right to Left or Left to Right?"
		}),
	},
} as ComponentMeta<typeof SearchBox>;

const Template: ComponentStory<typeof SearchBox> = (args) => (
	<SearchBox {...args} />
);
export const Overview = Template.bind({});

Overview.args = {
	onSearch: (searchValue: string) => action("onSearchHandler: " + searchValue)("Fired"),
	isRTL: false,
	dataLocale: {
		SEARCH_FOR_A_STORE: "Search for a store"
	}
};
