// @lk/ui-library/src/Component/V1/Checkout/StoreAddress/index.tsx
import React from "react";
import styled from "styled-components";

const Root = styled.div`
  display: flex;
  justify-content: space-between;
  margin: 20px;
  width: 90%;
`;

const AddressContainer = styled.div`
  width: 28vw;
`;

const Name = styled.div`
  font-weight: 700;
  font-size: 14px;
  margin-bottom: 3px;
`;

const Address = styled.div`
  font-weight: 500;
  color: #999999;
  font-size: 12px;
`;

const Timing = styled.div`
  margin-top: 10px;
  display: flex;
  align-items: center;
  gap: 5px;
  font-size: 14px;
`;

const Direction = styled.div`
  color: #18cfa8;
  font-size: 12px;
  font-weight: 600;
  margin: 4px auto;
`;
const Km = styled.span`
  color: #999999;
  font-size: 12px;
`;

const Box = styled.span`
  display: flex;
  flex-direction: column;
  align-items: center;
`;
const Lightdiv = styled.span`
   color: #999999;
   font-size: 12px;
`;

const getTime = (time: string) => {
	if (!time) return "";
	const a = time.split(":");
	return `${a[0]}:${a[1]}`;
};

const StoreAddress = ({ 
	dataLocale = {
		TIMING: "Timings:",
		DIRECTIONS: "DIRECTIONS",
		CALL_STORE: "CALL STORE",
		ALL_DAYS: "ALL_DAYS"
	},
	storeInfo }: any) => {
	return (
		<Root>
			<AddressContainer>
				<Name>{storeInfo.name}</Name>
				<Address>{storeInfo.address}</Address>
				<Timing>
					<span>{dataLocale.TIMING} </span>
					<Address>
						{getTime(storeInfo.openingTime)} {"AM"} {"-"}{" "}
						{getTime(storeInfo.closingTime)} {"PM"}
					</Address>
					<Lightdiv>
						{"("}
						{dataLocale.ALL_DAYS}
						{")"}
					</Lightdiv>
				</Timing>
			</AddressContainer>
			<Box>
				<img
					alt={"DIRECTIONS"}
					width={40}
					aria-label="Click Here For Directions"
					className="sa-img cursor-pointer"
					src="https://static.lenskart.com/media/mobile/universal/images/returnExchange/Directions.svg"
				/>
				<Direction>{dataLocale.DIRECTIONS}</Direction>
				<Km>{storeInfo.distance.humanReadable}</Km>
			</Box>
			<Box>
				<img
					alt={"CALL_STORE"}
					className="sa-img"
					src="https://static.lenskart.com/media/mobile/universal/images/returnExchange/CallStore.svg"
				/>
				<Direction>{dataLocale.CALL_STORE}</Direction>
				<Km>{storeInfo.telephone}</Km>
			</Box>
		</Root>
	);
};

export { StoreAddress };
