// @lk/ui-library/src/Component/V1/CheckoutMobile/DeleteAddressModal/DeleteAddressModal.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { DeleteAddressModal } from "./DeleteAddressModal.component";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/v1/CheckoutMobile/DeleteAddressModal",
	component: DeleteAddressModal,
	argTypes: {
		deleteAddress: generateArgTypes({
			type: "function",
			required: true,
			description: "deleteAddress handler to delete existing address",
			category: "Events",
		}),
		cancel: generateArgTypes({
			type: "function",
			required: true,
			description: "cancel turnoff delete Address Modal",
			category: "Events",
		}),
		show: generateArgTypes({
			type: "boolean",
			required: true,
			description: "This controls whether to show the modal or not",
			category: "Component",
			defaultValue: true
		}),
		title: generateArgTypes({
			type: "string",
			required: true,
			description: "Title of the modal",
			category: "Component",
			defaultValue: "Delete address ?"
		}),
		description: generateArgTypes({
			type: "string",
			required: true,
			description: "Description of the modal",
			category: "Component",
			defaultValue: "Are you sure you want to delete this address ?"
		})
	}
} as ComponentMeta<typeof DeleteAddressModal>;

const Template: ComponentStory<typeof DeleteAddressModal> = (args) => <DeleteAddressModal {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale:{
		DELETE_ADDRESS_QUESTION_MARK: "Delete address ?",
		ARE_YOU_SURE_YOU_WANT_TO_DELETE_ADDRESS: "Are you sure you want to delete this address ?",
		DELETE: "Delete",
		CANCEL: "Cancel"
	},
	font:TypographyENUM.lkSansRegular,
	id:2,	
};
