// @lk/ui-library/src/Component/V1/CheckoutMobile/DeleteAddressModal/styles.tsx
import { animated } from "react-spring";
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const DeleteAddressModalContainer = styled.div<{ styleFont: TypographyENUM; show: boolean}>`
	font-family:${props => props.styleFont};
    font-style: normal;
	display: flex;
    ${({ show }) => !show && "display: none;"}
    flex-direction: column;
    align-items: center;
    justify-content: center;
    padding: 24px;
    gap: 16px;
    position: fixed;
    inset: 0;
	#delete-address{
		background: rgba(0, 0, 66, 0.01);
		border: 1.5px solid #000042;
		color: #000042;
	}
	#cancel-delete-address{
		background: #11DAAC;
		color: #000042;
	}
`;

export const Backdrop = styled(animated.div)`
    background: #00004280;
    position: fixed;
    inset: 0;
    z-index: 1;
`;

export const Content = styled(animated.div)`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    padding: 24px;
    gap: 16px;
    max-width: 326px;
    background: var(--white);
    box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
    border-radius: 12px;
    z-index: 9999;
`;

export const Title = styled.div<{styleFont:TypographyENUM}>`
    font-family:${props => props.styleFont};
    font-style: normal;
    font-weight: var(--fw-regular);
    font-size: var(--fs-20);
    line-height: 32px;
    letter-spacing: -0.02em;
    color: var(--dark-blue-100);

`;

export const Description = styled.div<{styleFont:TypographyENUM}>`
   font-family:${props => props.styleFont};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-12);
    line-height: 18px;
    letter-spacing: -0.02em;
    color: var(--dark-blue-75);
`;

export const ButtonWrapper = styled.div`
    display: flex;
    gap: 16px;
    width: 100%;
`;

export const ContentWrapper = styled.div`
    display: flex;
    gap: 8px;
    flex-direction: column;
`;
