// @lk/ui-library/src/Component/V1/SignInForm/SignInForm.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import { TypographyENUM } from "../../../Types/general";
import SignInForm from "./index";

export default {
	title:"Components/V1/SignInForm",
	component: SignInForm,
	argTypes: {
		...generateGeneralArgs(),
		onSignIn: generateArgTypes({
			type: "function",
			required: false,
			description: "onSignIn handler for sign in form",
			category: "Events",
		}),
		onProceed: generateArgTypes({
			type: "function",
			required: false,
			description: "onProceed handler for sign in form",
			category: "Events",
		}),
		countryCode: generateArgTypes({
			type: "string",
			required: true,
			description: "country code",
			category: "Component",
			defaultValue: "+91",
		}),
		showOTPView: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show OTP view",
			category: "Component",
			defaultValue: false,
		}),
		isLoginSuccess: generateArgTypes({
			type: "boolean",
			required: true,
			description: "if login is successful",
			category: "Component",
			defaultValue: false,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show to right to left",
			category: "Status",
			defaultValue: false,
		}),
	}
} as ComponentMeta<typeof SignInForm>;

const Template: ComponentStory<typeof SignInForm> = args => <SignInForm {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"signin-form",
	dataLocale:{
		"MOBILE_EMAIL_TEXT": "Mobile / Email",
		"PROCEED": "Proceed",
		"T_C_APPLY": "T&C Apply",
		"WE_HAVE_SENT_AN_OTP": "We have sent an OTP to ",
		"PLEASE_ENTER_OTP": "Please enter OTP",
		"EDIT_NUMBER": "Edit Number",
		"BACK": "Back",
		"RESEND": "Resend",
	},
	font:TypographyENUM.defaultBook,
	onSignIn: () => console.log("api called and user signed in"),
	onProceed: () => console.log("api called and otp sent"),
	isLoginSuccess: false,
	showOTPView: false,
	countryCode:"+91",
	signInStatus:{
		type: null,
		showOTP: false,
		optSent: false,
		showPassword: false,
		passwordEntered: false,
		signInStatus: false,
		isLoading: false,
		isError: false,
		errorMessage: null
	}
};
