// @lk/ui-library/src/Component/V1/SignInForm/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const IconWrapper = styled.div`
	display:flex;
	cursor:pointer;
`;

export const EmailDisplay = styled.div`
	border: none;
	padding: 12px;
	width: 100%;
	display: block;
	outline: none;
	border-radius: 5px;
	color: #828282;
	margin-bottom: 18px;
	letter-spacing: 1px;
	font-family: futurastd-book;
	font-synthesis: none;
	font-size: 12px;
	position: relative;
	background-color: rgba(239, 239, 239, 0.5);
	display: flex;
    align-items: center;
    justify-content: space-between;
    font-size: 12px;
	button {
		outline: none;
		border: none;
		background: transparent;
		color: #3DB1A6;
		font-weight: normal;
		padding: 2px;
		letter-spacing: .7px;
		cursor: pointer;
	}
`;
export const InlineLoader = styled.img`
	display: block;
	width: 50px;
	margin: 0 auto;
`;

export const ErrorField = styled.div`
    color: #d90502;
    font-size: 14px;
`;

export const BackTextWrapper = styled.div`
	margin-left: 6px;
	letter-spacing: 0.5px;
	display: flex;
	flex-flow: column;
	justify-content: center;
	height: 11px;
	font-size: 12px;
`;

export const SignInFormContainer = styled.form<{styleFont:TypographyENUM}>`
padding:var(--pd-10);
display:flex;
flex-direction:column;
gap:var(--spacing-lg);
font-size:var(--fs-13);
font-family:${props => props.styleFont};
`;

export const UpdateOnWhatsappWrapper = styled.div<{isRTL: boolean}>`
display:flex;
align-items:center;
gap:var(--spacing-sm);
justify-content:center;
flex-direction: ${ props => props.isRTL ? "row-reverse" : ""};
`;
export const CheckboxInput = styled.input`
accent-color:var(--blashish-gray);
width:1.25em;
height:1.25em;
cursor: var(--pointer);
user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const ImageWhatsapp = styled.img`
max-height: 2em;
`;

export const CheckboxLabel = styled.label`
display:flex;
align-items:center;
justify-content:center;
gap:var(--spacing-sm);
cursor: var(--pointer);
`;

export const AnchorTag = styled.a`
font-size:var(--fs-12);
color:var(--dark-grey);
text-align:right;
&:hover{
	text-decoration:underline;
}
`;

export const OTPSentWrapper = styled.div`
font-size:var(--fs-12);
display:flex;
flex-direction:column;
gap:var(--spacing-md);
`;

export const BtnBack = styled.button`
	background:none;
	border:none;
	padding:0;
	width:max-content;
	cursor: var(--pointer);
	b {
		color: #3DB1A6;
		font-size: 15px;
		margin-right: 5px;
	}
`;

export const OTPSentTextWrapper = styled.div`
display:flex;
justify-content:var(--space-between);
align-items:center;
`;

export const OTPSentText = styled.p`
margin:0;
`;

export const InputWrapper = styled.div``;

export const ResendWrapper = styled.small`
display:flex;
justify-content:end;
text-align:right;
align-items:center;
gap:var(--spacing-sm);
;
`;
