// @lk/ui-library/src/Component/V1/StudioFlow/NoResults/NoResults.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const NoResultsWrapper = styled.div`
  padding: var(--pd-28) var(--pd-16) var(--pd-150);
`;

export const NoResultsImageContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  svg {
    width: 170px;
    height: 181px;
  }
`;

export const WarningContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
  padding: var(--pd-40) 0 0;
`;

export const WarningMessage = styled.div`
  max-width: 567px;
  padding: 0 var(--pd-32);
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: var(--fs-24);
  line-height: 36px;
  text-align: center;
  letter-spacing: -0.02em;
  color: var(--text);
`;

export const TryDifferentText = styled.div`
  padding-top: var(--pd-16);
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: var(--fs-14);
  line-height: 20px;
  text-align: center;
  letter-spacing: -0.02em;
  color: var(--text);
`;
