// @lk/ui-library/src/Component/V1/Checkout/AddAddressFormSa/components/AddressTextField/index.tsx
import React from "react";
// import { Cross } from "../../../Icons";
import { Cross } from "../../../../../../Icons";
import { CrossSection, FieldInput, LeftPositioned } from "./styles";

interface AddressTextFieldType {
	value: string | number,
	type?: string,
	onChange: (e: React.ChangeEvent<HTMLInputElement>) => void,
	placeholder?: string,
	children?: React.ReactNode,
	clearText?: () => void,
	onBlur?: (e: any) => void,
	props?: any[],
	maxLength?: number
}

const AddressTextField = ({ value, type = "text", onChange, placeholder = "", children, clearText, onBlur = () => null, maxLength, ...props }: AddressTextFieldType) => {

	return (
		<>
			{children && <LeftPositioned>{children}</LeftPositioned>}
			<FieldInput value={value} onChange={onChange} placeholder={placeholder} type={type} onBlur={onBlur}
				maxLength={maxLength}
				// focused={focused.toString()}
				{...props} />
			{value && clearText && <CrossSection onClick={clearText}><Cross/></CrossSection>}
		</>
	);
};

export default AddressTextField;
