// @lk/ui-library/src/Component/V1/MyOrderMobile/GetUpdates/Switch/Switch.styles.tsx
import styled from "styled-components";

export const Container = styled.div<{ switchEnabled: boolean }>`
	label {
		display: block;
		position: relative;
    	cursor: var(--pointer);
    	outline: none;
    	-webkit-user-select: none;
    	-moz-user-select: none;
    	-ms-user-select: none;
    	user-select: none;
    	padding: 2px;
    	width: 55px;
    	height: 25px;
    	background-color: var(--white);
    	border-radius: var(--border-radius-sm);
	}

	label {
		&::before {
			display: block;
    		position: absolute;
    		top: 1px;
    		left: 1px;
    		bottom: 1px;
    		content: "";
    		right: 1px;
			background-color: ${props => props.switchEnabled ? "#45C476" : "#8e9bad"};
    		border-radius: var(--border-radius-sm);
    		border-radius: white;
    		-webkit-transition: background 0.4s;
    		transition: background 0.4s;
		}

		&::after {
			display: block;
    		position: absolute;
    		top: 1px;
			${props => props.switchEnabled ? "left: 31px;" : "left: 1px"};
    		bottom: 1px;
    		content: "";
    		width: 24px;
    		background-color: var(--white);
    		border-radius: 100%;
    		box-shadow: 0 2px 5px var(--light-shadow);
    		-webkit-transition: left 0.4s ease-in-out;
			transition: left 0.4s ease-in-out;
		}
	}
`;
