// @lk/ui-library/src/Component/V1/Cart/Offers/Offers.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import Offers from "./index";
import { TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/v1/Cart/Offers",
	component: Offers,
	argTypes: {
		...generateGeneralArgs(),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of box",
			category: "Component",
			defaultValue: "100",
		}),
		subText: generateArgTypes({
			type: "string",
			required: false,
			description: "To set sub text of any offer",
			category: "Component",
			defaultValue: "Hurray! Buy 1 Get 1 applied to cart",
		}),
		descText: generateArgTypes({
			type: "string",
			required: false,
			description: "To set decription of offer",
			category: "Component",
			defaultValue: "You are saving ₹1600 on this order",
		})
	},
} as ComponentMeta<typeof Offers>;

const Template: ComponentStory<typeof Offers> = (args) => <Offers {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "offers",
	width: "100",
	tagText: "active",
	headText: "Gold Membership",
	subText:"Hurray! Buy 1 Get 1 applied to cart",
	descText: "You are saving ₹1600 on this order",
	font: TypographyENUM.defaultBook,
};
