// @lk/ui-library/src/Component/V1/Cart/Offers/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const OfferWrapper = styled.div <{ width: number, colorCode: string } >`
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: var(--pd-16) var(--pd-16);
  gap: var(--pd-12);
  background: ${(props) => props.colorCode};
  border: 1px solid #CECEDF;
  border-radius: var(--border-radius-xs);
`;

export const FlexWrapper = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  justify-content: var(--space-between);
`;

export const InfoIconWrapper = styled.span``;

export const BreakUp = styled.div`
  display: flex;
  flex-direction: row;
  padding: var(--pd-10) 0;
  justify-content: var(--space-between);
`;

export const HR = styled.hr`
  border: 1px dotted var(--purple-haze);
  width: 100%;
`;

export const SubText = styled.div<{styleFont: TypographyENUM}>`
  font-weight: var(--fw-regular);
  font-size: var(--fs-16);
  color: var(--dark-blue-100);
  width: 100%;
  font-family: ${(props) => props.styleFont};
`;

export const DescribeText = styled.div<{styleFont: TypographyENUM, isRTL?: boolean}>`
  font-weight: var(--fw-regular);
  font-size: var(--fs-14);
  color: var(--dark-blue-100);
  width: 100%;
  font-family: ${(props) => props.styleFont};
  ${props => props.isRTL ? "direction: ltr; text-align: right;" : ""}
`;

export const HeadText = styled.div<{styleFont: TypographyENUM}>`
  font-size: var(--fs-12);
  font-family: ${(props) => props.styleFont};
`;

export const Tag = styled.span`
  padding: 4px var(--pd-8);
  background: #e0c67c;
  border-radius: 99px;
  margin-left: var(--pd-5);
`;

export const GoldContainer = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  cursor: pointer;
  width: 100%;
`;

export const Img = styled.img<{isRTL: boolean}>`
  ${props => props.isRTL ? "transform : rotate(180deg)" : ""};
`;
