// @lk/ui-library/src/Component/V1/NewPayment/components/RadioGroup/NewRadioGroup.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";


export const RadioGroupContainer = styled.div<{row:number, column: number}>`
	display: grid;
	gap: 10px;
	align-items: center;
	grid-template-columns: ${ (props) => props.column ? `repeat(${props.column},1fr)` : "" };
	grid-template-rows: ${ (props) => props.row ? `repeat(${props.row},1fr)` : ""};
`;

export const InputContainer = styled.div`
	display	: flex;
	gap: 10px;
	align-items: center;
`;

export const RadioContainer = styled.div<{isSelected: boolean;}>`
    display: flex;
    align-items: center;
    justify-content: center;
    border: 1px solid ${props => props.isSelected ? "#0FBD95" : "#5C5C86"};
    width: 20px;
    height: 20px;
    cursor: pointer;
    background: ${props => props.isSelected ? "#D0FBF1" : "#FFFFFF"};
    border-radius: 10px;
`;

export const RadioInnerCircle = styled.div<{isSelected: boolean;}>`
    background: #0B8E6F;
    display: ${props => props.isSelected ? "block" : "none"};
    width: 8px;
    height: 8px;
    border-radius: 50%;
`;

export const Text = styled.label`
	font-family: ${TypographyENUM.lkSansRegular};
	font-style: normal;
	font-weight: 400;
	font-size: 16px;
	line-height: 150%;
	letter-spacing: 0.15px;
	color: #3C3C3C;
`;

export const Error = styled.div`
	font-family	: ${TypographyENUM.lkSansHairline};
	color: #f77171;
	position: absolute;
	font-size: 12px;
`;

export const SubmitButton = styled.button`
	margin: 1.3rem 0;
	padding: 5px;
	font-size: 14px;
	cursor: pointer;
`;
