// @lk/ui-library/src/Component/V1/CheckoutMobile/COD/COD.styles.tsx
import styled from "styled-components";

export const CashOnDelivery = styled.div<{borderVisible?: boolean}>`
  padding: ${props => props.borderVisible ? "var(--fs-16)" : "" };
  background: ${props => props.borderVisible ? "#FFFFFF" : "" };
  box-shadow: ${props => props.borderVisible ? "0px 0px 12px rgb(0 0 66 / 6%)" : "" };
  border-radius: ${props => props.borderVisible ? "var(--fs-12)" : "" };
  margin-top:24px;
  .sectionPayWrapper {
    margin-bottom: 0;
}
  @media screen and (max-width:767px){
	margin-top:0px;
	margin-bottom: 24px;
  }
`;

export const CODBottomSheet = styled.div`
  padding:16px;
  h2{
    font-family: 'LenskartSans-Regular';
    font-style: normal;
    font-weight: 400;
    font-size: 16px;
    line-height: 24px;
    letter-spacing: -0.02em;
    color: #000042;
    margin-bottom:25px;
  }
`;

export const Margin = styled.div`
  margin-bottom:20px;
`;

export const CaptchaSection = styled.div`
  display:flex;
  justify-content:space-around;
  align-items:center;
  margin-bottom:30px;

  a{
    font-family: 'LenskartSans-Regular';
    font-style: normal;
    font-weight: 700;
    font-size: 12px;
    line-height: 18px;
    letter-spacing: -0.02em;
    text-decoration-line: underline;
    text-transform: capitalize;
    color: #000042;
  }
`;

export const Captcha = styled.img`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 0px 16px;
  width: 50%;
  height: 56px;
  background: #FFFFFF;
  border: 1px solid #CECEDF;
  border-radius: 8px;
`;
