// @lk/ui-library/src/Component/Common/ToggleButton/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const CmnToggleLabelWrapper =
	styled.label <{checked: boolean, outerWidth: number, outerHeight: number, innerWidth: number, innerHeight: number, innerColor: string, outerColor: string }>`
	display: block;
	position: relative;
	cursor: var(--pointer);
	outline: none;
	user-select: none;
	padding: 2px;
	width: ${ props => props.outerWidth ? props.outerWidth : "40"}px;
	height: ${ props => props.outerHeight ? props.outerHeight : "20" }px;
	border-radius: 20px;
	&:before {
		display: block;
		position: absolute;
		top: 1px;
		left: 1px;
		bottom: 1px;
		content: "";
		right: 1px;
		background-color: ${(props) => (!props.checked ? "#dbdbea" : "#45C476")};
		border-radius: var(--border-radius-sm);
		border-radius: white;
		transition: background 0.4s;
		&:checked {
			background-color: #8e9bad;
		}
	}
	&:after {
		display: block;
		position: absolute;
		top: 2px;
		left: ${(props) => (!props.checked ? "3px" : "inherit")};
		right: ${(props) => (props.checked ? "3px" : "inherit")};
		bottom: 0px;
		content: "";
		width: ${ props => props.innerWidth ? props.innerWidth : "15"}px;
		height:  ${ props => props.innerHeight ? props.innerHeight : "15"}px;
		background-color: ${props => props.innerColor ? props.innerColor : "var(--white)" };
		border-radius: 100%;
		box-shadow: 0 2px 5px var(--light-shadow);
		transition: left 0.25s;
		&:checked {
			margin-left: 44px;
		}
	}
`;

export const ToggleWrapper = styled.div`
    display: flex;
    flex-direction: row;
	align-items: center;
`;

export const TextContainer = styled.div`
    margin-left: 10px;
	margin-right: 0px;
	font-family: ${TypographyENUM.lkSansBold};
	font-size: 12px;
`;
