// @lk/ui-library/src/Component/Common/ToggleButton/ToggleButton.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import ToggleButton from "./index";

export default {
	title: "Components/Common/ToggleButton",
	component: ToggleButton,
	argTypes: {
		value: generateArgTypes({
			type: "string",
			required: false,
			description: "Value that will be rendered",
			category: "Component",
			defaultValue: "Lorem Ipsum",
		}),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "change toggle button on / off",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show right to left",
			category: "Status",
		}),
	},
} as ComponentMeta<typeof ToggleButton>;

const Template: ComponentStory<typeof ToggleButton> = (args) => <ToggleButton {...args}/>;

export const Overview = Template.bind({});
Overview.args = {
	text: "3D",
	clickOnToggle: () => console.log("click on Toggle")
};
