// @lk/ui-library/src/Component/V1/PackageCard/PackageCard.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../StorybookHelper";
import PackageCard from "./index";
import { CurrencySymbol, CurrencyType } from "./PackageCard.types";
import { TypographyENUM } from "../../../Types/general";

export default {
	title: "Components/v1/PackageCard",
	component: PackageCard,
	argTypes: {
		...generateGeneralArgs(),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		}),
		showIcon: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to show only icon inside button",
			category: "Icon",
			defaultValue: "false",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for button",
			category: "Events",
		}),
	},
} as ComponentMeta<typeof PackageCard>;

const Template: ComponentStory<typeof PackageCard> = (args) => <PackageCard {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	width: "100",
	onClick: () => { console.log("Clicked");},
	id:"573eb04755206af4266f2820",
	imageUrl:"https://static1.lenskart.com/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/v/i/vincent-chase-vc-5158-golden-brown-c43-aviator-sunglasses_m_3573_1_3.jpg",
	packageTitle:"Brown Tinted",
	isSunglasses:true,
	warrantyTxt:"1 Year Warranty",
	specifications:[
		{
			groupType: "TINT",
			groupTitle: "",
			title: "Brown Color Non-Polarized Lenses",
			subTitle: "",
			showHoverIcon: false,
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		},
		{
			groupType: "UV_BLOCK",
			groupTitle: "",
			title: "UV 400 Protection",
			subTitle: "",
			showHoverIcon: false,
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		},
		{
			groupType: "WARRANTY",
			groupTitle: "",
			title: "1 Year Warranty*",
			subTitle: "",
			showHoverIcon: false,
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		},
		{
			groupType: "MATERIAL",
			groupTitle: "",
			title: "Lens Material : CR39",
			subTitle: "",
			showHoverIcon: false,
			hoverUrl: null,
			isEnabled: false,
			boldText: false
		},
		{
			groupType: "THICKNESS",
			groupTitle: "",
			title: "Index : 1.5",
			subTitle: "",
			showHoverIcon: false,
			hoverUrl: null,
			isEnabled: true,
			boldText: false
		}
	],
	dataLocale: {
		FRAME_PLUS_LENS: "Frame + Lens",
		TOLLTIP:"tolltip",
		OOPS_SOMETHING_WENT_WRONG:"Oops! something went wrong, please try again !",
		SELECTED:"SELECTED",
		SELECT_LENS:"SELECT LENS",
		INFO_IMG:"info img",
		INFO:"info",
		ZERO:"0",
		CURRENCY_CODE:"INR"
	},
	offerText:"BUY 1 GET 1 + 500 OFF. Voucher: POWER",
	price:{
		currency: CurrencyType.INR,
		basePrice: 1300,
		lkPrice: 0,
		symbol: CurrencySymbol.INR
	},
	isBluCut:true,
	blueCutLogo:"https://static.lenskart.com/media/desktop/img/blu_logo.png",
	framePrice:{
		currency: CurrencyType.INR,
		basePrice: 2000,
		lkPrice: 0,
		symbol: CurrencySymbol.INR
	},
	font:TypographyENUM.lkSansRegular
};
