// @lk/ui-library/src/Component/V1/PackageCard/styles.tsx
import styled, { css } from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const PackageHolder = styled.div`
	box-sizing: border-box;
	border-radius: 12px;
	overflow: hidden;
	border: 2px solid #fff;
	&:hover {
		border: 2px solid #00bac6;
	}
`;
export const PackageCardWrapper = styled.div <{ width: any, styleFont:TypographyENUM } >`
    background: var(--white);
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    text-align: center;
	width: ${props => props.width ? props.width : "33"}%;
    -webkit-box-flex: 0;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
    // border-radius: var(--border-radius-xs);
    // overflow: hidden;
    border: 2px solid transparent;
	cursor: var(--pointer);
	font-family: ${props => props.styleFont};
	padding: 0 5px;
`;

export const FlexWrapper = styled.div`
	display: flex;
	align-items: center;
	width: 100%;
	justify-content: var(--space-between);
`;
export const PackageCardHead = styled.div <{ isBluCut: boolean } >`
	display: flex;
    flex-direction: column;
	align-items: center;
	${props => props.isBluCut ? css`
		background-image: linear-gradient(180deg, #195da1 8.55%, #203c80 100%);
	` : css`
		background-color: #f5f5f5;
	`}
`;
export const PackageDetails = styled.div``;

export const PackageDetailsList = styled.ul`
  margin: 0;
  padding: 0;
  list-style: none;
  background: var(--white);
  letter-spacing: 1px;
`;

export const BluLogo = styled.img`
  max-height: 49px !important;
`;

export const PackageImage = styled.img`
  width:40%;
`;

export const BluLogoWrapper = styled.div`
  padding: 18px 0 6px;
`;

export const TextPrice = styled.div <{ isSunglasses: boolean, isBluCut: boolean } >`
color:${(props) => props.isBluCut ? "var(--white)" : "var(--whitish-gray)"};
font-size: var(--fs-14);
margin-bottom: 6px;
${props => props.isSunglasses && css`
	display: flex;
	justify-content: center;
	align-items: center;
`}
`;

export const WarrantyText = styled.div <{ isBluCut: boolean; } >`
	color: ${(props) => props.isBluCut ? "#faa61a" : "#4d4d4f"};
  direction: initial;
`;

export const StrongTextPrice = styled.div <{ isSunglasses: boolean, isBluCut: boolean } >`
	color: ${(props) => props.isBluCut ? "var(--white)" : "#00bac6"};
	display: ${(props) => props.isBluCut ? "unset" : "block"};
	margin-left: ${(props) => (props.isSunglasses ? "var(--pd-10)" : "")};
	font-size: var(--fs-18);
  font-family: "FuturaStd-Heavy";
`;

export const Title = styled.div`
  font-size: 16px;
  font-family: "FuturaStd-Heavy";
  letter-spacing: 1.5px;
  display: inline-grid;
  line-height: 22px;
	margin: 18px 0 10px 0;
	height: 28px;
  color: #414241;
`;

export const PackageButton = styled.button`
  width: 80%;
  cursor: var(--pointer);
  font-size: var(--fs-14);
  color: var(--white);
  border-radius: var(--border-radius-xxxs);
  letter-spacing: 1.5px;
  text-transform: var(--uppercase);
  line-height: normal;
  vertical-align: middle;
  outline: none;
  text-align: center;
  padding: 8px 0 7px;
  margin: 12px 0 18px;
  border: none;
`;

export const OfferLI = styled.li`
  color: #3bb3a9;
  font-family: "FuturaStd-Medium";
  font-weight: normal;
  font-size: var(--fs-13);
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: var(--pd-15) var(--pd-10);
  height: 55px;
  border-bottom: 1px solid rgba(65, 66, 65, 0.15);
`;

export const SpecsLI = styled.li`
  font-size: 13px;
  font-family: "FuturaStd-Book", "FuturaStd-Medium";
  font-synthesis: none;
  color: #3bb3a9;
  height: 45px;
  line-height: normal;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #99a0a9;
`;

export const PackageInfoWrapper = styled.span<{isRTL: boolean}>`
  width: 20px;
  cursor: var(--pointer);
  ${props => props.isRTL ? "margin-right: 5px" : "margin-left: 5px"};
  line-height: 5px;
`;

export const InfoIcon = styled.img`
  width: 100%;
  display: block;
`;

export const StrikedText = styled.div`
	font-weight: 600;
	color: #ccc;
    text-decoration: line-through;
`;

export const NewPrice = styled.div`
font-weight: 600;
color: #f18131;
`;

export const ExchangeInfoText = styled.span`
color: #faa61a;
border-radius: 4px;
font-size: 9px;
text-transform: uppercase;
display: inline-block;
`;
