// @lk/ui-library/src/Component/Common/BottomSheet/BottomSheet.Styles.tsx
import styled from "styled-components";

export const Root = styled.div<{backdropColor: string}>`
  background-color: ${({ backdropColor }) => backdropColor};
  pointer-events: auto;
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 100;
`;

export const Sheet = styled.div<{height:string, show: boolean, borderRadius: string, backgroundColor: string, top?:string, padding?:string, overflow?:boolean }>`
  background-color: ${props => props.backgroundColor || "var(--white)"};
  position: fixed;
  left: 0;
  bottom: 0;
  left: 0;
  top: ${props => props.top ? props.top: ""};
  width: 100%;
  margin-bottom: 0 !important;
  height: ${props => props.height ? props.height : "auto"};
  /* transform: ${(props) => (props.show ? "translateY(0%)" : "translateY(100%)")}; */
  /* transition: transform 0.25s ease; */
  z-index: 100;
  padding: ${props => props.padding || "15px"};
  border-radius: ${props => props.borderRadius || "10px 10px 0 0"};
  overflow:${props => props.overflow ? "auto" : "hidden"};
`;

export const Icon = styled.span<{isRTL?: boolean}>`
  position: absolute;
  padding: 4px 2px;
  right: ${props => props.isRTL ? "" : "14px"};
  left: ${props => props.isRTL ? "14px" : ""};
  cursor: pointer;
  display: flex;
  align-items: center;
  z-index: 99999999;
`;
