// @lk/ui-library/src/Component/Common/BottomSheet/BottomSheet.stories.tsx
import { action } from "@storybook/addon-actions";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import BottomSheet from "./BottomSheet.component";

export default {
	title: "Components/Common/BottomSheet",
	component: BottomSheet,
	argTypes: {
		show: generateArgTypes({
			type: "boolean",
			required: true,
			description: "To show the bottomsheet",
			category: "Component",
			defaultValue: true
		}),
		closebottomSheet: generateArgTypes({
			type: "function",
			required: true,
			description: "function to run when the bottomsheet is closed",
			category: "Component",
			defaultValue: () => null,
		}),
		borderRadius: generateArgTypes({
			type: "string",
			required: false,
			description: "To set custom border radius of the bottomsheet",
			category: "Component",
		}),
		backgroundColor: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the custom background color of the bottomsheet",
			category: "Component",
		}),
		showCrossIcon: generateArgTypes({
			type: "boolean",
			required: false,
			description: "To show or not to show the close icon",
			category: "Component",
		}),
		onBackdropClick: generateArgTypes({
			type: "function",
			required: false,
			description: "function which gets triggered when the user clicks on the backdrop",
			category: "Component"
		}),
		top: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the custom top position of the bottomsheet",
			category: "Component",
		}),
		padding: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the custom padding of the bottomsheet",
			category: "Component",
		})
	}
} as ComponentMeta<typeof BottomSheet>;

const Template: ComponentStory<typeof BottomSheet> = (args) => <BottomSheet {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	show: true,
	closebottomSheet: () => null,
	onBackdropClick: () => null,
};

export const Dummy = Template.bind({});
Dummy.args = {
	show: true,
	children: <><div style={{ padding: "50px" }}>this is content</div></>,
	closebottomSheet: () => {
		action("onClick")("On bottom sheet close click...");
		return null;
	},
	onBackdropClick: () => {
		action("onClick")("On bottom sheet close click...");
		return null;
	},
};
