// @lk/ui-library/src/Component/V1/CheckoutMobile/AwaitingPayment/AwaitingPayment.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const AwaitingPaymentScreen = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content:center;
  width: 100%;
  height: 100%;
  padding: var(--pd-32) var(--pd-16) var(--pd-70);
`;

export const AwaitingPaymentBody = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content:center;
  padding: var(--fs-16);
  width:100%;
  height: 100%;
`;

export const AwaiTimerWrapper = styled.div`
  position:absolute;
  inset: 0;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export const AwaitingTimer = styled.div`
  font-family: ${TypographyENUM.lkSerifNormal};
  font-style: normal;
  font-weight: 200;
  font-size: 32px;
  line-height: var(--fs-48);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  text-align:center;
  padding:0 var(--fs-10);
`;

export const AwaitingContent = styled.div`
  height: 100%;
  width:100%;
`;

export const AwaitingPaymentHeader = styled.div`
  margin: 10px 0 0;
  display:flex;
  justify-content:space-between;
  letter-spacing: -0.02em;
  width: 100%;
  span{
	font-size: 20px;
	font-family: ${TypographyENUM.lkSerifNormal};
	color: var(--dark-blue-100);
	font-weight: var(--fw-regular);
  }
`;

export const AwaitHead = styled.h2`
	font-size: 20px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-100);
	font-family: ${TypographyENUM.lkSerifNormal};
	font-style: normal;
	line-height: var(--fs-32);
	font-weight: var(--fw-regular);
`;

export const AwaitingHeading = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-medium);
  font-size: 16px;
  line-height: var(--border-radius-sm);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  margin-bottom: 5px;
  text-align:center;
  padding:0 var(--fs-10);
`;

export const AwaitingNotes = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  /* font-weight: var(--fw-regular); */
  font-size: 12px;
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  color: var(--text);
  text-align:center;
  margin: 0 auto 40px;
  max-width: 190px;
  strong{
	font-weight: var(--fw-bold);
	font-family: ${TypographyENUM.lkSansBold};
  }
`;
export const MinsRemaining = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 10px;
  line-height: var(--border-radius-sm);
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
`;
export const AwaitingProgress = styled.div`
  height:168px;
  margin:35px auto 32px;
  text-align:center;
  flex-direction: column;
  display:flex;
  width:168px;
  position:relative;
  background: #fff;
  position: relative;
  border-radius: 50%;
`;

export const AwaitingCancel = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-bold);
  font-size: var(--fs-14);
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  text-decoration-line: underline;
  color: var(--dark-blue-100);
  cursor: pointer;
  margin-top: 5px;
  text-align:center;
  width: 100%;
`;

export const AwaitingSheetContent = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-12);
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  margin-bottom:24px;
  color: var(--text);
  &.successful-text{
	font-size: var(--fs-10);
	line-height: var(--fs-16);
  }
`;
export const AwaitBtnPayment = styled.a`
	border: 1px solid #E2E2EE;
	filter: drop-shadow(0px 0px 4px rgba(0, 0, 66, 0.1));
	border-radius: 12px;
	display:flex;
	justify-content:space-between;
	padding:16px;
	font-weight: var(--fw-regular);
	font-size: var(--fs-14);
	line-height: var(--fs-24);
	letter-spacing: -0.02em;
	color: var(--text);
	cursor:pointer;
	font-family: ${TypographyENUM.lkSansRegular};
	margin-bottom:24px;
`;

export const AwaitingPaymentSection = styled.div`
  padding: 0;
`;

export const AwaitingSheetHeader = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-medium);
  font-size: var(--fs-16);
  line-height: var(--border-radius-sm);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  margin-bottom: var(--fs-24);
  display:flex;
  align-items:flex-start;
  padding-right:13%;
  svg{
	margin-right:10px;
  }
`;

export const AwaitBtnWrapper = styled.div`
  display:flex;
  align-items:center;
  justify-content:space-between;
  gap:20px;
  margin:30px 0 56px;
  button{
   border-radius:100px;
   font-size:16px;
   min-width: auto;
   width: 100%;
   font-family: ${TypographyENUM.lkSansBold};
   /* font-weight: var(--fw-bold); */
  }
`;

export const HeaderPrice = styled.div`
  font-family: var(--font-lkserif-normal);
  font-weight: 400;
  font-size: var(--fs-20);
  line-height: 32px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;
