// @lk/ui-library/src/Component/V1/CheckoutMobile/AwaitingPayment/AwaitingPayment.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { AwaitingPayment } from "./AwaitingPayment.component";
import { CurrencySymbol } from "./AwaitingPayment.types";

export default {
	title: "Components/v1/CheckoutMobile/AwaitingPayment",
	component: AwaitingPayment,
	argTypes: {
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			control: "object"
		}),
	}
} as ComponentMeta<typeof AwaitingPayment>;

const Template: ComponentStory<typeof AwaitingPayment> = (args) => <AwaitingPayment {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		"AWAITING_PAYMENT": "Awaiting payment",
		"APPROVE_BEFORE_TIMER_RUNS_OUT": "Please approve the payment on your UPI app before the timer runs out",
		"NOTE": "Note",
		"MINS_REMAINING": "Mins remaining",
		"DONT_CLOSE_UNTIL_TRANSACTION_COMPLETE": "Do not close this screen until the transaction is complete",
		"CANCEL_PAYMENT": "Cancel payment",
		"YES_CANCEL":"Yes, cancel",
		"NO":"No",
		"PAYMENT_UNSUCCESSFUL_RETRY_WITH_OTHER_METHOD":"Your payment was unsuccessful, retry with another method.",
		"UNSUCCESSFUL_DESCRIPTION":"Any amount debited from your account, will be credited back within 2-3 working days",
		"RETRY_PAYMENT": "Retry with other payment methods",
		"SURE_YOU_WANT_TO_CANCEL": "Are you sure you want to cancel payment?",
		"CANCELLATION_TEXT": "Cancellation will be successful only if you are yet to complete this payment on your UPI app"
	},
	price: 1780,
	currencySymbol: CurrencySymbol.INR,
};
