// @lk/ui-library/src/Component/V1/AlgoliaSearch/ProductSearch/styles.tsx
import styled from "styled-components";

export const CardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  background: white;
  width: 100%;
  margin-bottom: 10px;
  padding: 6px 0px 0px 14px;
`;

export const CardImageWrapper = styled.img<{
  enableBorder?: boolean;
  selected?: boolean;
}>`
  width: 96px;
  height: 80px;
  border: 1px solid #f2f2fd;
  border-radius: 13px;
  background: #ffffff;
  ${(props) => (props.enableBorder ? "border: 0.5px solid #E6E6EC" : "")};
  padding: 18px 4px;
  ${(props) => (props.selected ? "opacity:1 !important" : "")};
  margin-right: 8px;
  object-fit: contain;
`;

export const CardTitle = styled.div`
  font-weight: 700;
  font-family: "Plus Jakarta Sans-Bold";
  font-size: 16px;
  line-height: 20px;
  padding: 14px 0px;
`;
export const CardItemTileWrapper = styled.div<{
  selected?: boolean | undefined;
}>`
  display: flex;
  justify-content: space-between;
  width: 100%;
  align-items: center;
  height: 60px;
  ${(props) => (props.selected ? "color:#000042" : "")};
`;

export const CardItemContent = styled.div`
  display: flex;
  justify-content: space-between;
`;
export const ProductContentWrapper = styled.div`
  //   display: flex;
  position: relative;
`;
export const ProductContent = styled.div`
  display: flex;
  flex-direction: column;
  flex: 5;
  margin-bottom: 10px;
`;
export const ProductId = styled.div`
  font-size: 13px;
`;

export const ProductBrand = styled.div`
  font-size: 15px;
  font-family: "Plus Jakarta Sans-Bold";
  font-weight: 700;
  line-height: 20px;
  margin-top: 4px;
`;
export const ProductDetail = styled.div`
  font-size: 13px;
  line-height: 16px;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  overflow: hidden;
  text-overflow: ellipsis;
  -webkit-line-clamp: 2;
  margin-right: 10px;
  //   max-width: 230px;
  //   overflow: hidden;
  //   text-overflow: ellipsis;
  //   white-space: nowrap;
  //   @media screen and (max-width: 300px) {
  //     max-width: 155px;
  //   }
`;

export const CardItemAction = styled.div`
  margin-right: 27px;
  @media screen and (max-width: 300px) {
    margin-right: 40px;
  }
`;

export const CardItemTitle = styled.div`
  font-style: normal;
  font-family: "Plus Jakarta Sans-Medium";
  font-weight: 500;
  font-size: 15px;
  color: inherit;
  margin: 10px 0px 10px 10px;
  text-overflow: ellipsis;
  white-space: nowrap;
  max-width: 250px;
  overflow: hidden;
  @media screen and (max-width: 300px) {
    max-width: 155px;
  }
`;

export const CardItemSeperator = styled.div<{
  showBorder?: boolean | undefined;
}>`
  margin: 0px 0px 8px 108px;
  width: 68vw;
  ${(props) => (!props.showBorder ? "" : "border-bottom: 1px solid #E6E6EC")};
`;

export const CardIconWrapper = styled.div`
  margin: 10px;
`;

export const ImageWithLabelWrapper = styled.div<{
  //   inStock: boolean;
  //   isViewSimilar: boolean;
  //   showLabel: boolean;
  //   isAvailableStore: boolean;
}>`
  position: relative;
  display: inline-block;
`;

export const ImageLabel = styled.div`
  font-family: "Plus Jakarta Sans-Bold";
  font-weight: 700;
  font-size: 11px;
  line-height: 12px;
`;
export const ImageLabelContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  width: 96px;
  height: 20px;
  left: 0px;
  bottom: 0px;
  border-radius: 0px 0px 13px 13px;
`;

export const ViewSimilarBtn = styled.div`
  font-weight: 500;
  font-size: 13px;
  position: absolute;
  top: 0;
  right: 16px;
  display: flex;
  align-items: center;
  gap: 4px;
`;

export const FillUpIcon = styled.div``;

export const SearchProductCard = styled.div<{
  inStock: boolean;
  isViewSimilar: boolean;
  isAvailableStore: boolean;
}>`
  ${({ inStock, isAvailableStore, isViewSimilar }) => {
		if (inStock && isAvailableStore && isViewSimilar) {
			return `${ImageLabel} {
            color:#CC9B1C;
          }
          ${ImageLabelContainer} {
            background:#FFFAF0;
          }
          ${FillUpIcon}{
            display:none;
          }
          ${ProductBrand}{
            color:#9999B3;
          }
          ${ProductDetail}{
            color:#9999B3;
          }
          `;
			// Condition 1: inStock is true, isAvailableStore is true, and isViewSimilar is true
		} else if (inStock && !isAvailableStore && isViewSimilar) {
			return `
          ${ImageLabel} {
           display:none;
          }
          ${ImageLabelContainer} {
            display:none;
          }
          ${ViewSimilarBtn} {
            display:none;
          }
          
          
          
          `;
			// Condition 2: inStock is true, isAvailableStore is false, and isViewSimilar is true
		} else if (inStock && !isAvailableStore && !isViewSimilar) {
			return `
      ${ImageLabel} {
        display:none;
       }
       ${ImageLabelContainer} {
         display:none;
       }
       ${ViewSimilarBtn} {
         display:none;
       }
       
       
          `;
			// Condition 3: inStock is true, isAvailableStore is false, and isViewSimilar is false
		} else if (inStock && isAvailableStore && !isViewSimilar) {
			return `${ImageLabel} {
            color:#000042;
          }
          ${ImageLabelContainer} {
            background:#F2F2FD;
          }
          ${ViewSimilarBtn} {
            display:none;
          }
          ${CardImageWrapper}{
            opacity:0.5;
          }
          ${ProductBrand}{
            color:#9999B3;
          }
          ${ProductDetail}{
            color:#9999B3;
          }
          ${FillUpIcon}{
            display:none;
          }
          
          `;
			// Condition 4: inStock is true, isAvailableStore is true, and isViewSimilar is false
		} else if (!inStock && isAvailableStore && isViewSimilar) {
			return `${ImageLabel} {
            color:#CC4B3A;
          }
          ${ImageLabelContainer} {
            background:#FFEFED;
          }
          ${ProductBrand}{
            color:#9999B3;
          }
          ${ProductDetail}{
            color:#9999B3;
          }
          ${FillUpIcon}{
            display:none;
          }
          
          `;
			// Condition 5: inStock is false, isAvailableStore is true, and isViewSimilar is true
		} else if (!inStock && !isAvailableStore && isViewSimilar) {
			return `${ImageLabel} {
            color:#CC4B3A;
          }
          ${ImageLabelContainer} {
            background:#FFEFED;
          }
          ${ProductBrand}{
            color:#9999B3;
          }
          ${ProductDetail}{
            color:#9999B3;
          }
          ${FillUpIcon}{
            display:none;
          }
          
          `;
			// Condition 6: inStock is false, isAvailableStore is false, and isViewSimilar is true
		} else if (!inStock && isAvailableStore && !isViewSimilar) {
			return `${ImageLabel} {
            color:#000042;
          }
          ${ImageLabelContainer} {
            background:#F2F2FD;
          }
          ${ViewSimilarBtn} {
            display:none;
          }
          ${CardImageWrapper}{
            opacity:0.5;
          }
          ${ProductBrand}{
            color:#9999B3;
          }
          ${ProductDetail}{
            color:#9999B3;
          }
          ${FillUpIcon}{
            display:none;
          }
          
          `;
			// Condition 7: inStock is false, isAvailableStore is true, and isViewSimilar is false
		} else {
			return `${ImageLabel} {
           display:none;
          }
          ${ImageLabelContainer} {
            display:none;
          }
          ${ViewSimilarBtn} {
            display:none;
          }
           ${CardImageWrapper}{
            opacity:0.5;
          }
          ${ProductContent}{
            color:#9999B3;
          }
          ${FillUpIcon}{
            display:none;
          }

          
          `;
			// Default condition: none of the above conditions are met
		}
	}};
`;
