// @lk/ui-library/src/Component/V1/PDP/TechnicalInfoItem/TechnicalInfoItem.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";
import { TechnicalInfoItem } from "./TechnicalInfoItem.component";

export default {
	title: "Components/V1/PDP/TechnicalInfoItem",
	component: TechnicalInfoItem,
	argTypes: {
		itemKey: generateArgTypes({
			type: "string",
			required: true,
			description: "Key of Technical Information Item",
			category: "Component",
			defaultValue: "Brand Name",
		}),
		itemValue: generateArgTypes({
			type: "string",
			required: true,
			description: "Value of Technical Information Item",
			category: "Component",
			defaultValue: "Vincent Chase"
		}),
		itemLink: generateArgTypes({
			type: "string",
			required: false,
			description: "Link of Clickable Value",
			category: "Component",
			defaultValue: "https://www.lenskart.com/"
		}),
		itemTarget: generateArgTypes({
			type: "object",
			required: false,
			description: "Anchor Tag target",
			category: "Component",
			control: "object"
		}),
		isClickable: generateArgTypes({
			type: "boolean",
			required: false,
			description: "Is Value clickable link flag",
			category: "Component",
			defaultValue: false
		}),
		showMoreInfo: generateArgTypes({
			type: "boolean",
			required: false,
			description: "Show show more info icon",
			category: "Component",
			defaultValue: false
		})
	}
} as ComponentMeta<typeof TechnicalInfoItem>;

const Template: ComponentStory<typeof TechnicalInfoItem> = args => <TechnicalInfoItem {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	name: "Brand Name",
	value: "Vincent Chase",
	itemLink: "https://www.lenskart.in/",
	isClickable: true,
	showAdditionalInfo: true,
	font: TypographyENUM.defaultBook,
	triggerShowInfo: () => console.log("Trigger Show Info.")
};

export const notClickable = Template.bind({});
notClickable.args = {
	...Overview.args,
	isClickable:false
};

export const noAdditionalInfo = Template.bind({});
noAdditionalInfo.args = {
	...Overview.args,
	showAdditionalInfo:false
};
