// @lk/ui-library/src/Component/V1/PDP/TechnicalInfoItem/TechnicalInfoItem.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const ItemWrapper = styled.div<{styleFont:TypographyENUM}>`
  display: flex;
  align-items: center;
  font-family: ${props => props.styleFont};
  letter-spacing: 1px;
`;

export const ItemKey = styled.div`
  margin: var(--spacing-sm);
  font-size: var(--fs-13);
  color: var(--gray);
  flex-basis: 40%;
`;

export const ValueContainer = styled.div`
  display: flex;
  flex-basis: 60%;
`;
export const ItemValueSpan = styled.span`
  font-size: var(--fs-13);
  color: var(--black);
`;

export const ItemValue = styled.a`
  font-size: var(--fs-13);
  color:#329c92;
  cursor: var(--pointer);
`;

export const QuestionMarkIcon = styled.span`
  margin: auto var(--spacing-sm);
  width: 12px;
  height: 12px;
  font-size: var(--fs-10);
  cursor: var(--pointer);
  background-color: var(--black);
  color: var(--white);
  border-radius: 50%;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  transform: translateY(-1px);
`;

export const WistListButton = styled.div<{isShortListed?: boolean}>`
  svg{
    ${props => props.isShortListed ? "fill: #18cfa8" : ""};
	path{
	 stroke:#18cfa8;
	}
  }
`;
export const BuyChatButton = styled.div<{isShortListed?: boolean}>`
 button{
	max-width: 130px;
	min-width:inherit;
	text-align: left;
	font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
	font-weight: 500;
	img{
		height: 28px;
	}
 }
`;
