// @lk/ui-library/src/Component/V1/Header/MainHeader/SearchBar/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

const SearchBarHolder = styled.div`
	display: block;
	width: 100%;
	height: 100%;
	max-width: 736px;
	padding-left: 20px;
	padding-top: 2px;
	@media screen and (min-width: 1920px){
		max-width: 828px;
	}

	@media (max-width: 1220px) {
		max-width: 273px !important;
	}
	@media (max-width: 1579px) {
		max-width: 473px;
	}
	@media screen and (max-width:1300px){
		max-width: 407px;
	}
`;
const SearchBarContainer = styled.div`
	position: relative;
	width: 100%;
`;

const SearchInputField = styled.input`
	width: 100%;
	border-radius: 5px;
	z-index: 42;
    background: #fff;
    width: 100%;
    padding:4px 7px;
    margin: 0;
	position: relative;
	z-index: 2001;
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    border: 1px solid rgba(0,0,66,.5);
    border-radius: 5px;
    font-size: var(--fs-14);
    line-height: 24px;
    letter-spacing: -.02em;
    color: rgba(0,0,66,.5);
    font-family: ${TypographyENUM.lkSansRegular};
	outline: none;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

const SearchInputDropdown = styled.div<{
	isMsite:boolean
}>`
	display: block;
	position: absolute;
	width: 100%;
	height: auto;
	background-color: #fff;
	z-index: 2000;
	padding: ${props => props.isMsite ? "20px" : "10px"};
	border-width: 0 1px 1px 1px;
	border-color: rgba(0,0,66,.5);
	border-style: ${props => props.isMsite ? "none" : "solid"};
	left: 0;
	top: calc(100% - 5px);
	${props => props.isMsite ? "height: 100vh;overflow-y: auto;" : ""};
	${props => props.isMsite ? "border-top: 1px solid rgb(148 150 159 / 70%)" : ""};
`;

const SearchDropHead = styled.h3`
	margin: 15px 0 10px;
	font-style: normal;
	font-family: ${TypographyENUM.lkSansBold};
	font-size: 20px;
	line-height: 20px;
	font-weight: 900;
	letter-spacing: -0.02em;
	color: #27394e;
`;

const UL = styled.ul<{isProductDataList?: boolean}>`
	padding-left: 0;
	margin-left: 0;
	list-style-type: none;
	margin-top: 0;
    margin-bottom:${({ isProductDataList })=> isProductDataList ? "200px" : "10px"};
`;
const LI = styled.li`
	width: 100%;
	padding: 0px;
	list-style-type: none;
	display: list-item;
    text-align: -webkit-match-parent;
	cursor: var(--pointer);
`;
const A = styled.a`
	display: flex;
	font-size: var(--fs-14);
	padding: 6px 8px;
	text-decoration: none;
	color: var(--dark-blue-100);
	font-family: ${TypographyENUM.lkSansRegular};
	justify-content: space-between;
	align-items: center;
	&:hover {
		background: #fbf9f7 none repeat scroll 0 0;
		border-radius: var(--border-radius-xxxs);
	}
`;
const ProductLeftContainer = styled.div`
	display: flex;
	width: 85%;
	gap:10px;
`;
const ProductImg = styled.img`
	display: block;
	/* height: 100%; */
	border: 1px solid #d3d3d3;
	object-fit: contain;
	width: 100px;
`;
const ProductTextContainer = styled.div`
	display: flex;
	align-items: flex-start;
	justify-content: space-between;
	flex-direction: column;
	width: 100%;
`;
const ProductBrand = styled.h4`
	font-weight: 700;
	color: var(--dark-blue-100);
	font-size: 12px;
    line-height: 20px;
    letter-spacing: -0.02em;
    color: var(--dark-blue-100);
    font-family: ${TypographyENUM.lkSansRegular};
	margin: 0;
	padding: 0;
	&::before {
		display: none;
	}
	&::after {
		display: none;
	}
`;
const ProductInfo = styled.p`
	font-weight: 400;
	color: var(--dark-blue-100);
	font-size: 14px;
    line-height: 20px;
    letter-spacing: -0.02em;
    color: var(--dark-blue-100);
    font-family: ${TypographyENUM.lkSansBold};
	margin: 0;
	/* padding: 0 15px; */
`;
const ProductPrice = styled.h3<{ lineHeight?: string; }>`
	min-width: 15%;
	font-size: var(--fs-14);
	color: var(--dark-blue-100);
    line-height: ${({ lineHeight }) => lineHeight ?? "60px"};
    /* font-weight: 400; */
    letter-spacing: -0.02em;
    text-transform: capitalize;
    padding: 0;
	text-align: center;
    font-family: ${TypographyENUM.lkSansBold};
`;
export const SearchHeaderMsite = styled.div`
box-shadow: 0 2px 2px 0 rgb(148 150 159 / 30%);
padding-bottom: 10px;
display:flex;
`;
export const SearchContainer = styled.div`
position: fixed;
top: 0;
left: 0;
height: 100vh;
width: 100vw;
z-index: 101;
background: #fff;
-webkit-box-orient: vertical;
-webkit-box-direction: normal;
-ms-flex-direction: column;
flex-direction: column;
`
;

export const SearchBackButton = styled.span`
transform: rotate(180deg);
font-size: 24px;
padding: 0px 15px;
`;

export const SearchInputFieldMsite = styled.input`
font-size: 15px;
    border: transparent;
    outline: none;
    width: 100%;
    height: 40px;
	padding-top:10px;
    user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const SearchDropHeadMsite = styled.div`
font-size: 20px;
margin: 0px 0px 5px 6px;
font-family: ${TypographyENUM.lkSansBold};
color: var(--dark-blue-100);

`;
export const QrButtonContainer = styled.div`
	width: 100%;
    padding: 20px;
    background: #fff;
    position: fixed;
    bottom: 0;
    z-index: 10000;
`;

export const QrButton = styled.button`
border: 1.5px solid #a2a2b8;
border-radius: 100px;
width: 100%;
display: flex;
margin-bottom: 0;
font-weight: 400;
text-align: center;
padding: 6px 12px;
font-size: 14px;
line-height: 1.42857143;
color: #333;
background-color: #fff;
justify-content: center;
align-items: center;
gap: 10px;
`;

export const ScannerText = styled.span`
padding-left: 4px;
`;

export { SearchBarHolder, SearchBarContainer, SearchInputField, SearchInputDropdown, SearchDropHead, UL, LI, A, ProductImg, ProductTextContainer, ProductBrand, ProductInfo, ProductPrice, ProductLeftContainer };
