// @lk/ui-library/src/Component/V1/OldPayment/QRCode/qrCode.styles.tsx
import styled from "styled-components";

export const Wrapper = styled.div`
margin: 20px;
`;

export const Text = styled.div`
	text-align: center;
	font-size: var(--fs-14);
	font-weight: var(--fw-medium);
	font-family: var(--font-default-book);
	font-synthesis: none;
	color: var(--dark-gray);
`;
export const QRWrapper = styled.div`
    position: relative;
`;

export const QRBG = styled.div`
    margin: var(--pd-20);
    height: 200px;
    background: url(https://static.lenskart.com/media/desktop/img/QRBackground.png) no-repeat 50% 50%;
`;

export const QRButton = styled.button`
	color: var(--white);
    background: #00BBC6;
    font-size: var(--fs-14);
    display: block;
    border: none;
    padding: var(--pd-12) var(--pd-20);
    letter-spacing: 2px;
    border-radius: var(--border-radius-xxxs);
    min-width: 223px;
    font-weight: var(--fw-bold);
    position: absolute;
    left: 50%;
    top: 50%;
    -webkit-transform: translate(-50%, -50%);
    transform: translate(-50%, -50%);
    margin: 0 !important;
    text-transform: var(--uppercase);
	font-family: var(--font-default-heavy);
`;
