// @lk/ui-library/src/Component/V1/OldPayment/QRCode/qrCode.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import QRCode from "./index";

export default {
	title:"Components/V1/Payment/QRCode",
	component: QRCode,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: false,
			description: "Unique Id",
			category: "General",
			defaultValue: "near-by-store"
		})
	}
} as ComponentMeta<typeof QRCode>;

const Template: ComponentStory<typeof QRCode> = args => <QRCode {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"qr-code",
	createOrderPayment: () => console.log("Create Order Payment function call"),
	dataLocale: {
		PLEASE_CLICK_ON_GENERATE_QR_CODE:"Please click on Generate QR Code & Place Order",
		GENERATE_QR_CODE_PLACE_ORDER: "Generate QR Code & Place Order",
		YOU_WILL_BE_REDIRECTED_TO_NEW_PAGE:"You will be redirected to a new page to scan the QR Code"
	},
	frameProduct: false,
	loadingPlaceOrder: undefined,
	paymentCTA: { mainText: "Place order", subText: "(Submit Power in the Next Step)" },
	walletDetails: {
		"code": "payuwallet",
		"label": "UPI QR Code",
		"gatewayId": "PU:QRCODE",
		"enabled": true,
		"showCaptcha": false,
		"banks": [],
		"key": "qrcode_payu",
		"offers": []
	}
};
