// @lk/ui-library/src/Component/V1/Footer/Footer.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Footer from "./index";
import { MockFooterData, footerTopData } from "./MockData";
import { DeviceTypes } from "../../../Types/general";

export default {
	title: "Components/v1/Footer",
	component: Footer,
	argTypes: {},
} as ComponentMeta<typeof Footer>;

const Template: ComponentStory<typeof Footer> = (args) => <Footer {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	footerData: MockFooterData,
	footerTopData,
	dataLocale: {
		FOLLOW_US_AT: "FOLLOW US AT",
		DOWNLOAD_LENSKART_APP_BUY: "Download Lenskart App to buy",
		DOWNLOADAPP_PRODUCTTYPE: "Eyeglasses, Sunglasses and Contact Lenses",
		CAN_WE: "Can we",
		HELP: "Help",
		CHAT_WITH_US: "Chat With Us",
		FOLLOW_US: "Follow Us",
		ALL_RIGHTS_RESERVED: "All Rights Reserved",
		MORE_ABOUT_LENSKART: "More about Lenskart.com",
	},
	deviceType: DeviceTypes.MOBILE,
	footerDataMobile: {
		moreAbout: [
			{
				name: "Reviews",
				redirectUrl: "/reviews",
				isNewTab: false
			},
			{
				name: "Refer And Earn",
				redirectUrl: "/connect-and-earn-lkcash",
				isNewTab: false
			},
			{
				name: "Store Locator",
				redirectUrl: "https://stores.lenskart.com/",
				isNewTab: false
			},
			{
				name: "About Us",
				redirectUrl: "/about-us.html",
				isNewTab: false
			},
			{
				name: "FAQ",
				redirectUrl: "/gfaq",
				isNewTab: false
			}, { name: "T & C", redirectUrl: "/terms-and-conditions", isNewTab: false },
			{ name: "Be A Franchise", redirectUrl: "/franchise_enquiry", isNewTab: false },
			{ name: "Buying Guide", redirectUrl: "/buying-guide", isNewTab: false },
			{ name: "Download App", redirectUrl: "/mobile-app", isNewTab: false },
			{ name: "Frame Size", redirectUrl: "/frame-size", isNewTab: false },
			{ name: "Disclaimer", redirectUrl: "/disclaimer", isNewTab: false },
			{ name: "We Are Hiring", redirectUrl: "https://careers.lenskart.com/", isNewTab: false },
			{ name: "Privacy", redirectUrl: "/privacy-policy", isNewTab: false }
		],
		socialLinks: [
			{ name: "facebook", redirectUrl: "//www.facebook.com/Lenskartindia", isNewTab: true, class: "icon-facebook-squared" },
			{ name: "instagram", redirectUrl: "//www.instagram.com/lenskart/", isNewTab: true, class: "icon-instagram" },
			{ name: "pinterest", redirectUrl: "//in.pinterest.com/lenskartindia/", isNewTab: true, class: "icon-pinterest-circled" },
			{ name: "twitter", redirectUrl: "//twitter.com/Lenskart_com", isNewTab: true, class: "icon-twitter" }
		]
	}
};
