// @lk/ui-library/src/Component/V1/Footer/Footer.types.ts
import { DeviceTypes } from "../../../Types/general";
import { DataType } from "../Header/Header.types";

export interface FooterTypes {
	footerData: FooterDataType,
	dataLocale: DataType,
	footerTopData: FooterTopDataType;
	deviceType: DeviceTypes;
	footerDataMobile: FooterDataMobileType;
	openChatSprinkular: () => void;
	productType: any;
	subdirectoryPath: string
	isRTL: boolean
	country: string
}

export interface FooterDataType {
	mainSection: MainSectionType[];
	bottomRow: LinkType[];
	socialLinks: SocialLinkType[];
	applinks: AppLinksType[]
}
export interface AppLinksType {
	redirectUrl: string,
	imgUrl: string,
	name: string
}
export interface FooterDataMobileType {
	socialLinks: SocialLinkType[];
	moreAbout: MoreAboutType[]
}

export interface DropDataType {
	label: string,
	content: string
}

export interface FooterTopDataType {
	head: string,
	info: string[],
	dropData: DropDataType[]
}

interface MainSectionType {
	sectionTitle: string;
	sectionChildren: LinkType[];
}

interface LinkType {
	rel?: string;
	name: string;
	redirectUrl: string;
	isNewTab: boolean;
}
interface SocialLinkType {
	name: string;
	redirectUrl: string;
	isNewTab: boolean;
	class: string;
}

interface MoreAboutType {
	name: string;
	redirectUrl: string;
	isNewTab: boolean;
}
