// @lk/ui-library/src/Component/V1/Footer/styles.tsx
import styled, { css } from "styled-components";
import { TypographyENUM } from "../../../Types/general";

const colorPalette = {
	darkBlue100: "var(--dark-blue-100)",
	white: "var(--white)",
};

export const FooterContainer = styled.footer`
	background-color: ${colorPalette.darkBlue100};
	color: ${colorPalette.white};
	display: flex;
	flex-direction: column;
	font-family:  FuturaStd-Book, helvetica neue, Helvetica, Arial, sans-serif;
	font-synthesis: none;
	font-size: 13px;
	z-index: 2;
    position: relative;
`;

export const TopRow = styled.div`
	padding: 0 55px 35px;
	display: flex;
	justify-content: var(--space-between);
	flex-wrap: wrap;
`;

export const MainSection = styled.div`
	display: flex;
	gap: 100px;
	flex-wrap: wrap;
	justify-content: flex-start;
`;
export const MainSectionListTitle = styled.h2`
	line-height: 25px;
	font-size: var(--fs-20);
	font-weight: var(--fw-light);
	margin: 20px 0 10px;
	font-family: ${TypographyENUM.lkSansRegular};
`;
export const MainSectionListChildren = styled.ul`
	padding: 0;
	li {
		list-style-type: none;
		line-height: 17px;
		font-size: var(--fs-12);
		padding-top: var(--pd-10);
		margin-bottom: 10px;
		cursor: var(--pointer);
		font-family: ${TypographyENUM.lkSansRegular};
		letter-spacing: -.02em;
	}
`;

export const LinkWrapperOuter = styled.div<{isRTL: boolean}>`
	${props => props.isRTL ? "margin-right: 15px;" : "margin-left: 15px;"};
`;

export const MainSectionList = styled.div``;

export const DownloadAppSection = styled.div`
	padding: var(--pd-30);
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	flex-wrap: wrap;
`;

export const DownloadAppImages = styled.div`
	display: flex;
	gap: var(--spacing-sm);
	margin-bottom: 20px;
	flex-wrap: wrap;
	cursor: var(--pointer);
`;
export const DownloadAppText = styled.span`
	font-size: var(--fs-14);
	line-height: 1.42857;
	text-align: left;
	font-family: ${TypographyENUM.lkSansRegular};
`;

export const HorizontalLine = styled.div`
	border-top: 1px solid ${colorPalette.white};
	margin: 0 55px;
`;

export const BottomRow = styled.div`
	padding: 20px 55px;
	display: flex;
	justify-content: var(--space-between);
	flex-wrap: wrap;
`;

export const Links = styled.ul`
	display: flex;
	gap: var(--spacing-xl);
	align-items: center;
	padding: 0;
	flex-wrap: wrap;
	li {
		font-size: var(--fs-14);
		line-height: 1.42857;
		list-style-type: none;
		cursor: var(--pointer);
		font-family: ${TypographyENUM.lkSansRegular};
	}
`;

export const SocialLinks = styled.div`
	display: flex;
	align-items: center;
	flex-wrap: wrap;
	padding-right:20px;
`;

export const SocialLinkNew = styled.div`
font-size: 22px;
padding: 0 8px;
`;

export const SocialLinkText = styled.span<{msite?: boolean}>`
	font-size: var(--fs-12);
	line-height: 1.42857;
	padding-left: ${props => props.msite ? "0" : "10px"};
	text-transform: capitalize;
	font-family: ${TypographyENUM.lkSansRegular};
`;

export const SocialLinkIcons = styled.div<{msite?: boolean}>`
	display: flex;
	align-items: center;
	gap: var(--spacing-sm);
	cursor: var(--pointer);
	margin-top: 10px;
  margin-bottom: 5px;
	color: #18cfa8;

	${props => props.msite && css`
	margin-top: 5px;
	margin-left: -8px;
	`}
`;

export const LinkWrapper = styled.a`
	text-decoration: none;
	color: ${colorPalette.white};
	padding-inline-end: 0px;
	font-family: inherit;
`;


export const FooterTop = styled.div`
	margin-bottom: 40px;
`;

export const FooterHead = styled.div`
	font-family: ${TypographyENUM.lkSansMedium};
	font-size: 32px;
	line-height: 40px;
	letter-spacing: -.3px;
	color: #f4f4f4;
	padding-bottom: 15px;
	margin-top: 20px;
    margin-bottom: 10px;
	margin: 0.67em 0;
	padding: 0;
	font-weight: 400;
`;

export const FooterPara = styled.p`
	font-family: ${TypographyENUM.lkSansRegular};
	line-height: 20px;
	font-size: var(--fs-14);
	letter-spacing: -.02em;
	color: #fcfcfc;
	text-transform: capitalize;
	margin: 0 0 10px;
	padding: 0;
	a {
		color: #329c92;
		text-decoration: none;
		cursor: var(--pointer);
		background-color: transparent;
	}
`;

export const FooterDropContainer = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: flex-start;
`;

export const FooterDrop = styled.div`
	flex-basis: 33%;
    max-width: 450px;
    margin-right: 0;
    margin-left: 0;
	padding-right: 15px;
    padding-left: 15px;
	box-sizing: border-box;
    margin: 0;
    padding: 0;
`;
export const FooterDropLabel = styled.div`
	cursor: var(--pointer);
	padding: 0 0 8px;
	font-weight: 400;
	font-family: ${TypographyENUM.lkSansRegular};
	font-size: 16px;
	line-height: 20px;
	letter-spacing: -.02em;
	text-transform: var(--uppercase);
	color: #fff;
	i {
		float: right;
		padding: 2px;
		position: relative;
		width: 18px;
		height: 18px;
		right: 14px;
		top: 2px;
		display: inline-block;
		font: normal normal normal 14px/1 FontAwesome;
		font-size: inherit;
		text-rendering: auto;
		background: url(//static.lenskart.com/skin/frontend/base/default/img/sprite-v2.png) no-repeat 0 0;
		vertical-align: baseline;
		letter-spacing: -.02em;
		text-transform: var(--uppercase);
		color: #fff;
		cursor: var(--pointer);
		border: 1px solid #fff;
		display: flex;
		align-items: center;
		justify-content: center;
		border-radius: 50%;
	}
`;

export const FooterDropContent = styled.div<{isOpen: boolean;}>`
	font-family: ${TypographyENUM.lkSansMedium};
	font-size: var(--fs-13);
    line-height: 1.42857;
    border-right: 1px solid #ddd;
    border-bottom: 1px solid #ddd;
    border-left: 1px solid #ddd;
    overflow: hidden;
    max-height: ${props => props.isOpen ? "300vh" : "0"};
    -webkit-transition: max-height 300ms ease-in-out;
    transition: max-height 300ms ease-in-out;
	margin: 0;
    padding: 0;
`;

export const FooterDropPadding = styled.div`
	display: block;
	padding: 5px;
`;

export const FooterMobileWrapper = styled.div`
font-family: Roboto;
@media (min-width: 768px) {
	width: 100%;
}
background-color: #27394e;
padding: var(--pd-20);
border-top-left-radius: 5px;
border-top-right-radius: 5px;
box-shadow: 0 -3px 17px 1px #99a0a9;
color: #fff;

`;

export const DownLoadAppIconWrapper = styled.div`
	display: flex;
	align-items: center;
	margin-bottom: 15px;
	font-family: 'Roboto Slab';
`;

export const DownLoadIcon = styled.div`
	flex-basis: 15%;
`;

export const IconImage= styled.img`
max-width: 100%;
height: auto;
vertical-align: middle;
`;

export const DownLoadTextWarpper = styled.div<{isRTL: boolean}>`
	flex-basis: 85%;
	margin-left: ${props => props.isRTL ? 0 : 15}px;
	margin-right: ${props => props.isRTL ? 15 : 0}px;
`;

export const TextLink = styled.a`
font-weight: 600;
font-size: 14px;
color: #18cfa8;
text-decoration: none;
outline: none;
`;

export const HelpTextWrapper = styled.div`
margin-bottom: 5px
`;

export const HelpMeSubText = styled.div`
font-size: var(--fs-10);
font-family: 'Roboto Slab';
`;

export const HelpText = styled.div`
	font-weight: 600;
	font-size: 18px;
	font-family: 'Roboto Slab';
`;

export const ChatIconTextWrapper = styled.div`
	display: flex;
	align-items: center;
	vertical-align: middle;
	margin: 8px 0px 10px 0px;
`;

export const ChatIconImage= styled.img`
max-width: 100%;
height: auto;
vertical-align: middle;
`;

export const ChatText = styled.span<{isRTL: boolean}>`
	margin-left: ${props => props.isRTL ? 0 : 10}px;
	margin-right: ${props => props.isRTL ? 10 : 0}px;
	font-size: var(--fs-10);
`;

export const MoreAboutWrapper = styled.div<{msite?: boolean}>`
	${props => props.msite ? "margin-top: 5px;" : ""}
	div{
		font-family: Roboto;
		span,a{
			font-weight:300;
		}
	}
`;

export const AccordionItemWrapper= styled.div<{isRTL: boolean}>`
	display: grid;
	grid-template-columns: repeat(2,1fr);
	border-top: 1px solid #99a0a9;
	text-transform: capitalize;
	padding: 10px 0px;
	grid-gap: 5px;
	font-size: var(--fs-12);
	text-align: ${props => props.isRTL ? "right": "left"};
`;

export const SocialLinkWrapper = styled.div`
	padding: 5px 0px;
`;

export const FollowUsText = styled.span`
	text-transform: capitalize;
`;

export const SocialLinkMobileIcons = styled.div`
	font-size: var(--fs-24);
	margin-top: 5px;
`;

export const SocialLinkAnchor = styled.a`
	margin: 4px;
	color: #18cfa8;
`;

export const AllResevedTextWrapper = styled.div`
padding-top:4px;
display: flex;
border-top: 1px solid #99a0a9;
justify-content: space-between;
	font-size: var(--fs-10);
`;

export const AllReservedText = styled.span`
`;

export const LinkMainUrl = styled.a`
color: #18cfa8;
`;

export const SocialLink = styled.div`
color: #18cfa8;
 svg{
	width:30px;
	height:30px;
 }
`;
