// @lk/ui-library/src/Component/V1/MyOrderMobile/ReturnItemCard/Carousel/ReturnCarousel.styles.tsx
import styled from "styled-components";

export const Container = styled.div`
	margin-bottom: 20px;
	width: 100%;
	.slick-slider{
		width: 100%;
		img {
			width: 100%;
		}
	}
	.slick-dots {
    position: absolute;
    bottom: -20px;
    display: block;
    width: 100%;
    padding: 0;
    list-style: none;
    text-align: center;
    li {
      position: relative;
      display: inline-block;
      width: 20px;
      height: 20px;
      margin: 0 0px;
      padding: 0;
      cursor: pointer;
      button {
        font-size: 0;
        line-height: 0;
        display: block;
        width: 20px;
        height: 20px;
        padding: 5px;
        cursor: pointer;
        color: transparent;
        border: 0;
        outline: none;
        background: transparent;
        &:hover {
          outline: none;
        }
        &:hover:before {
          opacity: 1;
          color: #289f95;
        }
        &:before {
          font-family: "slick";
          font-size: 10px;
          line-height: 20px;
          position: absolute;
          top: 0;
          left: 0;
          width: 20px;
          height: 20px;
          content: "•";
          text-align: center;
          opacity: 0.25;
          color: black;
          -moz-osx-font-smoothing: grayscale;
        }
      }
      &.slick-active button:before {
        opacity: 1;
        color: #289f95;
      }
    }
  }
`;
