// @lk/ui-library/src/Component/V1/Cart/CartWishlist/CartWishlist.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";
import { SliderWrapper } from "../../../Common/Carousel/styles";

export const RecentlyViewedWrapper = styled.div<{ mobileView?: boolean, isRTL?: boolean }>`
  ${SliderWrapper} {
	/* ${props => props.mobileView ? "" : "padding-left: var(--pd-40); padding-right: var(--pd-40);"}; */
	.slick-track {
		margin-left: 0;
		margin-right: 0;
	}
  }
`;

export const ProductContainer = styled.div<{styleFont:TypographyENUM, mobileView?: boolean, isRTL?: boolean, width?: string, noMargin?: boolean}>`
  	padding: 0 8px var(--pd-15) 8px;
  	margin: 0;
	${props => props.isRTL ? (props.mobileView ? "margin-left: 20px;" : "margin-left: var(--spacing-md);") : (props.mobileView ? "margin-right: 20px;" : "margin-right: var(--spacing-md);")};
  	border-radius: ${props => props.mobileView ? "8px" : ""};
  	border: ${props => props.mobileView ? "1px solid #e2e2ee" : ""};
  	box-shadow: ${props => props.mobileView ? "0px 1px 4px rgb(0 0 0 / 10%)" : ""};
  	overflow: hidden;
  	background: var(--white);
  	font-family:${props => props.styleFont};
  	border: 1px solid var(--purple-haze);
	max-width: ${props => props.width ? props.width : "236px"};
	border-radius: 8px;
	${props => props.noMargin && "margin: 0"};
`;

export const ProductLink = styled.a`
	height: 100%;
	width: 100%;
	display: flex;
	flex-direction: column;
	gap: 8px;
	justify-content: center;
`;

export const ImgContainer = styled.div<{ isRTL?: boolean }>`
	/* display: flex;
	align-items: center;
	justify-content: center; */
	position: relative;
	padding: 16px 0;
	padding-bottom: 40px;
	div {
		position: absolute;
		${props => props.isRTL ? "right: 0" : "left: 0"};
		bottom: 0;
		padding: 4px 8px;
	}
`;

export const ProductImage = styled.img`
	width: 100%
`;

export const ProductNameContainer = styled.div`
	/* text-align: center; */
`;
export const CustomButtons = styled.div`
	display: flex;
	align-items: center;
	gap: 16px;
	direction: ltr;
	span {
		cursor: pointer;
	}
`;

export const ProductDetail = styled.div<{ isRTL?: boolean }>`
	display: flex;
	flex-direction: column;
	gap: 8px;
	align-items: flex-start;
	padding: 8px 0 0;
	direction: ${props => props.isRTL ? "rtl" : "ltr"};
`;

export const ProductFeature = styled.div`
	font-size: var(--fs-12);
	width: 95%;
	color: var(--dark-blue-50);
	font-family: ${TypographyENUM.lkSansRegular};
	span {
		display: block;
		width: 100%;
		overflow: hidden;
		text-overflow: ellipsis;
		white-space: nowrap;
  	}
`;

export const OfferText = styled.div<{ isRTL?: boolean }>`
	${props => props.isRTL ? "margin-right: var(--fs-6)" : "margin-left: var(--fs-6)"};
	color: rgb(15, 189, 149);
	font-size: var(--fs-12);
`;

export const Tax = styled.div<{isRTL?: boolean}>`
	${props => props.isRTL ? "margin-right: var(--fs-6)" : "margin-left: var(--fs-6)"};
	color: var(--dark-blue-10);
	font-size: 12px;
	font-family: ${TypographyENUM.lkSansRegular};
`;

export const StrickOffText = styled.span<{isRTL?: boolean}>`
	color: var(--dark-blue-10);
	font-weight: var(--fw-regular);
	font-family: var(--font-lksans-regular);
	${props => props.isRTL ? "margin-right: var(--fs-6)" : "margin-left: var(--fs-6)"};
	font-size: var(--fs-12);
	text-decoration: line-through;
`;

export const ProductName = styled.div`
	font-size: var(--fs-14);
	/* font-weight: var(--fw-bold); */
	line-height: 22px;
	color: var(--dark-blue-100);
	font-family: ${TypographyENUM.lkSansBold};
	word-break: break-all;
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
	width:143px;
`;

export const ProductPriceContainer = styled.div`
	/* text-align: center; */
`;

export const ProductPrice = styled.div`
	font-size: var(--fs-14);
	/* font-weight: var(--fw-bold); */
	line-height: 22px;
	color: var(--dark-blue-100);
	display: flex;
	font-family: ${TypographyENUM.lkSansBold};
`;

export const Heading = styled.div<{styleFont:TypographyENUM, mobileView?: boolean}>`
	font-weight: var(--fw-regular);
	font-size: ${props => props.mobileView ? "var(--fs-16);":"var(--fs-24);"}
	${props => props.mobileView ? "margin-top: var(--spacing-lg); margin-bottom: var(--spacing-lg);" : "margin-top: 22px;margin-bottom: 20px;line-height: 30px;"}
	font-family: ${TypographyENUM.lkSansRegular};
	color: var(--dark-blue-100);
	display: flex; gap: 20px;
	align-items: center;
	justify-content: space-between;
`;

export const MobileButtons = styled.div`
	display: flex;
	align-items: center;
	gap: 10px;
  	div {
		width: 28px;
		height: 28px;
		border-radius: 50%;
		box-shadow: 0px 1px 4px rgb(0 0 0 / 10%);
	img {
		width: 100%;
		height: 100%;
		display: block;
	}
  }
`;
