// @lk/ui-library/src/Component/V1/Cart/CartWishlist/CartWishlist.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import CartWishlist from "./CartWishlist.component";
import { productDataLocale, productListData } from "./stories-data";

export default {
	title: "Components/V1/Cart/CartWishlist",
	component: CartWishlist,
	argTypes: {
		productList: generateArgTypes({
			type: "array",
			required: true,
			description: "List of products",
			category: "Component",
			defaultValue: [],
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			defaultValue: productDataLocale(),
			control:"object"
		}),
	}
} as ComponentMeta<typeof CartWishlist>;

const Template: ComponentStory<typeof CartWishlist> = (args) => (<CartWishlist {...args} />);

export const Overview = Template.bind({});

Overview.args = {
	productList: productListData(),
	dataLocale: productDataLocale()
};
