// @lk/ui-library/src/Component/V1/PDP/ImagesGrid/ImagesGrid.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { ImagesGrid } from "./ImagesGrid.component";
import { imagesListData } from "./stories-data";

export default {
	title: "Components/V1/PDP/ImagesGrid",
	component: ImagesGrid,
	argTypes: {
		imagesList: generateArgTypes({
			type: "array",
			category: "Component",
			required: true,
			description: "Images list",
			defaultValue: imagesListData()
		}),
		itemClassName: generateArgTypes({
			type: "string",
			category: "Component",
			required: false,
			description: "item class name",
		}),
		containerClassName: generateArgTypes({
			type: "string",
			category: "Component",
			required: false,
			description: "container class name",
		}),
		containerId: generateArgTypes({
			type: "string",
			category: "Component",
			required: false,
			description: "container id",
		}),
		onClick: generateArgTypes({
			type: "function",
			category: "Event",
			required: false,
			description: "on click event ",
		}),
	}
} as ComponentMeta<typeof ImagesGrid>;

const Template: ComponentStory<typeof ImagesGrid> = (args) => <ImagesGrid {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	imagesList: imagesListData(),
};
