// @lk/ui-library/src/Component/V1/AlgoliaSearch/common.styles.tsx
import styled from "styled-components";

export const CardItemTileWrapper = styled.div<{selected?:boolean|undefined}>`
display:flex;
justify-content:space-between;
width: 100%;
align-items: center;
height: 48px;
${(props) => (props.selected ? "color:#000042" : "")};
`;

export const CardItemContent = styled.div`
// padding:4px;
display:flex;
justify-content:space-between;

`;
export const CardItemAction = styled.div`
margin-right: 27px;
@media screen and (max-width:300px){
    margin-right: 40px;
}
`;

export const CardItemTitle = styled.div`
font-style: normal;
font-family: 'Plus Jakarta Sans-Medium';
font-weight: 500;
font-size: 15px;
color: inherit;
margin: 10px 0px 10px 8px;
text-overflow: ellipsis;
white-space: nowrap;
max-width: 250px;
overflow: hidden;
@media screen and (max-width:300px){
    max-width: 155px;
}

`;

export const CardItemSeperator = styled.div<{showBorder?:boolean|undefined}>`
margin-left: 51px;
width: 84vw;
${(props) => (!props.showBorder ? "" : "border-bottom: 1px solid #E6E6EC")};
`;
export const CardImageWrapper = styled.img<{enableBorder?: boolean,selected?:boolean}>`
border-radius: 8px;
background: #FFFFFF;
${(props) => (props.enableBorder ? "border: 0.5px solid #E6E6EC" : "")};
${(props) => (props.selected ? "opacity:1 !important" : "")};
object-fit:contain;
`;

export const ProductContentWrapper = styled.div`
margin-left:10px;
display:flex;
flex-direction:column;
justify-content: center;
flex: 5;
`;
export const ProductId = styled.div`
font-size: 13px;
color: #66668E;
`; 
export const ProductBrand = styled.div`
font-size: 15px;
`; 
export const ProductDetail = styled.div`
font-size: 15px;
max-width: 250px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  @media screen and (max-width:300px){
    max-width: 155px;
}
`; 

export const CardIconWrapper = styled.div`
    margin: 10px;
`;
