// @lk/ui-library/src/Component/V1/PDP/ProductInfo/stories-data.ts
import { CurrencySymbol, CurrencyType } from "../../ProductCard/ProductCard.types";

export const colorOptionsData = () => {
	return [
		{
			"id": 146284,
			"color": "#000000",
			"colorID": 11378,
			"productURL": "https://www.lenskart.com/lenskart-air-lenskart-air-la-e11821-wayfarer-tr90-c2-eyeglasses.html",
			"price": {
				currency: CurrencyType.INR,
				"basePrice": 2000,
				"lkPrice": 999,
				symbol: CurrencySymbol.INR
			},
			"arModel": {
				"ios": "https://3d-asset-lookr.s3-ap-southeast-1.amazonaws.com/frameassets/146284/ios/146284.zip",
				"android": "https://3d-asset-lookr.s3-ap-southeast-1.amazonaws.com/frameassets/146284/android/sku_146284.glb"
			},
			"productImage": {
				"url": "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_g_9919_2.jpg",
				"hoverURL": "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_g_9919_2.jpg"
			}
		},
		{
			"id": 150550,
			"color": "#808080",
			"colorID": 13925,
			"productURL": "https://www.lenskart.com/lenskart-air-la-e11821-c4-eyeglasses.html",
			"price": {
				currency: CurrencyType.INR,
				"basePrice": 2000,
				"lkPrice": 999,
				symbol: CurrencySymbol.INR
			},
			"arModel": {
				"ios": null,
				"android": null
			},
			"productImage": {
				"url": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-c4-eyeglasses_G_0160.jpg",
				"hoverURL": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-c4-eyeglasses_G_0160.jpg"
			}
		},
		{
			"id": 150551,
			"color": "#ed7e07",
			"colorID": 11386,
			"productURL": "https://www.lenskart.com/lenskart-air-la-e11821-c5-eyeglasses.html",
			"price": {
				currency: CurrencyType.INR,
				"basePrice": 2000,
				"lkPrice": 999,
				symbol: CurrencySymbol.INR
			},
			"arModel": {
				"ios": null,
				"android": null
			},
			"productImage": {
				"url": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-c5-eyeglasses_G_0166.jpg",
				"hoverURL": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-c5-eyeglasses_G_0166.jpg"
			}
		},
		{
			"id": 145997,
			"color": "#800080",
			"colorID": 13948,
			"productURL": "https://www.lenskart.com/lenskart-air-la-e11821-full-rim-wayfarer-stainless-steel-c2-eyeglasses.html",
			"price": {
				currency: CurrencyType.INR,
				"basePrice": 2000,
				"lkPrice": 999,
				symbol: CurrencySymbol.INR
			},
			"arModel": {
				"ios": null,
				"android": null
			},
			"productImage": {
				"url": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-fall-rim-wfre-sts-steel-c2-eyeglasses_g_1068.jpg",
				"hoverURL": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-fall-rim-wfre-sts-steel-c2-eyeglasses_g_1068.jpg"
			}
		}
	];
};

export const priceData = () => {
	return {
		currency: CurrencyType.INR,
		basePrice:2000,
		lkPrice: 1199,
		symbol: CurrencySymbol.INR
	};
};


export const addToCartButtonsData = () => {
	return [
		{
			primaryText: "ADD TO CART",
			secondaryText: "For 1 year",
			onlyLens: true,
			withPower: false,

		},
		{
			primaryText: "ADD Power",
			onlyLens: false,
			withPower: false,
		}
	];
};
