// @lk/ui-library/src/Component/V1/HomePage/ImageCarousel/ImageCarousel.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import ImageCarousel from ".";
import { generateGeneralArgs } from "../../../../StorybookHelper";

export default {
	title: "Components/V1/HomePage/ImageCarousel",
	component: ImageCarousel,
	argTypes: {
		...generateGeneralArgs()
	}
} as ComponentMeta<typeof ImageCarousel>;

const Template: ComponentStory<typeof ImageCarousel> = (args) => <ImageCarousel {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "banner-section",
	showCategoryCard: (categoryCards: any, contentType?: string, cardCategory?: string) => console.log("clcick category", categoryCards, contentType, cardCategory),
	images: [
		{
			altText: "karan johar favorites",
			redirectUrl: "https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source: "https://static1.lenskart.com/media/desktop/img/Aug22/KJ-2.jpg",
			cardCategory: "eyeglasses",
			contentType: "Eyeglasses",
			categoryCards: [
				{
					"image": "https://static1.lenskart.com/media/desktop/img/Dec22/2-Jan/BlueZeroPower.jpg",
					"url": "/eyeglasses/offer/eyeglasses.html",
					"dtmObj": {
						"linkPageName": "home",
						"linkName": "web|home|eyeglasses",
						"category": "eyeglasses",
						"PageType": "home",
						"subcategory": "eyeglasses",
						"event": "generic-int-link"
					},
					"gaObj": {
						"event": "Banner Clicks",
						"category": "Banner Clicks",
						"action": "BOTTOM_SHEET",
						"label": "2_Eye_60percent"
					}
				},
				{
					"image": "https://static1.lenskart.com/media/desktop/img/Dec22/2-Jan/BlueWithPower.jpg",
					"url": "https://www.lenskart.com/eyeglasses.html",
					"dtmObj": {
						"linkPageName": "home",
						"linkName": "web|home|eyeglasses",
						"category": "eyeglasses",
						"PageType": "home",
						"subcategory": "eyeglasses",
						"event": "generic-int-link"
					},
					"gaObj": {
						"event": "Banner Clicks",
						"category": "Banner Clicks",
						"action": "BOTTOM_SHEET",
						"label": "1_air_vincent_chase_from_rs_1600"
					}
				}
			]
		},
		{
			altText: "rhea kapoor x lenskart",
			redirectUrl: "https://www.lenskart.com/eyeglasses/collections/rhea-kapoor-x-lenskart.html",
			source: "https://static1.lenskart.com/media/desktop/img/Sep22/Frame%203579.png",
			categoryCards: [],
			cardCategory: "eyeglasses",
			contentType: "Eyeglasses",
		},
		{
			altText: "flex collection",
			redirectUrl: "https://www.lenskart.com/eyeglasses/collections/tr-flex-collection.html",
			source: "https://static1.lenskart.com/media/desktop/img/Aug22/Frame%203583.png",
			categoryCards: [],
			contentType: "Eyeglasses",
			cardCategory: "eyeglasses",
		},
		{
			altText: "sleek steel eyeglases",
			redirectUrl: "https://www.lenskart.com/eyeglasses/promotions/sleek-steel.html",
			source: "https://static1.lenskart.com/media/desktop/img/Sep22/WEB%20(1).png",
			categoryCards: [],
			contentType: "Eyeglasses",
			cardCategory: "eyeglasses",
		},
		{
			altText: "john jacobs voyage",
			redirectUrl: "https://www.lenskart.com/sunglasses/promotions/jj-voyage.html",
			source: "https://static1.lenskart.com/media/desktop/img/Sep22/WEB-1.png",
			categoryCards: [],
			cardCategory: "eyeglasses",
			contentType: "Eyeglasses",
		},
		{
			altText: "kids eyeglasses",
			redirectUrl: "https://www.lenskart.com/eyeglasses/promotions/all-kids-eyeglasses.html",
			source: "https://static1.lenskart.com/media/desktop/img/Apr22/prescription%20web%20banners.png",
			categoryCards: [],
			contentType: "Eyeglasses",
			cardCategory: "eyeglasses",
		},
	]
};
