// @lk/ui-library/src/Component/V1/FooterDiv/FooterDiv.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../StorybookHelper";
import FooterDiv from "./index";

export default {
	title: "Components/v1/FooterDiv",
	component: FooterDiv,
	argTypes: {
		...generateGeneralArgs(),
		subtotalText: generateArgTypes({
			type: "string",
			required: true,
			description: "Subtotal text",
			category: "Component",
			defaultValue: "Subtotal",
		}),
		frameText: generateArgTypes({
			type: "string",
			required: false,
			description: "Breakup of subtotal",
			category: "Component",
			defaultValue: " (Frame + Lens): ",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for back press",
			category: "Events",
		}),
		price: generateArgTypes({
			type: "string",
			required: true,
			description: "Total price",
			category: "Component",
			defaultValue: "₹3000",
		}),
		buttonText: generateArgTypes({
			type: "string",
			required: true,
			description: "Button text",
			category: "Component",
			defaultValue: "Apply Coating",
		}),
		isDisabled: generateArgTypes({
			type: "boolean",
			required: true,
			description: "is Button disabled",
			category: "Component",
			defaultValue: true,
		})
	},
} as ComponentMeta<typeof FooterDiv>;

const Template: ComponentStory<typeof FooterDiv> = (args) => <FooterDiv {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	subtotalText: "Subtotal",
	frameText: " (Frame + Lens): ",
	price: "₹3000",
	buttonText: "Apply Coating",
	onClick: () => null,
	isDisabled: true
};
