// @lk/ui-library/src/Component/V1/FooterDiv/styles.tsx
import styled from "styled-components";

export const FooterDivContainer = styled.div`
	position: absolute;
    left: 0;
    bottom: 0;
    border-top: 1px solid rgba(60,60,60,.23);
    background-color: #fff;
    display: flex;
    align-items: center;
    justify-content: space-between;
    padding: 15px;
    width: 100%;
    z-index: 1;
	font-family: var(--font-default-book);
	font-synthesis: none;
    letter-spacing: 1px;
`;
export const SubtotalSubtext = styled.span`
	font-size: 18px;
    color: #3c3c3c;
	margin-right: 15px;
`;
export const SubtotalFrameText = styled.span`
	color: #999;
`;
export const SubtotalPriceTotal = styled.div`
	font-style: normal;
    font-weight: 700;
    font-size: 18px;
    color: #00bac6;
`;
export const FooterDivButton = styled.button<{disabled: boolean}>`
	background: ${props => props.disabled ? "#e0e0e0" : "#00bac6"};
	cursor: pointer;
    padding: 10px;
    height: 42px;
    border: none;
    color: #fff;
	border-radius: 4px;
	font-weight: 400;
	font-size: 16px;
    width: 260px;
`;
export const FooterDivAligner = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-between;
    direction: initial;
`;
