// @lk/ui-library/src/Component/V1/StudioFlow/MsiteNoResults/MsiteNoResults.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { MsiteNoResults } from "./MsiteNoResults.component";

export default {
	title: "Components/v1/StudioFlow/Mobile/MsiteNoResults",
	component: MsiteNoResults,
	argTypes: {

	}
} as ComponentMeta<typeof MsiteNoResults>;

const Template: ComponentStory<typeof MsiteNoResults> = (args) => <MsiteNoResults {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	items: [
		{
			id: 34225283,
			itemId: "147098",
			itemUrl: "/lenskart-air-la-e13852-c2-eyeglasses.html",
			itemImg: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e13852-c2-eyeglasses_G_2551.jpg",
			itemModel: "LA E13852",
			itemClassification: "eyeframe",
			itemQty: 1,
			itemPowerRequired: "POWER_REQUIRED",
			itemPrescriptionView: {
				powerType: "BIFOCAL",
				showPd: false,
				pdConfigAvailable: false
			},
			itemFrameType: "half_rim",
			itemFrameColor: "Gunmetal",
			itemLensType: "BIFOCAL",
			itemLensCategory: "EYEGLASS",
			itemPrescription: {
				powerType: "BIFOCAL"
			},
			itemContactDisposableType: "",
			itemOptions: [
				{
					type: "bifocal",
					name: "Premium Bi-Focal Lens (Circular/KT)_sg",
					name_en: "Premium Bi-Focal Lens (Circular/KT)_sg",
					label: "Premium Bifocal Lens (Circular KT)",
					label_en: "Premium Bifocal Lens (Circular KT)",
					catalogPrices: [
						{
							currencyCode: "SGD",
							value: 30,
							name: "Market Price"
						},
						{
							currencyCode: "SGD",
							value: 30,
							name: "Lenskart Price"
						}
					],
					id: 1356692,
					oid: "619e0f058398f93e82ed4ae9",
					sku: "bifocal_antiglare_lens_kt_sc2_sg",
					group: "Regular Bifocal KT SG"
				}
			],
			itemPriceObj: {
				itemName: "Gunmetal Grey Half Rim Rectangle Lenskart Air Classic LA E13852-C2 Eyeglasses",
				showStrikeOffPrice: true,
				strikeOffPrice: 98,
				itemPrice: 98
			},
			itemTotalPrice: 128,
			itemCatalogPrice: [
				{
					currencyCode: "SGD",
					value: 98,
					name: "Market Price"
				},
				{
					currencyCode: "SGD",
					value: 98,
					name: "Lenskart Price"
				}
			],
			itemSku: "eye:lenskart-air-la-e13852-c2-eyeglasses-bifocal_antiglare_lens_kt_sc2_sg;lens-package-market-price:30.0",
			itemBrandName: "Lenskart Air",
			itemFrameSize: "Medium",
			addOnPriceObj: {
				showStrikeOffPrice: false,
				strikeOffPrice: 0,
				itemName: "Premium Bi-Focal Lens (Circular/KT)_sg",
				itemPrice: 30
			},
			showRibbon: false,
			ribbonMessage: "",
			showOfferMessage: false,
			offerMessage: "",
			tierMembershipName: "",
			isFree: false,
			isLoyalty: false,
			loyaltyImage: "",
			itemName: "Gunmetal Grey Half Rim Rectangle Lenskart Air Classic LA E13852-C2 Eyeglasses",
			price: {
				currencyCode: "SGD",
				value: 128,
				name: "Gold VIP Price"
			}
		}
	],
	localeData: {
		SORRY_OUT_OF_STOCK: "Sorry, item(s) out of stock in stores",
		PLEASE_TRY_DIFFERENT: "Please try a different item, or visit a store near you",
		WISHLISTED: "Wishlisted",
		WISHLIST_ITEM: "Wishlist item"
	},
	wishlistedProductIds: ["146284"]
};
