// @lk/ui-library/src/Component/V1/StudioFlow/MsiteNoResults/MsiteNoResults.types.ts
import { JSON } from "../../../../Types/general";
import { CartItemPriceType, CartPrescriptionType } from "../../Cart/CartCard/CartCard.types";

export interface MsiteNoResultsPropTypes {
  items: CartItemType[];
  localeData: JSON;
  wishlistedProductIds: string[];
  saveToWishlist: (pid: number) => void;
}

interface CartItemType {
  id: number;
  itemId: string;
  itemUrl: string;
  itemImg: string;
  itemModel: string;
  itemClassification: string;
  itemQty: number;
  itemPowerRequired: string;
  itemFrameColor?: string;
  itemPrescription?: {
    powerType: string;
  };
  itemContactDisposableType?: string;
  itemPrescriptionView: CartPrescriptionType;
  itemFrameType: string;
  itemLensType: string;
  itemLensCategory: string;
  showRibbon: boolean;
  ribbonMessage: string;
  showOfferMessage: boolean;
  offerMessage: string;
  itemPriceObj: CartItemPriceType;
  addOnPriceObj: CartItemPriceType | null;
  itemOptions: itemOption[];
  itemTotalPrice?: number;
  itemCatalogPrice: itemCatalogPrice;
  itemSku: string;
  itemBrandName: string;
  itemFrameSize: string
  tierMembershipName: string;
  isFree: boolean;
  isLoyalty: boolean;
  loyaltyImage: string;
  itemName: string;
  price: Price;
}

type Price = {
  currencyCode: string;
  value: number;
  name: string;
}

type itemCatalogPrice = {
  currencyCode: string;
  value: number;
  name: string;
}[]

type itemOption = {
  type: string;
  name: string;
  name_en: string;
  label: string;
  label_en: string;
  catalogPrices: {
    currencyCode: string;
    value: number;
    name: string;
  }[];
  id: number;
  oid: string;
  sku: string;
  group: string;
};
