// @lk/ui-library/src/Component/V1/StudioFlow/MsiteNoResults/MsiteNoResults.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const NoResultsWrapper = styled.div`
  padding: var(--pd-28) var(--pd-16) var(--pd-150);
`;

export const NoResultsImageContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  svg {
    width: 170px;
    height: 181px;
  }
`;

export const WarningContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 8px;
  padding: var(--pd-20) 0;
`;

export const WarningIcon = styled.div`
  width: 20px;
  height: 20px;
`;

export const WarningMessage = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-16);
  line-height: 24px;
  text-align: center;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;

export const DashedLine = styled.div`
  border: 1px dashed var(--purple-haze);
`;

export const ProductContainer = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: center;
  gap: 16px;
  padding: var(--pd-16) 0;
  border-bottom: 1px dashed var(--purple-haze);
`;

export const ProductImageContainer = styled.div`
  max-width: 104px;
  width: 100%;
`;

export const ProductImage = styled.img`
  width: 100%;
`;

export const ProductDetails = styled.div``;

export const Name = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-12);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  padding-bottom: var(--pd-8);
`;

export const Lens = styled(Name)`
  padding-bottom: var(--pd-12);
`;

export const WishlistButton = styled.div`
  border: 1px solid var(--light-purple);
  border-radius: 100px;
  padding: var(--pd-6) var(--pd-12);
  display: flex;
  align-items: center;
  gap: 6px;
  width: fit-content;
`;

export const WishlistText = styled.div`
  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  font-weight: 700;
  font-size: var(--fs-12);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;

export const TryDifferentText = styled.div`
  padding-top: var(--pd-20);
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-12);
  line-height: 20px;
  text-align: center;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;
