// @lk/ui-library/src/Component/V1/PDP/RecentlyViewed/RecentlyViewed.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { RecentlyViewed } from "./RecentlyViewed.component";
import { productDataLocale, productListData } from "./stories-data";

export default {
	title: "Components/V1/PDP/RecentlyViewed",
	component: RecentlyViewed,
	argTypes: {
		productList: generateArgTypes({
			type: "array",
			required: true,
			description: "List of products",
			category: "Component",
			defaultValue: [],
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			defaultValue: productDataLocale(),
			control:"object"
		}),
	}
} as ComponentMeta<typeof RecentlyViewed>;

const Template: ComponentStory<typeof RecentlyViewed> = (args) => (<RecentlyViewed {...args} />);

export const Overview = Template.bind({});

Overview.args = {
	productList: productListData(),
	dataLocale: productDataLocale()
};
