// @lk/ui-library/src/Component/V1/PDP/RecentlyViewed/RecentlyViewed.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";
import { SliderWrapper } from "../../../Common/Carousel/styles";


export const RecentlyViewedWrapper = styled.div<{ mobileView?: boolean, isRTL?: boolean }>`
  ${SliderWrapper} {
	${props => props.mobileView ? "" : "padding-left: var(--pd-40); padding-right: var(--pd-40);"};
    .slick-next {
      margin-right: 0px;
	  transform: ${(props) => (props.isRTL ? "scaleX(-1)" : "")};
    }

    .slick-prev {
      margin-left: -20px;
	  ${props => props.isRTL ? "margin-right: -20px;" : ""};
	  transform: ${(props) => (props.isRTL ? "scaleX(-1)" : "")};

    }
	.slick-disabled{
		display:none !important;
	}
  }
  .slider-custom{
	.slick-track{
		margin-left:0px;
		margin-right:0px;
	}
  }
`;

export const ProductContainer = styled.div<{styleFont:TypographyENUM, mobileView?: boolean, isRTL?: boolean}>`
  padding: var(--pd-15);
  margin: ${props => props.mobileView ? "0 20px 0 0" : "0 var(--spacing-md)"};
  border-radius: ${props => props.mobileView ? "8px" : ""};
  border: ${props => props.mobileView ? "1px solid #e2e2ee" : ""};
  box-shadow: ${props => props.mobileView ? "0px 1px 4px rgb(0 0 0 / 10%)" : ""};
  overflow: hidden;
  background: var(--white);
  font-family:${props => props.styleFont};
  direction: ${props => props.isRTL ? "rtl" : ""};
`;

export const ProductLink = styled.a`
  height: 100%;
  width: 100%;
  min-height: 200px;
  display: flex;
  flex-direction: column;
  gap: 8px;
  justify-content: center;
`;

export const ImgContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const ProductImage = styled.img`
  width: 100%;
`;

export const ProductNameContainer = styled.div<{isRTL?: boolean}>``;
export const CustomButtons = styled.div``;
export const ProductDetail = styled.div``;
export const ProductFeature = styled.div``;
export const OfferText = styled.div``;

export const ProductName = styled.div`
  font-size: var(--fs-14);
  font-weight: var(--fw-bold);
  line-height: 22px;
  color: var(--light-gray-25);
`;

export const ProductPriceContainer = styled.div<{isRTL?: boolean}>``;

export const ProductPrice = styled.div`
  font-size: var(--fs-14);
  font-weight: var(--fw-bold);
  line-height: 22px;
  color: var(--light-gray-25);
`;

export const Heading = styled.div<{styleFont:TypographyENUM, mobileView?: boolean}>`
  font-weight: var(--fw-regular);
  font-size: var(--fs-16);
  ${props => props.mobileView ? "margin-top: var(--spacing-lg); margin-bottom: var(--spacing-lg);" : "margin-top: var(--spacing-xl); margin-bottom: var(--spacing-sm);"}
  font-family:${props => props.styleFont};
  color: var(--dark-blue-100);
  ${props => props.mobileView ? "display: flex; gap: 20px; align-items: center;" : ""};
`;

export const MobileButtons = styled.div`
  display: flex;
  align-items: center;
  gap: 10px;
  div {
	width: 28px;
	height: 28px;
	border-radius: 50%;
	box-shadow: 0px 1px 4px rgb(0 0 0 / 10%);
	img {
		width: 100%;
		height: 100%;
		display: block;
	}
  }
`;
