// @lk/ui-library/src/Component/V1/PDP/RecentlyViewed/stories-data.ts
import { CurrencySymbol, CurrencyType } from "../../../Types/price.types";

export const productListData = () => {
	return[
		{
			id: "1",
			brandName: "Vincent Chase",
			productName: "Aquacolor Premium Flirt Blue Color Contact Lenses  2 Lens Box Plano",
			imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13884-c5-eyeglasses_G_9955.jpg",
			productUrl: new URL("https://www.lenskart.com/product/brand-name"),
			prices: {
				symbol: CurrencySymbol.INR,
				currency: CurrencyType.INR,
				basePrice: 1500,
				lkPrice: 999
			},
			offer: 50,
		},
		{
			id: "2",
			brandName: "Vincent Chase Online",
			imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/Tortoise-Blue-Full-Rim-Wayfarer-Vincent-Chase-URBAN-RETRO-VC-S14466-C1-Sunglasses_vincent-chase-vc-s14466-c1-sunglasses_G_907418_02_2022.jpg",
			productUrl: new URL("https://www.lenskart.com/product/brand-name"),
			prices: {
				symbol: CurrencySymbol.INR,
				currency: CurrencyType.INR,
				basePrice: 2500,
				lkPrice: 1599
			},
		},
		{
			id: "7",
			brandName: "Aqualens",
			imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//a/q/aqualens-lens-lens-spray-100-ml__mg_9794_1.jpg",
			productUrl: new URL("https://www.lenskart.com/product/brand-name"),
			prices: {
				symbol: CurrencySymbol.INR,
				currency: CurrencyType.INR,
				basePrice: 150,
				lkPrice: 110
			},
		},
		{
			id: "3",
			brandName: "Brand Name",
			imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//a/q/aqualens-lens-lens-spray-100-ml__mg_9794_1.jpg",
			productUrl: new URL("https://www.lenskart.com/product/brand-name"),
			prices: {
				symbol: CurrencySymbol.INR,
				currency: CurrencyType.INR,
				basePrice: 150,
				lkPrice: 110
			},
		},
		{
			id: "4",
			brandName: "Brand Name",
			imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/Tortoise-Blue-Full-Rim-Wayfarer-Vincent-Chase-URBAN-RETRO-VC-S14466-C1-Sunglasses_vincent-chase-vc-s14466-c1-sunglasses_G_907418_02_2022.jpg",
			productUrl: new URL("https://www.lenskart.com/product/brand-name"),
			prices: {
				symbol: CurrencySymbol.INR,
				currency: CurrencyType.INR,
				basePrice: 150,
				lkPrice: 110
			},
		},
		{
			id: "5",
			brandName: "Brand Name",
			imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/Tortoise-Blue-Full-Rim-Wayfarer-Vincent-Chase-URBAN-RETRO-VC-S14466-C1-Sunglasses_vincent-chase-vc-s14466-c1-sunglasses_G_907418_02_2022.jpg",
			productUrl: new URL("https://www.lenskart.com/product/brand-name"),
			prices: {
				symbol: CurrencySymbol.INR,
				currency: CurrencyType.INR,
				basePrice: 150,
				lkPrice: 110
			},
		},
		{
			id: "6",
			brandName: "Brand Name",
			imageUrl: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/Tortoise-Blue-Full-Rim-Wayfarer-Vincent-Chase-URBAN-RETRO-VC-S14466-C1-Sunglasses_vincent-chase-vc-s14466-c1-sunglasses_G_907418_02_2022.jpg",
			productUrl: new URL("https://www.lenskart.com/product/brand-name"),
			prices: {
				symbol: CurrencySymbol.INR,
				currency: CurrencyType.INR,
				basePrice: 150,
				lkPrice: 110
			},
		},
	];
};

export const productDataLocale = () =>{
	return {
		RECENTLY_VIEWED_PRODUCTS: "Recently Viewed Products",
		ADD_ITEMS_FROM_YOUR_WISHLIST: "Add items from your wishlist"
	};
};
