// @lk/ui-library/src/Component/V1/PDP/Mobile/DeliveryOptions/DeliveryOptionsMobile.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { ComponentSizeENUM, TypographyENUM } from "../../../../../Types/general";
import DeliveryOptionsMobile from "./DeliveryOptionsMobile";

export default {
	title: "Components/V1/PDP/mobile/DeliveryOptionsMobile",
	component: DeliveryOptionsMobile,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		componentSize: generateArgTypes({
			type: "string",
			required: false,
			description: "Size of the component",
			category: "General",
			defaultValue: ComponentSizeENUM.small,
			control: "radio",
			options: ComponentSizeENUM,
		}),
		isReadOnly: generateArgTypes({
			type: "boolean",
			required: true,
			description: "read only or not",
			category: "Component",
			defaultValue: "false",
		}),
		onmouseClick: generateArgTypes({
			type: "function",
			required: false,
			description: "on Mouse click handler for rating",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show rating right to left",
			category: "",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		})
	}
} as ComponentMeta<typeof DeliveryOptionsMobile>;

const Template: ComponentStory<typeof DeliveryOptionsMobile> = args => <DeliveryOptionsMobile {...args}/>;
export const Overview = Template.bind({});
Overview.args = {
	
};
