// @lk/ui-library/src/Component/V1/PDP/Mobile/DeliveryOptions/DeliveryOptionsMobile.styles.tsx
import styled from "styled-components";

export const DeliveryOptionsContainer = styled.div`
	display: flex;
	flex-direction: column;
	gap: 10px;
	background-color: #eaeff4;
	padding: 10px 0px 0px 0px;
	margin:0 -5px;
`;

export const Title = styled.div<{ type?: string, color?: string, isHeading?:boolean }>`
	font-size: ${props => props.type === "primary" ? "20px" : "12px"};
	color: ${props => props.color || "var(--black)"};
	${({ isHeading })=> isHeading ? "margin-top: -10px;" : ""}
	font-family: 'LKSans-Black';
	&.dTitle{
		font-family: 'LKSans-Regular';
	}
`;

export const DeliveryInputContainer = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
	border-bottom: 1px solid #99a0a9;
`;
export const Input = styled.input`
	border: none;
	width: 100%;
	padding: 8px 0px;
	font-size:12px;
	font-family: 'LKSans-Regular';
	background-color: inherit;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
	&:focus {
		outline: none;
	}
`;

export const CurrentLocationIconContainer = styled.span`
	display: flex;
    justify-content: center;
    align-items: center;
    margin-right: 2px;
    font-size: 18px;
	opacity: 0.6;
`;

export const Loader = styled.span`
	@keyframes spin {
		0% {
			transform: rotate(0deg);
		}
		100% {
			transform: rotate(359deg);
		}
	}
	svg {
		animation: spin 2s infinite linear;
	}
`;
