// @lk/ui-library/src/Component/V1/AlgoliaSearch/Loader/styles.tsx
import styled, { keyframes } from "styled-components";

export const jumpingAnimation = keyframes`
    0{
    transform: translate(0, 0);
    -webkit-transform: translate(0, 0);
      }
    16%{ 
        transform: translate(0, -6px);
        -webkit-transform: translate(0, -6px);
        background-color: #9999B3;
      }
    33%{   
        transform: translate(0, 0);
        -webkit-transform: translate(0, 0);
      
      }
    100%{
         transform: translate(0, 0);
         -webkit-transform: translate(0, 0);
      }
`;

export const Dot = styled.span<{ height?: string; width?: string }>`
  display: inline-block;
  width: ${(props) => (props.width ? props.width : "8px")};
  height: ${(props) => (props.height ? props.height : "8px")};
  border-radius: 50%;
  margin: 0px 2px;
  background-color: #000042;
  &:nth-last-child(1) {
    animation: ${jumpingAnimation} 1.2s 0.7s linear infinite;
    -webkit-animation: ${jumpingAnimation} 1.2s 0.7s linear infinite;
  }
  &:nth-last-child(2) {
    animation: ${jumpingAnimation} 1.2s 0.4s linear infinite;
    -webkit-animation: ${jumpingAnimation} 1.2s 0.4s linear infinite;
  }
  &:nth-last-child(3) {
    animation: ${jumpingAnimation} 1.2s 0.2s linear infinite;
    -webkit-animation: ${jumpingAnimation} 1.2s 0.2s linear infinite;
  }
  &:nth-last-child(4) {
    animation: ${jumpingAnimation} 1.2s 0s linear infinite;
    -webkit-animation: ${jumpingAnimation} 1.2s 0s linear infinite;
  }
  @media screen and (max-width: 300px) {
    width: 6px;
    height: 6px;
    margin: 0px 1px;
  }
`;

export const TypingLoaderWrapper = styled.span<{ show?: boolean }>`
  visibility: ${(props) => (!props.show ? "hidden" : "visible")};
  ${Dot} {
    ${(props) => (!props.show ? "height: 0px" : "")};
    ${(props) => (!props.show ? "width: 0px" : "")};
    ${(props) => (!props.show ? "margin: 0px" : "")};
  }
`;
