// @lk/ui-library/src/Component/V1/StudioFlow/TopSection/TopSection.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import NeedHelpCta from "../../../Common/NeedHelpCta";
import { TopSection } from "./TopSection.component";

export default {
	title: "Components/v1/StudioFlow/Mobile/TopSection",
	component: TopSection,
	argTypes: {
		children: generateArgTypes({
			type: "node",
			category: "General",
			required: true,
			description: "Child component in the top section",
		}),
		goBack: generateArgTypes({
			type: "function",
			category: "General",
			required: true,
			description: "Back button click event handler"
		}),
		title: generateArgTypes({
			type: "string",
			category: "General",
			required: true,
			description: "Title to render"
		}),
	}
} as ComponentMeta<typeof TopSection>;

const Template: ComponentStory<typeof TopSection> = (args) => <TopSection {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	children: <NeedHelpCta
		localeData={{ NEED_HELP: "Need help", WHATSAPP_CHAT_URL: "https://api.whatsapp.com/send/?phone=918929853854&text=" }}
		isRendered={true}
		isBuyOnChat={true}
		whatsappChatMsg={"Hi Lenskart, help me get started! I am on <pageName> page of mobile site. I need help for product ID: <pid-no>".replace("<pageName>", "Payment").replace("<pid-no>", "147098")}
		phoneNumber={"18001020767"}
		onClick={() => {
			console.log("Clicked!");
		}}
	/>,
	title: "Select a store"
};
