// @lk/ui-library/src/Component/V1/CheckoutMobile/LKCashNew/styles.tsx
import styled, { css } from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const LKCashWrapper = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
	width: 100%;
	padding: var(--pd-16);
	background-color: var(--white);
	border-top-left-radius: var(--border-radius-xs);
	border-top-right-radius: var(--border-radius-xs);
`;

export const LKCashLeftSection = styled.div`
	display: flex;
	flex-direction: column;
	gap: 5px;
	flex: 1;
`;

export const LKCashPrimaryText = styled.div<{ lkCashApplied?: boolean}>`
	display: inline-flex;
	flex-direction: row;
	gap: 5px;
	font-family: ${TypographyENUM.lkSansBold};
	font-style: normal;
	font-weight: var(--fw-bold);
	font-size: var(--fs-12);
	line-height: 20px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-100);
`;

export const LKCashOfferText = styled.div<{ lkCashApplied?: boolean}>`
	display: flex;
	flex-direction: row;
	gap: 5px;
	${props => props.lkCashApplied ? css`
		font-family: ${TypographyENUM.lkSansBold};
		font-style: normal;
		font-weight: var(--fw-bold);
		font-size: var(--fs-12);
		line-height: 20px;
		letter-spacing: -0.02em;
		color: #489B1C;
	` : css`
		font-family: ${TypographyENUM.lkSansMedium};
		font-style: normal;
		font-weight: var(--fw-medium);
		font-size: var(--fs-12);
		line-height: 20px;
		letter-spacing: -0.02em;
		color: var(--text);
	`}
`;

export const LKCashRightSection = styled.div`
	display: flex;
	align-items: center;
	width: 30%;
	justify-content: flex-end;
	max-width: 100px;
	padding-right:10px;
`;

export const RemoveText = styled.div`
	font-family: ${TypographyENUM.lkSansBold};
	font-style: normal;
	font-weight: var(--fw-bold);
	font-size: var(--fs-12);
	line-height: 18px;
	display: flex;
	align-items: center;
	letter-spacing: -0.02em;
	text-decoration-line: underline;
	text-transform: capitalize;
	color: var(--dark-blue-100);
`;

export const RightArrowIcon = styled.div`
	display: flex;
	align-items: center;
	padding: 4px var(--pd-8);
	border: 1px solid var(--dark-blue-50);
	border-radius: 100px;
	cursor: var(--pointer);
`;
