// @lk/ui-library/src/Component/V1/PackageScreenNew/PackageFooter/PackageFooter.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const PackageFooterContainer = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-between;
	width: 100%;
	padding: 20px 46px;
	background-color: var(--warm-grey-25);
`;

export const LeftSection = styled.div`
	display: flex;
	flex-direction: column;
`;

export const Text = styled.div<{disabled?: boolean, isRTL?: boolean}>`
	font-family: ${TypographyENUM.lkSansBold};
	font-style: normal;
	/* font-weight: var(--fw-bold); */
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: ${props => props.disabled ? "var(--dark-blue-10)" : "var(--dark-blue-100)"};
	&.btnContinue{
		 ${props => props.isRTL ? "margin-left: 18px;" : "margin-right: 18px;"};
	}
`;

export const Price = styled.div`
	font-family: ${TypographyENUM.lkSansRegular};
	font-style: normal;
	font-weight: var(--fw-normal);
	font-size: 24px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-100);
	margin-top:10px;
`;

export const RightSection = styled.div`

`;

export const PackageButton = styled.button<{disabled?: boolean, isRTL?: boolean}>`
	display: flex;
	// flex-direction: ${props => props.isRTL ? "row-reverse" : "row"};
	justify-content: center;
	align-items: center;
	padding: 12px 20px;
	width: 291px;
	height: 48px;
	background-color: var(--vivid-green-100);
	background-color: ${props => !props?.disabled ? "var(--vivid-green-100)" : "#DBDBEA"};
	cursor: ${props => !props?.disabled ? "pointer" : "" };
	border-radius: 100px;
	font-size:16px;
	border:none;
	transition: box-shadow 0.2s ease-in-out;
	path{
		stroke: ${props => !props?.disabled ? "var(--dark-blue-100)" : "#9999B3"};
}
`;

export const IconWrapper = styled.div<{isRTL?: boolean}>`
		transform: ${props => props.isRTL ? "scale(-1)" : "scale(1)"};
`;
