// @lk/ui-library/src/Component/V1/PackageScreenNew/PackageFooter/PackageFooter.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

import PackageFooter from "./PackageFooter";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";

export default {
	title: "Components/v1/PackageScreenNew/PackageFooter",
	component: PackageFooter,
	argTypes: {
		...generateGeneralArgs(),
		subtotalText: generateArgTypes({
			type: "string",
			required: true,
			description: "Subtotal text",
			category: "Component",
			defaultValue: "Subtotal",
		}),
		frameText: generateArgTypes({
			type: "string",
			required: false,
			description: "Breakup of subtotal",
			category: "Component",
			defaultValue: " (Frame + Lens): ",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for back press",
			category: "Events",
		}),
		price: generateArgTypes({
			type: "string",
			required: true,
			description: "Total price",
			category: "Component",
			defaultValue: "₹3000",
		}),
		buttonText: generateArgTypes({
			type: "string",
			required: true,
			description: "Button text",
			category: "Component",
			defaultValue: "Apply Coating",
		}),
		isDisabled: generateArgTypes({
			type: "boolean",
			required: true,
			description: "is Button disabled",
			category: "Component",
			defaultValue: true,
		})
	},
} as ComponentMeta<typeof PackageFooter>;

const Template: ComponentStory<typeof PackageFooter> = (args) => <PackageFooter {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	subtotalText: "(Frame + Lens)",
	price: "₹3000",
	buttonText: "Apply Coating",
	onClick: () => null,
	isDisabled: false,
	showButton: true,
	dataLocale: {
		SUBTOTAL: "Subtotal"
	}
};
