// @lk/ui-library/src/Component/Common/HoverableDropdown/HoverableDropdown.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import HoverableDropdown from "./index";

export default {
	title: "Components/Common/HoverableDropdown",
	component: HoverableDropdown,
	argTypes: {
		children: generateArgTypes({
			type: "node",
			description: "The children of the component",
			required: true,
			category: "Component",
			defaultValue: ""
		}),
	}
} as ComponentMeta<typeof HoverableDropdown>;

const Template: ComponentStory<typeof HoverableDropdown> = (args) => (<HoverableDropdown {...args} />);

export const Overview = Template.bind({});

Overview.args = {
	children: (
		<>
			<HoverableDropdown.Header style={{ padding: "5px 10px" }}>Header</HoverableDropdown.Header>
			<HoverableDropdown.Menu>
				<HoverableDropdown.Option style={{ padding: "0 10px" }}>Option 1</HoverableDropdown.Option>
				<HoverableDropdown.Option style={{ padding: "0 10px" }}>Option 2</HoverableDropdown.Option>
				<HoverableDropdown.Option style={{ padding: "0 10px" }}>Option 3</HoverableDropdown.Option>
			</HoverableDropdown.Menu>
		</>
	)
};
