// @lk/ui-library/src/Component/Common/HoverableDropdown/styles.tsx
import styled from "styled-components";

export const DropdownHeaderWrapper = styled.div`
  background-color: var(--white);
  padding: 5px;
  border-radius: 10px;
`;

export const DropdownOptionWrapper = styled.div`
  cursor: var(--pointer);

  &:hover {
    background-color: var(--whitish-gray-25);
  }

  &:nth-last-child():hover {
    border-radius: 0 0 10px 10px;
  }
`;

export const DropdownMenuWrapper = styled.div`
  position: relative;

  .inner-wrapper {
    height: 0;
    transform: scaleY(0);
    transform-origin: top;
    opacity: 0;
    transition: opacity 0.15s ease-in-out;
    background-color: var(--white);
    border-radius: 10px;
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
  }
`;

export const DropdownWrapper = styled.div`
  display: flex;
  flex-direction: column;
  width: fit-content;
  z-index: 10;
  border-radius: 10px;

  &:hover {
    ${DropdownMenuWrapper} {
      .inner-wrapper {
        border-radius: 0 0 10px 10px;
        height: auto;
        transform: scaleY(1);
        opacity: 1;
      }
    }

    ${DropdownHeaderWrapper} {
      border-radius: 10px 10px 0 0;
    }
  }
`;
