// @lk/ui-library/src/Component/Mobile/Cart/CartProduct/CartProduct.styles.tsx
import styled, { css } from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const ProductCardContainer = styled.div<{ loyalty: boolean; }>`
    display: flex;
    flex-direction: column;
    gap: 16px;
    background: #ffffff;
    border-radius: 8px;
    flex: none;
    order: 1;
    flex-grow: 0;
    box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.08);
    position: relative;
    overflow: hidden;
    ${props => props.loyalty ? "padding: 0;" : ""};
`;
export const Ribbon = styled.div<{ topLeft: boolean; }>`
    width: 90px;
    height: 150px;
    overflow: hidden;
    position: absolute;
    ${props => props.topLeft ? `
    top: -10px;
    left: -10px;
    &:before,
    &:after {
        border-top-color: transparent;
        border-left-color: transparent;
    }
    &:after {
        bottom: 0;
            left: 0;
    }
    &:before {
        top: 0;
            right: 0;
    }
    span {
        right: -60px;
        top: 30px;
        transform: rotate(-45deg);
    }
    ` : ""}
    span {
        position: absolute;
        display: block;
        width: 225px;
        // width: 200px;
        // width: 150px;
        // top: 65px;
        // padding: 15px 0;
        background-color: #0fbd95;
        // box-shadow: 0 5px 10px rgba(0,0,0,.1);
        color: #fff;
        font-family: "LenskartSans-Bold";
        text-shadow: 0 1px 1px rgba(0, 0, 0, 0.2);
        text-transform: uppercase;
        text-align: center;
    }
`;
export const ProductCardImage = styled.div<{ loyalty: boolean; }>`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 0px 16px 16px;
    ${props => props.loyalty ? "padding: 0;" : ""};
`;
export const LoyaltyImage = styled.img<{ loyalty: boolean; }>`
    max-width: 100%;
    height: auto;
    vertical-align: middle;
${props => props.loyalty ? "width: 100vw;" : ""}
`;
export const ProductCardDesc = styled.div<{ loyalty: boolean; }>`
    letter-spacing: -0.02em;
    color: #000042;
    display: flex;
    flex-direction: column;
    gap: 16px;
    padding: 0px 16px 16px;
    ${props => props.loyalty ? "padding: 0px 16px 16px;" : ""}
`;
export const ItemDescRow = styled.div`
display: flex;
        flex-direction: row;
        align-items: baseline;
        justify-content: space-between;
        gap: 20px;
`;
export const ItemDescColor = styled.div``;
export const MarginStyles = css``;
export const ItemDescPrice = styled.div``;
export const ItemFreeText = styled.span`
margin-left: 5px;
font-family: ${TypographyENUM.lkSansBold};
            font-size: 12px;
            line-height: 18px;
            /* font-weight: 700; */
            color: #0fbd95;
            letter-spacing: -0.02em;
`;
export const NoPowerMessageContainer = styled.div`
display: flex;
        flex-direction: row;
        align-items: center;
        justify-content: space-between;
`;
export const NoPowerMessagePrimary = styled.div`
font-family: ${TypographyENUM.lkSansRegular};
            font-style: normal;
            font-weight: 400;
            font-size: 12px;
            line-height: 20px;
            letter-spacing: -0.02em;
            color: var(--dark-blue-100);
`;
export const NoPowerMessageSecondary = styled.div`
font-family: ${TypographyENUM.lkSansRegular};
            font-style: normal;
            font-weight: 400;
            font-size: 10px;
            line-height: 14px;
            text-align: right;
            letter-spacing: -0.02em;
            color: var(--text);
`;
export const DescLightDark = styled.div<{ colorDark?: boolean; descLight?: boolean; }>`
${props => props.descLight ? `
font-family: ${TypographyENUM.lkSansRegular};
font-size: 12px;
line-height: 18px;
font-weight: 400;
color: var(--dark-blue-10);
letter-spacing: -0.02em;
line-height: 20px;
` : ""}
/* width: 80%; */
${props => props.colorDark ? `
    color: var(--dark-blue-100);
` : ""}
`;
export const CartQuantity = styled.span`
  margin-left:3px;
`;
export const CartItemSectionSep = styled.div`
height: 0px;
        border: 1px dashed #e2e2ee;
        flex: none;
        align-self: stretch;
`;
export const ItemFinalPriceContainer = styled.div`
display: flex;
        flex-direction: column;
        width: 100%;
        gap: 20px;
`;
export const ItemDiscountContainer = styled.div`
display: flex;
            flex-direction: row;
            align-items: center;
            justify-content: space-between;
`;
export const TextRegular = styled.div`
font-family: ${TypographyENUM.lkSansRegular};
                font-style: normal;
                font-weight: 400;
                font-size: 12px;
                line-height: 20px;
                letter-spacing: -0.02em;
                color: var(--text);
`;
export const PriceDiscount = styled.div<{ bold: boolean; }>`
${props => props.bold ? `
font-family: ${TypographyENUM.lkSansBold};
font-style: normal;
font-size: 12px;
line-height: 20px;
letter-spacing: -0.02em;
color: var(--text);
` : ""}
color: #0fbd95;
`;
export const PriceFinal = styled.div<{ bold: boolean; }>`
color: var(--dark-blue-100);
${props => props.bold ? `
font-family: ${TypographyENUM.lkSansBold};
font-style: normal;
font-size: 12px;
line-height: 20px;
letter-spacing: -0.02em;
color: var(--text);
` : ""}
`;
export const FreeAmountText = styled.span`
color: #9999b3;
                    margin-right: 6px;
`;
export const FreePriceText = styled.span`
color:#0fbd95 ;
`;
export const RemoveItem = styled.div`
font-family: ${TypographyENUM.lkSansBold};
/* font-weight: 700; */
            font-size: 12px;
            line-height: 18px;
            color: var(--dark-blue-100);
            text-decoration: underline;
            text-transform: capitalize;
            cursor: pointer;
            letter-spacing: -0.02em;
`;
export const DuplicateItem = styled(RemoveItem)``;
export const CardOfferBanner = styled.div`
font-family: ${TypographyENUM.lkSansBold};
    font-size: 12px;
    line-height: 18px;
    /* font-weight: 700; */
    color: var(--text);
    letter-spacing: -0.02em;
    padding: 8px;
    background: linear-gradient(90deg, #f7f1de 0%, rgba(247, 241, 222, 0) 100%);
`;
export const KnowMoreContainer = styled.div`
display: flex;
	flex-direction: column;
	gap: 20px;
	align-items: center;
	padding-bottom: 20px;
	border-radius: 16px 16px 0 0;
    background-color: #fff;
    overflow: hidden;
	img {
		border-radius: 16px 16px 0 0;
	}
	a {
		text-decoration: none;
		color: inherit;
	}
`;
export const CloseIconContainer = styled.div`
display: flex;
		flex-direction: row;
		justify-content: center;
		align-items: center;
		padding: 8px;
		gap: 4px;

		// position: absolute;
		position: fixed;
		width: 40px;
		height: 40px;
		top: 16px;
		right: 16px;
		background: #ffffff;
		border-radius: 100px;
		cursor: pointer;
    img {
        max-width: 100%;
    height: auto;
    vertical-align: middle;
    }
`;
export const KnowMoreImage = styled.div`
img {
    max-width: 100%;
    height: auto;
    vertical-align: middle;
}
`;
export const KnowMoreCta = styled.div`
font-family: ${TypographyENUM.lkSansBold};
		/* font-weight: 700; */
		font-size: 12px;
		line-height: 20px;
		letter-spacing: -0.02em;
		text-decoration: underline;
`;
export const CloseIcon = styled.div`
display: flex;
	align-items: center;
	justify-content: center;
	color: var(--dark-blue-100);
	font-size: 16px;
	height: 32px;
	width: 32px;
`;
export const CloseModal = styled.div`
padding: 20px;
	display: flex;
	align-items: center;
	justify-content: center;
	z-index: 99999;
	border-radius: 100px;
	height: 32px;
	width: 32px;
	position: relative;
	left: 90%;
`;
export const FinalPrice = styled.div`
font-family: ${TypographyENUM.lkSansBold};
                font-style: normal;
                /* font-weight: 700; */
                font-size: 12px;
                line-height: 20px;
                letter-spacing: -0.02em;
                color: var(--text);
`;
