// @lk/ui-library/src/Component/V1/PackageScreenNew/PackageScreenNew.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import PackageScreenNew from "./PackageScreenNew";

export default {
	title: "Components/v1/PackageScreenNew/PackagesScreenNew",
	component: PackageScreenNew,
	argTypes: {
		...generateGeneralArgs(),
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General"
		}),
		rightSectionId: generateArgTypes({
			type: "string",
			required: false,
			description: "String used as unique identifier for rightSection",
			category: "General"
		}),
		leftSectionId: generateArgTypes({
			type: "string",
			required: false,
			description: "String used as unique identifier for leftSection",
			category: "General"
		}),
		sliderImages: generateArgTypes({
			type: "array",
			required: true,
			description: "Images to be shown in the left carousel",
			category: "General"
		}),
		sequences: generateArgTypes({
			type: "array",
			required: true,
			description: "Sequence Tabs",
			category: "General"
		}),
		heading: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier for leftSection",
			category: "General"
		}),
		onClosePackages: generateArgTypes({
			type: "function",
			required: true,
			description: "String used as unique identifier for leftSection",
			category: "General"
		}),
		isVisible: generateArgTypes({
			type: "boolean",
			required: true,
			description: "To show/hide menu",
			category: "General"
		}),
		onSelectPackage: generateArgTypes({
			type: "function",
			required: true,
			description: "Function triggered on selecting a package",
			category: "Event"
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show right to left",
			category: "Status"
		})
	}
} as ComponentMeta<typeof PackageScreenNew>;

const Template: ComponentStory<typeof PackageScreenNew> = (args) => <PackageScreenNew {...args} />;
export const Overview = Template.bind({});
Overview.args = {
	id: "packages-popup",
	rightSectionId: "slider-carousel-main",
	isVisible: true,
	sequences: [
		{
			topText: "Power Type",
			btText: "Bifocal/Progressive",
			currentlySelected: true,
			enabled: true,
			id: "abc1",
			onClick: (id) => console.log(`${id} is selected`)
		},
		{
			topText: "",
			btText: "Lens Type",
			currentlySelected: false,
			enabled: true,
			id: "abc2",
			onClick: (id) => console.log(`${id} is selected`)
		},
		{
			topText: "",
			btText: "Eye Power",
			currentlySelected: false,
			enabled: false,
			id: "abc3",
			onClick: (id) => console.log(`${id} is selected`)
		}
	],
	heading: "Full Rim",
	onClosePackages: () => console.log("Closed"),
	leftSectionId: "right-comp-main-packages",
	onSelectPackage: (id) => console.log(id),
	sliderImages: [
		{
			id: 0,
			imageUrl:
				"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/transparent-full-rim-rectangle-vincent-chase-classic-acetate-vc-e14999-c3-eyeglasses_g_0036_12_07_2022.jpg"
		},
		{
			id: 1,
			imageUrl:
				"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/transparent-full-rim-rectangle-vincent-chase-classic-acetate-vc-e14999-c3-eyeglasses_29_july_vc_m_f115539_204093_01_aug.jpg"
		},
		{
			id: 2,
			imageUrl:
				"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/transparent-full-rim-rectangle-vincent-chase-classic-acetate-vc-e14999-c3-eyeglasses_22_july_female110313_27_july.jpg"
		}
	]
};
