// @lk/ui-library/src/Component/V1/OldPayment/PaymentCard/paymentCard.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";
import { PaymentCard } from "./index";

export default {
	title:"Components/V1/Payment/PaymentCard",
	component: PaymentCard,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: false,
			description: "Unique Id",
			category: "General",
			defaultValue: "near-by-store"
		}),
		font: generateArgTypes({
			type: "string",
			required: false,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.defaultBook,
			control: "select",
			options: TypographyENUM,
		}),
		storeImg: generateArgTypes({
			type: "string",
			required: false,
			description: "Image fo the store the be displayed",
			category: "Component",
			defaultValue: "https://static.lenskart.com/media/desktop/img/pdp/visit_store.png",
		}),
		storeInfo: generateArgTypes({
			type: "string",
			required: false,
			description: "Info Text of the store",
			category: "Component",
			defaultValue: "Please Share Your Location To See Nearby Stores",
		}),
		onClickStoreLocator: generateArgTypes({
			type: "function",
			required: true,
			description: "onClick handler for near by store",
			category: "Events",
		}),
	}
} as ComponentMeta<typeof PaymentCard>;

const Template: ComponentStory<typeof PaymentCard> = args => <PaymentCard {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	createOrderPayment: () => console.log("Create Order Payment function call"),
	dataLocale: {
		SECURE: "SECURE",
		BANK_CARDS: "Bank Cards",
		PLEASE_ENTER_CARDHOLDER_NAME: "Please enter Cardholder name.",
		PLEASE_ENTER_VALID_CARD_NUMBER: "Please enter a valid card number.",
		PLEASE_ENTER_VALID_CVV_NUMBER: "Please enter a valid CVV number",
		PLEASE_ENTER_VALID_EXPIRY_DATE: "Please enter a valid expiry date.",
		LEARN_MORE_TEXT: "Learn More",
		EDIT: "Edit",
		ENTER_CARD_NUMBER: "Enter Card Number",
		EXPIRY_DATE: "Expiry Date",
		MM_YY: "MM/YYYY",
		HOW_TO_FIND_CVV: "How to find CVV?",
		CVV: "CVV",
		CARDHOLDER_NAME: "Cardholder Name",
		CLEAR_VALUES: "Clear Values",
		SECURE_CARD: "Secure this card as per RBI guidelines"
	},
	appliedPaymentOffer: "",
	applyOffer: () => console.log("Apply Code function call"),
	frameProduct: false,
	isApplyOfferEnable: true,
	loadingPlaceOrder: undefined,
	localeInfo: { countryCode: "in" },
	paymentCTA: { mainText: "Place order", subText: "(Submit Power in the Next Step)" },
	paymentKey: "cc",
	removeOfferPopup:() => console.log("Remove Offer Popup function call")
};
