// @lk/ui-library/src/Component/V1/OldPayment/PaymentCard/paymentCard.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const Flex = styled.div<{styledFont: TypographyENUM}>`
	display: flex;
	justify-content: space-between;
	font-family: ${(props) => props.styledFont};
	font-size: var(--fs-13);
`;

export const StripImage = styled.img`
	max-height: 20px;
`;

export const Label = styled.label<{showLabel: boolean}>`
	display: ${props => props.showLabel ? "block" : "none"} ;
`;

export const InputWrapper = styled.div<{width: string}>`
	width: ${props => props.width ? props.width + "%" : "100%" };
`;
export const Input = styled.input<{isError: boolean, styledFont: TypographyENUM}>`
	display: block;
    width: 100%;
    height: 34px;
    padding: var(--pd-5) var(--pd-12);
    font-size: var(--fs-14);
    line-height: 2px;
    color: var(--light-black);
    background-color: var(--white);
    background-image: none;
	outline: none;
    border: ${props => props.isError ? "1px solid var(--terracotta-100)" : "1px solid var(--whitish-grey)" } ;
    border-radius: var(--border-radius-xxxs);
	-o-transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s;
    transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s,-webkit-box-shadow ease-in-out 0.15s;
	font-family: ${(props) => props.styledFont};
	:focus{
		box-shadow: ${props => props.isError ? "inset 0 1px 1px rgb(0 0 0 / 8%), 0 0 6px #ce8483" : "inset 0 1px 1px rgb(0 0 0 / 8%), 0 0 8px rgb(102 175 233 / 60%)" } ;
	}
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const Form = styled.div`
    display: flex;
    flex-direction: column;
    gap: var(--pd-10);
`;

export const ClearValueButton = styled.button`
	display: none;
`;

export const PlaceOrderButton = styled.button`
	color: var(--white);
    background: var(--irish-blue);
    font-size: var(--fs-14);
    display: block;
    border: none;
    padding: var(--pd-12) var(--pd-20);
    letter-spacing: 2px;
    border-radius: var(--border-radius-xxxs);
    min-width: 223px;
    font-weight: var(--fw-bold);
	margin: var(--pd-20) 0;
    text-transform: var(--uppercase);
`;

const getCardImage = (cardType: string) => {
	if (cardType === "amex") {
		return "-130px";
	} else if (cardType === "visa") {
		return "-45px";
	} else if (cardType === "mastercard") {
		return "-2px";
	} else if (cardType === "maestro") {
		return "-88px" ;
	}
};

export const InputIcon = styled.div<{cardType: string}>`
  	background: url("https://static.lenskart.com/media/desktop/img/cards_icon1.jpg") no-repeat scroll -175px -4px transparent;
	/* display: inline-block; */
    height: 29px;
    width: 43px;
	display: flex;
    float: right;
    margin-right: var(--pd-5);
    margin-top: -30px;
	background-position-x: ${(props) =>getCardImage(props.cardType)} !important;
`;

export const ErrorDiv = styled.div<{styledFont: TypographyENUM}>`
	font-size: var(--fs-13);
	color: var(--terracotta-100);
	font-family: ${(props) => props.styledFont};
	margin-top: var(--pd-5);
    margin-bottom: var(--pd-10);
`;

export const SecureCardViewWrapper = styled.div<{styledFont: TypographyENUM}>`
	font-family: ${(props) => props.styledFont};
	font-size: var(--fs-13);
`;

export const LearnMoreText = styled.span`
	padding: var(--pd-5);
    font-weight: var(--fw-medium);
    font-size: var(--fs-12);
    color: var(--irish-blue);
`;
