// @lk/ui-library/src/Component/V1/HomePage/BannerWithTitle/BannerWithTitle.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import BannerWithTitle from ".";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import BannerSection from "../BannerSection";
// import PerfectFitSection from "../PerfectFitSection";

export default {
	title: "Components/V1/HomePage/BannerWithTitle",
	component: BannerWithTitle,
	argTypes: {
		...generateGeneralArgs(),
		title: generateArgTypes({
			type: "string",
			required: true,
			description: "title of banner section",
			category: "Component",
		})
	}
} as ComponentMeta<typeof BannerWithTitle>;

const Template: ComponentStory<typeof BannerWithTitle> = (args) => <BannerWithTitle {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "banner",
	title:"FIND THE PERFECT FIT",
	// bannerComponent: <PerfectFitSection id="perfect-fit"/>,
	bannerComponent: <BannerSection id="banner" images={[{
		altText:"colored contact lenses",
		redirectUrl:"https://www.lenskart.com/contact-lenses/contact-lens-type/colour-disposable-contact-lenses.html",
		source:"https://static1.lenskart.com/media/desktop/img/June22/contact-lens-more.jpg"
	},
	{
		altText:"contact lens solution",
		redirectUrl:"https://www.lenskart.com/contact-lenses/lens-solutions.html",
		source:"https://static1.lenskart.com/media/desktop/img/June22/contact-lens-more-1.jpg"
	},]}/>,
	componentWidth:"80%"
};
