// @lk/ui-library/src/Component/V1/Header/MainHeader/LastRow/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const DropdownContainer = styled.div<{active?:boolean, isRTL: boolean }>`
	position: absolute;
	top: 100%;
	/* ${props => !props.isRTL ? `left: ${!props.active ? " -10000px" : 0}` : `right: ${!props.active ? " -10000px" : 0}`}; */
	// ${props => props.isRTL ? "left:  -10000px" : "right:  -10000px"};
	margin-top: 1px;
	left:0px;
	right:0px;
	width: 96vw;
	background: #fff;
	padding: 0 0 30px;
	z-index: 102;
	max-width: 1920px;
	margin-left:auto;
	margin-right:auto;
	display:none;
	min-height:425px;
`;
export const LastRowContainer = styled.div`
	display: flex;
	align-items: center;
	margin: 7px -25px 0;
	padding: 0 25px;
	justify-content: var(--space-between);
	height: 60px;
	position: relative;
	background-color: #fbf9f7;
	@media screen and (min-width: 1920px){
		margin: 7px 0px 0;
	}
`;

export const LastRowLinkContainer = styled.div<{ isRTL?: boolean }>`
	height: 100%;
	display: flex;
	align-items: center;
	justify-content: center;
	background: #fbf9f7;
	font-family: ${TypographyENUM.lkSansBold};
	&:hover {
		${DropdownContainer} {
			${props => props.isRTL ? "right: 0" : "left: 0"};
			display:block;
			a{
				border:none !important;
			}
			a.storeBtn{
				border: 1px solid var(--dark-blue-100)!important;
			}
		}
		&:hover {
			a{
				border-bottom: 3px solid var(--dark-blue-100);
			}
		}
	}
`;

export const LastRowLink = styled.a`
	height: 100%;
	color: var(--dark-blue-100);
    line-height: 60px;
    /* font-weight: var(--fw-regular); */
    font-size: var(--fs-14);
    letter-spacing: 0.01em;
    text-transform: var(--uppercase);
    padding: 0;
    font-family: ${TypographyENUM.lkSansBold};
	cursor: var(--pointer);
	margin: 0 9px 0 11px;
	border-bottom: 3px solid #fbf9f7;
	white-space: nowrap;
	@media screen and (max-width:1360px){
		margin: 0 6px;
	}
	@media screen and (max-width:1180px){
		margin: 0 6px;
		font-size: var(--fs-13);
	}

	@media screen and (max-width:1099px){
		white-space: nowrap;
	}
	body{
		min-width:1024px;
	}

`;

export const HeaderImageLink = styled.img`
	display: block;
	height: 2.3VW;
	border-radius: 4px;
	margin: 0;
	padding: 0;
	margin-right: 4px;
`;
