// @lk/ui-library/src/Component/V1/MyAccount/MyOrder/Items/styles.tsx
import styled from "styled-components";

export const ItemWrapper = styled.div<{ mobileView?: boolean }>`
    border-top: ${props => props.mobileView ? "" : "1px solid #DDD"};
    background-color: #ffffff;
    margin: ${props => props.mobileView ? "10px 0px 15px 0;" : "5px 0px"};
    padding-bottom: ${props => props.mobileView ? "10px" : "5px"};
	${props => props.mobileView ? "	border-radius: 8px; -webkit-box-shadow: 0px 0.5px 1.5px rgb(0 0 0 / 25%); box-shadow: 0px 0.5px 1.5px rgb(0 0 0 / 25%);" : ""};
`;

export const PaymentPendingContainer = styled.div<{ mobileView?: boolean }>`
    padding: ${props => props.mobileView ? "12px 12px 12px 50px" : "15px 12px 15px 45px"};
    border-bottom: ${props => props.mobileView ? "" : "1px solid #eeeeee"};
    position: relative;
    margin: ${props => props.mobileView ? "" : "0px 20px"};
`;

export const PendingIcon = styled.span<{mobileView?: boolean;}>`
    top: ${props => props.mobileView ? "14px" : "18px"};
    left: ${props => props.mobileView ? "10px" : "0px"};
    width: 30px;
    height: 30px;
    position: absolute;
`;

export const PendingStausWrapper = styled.div`
    min-height: 35px;
    align-items: center;
    display: flex;
`;
export const StausSpan = styled.span`
    flex: 1;
`;

export const PendingStatusHeading = styled.div<{ mobileView?: boolean }>`
    color: #ee0000;
    font-size: 16px;
	${props => props.mobileView ? "font-size: 13px; font-family: var(--font-lksans-regular);" : "FuturaStd-Heavy"};
`;

export const PendingStatusSubText = styled.span<{ mobileView?: boolean }>`
    color: #999999;
	font-size: 13px;
	${props => props.mobileView ? "font-family: var(--font-lksans-medium); " : "font-family: var(--font-lksans-bold); font-size: var(--fs-14);"};
`;

export const ItemHeader = styled.div<{ mobileView?: boolean }>`
    padding: ${props => props.mobileView ? "15px 15px 0" : "15px 15px 10px 20px"};
	border-top: ${props => props.mobileView ? "1px solid #eeeeee" : ""};
`;

export const Item = styled.div`
    display: flex;
    justify-content: var(--space-between);
	font-family: var(--font-lksans-regular);
`;

export const A = styled.a<{ mobileView?: boolean }>`
	${props => props.mobileView ? "width: 100%" : ""};
    text-decoration: none;
`;
export const ReviewLink = styled.a`
    color: #00bac6;
    -webkit-tap-highlight-color: transparent;
    text-decoration: none;
    outline: none;
`;
export const ProductImageDescription = styled.div`
    display: flex;
	cursor:pointer;
`;

export const ProductImageWrapper = styled.div`
    flex-basis: 32%;
    padding-left: 5px;
`;

export const ProductImage = styled.img<{mobileView?: boolean}>`
    max-width: ${props => props.mobileView ? "82px" : "200px"};
    vertical-align: middle;
`;

export const ProductDescriptionContainer = styled.div`
    align-self: center;
    flex-basis: 68%;
    overflow: hidden;
    font-size: var(--fs-13);
`;

export const ProductName = styled.div<{ mobileView?: boolean }>`
    font-size: ${props => props.mobileView ? "13px" : "16px"};
    margin-bottom: 4px;
    color: #333;
`;

export const ProductOptions = styled.div`
	color: #999999;
	font-size: 13px;
	margin-bottom: 4px;
`;

export const ProductInfo = styled.div<{ mobileView?: boolean }>`
    color: ${props => props.mobileView ? "#cccccc" : "#999999"};
	font-family: ${props => props.mobileView ? "var(--font-lksans-regular)" : ""};
    font-size: ${props => props.mobileView ? "11px" : "var(--fs-12)"};
    margin-top: ${props => props.mobileView ? "8px" : "4px"};
`;

export const ProductPrice = styled.div`
    font-family: var(--font-lksans-bold);
    color: #333;
    font-size: 14px;
    margin-top: 10px;
    padding-bottom: 10px;
    line-height: 18px;
`;

export const PriceSymbol = styled.span``;

export const ItemStatusContainer = styled.div<{ mobileView?: boolean }>`
padding: ${props => props.mobileView ? "12px 12px 12px 50px": "15px 12px 15px 45px"};
border-bottom: ${props => props.mobileView ? "": "1px solid #eeeeee"};
position: relative;
margin: ${props => props.mobileView ? "": "0px 20px"};
`;

export const StatusIcon = styled.span<{mobileView?: boolean}>`
top: ${props => props.mobileView ? "14px" : "18px"};
left: ${props => props.mobileView ? "10px" : "0px"};
width: 30px;
height: 30px;
position: absolute;
`;

export const ItemStatusWarpper = styled.div`
flex-basis: 32%;
padding-left: 5px;
`;

export const ItemStatusHeading = styled.div<{ mobileView?: boolean }>`
    color: #faa61a;;
	font-family: ${props => props.mobileView ? "var(--font-lksans-bold)" : "FuturaStd-Heavy"};
	font-size: ${props => props.mobileView ? "var(--fs-13)" : "var(--fs-16)"};
`;

export const ItemStatusSubText = styled.span<{ mobileView?: boolean, deliveryDate?: number }>`
    font-family: ${props => props.mobileView ? "var(--font-lksans-medium)" : "var(--font-lksans-bold)"};
    color: #faa61a;
    font-size: var(--fs-14);
	${props => props.deliveryDate ? "" : "display: block; font-family: var(--font-lksans-bold); font-size: var(--fs-13); padding: 7px 0;"}
`;

export const ContentWrapper = styled.div`
    backgroud: #fff;
`;

export const ContentList = styled.div`
    padding: var(--pd-20) var(--pd-10) var(--pd-5) var(--pd-10);
    background-color: #fff;
    border-radius: 8px;
    padding: 20px;
`;

export const PowerTable = styled.div<{ mobileView?: boolean }>`
    margin: ${props => props.mobileView ? "0 10px" : "10px 15px"};
`;

export const AccordionConatiner = styled.div`
    margin: 10px 15px;
`;

export const ItemReviewContainer = styled.div<{mobileView?: boolean}>`
    border-bottom: 1px solid #eee;
    -webkit-box-pack: justify;
    margin-bottom: 15px;
    padding-bottom: 10px;
    justify-content: space-between;
    ${({ mobileView }) => mobileView ? "display: flex;" : "display: none;"}
`;
