// @lk/ui-library/src/Component/V1/PackageScreenNew/PackageItem/PackageItem.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const PackageItemWrapper = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
	background-color: var(--white);
	box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
	border-radius: 12px;
	padding: 24px;
	gap: 24px;
	cursor: pointer;
	transition: box-shadow 0.2s ease-in-out;
	&:hover {
			box-shadow: 0 3px 16px 0 rgba(0, 0, 0, 0.1);
		}
`;


export const ImageWrapper = styled.div`

`;

export const RightSection = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	flex: 1;
	svg{
		width:16px;
		height:14px;
	}
`;


export const ContentWrapper = styled.div`
	flex: 1;
`;

export const IconContainer = styled.div<{isRTL: boolean}>`
	transform: ${props => props.isRTL ? "scaleX(-1)": ""};
`;
export const Title = styled.div`
	font-family: ${TypographyENUM.lkSansBold};
	/* font-weight: 700; */
	font-size: 14px;
	line-height: 20px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-100);
	margin-bottom:5px;
`;
export const Description = styled.div`
	font-family: ${TypographyENUM.lkSansRegular};
	font-weight: 400;
	font-size: 14px;
	line-height: 20px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-75);
`;
