// @lk/ui-library/src/Component/Common/HoverableMenu/HoverableMenu.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import HoverableMenu from "./HoverableMenu.component";

export default {
	title: "Components/common/HoverableMenu",
	component: HoverableMenu,
	argTypes: {
		header: generateArgTypes({
			type: "object",
			required: true,
			description: "header content of hoverableMenu",
			category: "",
		}),
		items: generateArgTypes({
			type: "array",
			required: true,
			description: "items inside Menu",
			category: "",
		}),
		viewAllVisible: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show view all link",
			category: "",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "display right to left",
			category: "",
		}),
	}
} as ComponentMeta<typeof HoverableMenu>;

const Template: ComponentStory<typeof HoverableMenu> = (args) => (<HoverableMenu {...args} />);

export const Overview = Template.bind({});

Overview.args = {
	header: {
		link: "https://www.lenskart.com/eyeglasses.html",
		imgUrl: "https://static1.lenskart.com/media/desktop/img/Apr22/a2.png",
		title: "Eyeglasses"
	},
	items: [
		{
			link: "https://www.lenskart.com/sunglasses/special/vincent-chase-bestsellers.html",
			title: "Vincent Chase",
			category: "with Gold Membership",
			price: {
				currencyName: "INR",
				currencySymbol: "₹",
				lenskartPrice: "1299",
				marketPrice: ""
			}
		},
		{
			link: "https://www.lenskart.com/sunglasses/special/vincent-chase-bestsellers.html",
			title: "Vincent Chase",
			category: "with Gold Membership",
			price: {
				currencyName: "INR",
				currencySymbol: "₹",
				lenskartPrice: "1299",
				marketPrice: ""
			}
		},
		{
			link: "https://www.lenskart.com/sunglasses/special/vincent-chase-bestsellers.html",
			title: "Vincent Chase",
			category: "with Gold Membership",
			price: {
				currencyName: "INR",
				currencySymbol: "₹",
				lenskartPrice: "1299",
				marketPrice: ""
			}
		},
	],
	viewAllVisible: true,
	isRTL: false,
};
