// @lk/ui-library/src/Component/Common/HoverableMenu/HoverableMenu.styles.tsx
import styled from "styled-components";
import HoverableDropdown from "../HoverableDropdown";

export const MenuHeader = styled(HoverableDropdown.Header)`
  text-align: center;
  font-size: 15px;
  letter-spacing: 0.5px;
  color: var(--whitish-gray-10);
`;

export const MenuList = styled(HoverableDropdown.Menu)`
  .inner-wrapper {
    box-shadow: 0px 8px 16px 0px rgb(0 0 0 / 20%);
  }
`;


export const HeaderImg = styled.img`
  width: 100%;
  height: 100%;
`;

export const HeaderText = styled.div`
  padding: 4px 0;
  font-size: 15px;
  letter-spacing: 0.5px;
`;

export const Menu = styled(HoverableDropdown)<{isRTL:boolean}>`

  &:hover {
    box-shadow: 0px 8px 16px 0px rgb(0 0 0 / 20%);

    ${MenuHeader} {
      border-bottom: 1px solid var(--whitish-gray-15);
    }
  }
  width: 16%;
  direction: ${(props) => props.isRTL ? "rtl" : "ltr"};
`;

export const MenuItem = styled(HoverableDropdown.Option)`
  padding: 0px 8px;
  border-bottom: 1px solid var(--whitish-gray-15);
  font-size: 12px;

  a, table {
    width: 100%;
  }

  .view-all-btn {
    padding: 12px 6px;
    color: var(--blashish-gray);
  }

  &:last-child {
    border: none;
    border-radius: 0 0 10px 10px;
  }
`;

export const Row = styled.tr`
  .bold {
    font-weight: bold;
  }
`;
