// @lk/ui-library/src/Component/V1/MyOrderMobile/ReturnRefundMobilePage/ReturnRefundMobile.styles.tsx
import styled from "styled-components";

export const Container = styled.div<{fullHeight: boolean;}>`
	/* padding: 10px; */
	background-color: #eaeff4;
	padding: 10px 0;
	height: ${props => props.fullHeight ? "88vh" : "auto"};
`;

export const DropContainerHeader = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
`;
export const FlexStartSpaced = styled.div`
	display: flex;
    justify-content: space-between;
    align-items: flex-start;
	a {
		font-weight: 700;
		text-transform: uppercase;
		cursor: pointer;
		color: #18cfa8;
		font-size: 13px;
		text-transform: uppercase;
	}
`;
export const DropContainer = styled.div<{ showItem: boolean }>`
	height: ${props => props.showItem ? "161.5px" : "0px"};
	margin-top: ${props => props.showItem ? "5px" : "0px"};;
	overflow: hidden;
	transition: height 0.3s ease-in-out, margin-top 0.5s ease-out;

`;

export const DropButton = styled.span<{ showItem: boolean }>`
	border: 1px solid #eaeff4;
    border-radius: 25px 25px;
	padding: 5px 15px;
	font-size: 11px;
	font-family: var(--font-lksans-regular);
	color: #666666;
	display: flex;
	gap: 5px;
	svg {
		color: #18cfa8;
		align-self: center;
		transform: ${props => props.showItem ? "rotate(180deg)" : ""};
		transition: transform 0.5s ease-out;
	}
`;

export const ItemCardContainer = styled.div`
	background-color: #fff;
	padding: 10px 10px 12px 10px;
	margin-bottom: 10px;
`;

export const ItemTitle = styled.div`
	font-family: var(--font-lksans-bold);
	color: #333333;
	font-size: 13px;
`;

export const AddressContainer = styled.div`
	background-color: #fff;
	margin-bottom: 10px;
	padding: 10px;
	color: #333333;
	font-family: var(--font-lksans-bold);
	font-size: 13px;
`;

export const ReturnOptions = styled.div`
	background-color: #fff;
	padding: 10px 10px 12px 10px;
`;

export const ReturnOptionsTitle = styled.div`
	font-family: var(--font-lksans-bold);
	color: #333;
	font-size: 13px;
	margin-bottom: 5px;
`;

export const ReturnOptionsSub = styled.div`
	font-size: 11px;
	font-family: var(--font-lksans-regular);
	color: #999999;
	padding: 5px 0 10px;
`;

export const ButtonContainer = styled.div`
	width: 100%;
	padding: 8px;
	position: fixed;
	bottom: 0;
	left: 0;
	background: #fff;
	button {
		width: 100%;
		background-color: #18CFA8;
		color: #ffffff;
		border: none;
		text-transform: uppercase;
		padding: 15px 0;
		border-radius: 4px;
		letter-spacing: 1px;
		text-align: center;
		font-family: var(--font-lksans-bold);
		font-size: 15px;
	}
`;
