// @lk/ui-library/src/Component/V1/MyOrderMobile/ReturnRefundMobilePage/ReturnRefundMobile.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import ReturnRefundMobile from "./ReturnRefundMobile.component";

export default {
	title: "Components/v1/MyOrder/ReturnRefundMobile",
	component: ReturnRefundMobile,
	args: {
	},
} as ComponentMeta<typeof ReturnRefundMobile>;

const Template: ComponentStory<typeof ReturnRefundMobile> = (args) => <ReturnRefundMobile {...args}/>;

export const Overview = Template.bind({});
Overview.args = {
	defaultValue: "store",
	refundOptions: [
		{
			iconUrl: "https://static.lenskart.com/media/mobile/universal/images/returnExchange/Steps_Store.svg",
			optionTitle: "Visit a Nearby Store",
			subtext: [
				"Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odit eum ea excepturi.",
				"Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ut ullam quaerat suscipit officiis neque.",
				"Lorem ipsum dolor sit amet consectetur, adipisicing elit. Laudantium provident doloribus adipisci perferendis!",
			],
			tagText: "Most Popular",
			value: "store",
		},
		{
			iconUrl: "https://static.lenskart.com/media/mobile/universal/images/returnExchange/SchedulePickup.svg",
			optionTitle: "Schedule a Pickup",
			subtext: [
				"Lorem ipsum dolor sit amet, consectetur adipisicing elit. Odit eum ea excepturi.",
				"Lorem ipsum dolor, sit amet consectetur adipisicing elit. Ut ullam quaerat suscipit officiis neque.",
			],
			mostPopular: true,
			value: "pickup",
		},
	],
};
