// @lk/ui-library/src/Component/V1/StudioFlow/MsiteGenderRadio/MsiteGenderRadio.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const Label = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: var(--fs-16);
  line-height: 24px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  padding-bottom: var(--pd-18);
`;

export const InputContainer = styled.div<{ isSelected: boolean; }>`
	display	: flex;
	gap: 10px;
	align-items: center;
  padding: var(--pd-6) var(--pd-16);
  background: ${({ isSelected }) => isSelected ? "var(--ghost-white)" : "var(--white)"};
  border: 1px solid ${({ isSelected }) => isSelected ? "var(--text)" : "var(--light-purple-10)"};
  border-radius: 99px;
  transition: 0.3s ease all;
`;

export const Text = styled.label`
	font-family: ${TypographyENUM.lkSansRegular};
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 150%;
	letter-spacing: 0.15px;
	color: #3C3C3C;
`;

export const RadioGroupContainer = styled.div<{isMsite?:boolean}>`
	display: flex;
	gap: ${props => props.isMsite? "10px":"50px"};
	align-items: center;
`;
