// @lk/ui-library/src/Component/Common/BottomSheetNew/BottomSheetNew.stories.tsx
import { action } from "@storybook/addon-actions";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import BottomSheetNew from "./index";

export default{
	title: "Components/Common/BottomSheetNew",
	component: BottomSheetNew,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: false,
			description: "String used as unique identifier",
			category: "General"
		}),
		show: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show or hide bottomsheet",
			category: "Component",
			defaultValue: true
		}),
		fullPage: generateArgTypes({
			type: "boolean",
			required: false,
			description: "show bottomsheet full page or not",
			category: "Component",
			defaultValue: false
		}),
		closeBottomSheet: generateArgTypes({
			type: "function",
			required: true,
			description: "handler function to close bottomsheet",
			category: "Component",
		}),
		children: generateArgTypes({
			type: "node",
			category: "Component",
			required: true,
			description: "Bottomsheet children component"
		}),
		backdrop: generateArgTypes({
			type: "boolean",
			category: "Component",
			required: false,
			description: "Whether or not close bottomsheet on clicking backdrop",
			defaultValue: true
		}),
		borderRadius: generateArgTypes({
			type: "boolean",
			category: "Component",
			required: false,
			description: "show border radius of Bottomsheet container",
			defaultValue: false
		})
	}
} as ComponentMeta<typeof BottomSheetNew>;

const Template: ComponentStory<typeof BottomSheetNew> = (args) => <BottomSheetNew {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	show: true,
	fullPage: false,
	backdrop: true
};

export const Dummy = Template.bind({});
Dummy.args = {
	show: true,
	fullPage: false,
	children: <><div style={{ padding: "50px" }}>this is content</div></>,
	closeBottomSheet: () => {
		action("onClick")("On bottom sheet close click...");
	}
};
