// @lk/ui-library/src/Component/Common/BottomSheetNew/BottomSheetNew.styles.tsx
import styled, { keyframes } from "styled-components";

export const BottomSheetWrapper = styled.div`
	background-color: var(--white);
	position: fixed;
	left: 0;
	top: 0;
	width: 100%;
	height: 100%;
	overflow: hidden;
	z-index: 200;
`;

export const BottomSheetOverlay = styled.div<{backdrop?:boolean}>`
	background-color: rgba(33,33,33,.75);
  pointer-events: auto;
	transition: opacity 0.45s;
    position: fixed;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
    z-index: 50;
`;

export const animation = keyframes`
	from{
		bottom: -100px;
	}
	to{
		bottom: 0;
	}
`;

export const BottomSheetContent = styled.div<{borderRadius?: boolean, fullPage?: boolean, show?:boolean}>`
	position: absolute;
	left: 0;
	right: 0;
	bottom: 0;
	background-color: var(--white);
	z-index: 101;
	border-radius: ${(props) => props.borderRadius ? "16px 16px 0 0" : ""};
	height: ${props => props.fullPage ? "100%" : ""};
	animation: ${animation} 0.5s ease;
	/* transform: ${(props) => (props.show ? "translateY(0%)" : "translateY(100%)")};
	transition: transform 0.25s ease; */
`;
export const Icon = styled.div`
	position: absolute;
	right: 16px;
	top: 16px;
`;
