// @lk/ui-library/src/Component/V1/CheckoutMobile/PostCheckoutSection/PostCheckout.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const PaymentSummaryContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 0 auto;
  height: 100vh;
  padding: 50px 16px 40px 16px;
`;
export const IconContainer = styled.div<{ success: boolean }>`
  margin-bottom: var(--fs-20);
  svg {
    height: ${(props) => (props.success ? "76px" : "40px")};
    width: ${(props) => (props.success ? "76px" : "40px")};
  }
`;
export const PaymentHeading = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  color: var(--dark-blue-100);
  font-weight: var(--fw-regular);
  font-size: var(--fs-16);
  line-height: var(--fs-24);
  text-align: center;
  letter-spacing: -0.02em;
`;
export const ThankyouPaymentHeading = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  color: var(--dark-blue-100);
  font-weight: var(--fw-regular);
  font-size: var(--fs-16);
  line-height: var(--fs-24);
  text-align: center;
  letter-spacing: -0.02em;
  padding-bottom:25px;
`;
// export const SubHead = styled.div`
//   font-family: ${TypographyENUM.lkSansBold};
//   font-style: normal;
//   font-weight: var(--fw-bold);
//   font-size: var(--fs-14);
//   line-height: var(--fs-20);
//   text-align: center;
//   letter-spacing: -0.02em;
//   color: var(--text);

//   padding: 10px 0;
// `;
export const MainInfo = styled.p`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: var(--fs-12);
  line-height: var(--fs-20);
  letter-spacing: -0.02em;
  color: var(--text);
  padding: 0px 0 var(--fs-20) 0;
  text-align: center;
`;
export const PrescriptionSection = styled.div<{success?:boolean}>`
  margin-bottom: 20px;
  margin-top:${props => props.success ? "0" : "20px"};
`;
export const WhatsNextBox = styled.div`
  display: flex;
  flex-direction: column;
  padding: 18px var(--fs-16);
  background: var(--white);
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.12);
  border-radius: var(--fs-12);
  min-width: 360px;
`;
export const WhatsNext = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 10px;
  width: 100%;
`;
export const BoxHead = styled.h3`
  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  /* font-weight: var(--fw-bold); */
  font-size: var(--fs-14);
  line-height: var(--fs-24);
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;
export const BoxMainText = styled.div`
  p {
    font-family: ${TypographyENUM.lkSansRegular};
    font-style: normal;
    font-weight: var(--fw-regular);
    font-size: var(--fs-12);
    line-height: var(--fs-20);
    color: var(--text);
    letter-spacing: -0.02em;
  }

  padding-bottom: 16px;
  margin-bottom: 14px;
  border-style: dashed;
  border-color: #d3d3d3;
  border-width: 0 0 1px 0;
`;

export const BottomButtonSection = styled.div`
  position: fixed;
  bottom: 30px;
  width: 90%;
  display: flex;
  flex-direction: column;
  gap: 8px;

  Button{
    padding:var(--fs-12) var(--fs-20);

    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    /* font-weight: var(--fw-bold); */
    font-size: var(--fs-16);
    line-height: var(--fs-24);
    letter-spacing: -0.02em;
    color: var(--dark-blue-100);
  }

  IconRight{
    margin-left: 15px;
  }
`;

export const AnimatedCheckBody = styled.div`
  width: 100%;
  height: 100vh;
  display: flex;
  align-items: center;
  justify-content: center;
`;

// export const AmountDebited = styled.div`
//   font-family: ${TypographyENUM.lkSansRegular};
//   font-style: normal;
//   font-weight: var(--fw-regular);
//   font-size: var(--fs-12);
//   line-height: var(--fs-20);
//   text-align: center;
//   letter-spacing: -0.02em;
//   color: var(--text);

//   padding-top:var(--fs-20);
// `;

export const ViewOrder = styled.div`
  text-decoration: underline;
  font-family: ${TypographyENUM.lkSansBold};
  display: inline-block;
`;

export const Icon = styled.div`
  margin-left: 15px;
  margin-bottom: -5px;
`;

export const RightArrowOuter = styled.div<{isRTL?: boolean}>`
height: 26px;
width: 38px;
border-radius: 40px;
border: 1px solid grey;
padding-left: ${props => props.isRTL ? "" : "11px"};
padding-top: 4px;

transform: ${props => props.isRTL ? "rotate(180deg)" : ""};

padding-right: ${props => props.isRTL ? "10px" : ""};
`;

export const SecondSection =styled.div`
  display:flex;
  justify-content:space-between;
  align-items:center;
`;

export const PowerAndPdButton = styled.div`
  font-weight: 600;
  font-size: 11px;
  padding: 12px;

  text-transform:uppercase;
  color:white;

  background :#18cfa8;
  margin-top:20px;
`;

export const ButtonWrapper = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
`;
