// @lk/ui-library/src/Component/Common/ApplyCouponBar/ApplyCouponBar.stories.tsx
import React from "react";
import ApplyCouponBar from "./ApplyCouponBar.component";
import { ComponentMeta, ComponentStory } from "@storybook/react";

export default {
	title: "Components/Common/ApplyCouponBar",
	component: ApplyCouponBar,
	argTypes: {

	}
} as ComponentMeta<typeof ApplyCouponBar>;

const Template: ComponentStory<typeof ApplyCouponBar> = (args) => <ApplyCouponBar {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	mobileView: true,
	// children:
	// <Coupon id="Coupon"
	// 	width="100"
	// 	brandImage="https://upload.wikimedia.org/wikipedia/commons/thumb/4/42/Paytm_logo.png/640px-Paytm_logo.png"
	// 	headText="FLAT100"
	// 	subText="Save ₹100 with FLAT100"
	// 	descText="Use voucher SINGLE & get Extra 20% Off on single product!"
	// 	onClick={() => {
	// 		action("onClick")("Apply Coupon Clicked");
	// 	} }
	// 	font={TypographyENUM.defaultBook}
	// 	termsAndCondition={"Cant be merged with Buy 1 Get 1 Free or any other offer"}

	// />,
	applicableGvs: [
		{
			code: "SINGLE",
			description: "Use voucher SINGLE & get Extra 20% Off on single product!",
			heading: "EXTRA 20% OFF ON SINGLE PRODUCT",
			priority: 1,
			termsAndConditions: "Cant be merged with Buy 1 Get 1 Free or any other offer",
			autoApplicable: false,
			brandImage: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png",
		},
	],
	showSidebar: true,
	sorryNoCoupon: "Sorry! No Coupons available at this moment.",
	isRTL: false,
};
