// @lk/ui-library/src/Component/Common/ApplyCouponBar/ApplyCouponBar.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const ManualApplyCoupon = styled.div<{ font: TypographyENUM }>`
	font-family: ${(props) => props.font} !important;
	position:relative;
	// &::after {
	// 	content: "";
	// 	width: 100vw;
	// 	height: 1px;
	// 	background-color: rgba(0,0,0,0.1);
	// 	display: block;
	// 	margin-top: 30px;
	// 	margin-bottom: 20px;
	// 	margin-left: -30px;
	// }
`;

export const Flex = styled.div<{ justify?: boolean, isRTL?: boolean }>`
  	display: flex;
  	align-items: center;
	/* flex-direction: ${props => props.isRTL ? "row-reverse" : "row"}; */
	${props => props.justify &&
		`justify-content: space-between;
		width: 100%;`}
`;

export const ApplyCouponInput = styled.input<{ isRTL?: boolean }>`
  width: 100%;
  height: 60px;
  ${(props) => (props.isRTL ? "padding-right: 10px" : "padding-left: 10px")};
  ${(props) =>
		props.isRTL ? "border-left: 0 !important" : "border-right: 0 !important"};
  border-radius: ${(props) =>
		props.isRTL ? "0 12px 12px 0" : "12px 0px 0px 12px"};
  text-transform: uppercase;
  border: 1px solid #cecedf;
  font-size: 16px;
  font-family: ${TypographyENUM.lkSansRegular};
  padding: 20px;
  box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
  :focus {
    outline: none;
  }
  user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;
export const ApplyButtonNew = styled.div<{ isRTL?: boolean }>`
	font-family: ${TypographyENUM.lkSansRegular};
	font-style: normal;
	font-weight: 700;
	font-size: 12px;
	line-height: 20px;
	letter-spacing: -0.02em;
	text-decoration-line: underline;
	color: #000042;
	position:absolute;
	right: ${props => props.isRTL ? "" : 20}px;
	left: ${props => props.isRTL ? 20 : ""}px;

`;
export const ApplyButton = styled.button<{
	isActive?: boolean;
	isRTL?: boolean;
}>`
	cursor: ${props => props.isActive ? "pointer" : "not-allowed"};

  font-size: 15px;
  font-family: var(--font-lksans-bold);
  height: 60px;
  background: white;
  border-radius: ${(props) =>
		!props.isRTL ? "0 12px 12px 0" : "12px 0px 0px 12px"};
  ${(props) =>
		!props.isRTL ? "border-left: 0 !important" : "border-right: 0 !important"};
  // ${(props) => (props.isRTL ? "padding-left: 10px" : "padding-right: 10px")};
  padding: 20px;
  border: 1px solid #cecedf;
  box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
  color: ${(props) =>
		props.isActive ? "var(--vivid-green-100)" : "var(--light-gray-25)"};
`;
// export const ApplyButton = styled.button<{ isActive?: boolean }>`
// 	cursor: ${props => props.isActive ? "pointer" : "not-allowed"};
//   	font-size: 15px;
//   	font-family: var(--font-lksans-bold);
//   	height: 60px;
//   	border-left: 0 !important;
//   	background: white;
//   	border-radius: 0 12px 12px 0;
//   	/* padding-right: 10px; */
//   	padding: 20px;
//   	border: 1px solid #cecedf;
//   	box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
//   	color: ${(props) => props.isActive ? "var(--vivid-green-100)" : "var(--light-gray-25)"};
// `;

export const ErrorMessage = styled.div`
	color: red;
	margin-top: 5px;
	font-size: 14px;
`;

export const CouponTitle = styled.span`
	font-family: ${TypographyENUM.lkSansBold};
	font-size: 14px;
	line-height: 16px;
	padding: 10px 10px 0 10px;
	letter-spacing: -0.3px;
	color: #333333;
`;

export const Content = styled.div`
  /* padding: var(--pd-10); */
  gap: 15px;
  display: flex;
  flex-direction: column;
`;


export const ManualApplyCouponNew = styled.div<{ isRTL?: boolean }>`
	display: flex;
	/* flex-direction: row; */
	flex-direction: ${props => props.isRTL ? "row-reverse" : "row"};
	align-items: center;
	justify-content: space-between;
	width: 100%;
	gap: 16px;
`;

export const InputContainer = styled.div`
	width: 100%;
`;

export const ApplyButtonNewDesign = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: center;
	align-items: center;
	padding: 14px 20px;
	gap: 8px;
	min-width: 87px;
	font-weight: 700;
	font-size: 16px;
	line-height: 20px;
	font-family: var(--font-lksans-bold);
	height: 58px;
	border: 1.5px solid var(--dark-blue-100);
	border-radius: 12px;
	color: #000042;
	cursor:pointer;
`;

export const ApplyCouponInputNew = styled.input`
  width: 100%;
  height: 60px;
  border-radius: 8px;
  text-transform: uppercase;
  border: 1px solid #A2A2B8;
  font-size: 16px;
  font-family: ${TypographyENUM.lkSansRegular};
  padding: 16px;
  box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
  :focus {
    outline: none;
  }
  user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const LineThroughHeadingContainer = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-between;
	gap: 10px;
	width: 100%;
	margin: 15px 0 10px;
`;

export const Line = styled.div`
width: 100%;
height: 0px;
border: 1px dashed #CECEDF;
`;

export const LineHeading = styled.div`
	font-family: ${TypographyENUM.lkSansBold};
	font-style: normal;
	font-weight: 700;
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: #66668E;
	min-width: fit-content;
`;
