// @lk/ui-library/src/Component/Common/PrimaryButton/PrimaryButton.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from "../../../Types/general";
import PrimaryButton from "./index";

export default {
	title:"Components/Common/PrimaryButton",
	component: PrimaryButton,
	argTypes:{
		...generateGeneralArgs(),
		height: generateArgTypes({
			type: "string",
			required: false,
			description: "Height of the component",
			category: "Component"
		}),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "Width of the component",
			category: "Component"
		}),
		primaryText: generateArgTypes({
			type: "string",
			required: false,
			description: "Primary text of the component",
			category: "Component"
		}),
		secondaryText: generateArgTypes({
			type: "string",
			required: false,
			description: "Secondary of the component",
			category: "Component"
		}),
		borderColor: generateArgTypes({
			type: "string",
			required: false,
			description: "border color",
			category: "Component",
			defaultValue:"var(--turquoise)"
		}),
		onClick: generateArgTypes({
			type:"function",
			required:false,
			description:"on click handler for button",
			category:"Events",
		})
	}
} as ComponentMeta<typeof PrimaryButton>;

const Template: ComponentStory<typeof PrimaryButton> = args => <PrimaryButton {...args}/>;

export const Overview = Template.bind({});
Overview.args = {
	id:"primary-button",
	height:"34px",
	width:"100px",
	primaryText:"Primary text",
	secondaryText:"Secondary text",
	theme:ThemeENUM.primary,
	font:TypographyENUM.defaultBook,
	componentSize:ComponentSizeENUM.medium,
	icon: ""
};
