// @lk/ui-library/src/Component/Common/PrimaryButton/styles.tsx
import styled from "styled-components";
import { ComponentSizeENUM } from "../../../Types/general";

const getComponentSize = (size: ComponentSizeENUM | undefined) => {
	if (size === ComponentSizeENUM.small) {
		return "var(--fs-12)";
	} else if (size === ComponentSizeENUM.medium) {
		return "var(--fs-14)";
	} else {
		return "var(--fs-16)";
	}
};

export const Img = styled.img`
	height: 24px;
	margin-right:10px;
`;

export const PrimaryButtonWrapper = styled.button<{
	width:string,
	height:string,
	styleFont:string,
	color: string | undefined
	backgroundColor: string | undefined
	styleBorder: string | undefined
	mobileView?: boolean
	icon?:string
	padding?:string
	positionRight?: boolean
	borderRadius?:string
}>`
	width: ${ props => props.mobileView ? "100%" : "" };
	display:flex;
	flex-direction: ${props => props.icon !== "" ? "row" : "column"};
	justify-content:center;
	align-items:center;
	gap:2px;
	color:${(props) => props.color ? props.color : "var(--turquoise)"};
	min-width:${props => props.width};
	min-height:${props => props.height};
	/* padding:var(--pd-8) var(--pd-5); */
	padding: ${props => props.padding || "var(--pd-8) var(--pd-5)" };
	background-color:${(props) => props.backgroundColor ? props.backgroundColor: "var(--turquoise)"};
	border: 1px solid ${props => props.styleBorder ? props.styleBorder: "var(--turquoise)"};
	border-radius:${props => props.borderRadius ? `var(--border-radius-${props.borderRadius})` : "var(--border-radius-xxs)"};
	cursor: var(--pointer);
	font-family:${props => props.styleFont};
	${props => props.positionRight ? "margin-left: auto" : "margin-right: auto"};
	+ #btn-primary{
		 background:none;
		 color: #00bac6;
	}
`;

export const PrimaryText = styled.p< {componentSize:ComponentSizeENUM | undefined, weight?: number}>`
margin:0;
font-size:${props => getComponentSize(props.componentSize)};
letter-spacing: 1px;
font-family: inherit !important;
${props => props?.weight && `font-weight: ${props.weight}`};
`;

export const SecondaryText = styled.small`
margin:0;
display:block;
`;
