// @lk/ui-library/src/Component/V1/MyOrderMobile/InfoPopMobile/InfoPopMobile.component.tsx
import React, { useEffect } from "react";
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

const InfoText = styled.div<{ isOpen: boolean, timeVal: number }>`
	position: fixed;
	background-color: #0b0b35;
	color: white;
	display: inline;
	transition: bottom ease-in-out;
	transition-duration: ${props => props.timeVal}ms;
	text-align: center;
	margin: auto;
	left: 50%;
	transform: translateX(-50%);
	padding: 15px;
	border-radius: 5px;
	font-family: ${TypographyENUM.lkSansRegular};
	font-size: 10px;
	z-index: 999999;
	bottom: ${props => props.isOpen ? "24px" : "-60px"};
	width: max-content;
`;

interface InfoPopMobileTypes {
	isOpen: boolean;
	closeHandler: () => void;
	time?: number;
	text: string;
}

const InfoPopMobile = ({ isOpen, closeHandler, time, text }: InfoPopMobileTypes) => {

	const timeVal = time ? time : 1500;

	let timerVar: NodeJS.Timeout | undefined;
	useEffect(() => {
		console.log(isOpen);
		if(isOpen) {
			timerVar = setTimeout(() => {
				closeHandler();
			},timeVal);
		}

		return () => {
			if(timerVar) {
				clearTimeout(timerVar);
			}
		};
	},[isOpen]);

	return (
		<InfoText isOpen={isOpen} timeVal={timeVal/2}>
			{text}
		</InfoText>
	);
};

export default InfoPopMobile;
