// @lk/ui-library/src/Component/V1/PackageScreenNew/AboutEye/AboutEyePower.styles.tsx
import styled, { keyframes } from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

const animateRight = keyframes`
  from {
		transform: translate3d(30%, 0, 0);
	}
	to {
		transform: translate3d(0, 0, 0);
	}
`;

export const AboutEyePowerContainer = styled.div`
	// display: flex;
	// align-items: center;
	// flex-direction: column;
	// justify-content: center;
	flex: 1;
	gap: 20px;
	margin:4% 14%;
	animation-name: ${animateRight};
	animation-duration: 0.5s;
`;

export const Heading = styled.div`
	h2{
		font-family: ${TypographyENUM.lkSerifNormal};
		font-style: normal;
		font-weight: 200;
		font-size: 32px;
		line-height: 40px;
		letter-spacing: -0.02em;
		color: var(--dark-blue-100);
		margin-bottom:20px;
	}
`;

export const SubHeadingContainer = styled.div`
	display: flex;
	flex-direction: column;
	gap: 26px;
`;

export const SubHeading = styled.div`
	font-family: ${TypographyENUM.lkSansBold};
	font-style: normal;
	/* font-weight: 700; */
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: var(--text);
`;

export const RowContainer = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	gap: 10px;
	svg{
		width:32px;
		height:32px;
	}
`;
