// @lk/ui-library/src/Component/V1/PDP/Review/ReviewDetails/ReviewDetails.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const ReviewDetailsContainer = styled.div<{maxHeight:string,styleFont:TypographyENUM}>`
display:flex;
flex-direction:column;
background-color:var(--white);
max-height:${props => props.maxHeight};
font-family:${props => props.styleFont};

`;

export const ReviewsTitle = styled.p`
font-weight: var(--fw-medium);
text-align: center;
font-size:var(--fs-28);
color:var(--blackish-gray);
margin:0;
`;

export const ReviewHeader = styled.div`
padding:var(--pd-40);
display:flex;
flex-direction:column;
gap:var(--pd-40);
box-shadow: 0 0 16px rgb(0 0 0 / 10%);
height:max-content;
`;

export const ProductDetails = styled.div<{isRatingtext: boolean}>`
display:flex;
flex-direction:column;
justify-content:center;
gap:var(--spacing-sm);
font-weight:var(--bold);
max-width: ${props => props.isRatingtext ?"70%":"30%"};
overflow-wrap: break-word;
`;

export const ProductDetailsWrapper = styled.div<{isRTL: boolean}>`
display:flex;
justify-content: var(--space-between);
align-items:center;
flex-direction: ${props => props.isRTL ? "row-reverse": ""}
`;

export const ProductBrand = styled.p<{isRTL: boolean}>`
margin:0;
font-size:var(--fs-14);
color:var(--whitish-gray);
text-align: ${props => props.isRTL ? "end": ""};
font-family: ${TypographyENUM.lkSansBold};
letter-spacing: -.02em;
`;

export const ProductName = styled.p<{isRTL: boolean}>`
margin:0;
font-size:var(--fs-16);
// color:var(--serene-gray);
letter-spacing: 1px;
text-align: ${props => props.isRTL ? "end": ""};
`;

export const WriteReview = styled.button<{styleFont:TypographyENUM}>`
text-align:right;
background:none;
border:none;
cursor: var(--pointer);
margin:0;
font-size:var(--fs-14);
color:var(--irish-blue);
font-family:${props => props.styleFont};
`;

export const ReviewListWrapper = styled.div`
overflow:scroll;
padding:var(--pd-20) var(--pd-40);
`;

export const LoaderDiv = styled.div`
    position: absolute;
    bottom: 0;
    left: 0;
    right: 0;
    background-color: white;

    img{
        width: 100%;
        height: 100%;
    }

`;
