// @lk/ui-library/src/Component/V1/PDP/Review/ReviewDetails/ReviewDetails.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import { ReviewDetails } from "./ReviewDetails.component";

export default {
	title: "Components/V1/PDP/Review/ReviewDetails",
	component: ReviewDetails,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		maxHeight: generateArgTypes({
			type: "string",
			required: true,
			description: "component max height",
			category: "Component",
		}),
		productBrand: generateArgTypes({
			type: "string",
			required: true,
			description: "product brand",
			category: "Component",
		}),
		productName: generateArgTypes({
			type: "string",
			required: true,
			description: "product name",
			category: "Component",
		}),
		numberOfReviews: generateArgTypes({
			type: "number",
			required: true,
			description: "number of reviews",
			category: "Component",
		}),
		avgRating: generateArgTypes({
			type: "number",
			required: true,
			description: "product avg rating",
			category: "Component",
		}),
		bottomBoxId: generateArgTypes({
			type: "string",
			required: true,
			description: "bottom box id",
			category: "Component",
		}),
		writeReviewHandler: generateArgTypes({
			type: "function",
			required: false,
			description: "event on write review",
			category: "Event",
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			defaultValue:{
				WRITE_A_REVIEW: "Write A Review",
				REVIEWS: "Reviews",
			},
			control:"object"
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show right to left",
			category: "Status",
			defaultValue: "false"
		}),
	}
} as ComponentMeta<typeof ReviewDetails>;

const Template: ComponentStory<typeof ReviewDetails> = args => <ReviewDetails {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"review-details",
	maxHeight:"400px",
	productBrand:"Lenskart Air",
	productName:"Tortoise Full Rim Wayfarer Eyeglasses",
	numberOfReviews: 309,
	avgRating: 4.75,
	bottomBoxId: "xyz-btm-box-id",
	dataLocale:{
		WRITE_A_REVIEW: "Write A Review",
		REVIEWS: "Reviews",
	},
	font:TypographyENUM.defaultBook,
	writeReviewHandler: () => null
};
