// @lk/ui-library/src/Component/V1/FilterComponent/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const FilterWrapper = styled.div<{ innerHeight?: number }>`
font-family: ${TypographyENUM.lkSansMedium};
  overflow: hidden;
  display:flex;
  background-color:transparent;
  color:#27394e;
  width: 100vw;
  height: calc(${props => props.innerHeight}px - 115px);
`;

export const FilterTypesWrapper = styled.div`
display:flex;
flex-direction:column;
width:36%;
height:100%;
overflow-y: scroll;
-ms-overflow-style: none;
scrollbar-width: none;
background-color: white;

&::-webkit-scrollbar {
  display: none;
}

&:first-child {
  border-top: 1px solid #18cfa8;
}
`;


export const FilterHeaderWrapper = styled.div`
font-family: ${TypographyENUM.lkSansBold};
width: 100%;
z-index: 2;
background: #fff;
color:#27394e;
`;
export const FilterHeader = styled.div`
height: 55px;
position: relative;
align-items: center;
display: flex;
`;
export const FilterHeaderText = styled.span`
font-size: 22px;
`;

export const FilterHeaderIcon = styled.span`
&:before {
  content: "";
}
margin-right:20px;
margin-left: 20px;
color: #18cfa8;
font-size: var(--fs-22);
`;

export const FilterType = styled.div<{isSelected: boolean}>`
display:flex;
flex-direction:row;
align-items:center;
border-bottom: 1px solid #18cfa8;
background: ${props => props.isSelected ? "#18cfa8" : "#fff" };
position: relative;
`;


export const FilterTypes = styled.div<{isSelected: boolean}>`
	display:flex;
	text-transform: capitalize;
	font-size: var(--fs-12);
	min-height: 50px;
	align-items: center;
	min-width: 115px;
	padding: 5px 10px;
	width: 100%;
	/* max-width: min-content; */
	align-items: center;
	padding-left: 20px;
	background-color: ${(props) => props.isSelected ? "#18cfa8" : "#fff"};
  	color: ${(props) => props.isSelected ? "#fff" : null};
    justify-content: space-between;
`;

export const FilterIndicator = styled.span<{
  isSelected:boolean
}>`
height: 5px;
    width: 5px;
    left: 4px;
    background-color:${(props) =>
		props.isSelected ? "#fff" : "#18cfa8"};
    border-radius: 50%;
    position:absolute;
    margin-right:2px;
`;

export const FilterOptionsWrapper = styled.div`
display:flex;
flex-direction:column;
width:64%;
height:100%;
overflow-y: scroll;
background-color:#eaeff4;
`;

export const FilterOptions= styled.div`
display:flex;
flex-direction:row;
justify-content: space-between;
padding:10px 20px;
height: 50px;
align-items: center;
background-color:#fff;
`;

export const FilterOptionTitle= styled.span`
font-size: var(--fs-12);
text-transform: capitalize;
`;
export const FilterOptionCheckbox= styled.input.attrs({ type: "checkbox" })`
&:before {
  content: "";
}
margin: 0 4px;
transform: scale(1.5);
user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;
export const FilterApplyBtnWrapper= styled.div`
font-family: ${TypographyENUM.lkSansRegular};
bottom: 0;
    width: 100%;
    z-index: 10;
    top: auto;
    position: fixed;
    padding:10px;
    display: flex;
	background:#fff;
   `;
export const FilterBtn = styled.span<{
  isApply:boolean
}>`

flex: 1;
text-align: center;
text-transform: uppercase;
border-radius: 4px;
letter-spacing: 1px;
border: 1px solid #18cfa8;
padding:10px 10px;
margin:0 5px;
background-color:${(props) =>
		props.isApply ? "#18cfa8" : null};
color:${(props) =>
		props.isApply ? "#fff" : null}
`;

export const FilterCount= styled.span<{
  isSelected:boolean
}>`
min-width: 15px;
font-size: var(--fs-14);
right: 10px;
color:${(props) =>
		props.isSelected ? "#fff" : "#18cfa8"};
`;

export const Root = styled.div`
  position: fixed;
  top: 0;
  z-index: 100;
  bottom: 0;
  left: 0;
  right: 0;
`;
