// @lk/ui-library/src/Component/V1/PhonecodeDropdown/styles.tsx
import styled, { css } from "styled-components";

const CommonCss = css<{width?: string}>`
	border: none;
	background: none;
	box-shadow: none;
	color: #999999;
	opacity: 1;
	left: 0;
	outline: none;
`;
export const CountryCodeSelect = styled.select<{width?: string,incCountryCodeFont?: boolean, showSvgArrow?: boolean,msiteAddressPage?: boolean, isMobileView?: boolean}>`
	${CommonCss}
	width: ${props => props.width ? props.width : "40px"};

	${props => props.incCountryCodeFont ? "font-size:16px;" : ""};

	${props => props.showSvgArrow && css`
	appearance: none;
  ${css`
    background-image: url(https://static.lenskart.com/media/Assets/Images/down-arrow.svg);
  `}
	background-position: 87% 11%;
	background-repeat: no-repeat;
	`}
	
	${props => props.msiteAddressPage && css`
	background-position: 119% 11%;
	`};

	${props => props.isMobileView && css`
	background-position: 116% 11%;
	`}
`;

export const CountryCodeField = styled.input<{width?: string}>`
	${CommonCss}
	// width: ${props => props.width ? props.width : "40px"};
    font-size: 16px;
	user-select: initial;
	color: #010042;
	&::before, &::after {
    	user-select: initial;
	}
`;
