// @lk/ui-library/src/Component/V1/PlaceholderDropdown/Dropdown.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../StorybookHelper";
import {
	ComponentSizeENUM,
	TypographyENUM,
} from "../../../Types/general";
import Dropdown from "./index";

export default {
	title: "Components/v1/PlaceholderDropdown",
	component: Dropdown,
	argTypes: {
		...generateGeneralArgs(),
		options: generateArgTypes({
			type: "array",
			required: true,
			description: "Show dropdown options",
			category: "Component",
		}),
		index: generateArgTypes({
			type: "number",
			required: true,
			description: "Show cuurent active index",
			category: "Component",
		}),
	},
} as ComponentMeta<typeof Dropdown>;

const Template: ComponentStory<typeof Dropdown> = (arges) => (
	<Dropdown {...arges} />
);

export const Overview = Template.bind({});
Overview.args = {
	id: "dropdrow-id",
	componentSize: ComponentSizeENUM.medium,
	font: TypographyENUM.serif,
	options: [{ key:"Your Picks", value:"your_picks" },
		{ key:"Discover", value:"dicover" },
		{ key:"Categories", value:"categories" },
		{ key:"Gold", value:"gold" },
		{ key:"Collabs", value:"collabs" },
		{ key:"Labs", value:"labs" }],
	SelectedIndex: 0,
	placeholder: "Placeholder",
	handleChange: () => null,
};
