// @lk/ui-library/src/Component/V1/PlaceholderDropdown/styles.tsx
import styled, { css } from "styled-components";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";

const getSize = (size: ComponentSizeENUM) => {
	if (size === "small") return "var(--fs-12)";
	if (size === "medium") return "var(--fs-16)";
	else return "var(--fs-18)";
};

export const DropdownWrapper = styled.div<{
	styleSize: ComponentSizeENUM;
	styleFont: TypographyENUM;
	hasError?: boolean;
}>`
	font-family: ${(props) => props.styleFont};
	font-size: ${(props) => getSize(props.styleSize)};

	padding: 0;
    margin: 0 0 10px 0;
    width: 100%;
    border-radius: 4px;
    overflow: hidden;
		${css`
    background-image: url(https://static.lenskart.com/media/Assets/Images/down-arrow.svg);
  `}

	background-position: 96% 50%;
	background-repeat: no-repeat;
    border: 1px solid ${({ hasError }) => hasError ? "red" : "#d3d3d3"};
`;

export const DropdownLabel = styled.label`
	font-family: inherit;
	font-size: 1em;
	letter-spacing: 1.5px;
	text-transform: var(--uppercase);
	color: var(--vivid-green-100);
	line-height: 30px;
`;

export const SelectBox = styled.select`
	padding: 14px 24px 14px 8px;
    width: 100%;
    border: none;
    font-size: 14px;
    outline: none;
    -webkit-box-shadow: none;
    box-shadow: none;
    background-color: transparent;
    background-image: none;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
    white-space: pre-wrap; /* Allows line breaks in the placeholder text */
    word-wrap: break-word; /* Allows long words to break onto the next line */
	cursor:pointer;
`;

export const Option = styled.option`
	font-family: inherit;
	font-size: 1em;
	line-height: 1em;
`;
