// @lk/ui-library/src/Component/Mobile/Carousel/Carousel.styles.tsx
import styled, { css } from "styled-components";

export const CarouselMain = styled.section`
    box-sizing: border-box;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    -webkit-touch-callout: none;
    -khtml-user-select: none;
    -ms-touch-action: pan-y;
    touch-action: pan-y;
    -webkit-tap-highlight-color: transparent;
    position: relative;
    width: 100vw;
`;

const SlideCommonCss = css`
    position: relative;
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    -o-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
`;
export const CarouselSlideContainer = styled.div<{ isPfu: boolean; }>`
    ${SlideCommonCss};
    overflow: hidden;
    margin: 0;
    padding: 0;
    ${props => props.isPfu ? `
        display: contents;
        width: 358px;
        height: 96vh;
    ` : ""}
`;
export const CarouselTrack = styled.div`
    ${SlideCommonCss};
    display: flex;
`;
export const CarouselSlide = styled.div<{ isPfu: boolean; active: boolean; }>`
    ${props => !props.isPfu ? `
        img {
            width: 100%;
            max-width: 100vw;
        }
    ` : ""}
`;
const CarouselDirCss = css`
    position: absolute;
    display: block;
    height: 100%;
    width: 40px;
    cursor: pointer;
    top: 50%;
    -webkit-transform: translateY(-50%);
    -ms-transform: translateY(-50%);
    transform: translateY(-50%);
    padding: 0;
    border: none;
    z-index: 1;
    background: transparent;
    color: transparent;
    outline: none;
    &:focus,
    &:hover {
        background: transparent;
        color: transparent;
        outline: none;
    }
    &:before {
        content: "\27E8";
        margin-left: 5px;
        color: #9e9e9e;
        opacity: 1;
        font-size: 40px;
    }
`;
export const CarouselPrev = styled.button<{ isPfu: boolean; }>`
    ${CarouselDirCss};
    left: 0;
    ${props => props.isPfu ? `
        margin-left: 4px;
        margin-top: -4px;
    ` : ""}
`;
export const CarouselNext = styled.button<{ isPfu: boolean; }>`
    ${CarouselDirCss};
    right: 0;
    &:before {
        margin-right: 5px;
    }
`;
export const NavDotsContainer = styled.div`
  bottom: 30px;
    position: relative;
    display: flex;
    justify-content: center;
`;
export const NavDot = styled.div<{ active: boolean }>`
    height: 10px;
    width: 10px;
    margin: 0 5px;
    background-color: ${props => props.active ? "#18cfa8" : "#fff"};
    border-radius: 50%;
    transition: background-color 0.6s ease;
    border: 1px solid #ddd;
`;
export const DotContainer = styled.div`
    width: 100%;
    position: absolute;
    bottom: 11vh;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    gap: 10px;
    padding: 0 20px;
    z-index: 1;
`;
export const DotItem = styled.div`
    width: 8px;
    height: 8px;
    background: #FFFFFF;
    border: 0.5px solid #A2A2B8;
    border-radius: 99px;
`;
