// @lk/ui-library/src/Component/Common/Dropdown/Dropdown.styles.tsx
import styled from "styled-components";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";

const getSize = (size: ComponentSizeENUM) => {
	if (size === "small") return "var(--fs-12)";
	if (size === "medium") return "var(--fs-16)";
	else return "var(--fs-18)";
};

export const DropdownWrapper = styled.div<{
	styleSize: ComponentSizeENUM
	styleFont: TypographyENUM
}>`
    font-family: 'LKFuturaStd-Heavy';
	font-size: ${(props) => getSize(props.styleSize)};
	display: flex;
	align-items: center;
`;

export const SortIconContainer = styled.span`
	color: #33b7aa;
	margin-top: 2px;
	margin-right: 2px;
`;

export const DropdownLabel = styled.label`
	font-family: inherit;
	font-size: 1em;
	letter-spacing: 1.5px;
	text-transform: var(--uppercase);
	color: #33b7aa;
	line-height: 30px;
	font-weight: 900;
	margin-right: 6px;
`;

export const SelectBox = styled.select`
	border-radius: 0;
    -webkit-appearance: initial;
    -moz-appearance: initial;
    padding: 3px 38px 3px 8px;
    background: var(--white) url(//static.lenskart.com/media/desktop/img/select-bg.png) no-repeat scroll right center!important;
    cursor: pointer;
    border: 1px solid var(--light-grey);
	font-family: var(--font-default-book);
	font-synthesis: none;
	font-size: 1em;
	line-height: 22px;
`;

export const Option = styled.option`
	font-family: inherit;
	font-size: 1em;
	line-height: 1em;
`;
