// @lk/ui-library/src/Component/Common/Dropdown/Dropdown.stories.tsx
import { action } from "@storybook/addon-actions";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";
import Dropdown from "./Dropdown.component";

export default {
	title: "Components/common/Dropdown",
	component: Dropdown,
	argTypes: {
		...generateGeneralArgs(),
		options: generateArgTypes({
			type: "array",
			required: true,
			description: "Show dropdown options",
			category: "Component",
		}),
		index: generateArgTypes({
			type: "number",
			required: true,
			description: "Show current active index",
			category: "Component",
		}),
		isRTL : generateArgTypes({
			type: "boolean",
			required: true,
			description: "display content right to left",
			category: "",
		})
	},
} as ComponentMeta<typeof Dropdown>;

const Template: ComponentStory<typeof Dropdown> = (arges) => (
	<Dropdown {...arges} />
);

export const Overview = Template.bind({});
Overview.args = {
	id: "dropdrow-id",
	componentSize: ComponentSizeENUM.medium,
	font: TypographyENUM.serif,
	options: [
		{ key:"Your Picks", value:"your_picks" },
		{ key:"Discover", value:"discover" },
		{ key:"Categories", value:"categories" },
		{ key:"Gold", value:"gold" },
		{ key:"Collabs", value:"collabs" },
		{ key:"Labs", value:"labs" }
	],
	SelectedIndex: 0,
	dataLocale:{
		SORT_BY:"Sort By"
	},
	handleChange: (e) => { action("handleChange")(e.target.value); },
	isRTL: false,
};
