// @lk/ui-library/src/Component/V1/OldPayment/NetBanking/netBanking.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";
import { NetBanking } from "./index";

export default {
	title:"Components/V1/Payment/NetBanking",
	component: NetBanking,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: false,
			description: "Unique Id",
			category: "General",
			defaultValue: "near-by-store"
		}),
		font: generateArgTypes({
			type: "string",
			required: false,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.defaultBook,
			control: "select",
			options: TypographyENUM,
		}),
		storeImg: generateArgTypes({
			type: "string",
			required: false,
			description: "Image fo the store the be displayed",
			category: "Component",
			defaultValue: "https://static.lenskart.com/media/desktop/img/pdp/visit_store.png",
		}),
		storeInfo: generateArgTypes({
			type: "string",
			required: false,
			description: "Info Text of the store",
			category: "Component",
			defaultValue: "Please Share Your Location To See Nearby Stores",
		}),
		onClickStoreLocator: generateArgTypes({
			type: "function",
			required: true,
			description: "onClick handler for near by store",
			category: "Events",
		}),
	}
} as ComponentMeta<typeof NetBanking>;

const Template: ComponentStory<typeof NetBanking> = args => <NetBanking {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	applyOffer: () => console.log("Apply Code function cal"),
	banksList: [{
		"name": "HDFC Bank",
		"bankcode": "PU:HDFB",
		"logoImageUrl": "https://static5.lenskart.com/images/cust_mailer/feb-26th/HDFC.png",
		"offers": []
	},{
		"name": "State Bank of India",
		"bankcode": "PU:SBIB",
		"logoImageUrl": "https://static5.lenskart.com/images/cust_mailer/feb-26th/sbi.png",
		"offers": []
	},{
		"name": "State Bank of India",
		"bankcode": "PU:SBIB",
		"logoImageUrl": "https://static5.lenskart.com/images/cust_mailer/feb-26th/sbi.png",
		"offers": []
	},{
		"name": "State Bank of India",
		"bankcode": "PU:SBIB",
		"logoImageUrl": "https://static5.lenskart.com/images/cust_mailer/feb-26th/sbi.png",
		"offers": []
	},{
		"name": "State Bank of India",
		"bankcode": "PU:SBIB",
		"logoImageUrl": "https://static5.lenskart.com/images/cust_mailer/feb-26th/sbi.png",
		"offers": []
	},{
		"name": "State Bank of India",
		"bankcode": "PU:SBIB",
		"logoImageUrl": "https://static5.lenskart.com/images/cust_mailer/feb-26th/sbi.png",
		"offers": []
	}],
	createOrderPayment: () => console.log("Create Order Payment function call"),
	dataLocale: { PAY_USING_NETBANKING: "Pay using Netbanking", SHOW_MORE: "Show More" },
	frameProduct: false,
	loadingPlaceOrder: false,
	paymentCTA: "mainText: 'Place order', subText: '(Submit Power in the Next Step)"
};
