// @lk/ui-library/src/Component/V1/OldPayment/NetBanking/netBanking.styles.tsx
import styled from "styled-components";

export const Heading3 = styled.h3`
	font-family: var(--font-default-medium);
	font-size: var(--fs-13);
    color: var(--dark-gray);
    margin: var(--pd-15) 0;
`;

export const ListWrapper = styled.ul`
	margin: 0;
    padding: 0;
	max-height: 305px;
    overflow: scroll;
`;

export const BankListWrapper = styled.li`
	padding: var(--pd-15) var(--pd-20);
    background: var(--white);
    border: 1px solid var(--white);
    -webkit-box-sizing: border-box;
    box-sizing: border-box;
    border-radius: var(--border-radius-xxxs);
    margin: var(--pd-10) 0 0 0;
    display: flex;
    justify-content: center;
    align-items: center;
    cursor: pointer;
    font-size: var(--fs-15);
    color: var(--dark-gray);
    list-style-type: none;
	font-family: var(--font-default-heavy) ;
`;


export const BankIcon = styled.span``;

export const BankIconImage = styled.img`
	width: 28px;
    height: 28px;
    margin-right: var(--pd-15);
    vertical-align: middle;
`;

export const ArrowIcon = styled.i`
	margin-left: auto;
    font-style: normal;
`;

export const ShowMore = styled.div`
	display: flex;
    align-items: center;
    gap: var(--pd-5);
    justify-content: center;
	font-weight: var(--fw-bold);
	color: var(--irish-blue);
	font-size: var(--fs-13);
	margin-top: var(--pd-15);
	font-family: var(--font-default-book) ;
	font-synthesis: none;
	cursor: pointer;
`;
