// @lk/ui-library/src/Component/V1/CheckoutMobile/PayLaterBottomsheet/PayLaterBottomsheet.styles.tsx
import styled from "styled-components";

export const BottomSheetWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    background-color: #ffffff;
    min-height: 50vh;
    height: fit-content;
    padding: 20px 13px;
`;

export const Header = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-bottom: 20px;
`;

export const Title = styled.div`
  font-weight: 500;
  font-size: 20px;
  line-height: 160%;
  display: flex;
  align-items: center;
  letter-spacing: 0.15px;
  color: #3c3c3c;
  text-align: center;
  margin-bottom: 4px;
`;

export const Body = styled.div`
  margin: 10px 0px;
  padding: 0px 17px;
  text-align: center;
`;

export const Desc = styled.div`
  margin-bottom: 10px;
  justify-content: center;
  text-align: center;
  font-weight: 700;
  font-size: 14px;
  line-height: 143%;
  display: flex;
  align-items: center;
  letter-spacing: 0.15px;
  color: rgba(60, 60, 60, 0.54);
`;

export const SubmitButton = styled.div`
  text-transform: uppercase;
  width: 90%;
  padding: 8px 0px;
  margin: 12px;
  cursor: pointer;
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 4px;
  font-weight: 500;
  font-size: 15px;
  line-height: 26px;
  letter-spacing: 0.46px;
  text-transform: uppercase;
  background: #17c89e;
  box-shadow: 0px 3px 1px -2px rgba(0, 0, 0, 0.2),
    0px 2px 2px rgba(0, 0, 0, 0.14), 0px 1px 5px rgba(0, 0, 0, 0.12);
  border-radius: 4px;
  color: white;
`;

export const LaterButton = styled.div`
  font-weight: 500;
  font-size: 15px;
  line-height: 26px;
  letter-spacing: 0.46px;
  text-transform: uppercase;
  color: #17c89e;
`;

export const ButtonContainer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
`;
