// @lk/ui-library/src/Component/V1/HomePage/ProductImageCarousel/ProductImageCarousel.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import ProductImageCarousel from ".";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/V1/HomePage/ProductImageCarousel",
	component: ProductImageCarousel,
	argTypes: {
		...generateGeneralArgs(),
		title:generateArgTypes({
			type: "string",
			required: true,
			description: "title",
			category: "Component",
		}),
		onViewRangeClick:generateArgTypes({
			type: "function",
			required: true,
			description: "function called on view range button click",
			category: "Event",
		}),
	}
} as ComponentMeta<typeof ProductImageCarousel>;

const Template: ComponentStory<typeof ProductImageCarousel> = (args) => <ProductImageCarousel {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "banner-section",
	images:[
		{
			altText:"karan johar favorites",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/Black-Red-Full-Rim-Rectangle-Vincent-Chase-VINTAGE-VC-S13969-C1-Polarized-Sunglasses_vincent-chase-vc-s13969-c1-sunglasses_sunglasses_sunglasses_g_628518_02_2022.jpg"
		},
		{
			altText:"karan johar favorites",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-s14085-c2-c2-sunglasses_vincent-chase-vc-s14085-c2-c2-sunglasses_sunglasses_G_2293.jpg"
		},
		{
			altText:"karan johar favorites",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/black-green-full-rim-cat-eye-vincent-chase-polarized-vintage-vc-s13123-c18-sunglasses_g_3846_7_05_22.jpg"
		},
		{
			altText:"karan johar favorites",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-polarized-vc-s11075-c5-sunglasses_sunglasses_g_5035_1_1_1.jpg"
		},
		{
			altText:"karan johar favorites",
			redirectUrl:"https://www.lenskart.com/eyewear/collections/karan-johar-favorites.html",
			source:"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/371x178/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-s12585-c8-sunglasses_sunglasses_img_2170.jpg"
		},

	],
	title:"EYEGLASSES",
	font:TypographyENUM.defaultBook,
	dataLocale:{
		"VIEW_RANGE": "VIEW RANGE",
	},
	onViewRangeClick:()=>console.log("view range clicked")
};
