// @lk/ui-library/src/Component/V1/MyAccount/Mobile/RefundDetails/RefundDetails.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import { RefundDetails } from "./RefundDetails.component";
import { CurrencySymbol, CurrencyType } from "./RefundDetails.type";

export default {
	title: "Components/V1/MyAccount/Mobile/RefundDetails",
	component: RefundDetails,
	argTypes: {
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "radio",
			options: TypographyENUM,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left ",
			category: "Status",
			defaultValue: "false",
		})
	}
} as ComponentMeta<typeof RefundDetails>;

const Template: ComponentStory<typeof RefundDetails> = args => <RefundDetails {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	id: "refund-details",
	font: TypographyENUM.defaultBook,
	UrlImage: "https://static.lenskart.com/media/mobile/universal/images/returnExchange/TickGreen.svg",
	dataLocale: {
		REFUND_HEADING:"Refund Completed",
		REFUND_TEXT: "Your refund has been successfully done via Store Credit. You can instantly redeem these against any purchase",
		REFUND_METHOD:"Refund Method",
		REFUND_AMOUNT:"Refund Amount",
		REFUND_CREDIT_AMOUNT:"Store Credit Amount",
		REFUND_CODE_HEAD_TEXT:"Store Credit Code",
		COPY_CODE_TEXT:"Copy Code",
		HOW_TO_USE_STORE_CREDIT:"How to use Store Credit",
		REFUND_CODE_TEXT:"Apply Store Credit at payment step during your next purchase. We will also be sending this code to you via E-mail and SMS.",
		cancellationReasonText:"Cancellation Reason",
		changedMindText:"Changed Mind",
		CONTINUE_SHOPPING: "Continue Shopping",
	},
	"price": {
		"currency": CurrencyType.INR,
		"basePrice": 2000,
		"lkPrice": 0,
		"symbol": CurrencySymbol.INR
	},
};
