// @lk/ui-library/src/Component/V1/MyAccount/Mobile/RefundDetails/RefundDetails.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";


export const RefundInfoWrapper = styled.div<{styleFont:TypographyENUM}>`
	flex-grow:1;
	display:flex;
	flex-direction:column;
	gap:var(--pd-10);
	font-family:${props => props.styleFont};
	background:var(--white);
	padding:10px;
`;
export const RefundCompleteWrapper = styled.div`
   text-align:center;
   padding-top:10px;
`;
export const RefundInfoText = styled.p`
  text-align:center;
  padding-top:10px;
  color:var(--gray);
  font-size:var(--fs-12);
`;
export const RefundHead = styled.div`
  text-align:center;
  padding-top:10px;
  color:var(--black);
  font-size:var(--fs-16);
  font-weight: var(--fw-medium);
`;

export const RefundAmountWrapper = styled.div`
	display:flex;
	justify-content:space-between;
	margin-top:30px;
`;

export const RefundMethod = styled.span`
  margin-bottom:10px;
  color:var(--gray);
  font-size:var(--fs-14);
  font-weight: var(--fw-medium);
  line-height:16px;
  text-transform:uppercase;
`;
export const RefundAmount = styled.span`
  margin-bottom:10px;
  color:var(--gray);
  font-size:var(--fs-14);
  font-weight: var(--fw-medium);
  text-transform:uppercase;
  line-height:16px;
`;

export const SuccessImage = styled.div`
  text-align:center;
  img{
	max-width:40px;
  }
`;

export const RefundStoreCredit = styled.div`
	display:flex;
	justify-content:space-between;
	margin-top:0px;
`;
export const RefundStoreCode = styled.div`
	margin-top:20px;
	border-top: 1px solid #ddd;
	padding-top:5px;
`;

export const RefundCodeText = styled.p`
  text-align:center;
  padding-top:10px;
  color:var(--gray);
  font-size:var(--fs-12);
`;
export const RefundCodeHead = styled.div`
  text-align:center;
  padding-top:10px;
  color:#424242;
  font-size:var(--fs-14);
  font-weight: var(--fw-medium);
`;

export const RefundInputCode = styled.div`
  text-align:center;
  padding-top:15px;
  input{
	color:var(--gray);
	font-size:var(--fs-20);
	border: 2px dashed #f18132;
	color: #f18132;
	line-height:24px;
	padding:8px 17px;
	border-radius: 4px;
	text-align:center;
	font-weight: var(--fw-bold);
  }
`;

export const RefundStoreCreditAmount = styled.div`
	color:#424242;
	font-size:var(--fs-14);
	font-weight: var(--fw-medium);
	line-height:16px;
`;

export const CopyCode = styled.div`
	color:#18cfa8;
	font-size:var(--fs-12);
	font-weight: var(--fw-medium);
	line-height:16px;
	padding:10px 0 15px;
`;
export const UseStoreCredit = styled.div`
	color:#18cfa8;
	font-size:var(--fs-12);
	font-weight: var(--fw-medium);
	line-height:16px;
`;

export const RefundAmountPrice = styled.div`
	color:var(--black);
	font-size:var(--fs-14);
	font-weight: var(--fw-medium);
	text-transform:uppercase;
	line-height:16px;
`;

export const RefundCancalation = styled.div`

`;

export const CancellationReason = styled.div`
  border-bottom: 1px solid #eaeff4;
  color:var(--whitish-gray);
  font-size:var(--fs-14);
  font-weight: var(--fw-medium);
  padding:10px;
`;

export const ChangedMind = styled.div`
	color:#27394e;
	font-size:var(--fs-12);
	font-weight: var(--fw-medium);
	padding:10px;
`;

export const RefundBtnWrapper = styled.div`
	position:fixed;
	left:10px;
	right:10px;
	bottom:0;
`;
