// @lk/ui-library/src/Component/Common/Textarea/Textarea.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import FormTextarea from "./index";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import { ComponentSizeENUM } from "../../../Types/general";

export default {
	title: "Components/Common/Textarea",
	component: FormTextarea,
	argTypes: {
		value: generateArgTypes({
			type: "string",
			required: false,
			description: "Value that will be rendered",
			category: "Component",
			defaultValue: "Lorem Ipsum",
		}),
		placeholder: generateArgTypes({
			type: "string",
			required: false,
			description: "Placeholder that will be rendered",
			category: "Component",
			defaultValue: "Lorem Ipsum placeholder",
		}),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		}),
		disabled: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to enable/disable the input",
			category: "Component",
			defaultValue: "false",
		}),
		readonly: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to set readOnly condition in input",
			category: "Component",
			defaultValue: "false",
		}),
		error: generateArgTypes({
			type: "boolean",
			required: false,
			description: "Boolean to set error in input",
			category: "Component",
			defaultValue: "false",
		}),
		onChange: generateArgTypes({
			type: "function",
			required: false,
			description: "onChange handler for input",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show right to left",
			category: "Status",
		}),
	},
} as ComponentMeta<typeof FormTextarea>;

const Template: ComponentStory<typeof FormTextarea> = (args) => <FormTextarea {...args}/>;

export const Fields = Template.bind({});
Fields.args = {
	value: "Sample Value",
	label:"First Name",
	componentSize: ComponentSizeENUM.large,
	disabled: false,
	error: false,
	readonly: false,
	width: "50%",
};
