// @lk/ui-library/src/Component/Common/SwitchButton/styles.tsx
import styled from "styled-components";
import tokens from "../../../Styles/tokens";
import { ComponentSizeENUM } from "../../../Types/general";

export const SwitchButtonWrapper = styled.div`
	display: flex;
	align-items: center;
	font-size: 1.2vw;
	line-height: 1.4;
	letter-spacing: 1.5px;
	font-family: var(--font-default-book);
	font-synthesis: none;
`;

export const SwitchWrapper = styled.div`
	padding-right: var(--pd-20);
	padding-left: var(--pd-20);
`;

export const CmnToggleInputWrapper = styled.input`
	position: absolute;
	visibility: hidden;
	padding: 2px;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const LeftSpanWrapper = styled.span<{
	checked: boolean
	styleSize: ComponentSizeENUM
}>`
	cursor: var(--pointer);
	color: ${(props) =>
		!props.checked ? "var(--blashish-gray)" : "var(--black)"};
	//font-size: ${(props) => tokens.fontSize[props.styleSize]};
	font-size: 12px;
    font-weight: ${(props) =>
		!props.checked ? "400" : "600"};
		font-family: 'LKFuturaStd-Heavy';
`;

export const RightSpanWrapper = styled.span<{
	checked: boolean
	styleSize: ComponentSizeENUM
}>`
	cursor: var(--pointer);
	color: ${(props) =>
		!props.checked ? "var(--black)" : "var(--blashish-gray)"};
	font-size: ${(props) => tokens.fontSize[props.styleSize]};
	font-size: 12px;
    font-weight: ${(props) =>
		!props.checked ? "400" : "600"};
`;

export const CmnToggleLabelWrapper = styled.label<{
	checked: boolean
}>`
	display: block;
	position: relative;
	cursor: var(--pointer);
	outline: none;
	user-select: none;
	padding: 2px;
	width: 70px;
	height: 25px;
	background-color: var(--white);
	border-radius: var(--border-radius-xl);
	&:before {
		display: block;
		position: absolute;
		top: 1px;
		left: 1px;
		bottom: 1px;
		content: '';
		right: 1px;
		background-color: var(--blashish-gray);
		border-radius: var(--border-radius-xl);
		transition: background 0.4s;
		&:checked {
			background-color: var(--blashish-gray);
		}
	}
	&:after {
		display: block;
		position: absolute;
		top: 1px;
		left: ${(props) => (!props.checked ? "1px" : "inherit")};
		right: ${(props) => (props.checked ? "1px" : "inherit")};
		bottom: 1px;
		content: '';
		width: 24px;
		background-color: var(--white);
		border-radius: 100%;
		box-shadow: 0 2px 5px var(--light-shadow);
		transition: margin 0.4s;
		&:checked {
			margin-left: 44px;
		}
	}
`;
