// @lk/ui-library/src/Component/NewHtoFlow/HtoSixSignsCoursel/HtoSixSignsCoursel.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import HtoSixSignsCoursel from "./index";


export default {
	title: "NewHtoFlow/HtoSixSignsCoursel",
	component: HtoSixSignsCoursel,
	argTypes: {}
} as ComponentMeta<typeof HtoSixSignsCoursel>;

const Template: ComponentStory<typeof HtoSixSignsCoursel> = (args) => <HtoSixSignsCoursel {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		SIX: "6",
		SIGNS_YOU: "Signs you",
		NEED_AN_EYETEST: " need an eyetest"
	},
	courselConfig: {
		slidesToShow: 2,
		autoPlay: true,
		autoplaySpeed: 2000
	},
	courselArrayObject: [
		{
			heading1: "Frequent", heading2: "Headeches", icon: "https://static.lenskart.com/media/desktop/img/HTO/Person.svg",
		},
		{
			heading1: "Difficulty", heading2: "Reading", icon: "https://static.lenskart.com/media/desktop/img/HTO/A_char.svg",
		},
		{
			heading1: "Hazy", heading2: "Vision", icon: "https://static.lenskart.com/media/desktop/img/HTO/EyeMulti.svg"
		},
		{
			heading1: "Bright Light", heading2: "Discomfort", icon: "https://static.lenskart.com/media/desktop/img/HTO/Eye.svg"
		},
		{
			heading1: "Water Eyes", heading2: "Discomfort", icon: "https://static.lenskart.com/media/desktop/img/HTO/EyeWithDrop.svg",
		},
		{
			heading1: "Frequent", heading2: "Neck Pain", icon: "https://static.lenskart.com/media/desktop/img/HTO/PersonTwo.svg"
		}
	]
};
