// @lk/ui-library/src/Component/NewHtoFlow/HtoSixSignsCoursel/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const HtoSixSignsCourselWrapper = styled.div`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: flex-start;
    padding: 24px 16px;
    gap: 16px;
    background: #F7F2ED;
    width: 100%;
`;

export const SixSignsWrapper = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 0px 16px 0px 0px;
    gap: 8px;
`;

export const SixNumber = styled.div`
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: 800;
    font-size: 56px;
    line-height: 64px;
    display: flex;
    align-items: center;
    color: #010043;
`;

export const SixText = styled.div`
    font-family:  ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-16);
    line-height: 20px;
    color: #000042;
`;

export const CousealWrapper = styled.div`
    width: 100%;
`;
export const CourselContainer = styled.div`
padding: 0px 5px;
overflow: hidden;
.slick-track {
    display: flex !important;
    flex-direction: row;
    justify-content: space-between !important;
};
 `;
