// @lk/ui-library/src/Component/V1/Header/MainHeader/Dropdown/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

//gender => col1
export const GenderBox = styled.div<{active: boolean;}>`
	background-color: ${props => props.active ? "#f2ece6" : "#fff"};
    display: flex;
    padding: 10px 10px 10px 25px;
    cursor: var(--pointer);
    align-items: center;
    font-weight: var(--fw-bold);
    font-size: var(--fs-14);
    border-radius: 5px;
    margin-left: 30px;
    color: var(--dark-blue-100);
    font-family: ${TypographyENUM.lkSansRegular};
	&:hover {
		background-color: #f2ece6;
	}
`;

export const GenderImg = styled.img`
	display: block;
	margin: 0;
	margin: auto 12px;
	padding: 0;
	max-width: 48px;
`;

export const GenderText = styled.span`
	width: 50%;
	display: inline-block;
	text-transform: capitalize;
	margin: 0;
	padding: 0;
`;
export const ArrowRight = styled.i`
	padding-left: 30px;
	color: var(--dark-blue-100);
	&:before{
		content: "\f105";
	}
`;

//categories => col2
export const CategoryBox = styled.a<{active: boolean;}>`
	background-color: ${props => props.active ? "#f2ece6" : "#fff"};
	padding: 16px 10px 16px 25px;
	margin: 0;
	display: flex;
    align-items: center;
    justify-content: var(--space-between);
    min-height: 48px;
    color: var(--dark-blue-100);
    font-size: var(--fs-13);
	cursor: var(--pointer);
	white-space: nowrap;
    letter-spacing: 1px;
	text-decoration: none;
	&:hover {
		background-color: #f2ece6;
	}
`;

export const CategoryTextContainer = styled.div`
	display: flex;
	flex-direction: column;
	font-size: var(--fs-14);
	text-transform: capitalize;
`;

export const CategoryText = styled.span`
	display: block;
	color: var(--dark-blue-100);
	height: auto;
	line-height: 18px;
	font-family: ${TypographyENUM.lkSansRegular};
`;

//subcategories => col3.....n

export const SubMenuHead = styled.a`
	color: var(--dark-blue-100);
	/* font-weight: var(--fw-bold); */
	text-transform: capitalize;
	/* font-size: var(--fs-14); */
	display: block;
	text-decoration: none;
	font-family: ${TypographyENUM.lkSansBold};
	text-decoration: none;
	padding: 20px 20px 0 20px;
`;

export const HoverIcon = styled.div<{ isRTL?: boolean }>`
	font-size: 7px;
	opacity: 0;
	margin: ${props => props.isRTL ? "0 -35px 0 0" : "0 0 0 -5px"};
	display: flex;
	align-items: center;
	svg {
		path {
			stroke: rgba(0,0,66,.5);
		}
	}
`;

export const SubMenuLinkContainer = styled.div`
	display: flex;
	align-items: center;
	margin-bottom: 10px;
	margin-right: 5px;
    height: auto;
	&:hover {
		${HoverIcon} {
			opacity: 1;
		}
	}
`;

export const SubMenuLinkText = styled.a`
	display: block;
	font-size: 12px;
	line-height: 20px;
	font-family: ${TypographyENUM.lkSansRegular};
	padding: 0 20px 0 20px;
	cursor: var(--pointer);
	color: rgba(0,0,66,.5);
	&:hover {
		color: var(--dark-blue-100);
	}
`;

export const Icon = styled.div`
	font-size: 7px;
	display: flex;
	align-items: center;
	svg {
		path {
			stroke: var(--dark-blue-100);
		}
	}
`;
