// @lk/ui-library/src/Component/V1/FilterItem/FilterItem.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import FilterItem from "./index";
import { generateArgTypes } from "../../../StorybookHelper";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";

export default {
	title: "Components/v1/FilterItem",
	component: FilterItem,
	argTypes: {
		checked: generateArgTypes({
			type: "boolean",
			required: true,
			description: "status of checked of FilterItem",
			category: "Component",
		}),
		componentSize: generateArgTypes({
			type: "string",
			required: false,
			description: "Size of the component",
			category: "General",
			defaultValue: ComponentSizeENUM.small,
			control: "FilterItem",
			options: ComponentSizeENUM,
		}),
		disabled: generateArgTypes({
			type: "boolean",
			required: true,
			description: "is FilterItem disabled",
			category: "Component",
		}),
		imageUrl: generateArgTypes({
			type: "string",
			required: false,
			description: "Insert image url for image FilterItem button",
			category: "Component",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for filter",
			category: "Events",
		}),
	},
} as ComponentMeta<typeof FilterItem>;

const Template: ComponentStory<typeof FilterItem> = (args) => <FilterItem {...args} />;

export const ImageFilterItem = Template.bind({});
ImageFilterItem.args = {
	id: "FilterItem-image",
	checked: false,
	componentSize: ComponentSizeENUM.medium,
	font: TypographyENUM.serif,
	disabled: false,
	imageUrl:
		"https://static.lenskart.com/images/cust_mailer/Eyeglass/FullRim.png",
};
