// @lk/ui-library/src/Component/V1/FilterItem/styles.tsx
import styled from "styled-components";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";


export const FilterItemImageWrapper = styled.div`
	display: flex;
	flex-flow: column;
	justify-content: center;
	background-color: var(--white);
	width: 68px;
	height: 60px;
	cursor: var(--pointer);
`;

export const ImageSection = styled.div`
	display: flex;
	justify-content: center;
	filter: opacity(0.6);
`;

export const ImageItem = styled.img`
	width: 56px;
	height: 28px;
`;

export const ImageTextSection = styled.div`
	display: flex;
	justify-content: center;
	font-size: 12px;
	margin-bottom: 2.5px;
	margin-top: 2.5px;
	color: #2a2a2a;
`;

export const ImageText = styled.span`
	// text-overflow: ellipsis;
    // overflow: hidden;
    // white-space: nowrap;
	color: #2a2a2a;
    line-height: 12px;
    letter-spacing: 1px;
	font-family: ${TypographyENUM.defaultBook};
	font-synthesis: none;
	opacity: 0.6;
`;

export const FilterItemContainer = styled.label`
	display: flex;
	align-items: center;
	width: fit-content;
	> input {
		display: none;
	}
	> .image {
		cursor: var(--pointer);
		border: 1px solid rgba(205,205,205,0.6);
	}
	> .image:hover {
		border: 1px solid var(--black);
	}
	> .image:hover span{
		color:#000;
	}
	> input:checked + .image {
		border: 2px solid #1cbbb4;
	}
`;

export const FilterItemWrapper = styled.input<{
	styleSize: ComponentSizeENUM
}>`
	cursor: var(--pointer);

	&:checked {
		border-color: var(--terracotta-100);
	}
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;
export const LabelWrapper = styled.span<{
	styleFont: TypographyENUM
	styleSize: ComponentSizeENUM
}>`
	margin-left: var(--spacing-md);
	font-family: ${(props) => props.styleFont};
`;

export const Image = styled.div`
	opacity: var(--moderate-opacity);
	width: 60px;
	height: 60px;
	/* background-position: center center; */
	background-position: center -25%;
	background-color: var(--white);
	display: inline-block;
	margin: 10px;
	display: flex;
    flex-direction: column;
    justify-content: flex-end;
    align-items: center;
	margin: 0;

	&:hover {
		opacity: var(--default-opacity);
		background-position: center -25%;
	}
`;

export const Title = styled.p`
    margin: 0;
    font-size: 10px;
    position: relative;
    top: -4px;
`;
