// @lk/ui-library/src/Component/V1/AdvancedDropdown/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

// .dropd-arrow-icon {
// 	transition: transform 300ms ease-in-out;
// 	cursor: pointer;
// 	fill: currentColor;
// 	width: 1em;
// 	height: 1em;
// 	display: inline-block;
// 	font-size: 1.5rem;
// 	transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
// 	flex-shrink: 0;
// 	user-select: none;
//   }
export const DropCustomSelect = styled.div<{ disabled: boolean, font: TypographyENUM; fullWidth: boolean; }>`
	display: flex;
	flex-direction: column;
	text-align: center;
	border: 1px solid #c3c3c3;
	border-radius: 12px;
	text-align: start;
	cursor: pointer;
	min-height: 40px;
	width: ${props => props.fullWidth ? "100%" : "431px"};
	&:hover {
		border: 1px solid #000;
	}
	${props => props.disabled && `
		pointer-events: none;
		opacity: 0.4;
	`}
`;
//   .dropd-custom-select {
// 	display: flex;
// 	flex-direction: column;
// 	text-align: center;
// 	border: 1px solid #c3c3c3;
// 	border-radius: 3px;
// 	text-align: start;
// 	cursor: pointer;
// 	min-height: 40px;
// 	&:hover {
// 	  border: 1px solid #000;
// 	}
//   }
//   .disabled {
// 	pointer-events: none;
// 	opacity: 0.4;
//   }
export const DropSelectMain = styled.div`
	width: 100%;
	height: 100%;
	display: flex;
	border-radius: 12px;
	position: relative;
`;
//   .dropd-select-main {
// 	width: 100%;
// 	height: 100%;
// 	display: flex;
// 	border-radius: 3px;
// 	position: relative;
//   }
export const DropMainText = styled.div<{disabled?: boolean}>`
	width: 100%;
	padding: 18.5px 14px;
	cursor: pointer;
	height: 56px;
	span {
		color: ${props => props.disabled ? "" : "#a9a9a9"};
	}
`;
//   .dropd-main-text {
// 	width: 100%;
// 	padding: 18.5px 14px;
// 	cursor: pointer;
// 	height: 56px;
// 	span {
// 	  color: #a9a9a9;
// 	}
//   }
export const DropdDownIconContainer = styled.div<{isRTL: boolean}>`
	height: 100%;
	min-width: 40px;
	display: flex;
	align-items: center;
	justify-content: center;
	position: absolute;
	${props => props.isRTL ? "left: 0" : "right: 0"};
	top: 0;
`;
//   .dropd-dropdown-icon-container {
// 	height: 100%;
// 	min-width: 40px;
// 	display: flex;
// 	align-items: center;
// 	justify-content: center;
// 	position: absolute;
// 	right: 0;
// 	top: 0;
//   }
export const DropdDropLabel = styled.label<{isRTL: boolean, disabled?: boolean}>`
	position: absolute;
	top: 0;
	${props => props.isRTL ? "right: 5px" : "left: 5px"};
	font-size: 11px;
	font-weight: 400;
	transform: translateY(-50%);
	opacity: 1;
	z-index: 2;
	background-color: #fff;
	padding: 0 4px;
	color: ${props => props.disabled ? "" : "rgba(0, 0, 0, 0.54)"};
`;
//   .dropd-drop-label {
// 	position: absolute;
// 	top: 0;
// 	left: 5px;
// 	font-size: 11px;
// 	font-weight: 400;
// 	transform: translateY(-50%);
// 	opacity: 1;
// 	z-index: 2;
// 	background-color: #fff;
// 	padding: 0 4px;
// 	color: rgba(0, 0, 0, 0.54);
//   }
export const DropMainInput = styled.input`
	left: 0;
	width: 100%;
	bottom: 0;
	opacity: 0;
	position: absolute;
	pointer-events: none;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;
//   .dropd-main-input {
// 	left: 0;
// 	width: 100%;
// 	bottom: 0;
// 	opacity: 0;
// 	position: absolute;
// 	pointer-events: none;
//   }
