// @lk/ui-library/src/Component/V1/CheckoutMobile/UpiIntentBottomsheet/UpiIntentBottomSheet.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../../StorybookHelper";
import React from "react";
import { UpiIntentBottomSheet } from "./UpiIntentBottomSheet.component";

export default{
	title: "Components/v1/CheckoutMobile/UpiIntentBottomSheet",
	component: UpiIntentBottomSheet,
	argTypes: {
		show: generateArgTypes({
			type: "boolean",
			required: false,
			description: "State that renders the component",
			category: "Component"
		}),
	}
} as ComponentMeta<typeof UpiIntentBottomSheet>;

const Template: ComponentStory<typeof UpiIntentBottomSheet> = (args) => <UpiIntentBottomSheet {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	dataLocale: {
		"MORE": "more",
		"ADD_NEW_ID": "Add new UPI ID",
		"UPI": "Upi",
		"YOUR_UPI_ID": "Your UPI ID",
		"VERIFT_AND_PAY": "Verify & Pay",
		"GOOGLE_PAY": "Google Pay",
		"MORE_UPI_APPS": "More UPI Apps",
		"INTENT_FLOW": "Intent Flow",
		"PAY_VIA_UPI_APPS": "Pay via UPI Apps",
		"PROCEED_TO_GET_INSTALLED_UPI_APPS": "Proceed to get a list of all installed UPI Apps",
		"PROCEED_TO_PAY": "Proceed to pay",
		"CHOOSE_OTHER_PAYMENT_OPTIONS": "Choose other payment options",
		"SELECT_APP_FROM_BELOW_LIST": "Please select an app from below list to make the payment.",
		"NOTE": "Note",
		"DONT_CLOSE_TILL_TRANSACTION_COMPLETE": "Do not close this screen until the transaction is complete",
		"CANCEL_PAYMENT": "Cancel payment",
		"AWAITING_PAYMENT": "Awaiting payment",
		"APPROVE_BEFORE_TIMER_RUNS_OUT": "Please approve the payment on your UPI app before the timer runs out",
		"MINS_REMAINING": "Mins remaining",
		"DONT_CLOSE_UNTIL_TRANSACTION_COMPLETE": "Do not close this screen until the transaction is complete",
		"YES_CANCEL":"Yes, cancel",
		"NO":"No",
		"PAYMENT_UNSUCCESSFUL_RETRY_WITH_OTHER_METHOD":"Your payment was unsuccessful, retry with another method.",
		"UNSUCCESSFUL_DESCRIPTION":"Any amount debited from your account, will be credited back within 2-3 working days",
		"RETRY_PAYMENT": "Retry with other payment methods",
		"SURE_YOU_WANT_TO_CANCEL": "Are you sure you want to cancel payment?",
		"CANCELLATION_TEXT": "Cancellation will be successful only if you are yet to complete this payment on your UPI app"
	},
};
