// @lk/ui-library/src/Component/V1/CheckoutMobile/UpiIntentBottomsheet/UpiIntentBottomSheet.styles.tsx
import styled from "styled-components";

export const IntentSheetContainer = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  justify-content: space-between;
  text-align: center;
`;

export const IntentSheetContent = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100%;
  width: 100%;
`;

export const IntentSheetHeader = styled.div`
  line-height: 20px;
  font-size: var(--fs-14);
  font-family: var(--font-lksans-regular);
`;

export const LoaderBlock = styled.div`
  margin-bottom: 50px;
  margin-top: 35px;

  img {
    max-width: 100px;
  }
`;

export const NoteSection = styled.div`
  padding: 0 var(--pd-16) var(--pd-70);
  font-size: var(--fs-13px);
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;

  button {
    font-family: var(--font-lksans-bold);
    font-style: normal;
    /* font-weight: 700; */
    font-size: var(--fs-14);
    line-height: 24px;
  }
`;

export const Note = styled.div`
  font-family: var(--font-lksans-regular);
  font-weight: 400;
  font-size: var(--fs-12);
  line-height: 20px;
  text-align: center;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  padding: 0 15%;
  padding-bottom: var(--pd-32);

  strong {
    font-family: 'LenskartSans-Bold';
    font-style: normal;
    font-weight: 700;
  }
`;
