// @lk/ui-library/src/Component/V1/GridConfigration/GridConfigration.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import GridConfigration from "./index";

export default {
	title: "Components/v1/GridConfigration",
	component: GridConfigration,
	argTypes: {
		...generateGeneralArgs(),
		index: generateArgTypes({
			type: "number",
			required: false,
			description: "Index of product image",
			category: "Component",
			defaultValue: 0,
		}),
		imageSrc: generateArgTypes({
			type: "string",
			required: true,
			description: "Image source",
			category: "Component",
			defaultValue: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "Function to be called on click",
			category: "Component",
			defaultValue: () => null,
		}),
	}
} as ComponentMeta<typeof GridConfigration>;

const Template: ComponentStory<typeof GridConfigration> = (args) => <GridConfigration {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {

	},
	categoryCardsType: "Eyeglass",
	category: "Eyeglassess",
	bottomSheetData: [
		{
			dtmObj: {
				PageType: "home",
				category: "eyeglasses",
				event: "generic-int-link",
				linkName: "web|home|eyeglasses",
				linkPageName: "home",
				subcategory: "eyeglasses"
			},
			gaObj: {
				action: "BOTTOM_SHEET",
				category: "Banner Clicks",
				event: "Banner Clicks",
				label: "2_Eye_60percent"
			},
			image: "https://static1.lenskart.com/media/desktop/img/Dec22/15-Dec/eye-bottom1.jpg",
			url: "/eyeglasses/offer/eyeglasses.html"
		},
		{
			dtmObj: {
				PageType: "home",
				category: "eyeglasses",
				event: "generic-int-link",
				linkName: "web|home|eyeglasses",
				linkPageName: "home",
				subcategory: "eyeglasses"
			},
			gaObj: {
				action: "BOTTOM_SHEET",
				category: "Banner Clicks",
				event: "Banner Clicks",
				label: "1_air_vincent_chase_from_rs_1600"
			},
			image: "https://static1.lenskart.com/media/desktop/img/Dec22/12dec/bottom-sheet/Eye-1.jpg",
			url: "https://www.lenskart.com/eyeglasses.html"
		},
		{
			dtmObj: {
				PageType: "home",
				category: "eyeglasses",
				event: "generic-int-link",
				linkName: "web|home|eyeglasses",
				linkPageName: "home",
				subcategory: "eyeglasses",
			},
			gaObj:{
				action: "BOTTOM_SHEET",
				category: "Banner Clicks",
				event: "Banner Clicks",
				label: "2_john_jacobs_from_rs_3000",
			},
			image: "https://static1.lenskart.com/media/desktop/img/Dec22/15-Dec/JJ-Eye.jpg",
			url: "/premium-eyeglasses",
		},
		{
			dtmObj: {
				PageType: "home",
				category: "eyeglasses",
				event:"generic-int-link",
				linkName: "web|home|eyeglasses",
				linkPageName: "home",
				subcategory: "eyeglasses"
			},
			gaObj: {
				action: "BOTTOM_SHEET",
				category: "Banner Clicks",
				event:"Banner Clicks",
				label:"2_Eye_60percent",
			},
			image: "https://static1.lenskart.com/media/desktop/img/Dec22/12dec/bottom-sheet/Eye-3.jpg",
			url: "/eyeglasses/offer/eyeglasses.html"
		}
	]

};
