// @lk/ui-library/src/Component/V1/AdvancedDropdown/Menu/styles.tsx
import styled from "styled-components";

export const MenuOuterPageContainer = styled.div`
	position: fixed;
	z-index: 1300;
	inset: 0px;
	top: 0;
	left: 0;
	width: 100vw;
	height: 100vh;
`;
export const MenuInnerBox = styled.div`
	opacity: 1;
	transform: none;
	min-width: 251px;
	transition: opacity 435ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,
	transform 290ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
	transform-origin: 0px 649.87px;
	max-height: calc(100% - 96px);
	-webkit-overflow-scrolling: touch;
	outline: 0;
	position: absolute;
	min-width: 16px;
	min-height: 16px;
	overflow-x: hidden;
	overflow-y: auto;
	box-shadow: 0px 5px 5px -3px rgba(0, 0, 0, 0.2), 0px 8px 10px 1px rgba(0, 0, 0, 0.14),
	0px 3px 14px 2px rgba(0, 0, 0, 0.12);
	border-radius: 4px;
	color: rgba(0, 0, 0, 0.87);
	transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
	background-color: #fff;
`;

export const MenuFullpageCoverage = styled.div`
	-webkit-tap-highlight-color: transparent;
`;

export const MenuMenuList = styled.div`
	padding-right: 0px;
	outline: 0;
	padding-top: 8px;
	padding-bottom: 8px;
	margin: 0;
	padding: 0;
	position: relative;
	list-style: none;
	display: block;
	list-style-type: disc;
	color: rgba(0, 0, 0, 0.87);
`;
