// @lk/ui-library/src/Component/V1/CheckoutMobile/MorePaymentOptions/Input/Input.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { FormInput } from "./Input.component";
import React from "react";
import { inputs, typeENUM } from "./Input.types";
import { generateArgTypes, generateGeneralArgs } from "../../../../../StorybookHelper";
import { ComponentSizeENUM, TypographyENUM } from "../../../../../Types/general";

export default {
	title: "Components/Common/CheckoutMobile/Input",
	component: FormInput,
	argTypes: {
		...generateGeneralArgs(), 
		value: generateArgTypes({
			type: "string",
			required: false,
			description: "Value that will be rendered",
			category: "Component",
			defaultValue: "Lorem Ipsum",
		}),
		placeholder: generateArgTypes({
			type: "string",
			required: false,
			description: "Placeholder that will be rendered",
			category: "Component",
			defaultValue: "Lorem Ipsum placeholder",
		}),
		label: generateArgTypes({
			type: "string",
			required: false,
			description: "Label that will be rendered",
			category: "Component",
			defaultValue: "",
		}),
		type: generateArgTypes({
			type: "string",
			required: true,
			description: "Type of input will be rendered",
			category: "Component",
			defaultValue: typeENUM.text,
			control: "radio",
			options: typeENUM,
		}),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		}),
		onChange: generateArgTypes({
			type: "function",
			required: false,
			description: "onChange handler for input",
			category: "Events",
		}),
	},
} as ComponentMeta<typeof FormInput>;

const Template: ComponentStory<typeof FormInput> = (args) => <FormInput {...args}/>;

export const inputWithoutError = Template.bind({});
inputWithoutError.args = {
	label:"First Name",
	componentSize: ComponentSizeENUM.medium,
	disabled: false,
	readonly: false,
	width: "50%",
	type: inputs.type.text,
	font:TypographyENUM.defaultBook
};
