// @lk/ui-library/src/Component/V1/CheckoutMobile/MorePaymentOptions/Input/Input.styles.tsx
import styled from "styled-components";
import { ComponentSizeENUM , TypographyENUM } from "../../../../../Types/general";
import { fontSize } from "./helper";


export const InputField= styled.input`
	background:none;
	border: 1px solid #CECEDF;
	padding: 0px 55px;
	width: 100%;
	display: block;
	outline: none;
	height: 56px;
	background: #FFFFFF;
	border-radius: 12px;
	&::placeholder{
		font-weight: 400;
		font-size: 14px;
		line-height: 24px;
		letter-spacing: -0.02em;
		color: #66668E;
		font-family: ${TypographyENUM.lkSansRegular};
	}

	font-weight: 400;
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #66668E;
  user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const CrossIcon = styled.span`
	position:absolute;
	right:18px;
	top:18px;
`;


export const InputContainer = styled.div<{width:string, value:string | undefined, styleSize?: ComponentSizeENUM, styleFont?: TypographyENUM}>`
	position: relative;
	display: flex;
	height:max-content;
	flex-direction: column;
	flex-grow:1;
	width: ${(props) => props.width};
`;

export const InputWrapper = styled.div<{ styleSize:ComponentSizeENUM, styleFont:TypographyENUM}>`
	font-size: var(${(props) => fontSize[props.styleSize]});
	font-family:${props => props.styleFont};
	width: 100%;
`;
