// @lk/ui-library/src/Component/V1/CheckoutMobile/MorePaymentOptions/Input/Input.types.tsx
import React from "react";
import { ComponentSizeENUM, TypographyENUM } from "../../../../../Types/general";
export interface InputType {
	setClearInput?:(val:boolean) => void;
	placeholder?: string
	type: string
  getInputValue: (val:string) => void
	componentSize?: ComponentSizeENUM
	width: string
	id?: string
	clearInput?:boolean
	disabled?: boolean
	readonly?: boolean
	hasError?: boolean
	errorMessage?:string;
	label?:string
	font?:TypographyENUM
	onValueChange?: (value: React.ChangeEvent<HTMLInputElement>) => void
	includeButton?:boolean
	buttonText?:string 
	buttonPosition?:positionENUM
	labelPosition?:positionENUM
	buttonOnClick?:() => void
	// will add functionalities for below parameters later
	// includeIcon?:boolean
	// iconText?:string
	// iconPosition?:positionENUM
	// iconOnClick?:() => void
	countryCodePrefix?:string,
	onKeyUpChange?: (value: React.ChangeEvent<HTMLInputElement>) => void
	dir?: string
	labelFontSize?: string
	// suffix?:string
}

export enum positionENUM{
	right="right",
	left="left",
	// top="top",
}

export enum typeENUM {
	text = "text",
	password = "password",
	email = "email",
	number = "number",
	dir = "auto"
}

export const inputs = {
	type: typeENUM,
};
