// @lk/ui-library/src/Component/Common/Carousel/Carousel.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import Carousel from "./index";

export default {
	title: "Components/Common/Carousel",
	component: Carousel,
	argTypes: {
		initialSlide: generateArgTypes({
			type: "number",
			category: "Component",
			required: false,
			description: "Initial slide index",
			defaultValue: 0,
		}),
		responsive: generateArgTypes({
			type: "array",
			category: "Component",
			required: false,
			description: "Responsive settings",
			defaultValue: ""
		}),
		slidesToShow: generateArgTypes({
			type: "number",
			category: "Component",
			required: false,
			description: "Number of slides to show",
			defaultValue: 7,
		}),
		children: generateArgTypes({
			type: "node",
			category: "Component",
			required: true,
			description: "Slides",
			defaultValue: ""
		}),
	}
} as ComponentMeta<typeof Carousel>;

const Template: ComponentStory<typeof Carousel> = (args) => <Carousel {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	initialSlide: 0,
	responsive: [
		{
			"breakpoint": 1500,
			"settings": {
				"slidesToShow": 6
			}
		},
		{
			"breakpoint": 1024,
			"settings": {
				"slidesToShow": 5
			}
		},
		{
			"breakpoint": 600,
			"settings": {
				"slidesToShow": 5
			}
		},
		{
			"breakpoint": 480,
			"settings": {
				"slidesToShow": 3
			}
		}
	],
	slidesToShow: 7,
	wrapperClassName: "slider-custom",
	children: "",
};
