// @lk/ui-library/src/Component/Common/Carousel/styles.tsx
import styled, { css } from "styled-components";

export const SliderWrapper = styled.div<{ mobileView?: boolean, isRTL?: boolean}>`
  &.custom-zoom-image {
	margin-bottom:30px;
    .slick-next {
      width: 80px;
      height: 80px;
      background-size: contain !important;
      background-color: transparent !important;
      text-indent: -9999px;
      ${props => !props.isRTL && css`
      background-image: url("https://static.lenskart.com/media/desktop/img/menu/icon-arrow-right.svg");
      `}

      ${props => props.isRTL && css`
      background-image: url("https://static.lenskart.com/media/desktop/img/menu/icon-arrow-left.svg");
      `}

      &:hover {
        width: 80px;
        height: 80px;
        background-size: contain !important;
        background-color: transparent !important;
        text-indent: -9999px;
        ${props => !props.isRTL && css`
      background-image: url("https://static.lenskart.com/media/desktop/img/menu/icon-arrow-right.svg");
      `}

      ${props => props.isRTL && css`
      background-image: url("https://static.lenskart.com/media/desktop/img/menu/icon-arrow-left.svg");
      `}
      }

      ${props => props.isRTL && css`
      margin-left: -16px;
    `};

    ${props => !props.isRTL && css`
      margin-right: -16px;
    `};
    }
    .slick-prev {
      width: 80px;
      height: 80px;
      background-size: contain !important;
      background-color: transparent !important;
      text-indent: -9999px;

      ${props => !props.isRTL && css`
      background-image: url("https://static.lenskart.com/media/desktop/img/menu/icon-arrow-left.svg");
      `}

      ${props => props.isRTL && css`
      background-image: url("https://static.lenskart.com/media/desktop/img/menu/icon-arrow-right.svg");
      `}

      /* left: -32px; */
      &:hover {
        width: 80px;
        height: 80px;
        background-size: contain !important;
        background-color: transparent !important;
        text-indent: -9999px;
        ${props => !props.isRTL && css`
      background-image: url("https://static.lenskart.com/media/desktop/img/menu/icon-arrow-left.svg");
      `}

      ${props => props.isRTL && css`
      background-image: url("https://static.lenskart.com/media/desktop/img/menu/icon-arrow-right.svg");
      `}
        /* left: -32px; */
      }

      ${props => props.isRTL && css`
        margin-right: -16px;
      `};

      ${props => !props.isRTL && css`
        margin-left: -16px;
      `};

    }
    .slick-slide .sliderImg {
      margin: 0 auto;
    }
    &_desc {
      max-width: 1000px;
      text-align: center;
      margin: 20px auto;
    }
  }

  &.slider-custom {
    .slick-next::before, .slick-prev::before {
      content: '';
    }
    .slick-prev,
    .slick-next {
      height: 50px;
      width: 50px;
      border-radius: 50%;
      background: #ffffff !important;
	  box-shadow: ${props => props.mobileView ? "" : "0px 4px 4px rgba(0, 0, 0, 0.05)"};
      padding: 20px;
      margin-bottom: 20px;
      background-repeat: no-repeat !important;
      background-position: center !important;
      stroke: rgb(102, 102, 102);

      &:hover {
        height: 50px;
        width: 50px;
      }
    }
    .slick-prev {
      background-size: contain;
    }
    .slick-next {
      background-size: contain;
    }
  };
   &.slick-active button:before {
      opacity: 1;
      color: #289f95;
    }

  .slider-main.carousel-comp-slider {
    img {
      width: 100%;
    }
  }

  &.home-page-slider {
    .slick-next:before, .slick-prev:before {
		display: block;
		color: var(--white);
		font-size:var(--fs-40);
		position:absolute;
		bottom:0;
	}
    .slick-prev:before {
		content: '❮';
		left: var(--spacing-xl);
	}
    .slick-next:before {
		content: '❯';
		right: var(--spacing-xl);
	}

    .slick-prev,
    .slick-next {
		&:hover::before{
			transform:scale(1.1);
		}
    }
  };
  .slick-dots {
    position: absolute;
    bottom: 0px;
    display: block;
    width: 100%;
    padding: 0;
    list-style: none;
    text-align: center;
    li {
      position: relative;
      display: inline-block;
      width: 20px;
      height: 20px;
      margin: 0 5px;
      padding: 0;
      cursor: pointer;
      button {
        font-size: 0;
        line-height: 0;
        display: block;
        width: 20px;
        height: 20px;
        padding: 5px;
        cursor: pointer;
        color: transparent;
        border: 0;
        outline: none;
        background: transparent;
        &:hover {
          outline: none;
        }
        &:hover:before {
          opacity: 1;
          color: #289f95;
        }
        &:before {
          font-family: "slick";
          font-size: 50px;
          line-height: 20px;
          position: absolute;
          top: 0;
          left: 0;
          width: 20px;
          height: 20px;
          content: "•";
          text-align: center;
          opacity: 0.25;
          color: black;
          -moz-osx-font-smoothing: grayscale;
        }
      }
      &.slick-active button:before {
        opacity: 1;
        color: #289f95;
      }
    }
  };
`;

export const SlideWrapper = styled.div<{ isCurrentSlide?: boolean }>`
  ${({ isCurrentSlide }) => isCurrentSlide && `
      // css when slide is selected
    `}

  &.custom-slide {
    img {
      cursor: var(--pointer);
    }
  }
`;

export const Input = styled.input`
  color: #000042 !important;
  accent-color: #000042;
  margin-top: 10px;
  width: 75px;
  height: 0px;
  border: 2px solid #000042;
  position: absolute;
  left: 50%;
  transform: translate(-50%, -50%);
`;
