// @lk/ui-library/src/Component/V1/PDP/ImageModal/ImageModal.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";
import { SliderWrapper } from "../../../Common/Carousel/styles";

export const ImageModalWrapper = styled.div`
  ${SliderWrapper} {
    padding-left: var(--pd-40);
    padding-right: var(--pd-40);

    .slick-next {
      margin-right: -20px;
    }

    .slick-prev {
      margin-left: -20px;
    }
  }
`;

export const ProductContainer = styled.div<{styleFont:TypographyENUM, isSelectedImage:boolean}>`
  text-align: center;
  padding: var(--pd-10);
  margin: 0 var(--spacing-md);
  overflow: hidden;
  background: var(--white);
  font-family:${props => props.styleFont};
  border: ${ props => props.isSelectedImage ? "2px solid #2ba2c9" : "2px solid black"};
  border-radius: 10%;
  cursor: pointer;
`;

export const CarouselWrapper = styled.div`
margin-top: var(--spacing-md);
`;

export const ImgContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  &#zoom-image-container{
	img{
		height:auto !important;
	}
  }
`;

export const ProductImage = styled.img`
  width: 100%;
`;

export const Image = styled.img`{
    width: 45%;
}`;
