// @lk/ui-library/src/Component/V1/NewPayment/components/CardImage.tsx
// import { PlusIcon } from "../../../../Icons";
import React from "react";
import styled from "styled-components";

const ImageContainer = styled.div<{isSelected: boolean; leftMargin: boolean; rightMargin: boolean;}>`
    width: 46px;
    height: 32px;
    background: ${props => props.isSelected ? "#F5F5FF" : "#FFFFFF"};
    border: 1px solid #E2E2EE;
    border-radius: 4px;
    display: flex;
    align-items: center;
    justify-content: center;
    margin-top: 0;
    margin-bottom: 0;
    ${props => props.leftMargin && `
        margin-left: 16px;
    `}
    // ${props => props.rightMargin && `
    //     margin-right: 16px;
    // `}
`;
const ImageDisplay = styled.img`
  width: 46px;
  height: 32px;
  object-fit: contain;
  display: block;
  margin: 0;
  padding: 0;
`;
interface CardImageType {
	src: string,
	alt: string,
	isSelected: boolean,
	leftMargin?: boolean,
	rightMargin?: boolean
}
const CardImage = ({ src = "", alt="", isSelected = false, leftMargin = true, rightMargin = true }: CardImageType) => {
	return (
		<ImageContainer className="netCartImg" isSelected={isSelected} leftMargin={leftMargin} rightMargin={rightMargin}>
			{/* {isSelected ? <PlusIcon /> :  */}
			<ImageDisplay src={src} alt={alt} />
		</ImageContainer>
	);
};

export { CardImage };
