// @lk/ui-library/src/Component/V1/NewPayment/components/PaymentAddress.tsx
import React from "react";
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

const PaymentAddressWrapper = styled.div<{width?:string, fullWidth:boolean}>`
	box-sizing: border-box;
	display: flex;
	width: ${props => props.width ? "props.width" : "328px"};
	width: ${props => props.fullWidth ? "100%" : ""};
	/* height: 80px; */
	border-radius: 10px;
	padding: 16px;
	gap: 12px;
	font-family: ${TypographyENUM.lkSansRegular};
	font-size: 12px;
	justify-content: space-between;
	align-items: center;
	box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
	line-height: 20px;
	background-color: white;
`;

const Left = styled.div<{isRTL: boolean}>`
	${props => props.isRTL ? "direction: initial" : ""};
	width: 70%;
`;

const Head = styled.div<{isRTL: boolean}>`
	display: flex;
	color:#333368;
	gap: 5px;
	${props => props.isRTL ? "justify-content: end" : ""};
`;

const HeadInfo = styled.div`

`;

const HeadText = styled.div`
	font-family: ${TypographyENUM.lkSansBold};
`;

const Address = styled.div<{isRTL: boolean}>`
	color: #66668E;
	width: 100%;
	height: 20px;
	white-space: nowrap;
  	overflow: hidden;
  	text-overflow: ellipsis;
	flex-grow: 1;
	${props => props.isRTL ? "display:flex;justify-content: end;" : ""};
`;

const Right = styled.div`
`;

const AddressButton = styled.button`
	font-family: ${TypographyENUM.lkSansBold};
	color: #000042;
	background-color: transparent;
	border: none;
	text-decoration: underline;
	cursor: pointer;
`;

interface PaymentAddressType {
	headInfo: string;
	headText: string;
	btnText: string;
	onBtnClick: () => void;
	mainText: string;
	font: TypographyENUM;
	fullWidth: boolean;
	width?: string;
	isRTL?: boolean;
}

const PaymentAddress = ({ headInfo="Deliver to: ", headText="John Doe", btnText="Change", onBtnClick, mainText="A201, Gina Shalom, Old Madras Road, Battara Junction, Krishnarajapuram", width, fullWidth=false, isRTL = false }:PaymentAddressType) => {

	return (
		<PaymentAddressWrapper width={width} fullWidth={fullWidth}>
			<Left isRTL={isRTL}>
				<Head isRTL={isRTL}>
					<HeadInfo>
						{headInfo}
					</HeadInfo>
					<HeadText>
						{headText}
					</HeadText>
				</Head>
				<Address isRTL={isRTL}>
					{mainText}
				</Address>
			</Left>

			<Right>
				<AddressButton onClick={onBtnClick}>
					{btnText}
				</AddressButton>
			</Right>
		</PaymentAddressWrapper>
	);
};

export { PaymentAddress };
