// @lk/ui-library/src/Component/Common/Swatch/Swatch.types.ts
import { ComponentSizeENUM } from "../../../Types/general";
export interface SwatchTypes {
	id: string
	colors: ColorOptionType[]
	onmouseOver?: () => void
	updateImageURL?: (newURL?:string) => void
	productImageURL?:string
	componentSize?: ComponentSizeENUM
	selectedIndex?:number|undefined
	selectedIndexColor:string
	hoverColor?:string
	hoverRotate:boolean
	isRTL: boolean
	device?: string
	setproductUrlMobile?: (props: string) => void
	triggerCTAGA: ()=>void
}

export interface imageHoverType {
	url: string,
	hoverURL: string
}

	
export interface ColorOptionType {
	id: number,
	color: string,
	secondaryColor?: string,
	glassColor?: string,
	colorID: number,
	productURL: string,
	price: PriceType,
	arModel: ARModelType,
	productImage: imageHoverType,
	frameColorImage: string
}

export interface ARModelType {
	android: string | null,
	ios: string | null
}


export enum CurrencySymbol {
	INR="₹",
	SGD="$",
	AED = "AED"
}

export enum CurrencyType {
	INR="INR",
	SGD="SGD",
	AED="AED"
}

export interface PriceType {
	symbol: CurrencySymbol,
	currency: CurrencyType,
	basePrice: number,
	lkPrice: number
}
