// @lk/ui-library/src/Component/Common/Swatch/Swatch.stories.tsx
/* eslint-disable @typescript-eslint/no-empty-function */
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../StorybookHelper";
import Swatch from "./Swatch.component";
import { ComponentSizeENUM, CurrencySymbol, CurrencyType } from "../../../Types/general";

export default {
	title: "Components/Common/Swatch",
	component: Swatch,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		componentSize: generateArgTypes({
			type: "string",
			required: false,
			description: "Size of the component",
			category: "General",
			defaultValue: ComponentSizeENUM.small,
			control: "radio",
			options: ComponentSizeENUM,
		}),
		colors: generateArgTypes({
			type: "array",
			required: true,
			description: "array of values to set color, secondaryColor, URL, etc...",
			category: "Component",
		}),
		onmouseOver: generateArgTypes({
			type: "function",
			required: false,
			description: "on Mouse hover handler for swatch",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show right to left",
			category: "Status",
		}),
	},
} as ComponentMeta<typeof Swatch>;

const Template: ComponentStory<typeof Swatch> = (args) => <Swatch {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "swatch",
	colors: [{
		"id": 146284,
		"color": "#000000",
		"secondaryColor": "red",
		"colorID": 11378,
		"productURL": "https://www.lenskart.com/lenskart-air-lenskart-air-la-e11821-wayfarer-tr90-c2-eyeglasses.html",
		"price": {
			"currency": CurrencyType.INR,
			"basePrice": 2000,
			"lkPrice": 999,
			"symbol": CurrencySymbol.INR
		},
		"arModel": {
			"ios": "https://3d-asset-lookr.s3-ap-southeast-1.amazonaws.com/frameassets/146284/ios/146284.zip",
			"android": "https://3d-asset-lookr.s3-ap-southeast-1.amazonaws.com/frameassets/146284/android/sku_146284.glb"
		},
		"productImage": {
			"url": "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_g_9919_2.jpg",
			"hoverURL": "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_g_9919_2.jpg"
		}
	}],

};

export const MultiColorsSwatches = Template.bind({});
MultiColorsSwatches.args = {
	id: "swatch",
	colors: [
		{
			"id": 146284,
			"color": "#000000",
			"secondaryColor": "red",
			"colorID": 11378,
			"productURL": "https://www.lenskart.com/lenskart-air-lenskart-air-la-e11821-wayfarer-tr90-c2-eyeglasses.html",
			"price": {
				"currency": CurrencyType.INR,
				"basePrice": 2000,
				"lkPrice": 999,
				"symbol": CurrencySymbol.INR
			},
			"arModel": {
				"ios": "https://3d-asset-lookr.s3-ap-southeast-1.amazonaws.com/frameassets/146284/ios/146284.zip",
				"android": "https://3d-asset-lookr.s3-ap-southeast-1.amazonaws.com/frameassets/146284/android/sku_146284.glb"
			},
			"productImage": {
				"url": "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_g_9919_2.jpg",
				"hoverURL": "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_lenskart-air-lnewa-e11821-wayfarer-tr90-c2-eyeglnewasses_g_9919_2.jpg"
			}
		},
		{
			"id": 150550,
			"color": "#808080",
			"colorID": 13925,
			"productURL": "https://www.lenskart.com/lenskart-air-la-e11821-c4-eyeglasses.html",
			"price": {
				"currency": CurrencyType.INR,
				"basePrice": 2000,
				"lkPrice": 999,
				"symbol": CurrencySymbol.INR
			},
			"arModel": {
				"ios": null,
				"android": null
			},
			"productImage": {
				"url": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-c4-eyeglasses_G_0160.jpg",
				"hoverURL": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-c4-eyeglasses_G_0160.jpg"
			}
		},
		{
			"id": 150551,
			"color": "#ed7e07",
			"glassColor": "#0000ff",
			"colorID": 11386,
			"productURL": "https://www.lenskart.com/lenskart-air-la-e11821-c5-eyeglasses.html",
			"price": {
				"currency": CurrencyType.INR,
				"basePrice": 2000,
				"lkPrice": 999,
				"symbol": CurrencySymbol.INR
			},
			"arModel": {
				"ios": null,
				"android": null
			},
			"productImage": {
				"url": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-c5-eyeglasses_G_0166.jpg",
				"hoverURL": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-c5-eyeglasses_G_0166.jpg"
			}
		},
		{
			"id": 150552,
			"color": "#0000ff",
			"glassColor": "#800080",
			"colorID": 11375,
			"productURL": "https://www.lenskart.com/lenskart-air-la-e11821-c6-eyeglasses.html",
			"price": {
				"currency": CurrencyType.INR,
				"basePrice": 2000,
				"lkPrice": 999,
				"symbol": CurrencySymbol.INR
			},
			"arModel": {
				"ios": null,
				"android": null
			},
			"productImage": {
				"url": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/i/Blue-Silver-Full-Rim-Wayfarer-Lenskart-Air-Signia-LA-E11821-C6-Eyeglasses_lenskart-air-la-e11821-c6-eyeglasses_G_862707_02_2022.jpg",
				"hoverURL": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/i/Blue-Silver-Full-Rim-Wayfarer-Lenskart-Air-Signia-LA-E11821-C6-Eyeglasses_lenskart-air-la-e11821-c6-eyeglasses_G_862707_02_2022.jpg"
			}
		},
		{
			"id": 145997,
			"color": "#800080",
			"glassColor": "#808080",
			"colorID": 13948,
			"productURL": "https://www.lenskart.com/lenskart-air-la-e11821-full-rim-wayfarer-stainless-steel-c2-eyeglasses.html",
			"price": {
				"currency": CurrencyType.INR,
				"basePrice": 2000,
				"lkPrice": 999,
				"symbol": CurrencySymbol.INR
			},
			"arModel": {
				"ios": null,
				"android": null
			},
			"productImage": {
				"url": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-fall-rim-wfre-sts-steel-c2-eyeglasses_g_1068.jpg",
				"hoverURL": "https://static1.lenskart.com/media/catalog/product/cache/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//l/e/lenskart-air-la-e11821-fall-rim-wfre-sts-steel-c2-eyeglasses_g_1068.jpg"
			}
		}
	],
	componentSize: ComponentSizeENUM.large,
	selectedIndexColor:"var(--turquoise)"
};
