// @lk/ui-library/src/Component/V1/PDP/AboutEyePowerBox/AboutEyePowerBox.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import { AboutEyePowerBox } from "./AboutEyePowerBox.component";

export default {
	title: "Components/v1/AboutEyePowerBox",
	component: AboutEyePowerBox,
	argTypes: {
		...generateGeneralArgs(),
		infoSections: generateArgTypes({
			type: "array",
			required: true,
			description: "Coating Image",
			category: "Component",
			defaultValue: [{ image: "https://static.lenskart.com/media/desktop/img/pdp/right.png", text: "Power can be submitted within 10 days of order placement" }],
		}),
		title: generateArgTypes({
			type: "string",
			required: true,
			description: "Title",
			category: "Component",
			defaultValue: "What about my eye power?",
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "dataLocale",
			category: "Component",
			defaultValue: { SUBMIT_EYE_POWER_TEXT: "Submit eye power", ENTER_MANUALLY: "Enter Manually", PHONE_ASSISTANCE: "Phone assistance", UPLOAD_PRESCRIPTION: "Upload prescription", HOME_EYE_CHECKUP: "Home eye checkup", I_HAVE_READ_AND_UNDERSTOOD_THE_TERMS_OF_POWER_SUBMISSION: "I have read and understood the terms of power submission" },
		}),
	},
} as ComponentMeta<typeof AboutEyePowerBox>;

const Template: ComponentStory<typeof AboutEyePowerBox> = (args) => <AboutEyePowerBox {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	infoSections: ["Power can be submitted within 10 days of order placement"],
	dataLocale: { SUBMIT_EYE_POWER_TEXT: "Submit eye power", ENTER_MANUALLY: "Enter Manually", PHONE_ASSISTANCE: "Phone assistance", UPLOAD_PRESCRIPTION: "Upload prescription", HOME_EYE_CHECKUP: "Home eye checkup", I_HAVE_READ_AND_UNDERSTOOD_THE_TERMS_OF_POWER_SUBMISSION: "I have read and understood the terms of power submission" },
};
