// @lk/ui-library/src/Component/V1/PDP/AboutEyePowerBox/AboutEyePowerbox.styles.tsx
import styled from "styled-components";

export const EyeBoxContainer = styled.div`

`;
export const MainSection = styled.div`
    width: 580px;
	border: 1px solid #e5e8e9;
	border-radius: 4px;
	margin: 0 auto;
	padding: 15px;
	font-family:  FuturaStd-Book, helvetica neue, Helvetica, Arial, sans-serif;
	font-synthesis: none;
    font-size: 13px;
    line-height: 1.42857;
    color: #333;
`;
export const Title = styled.div`
	font-weight: 700;
	text-align: center;
	padding-bottom: 15px;
	font-size: 22px;
	font-family: var(--font-default-bold);
`;
export const Flex = styled.div`
	display: flex;
	margin: 0 auto;
	width: 80%;
	justify-content: space-between;
`;
export const TipFlex = styled.div`
	display: flex;
	margin-bottom: 10px;
	text-align: center;
	align-items: center;
	font-family: var(--font-default-book);
	font-synthesis: none;
`;
export const HighlightsSection = styled.div`
	width: 100%;
	padding: 15px 0;
`;
export const BottomSection = styled.div`
	border-top: 1px solid #ccc;
	padding: 15px 0;
`;
export const InfoSection = styled(Flex)`
	align-items: center;
	justify-content: center;
	margin-bottom: 25px;
	margin-top: 15px;
    font-style: normal;
    font-weight: 700;
    font-size: 16px;
    line-height: 20px;
    color: #00bac6;
	font-family: var(--font-default-book);
	font-synthesis: none;
`;
export const CheckBox = styled.img<{isRTL: boolean}>`
	${props => props.isRTL ? "padding-left: 10px" : "padding-right: 10px"};
	max-width: 100%;
	display: block;
`;
