// @lk/ui-library/src/Component/NewHtoFlow/ProductNamePrice/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const ProductNamePriceWrapper = styled.div`
    display: flex;
    flex-direction: column;
    gap: 8px;
`;

export const ProductNameWrapper = styled.div`
    color: var(--dark-blue-100);
    font-style: normal;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    width: 100%;
`;

export const Heading = styled.div`
    font-weight: 800;
    font-size: var(--fs-18);
    line-height: 24px;
    font-family: ${TypographyENUM.lkSansBold};
    width: 75%;
`;

export const SubHeading = styled.div`
    font-weight:  var(--fw-medium);
    font-size: var(--fs-12);
    line-height: 16px;
    font-family: ${TypographyENUM.lkSansMedium}
`;

export const ProductPriceWrapper = styled.div`
    display: flex;
    flex-direction: row;
    // width: 25%;
`;

export const ProductActualPrice = styled.div`
    font-style: normal;
    font-weight: var(--fw-medium);
    font-size: var(--fs-14);
    line-height: 20px;
    text-decoration-line: line-through;
    color: #66668E;
    font-family: ${TypographyENUM.lkSansMedium};
    white-space: nowrap;
`;

export const ProductSellingPrice = styled.div`
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-16);
    line-height: 20px;
    font-family: ${TypographyENUM.lkSansBold};
    margin-left: 10px;
    white-space: nowrap;
`;

export const ProductDescription = styled.div`
`;
