// @lk/ui-library/src/Component/V1/MyAccount/MyOrder/PricebreakUp/PriceBreakUp.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import PriceBreakUpOD from ".";

export default {
	title: "Components/V1/MyAccount/PricebreakUpOD",
	component: PriceBreakUpOD
} as ComponentMeta<typeof PriceBreakUpOD>;

const Template: ComponentStory<typeof PriceBreakUpOD> = (args) => <PriceBreakUpOD {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		CONVENIENCE_FEES: "Convenience Fees",
		NET_AMOUNT: "Net Amount",
		STORE_CREDIT: "Store Credit",
		TOTAL: "Total",
		OLD_PRODUCT_EXCHANGE_VALUE: "Old Product Exchange Value",
		GST_COLLECTED: "GST Collected",
		PREPAID_DISCOUNT: "Prepaid Discount",
		PRICE_BREAKUP: "Price Breakup",
		VOUCHER_DISCOUNT: "Voucher Discount",
		SUBTOTAL: "Subtotal",
		LK_CASH: "LK Cash",
		LK_CASH_PLUS: "LK Cash+",
		COUPON_DISCOUNT: "Coupon Discount",
		AUTO_DISCOUNT: "Auto Discount",
		CURRENCY_SYMBOL: "₹",
	},
	amount: {
		currencyCode: "INR",
		discounts: [
			{
				type: "implicit",
				amount: 9200
			},
			{
				code: "FORYOU",
				type: "gv",
				amount: 100
			}
		],
		totalDiscount: 9300,
		taxes: [
			{
				name: "VAT",
				amount: 1092
			}
		],
		totalTax: 1092,
		shipping: 0,
		subTotal: 18400,
		total: 10192
	},
	mobileView: true,
};
