// @lk/ui-library/src/Component/V1/MyAccount/MyOrder/PricebreakUp/styles.tsx
import styled from "styled-components";

export const PriceBreakUpContainer = styled.div<{ mobileView?: boolean }>`
	font-family: ${props => props.mobileView ? "var(--font-lksans-regular)" : "FuturaStd-Book, helvetica neue, Helvetica, Arial, sans-serif"};
	font-synthesis: none;
	line-height: ${props => props.mobileView ? "1.2875" : ""};
	${props => props.mobileView ? `padding-bottom: 70px;
    padding-top: 20px;
    padding-left: 15px;
    padding-right: 15px;
    background-color: #fafbfc;` : ""}
`;

export const Span = styled.span<{ mobileView?: boolean }>`
	color: ${props => props.mobileView ? "#333333" : "#999999"};
	font-family: ${props => props.mobileView ? "var(--font-lksans-regular)" : "FuturaStd-Book, helvetica neue, Helvetica, Arial, sans-serif"};
`;

export const PriceBreakUpText = styled.div<{ mobileView?: boolean }>`
	${props => props.mobileView ? "font-size: 16px; color: #333333; font-family: var(--font-lksans-bold);" : "font-size: 14px; font-family: var(--font-default-heavy); color: #333;"}
	margin-bottom: 20px;
`;

export const SubTotalText = styled.div<{ mobileView?: boolean }>`
	color: ${props => props.mobileView ? "#333333" : "#999999"};
	font-size: 14px;
`;

export const Price = styled.span<{ mobileView?: boolean, isRTL?: boolean }>`
	color: ${props => props.mobileView ? "#333333" : "#999999"};
	font-size: 14px;
  float: ${props => props.isRTL ? "left" : "right"};
`;

export const DiscountConatiner = styled.div<{ mobileView?: boolean }>`
	color: ${props => props.mobileView ? "#333333" : "#999999"};
	font-size: 14px;
	margin-top: 4px;
`;

export const TaxWrapper = styled.div`
	border-top: 1px solid #eeeeee;
	border-bottom: 1px solid #eeeeee;
	margin: 12px 0px 10px 0px;
	padding: 12px 0px;
`;

export const NetAmount = styled.div<{ mobileView?: boolean }>`
	font-family: ${props => props.mobileView ? "var(--font-lksans-bold)" : "FuturaStd-Book, helvetica neue, Helvetica, Arial, sans-serif"};
	color: ${props => props.mobileView ? "#333333" : "#333"};
	font-size: 14px;
	margin-bottom: 10px;
	font-synthesis: none;
`;

export const NetAmountSpan = styled.span`
	font-weight: 700;
`;

export const NetAmountPriceSpan = styled.span<{isRTL?: boolean}>`
	float: ${props => props.isRTL ? "left" : "right"};
	font-weight: 700;
`;

export const GSTCollected = styled.div<{ mobileView?: boolean }>`
	color: ${props => props.mobileView ? "#666666" : "#999999"};
	font-size: 12px;
	margin-bottom: 4px;
	${Span} {
		color: ${props => props.mobileView ? "#666666" : "" };
	}
	${Price} {
		font-size: 12px;
		color: ${props => props.mobileView ? "#666666" : "" };
	}
`;

export const PrePaidDiscount = styled.div<{ mobileView?: boolean }>`
	color: ${props => props.mobileView ? "#333333" : "#999999"};
	font-size: 12px;
	margin-bottom: 4px;
`;

export const OldorderDiscount = styled.div<{ mobileView?: boolean }>`
	color: ${props => props.mobileView ? "#333333" : "#999999"};
	font-size: 12px;
	margin-bottom: 4px;
`;

export const StoreCreditDiscount = styled.div<{ mobileView?: boolean }>`
	color: ${props => props.mobileView ? "#333333" : "#999999"};
	font-size: 12px;
	margin-bottom: 4px;
`;

export const ConvinceFess = styled.div<{ mobileView?: boolean }>`
	color: ${props => props.mobileView ? "#666666" : "#999999"};
	font-size: 12px;
	margin-bottom: 4px;
	width: 100%;
	${Span} {
		color: ${props => props.mobileView ? "#666666" : "" };
	}
	${Price} {
		font-size: 12px;
		color: ${props => props.mobileView ? "#666666" : "" };
	}
`;

export const TotalAmount = styled.div<{ mobileView?: boolean }>`
	font-family: ${props => props.mobileView ? "var(--font-lksans-bold)" : "FuturaStd-Book, helvetica neue, Helvetica, Arial, sans-serif"};
	color: ${props => props.mobileView ? "#333333" : "#333"};
	font-size: 14px;
	margin-bottom: 4px;
	font-synthesis: none;
`;

export const TotalSpan = styled.span`
	font-weight: 700;
`;

export const TotalPriceSpan = styled.span<{ mobileView?: boolean, isRTL?: boolean }>`
	color: ${props => props.mobileView ? "#333333" : "#333"};
	font-size: 14px;
	font-weight: 700;
  float: ${props => props.isRTL ? "left" : "right"};
`;
