// @lk/ui-library/src/Component/V1/ToolTip/styles.tsx
import styled, { css } from "styled-components";

export const TooltipWrapper = styled.div<{direction: string}>`
	position: fixed;
    z-index: 10000;
    display: block;
    font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
    font-style: normal;
    font-weight: var(--fw-regular);
    line-height: 1.42857143;
    line-break: auto;
    text-align: left;
    text-align: start;
    text-decoration: none;
    text-shadow: none;
    text-transform: none;
    letter-spacing: normal;
    word-break: normal;
    word-spacing: normal;
    word-wrap: normal;
    white-space: normal;
    font-size: 12px;
    filter: alpha(opacity=0);
    // opacity: 0;

    // opacity: 0;
    // -webkit-transition: opacity 0.15s linear;
    // -o-transition: opacity 0.15s linear;
    // transition: opacity 0.15s linear;
  .tooltip-inner {
    background: #fff;
    /* width: 360px; */
    max-width: 360px;
    /* height: 180px; */
    vertical-align: middle;
    border: 1px solid #414241;
    border-radius: 6px;
    padding: 0px;
    box-sizing: border-box;
    box-shadow: 12px 12px 24px 0 rgba(0, 0, 0, 0.12);
    display: flex;
    justify-content: center;
    align-items: center;
    img {
      max-height: 100%;
    }
  }

  .tooltip-inner {
	max-width: 200px;
	padding: 3px 8px;
	color: #fff;
	text-align: center;
	background-color: #000; 
	border-radius: 4px;
  }

    .tooltip-arrow {
		width: 0;
		height: 0;
		${props => props.direction === "top" && css`
			border-left: 5px solid transparent;
			border-right: 5px solid transparent;
			border-top: 5px solid #000;
		`}
		${props => props.direction === "left" && css`
			border-top: 5px solid transparent;
			border-bottom: 5px solid transparent;
			border-left: 5px solid #000;
		`}
		${props => props.direction === "bottom" && css`
			border-left: 5px solid transparent;
			border-right: 5px solid transparent;
			border-bottom: 5px solid #000;
		`}
		${props => props.direction === "right" && css`
			border-bottom: 5px solid transparent;
			border-right: 5px solid #000;
			border-top: 5px solid transparent;
		`}
	}

  /* .tooltip.in {
  opacity: 0.9;
} */
`;
