// @lk/ui-library/src/Component/V1/MyAccount/StoreCredit/StoreCredit.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import StoreCredit from "./index";

export default {
	title: "Components/V1/MyAccount/StoreCredit",
	component: StoreCredit,
	argTypes: {
		...generateGeneralArgs(),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left onKeepMEPosted",
			category: "Staus",
		}),
	}
} as ComponentMeta<typeof StoreCredit>;

const Template: ComponentStory<typeof StoreCredit> = (args) => <StoreCredit {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		HELLO: "Hello",
		MY_ORDERS: "My Orders",
		ORDERS_PAGE: "Orders Page",
		ORDER: "Order",
		DATE: "Date",
		SHIP_TO: "Ship To",
		ORDER_TOTAL: "Order Total",
		CHECK_STORE_CREDIT_BALANCE: "Check Store Credit Balance",
		ENTER_YOUR_STORE_CREDIT_CODE: "Enter your store credit code*",
		SEND_CREDIT_CODE: "Send Credit Code",
		ERRORMESSAGE: "This is required",
		TIME: "Time",
		AMOUNT: "Amount",
		COMMENTS: "Comments",
		EXTRA_CONTENT: "Extra Content"
	},
	firstName: "Test",
	storeCreditList: [
		{
			id: 1930487280,
			status: {
				status: "CLOSED",
				state: "CLOSED",
				returnable: false,
				cancellable: false,
				comment: "closed",
				timestamp: 1667148002009,
				trackingStatus: "DISPATCHED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "DISPATCHED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [
					{
						type: "implicit",
						amount: 0
					}
				],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 240.12
					}
				],
				totalTax: 240.12,
				shipping: 0,
				subTotal: 2001,
				total: 2241.12
			},
			shippingAddress: {
				id: 40976063,
				firstName: "test",
				lastName: "test",
				phone: "8217825953",
				phoneCode: "+91",
				email: "8217825953@lenskartomni.com",
				addressType: "shipping",
				addressline1: "sdjgajhsd",
				city: "Gurgaon",
				state: "Haryana",
				postcode: "122004",
				country: "IN",
				gender: "unknown"
			},
			createdAt: 1667145983689
		},
		{
			id: 1930487279,
			status: {
				status: "CLOSED",
				state: "CLOSED",
				returnable: false,
				cancellable: false,
				comment: "closed",
				timestamp: 1667147174485,
				trackingStatus: "RETURN_REQUESTED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "DELIVERED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [
					{
						type: "implicit",
						amount: 0
					}
				],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 240.12
					}
				],
				totalTax: 240.12,
				shipping: 0,
				subTotal: 2001,
				total: 2241.12
			},
			shippingAddress: {
				id: 40976061,
				firstName: "Kavya",
				lastName: "Kav",
				phone: "8217825953",
				phoneCode: "+91",
				email: "8217825953@lenskartomni.com",
				addressType: "shipping",
				addressline1: "test11 test11",
				addressline2: "zkchjzkxzxczxczxcxxz",
				city: "GURUGRAM",
				state: "HARYANA",
				postcode: "122001",
				country: "IN",
				gender: "unknown"
			},
			createdAt: 1667145892042
		},
		{
			id: 1930487026,
			status: {
				status: "RETURN_REFUNDED_STORE_CREDIT",
				state: "CLOSED",
				returnable: false,
				cancellable: false,
				comment: "return_refunded_store_credit",
				timestamp: 1666939877248,
				trackingStatus: "DISPATCHED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "DISPATCHED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 240.12
					}
				],
				totalTax: 240.12,
				shipping: 0,
				subTotal: 2001,
				total: 2241.12
			},
			shippingAddress: {
				id: 40975575,
				firstName: "test",
				lastName: "test",
				phone: "8217825953",
				phoneCode: "+91",
				email: "dshksj@jfdhdj.com",
				addressType: "shipping",
				addressline1: "jhgjhfghfhjfhj",
				addressline2: "hjvhjvhjv",
				city: "Gurgaon",
				state: "Haryana",
				postcode: "122004",
				country: "IN"
			},
			createdAt: 1666934597838
		},
		{
			id: 1930476118,
			status: {
				status: "PROCESSING",
				state: "PROCESSING",
				returnable: false,
				cancellable: false,
				comment: "Successful Payment with cc",
				timestamp: 1661866567043,
				trackingStatus: "IN_HOUSE_PROCESSING",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "DELIVERED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 480
					}
				],
				totalTax: 480,
				shipping: 0,
				subTotal: 4000,
				total: 4480
			},
			shippingAddress: {
				id: 40954929,
				firstName: "test",
				lastName: "test",
				phone: "8217825953",
				phoneCode: "+91",
				email: "kljnkljl@hbjh.com",
				addressType: "shipping",
				addressline1: "lkdshfjdksh",
				addressline2: "",
				city: "Gurgaon",
				floor: 0,
				liftAvailable: false,
				state: "Haryana",
				postcode: "122004",
				country: "IN",
				gender: "female"
			},
			createdAt: 1661866555966
		},
		{
			id: 1930475677,
			status: {
				status: "PROCESSING",
				state: "PROCESSING",
				returnable: false,
				cancellable: false,
				comment: "Order placed using cod",
				timestamp: 1661755742278,
				trackingStatus: "ORDER_PLACED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "ORDER_PLACED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [
					{
						type: "implicit",
						amount: 0
					},
					{
						code: "LENS-GAGAN",
						type: "gv",
						amount: 500
					}
				],
				totalDiscount: 500,
				taxes: [
					{
						name: "VAT",
						amount: 967.68
					}
				],
				totalTax: 967.68,
				shipping: 0,
				subTotal: 5300,
				total: 5767.68
			},
			shippingAddress: {
				id: 40954077,
				firstName: "test",
				lastName: "test",
				phone: "8217825953",
				phoneCode: "+91",
				email: "jkdfhkj@jkdhfj.com",
				addressType: "shipping",
				addressline1: "jhgjhgjhghj",
				city: "Gurgaon",
				state: "Haryana",
				postcode: "122004",
				country: "IN",
				gender: "female"
			},
			createdAt: 1661755654404
		},
		{
			id: 1930471836,
			status: {
				status: "CLOSED",
				state: "CLOSED",
				returnable: false,
				cancellable: false,
				comment: "closed",
				timestamp: 1660024035238,
				trackingStatus: "RETURN_REQUESTED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "DELIVERED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 0
					}
				],
				totalTax: 0,
				shipping: 0,
				subTotal: 2001,
				total: 2001
			},
			shippingAddress: {
				id: 40946675,
				firstName: "test",
				lastName: "test",
				phone: "8217825953",
				phoneCode: "+91",
				email: "8217825953@lenskartomni.com",
				addressType: "shipping",
				addressline1: "jhgjhgjhghj",
				addressline2: "",
				city: "Gurgaon",
				floor: 0,
				liftAvailable: false,
				state: "Haryana",
				postcode: "122004",
				country: "IN"
			},
			createdAt: 1659958365506
		},
		{
			id: 1930467711,
			status: {
				status: "CLOSED",
				state: "CLOSED",
				returnable: false,
				cancellable: false,
				comment: "closed",
				timestamp: 1658413176237,
				trackingStatus: "RETURN_REQUESTED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "DELIVERED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [
					{
						type: "implicit",
						amount: 0
					}
				],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 120
					}
				],
				totalTax: 120,
				shipping: 0,
				subTotal: 1000,
				total: 1120
			},
			shippingAddress: {
				id: 40938825,
				firstName: "test",
				lastName: "test",
				phone: "8217825953",
				phoneCode: "+91",
				email: "sajgsajh@shghds.com",
				addressType: "shipping",
				addressline1: "ashdghjsad",
				addressline2: "asdhagsdhja",
				city: "Gurgaon",
				state: "Haryana",
				postcode: "122004",
				country: "IN",
				gender: "unknown"
			},
			createdAt: 1658225380068
		},
		{
			id: 1930464794,
			status: {
				status: "PREDELIVERY_REFUND",
				state: "CLOSED",
				returnable: false,
				cancellable: false,
				comment: "predelivery_refund",
				timestamp: 1656660885386,
				trackingStatus: "IN_HOUSE_PROCESSING",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "DELIVERED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 192
					}
				],
				totalTax: 192,
				shipping: 0,
				subTotal: 1600,
				total: 1792
			},
			shippingAddress: {
				id: 40933241,
				firstName: "Kavya",
				lastName: "Kav",
				phone: "8217825953",
				phoneCode: "+91",
				email: "kavyaar34@gmail.com",
				addressType: "shipping",
				addressline1: "test11 test11",
				addressline2: "zkchjzkxzxczxczxcxxz",
				city: "GURUGRAM",
				floor: 0,
				liftAvailable: false,
				state: "HARYANA",
				postcode: "122001",
				country: "IN",
				locality: "",
				gender: "unknown"
			},
			createdAt: 1656649797555
		},
		{
			id: 1930464020,
			status: {
				status: "PENDING",
				state: "NEW",
				returnable: false,
				cancellable: false,
				comment: "Power is updated via 'Edit Power' by deepc@valyoo.in",
				timestamp: 1656409611207,
				trackingStatus: "PAYMENT_NOT_INITIATED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "ORDER_PLACED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [
					{
						type: "implicit",
						amount: 0
					}
				],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 1012.08
					}
				],
				totalTax: 1012.08,
				shipping: 0,
				subTotal: 7300,
				total: 8312.08
			},
			shippingAddress: {
				id: 40931873,
				firstName: "test",
				lastName: "test",
				phone: "8217825953",
				phoneCode: "+91",
				email: "kavya.ar@lenskart.com",
				addressType: "shipping",
				addressline1: "hjsgxjhs",
				addressline2: "sdjgajskhgdaasd",
				city: "Gurgaon",
				state: "Haryana",
				postcode: "122004",
				country: "IN",
				gender: "male"
			},
			createdAt: 1656409403587
		},
		{
			id: 1930461879,
			status: {
				status: "CLOSED",
				state: "CLOSED",
				returnable: false,
				cancellable: false,
				comment: "closed",
				timestamp: 1659334485148,
				trackingStatus: "RETURN_REQUESTED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "DELIVERED"
			},
			amount: {
				currencyCode: "INR",
				discounts: [
					{
						type: "implicit",
						amount: 0
					}
				],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 192
					}
				],
				totalTax: 192,
				shipping: 0,
				subTotal: 1600,
				total: 1792
			},
			shippingAddress: {
				id: 40927871,
				firstName: "test",
				lastName: "test",
				phone: "8217825953",
				phoneCode: "+91",
				email: "kavya.ar@lenskart.com",
				addressType: "shipping",
				addressline1: "hjsgxjhs",
				addressline2: "sdjgajskhgdaasd",
				city: "Gurgaon",
				state: "Haryana",
				postcode: "122004",
				country: "IN",
				gender: "unknown"
			},
			createdAt: 1655380162498
		}
	],
	onCheckStoreCredit: () => console.log("Store credit is valid"),
	onClickStoreCreditOrderID: () => console.log("Store Credit Order Id"),
	onClickStoreCreditCodeBySms: () => console.log("store credit code by sms")
};
