// @lk/ui-library/src/Component/V1/MyAccount/StoreCredit/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";
import { positionENUM } from "./StoreCredit.types";

export const StoreCreditWrapper = styled.div`
    background-color: #ffffff;
`;

export const UserName = styled.div`
	color: rgb(109, 111, 113);
	letter-spacing: 1.3px;
	margin-top: 30px;
	font-family: "FuturaStd-Heavy";
	margin-bottom: 20px;
`;

export const Heading4 = styled.h4`
	font-weight: 400px;
	font-size: 18px;
`;

export const HeaderWrapper = styled.div`
`;

export const Heading = styled.h2`
	font-family: "FuturaStd-Medium";
	font-weight: 400;
	font-size: 30px;
	margin-top: 20px;
	margin-bottom: 10px;
	line-height: 1.1;
	color: inherit;
`;

export const StoreCreditWrap = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
`;

export const P = styled.p`
	color: rgb(109, 111, 113);
	text-transform: uppercase;
	margin: 0 0 10px;
`;

export const UL = styled.ul``;

export const OrderListingWrapper = styled.div`
.text-link {
    color: #329c92;
    cursor: pointer;
};
.text-weight {
    font-weight: 600;
}
`;

export const Table = styled.table`
	width: 100%;
	max-width: 100%;
	margin-bottom: 20px;
	background-color: transparent;
	border-collapse: collapse;
	border-spacing: 0;
	border-color: grey;
	font-size: 14px;
`;

export const TableHead = styled.thead`
font-weight: bold;
`;

export const TableRow = styled.tr``;

export const TableHeading = styled.th`
	padding: 8px;
	line-height: 1.42857143;
	text-align: left;
	font-weight: bold;
	font-size: 14px;
`;

export const TableBody = styled.tbody`
    tr:nth-of-type(odd) {
        background-color: #f9f9f9;
    };
`;

export const TableData = styled.td`
padding: 8px;
line-height: 1.42857143;
vertical-align: top;
border-top: 1px solid #ddd;
`;

export const Span = styled.span``;

export const InputField =
	styled.input <
	{ error: boolean, disabled: boolean, prefixText: boolean, isRTL: boolean, width: string } >`
    width: ${(props) => (props.width ? props.width : "")}%;
    border: 1px solid ${props => props.error ? "#a94442" : "var(--serene-gray)"};
    border-radius: 4px;
    z-index: 42;
    background: #fff;
    padding: 6px;
    margin: 0;
    position: relative;
    letter-spacing: 1px;
	box-shadow: inset 0 1px 1px rgb(0 0 0 / 8%);
    box-sizing: border-box;
    font-size: var(--fs-14);
    line-height: 20px;
    letter-spacing: -.02em;
    color: rgba(0,0,66,.5);
    font-family: ${TypographyENUM.lkSansRegular};
    outline: none;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const InputLabel =
	styled.label <
	{ prefixText: boolean, labelPosition: positionENUM, error:boolean } >`
	font-size: 1em;
	margin-bottom: 0.4em;
	${(props) => props.labelPosition}:${(props) => (props.prefixText ? "2em" : "0")};
	color: ${props => props.error ? "#a94442" : "#333" };
    display: inline-block;
    max-width: 100%;
    margin-bottom: 5px;
`;

export const ButtonWrapper = styled.div`
	margin-top: 2%;
	width: 100%;
`;

export const ErrorMessage = styled.p`
	color: var(--error);
	margin: 0;
	padding-top: var(--pd-5);
	font-size: 0.8em;
`;

export const InputWrapper = styled.div`
	width: 50%;
	margin-bottom: 5px;
	display: flex;
	flex-direction: column;
`;

export const DividerWrapper = styled.div`
	display: block;
	margin: 12px 0;
	height: 1px;
	padding: 0;
`;

export const Border = styled.div<{id: string}>`
    border: 1px dashed rgb(204, 204, 204);
`;

export const StoreCreditHeading = styled.h4`
	font-size: 18px;
	font-family: 'LKFuturaStd-Medium';
	font-weight: 400;
	margin-top: 10px;
	margin-bottom: 10px;
`;
