// @lk/ui-library/src/Component/Common/PopUpText/PopUpText.styles.tsx
import styled, { css, keyframes } from "styled-components";
import { devices } from "../../../helper";
import { TypographyENUM } from "../../../Types/general";

export const breatheAnimation = keyframes`
 0% { transform: translateY(-80px); opacity: 0.7;}
 100% { transform: translateY(0px); opacity: 1; }
`;

export const Root = styled.div<{isRTL: boolean, mobileView?: boolean, top?: boolean}>`
  z-index: 11;
  width: 100%;
  position: ${props => props.mobileView ? "fixed" : "sticky"};
//   width: 100%;
//   top: 8%;
//   right: 50%;
//   transform: translate(50%, -50%);
  transform: translateY(0px);
  opacity: 1;
  animation: ${breatheAnimation} 250ms ease;

  ${props => props.top && css`
    position: fixed;
    top: 8%;
    width: 65%;
  `};

  @media ${devices.laptop} {
    // width: calc(70% - 55px);
    ${props => props.isRTL ? "right: 20px" : ""};
    transform: none;
  }
`;

export const PopUpContent = styled.div`
  position: absolute;
  width: 100%;
  padding: 32px 16px;
  gap: 10px;
  display: flex;
  align-items: center;
  justify-content: space-between;
  height: 52px;
  z-index: 12;
  background: #e8fdff;
  box-shadow: 0px 1px 4px rgba(0, 0, 0, 0.1);
  border-radius: 12px;
  cursor: pointer;

  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  /* font-weight: 700; */
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #000042;
`;
