// @lk/ui-library/src/Component/Common/DesktopPopUp/DesktopPopUp.styles.tsx
import styled from "styled-components";
import { animated } from "react-spring";
import { TypographyENUM } from "../../../Types/general";

export const ScreenContainer = styled.div`
  background: rgba(0, 0, 66, 0.5);
    box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    display: flex;
    align-items: center;
    justify-content: center;
    position: fixed;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    z-index: 1000;
`;

export const PopUpModal = styled.div`
  position:relative;
`;

export const CrossSection = styled.div`
  cursor:pointer;
  margin-left:20px;
`;

export const MainSection = styled(animated.div)``;

export const Header = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: 24px;
  line-height: 36px;
  letter-spacing: -0.02em;
  color: #333368;
`;

export const Description = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #66668E;
`;

export const HeaderContent = styled.div`
  display:flex;
  align-items:center;
  justify-content:space-between;
  width: 100%;
`;

export const PopUp = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  z-index: 1000;
  position: fixed;
  left: 36%;
  top: 50%;
  transform: translateY(-50%);
`;

export const PopUpModalContent = styled.div`
  padding: 24px;
  background: #FFFFFF;
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
  border-radius: 12px;
  background: white;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
`;
