// @lk/ui-library/src/Component/V1/FilterSortContainer/FilterSortContainer.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import FilterSortContainer from "./index";
export default {
	title: "Components/v1/FilterSortBtn",
	component: FilterSortContainer,
} as ComponentMeta<typeof FilterSortContainer>;
const Template: ComponentStory<typeof FilterSortContainer> = (args) => <FilterSortContainer {...args} />;
export const Overview = Template.bind({});
Overview.args = {
	bottomSheetData: {
		title:"Test",
		data:[],
		type:"checkbox",
		sortBy:"string",
		filterList:[]
	},
	showBottomSheet: true,
	closeBottomSheet: () => console.log("close boottom sheet"),
	openBottomSheet: (id: any) => console.log("id", id),

	selectedSize: true,
	onSizeTitleClick: (id: any) => console.log(id),
	onSortClick: (id: any) => console.log(id),
	onClearClick: () => console.log("clear"),
	onApplyClick: () => console.log("apply"),
	sortBy: "sort by",
	dataLocale: {
		CLEAR_ALL: "CLEAR ALL",
		ALL_SIZE: "All Size",
		SORT_BY: "Sort By",
		FILTERS: "Filters",
		APPLY: "APPLY"
	}
};
