// @lk/ui-library/src/Component/V1/FilterSortContainer/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const FilterSortBtnWrapper = styled.div`
z-index: 11;
font-family: ${TypographyENUM.lkSansRegular};
height: 65px;
padding:10px;
display:flex;
justify-content: space-around;
background-color: #fff;
position:fixed;
bottom:0;
width:100%;
border-top: 1px solid #eaeff4;
`;

export const FilterSortBtn = styled.div<{isRTL: boolean}>`
// ${props => props.isRTL ? "border-left" : "border-right"}: 1px solid #dedede;
display:flex;
align-items:center;
width: 33%;
justify-content: center;
&.all-size{
	justify-content: left;
	padding-left: 5px;
}
// &:last-child {
//   ${props => props.isRTL ? "border-left" : "border-right"}: none;
// };
`;


export const FilterSortBtnText = styled.span`
font-family: ${TypographyENUM.lkSansBold};
font-weight: 700;
font-size: 14px;
letter-spacing: -.02em;
color: #000042;
line-height: 18px;
`;

export const FilterSortBtnIcon = styled.span`
&:before {
  content: " ";
}
margin: 4px 4px 0px 4px;
`;


export const BottomSheetBtnWrapper= styled.div`
font-family: ${TypographyENUM.lkSansRegular};
    bottom: 3px;
    width: 100%;
    z-index: 10;
    top: auto;
    position: fixed;
    padding: 5px 10px;
    display: flex;
	left:0;
   `;
export const BottomSheetBtn = styled.span<{
  isApply:boolean
}>`

flex: 1;
text-align: center;
text-transform: uppercase;
border-radius: 4px;
letter-spacing: 1px;
border: 1px solid #18cfa8;
padding:10px 10px;
margin:0 5px;
background-color:${(props) =>
		props.isApply ? "#18cfa8" : null};
color:${(props) =>
		props.isApply ? "#fff" : "#18cfa8"}
`;

export const BottomSheetTitle= styled.div`
font-family: ${TypographyENUM.lkSansBold};
    padding: 5px 14px 15px;
    font-size: 20px;
    /* font-weight: 700; */
	display: flex;
    align-items: center;
    gap: 42px;
    border-bottom: 1px solid #f0f0f0;
	svg{
		margin-left: -13px;
		width: 28px;
		height: 28px;
		margin-top: -3px;
	}
`;
export const BottomSheetTile= styled.div`
   font-family: ${TypographyENUM.lkSansMedium};
   font-size:14px;
   border-bottom: 1px solid #f0f0f0;
   justify-content: space-between;
   display:flex;
   padding: 15px 10px;
   margin-bottom: 10px;
   background-color: #fff;
   `;

export const BottomSheetListTitle = styled.span`
   text-transform: capitalize;
`;
export const BottomSheetListRadioWrapper = styled.label`

`;
export const BottomSheetListRadio= styled.input.attrs({ type: "radio" })`
&:before {
  content: "";
}
transform: scale(1.5);
user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;
export const BottomSheetIndicator = styled.span`
    width: 10px;
    height: 10px;
    top: -15px;
    left: 8px;
    background-color:#18cfa8;
    border-radius: 50%;
    position: relative;
`;

export const Divider = styled.div`
border-left: 1px solid #ccc;
height: 100%;
margin: 0 10px;
`;
