// @lk/ui-library/src/Component/NewHtoFlow/WhatToExcept/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const WhatToExceptWrapper = styled.div`

`;


export const WhatToExceptText = styled.div`
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: var(--fs-16);
    line-height: 20px;
    color: var(--dark-blue-100);
    margin-bottom: 10px;
`;

export const IconTextWrapper =styled.div`
    display: flex;
    flex-direction: column;
    gap: 12px;
`;

export const IconTextContainer = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
`;

export const IconWrapper = styled.div`
`;

export const Icon = styled.img`
    width: 30;
    height: 30;
`;

export const TextWrapper = styled.div`
    margin-left: 10px;
    font-family: ${TypographyENUM.lkSansMedium};
    font-style: normal;
    font-weight: var(--fw-medium);
    font-size: var(--fs-14);
    line-height: 20px;
    color: var(--dark-blue-100);
`;
