// @lk/ui-library/src/Component/NewHtoFlow/TermsAndConditions/TermsAndConditions.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import TermsAndCondition from ".";

export default {
	title: "NewHtoFlow/TermsAndCondition",
	component: TermsAndCondition,
	argTypes: {}
} as ComponentMeta<typeof TermsAndCondition>;

const Template: ComponentStory<typeof TermsAndCondition> = (args) => <TermsAndCondition {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		"TERMS_AND_CONDITION": "Terms and conditions",
	},
	redisCommonData: {
		TERMS_AND_CONDITIONS: "[\"The age group of the person whose eyes are getting tested should be between 14 and 75 years\",\"The person whose eyes are getting tested should be in good physical health\",\"If the person whose eyes are getting tested has high blood pressure or high blood sugar, their condition should be under control at the time of the check-up\",\"The eye check-up location must be situated on the third floor or below. If the location is on the fourth floor or higher, the building must have a functional lift in working condition to reach the eye check-up area\",\"Prior to the eye check-up, it is required that the person whose eyes are getting tested and anyone else residing in their household is free from symptoms of a cold, cough, or flu\",\"A minimum clear space of 10-12 feet in a straight line is required to perform the eye check-up.\"]"
	}
};
