// @lk/ui-library/src/Component/V1/MyAccount/VoucherBalance/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";
import { positionENUM } from "./VoucherBalance.types";


export const HeaderWrapper = styled.div`
font-family: "FuturaStd-Medium";
`;

export const Heading = styled.h2`
    margin-top: 20px;
    margin-bottom: 10px;
    font-family: "FuturaStd-Medium";
    font-weight: 400;
    font-size: 30px;
`;

export const InputField = styled.input<{error:boolean, disabled:boolean, prefixText:boolean, isRTL?: boolean, width: string}>`
    width: ${props => props.width ? props.width : ""}%;
    border: 1px solid ${props => props.error ? "#a94442" : "#843534"};
    border-radius: 4px;
    z-index: 42;
    background: #fff;
    padding: 6px 12px;
    box-shadow: inset 0 1px 1px rgb(0 0 0 / 8%);
    margin: 0;
    position: relative;
    letter-spacing: 1px;
    box-sizing: border-box;
    font-size: var(--fs-14);
    line-height: 20px;
    letter-spacing: -.02em;
    color: rgba(0,0,66,.5);
    font-family: ${TypographyENUM.lkSansRegular};
    outline: none;
    transition: border-color ease-in-out 0.15s,box-shadow ease-in-out 0.15s,-webkit-box-shadow ease-in-out 0.15s;
    :focus {
        outline: none;
        box-shadow: ${ props => props.error ? "inset 0 1px 1px rgba(0,0,0,.075), 0 0 6px #ce8483" : "inset 0 1px 1px rgba(0,0,0,.075), 0 0 8px rgba(102,175,233,.6); auto 1px"};
    }
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const InputLabel= styled.label<{prefixText:boolean, labelPosition?: positionENUM, error:boolean }>`
	font-size: 13px;
	margin-bottom: 0.4em;
	${props => props.labelPosition}:${props => props.prefixText ? "2em" : "0"};
	pointer-events: none;
	color: ${props => props.error ? "#a94442" : "#333" };
    display: inline-block;
    max-width: 100%;
    margin-bottom: 5px;
    font-weight: 700;
    line-height: 1.42857;
    font-family:  FuturaStd-Book, helvetica neue, Helvetica, Arial, sans-serif;
`;

export const ButtonWrapper = styled.div`
    margin-top: 2%;
    width: 100%
    text-transform: uppercase;
    font-family: "FuturaStd-Medium";
    font-size: 14px;
    letter-spacing: 1px;
    white-space: initial;
    line-height: 1.42857143;
    border: 1px solid transparent;
	button{
		text-transform:uppercase;
	}
`;

export const ErrorMessage = styled.p`
	color:var(--error);
	margin:0;
	padding-top:var(--pd-5);
	font-size:0.8em;
`;

export const InputWrapper = styled.div`
    width: 50%;
    margin-bottom: 15px;
    display: flex;
    flex-direction: column;
`;

export const Border = styled.div<{id: string}>`
    border: 1px dashed rgb(204, 204, 204);
    width: 66%;
`;
