// @lk/ui-library/src/Component/V1/CheckoutMobile/Grabpay/Grabpay.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { Grabpay } from "./Grabpay.component";

export default{
	title: "Components/v1/CheckoutMobile/Grabpay",
	component: Grabpay,
	argTypes: {
		methodData: generateArgTypes({
			category: "General",
			required: true,
			type: "object",
			description: "Contains the payment method data",
		})
	}
} as ComponentMeta<typeof Grabpay>;

const Template: ComponentStory<typeof Grabpay> = (args) => <Grabpay {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	localeData : {
		"ATOME": "Atome"
	},
	groupData: { methods:[
		{
			logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png",
			key: "ATOME",
			label: "ATOME",
			offerText: "Flat 50% off"
		},
		{
			logoImageUrl: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png",
			key: "HOOLAH",
			label: "HOOLAH"
		}
	] },
	renderPrimerUI: (key) => console.log({ key }),
	setPayLaterSelected: () => null
};
