// @lk/ui-library/src/Component/V1/CheckoutMobile/SavedAddressesTiles/SavedAddressesTiles.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { SavedAddressTiles } from "./SavedAddressesTiles.component";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/v1/CheckoutMobile/SavedAddressTiles",
	component: SavedAddressTiles,
	argTypes: {
		editAddress: generateArgTypes({
			type: "function",
			required: true,
			description: "editAddress handler to edit existing address",
			category: "Events",
		}),
		deleteAddress: generateArgTypes({
			type: "function",
			required: true,
			description: "deleteAddress handler to delete existing address",
			category: "Events",
		}),
		isShowMore: generateArgTypes({
			type: "boolean",
			required: true,
			description: "isShowMore state indicates whether need to show all address or not",
			category: "State",
			defaultValue: false
		}),
		handleShowMoreClick: generateArgTypes({
			type: "function",
			required: true,
			description: "handles whether to show all address tiles or few",
			category: "Events",
		})
	}
} as ComponentMeta<typeof SavedAddressTiles>;

const Template: ComponentStory<typeof SavedAddressTiles> = (args) => <SavedAddressTiles {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale:{
		EDIT: "Edit",
		DELETE: "Delete",
		PH: "Ph:",
		EXPECTED_DELIVERY_BY: "Expected delivery by",
		SAVED_ADDRESSESS: "Saved addresses",
		SHOW_MORE: "Show more",
		SHOW_LESS: "Show Less"
	},
	
	addresses:[
		{
			addressline1: "test",
			addressline2: "test",
			city: "Noida",
			country: "IN",
			firstName: "test",
			lastName: "test",
			phone: "9999999999",
			postcode: "878787",
			state: "CHANDIGARH",
			addressLabel: "Home",
			id:1
		},
		{
			addressline1: "test",
			addressline2: "test",
			city: "Noida",
			country: "IN",
			firstName: "test",
			lastName: "test",
			phone: "9999999999",
			postcode: "878787",
			state: "CHANDIGARH",
			addressLabel: "Home",
			id:2
		},
		{
			addressline1: "test",
			addressline2: "test",
			city: "Noida",
			country: "IN",
			firstName: "test",
			lastName: "test",
			phone: "9999999999",
			postcode: "878787",
			state: "CHANDIGARH",
			addressLabel: "Home",
			id: 3
		}
	],
	font:TypographyENUM.lkSansRegular,
};
