// @lk/ui-library/src/Component/Common/FloatingLabelDropdown/styles.tsx
import styled from "styled-components";
import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";

const getSize = (size: ComponentSizeENUM) => {
	if (size === "small") return "var(--fs-12)";
	if (size === "medium") return "var(--fs-16)";
	if (size === "extraLarge") return "var(--fs-24)";
	else return "var(--fs-18)";
};

export const DropdownWrapper = styled.div<{
	styleSize: ComponentSizeENUM
	styleFont: TypographyENUM
}>`
	font-family: ${(props) => props.styleFont};
	font-size: ${(props) => getSize(props.styleSize)};
	padding: 0;
    margin: 0 0 10px 0;
    width: 100%;
    border-radius: 4px;
    overflow: hidden;
    border: 1px solid #d3d3d3;
	position: relative;
	display: flex;
	justify-content: space-between;
	align-items: center;
	background-color: field;
`;

export const DropdownLabel = styled.label`
	position: absolute;
	font-size: 12px;
	pointer-events: none;
	-webkit-transition: 0.2s ease all;
	transition: 200ms cubic-bezier(0, 0, 0, 1) 0ms;
	margin-bottom: 5px;
	left:  "16px";
	color: var(--text);
    line-height: 20px;
	color: var(--text);
    top: -4px;
	text-transform: capitalize;
    margin-top: 5px;
    padding: 4px 16px;
`;

export const SelectBox = styled.select<{disabled?:boolean}>`
	padding: 22px 16px 10px;
    width: 100%;
    border: none;
    font-size: 14px;
    outline: none;
    -webkit-box-shadow: none;
    box-shadow: none;
    -webkit-appearance: none;
    -moz-appearance: none;
    appearance: none;
	color: var(--text);
	font-weight: 400;
    line-height: 24px;
    letter-spacing: -0.02em;
	cursor: ${(props) => (props.disabled ? "not-allowed" : "auto")};
`;

export const Option = styled.option`
	font-family: inherit;
	font-size: 1em;
	line-height: 1em;
`;
