// @lk/ui-library/src/Component/V1/Cart/KnowMoreModal/KnowMoreModal.styles.tsx
import styled from "styled-components";

export const KnowMoreContainer = styled.div`
  z-index: 1050 !important;
	width: 100%;
	height: 100%;
	position: fixed;
	display: flex;
	align-items: center;
	justify-content: center;
	top: 0;
	left: 0;
	cursor: auto;
`;

export const Backdrop = styled.div`
  position: fixed;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-color: rgba(0,0,0,0.5);
  transition: opacity 0.15s linear;
  z-index: 1;
`;

export const Content = styled.div`
  a {
    text-decoration: none;
    color: inherit;
  }
  position: relative;
  margin: 100px auto;
  // width: fit-content;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding-bottom: 20px;
  gap: 20px;
  z-index: 2;
  background: #fff;
  border-radius: 12px;
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #000042;
`;

export const Close = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 8px;
  gap: 4px;

  position: absolute;
  // position: fixed;
  width: 40px;
  height: 40px;
  top: 16px;
  right: 16px;
  background: #ffffff;
  border-radius: 100px;
  cursor: pointer;
`;

export const ModalImageContainer = styled.div`
  img {
    border-radius: 12px;
    height: 720px;
    transition: all ease 0.5s;
    @media screen and(max-width: 1441px) {
      height: 620px;
    }
    @media screen and (max-width: 1366px) {
      height: 420px;
    }
    // @media screen and (max-width: 1200px) {
    // 	height: 520px;
    // }
    // @media screen and (max-width: 800px) {
    // 	height: 420px;
    // }
    @media screen and (max-width: 720px) {
      height: 380px;
    }
  }
`;

export const ModalImage = styled.img``;

export const AnchorText = styled.a`
  text-decoration: underline;
  cursor: pointer;
`;

export const Text = styled.div`
  text-decoration: underline;
  cursor: pointer;
`;
