// @lk/ui-library/src/Component/V1/Cart/KnowMoreModal/KnowMoreModal.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import KnowMoreModal from "./KnowMoreModal.component";

export default {
	title: "Components/V1/Cart/KnowMoreModal",
	component: KnowMoreModal,
	argTypes: {
		show: generateArgTypes({
			category: "General",
			description: "Show or hide the modal",
			required: true,
			type: "boolean",
			defaultValue: false,
		}),
		onHide: generateArgTypes({
			category: "General",
			description: "Function to run when the close button is clicked",
			required: true,
			type: "function"
		}),
		imagePath: generateArgTypes({
			category: "General",
			description: "Path of the image to render in the modal",
			required: true,
			type: "string"
		}),
		url: generateArgTypes({
			category: "General",
			description: "Url where the user will be redirected on click of the CTA",
			required: true,
			type: "string"
		}),
		cta: generateArgTypes({
			category: "General",
			description: "The text to show in the modal's primary button",
			required: true,
			type: "string"
		})
	}
} as ComponentMeta<typeof KnowMoreModal>;

const Template: ComponentStory<typeof KnowMoreModal> = args => <KnowMoreModal {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	show: false,
	onHide: () => null,
	cta: "Know more",
	imagePath: "https://static1.lenskart.com/media/desktop/img/Feb22/18-Feb/gold_bottom.jpg",
	url: "https://www.lenskart.com/loyalty"
};
