// @lk/ui-library/src/Component/V1/MyAccount/Mobile/ExchangeReturnDetails/ExchangeReturnDetails.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const ExchangeInfoWrapper = styled.div<{styleFont:TypographyENUM}>`
	flex-grow:1;
	display:flex;
	flex-direction:column;
	gap:var(--pd-10);
	font-family:${props => props.styleFont};
	padding-bottom:50px;
`;

export const ProductReturnWrapper = styled.div`
	background:#fff;
	padding:10px 0;
`;
export const ExchangePrice = styled.div`
	margin-top:10px;
`;
export const NeedHeading = styled.div``;
export const ExchangeProductInfo = styled.div`
	display:flex;
	justify-content:space-between;
	padding:10px 15px;
	align-items:center;
	border-bottom: 1px solid #eaeff4;
	border-top: 1px solid #eaeff4;
`;
export const ExchangeProductImage = styled.div`
   max-width:120px;
   img{
	 max-width:100%;
   }
`;

export const ExchangeProductDes = styled.div`
   padding-left:10px;
`;

export const NeedHelp = styled.div`
	font-size:var(--fs-12);
	color:var(--whitish-gray);
	margin-top:5px;
`;

export const ExchangeBtnWrapper = styled.div`
	position:fixed;
	left:10px;
	right:10px;
	bottom:0;
	margin: auto -10px;
	padding: 10px;
	border: 1px solid #eaeff4;
	background: #fff;
	font-weight: var(--fw-bold);
	button[disabled]{
		background-color: #ccc;
        pointer-events: none;
		border:1px solid #ccc;
	}
	button{
		text-transform:uppercase;
	}
`;

export const ProductTexTitle = styled.strong`
    font-size:var(--fs-14);
	font-weight: var(--fw-bold);
	margin-bottom:8px;
	display:block;
	color: #333;
`;

export const NeedHelpWrapper = styled.div`
    font-size:var(--fs-16);
	font-weight: var(--fw-bold);
	margin-bottom:6px;
	text-align:center;
	background:#fff;
	padding:20px 10px;
	img{
		max-width:40px;
	}
`;

export const ProductTextDes = styled.p`
	font-size:var(--fs-12);
	text-align:left;
	font-family:var(--font-lksans-regular);
	color:var(--whitish-gray);
	line-height:18px;
	margin-bottom:5px;
`;
export const ChatWithUs = styled.div`
	display:flex;
	justify-content:center;
	margin-top:10px;
	p{
		color:var(--turquoise);
		font-size:var(--fs-12);
	    font-weight: var(--fw-bold);
	}

`;

export const ReturnTitle = styled.div`
	font-size:var(--fs-14);
	/* font-weight: var(--fw-bold); */
	text-align:left;
	padding:0 var(--pd-10) var(--pd-15) var(--pd-10);
	font-family: var(--font-lksans-bold);
`;
export const ProductSku = styled.div`
    font-size:var(--fs-12);
	text-align:left;
	font-family:  var(--font-lksans-regular);
	color:var(--whitish-gray);
`;

export const NeedAddress = styled.div`
	font-size:var(--fs-14);
	font-weight: var(--fw-normal);
	text-align:left;
	padding:var(--pd-12) var(--pd-10) 2px var(--pd-10);
`;
export const CancelledWrapper = styled.div`
	background:#fff;
	padding:20px 10px;
`;

export const CancelledTitle = styled.div`
	font-size:var(--fs-18);
	font-weight: var(--fw-bold);
	text-align:center;
	padding:0 var(--pd-10) 10px 0;
`;

export const CancelledImage = styled.div`
	margin-bottom:6px;
	text-align:center;
	img{
		width:50px;
	}
`;
export const StoreAddress = styled.div`
	margin-bottom:6px;
	text-align:center;
`;
export const AddressStoreWrapper = styled.div``;
export const StoreCallAction = styled.div`
 display:flex;
 justify-content:flex-end;
 gap:10px;
  i{
	display:flex;
	align-items:center;
	border-radius:50%;
	border:1px solid #eee;
	justify-content:center;
	width:45px;
	height:45px;
	margin:0 auto 5px;
	svg{
		width:20px;
		height:20px
	}
  }
`;

export const DirectionBtn = styled.div`
   font-size:var(--fs-12);
   font-weight: var(--fw-bold);
   white-space: nowrap;
`;
export const CallActionBtn = styled.div`
   font-size:var(--fs-12);
   /* font-weight: var(--fw-bold); */
   font-family:  var(--font-lksans-bold);
   white-space: nowrap;
`;

export const StoreddressText = styled.div`
    font-size:var(--fs-12);
	/* font-weight: var(--fw-bold); */
	padding:0 var(--pd-10) 10px 0;
	color:var(--whitish-gray);
	font-family:  var(--font-lksans-bold);
`;
export const StoreNameText = styled.div`
    font-size:var(--fs-14);
	/* font-weight: var(--fw-bold); */
	font-family: var(--font-lksans-bold);
	padding:0 0 5px 0;
`;
export const CompleteStoreAddress = styled.div`
	font-size:var(--fs-12);
	font-weight: var(--fw-bold);
	padding:0 var(--pd-25) 10px 0;
`;
export const P = styled.div`
  color:var(--whitish-gray);
`;
export const StoreTimings = styled.div`
	font-size:var(--fs-12);
	/* font-weight: var(--fw-bold); */
	font-family: var(--font-lksans-bold);
	padding:10px 0 5px 0;
	span{
	   font-weight: var(--fw-normal);
	   font-family: var(--font-lksans-regular);
	}
`;

export const CompleteStoreAddressWrapper = styled.div`
	display:flex;
	justify-content:space-between;
	align-items:center;
`;
