// @lk/ui-library/src/Component/V1/MyAccount/Mobile/ExchangeReturnDetails/ExchangeReturnDetails.type.tsx
import { ComponentSizeENUM, DataType, TypographyENUM } from "../../../../../Types/general";

export interface ExchangeReturnDetailsType{
	id: string,
	pid: number;
	price: PriceType;
	componentSize: ComponentSizeENUM,
	font: TypographyENUM,
	dataLocale:DataType;
	SHARE_YOUR_FEEDBACK:string,
	PRODUCT_IMAGE:string,
	PRODUCT_TITLE:string,
	PrimaryText:string,
	TICK_TEXT:string,
	textChat:string,
	NEED_HELP:string,
	PRODUCT_DESCRIPTION:string,
	LENS_DESCRIPTION:string,
	SKU:string,
	framePrice?:PriceType
	status: Array<ConfigDataType>,
	isCurrent:string,
	isCancelled:string,
	CROSS_ICON:string,
	CANCELLED_TITLE:string,
	storeAddress:DataType,
	shippingAddress:DataType,
	isRTL:boolean,
}

export interface ConfigDataType {
	id:number,
	date:string,
	title:string,
	icon:React.ReactNode,
}

export enum CurrencySymbol {
	INR="₹",
	SGD="$",
	AED = "AED"
}

export enum CurrencyType {
	INR="INR",
	SGD="SGD",
	AED="AED"
}

export interface PriceType {
	symbol: CurrencySymbol,
	currency: CurrencyType,
	basePrice: number,
	lkPrice: number,
}
