// @lk/ui-library/src/Component/V1/MyAccount/Mobile/ExchangeReturnDetails/ExchangeReturnDetails.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { Cross, Tick } from "../../../../../Icons";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { CurrencyType, TypographyENUM } from "../../../../../Types/general";
import { ExchangeReturnDetails } from "./ExchangeReturnDetails.component";
import { CurrencySymbol } from "./ExchangeReturnDetails.type";

export default {
	title: "Components/V1/MyAccount/Mobile/ExchangeReturnDetails",
	component: ExchangeReturnDetails,
	argTypes: {
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.lkSansRegular,
			control: "radio",
			options: TypographyENUM,
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			control: "object"
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left ",
			category: "Status",
			defaultValue: "false",
		}),
	}
} as ComponentMeta<typeof ExchangeReturnDetails>;

const Template: ComponentStory<typeof ExchangeReturnDetails> = args => <ExchangeReturnDetails {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	id: "exchange-return-details",
	font: TypographyENUM.lkSansRegular,
	dataLocale: {
		SKU: "150604",
		PRODUCT_TITLE: "Lenskart Air",
		PRODUCT_DESCRIPTION: "Gunmetal Aviator Full Rim",
		LENS_DESCRIPTION: "+ Bluecut Fully Loaded Anti Glare Thin Lens",
		TITLE: "Product to Return",
		WE_NEED_YOUR_ADDRESS_TO_PROCEED: "We need your address to proceed",
		PrimaryText: "Continue Shopping",
		TextChat: "Chat With Us",
		PRODUCT_STATUS: "Current Status",
		CANCELLED_TITLE: "Return Cancelled",
	},
	isCurrent: "current",
	isCancelled: "cancelled",
	SHARE_YOUR_FEEDBACK: "Share Your Feedback",
	PRODUCT_IMAGE: "https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//l/i/sky-blue-gunmetal-full-rim-rectangle-lenskart-air-signia-la-e14573-c2-eyeglasses_g_4977_6_13_22.jpg",
	TICK_TEXT: "https://static.lenskart.com/media/mobile/universal/images/returnExchange/TickGreen.svg",
	CROSS_ICON: "https://static1.lenskart.com/media/desktop/img/DesignStudioIcons/CrossWithBorder.svg",
	NEED_HELP: "Need Help?",
	price: {
		currency: CurrencyType.INR,
		basePrice: 1100,
		lkPrice: 0,
		symbol: CurrencySymbol.INR
	},
	framePrice: {
		currency: CurrencyType.INR,
		basePrice: 2000,
		lkPrice: 1100,
		symbol: CurrencySymbol.INR
	},
	status: [
		{
			"id": 1,
			"title": "Pickup Scheduled",
			"date": "7 Feb",
			"icon": <Tick />
		},
		{
			"id": 2,
			"title": "Pickup Cancelled",
			"date": "7 Feb",
			"icon": <Cross />,
		},
		{
			"id": 3,
			"title": "Store Visit Scheduled",
			"date": "7 Feb",
			"icon": <Tick />
		},
		{
			"id": 4,
			"title": "Pickup Cancelled",
			"date": "7 Feb",
			"icon": <Cross />,
		}
	],
	storeAddress: {
		SYMBOL_COLON: ":",
		STORE_ADDRESS: "Store Address",
		INDIA: "India",
		TIMINGS:"Timings",
		WORKING_HOURS:" 10:30 AM - 9:30 PM (All Days)"
	},
	shippingAddress: {
		storeName:"Flagship at Connaught Place",
		addressType: "store",
		addressline1: "K 26, Ground Floor Connaught Place New Delhi, 110001 ",
		addressline2: "Opposite PVR Plaza Cinema",
		phone: "428891059",
		phoneCode: "+91",
		postcode: "110001",
		state: " New Delhi",
		direction:"Direction",
		callStore:"Call Store",
	}
};
