// @lk/ui-library/src/Component/Common/Toast/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";


export const ToastWrapper = styled.div<{
	color?: string,
	height?: string,
	width?: string,
	bottom?: string,
	borderColor?:string,
	borderRadius?:string
}>`
z-index: 999999999999;
 ${props => props.width ? `width:${props.width}` : ""};
height: ${props => props.height ? props.height : "none"};
bottom: ${props => props.bottom ? props.bottom : "100px"};
text-align: center;
font-family: ${TypographyENUM.defaultBook};
font-synthesis: none;
position: fixed;
transition: all 0.4s ease-in-out;
background-color:${props => props.color ? props.color : "orange"};
justify-content: center;
align-items: center;
display: flex;
/* border-radius:4px; */
border-radius: ${props => props.borderRadius || "4px"};
border: ${props => props.borderColor || ""};
right: 50%!important;
transform: translateX(50%);
`;

export const ToastText = styled.span<{
	textColor?: string,
	fontSize?: string,
	fontFamily?: TypographyENUM | string,
	padding?: string,
}>`

color: ${props => props.textColor ? props.textColor : "white"};
padding: ${props => props.padding ? props.padding : "8px 15px"};
width: max-content;
font-size:${props => props.fontSize ? props.fontSize : "12px"};
font-size:${props => props.fontFamily ? `${props.fontFamily}` : `${TypographyENUM.lkSansRegular}`};

`;
