// @lk/ui-library/src/Component/V1/PackageScreenNew/Coating/PackageCoating.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import PackageCoating from "./PackageCoating";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import { ComponentSizeENUM } from "../../../../Types/general";

export default {
	title: "Components/v1/PackageScreenNew/PackageCoating",
	component: PackageCoating,
	argTypes: {
		...generateGeneralArgs(),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		}),
		showIcon: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean to show only icon inside button",
			category: "Icon",
			defaultValue: "false",
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for button",
			category: "Events",
		}),
		isSelected: generateArgTypes({
			type: "boolean",
			required: false,
			description: "Whether the Radio is selected or not",
			category: "Events",
		}),
		title: generateArgTypes({
			type: "string",
			required: false,
			description: "title of radio",
			category: "Component",
		}),
		packageDesc: generateArgTypes({
			type: "string",
			required: false,
			description: "The description of PackageRadio",
			category: "Component",
		}),
		imageUrl: generateArgTypes({
			type: "string",
			required: false,
			description: "Image of PackageRadio",
			category: "Component",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: false,
			description: "show to right to left",
			category: "Status",
		})
	},
} as ComponentMeta<typeof PackageCoating>;

const Template: ComponentStory<typeof PackageCoating> = (args) => <PackageCoating {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "single_vision",
	width: "100",
	componentSize: ComponentSizeENUM.small,
	isSelected: true,
	onClick: () => { console.log("Clicked");},
	title: "Package",
	packageDesc: "Package Description",
	imageUrl: "https://static.lenskart.com/media/desktop/img/pdp/single_vision.png",
	addOnData: [
		{
			"id": "563ca5151314273042bc6266",
			"imageUrl": "https://static.lenskart.com/images/cust_mailer/1-Aug-18/scratch-resistant.jpg",
			"price": {
				"currency": "INR",
				"basePrice": 150,
				"lkPrice": 150,
				"symbol": "₹"
			},
			"title": "Scratch Resistant"
		},
		{
			"id": "563ca5151314273042bc1233",
			// "imageUrl": "https://static.lenskart.com/images/cust_mailer/1-Aug-18/scratch-resistant.jpg",
			"imageUrl": "",
			"price": {
				"currency": "INR",
				"basePrice": 350,
				"lkPrice": 250,
				"symbol": "₹"
			},
			"title": "Anti Fog Coating"
		}
	]
};
