// @lk/ui-library/src/Component/V1/PackageScreenNew/Coating/PackageCoating.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const PackageCoatingContainer = styled.div`
	display: flex;
	flex-direction: column;
	gap: var(--pd-16);
	flex: 1;
	padding: 28px 46px var(--pd-20);
`;

export const PackageCoatingSelection = styled.div<{selected?: boolean}>`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	background-color: var(--white);
	cursor: pointer;
	box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
	border-radius: 12px;
	transition: box-shadow 0.2s ease-in-out;
	padding: 10px;
	border: ${props => props.selected ? "1px solid var(--dark-blue-100)" : ""};
	&:hover {
			box-shadow: 0 3px 16px 0 rgba(0, 0, 0, 0.1);
		}
`;

export const ImageWrapper = styled.div`
	img{
		max-width: 250px;
	}
`;

export const RightSection = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	flex: 1;
	padding: var(--pd-20) 10px var(--pd-20) var(--pd-20);
	svg{
		width:18px;
		height:14px;
	  }
`;

export const ContentWrapper = styled.div`

`;

export const Title = styled.div`
	font-family: ${TypographyENUM.lkSerifNormal};
	font-style: normal;
	font-weight: var(--fw-light);
	font-size: 20px;
	line-height: 30px;
	display: flex;
	letter-spacing: -0.02em;
	color: var(--dark-bluw-100);
`;
export const Price = styled.div`
	font-family: ${TypographyENUM.lkSansBold};
	font-style: normal;
	/* font-weight: var(--fw-bold); */
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: var(--text);
`;

export const SkipCoating = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	gap: 20px;
	background-color: var(--white);
	padding: var(--pd-20);
	box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
	border-radius: 12px;
	cursor: pointer;
	transition: box-shadow 0.2s ease-in-out;
	&:hover {
			box-shadow: 0 3px 16px 0 rgba(0, 0, 0, 0.1);
		}
`;

export const TextWrapper = styled.div`
	flex: 1;
	font-family: ${TypographyENUM.lkSansBold};
	font-style: normal;
	/* font-weight: var(--fw-bold); */
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: var(--text);
`;
