// @lk/ui-library/src/Component/V1/CheckoutMobile/PostCheckoutSection/CommonComponents/AnimatedCheck.tsx
import React from "react";
import styled, { keyframes } from "styled-components";

const strokeAnimation = keyframes`
100% {
    stroke-dashoffset: 0;
  }
`;
const scaleAnimation = keyframes`
0%, 100% {
    transform: none;
  }
  50% {
    transform: scale3d(1.1, 1.1, 1);
  }
`;
const fillAnimation = keyframes`
100% {
    box-shadow: inset 0px 0px 0px 30px #000042;
  }
`;
const borderAnimation = keyframes`
	0% {
		border: 0px solid rgba(0, 0, 66, 0.4);
	}
	100% {
		border: 10px solid rgba(0, 0, 66, 0.4);
	}
`;
const CheckMarkSvg = styled.svg`
  width: 56px;
  height: 56px;
  border-radius: 50%;
  display: block;
  stroke-width: 2;
//   stroke: #fff;
  stroke-miterlimit: 10;
//   margin: 0 auto;
  box-shadow: inset 0px 0px 0px #000042;
  animation: ${fillAnimation} 0.4s ease-in-out 0.4s forwards, ${borderAnimation} 0.4s ease-in-out 1.2s forwards, ${scaleAnimation} 0.3s ease-in-out 0.9s both;

//   border: 10px solid #000;
    box-sizing: content-box;
    outline: none;
`;
const CheckMarkCircle = styled.circle`
	transform-origin: 50% 50%;
	stroke-dasharray: 48;
	stroke-dashoffset: 48;
	animation: ${strokeAnimation} 0.3s cubic-bezier(0.65, 0, 0.45, 1) 0.8s forwards;
`;
const CheckMarkCheck = styled.path`
	transform-origin: 50% 50%;
	stroke-dasharray: 48;
	stroke-dashoffset: 48;
	stroke: #fff;
	animation: ${strokeAnimation} 0.3s cubic-bezier(0.65, 0, 0.45, 1) 0.8s forwards;
`;
const AnimatedCheck = () => {
	return (
		<CheckMarkSvg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 52 52">
			<CheckMarkCircle cx="26" cy="26" r="25" fill="none"/>
			<CheckMarkCheck fill="none" d="M14.1 27.2l7.1 7.2 16.7-16.8"/>
		</CheckMarkSvg>
	);
};

export { AnimatedCheck };
