// @lk/ui-library/src/Component/V1/Header/HeaderNav/HeaderNav.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import HeaderNav from "./index";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
export default {
	title: "Components/v1/HeaderNav",
	component: HeaderNav,
	argTypes: {
		...generateGeneralArgs(),
		id: generateArgTypes({
			type: "number",
			required: true,
			description: "id",
			category: "Component",
		}),
		isVisible: generateArgTypes({
			type: "boolean",
			required: false,
			description: "visible",
			category: "Component",
		}),
		signedInData: generateArgTypes({
			type: "object",
			required: false,
			description: "onClick handler for back press",
			category: "Component",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: false,
			description: "rtl direction",
			category: "Component",
		}),
	},
} as ComponentMeta<typeof HeaderNav>;
const Template: ComponentStory<typeof HeaderNav> = (args) => <HeaderNav {...args}/>;
export const Overview = Template.bind({});

Overview.args = {
	id : 1,	
	isVisible : true,
	signedInData : {
		isSignedIn: true,
		username: "ABC"
	},
	configData : {},
	logoutHandler : () => console.log("logout clicked"),
	dataLocale : {
		MY_ORDERS: "My orders",
		LOGOUT: "Logout",
		CONTACT_US: "Contact Us",
		REVIEWS_AND_RATINGS: "Reviews and Ratings",
		SHOP_INSTAGRAM: "Shop Instagram",
		LENSKART_INTERNATIONAL: "Lenskart International",
		MANAGE_NOTIFICATIONS: "Manage Notifications",
		HI: "Hi", 
		SYMBOL_COMMA: ",", 
		HI_GUEST: "Hi Specsy!",
		DAYS_LEFT_TEXT: "Days left",
		LOGIN_SIGNUP_TEXT: "Login/SignUp",
		OFFER_TEXT_FOR_GOLD: "Get GOLD Membership",
		LOGIN_OR_REGISTER: "Login Or Register",
		GET_GOLD_MEMBERSHIP: "Get Gold Membership"
	},
	userInfo : {},
	goToLoginOrRegister : () => console.log("login/register clicked"),
	isRTL: false
};
