// @lk/ui-library/src/Component/V1/Header/HeaderNav/mockValues3.ts

export const mockData2 = {
	data: [{
		categorylabel: "",
		items: [

			{
				label: "Contact Us",
				link: "/"
			},
			{
				label: "Review & Ratings ",
				link: "/"
			},
			{
				label: "Shop Instagram",
				link: "/"
			},
			{
				label: "Lenskart International ",
				link: "/"
			}

		],
	},
	{
		title: "Highlights",
		items: [

			{
				label: "Check Frame Size",
				link: "/"
			},
			{
				label: "Gold Membership ",
				link: "/"
			},
			{
				label: "Try Frames in 3D",
				link: "/"
			},
			{
				label: "Download App ",
				link: "/"
			}

		],
	},


	{
		"label": "Men",
		"catItems": [
			{
				"dtmParent": "men",
				"dtmCategory": "eyeglasses",
				"routerLink": "/eyeglasses/find-eyewear/men/subcategories",
				"label": "Eyeglasses",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Eyeglasses"
				}
			},
			{
				"dtmParent": "men",
				"dtmCategory": "sunglasses",
				"routerLink": "/sunglasses/find-eyewear/men/subcategories",
				"label": "Sunglasses",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Sunglasses"
				}
			},
			{
				"dtmParent": "men",
				"dtmCategory": "power sunglasses",
				"routerLink": "/men/power-sunglasses.html",
				"label": "Prescription Sunglasses",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Prescription Sunglasses"
				}
			},
			{
				"dtmParent": "men",
				"dtmCategory": "contact-lenses",
				"routerLink": "/contact-lenses.html",
				"label": "Contact Lenses",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Contact Lenses"
				}
			},
			{
				"dtmParent": "men",
				"dtmCategory": "eyeglasses",
				"routerLink": "/zero-power/find-eyewear/men/subcategories",
				"label": "Zero Power Eyeglasses",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Zero Power Eyeglasses"
				}
			},
			{
				"dtmParent": "men",
				"dtmCategory": "latest-collection",
				"routerLink": "/latest-collection/find-eyewear/men/subcategories",
				"label": "Latest Collection",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Latest Collection"
				}
			}
		]
	},

	{
		"label": "Women",
		"catItems": [
			{
				"dtmParent": "women",
				"dtmCategory": "eyeglasses",
				"routerLink": "/eyeglasses/find-eyewear/women/subcategories",
				"label": "Eyeglasses",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Eyeglasses"
				}
			},
			{
				"dtmParent": "women",
				"dtmCategory": "sunglasses",
				"routerLink": "/sunglasses/find-eyewear/women/subcategories",
				"label": "Sunglasses"
			},
			{
				"dtmParent": "women",
				"dtmCategory": "power sunglasses",
				"routerLink": "/women/power-sunglasses.html",
				"label": "Prescription Sunglasses",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Prescription Sunglasses"
				}
			},
			{
				"dtmParent": "women",
				"dtmCategory": "contact-lenses",
				"routerLink": "/contact-lenses.html",
				"label": "Contact Lenses",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Contact Lenses"
				}
			},
			{
				"dtmParent": "women",
				"dtmCategory": "eyeglasses",
				"routerLink": "/zero-power/find-eyewear/women/subcategories",
				"label": "Zero Power Eyeglasses",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Zero Power Eyeglasses"
				}
			},
			{
				"dtmParent": "women",
				"dtmCategory": "latest-collection",
				"routerLink": "/latest-collection/find-eyewear/women/subcategories",
				"label": "Latest Collection",
				"gaObj": {
					"event": "Menu_Clicks",
					"category": "Menu_Clicks",
					"action": "TYPE_MENU",
					"label": "Latest Collection"
				}
			}
		]
	},
	{
		"label": "Kids",
		"catItems": [
			{
				"dtmParent": "kids",
				"dtmCategory": "eyeglasses",
				"routerLink": "/eyeglasses/find-eyewear/kids/subcategories",
				"label": "Eyeglasses"
			},
			{
				"dtmParent": "kids",
				"dtmCategory": "sunglasses",
				"routerLink": "/sunglasses/find-eyewear/kids/subcategories",
				"label": "Sunglasses"
			},
			{
				"dtmParent": "kids",
				"dtmCategory": "eyeglasses",
				"routerLink": "/zero-power/find-eyewear/kids/subcategories",
				"label": "Zero Power Eyeglasses"
			},
			{
				"dtmParent": "kids",
				"dtmCategory": "latest-collection",
				"routerLink": "/latest-collection/find-eyewear/kids/subcategories",
				"label": "Latest Collection"
			}
		]
	},


	]


};

export const menuData: any =

	[
		{
			"categoryTitle": "SHOP",
			"isShopMenu": true,
			"items": [
				{
					"dtmParent": "",
					"dtmCategory": "frame-size-guide",
					"routerLink": "/frame-size-guide",
					"label": "Frame Size Guide",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Frame Size Guide"
					}
				},
				{
					"dtmParent": "",
					"dtmCategory": "eyeglasses",
					"routerLink": "/eyeglasses.html",
					"label": "Eyeglasses",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Eyeglasses"
					}
				},
				{
					"dtmParent": "",
					"dtmCategory": "sunglasses",
					"routerLink": "/sunglasses.html",
					"label": "Sunglasses",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Eyeglasses"
					}
				},
				{
					"dtmParent": "",
					"dtmCategory": "contact-lenses",
					"routerLink": "/contact-lenses.html",
					"label": "Contact Lenses",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Contact Lenses"
					}
				},
				{
					"dtmParent": "",
					"dtmCategory": "power sunglasses",
					"routerLink": "/men/power-sunglasses.html",
					"label": "Power Sunglasses",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Power Sunglasses"
					}
				},
				{
					"label": "Men",
					"catItems": [
						{
							"dtmParent": "men",
							"dtmCategory": "eyeglasses",
							"routerLink": "/eyeglasses/find-eyewear/men/subcategories",
							"label": "Eyeglasses",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Eyeglasses"
							}
						},
						{
							"dtmParent": "men",
							"dtmCategory": "sunglasses",
							"routerLink": "/sunglasses/find-eyewear/men/subcategories",
							"label": "Sunglasses",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Sunglasses"
							}
						},
						{
							"dtmParent": "men",
							"dtmCategory": "power sunglasses",
							"routerLink": "/men/power-sunglasses.html",
							"label": "Prescription Sunglasses",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Prescription Sunglasses"
							}
						},
						{
							"dtmParent": "men",
							"dtmCategory": "contact-lenses",
							"routerLink": "/contact-lenses.html",
							"label": "Contact Lenses",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Contact Lenses"
							}
						},
						{
							"dtmParent": "men",
							"dtmCategory": "eyeglasses",
							"routerLink": "/zero-power/find-eyewear/men/subcategories",
							"label": "Zero Power Eyeglasses",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Zero Power Eyeglasses"
							}
						},
						{
							"dtmParent": "men",
							"dtmCategory": "latest-collection",
							"routerLink": "/latest-collection/find-eyewear/men/subcategories",
							"label": "Latest Collection",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Latest Collection"
							}
						}
					]
				},
				{
					"label": "Women",
					"catItems": [
						{
							"dtmParent": "women",
							"dtmCategory": "eyeglasses",
							"routerLink": "/eyeglasses/find-eyewear/women/subcategories",
							"label": "Eyeglasses",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Eyeglasses"
							}
						},
						{
							"dtmParent": "women",
							"dtmCategory": "sunglasses",
							"routerLink": "/sunglasses/find-eyewear/women/subcategories",
							"label": "Sunglasses"
						},
						{
							"dtmParent": "women",
							"dtmCategory": "power sunglasses",
							"routerLink": "/women/power-sunglasses.html",
							"label": "Prescription Sunglasses",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Prescription Sunglasses"
							}
						},
						{
							"dtmParent": "women",
							"dtmCategory": "contact-lenses",
							"routerLink": "/contact-lenses.html",
							"label": "Contact Lenses",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Contact Lenses"
							}
						},
						{
							"dtmParent": "women",
							"dtmCategory": "eyeglasses",
							"routerLink": "/zero-power/find-eyewear/women/subcategories",
							"label": "Zero Power Eyeglasses",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Zero Power Eyeglasses"
							}
						},
						{
							"dtmParent": "women",
							"dtmCategory": "latest-collection",
							"routerLink": "/latest-collection/find-eyewear/women/subcategories",
							"label": "Latest Collection",
							"gaObj": {
								"event": "Menu_Clicks",
								"category": "Menu_Clicks",
								"action": "TYPE_MENU",
								"label": "Latest Collection"
							}
						}
					]
				},
				{
					"label": "Kids",
					"catItems": [
						{
							"dtmParent": "kids",
							"dtmCategory": "eyeglasses",
							"routerLink": "/eyeglasses/find-eyewear/kids/subcategories",
							"label": "Eyeglasses"
						},
						{
							"dtmParent": "kids",
							"dtmCategory": "sunglasses",
							"routerLink": "/sunglasses/find-eyewear/kids/subcategories",
							"label": "Sunglasses"
						},
						{
							"dtmParent": "kids",
							"dtmCategory": "eyeglasses",
							"routerLink": "/zero-power/find-eyewear/kids/subcategories",
							"label": "Zero Power Eyeglasses"
						},
						{
							"dtmParent": "kids",
							"dtmCategory": "latest-collection",
							"routerLink": "/latest-collection/find-eyewear/kids/subcategories",
							"label": "Latest Collection"
						}
					]
				}
			]
		},
		{
			"categoryTitle": "OUR SERVICES",
			"items": [
				{
					"dtmParent": "services",
					"dtmCategory": "home eye check up",
					"routerLink": "/home-try-on/phone",
					"label": "Home Eye Check-up",
					"hideMenuItem": false,
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Home Eye Check-up"
					}
				},
				{
					"dtmParent": "services",
					"dtmCategory": "loyalty",
					"routerLink": "/loyalty",
					"label": "Lenskart Gold",
					"hideMenuItem": true,
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Lenskart Gold"
					}
				},
				{
					"dtmParent": "services",
					"dtmCategory": "locate stores",
					"href": "https://stores.lenskart.com/",
					"label": "Store Locator",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Store Locator"
					}
				},
				{
					"dtmParent": "services",
					"dtmCategory": "refer and earn",
					"routerLink": "/refer-and-earn-paytm",
					"label": "Refer & Earn",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Refer & Earn"
					}
				}
			]
		},
		{
			"categoryTitle": "FAQ's & POLICIES",
			"items": [
				{
					"dtmParent": "faq and policies",
					"dtmCategory": "frequently asked questions",
					"routerLink": "/gfaq",
					"label": "Frequently Asked Questions",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Frequently Asked Questions"
					}
				},
				{
					"dtmParent": "faq and policies",
					"dtmCategory": "cancellation and returns policy",
					"routerLink": "/cancellation.html",
					"label": "Cancellation & Returns Policy",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Cancellation & Returns Policy"
					}
				},
				{
					"dtmParent": "cobrowsing",
					"dtmCategory": "cobrowsing",
					"routerLink": "/cobrowsing",
					"label": "Cobrowsing",
					"gaObj": {
						"event": "Menu_Clicks",
						"category": "Menu_Clicks",
						"action": "TYPE_MENU",
						"label": "Cobrowsing"
					}
				}
			]
		}
	];

// [
// 	{
// 		"categoryTitle": "HIGHLIGHTS",
// 		"isShopMenu": true,
// 		"items": [
// 			{
// 				"dtmParent": "",
// 				"dtmCategory": "cms-frame-size",
// 				"routerLink": "/cms-frame-size",
// 				"label": "Check Frame Size",
// 				"gaObj": {
// 					"event": "Menu_Clicks",
// 					"category": "Menu_Clicks",
// 					"action": "TYPE_MENU",
// 					"label": "Frame Size Guide"
// 				}
// 			},
// 			{
// 				"dtmParent": "",
// 				"dtmCategory": "loyalty",
// 				"routerLink": "/loyalty",
// 				"label": "Gold Membership",
// 				"gaObj": {
// 					"event": "Menu_Clicks",
// 					"category": "Menu_Clicks",
// 					"action": "TYPE_MENU",
// 					"label": "Gold Membership"
// 				}
// 			},
// 			{
// 				"dtmParent": "",
// 				"dtmCategory": "compare-looks",
// 				"routerLink": "/compare-looks",
// 				"label": "Try Frames in 3D",
// 				"gaObj": {
// 					"event": "Menu_Clicks",
// 					"category": "Menu_Clicks",
// 					"action": "TYPE_MENU",
// 					"label": "Try Frames in 3D"
// 				}
// 			},
// 		]
// 	},
// 	{
// 		"categoryTitle": "SHOP",
// 		"isShopMenu": true,
// 		"items": [
// 			{
// 				"label": "Men",
// 				"catItems": [
// 					{
// 						"dtmParent": "men",
// 						"dtmCategory": "eyeglasses",
// 						"routerLink": "/eyeglasses/marketing/bestsellers-premium-eyeglasses.html",
// 						"label": "John Jacobs Eyeglasses - From SAR 650",
// 						"gaObj": {
// 							"event": "Menu_Clicks",
// 							"category": "Menu_Clicks",
// 							"action": "TYPE_MENU",
// 							"label": "John Jacobs Eyeglasses - From SAR 650"
// 						}
// 					},
// 					{
// 						"dtmParent": "men",
// 						"dtmCategory": "sunglasses",
// 						"routerLink": "/eyeglasses/collections/all-computer-glasses.html",
// 						"label": "Computer Glasses - From SAR 199",
// 						"gaObj": {
// 							"event": "Menu_Clicks",
// 							"category": "Menu_Clicks",
// 							"action": "TYPE_MENU",
// 							"label": "Computer Glasses - From SAR 199"
// 						}
// 					},
// 					{
// 						"dtmParent": "men",
// 						"dtmCategory": "power sunglasses",
// 						"routerLink": "/sunglasses/special/vincent-chase-bestsellers.html",
// 						"label": "Sunglasses - From SAR 199",
// 						"gaObj": {
// 							"event": "Menu_Clicks",
// 							"category": "Menu_Clicks",
// 							"action": "TYPE_MENU",
// 							"label": "Sunglasses - From SAR 199"
// 						}
// 					},
// 				]
// 			},
// 			{
// 				"label": "Women",
// 				"catItems": [
// 					{
// 						"dtmParent": "women",
// 						"dtmCategory": "eyeglasses",
// 						"routerLink": "/eyeglasses/marketing/bestsellers-premium-eyeglasses.html",
// 						"label": "John Jacobs Eyeglasses - From SAR 650",
// 						"gaObj": {
// 							"event": "Menu_Clicks",
// 							"category": "Menu_Clicks",
// 							"action": "TYPE_MENU",
// 							"label": "John Jacobs Eyeglasses - From SAR 650"
// 						}
// 					},
// 					{
// 						"dtmParent": "women",
// 						"dtmCategory": "Computer Glasses - From SAR 199",
// 						"routerLink": "/eyeglasses/collections/all-computer-glasses.html",
// 						"label": "Computer Glasses - From SAR 199"
// 					},
// 					{
// 						"dtmParent": "women",
// 						"dtmCategory": "power sunglasses",
// 						"routerLink": "/sunglasses/special/vincent-chase-bestsellers.html",
// 						"gaObj": {
// 							"event": "Menu_Clicks",
// 							"category": "Menu_Clicks",
// 							"action": "TYPE_MENU",
// 							"label": "Sunglasses - From SAR 199"
// 						}
// 					},
// 				]
// 			},
// 			{
// 				"label": "Kids",
// 				"catItems": [
// 					{
// 						"dtmParent": "kids",
// 						"dtmCategory": "eyeglasses",
// 						"routerLink": "/eyeglasses/promotions/all-kids-eyeglasses.html",
// 						"label": "With Power Lenses - From SAR 149"
// 					},
// 					{
// 						"dtmParent": "kids",
// 						"dtmCategory": "sunglasses",
// 						"routerLink": "/sa/sunglasses.html",
// 						"label": "Sunglasses - From SAR 149"
// 					},
// 				]
// 			}
// 		]
// 	},

// 	{
// 		"categoryTitle": "FAQ's & POLICIES",
// 		"items": [
// 			{
// 				"dtmParent": "faq and policies",
// 				"dtmCategory": "cancellation and returns policy",
// 				"routerLink": "/cancellation.html",
// 				"label": "Cancellation & Returns Policy",
// 				"gaObj": {
// 					"event": "Menu_Clicks",
// 					"category": "Menu_Clicks",
// 					"action": "TYPE_MENU",
// 					"label": "Cancellation & Returns Policy"
// 				}
// 			},
// 			{
// 				"dtmParent": "about-us",
// 				"dtmCategory": "about-us",
// 				"routerLink": "/about-us.html",
// 				"label": "About Us",
// 				"gaObj": {
// 					"event": "Menu_Clicks",
// 					"category": "Menu_Clicks",
// 					"action": "TYPE_MENU",
// 					"label": "About Us"
// 				}
// 			}
// 		]
// 	}
// ];
