// @lk/ui-library/src/Component/V1/Header/HeaderNav/styles.tsx
import styled from "styled-components";

export const NavigationWrapper = styled.div`
position: fixed;
left: 0;
top: 48px;
width: 100%;
/* height: calc(100% - 48px); */
overflow: hidden;
z-index: 105;
direction: ltr;
`;

export const NavigationContainer = styled.div<{innerHeight?: number}>`
position: relative;
overflow-y: scroll;
height: ${props => props?.innerHeight ? `calc(${props.innerHeight}px - 48px)` : "calc(100vh - 48px)"} ;
overflow-x:hidden;
&::-webkit-scrollbar {
	display: none;
  }
`;
export const NavCollaspeContainer = styled.div`
transition-duration: 1s;
    transition-timing-function: ease-in;
    max-height: 10000px;
`;

export const NavMenuContainer = styled.div`
background: #27394e;
    color: #fff;
    padding: 15px 13px 10px;
    font: 500 14px/20px Roboto,Arial,Helvetica,sans-serif;
`;

export const NavUserContainer = styled.div<{isRTL?: boolean,}>`
display: flex;
flex-direction: row;
gap: 15px;
align-items: center;
width: 100%;
flex-direction: ${props => !props.isRTL ? "row" : "row-reverse"};
`;

export const NavUserContainerWrapper = styled.div<{isRTL?: boolean}>`
border-bottom: 2px solid #18cfa8;
padding-bottom: 8px;
`;

export const NavUser = styled.div`

display: flex;
flex-direction:column;
`;
export const NavAvatar = styled.img`
`;

export const NavUserInfo = styled.div`
display: flex;
flex-direction:column;
`;


export const NavUserText = styled.div<{isRTL?: boolean}>`
font-weight: 500;
font-size: 16px;
margin-bottom: 5px;
${props => props.isRTL ? "direction: rtl" : ""};
`;
export const NavUserBelowText = styled.div `
line-height: 18px;
font-size: 13px;
color: #ccc;
`;
export const NavUserActionBtn = styled.a `
width: 100%;
line-height: 32px;
font-size: 12px;
text-transform: uppercase;
border-radius: 4px;
text-align: center;
background-color: #00bac6;
margin-top: 10px;
border-bottom: 2px solid #18cfa8;
width:100%;
`;

export const NavUnitContainer = styled.div`
padding: 0 0 64px;

`;

export const NavItem = styled.a<{isRTL?: boolean}>`
display: flex;
flex-direction: ${props => !props.isRTL ? "row" : "row-reverse"};
align-items: center;
justify-content: space-between;
padding-right: ${props => !props.isRTL ? "15px" : ""};
padding-left: ${props => props.isRTL ? "15px" : ""};
/* width: 100%; */
border-bottom: 1px solid rgba(255,255,255,.2);

`;

export const NavLink = styled.span<{isRTL: boolean}>`
display: block;
    position: relative;
    margin: -1px -13px;
    padding: 16px ${props => props.isRTL ? "16px" : "40px"} 15px 15px;
    color: #fff;
/* &:after{
  content: ">";
  position: absolute;
  top: 50%;
  right: 15px;
  margin: -10px 0 0;
} */


`;

export const ArrowRight = styled.div`
`;

export const NavTitle = styled.div<{isRTL?: boolean}>`
font-weight: 700;
text-transform:uppercase;
display: flex;
flex-direction: ${props => !props.isRTL ? "row" : "row-reverse"};
`;

export const AccordionWrapper = styled.div`
// font: 500 14px/20px Roboto,Arial,Helvetica,sans-serif;
`;

export const TierWrapper = styled.div`
margin-top:2px;
`;
export const TierName = styled.span `
line-height: 24px;
padding: 6px 12px;
text-transform: uppercase;
border-radius: 8px;
background-color: #ffb400;
color: #333;
font-size:10px;
`;

export const DaysWrapper = styled.div `
margin-top: 10px;
display: flex;
`;
export const DaysContainer = styled.div `
margin-right: 25px;
`;
export const DaysText = styled.p `
font-size: 12px;
margin-bottom: 5px;
color: #ccc;
`;

export const DayCount = styled.div `
line-height: 18px;
font-weight: 700;
font-size: 18px;
`;
