// @lk/ui-library/src/Component/V1/PDP/Mobile/LensSelection/LensSelection.styles.tsx
import styled from "styled-components";

export const LensSelectionContainer = styled.div`
	display: flex;
	flex-direction: column;
	/* text-align: center; */
	width: 100%;
	background-color: #eaeff4;
	gap: 10px;
`;

export const Title = styled.div`
	font-weight: 500;
	font-family: sans-serif;
	display: flex;
	align-items: center;
	justify-content: center;
`;

export const LensOptionsContainer = styled.div`
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	justify-content: center;
	gap: 10px;
	/* padding: 0 15px; */
`;

export const LensSelectionCard = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
	background-color: var(--white);
	width: 100%;
	padding: 10px;
	gap: 20px;
`;

export const ImageContainer = styled.div`
	img{
		width: 13.2vw;
	}
`;

export const TextContainer = styled.div`
	width: 64vw;
`;

export const Text = styled.div<{type?:string}>`
	font-size: ${props => props.type === "primary" ? "14px" : ""};
	color: ${props => props.type === "primary" ? "#27394e" : "#99a0a"};
	font-weight: ${props => props.type === "primary" ? "700" : "400"};
`;
export const IconContainer = styled.div`
	font-size: 30px;
	color: #18cfa8;
	font-weight: 700;
`;

export const PriceContainer = styled.div`
	background-color: var(--white);
	display: flex;
	align-items: center;
	justify-content: space-between;
	padding: 10px;
`;

export const LeftSection = styled.div`
	display: flex;
	flex-direction: column;
`;
export const BuyNowContainer = styled.div`
	background-color: #eaeff4;
	color: var(--white);
	display: flex;
	flex: 0.6;
	align-items: center;
	justify-content: center;
	border-radius: 4px;
	letter-spacing: 1px;
	font-weight: 700;
	font-size: 18px;
	padding: 5px 0;
	text-transform: uppercase;
`;
