// @lk/ui-library/src/Component/V1/PDP/Mobile/LensSelection/LensSelection.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import { prescriptionType } from "../../../PackageRadioGroup/mockValue";
import LensSelection from "./LensSelection";

export default {
	title:"Components/V1/PDP/Mobile/LensSelection",
	component: LensSelection,
	argTypes:{
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "radio",
			options: TypographyENUM,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left ",
			category: "Status",
			defaultValue: "false",
		}),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		}),
		selected: generateArgTypes({
			type: "boolean",
			required: false,
			description: "show checkbox",
			category: "Component",
			defaultValue: true,
		}),
		onClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for button",
			category: "Events",
		}),
	}
} as ComponentMeta<typeof LensSelection>;

const Template: ComponentStory<typeof LensSelection> = args => <LensSelection {...args}/>;

export const Overview = Template.bind({});
const productLensTypeDescSample = {
	"Single Vision": {
		"desc": "For distance or near vision <br/>(Thin, anti-glare, blue-cut options)",
		"image": true
	},
	"Bifocal/Progressive": {
		"desc": "Bifocal and Progressives <br/>(For two powers in same lens)",
		"image": true
	},
	"Zero Power": {
		"desc": "For computer smartphones & fashion <br/>(Anti-glare and blue-cut options)",
		"image": true
	},
	"Frame Only": {
		"desc": "Buy only the frame without any lenses",
		"icon": "icon-without_power"
	},
	"Tinted Single Vision": {
		"desc": "(Photochromatic, Grey, Brown, Dark Green Color Options)<br/>Available in Single Vision Only",
		"image": true
	}
};
Overview.args = {
	title: "Choose Lens Type",
	prescriptionType: prescriptionType,
	productLensTypeDesc: productLensTypeDescSample,
	category: "Eyewear",
	price: {
		"currency": "INR",
		"basePrice": 1199,
		"lkPrice": 1199,
		"symbol": "₹"
	},
	buyWithCallConfig: {
		buyonchat: true,
		buyonchat_link: "https://api.whatsapp.com/send?phone=918929853854&text=Hi Lenskart, help me get started!",
		contact_lens: {
			tel: "8470007367"
		},
		cta: {
			icon: "https://static.lenskart.com/media/desktop/img/09-jan-20/call_icon.png",
			iconGreen: "https://static.lenskart.com/media/mobile/images/call_icon_green.png",
			isShown: true,
			text: "Help?",
			whatsappIcon: "https://static1.lenskart.com/media/mobile/images/whatsapp-smallicon.png",
			whatsappIconGreen: "https://static.lenskart.com/media/mobile/images/whatsapp-smallicon-green.png"
		},
		description: "Tap to whatsapp our eyewear expert",
		enable_new_flow: true,
		enabled: true,
		eyeglasses: {
			tel: "18001020767"
		},
		floating_image: "https://static1.lenskart.com/media/desktop/img/buyonchat1.png",
		icon: "https://static1.lenskart.com/media/mobile/images/whatsapp-gogreen.png",
		pdpbuyonchat_icon: "https://static1.lenskart.com/media/mobile/images/small-whatsapp.png",
		pdpbuyonchat_text: "Buy on Chat",
		sunglasses: {
			tel: "18001020767"
		},
		title: "Buy on Chat"
	}
};
