// @lk/ui-library/src/Component/V1/AddressInputField/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const FieldInpu = styled.input`
	width: 100%;
	height: 100%;
	border: none;
	outline: none;
	background: transparent;
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;
export const LeftPositioned = styled.div`
	// position: absolute;
	left: 0;
	// width: 48px;
	width: fit-content;
	// top: 25%;
`;

export const Label = styled.label<{isRTL: boolean; prefixText: boolean, left: number}>`
	// font-family: 'Lenskart Sans';
	font-style: normal;
	font-weight: 400;
	    pointer-events: none;
	font-size: 16px;
	line-height: 24px;
	letter-spacing: -0.02em;
	color: #333368;
	margin-bottom: 8px;
	position: absolute;
    top: 17px;
    z-index: 1;
	left: ${props => props.left}px;
`;

export const Box = styled.div<{font: TypographyENUM, value: boolean}>`
  font-family: ${props => props.font};
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	// width: fit-content;
	width: 100%;
	position: relative;

	input::placeholder {
		font-family: ${props => props.font};
	}

	&:focus-within {
		${Label} {
		transform: translate(0,-1.4em);
		margin-bottom: 5px;
		left: 16px;
		font-size:0.8em;
		font-weight:var(--fw-regular);
		}
	}

	${props => props.value && `${Label}{
		transform: translate(0,-1.4em);
		font-size:0.8em;
		left: 16px;
		margin-bottom: 5px;
		font-weight:var(--fw-regular);
	}`}
`;

export const SubText = styled.div<{isError?: boolean}>`
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 20px;
	letter-spacing: -0.02em;
	color: ${props => props.isError ? "#d90502" : "#66668E"};
	margin-top: 8px;
`;
export const SubTextSpan = styled.span`
	font-style: normal;
	font-weight: 400;
	font-size: 14px;
	line-height: 20px;
	letter-spacing: -0.02em;
	color: #ec3331;
	margin-top: 8px;
`;

export const CrossSection = styled.div`
	position: absolute;
	right: 10px;
	cursor: pointer;
	height: 12px;
	width: 12px;
	display: flex;
    align-items: center;
    justify-content: center;
    top: 40%;
`;

export const FieldContainer = styled.div<{fullWidth?: boolean, line?: number, disabled?: boolean, isRTL?: boolean, prefixText: boolean, label: boolean}>`
	display: flex;
	flex-direction: row;
	align-items: center;
	// padding: 6px 0px;
	padding: 16px;
	gap: 8px;
	width: ${props => props.fullWidth ? "100%" : "431px"};
	height: ${props => props.line ? `${props.line * 56}px` : "56px"};
	background: ${props => props.disabled ? "rgb(227, 227, 227)" : "#FFFFFF"};
	border: 1px solid #E2E2EE;
	box-shadow: 0px 1px 2px rgba(16, 24, 40, 0.05);
	border-radius: 12px;
	flex: none;
	align-self: stretch;
	flex-grow: 0;
	position: relative;
	${props => props.label && "padding-top: 23px;padding-bottom: 10px;"}


	${props => props.isRTL ? "direction: ltr;" : ""};

`;
