// @lk/ui-library/src/Component/V1/Header/MainHeader/StoreLayout/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const StoreLayoutHolder = styled.div`
	display: flex;
	height: 100%;
	width: 100%;
`;
export const StoreMainContainer = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	margin-left: 20px;
    width: 100%;
	justify-content: center;
`;

export const StoreHead = styled.h3`
	font-family: ${TypographyENUM.lkSansRegular};
	font-style: normal;
	font-weight: var(--fw-regular);
	line-height: 48px;
	text-align: center;
	letter-spacing: -.02em;
	color: var(--dark-blue-100);
	font-size: 40px;
	line-height: 84px;
	margin-top: 20px;
    margin-bottom: 10px
`;

export const StoreContent = styled.p`
	font-family: ${TypographyENUM.lkSansHairline};
	font-style: normal;
	font-size: 20px;
	line-height: 36px;
	text-align: center;
	letter-spacing: -.02em;
	color: var(--dark-blue-100);
	margin-bottom: 15px;
`;

export const StoreButton = styled.a`
	margin-top: 10px;
	padding: 16px 22px;
    width: auto;
    height: 56px;
    border: 1px solid var(--dark-blue-100);
    border-radius: 28px;
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-size: 18px;
    line-height: 140%;
    letter-spacing: -.02em;
    color: var(--dark-blue-100);
    text-align: center;
    background: #fff;
    outline: none;
    cursor: var(--pointer);
	/* font-weight: 600; */
	margin-right: 20px;
	display: inline-block;
    margin-bottom: 0;
	text-decoration: none;
	&:hover{
		background-color: var(--dark-blue-100);
		color: #fff;
	}
`;

export const StoreImageContainer = styled.a`
	display: block;
`;

export const StoreImage = styled.img`
	position: relative;
	min-width: 700px;
	height: 425px;
`;

export const StoreImageGroup = styled.div`
	display: flex;
	align-items: center;
	justify-content: space-evenly;
	margin: 0 20px;
`;
export const CityContainer = styled.a`
	display: flex;
	flex-direction: column;
	margin: 0 10px;
`;
export const CityImage = styled.img`
	display: block;
	margin: 0;
	padding: 0;
	height: 79px;
`;
export const CityName = styled.div`
	color: var(--dark-blue-100);
	display: block;
	text-align: center;
	text-transform: capitalize;
	font-size: var(--fs-14);
`;
