// @lk/ui-library/src/Component/V1/PDP/TechnicalInformation/TechnicalInfo.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";
import { TechnicalInfoData } from "./stories-data";
import { TechnicalInfo } from "./TechnicalInfo.component";

export default {
	title:"Components/V1/PDP/TechnicalInfo",
	component: TechnicalInfo,
	argTypes:{
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "radio",
			options: TypographyENUM,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left ",
			category: "Status",
			defaultValue: "false",
		})
	}
} as ComponentMeta<typeof TechnicalInfo>;

const Template: ComponentStory<typeof TechnicalInfo> = args => <TechnicalInfo {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"tech-info-card",
	data: TechnicalInfoData(),
	font:TypographyENUM.defaultBook,
};
