// @lk/ui-library/src/Component/V1/PDP/TechnicalInformation/TechnicalInfo.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const InfoWrapper = styled.div<{styleFont:TypographyENUM, isRTL: boolean}>`
	display:flex;
	flex-direction:column;
	align-items: ${ props => props.isRTL ? "end" : "start"};
	font-family:${props => props.styleFont};
`;

export const TechInfoLine = styled.div<{isRTL: boolean}>`
	display: flex;
	justify-content: flex-start;
	margin-bottom: var(--spacing-sm);
	flex-direction: ${props => props.isRTL ? "row-reverse" : ""};
`;

export const TechInfoKey = styled.span<{isRTL: boolean}>`
	margin:0;
	padding-left: ${ props => props.isRTL ? "var(--pd-8)" : "0"};
	text-align: start;
	min-width: ${ props => props.isRTL ? "" : "135px"};
	color: var(--gray);
	font-size: var(--fs-13);
	font-family: ${TypographyENUM.defaultLight};
`;

export const TechInfoVal = styled.span<{isRTL: boolean}>`
	margin:0;
	padding-left: ${ props => props.isRTL ? "0" : "var(--pd-8)"};
	padding-right: ${ props => props.isRTL ? "var(--pd-50)" : "0"};
	font-weight: var(--fw-medium);
	font-family: ${TypographyENUM.defaultBook};
	font-synthesis: none;
	color: var(--black);
	font-size: var(--fs-13);
	min-width: ${ props => props.isRTL ? "135px" : ""};
	text-align: ${ props => props.isRTL ? "end" : ""};
`;

export const TechInfoLink = styled.div`
	color: var(--irish-blue);
	cursor: var(--pointer);
	font-size: var(--fs-13);
	margin-bottom: var(--spacing-sm);
`;

export const QuestionMarkLink = styled.a`
	margin-left: var(--spacing-sm);
	width: 12px;
	height: 12px;
	font-size: var(--fs-10);
	cursor: var(--pointer);
	background-color: var(--black);
	color: var(--white);
	border-radius: 50%;
	display: inline-flex;
	align-items: center;
	justify-content: center;
	transform: translateY(-1px);
`;
