// @lk/ui-library/src/Component/V1/StudioFlow/NearestStore/NearestStore.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { NearestStore } from "./NearestStore.component";

export default {
	title: "Components/v1/StudioFlow/NearestStore",
	component: NearestStore,
	argTypes: {
		isSelected: generateArgTypes({
			type: "boolean",
			category: "Component",
			required: true,
			description: "Tells whether the component is selected or not",
			defaultValue: false
		}),
		store: generateArgTypes({
			type: "object",
			category: "Component",
			required: true,
			description: "Store object which is used to render data in the object",
			defaultValue: {
				"code": "LKST401",
				"storeType": "LENSKART",
				"name": "Funan",
				"address": "Basement 1, #B1-08/09, 107, North Bridge Road, Funan, Funan, Opposite Supreme Court, Downtown, Singapore, 179105",
				"openingTime": "12:00 PM",
				"closingTime": "09:00 PM",
				"area": "Funan",
				"landmark": "Opposite Supreme Court",
				"city": "Downtown",
				"state": "Singapore",
				"country": "Singapore",
				"telephone": "69701335",
				"phoneCode": "+65",
				"longitude": "103.850301",
				"latitude": "1.291066",
				"url": "https://stores.lenskart.com/lenskart-com-Funan-Funan-Singapore-391092/Home#book",
				"shortUrl": "http://lnskt.co/LX65uE4E",
				"googleUrl": "https://maps.google.com/maps?cid=10076176020890546085",
				"pincode": "179105",
				"email": "funan@lenskart.sg",
				"isAvailable": true,
				"geoDistance": 33.54
			}
		})
	},
} as ComponentMeta<typeof NearestStore>;

const Template: ComponentStory<typeof NearestStore> = (args) => (
	<NearestStore {...args} />
);
export const Overview = Template.bind({});

Overview.args = {
	isSelected: false,
	store: {
		"code": "LKST401",
		"storeType": "LENSKART",
		"name": "Funan",
		"address": "Basement 1, #B1-08/09, 107, North Bridge Road, Funan, Funan, Opposite Supreme Court, Downtown, Singapore, 179105",
		"openingTime": "12:00 PM",
		"closingTime": "09:00 PM",
		"area": "Funan",
		"landmark": "Opposite Supreme Court",
		"city": "Downtown",
		"state": "Singapore",
		"country": "Singapore",
		"telephone": "69701335",
		"phoneCode": "+65",
		"longitude": "103.850301",
		"latitude": "1.291066",
		"url": "https://stores.lenskart.com/lenskart-com-Funan-Funan-Singapore-391092/Home#book",
		"shortUrl": "http://lnskt.co/LX65uE4E",
		"googleUrl": "https://maps.google.com/maps?cid=10076176020890546085",
		"pincode": "179105",
		"email": "funan@lenskart.sg",
		"isAvailable": true,
		"geoDistance": 33.54
	}
};
