// @lk/ui-library/src/Component/V1/StudioFlow/NearestStore/NearestStore.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const NearestStoreWrapper = styled.div<{ isSelected: boolean; }>`
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
  border-radius: 12px;
  padding: var(--pd-16);
  cursor: pointer;
`;

export const TopSection = styled.div`
  display: flex;
  align-items: flex-start;
  gap: 12px;
`;

export const ImageContainer = styled.div`
  max-width: 80px;
  width: 100%;
  max-height: 80px;
  height: 100%;

`;

export const Image = styled.img`
  width: 100%;
  object-fit: cover;
`;

export const DetailsSection = styled.div`
  display: flex;
  flex-direction: column;
  gap: 12px;
  width: 100%;
`;

export const NameAndCheckbox = styled.div`
  display: flex;
  align-items: flex-start;
  justify-content: space-between;

`;

export const Name = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-weight: 700;
  font-size: var(--fs-16);
  line-height: 24px;
  display: flex;
  align-items: flex-end;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;

export const AddressSection = styled.div`
  max-width: 296px;
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: var(--fs-16);
  line-height: 24px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
`;

export const BottomSection = styled.div`
  border-top: 1px dashed var(--light-purple-10);
  margin-top: var(--mr-12);
  padding-top: var(--pd-12);
`;

export const Distance = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: var(--fs-14);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--text);
`;
