// @lk/ui-library/src/Component/V1/StudioFlow/RadioStore/RadioStore.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { RadioStore } from "./RadioStore.component";

export default {
	title: "Components/v1/StudioFlow/Mobile/RadioStore",
	component: RadioStore,
	argTypes: {
		selected: generateArgTypes({
			type: "boolean",
			category: "General",
			required: true,
			description: "Tells if the Radio button is selected or not",
		}),
		onClick: generateArgTypes({
			type: "function",
			category: "General",
			required: true,
			description: "Click Event handler of the Radio Card"
		})
	},
} as ComponentMeta<typeof RadioStore>;

const Template: ComponentStory<typeof RadioStore> = (args) => <RadioStore {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	localeData: {
		CALL_STORE: "Call store",
		VIEW_ON_MAP: "View on map"
	},
	onClick: (store) => console.log({ store }),
	selected: false,
	store: {
		"code": "LKST401",
		"storeType": "LENSKART",
		"name": "Funan",
		"address": "Basement 1, #B1-08/09, 107, North Bridge Road, Funan, Funan, Opposite Supreme Court, Downtown, Singapore, 179105",
		"openingTime": "12:00 PM",
		"closingTime": "09:00 PM",
		"area": "Funan",
		"landmark": "Opposite Supreme Court",
		"city": "Downtown",
		"state": "Singapore",
		"country": "Singapore",
		"telephone": "69701335",
		"phoneCode": "+65",
		"longitude": "103.850301",
		"latitude": "1.291066",
		"url": "https://stores.lenskart.com/lenskart-com-Funan-Funan-Singapore-391092/Home#book",
		"shortUrl": "http://lnskt.co/LX65uE4E",
		"googleUrl": "https://maps.google.com/maps?cid=10076176020890546085",
		"pincode": "179105",
		"email": "funan@lenskart.sg"
	}
};
