// @lk/ui-library/src/Component/V1/StudioFlow/RadioStore/RadioStore.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const RadioCardWrapper = styled.div<{ isSelected: boolean; }>`
  /* background: ${({ isSelected }) => isSelected ? "var(--ghost-white)" : "var(--white)"};
  border: 1px solid ${({ isSelected }) => isSelected ? "var(--dark-blue-50)" : "var(--white)"}; */
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
  border-radius: var(--border-radius-xs);
  padding: var(--pd-16);
`;

export const NameStateContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding-bottom: 4px;
`;

export const Name = styled.div`
  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  font-weight: 700;
  font-size: var(--fs-12);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
`;

export const IconContainer = styled.div``;

export const StoreDetails = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-12);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--text);
  padding-bottom: 4px;

`;

export const HiddenPart = styled.div<{ isSelected: boolean; }>`
`;

export const Address = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 500;
  font-size: var(--fs-12);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--text);
  padding-bottom: 12px;
`;

export const DashedLine = styled.div`
  border-bottom: 1px dashed var(--light-purple-10);
  margin-bottom: 12px;
`;

export const ActionContainer = styled.div`
  display: flex;
  align-items: center;
  gap: 16px;
`;

export const StoreActionButton = styled.a`
  all: unset;
  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  font-weight: 700;
  font-size: var(--fs-12);
  line-height: 20px;
  display: flex;
  align-items: center;
  text-align: center;
  letter-spacing: -0.02em;
  text-decoration-line: underline;
  color: var(--dark-blue-100);
`;
