// @lk/ui-library/src/Component/Common/Panel/Panel.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";
interface Props {
  expand: boolean;
  styledFont?: TypographyENUM;
  isRTL:boolean;
}

export const PanelWrapper = styled.div`
  /* margin-top: var(--mg-xs); */
  padding: var(--pd-xs);
`;

export const TitleBox = styled.div<Props>`
	cursor: pointer;
	padding: 15px 10px 10px 15px;
	background-color: ${props => props.expand ? "var(--irish-blue)" : "var(--warm-grey-100)"};
	border-radius: ${props => props.expand ? "10px 10px 0px 0px" : "10px 10px 10px 10px"};
	display: flex;
	flex-direction: ${props => props.isRTL ? "row-reverse": ""};
  	justify-content: space-between;
  	height: auto;
  	color: ${props =>
		props.expand ? "var(--white)" : "var(--black)"};
  	font-weight: var(--fw-md);
  	align-items: center;
  	font-family: ${props => props.styledFont};
  	text-transform: uppercase;
`;

export const TitleWrapper = styled.div<{isRTL:boolean}>`
	display: flex;
	width: 100%;
	flex-direction: column;
	// border: 2px solid black;
	justify-content: flex-end;
	align-items: ${ props => props.isRTL ? "flex-end" : "" };
`;
export const Title = styled.span`
//   width: calc(100% - 20px);
  display: flex;
  align-items: center;
`;

export const Subtitle = styled.span`
  color: #4caf50;
  font-size: 16px;
  letter-spacing: 0.15px;
  margin-top: 5px;
`;

export const Content = styled.div<{isRTL:boolean}>`
  display: flex;
  /* padding: 0px 15px 15px 15px; */
  background: var(--grey-100);
  border-radius: 0px 0px 10px 10px;
  border: 1px solid #eeeeee;
  padding: 15px;
  flex-direction: ${ props => props.isRTL ? "row-reverse" : ""};
`;

export const ImageWrapper = styled.div`
	margin: 4px 6px 2px 2px;
	width: 70px;
	img {
		width: 100%;
	}
`;
