// @lk/ui-library/src/Component/Common/Panel/Panel.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Panel from "./Panel.component";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
// import { ComponentSizeENUM, TypographyENUM } from "../../../Types/general";

export default {
	title: "Components/Common/Panel",
	component: Panel,
	argTypes: {
		children: generateArgTypes({
			type: "node",
			required: false,
			description: "Children Component",
			category: ""
		}),
		title: generateArgTypes({
			type: "string",
			required: true,
			description: "Title of Panel",
			category: "",
		}),
		expand: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Expand ? True : False",
			category: "",
		}),
		src: generateArgTypes({
			type: "string",
			required: false,
			description: "Image Source",
			category: "",
		}),
		subtitle: generateArgTypes({
			type: "string",
			required: false,
			description: "Subtitle of Panel",
			category: "",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Flips component to read from Right to Left",
			category: "",
		}),
		onTitleClick: generateArgTypes({
			type: "function",
			required: true,
			description: "handles expand function for panel",
			category: "",
		})
	},
} as ComponentMeta<typeof Panel>;

const Template: ComponentStory<typeof Panel> = (args) => <Panel {...args}/>;

export const Overview = Template.bind({});
Overview.args = {
	children: <span>Child Elements</span>,
	title: "Title",
	expand: true,
	src: "",
	subtitle: "Subtitle",
	isRTL: false,
};

export const expandFalse = Template.bind({});
expandFalse.args = {
	...Overview.args,
	subtitle: "Expand False",
	expand: false,
};

export const imgSrc = Template.bind({});
imgSrc.args = {
	...Overview.args,
	title: "Image Version",
	src: "https://1000logos.net/wp-content/uploads/2022/10/Lenskart-Logo.png",
};

export const imgSrcExpandFalse = Template.bind({});
imgSrcExpandFalse.args = {
	...imgSrc.args,
	subtitle: "Expand False",
	expand: false
};

export const rtlEnable = Template.bind({});
rtlEnable.args = {
	...imgSrc.args,
	subtitle: "Expand False",
	isRTL: true,
};
