// @lk/ui-library/src/Component/V1/StudioFlow/BookAppointment/Desktop/styles.tsx
import styled from "styled-components";

export const BookAppointmentWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  background-color: #fbf9f7;
  /* padding: 50px 175px; */
  padding: 50px 12% 60px 12%;
  gap: 40px;
  height: 100%;
  width: 100%;
  min-height:calc(100vh - 65px);
  font-synthesis: none;
`;
export const Heading = styled.div`
  font-style: normal;
  font-weight: 400;
  font-size: 24px;
  line-height: 36px;
  letter-spacing: -0.02em;
  color: #000042;
  align-self: flex-start;
  text-transform: capitalize;
`;

export const ErrorBlock = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  align-self: center;
  padding: 8px 12px;
  gap: 12px;
  width: 328px;
  height: 48px;
  background: #ffebea;
  border: 1px solid #e34934;
  border-radius: 8px;
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 500;
  font-size: 10px;
  line-height: 16px;
  letter-spacing: -0.02em;
  color: #000042;
`;

export const AppointmentCalendarContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  gap: 24px;
  max-width: 720px;
  background: #ffffff;
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
  border-radius: 12px;
  overflow: hidden;
  padding: 20px 0;
  /* padding: 30px 0; */
`;

export const StoreNameContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: space-between;
  gap: 5px;
  /* padding: 20px 20px 0 20px; */
  padding: 0 20px;
`;
export const StoreName = styled.div<{
  alert?: boolean;
}>`
  font-family: var(--font-lksans-bold);
  font-style: normal;
  font-weight: 700;
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.02em;
  /* color: var(--dark-blue-100); */
  color: ${(props) => (props.alert ? "red" : "var(--dark-blue-100)")};
  display: flex;
  gap: 10px;
  width:100%;
`;

export const StoreIcon = styled.img``;

export const StoreAddress = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
  width: 200px;
`;

export const StoreDateSelector = styled.div`
  background: #fbf9f7;
  width: 100%;
`;
export const DateWrapperInner = styled.div`
  display: flex;
  align-items: flex-start;
  gap: 10px;
  width: 100%;
  overflow-x: scroll;
  align-items:center;
`;

export const DateHeading = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 28px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  padding:20px 0 10px 0px;
`;
export const ChooseDateHeading = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 400;
  font-size: 18px;
  line-height: 28px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);
  padding:20px 0 10px 20px;
`;

export const DatesWrapper = styled.div`
  padding: 0 20px;
  display: flex;
  flex-direction: column;
  gap: 10px;
`;

export const DateWrapper = styled.div`
  display: flex;
  flex-direction: column;
`;
export const DateItem = styled.div<{ isSelected?: any }>`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 8px 16px 8px 12px;
  gap: 4px;
  width: 72px;
  height: 72px;
  text-transform: capitalize;
  border: 1px solid;
  border-radius: 12px;
  background: ${(props) => (props.isSelected ? "#F5F5FF" : "#FFFFFF")};
  border-color: ${(props) => (props.isSelected ? "#000042" : "#CECEDF")};
  cursor: pointer;
`;

export const DateDate = styled.div``;

export const DateDay = styled.div``;

export const DateSubHeading = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
`;

export const DateContainer = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
  gap: 10px;
  /* width: 528px; */ //! check later
  overflow: scroll;
//   padding-right: 75px;
  //* to hide scroll bars
  ::-webkit-scrollbar {
    width: 0px;
    background: none;
  }
  ::-webkit-scrollbar-track {
    width: 0;
    background: none;
  }
`;

export const StoreTimeSelector = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  padding: 0 20px;
  gap: 10px;
  width: 100%;
`;

export const LoaderContainer = styled.div``;

export const TimeWrapper = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
`;

export const TimeContainer = styled.div`
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  gap: 15px;
`;

export const TimeItem = styled.div<{
  isTimeSelected: any;
  isTimeAvaialble: any;
}>`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 8px 12px;
  gap: 2px;
  border: 1px solid;
  border-radius: 99px;
  background: ${(props) => (props.isTimeSelected ? "#F5F5FF" : "#FFFFFF")};
  border-color: ${(props) => (props.isTimeSelected ? "#333368" : "#CECEDF")};
  color: ${(props) => (props.isTimeAvaialble ? "#000042" : "#BDBDD3")};
  cursor: ${(props) => (props.isTimeAvaialble ? "pointer" : "")};
  pointer-events: ${(props) => (props.isTimeAvaialble ? "auto" : "none")};
`;

export const Time = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 700;
  font-size: 14px;
  line-height: 24px;
  display: flex;
  align-items: center;
  letter-spacing: -0.02em;
  /* color: #000042; */
`;

export const TimeValue = styled.div`
  font-family: var(--font-lksans-bold);
  font-style: normal;
  font-weight: 700;
  font-size: 14px;
  line-height: 24px;
  display: flex;
  align-items: center;
  letter-spacing: -0.02em;
  /* color: #000042;  */
  text-transform: lowercase;
`;

export const BookAppointmentCTAContainer = styled.div<{
  enabled?: boolean;
  bookSlotLoading?: boolean;
  primary?: boolean;
}>`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  gap: 20px;
  align-self: center;
  padding: 16px 100px;
  width: 500px;
  height: 56px;
  background: ${(props) =>
		props.enabled && !props.bookSlotLoading ? "#11daac" : "#BDBDD3"};
  border-radius: 100px;
  cursor: ${(props) => (props.enabled ? "pointer" : "auto")};
  pointer-events: ${(props) => (!props.enabled ? "none" : "auto")};
  &:hover {
    box-shadow: 0 3px 16px 0 rgba(0, 0, 0, 0.1);
    overflow: hidden;
    position: relative;
    z-index: 1;
  }
  svg path{
    stroke: var(--dark-blue-100)
  }
`;

export const PayNowCTAContainer = styled.div<{ primary?: boolean }>`
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  gap: 20px;
  align-self: center;
  padding: 16px 100px;
  width: 475px;
  height: 56px;
  border-radius: 100px;
  background: ${(props) => (props.primary ? "#11daac" : "")};
  border: ${(props) => (!props.primary ? "1.5px solid #737397" : "")};
  margin: 0 20px;
  cursor: pointer;
  &:hover {
    box-shadow: 0 3px 16px 0 rgba(0, 0, 0, 0.1);
    overflow: hidden;
    position: relative;
    z-index: 1;
  }
  svg path{
    stroke: var(--dark-blue-100)
  }
`;

export const BookAppointmentCTA = styled.div`
  font-family: var(--font-lksans-bold);
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  display: flex;
  align-items: center;
  letter-spacing: -0.02em;
  color: #000042;
  /* min-width: 230px; */
`;

export const ConfirmAppointmentScreen = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 20px;
`;

export const ConfirmHeading = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 400;
  font-size: 32px;
  line-height: 48px;
  text-align: center;
  letter-spacing: -0.02em;
  color: #000042;
`;

export const BookingDetails = styled.div`
  background: #ffffff;
  box-shadow: 0px 4px 12px rgba(243, 233, 226, 0.9);
  border-radius: 12px;
  display: flex;
  flex-direction: column;
  gap: 20px;
  padding: 30px 50px;
`;

export const Section = styled.div`
  display: flex;
  flex-direction: column;
  gap: 20px;
`;

export const SubHeading = styled.div`
  font-family: var(--font-lksans-medium);
  font-style: normal;
  font-weight: 500;
  font-size: 12px;
  line-height: 20px;
  text-align: center;
  letter-spacing: -0.02em;
  color: #333368;
`;

export const Row = styled.div`
  display: flex;
  flex-direction: column;
  gap: 10px;
`;

export const RowItem = styled.div`
  display: flex;
  gap: 20px;
  align-items: center;
`;
