// @lk/ui-library/src/Component/V1/PDP/Mobile/ItemDelivery/ItemDelivery.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { ComponentSizeENUM, TypographyENUM } from "../../../../../Types/general";
import ItemDelivery from "./ItemDelivery";

export default {
	title: "Components/V1/PDP/mobile/ItemDelivery",
	component: ItemDelivery,
	argTypes:{
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		componentSize: generateArgTypes({
			type: "string",
			required: false,
			description: "Size of the component",
			category: "General",
			defaultValue: ComponentSizeENUM.small,
			control: "radio",
			options: ComponentSizeENUM,
		}),
		isReadOnly: generateArgTypes({
			type: "boolean",
			required: true,
			description: "read only or not",
			category: "Component",
			defaultValue: "false",
		}),
		onmouseClick: generateArgTypes({
			type: "function",
			required: false,
			description: "on Mouse click handler for rating",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show rating right to left",
			category: "",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		})
	}
} as ComponentMeta<typeof ItemDelivery>;

const Template: ComponentStory<typeof ItemDelivery> = args => <ItemDelivery {...args}/>;
export const Overview = Template.bind({});
Overview.args = {
	localeData: {
		DELIVERY_MSITE_TEXT: "Delivery at your doorsteps within", DAYS_TEXT_FRAME: "5 - 7 Days", 
		DAYS_TEXT_CONTACT: "2 - 4 Days"
	},
	productClassification: "eyeglasses"
};
