// @lk/ui-library/src/Component/V1/PackagesScreen/RightSection/styles.tsx
import styled from "styled-components";

export const RightSectionContainer = styled.div`
	width: 70vw;
	vertical-align: top;
	height: 100%;
	position: relative;
`;

export const Icon = styled.img`
	cursor: pointer;
`;

export const RightTop = styled.div`
	padding: 15px 40px 0 40px;
	background-color: #fff;
    -webkit-box-shadow: 0 0 16px rgb(0 0 0 / 10%);
    box-shadow: 0 0 16px rgb(0 0 0 / 10%);
    position: absolute;
    width: 100%;
    top: 0;
    z-index: 1;
`;
export const RightBottom = styled.div`
	margin: 0;
	margin-top: 150px;
	overflow-y: scroll;
	height: calc(100vh - 150px);
	display: flex;
	align-items: center;
	justify-content: center;
	box-sizing: border-box;
    padding: 0;
`;

export const HeadContainer = styled.div`
display: flex;
box-sizing: border-box;
    margin: 0;
    padding: 0;
`;

export const Head = styled.div`
	-webkit-box-flex: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
    font-family: rajdhani-regular;
	font-weight: 600;
	text-align: center;
    text-align: -webkit-center;
	font-size: 28px;
`;
export const Cross = styled.img`
	margin: 0;
	padding: 0;
	display: block;
	cursor: var(--pointer);
`;
export const SequenceContainer = styled.div`
	-webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
	display: flex;
	text-align: -webkit-center;
	margin-top: 10px;
	margin: 0;
    padding: 0;
`;
export const SqHolder = styled.div<{hide: boolean}>`
	text-align: left;
	display: flex;
	${props => props.hide && "display: none"};
	font-family: rajdhani-regular;
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: 16px;
    line-height: 20px;
    color: #c5c3c3;
    cursor: var(--pointer);
    -ms-flex-item-align: end;
    align-self: flex-end;
	margin: 0;
    padding: 0;
`;
export const SqTextContainer = styled.div<{enabled: boolean;currentlySelected: boolean;}>`
	padding: 10px;
	-webkit-box-orient: vertical;
	-webkit-box-direction: normal;
	-ms-flex-direction: column;
	flex-direction: column;
	margin-top: 3px;
	display: flex;
	cursor: ${props => props.enabled ? "pointer" : "not-allowed"};
	${props => props.currentlySelected && "border-bottom: 2px solid #00bac6"};
	${props => props.currentlySelected && "color: #00bac6"};
`;
export const SqTopText = styled.div`
	color: #8b9a98;
	font-style: normal;
	font-weight: 700;
	font-size: 18px;
	line-height: 14px;
	margin: 0;
    padding: 0;
`;
export const SqBottomText = styled.div<{currentlySelected: boolean;}>`
	${props => props.currentlySelected && "color: #00bac6"};
`;
export const SqDividerLine = styled.div<{hasTopText: boolean;}>`
	width: 90px;
	border-bottom: 2px dashed #999;
	display: -webkit-box;
	display: -ms-flexbox;
	display: flex;
	-ms-flex-item-align: center;
	align-self: center;
	padding: 0;
	margin: 0;
`;
