// @lk/ui-library/src/Component/V1/MyAccount/Mobile/FeedbackPopUp/FeedbackPopUp.styles.tsx
import styled from "styled-components";

export const PopUpWrapper = styled.div`
  position: fixed;
  inset: 0;
  width: 100%;
  height: 100%;
  font: 12px/1.2857 "Roboto",sans-serif,"Arial","Helvetica",sans-serif;
  z-index: 101;
`;

export const Backdrop = styled.div`
  position: absolute;
  inset: 0;
  width: 100%;
  height: 100%;
  background-color: rgba(33,33,33,.75);
`;

export const Content = styled.div`
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: white;
  z-index: 101;
  min-height: 130px;
  min-width: 240px;
  border-radius: 4px;
`;

export const Container = styled.div`
  text-align: center;
  padding: var(--pd-20);
  min-width: 80vw;
`;

export const IconContainer = styled.div`
  display: flex;
  justify-content: center;
`;

export const SuccessIcon = styled.span`
  width: 40px;
  height: 40px;
  display: block;
  border-radius: 50%;
  text-align: center;
  font-size: 22px;
  border: 1px solid;
  color: #0ecda4;
  background: #fff;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  justify-content: center;

  &:before {
    content: "";
    font-family: fontawesome;
  }
`;

export const Heading = styled.div`
  font-size: var(--fs-18);
  font-weight: 500;
  margin-top: 8px;
`;

export const Text = styled.div<{ marginTop: string; }>`
  font-size: var(--fs-13);
  margin-top: ${({ marginTop }) => marginTop};
`;

export const Link = styled.span`
  text-align: center;
    text-transform: uppercase;
    border-radius: 4px;
    letter-spacing: 1px;
    background-color: var(--topaz);
    margin-top: 20px;
    color: white;
    font-size: var(--fs-12);
    font-weight: 700;
    display: block;
    padding: var(--pd-8);
`;
