// @lk/ui-library/src/Component/V1/MyAccount/Mobile/FeedbackPopUp/FeedbackPopUp.stories.tsx
import { action } from "@storybook/addon-actions";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import FeedbackPopUp from "./FeedbackPopUp.component";

export default {
	title:"Components/V1/MyAccount/Mobile/FeedbackPopUp",
	component: FeedbackPopUp,
	argTypes: {
		show: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Whether to show the popup or not",
			category: "General",
			defaultValue: false
		})
	}
} as ComponentMeta<typeof FeedbackPopUp>;

const Template: ComponentStory<typeof FeedbackPopUp> = args => <FeedbackPopUp {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		THANK_YOU: "Thank You",
		SYMBOL_EXCLAIMATION: "!",
		FOR_POSTING_REVIEW: "For posting a review.",
		REVIEW_PENDING_FOR_PUBLISH: "Your review was sent successfully and is now wating for our staff to publish it.",
		CONTINUE_SHOPPING: "CONTINUE SHOPPING"
	},
	onClick: () => action("Continue Shopping Clicked"),
	show: false
};
