// @lk/ui-library/src/Component/V1/ProductCardNew/styles.tsx
import styled from "styled-components";
import { devices } from "../../../helper";
import { TypographyENUM } from "../../../Types/general";

export const WishlistContainer = styled.div<{mobile: boolean,isDitto: boolean, isCygnus:boolean, isGridView: boolean}>`
position:absolute;
right:0;
top:0;
display:flex;
flex-direction:column;
align-items:center;
justify-content:center;
gap:2px;
padding:var(--pd-15);
z-index: 10;
transition: all 0.3s;
padding: ${props => props.mobile && props.isDitto && props.isCygnus? "20px 15px 15px 15px;" : "15px"};
${props => !props.mobile && props.isDitto && props.isCygnus? "background: #ffffff;padding: 8px; border-radius: 50%;right: 18px;top: 18px;" : ""}
@media screen and (max-width:767px){
	padding: ${props => props.mobile && props.isDitto && props.isCygnus? "7px 7px 15px 15px;" : "7px 7px 15px 15px"};
	margin: ${props => !props.isGridView && props.mobile && props.isDitto && props.isCygnus? "7px 7px 0 0" : ""};
}
&:hover .wishlist-icon{
${props => !props.mobile && "fill: var(--wishlist);"}
transition: all 0.3s;
transform: ${props => props.mobile ? "scale(1.1) translate(1px, 2px)" : "scale(0.9)"};
cursor: var(--pointer);
}
	svg{
		${props => props.isDitto && props.isCygnus? "width:24px;height:22px;margin-top: 2px;" : ""}
		 transform: scale(0.89);
		 width: 25px;
		 @media screen and (max-width:767px){
			height:24px;
			margin-top: 0px;
		 }
	}
`;

export const ProductImage = styled.img<{isDitto: boolean, isGridView: boolean, isMobile: boolean, width: number, isCygnus:boolean}>`
padding:${props => props.isDitto ? "0" : props.isGridView ? "var(--pd-40) 0 0 0" : "var(--pd-40) 0 0 0"};
// margin-bottom: ${props => props.isDitto ? "" : "var(--spacing-md)"};
display: ${props => props.isDitto ? "block" : ""};
z-index: ${props => props.isDitto ? "2" : ""};
// ${props => !props.isMobile && "max-height: 320px;"}
// max-height: 100vw;
max-height: ${props => props.isDitto && props.isCygnus ? "242px" : ""};
width: ${props => props.isDitto && props.isCygnus? "95%" : "100%"};
min-height: ${props => props.isDitto && props.isCygnus? "242px" : ""};
margin: ${props => props.isDitto ? "0px auto" : "" };
position: relative;
object-fit: cover;
${props => props.isMobile && props.isDitto && " width: 100%; object-fit: cover;"}
// height: ${props => (props.width / 628) * 301 > 220 ? (props.width / 628) * 301 : 220}px;
${props => props.isGridView && "height: auto; margin-bottom: 0; object-fit: cover; max-height: 140px;"}
padding: ${props => props.isDitto && props.isCygnus ? "0px 0 10px" : "10px" };
// @media ${devices.mobileMMaxWidth} {
// 	${props => props.isMobile && props.isDitto && "padding: 10px; width: 100%; object-fit: cover;"}
// };
@media screen and (max-width:1199px) and (min-width:768px){
	${props => props.isGridView && "max-height: inherit;"}
}
@media screen and (max-width:767px){
	max-height: ${props => props.isDitto && props.isCygnus ? "320px" : ""};
	width: ${props => props.isDitto && props.isCygnus ? "100%" : "100%"};
	min-height: ${props => props.isDitto && props.isGridView && props.isCygnus? "125px" : ""};
	max-height: ${props => props.isDitto && props.isGridView && props.isCygnus ? "125px" : ""};
 }
`;

export const ProductTitle = styled.p<{showTitle:boolean, isRTL: boolean, isGridView: boolean}>`
word-wrap: break-word;
text-overflow: ellipsis;
overflow: hidden;
max-width: 100%;
font-family: var(--font-lksans-bold);
font-size: ${props => props.isGridView ? "var(--fs-12)" : "var(--fs-14)"};
/* font-weight:var(--fw-bold); */
margin:0;
text-align: ${props => props.isRTL ? "end" : ""};
display: ${props => props.showTitle ? "" : "none"};
`;

export const ProductSizeContainer = styled.div<{isRTL: boolean, isGridView: boolean }>`
display:flex;
height: 22px;
flex-wrap:wrap;
margin: ${props => props.isGridView ? "0" : "-5px 0"};
align-items:center;
justify-content:var(--space-between);
flex-direction: ${props => props.isRTL ? "row-reverse" : ""};
@media screen and (max-width:767px){
	height: auto;
}
`;

export const ProductSize = styled.span<{isGridView: boolean}>`
color:var(--text);
font-weight: var(--fw-regular);
font-size: ${props => props.isGridView ? "var(--fs-10)" : "var(--fs-12)"};
font-family: var(--font-lksans-regular);
// min-height: 29px;
@media screen and (max-width:767px){
	max-width: ${props => props.isGridView ? "" : "60%"};
}
`;

export const ProductUsage = styled.span<{isGridView: boolean}>`
color:var(--text);
font-weight: var(--fw-regular);
font-size: ${props => props.isGridView ? "var(--fs-10)" : "var(--fs-12)"};
font-family: var(--font-lksans-regular);
`;

export const ProductDetailsContainer = styled.div<{isDitto: boolean, isRTL: boolean,}>`
display:flex;
flex-direction:column;
gap:var(--spacing-sm);
padding: ${props => props.isDitto ? "var(--pd-15)" : "var(--pd-15)"};
padding-left: var(--pd-12);
direction: ${props => props.isRTL ? "ltr" : ""};
padding-bottom:0px;
& [data-testid="pricing-test"]{
	padding-bottom:6px;
}
@media screen and (max-width:767px){
	padding-bottom:var(--pd-15);
}
`;
export const AnchorWrapper = styled.a`
	display: flex;
	flex-direction: column;
	justify-content: space-between;
	height: 100%;
 	cursor:pointer;
`;


export const ProductContainer = styled.div<{width: number, isGridView: boolean;}>`
display:flex;
flex-direction:column;
//width:${props => props.width ? `${props.width}px` : "33%"};
-webkit-tap-highlight-color: transparent;
@media ${devices.mobileS} {
	max-width: 100%;
	width: ${props => props.isGridView ? "49%" : "100%"};
	// width: 100%;
	margin: ${props => props.isGridView ? "2px" : "10px 0px"};
};
background-color:white;
border: 1px solid var(--border);
border-radius:var(--border-radius-xxs);
position:relative;
justify-content: space-between;

&:hover{
	transition: box-shadow 0.2s ease-in-out, -webkit-box-shadow 0.2s ease-in-out;
	box-shadow: 0 3px 16px 0 rgba(0, 0, 0, 0.11);
};
@media ${devices.laptop} {
	max-width: 32%;
	width:${props => props.width ? `${props.width}px` : "33%"};
	margin: 10px;
};
@media ${devices.desktop} {
	max-width: 32%;
	width:${props => props.width ? `${props.width}px` : "33%"};
	margin: 10px;
};

& #product-rating{
	font-size:${props => props.isGridView ? "11px" : "14px"};
	margin-top: 10px;
}
`;

export const ImageContainer = styled.div<{isDitto: boolean, url?: string, isGridView: boolean, isCygnus:boolean}>`
	display: flex;
	justify-content: center;
	justify-content: center;
	width: 100%;
	display: block;
	padding-top: ${props => props.isDitto && props.isCygnus ? "12px" : "26px"};
	position: relative;
	// height: ${props => props.isDitto ? "56%" : "100%"};
	// background: ${props => props.isDitto ? `url(${props.url}) center no-repeat`: "none"};

	left: ${props => props.isDitto ? "0px" : ""};
    top: ${props => props.isDitto ? "0px" : ""};
    width: ${props => props.isDitto ? "100%" : ""};
	// min-height: ${props => props.isGridView ? "100px" : "220px"};
// 	-ms-filter: ${props => props.isDitto ? "progid:DXImageTransform.Microsoft.Alpha(Opacity=50)" : ""};
//   filter: ${props => props.isDitto ? "alpha(opacity=50)" : "" };
//   -khtml-opacity: ${props => props.isDitto ? "0.5" : ""};
	// &:before {

  	// }
	// opacity: ${props => props.isDitto ? "0.85" : ""};
    z-index: ${props => props.isDitto ? "1" : ""};
    background-size: ${props => props.isDitto ? "cover" : ""};
    background-repeat: ${props => props.isDitto ? "no-repeat" : ""};
	position: relative;
	@media screen and (max-width:767px){
		padding: ${props => props.isDitto && props.isCygnus ? "10px 10px 0 10px" : "26px 0 0 0"};
		min-height: ${props => props.isGridView ? "125px" : ""};
	}
`;

export const ColorOptionsContainer = styled.div<{isDitto: boolean, isRTL?: boolean}>`
display:${props => props.isDitto ? "none": "flex"};
flex-wrap:wrap;
gap:2px;
font-size:var(--fs-12);
align-items:center;

${props => props.isRTL ? "margin-right:auto;" : ""};
${props => !props.isRTL ? "margin-left:auto;" : ""};

@media (max-width: 1296px) and (min-width: 1022px) {
	padding-top: 7px;
}
`;

export const OfferContainer = styled.div<{isRTL: boolean, offerText: string, isGridView: boolean}>`
background: ${props => props.isRTL ? "linear-gradient(270deg,#F7F1DE 0%,rgba(247,241,222,0) 100%)" : "linear-gradient(90deg,#F7F1DE 0%,rgba(247,241,222,0) 100%)"};
padding:var(--pd-10) var(--pd-15);
border-bottom-left-radius: var(--border-radius-xxs);
color: var(--gold);
/* font-weight: var(--fw-bold); */
// font-size: var(--fs-12);
font-size: ${props => props.isGridView ? "var(--fs-10)" : "var(--fs-12)"};
font-family: var(--font-lksans-bold);
text-align: ${ props => props.isRTL ? "end" : ""};
margin: ${props => props.isGridView ? "-15px -12px" : "-15px -12px"};
margin-top: -4px;
visibility: ${props => props.offerText ? "visible" : "hidden"};
margin-bottom: 0;
border-bottom-right-radius: var(--border-radius-xxs);
@media screen and (max-width:767px){
 margin:-0px -12px -15px -12px ;
}
`;

export const RatingStarContainer = styled.div<{isRTL: boolean, visible: boolean}>`
	width: 100%;
	${props => !props.visible && "visibility: hidden"};
	display: flex;
	// justify-content: ${props => props.isRTL ? "flex-end" : "flex-start"};
	justify-content: space-between;
	direction: ${props => props.isRTL ? "rtl" : "ltr"};
	margin-top:-17px;
	margin-bottom:10px;
`;

export const AnchorCount = styled.a`
text-decoration: none;
font-family: ${TypographyENUM.lkSansBold};
font-size: 12px;
/* font-weight: 700; */
line-height: 18px;
letter-spacing: -.02em;
color: #333368;
`;

export const HeartIconWrapper = styled.div<{isGridView: boolean}>`
	font-size: var(--fs-13);
	scale: ${props => props.isGridView ? "0.55" : "1"};
	background: rgb(255, 255, 255);
	padding: 8px;
	border-radius: 50%;
	display: flex;

	.wishlist-icon{
		transform: translate(1px, 2px);
	}

`;

export const OfferContainerOuter = styled.div`
	/* height: 20px; */
`;

export const ImageLoaderComponent = styled.div`
	z-index: 8;
	width: 100%;
	height: 100%;
	position: absolute;
	top: 0;
	left: 0;
`;

export const Button = styled.button`
	border: none;
    background: #fff;
	margin-top: 10px;
    box-shadow: 0 1px 4px rgba(0,0,66,.1);
    border-radius: 100px;
    font-family: ${TypographyENUM.lkSansBold};
    font-style: normal;
    font-weight: 700;
    font-size: 12px;
    color: #000042;
    padding: 6px 12px;
    display: flex;
    align-items: center;
    gap: 8px;
    transition: .3s ease all;
    cursor: pointer;
	height: 30.17px;
`;
