// @lk/ui-library/src/Component/V1/NewPayment/ContactLensConsentCheckbox/ContactLensConsentCheckbox.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import ContactLensConsentCheckbox from "./index";

export default {
	title:"Components/V1/ContactLensConsentCheckbox",
	component: ContactLensConsentCheckbox,
	argTypes: {
		...generateGeneralArgs(),
		onSignIn: generateArgTypes({
			type: "function",
			required: false,
			description: "onSignIn handler for sign in form",
			category: "Events",
		}),
		onProceed: generateArgTypes({
			type: "function",
			required: false,
			description: "onProceed handler for sign in form",
			category: "Events",
		}),
		countryCode: generateArgTypes({
			type: "string",
			required: true,
			description: "country code",
			category: "Component",
			defaultValue: "+91",
		}),
		showOTPView: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show OTP view",
			category: "Component",
			defaultValue: false,
		}),
		isLoginSuccess: generateArgTypes({
			type: "boolean",
			required: true,
			description: "if login is successful",
			category: "Component",
			defaultValue: false,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show to right to left",
			category: "Status",
			defaultValue: false,
		}),
	}
} as ComponentMeta<typeof ContactLensConsentCheckbox>;

const Template: ComponentStory<typeof ContactLensConsentCheckbox> = (args) => <ContactLensConsentCheckbox {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	checked: false,
	dataLocale: {
		CL_BY_PROCEEDING: "By proceeding, you agree to our",
		CL_TERMS_AND_CONDITIONS_TITLE: "Terms and Conditions",
		CL_FOR_CONTACT_LENSES: "for contact lens purchases.",

	}
};
