// @lk/ui-library/src/Component/V1/NewPayment/ContactLensConsentCheckbox/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const ContactLensConsentCheckboxWrapper = styled.div`
margin-bottom:15px;
`;

export const Divider = styled.div`
width: 100%;
margin-top: 15px;
margin-bottom: 10px;
`;

export const HorizontalLine =styled.hr`
    border: none;
    height: 1px;
    background-color: rgba(0,0,0,.12);
`;

export const ConsentContainer = styled.div`
    padding: 10px;
    input[type=checkbox] {
        opacity: 0;
        position: absolute;
        width: 24px;
        height: 24px;
        cursor: pointer;
    };
    label {
        display: flex;
        font-weight: normal;
        padding: 0 3px 0px 30px;
        background: url("https://static.lenskart.com/media/desktop/img/uncheckbox_filter.png") no-repeat 0 0;
        line-height: 17px;
		cursor:pointer;
    };
    input[type=checkbox]:checked+span {
        background: url("https://static.lenskart.com/media/desktop/img/checkbox_filter.png") no-repeat 0 0;
        margin-left: -30px;
        padding-left: 30px;
        line-height: 17px;
    };
    label span {
        display: flex;
        align-items: center;
        min-height: 24px;
		@media (max-width: 767px) {
			font-size:11px;
			margin-top:0px;
		}
    };
    a {
        color: #329c92;
        text-decoration: none;
        cursor: pointer;
    }
`;

export const TermsCondition = styled.span`
    display: flex;
    align-items: center;
    min-height: 24px;
    font-size: 13px;
    color: #333;
    font-weight: 400;
    line-height: 17px;
    cursor: pointer;
    font-family: ${TypographyENUM.lkSansBold};
`;

export const ModalTermsAndConditionsWrapper = styled.div`
    position: fixed;
    margin: 0 auto;
    top: 0;
    left: 0;
    padding-top: 10px;
    z-index: 1200 !important;
    overflow-x: hidden;
    overflow-y: hidden;
    height: 100vh;
    width: 100%;
    align-items: center;
    display: flex;
	background:rgba(0,0,0,0.5);
`;

export const TermsConditionUl = styled.ul``;

export const TermsAndConditionsBox = styled.div`
    position: relative;
    background-color: #fff;
    background-clip: padding-box;
    border: 1px solid #999;
    border: 1px solid rgba(0,0,0,.2);
    border-radius: 6px;
    outline: 0;
    box-shadow: 0 5px 15px rgba(0,0,0,.5);
    width: 600px;
    margin: auto;
	font-family: ${TypographyENUM.defaultMedium};
`;

export const ModalHeader = styled.div<{mobilesite: boolean}>`
    padding: ${props => props.mobilesite ? "10px 0" : "15px"};
    margin: ${props => props.mobilesite ? "0 0 15px" : "" };
    border-bottom: 1px solid #e5e5e5;
    min-height: 55px;
`;

export const CloseButton = styled.div`
	display: flex;
	justify-content: flex-end;
	width: 100%;
	margin-top: -21px;
    margin-bottom: 8px
	margin-right: 10px;
	cursor:pointer;
`;

export const ModalTitle = styled.div<{mobilesite: boolean}>`
    font-weight: ${props => props.mobilesite ? "400" : "600"};
    font-size: ${props => props.mobilesite ? "16" : "18" }px;
    line-height: ${props => props.mobilesite ? "24" : "" }px;
    letter-spacing: ${props => props.mobilesite ? "-0.02" : ""}em;
    color: ${props => props.mobilesite ? "#000042" : "#333"};
    text-align: ${props => props.mobilesite ? "left" : "center" };
`;

export const ModalBody = styled.div<{mobilesite: boolean}>`
    position: relative;
    padding: ${props => props.mobilesite ? "0" : "30" }px;
    height: 100%;
`;

export const ConsentHeader = styled.div<{mobilesite: boolean}>`
    font-weight: ${props => props.mobilesite ? "400" : "600"};
    font-size: ${props => props.mobilesite ? "12" : "16"}px;
    margin-bottom: 15px;
    margin-top: ${props => props.mobilesite ? "15" : "0"}px;
    line-height:  ${props => props.mobilesite ? "18" : "" }px;
    letter-spacing: -0.02em;
    text-align: left;
    font-family: ${props => props.mobilesite ? `${TypographyENUM.lkSansRegular}, ${TypographyENUM.serif}` : "" };
    color: ${props => props.mobilesite ? "#333368" : "#333"};
`;

export const ConsentPoints = styled.li<{mobilesite: boolean}>`
    margin-bottom: 10px;
    margin-left: 10px;
    margin-right: 10px;
    list-style-type: disc;
    font-weight: ${props => props.mobilesite ? "400" : "600"};
    font-size: ${props => props.mobilesite ? "12" : "13"}px;
    line-height: ${props => props.mobilesite ? "18px" : "18px" };
    letter-spacing: -0.02em;
    text-align: left;
    font-family: ${TypographyENUM.defaultBook}, ${TypographyENUM.serif};
    color: ${props => props.mobilesite ? "#333368" : "#333"};
	a{
		color: #329c92;
	}
	&:maker{
		display:none;
	}
`;

export const BottomSheetWrapper = styled.div`
    font-family: ${TypographyENUM.lkSansRegular};
    color: #000042;
`;
