// @lk/ui-library/src/Component/Common/ToastMessage/ToastMessage.styles.tsx
import styled, { keyframes } from "styled-components";
import { TypographyENUM } from "../../../Types/general";

const animateRight = keyframes`
  from {
		transform: translateX(100%);
	}
	to {
		transform: translateX(0);
	}
`;


export const ToastWrapper = styled.div<{
	color?:string,position?:string
}>`
  z-index: 999999999999;
  padding:2px 10px;
  text-align: center;
  position:${props => props.position ? props.position : "fixed"};
  background-color:${props => props.color ? props.color : "orange"};
  justify-content: center;
  display: flex;
  border-radius:4px;
  top:20px;
  right:20px;
  opacity:0.9;
  transition:0.3s ease;

  &:hover{
    box-shadow: 0 0 12px #fff;
    opacity:1;
  }

  animation: ${animateRight} 0.7s;
`;

export const ToastText = styled.span`
  color: white;
  width: 39vw;
  font-size: 16px;
  font-family:${TypographyENUM.lkSansRegular};
  padding-top:10px;
  padding-right:20px;
  padding-bottom:10px;
`;


export const ToastCross = styled.div`
  font-size: 12px;
  color: white;
  padding-top:5px;
`;

export const IconWrapper = styled.span<{showIcon: boolean}>`
  ${props => !props.showIcon && "display: none;" }
`;
