// @lk/ui-library/src/Component/V1/MyAccount/Mobile/ExchangeFilters/ExchangeFilter.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import ExchangeFilter from "./index";
import { filters } from "./dataValue";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import { FilterTypes, OptionTypes } from "./ExchangeFilter.types";

export default {
	title: "Components/V1/MyAccount/Mobile/ExchangeFilter",
	component: ExchangeFilter,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Boolean Text left to right",
			category: "Component",
			defaultValue: "false",
		}),
		filters: generateArgTypes({
			type: "array",
			required: true,
			description: "Filters Data",
			category: "Component",
			control: "array"
		})
	},
} as ComponentMeta<typeof ExchangeFilter>;

const Template: ComponentStory<typeof ExchangeFilter> = (args) => (
	<ExchangeFilter {...args} />
);
export const Overview = Template.bind({});
Overview.args = {
	id: "ExchangeFilter",
	FilterTitle:"Please tell us the problem with the product",
	AdditionComment:"Additional Comments (Optional)",
	PrimaryText:"Submit Reason",
	filters: filters,
	font: TypographyENUM.defaultBook,
	isRTL: false,
	onChange: ({ event, item, selectedOption }: {item: FilterTypes, selectedOption: OptionTypes, event: any}) => console.log(event, item, selectedOption)

};
