// @lk/ui-library/src/Component/V1/MyAccount/Mobile/ExchangeFilters/ExchangeFilter.types.ts
import { ReactNode } from "react";
import { TypographyENUM } from "../../../../../Types/general";

export interface ExchangeFilterGroupTypes {
	id: string
	filters: Array<FilterTypes>,
	font: TypographyENUM
	filterSelected: any
	onSubmit: () => void;
	type: filterTypeENUM
	canCollapse?: boolean
	showBorderTop?: boolean,
	onChange: ({ mainOptionId, secondaryOptionId }: {mainOptionId?: string, secondaryOptionId?: string}) => void
	// onClick: ({ event, item, selectedOption }: {item: FilterTypes, selectedOption: OptionTypes, event: React.MouseEvent<HTMLInputElement, MouseEvent>}) => null
	isRTL?: boolean
	defaultExpand?: number;
	padding?: boolean
	FilterTitle:string,
	AdditionComment:string,
	PrimaryText:{
		enable: string,
		disable: string,
	},
	selectedMainOptionId: string, 
	selectedSecondaryOptionId: string
}

export interface FilterTypes {
	reason: string
	id: string
	type: string,
	canCollapse: boolean,
	showBorderTop: boolean,
	selectedOption?: string[]
	secondaryReasons?: Array<OptionTypes>
	component?: () => ReactNode
	isRTL?: boolean
	height?: string | undefined
}
export interface OptionTypes {
	reason: string
	productsCount: number
	imageUrl?: string
	id: string
}

export interface AccordionTypes {
	id: string
	children: ReactNode
	title: string
	isMaxHeight: boolean
	expand: boolean
	showBorderTop: boolean
	canCollapse: boolean
	isComponent: boolean
	handleExpand: () => void,
	font: TypographyENUM
	isRTL?: boolean
	padding?: boolean
	height?: string | undefined
}

export enum filterTypeENUM {
	IMAGE_SELECTION = "IMAGE_SELECTION",
	CHECKBOX = "CHECKBOX",
	COMPONENT="COMPONENT"
}

export type handleClickAttrs = {
	// item: FilterTypes
	// selectedOption: OptionTypes
	// event: any
	mainOptionId: string, 
	secondaryOptionId: string
}
