// @lk/ui-library/src/Component/V1/MyAccount/Mobile/ExchangeFilters/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";
interface Props {
	expand?: boolean
	column?: boolean
	isMaxHeight?: boolean
	isRTL?: boolean
	padding?: boolean
	height?: string | undefined
}

export const ExchangeFilterWrapper = styled.div`
 background:#fff;
 padding:10px;
`;

export const ExchangeFilterInner
= styled.div`
   border:1px solid var(--light-grey);
   border-radius: 3px;
   background:#fff;
`;
export const AccordionWrapper = styled.div<{borderTopStyle: boolean, expand: boolean, styleFont: TypographyENUM, isRTL: boolean}>`
	border-top: ${(props) => props.borderTopStyle === true ? "1px solid var(--light-grey) !important" : "none"};
	text-align: left;
	font-size: var(--fs-14);
	font-family: ${(props) => props.styleFont ? props.styleFont : TypographyENUM.defaultBook};
	font-synthesis: none;
	font-weight: bold;
	letter-spacing: 1.3px;
	color: var(--darker-grey);
	border: 0px solid var(--transparent);
	&:last-child {
		border-bottom: none;
	}
	&:first-child {
		border-top: none !important;
	}
`;

export const TitleBox = styled.div<{styleComponent: boolean, expand: boolean, isRTL: boolean, padding: boolean}>`
	padding: ${(props) => props.styleComponent ? "var(--pd-20) var(--pd-10)" : "var(--pd-15) var(--pd-10) var(--pd-10) var(--pd-10)"};
	background-color: ${(props) =>
		props.expand === true ? "var(--turquoise-blue-dark)" : "var(--grey-100)"};
	border-radius: ${(props) =>
		props.expand === true ? "10px 10px 0px 0px" : "10px 10px 10px 10px"};
	display: flex;
	justify-content: var(--space-between);
	height: auto;
	color: ${(props) =>
		props.expand === true ? "var(--text-color)" : "var(--text-color-black)"};
	font-weight: var(--fw-md);
	align-items: center;
	flex-direction: ${props => props.isRTL ? "row-reverse" : "row"};
	${props => props.padding && "padding: var(--pd-20) 0"}
`;

export const TitleWrapper = styled.div`
	display: flex;
	flex-direction: column;
	width: 100%;
`;

export const Title = styled.span<{
	isRTL: boolean
}>`
	width: calc(100% - 20px);
	display: ${props => props.isRTL ? "table" : "flex"};
	text-align: ${props => props.isRTL ? "end" : "initial"};
	align-items: center;
	font-size: var(--fs-14);
	letter-spacing: 1.3px;
	font-weight: 400;
	color: #424242;
	font-family:  var(--font-lksans-regular);
`;

export const Content = styled.div<Props>`
    padding:0 0 0 15px;
	transform-origin: top;
	background: var(--grey-100);
	border-radius: 0px 0px var(--pd-10) var(--pd-10);
	height: ${props => props.height ? props.height : "auto"};
	${props => !props.expand && "height: 0"};
	transform: ${props => props.expand ? "scaleY(1)" : "scaleY(0)"};
	transition: 0.3s ease-in-out;
	overflow: hidden;
	/* width */
	::-webkit-scrollbar {
		width: 5px;
	}

	/* Track */
	::-webkit-scrollbar-track {
		border-radius: 0;
	}

	/* Handle */
	::-webkit-scrollbar-thumb {
		background: var(--whitish-gray);
		border-radius: 5px;
	}

	/* Handle on hover */
	::-webkit-scrollbar-thumb:hover {
		background: var(--black);
	}
`;


export const CheckBoxLabel = styled.span<{
	isRTL: boolean
}>`
	font-size: var(--fs-14);
	font-family:  var(--font-lksans-regular);
	padding: 10px 0px 10px 24px;
	color: #424242;
	font-weight:400;
	position:relative;
	&:hover {
		color: var(--black);
	}
	&:before{
		left: 0;
		top: 12px;
		width: 14px;
		height: 14px;
		border: 1px solid #ccc;
		border-radius:50%;
		border: 1px solid #ccc;
		position:absolute;
		content:"";
	}
`;

export const UpArrow = styled.span`
	width: 14px;
	height: 14px;
	border: 1px solid #ccc;
	margin-top: 0px;
	margin-right:10px;
	display:block;
	border-radius:50%;
`;

export const DownArrow = styled.span`
	width: 10px;
	height: 10px;
	background: #18cfa8;
	margin-top: 0px;
	margin-right:10px;
	display:block;
	border-radius:50%;
	border: 1px solid #ccc;
	position:relative;
	margin-left:4px;
	&:before{
		content: "";
		width: 14px;
		height: 14px;
		border: 1px solid #18cfa8;
		position: absolute;
		border-radius: 50%;
		left: -3px;
		right: -3px;
		top: -3px;
		bottom: -3px;
	}
`;
export const Label = styled.label`
	display: flex;
    align-items: left;
	color:#424242;
`;

export const FilterItemsWrapper = styled.div<Props>`
	padding-left:5px;
	input{
		display:none;
		&:checked + span:after{
			left: 2px;
			top: 14px;
			width: 10px;
			height: 10px;
			background: #18cfa8;
			border-radius:50%;
			position:absolute;
			content:"";
		}
	}
`;

export const FilterHeading = styled.div<Props>`
	font-size: var(--fs-14);
	font-family:  var(--font-lksans-bold);
	padding: 10px 0px 10px 10px;
	color: #424242;
	/* font-weight:500; */
	&.margin-top10{
		margin-top:10px;
	}
`;
export const TextareaWrapper = styled.div<Props>`
margin-bottom: 20px;
& textarea{
	border:1px solid #ddd;
	font-family:  var(--font-lksans-regular);
	border-radius: 4px;
    padding: 8px;
	margin-bottom: 50px;
	min-height: 80px;
	color: #424242;
	&::-webkit-input-placeholder{
		color: #424242;
	}
	&::-ms-input-placeholder{
		color: #424242;
	}
	&::-placeholder{
		color: #424242;
	}
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
}`;

export const ReasonTitle = styled.span`
	font-size: 13px;
`;
