// @lk/ui-library/src/Component/V1/Checkout/Shipping/index.tsx

import React from "react";
import styled from "styled-components";
import {
	AddressColumnFlex,
	AddressData,
	ChangeButton,
	HeaderLeft,
	HeaderText,
	LeftOuterMost,
	LeftRoot,
} from "../styles";
import PrimaryButton from "../../../Common/PrimaryButton";
import { ComponentSizeENUM, ThemeENUM, TypographyENUM } from "../../../../Types/general";

const Body = styled.div`
  margin-left: 33px;
  margin-right: 20px;
  margin-top: 5px;
  margin-bottom: 5px;
`;
const SubHead = styled.div`
  display: flex;
  justify-content: space-between;
  color: #999999;
  font-size: 12px;
  align-items: center;
`;

const SummaryAdd = styled.div`
  font-size: 12px;
  font-weight: 400;
  margin-top: 5px;
`;

const DarkDiv = styled.div`
   color: #333;
`;
const DivMar = styled.div`
   margin-bottom: 3px;
`;

const Shipping = ({
	dataLocale = {
		HEADER: "4. Where do you want to ship the Exchange Item?",
		SHIPPING_ADDRESS: "SHIPPING ADDRESS",
		SUBHEADER: "We need your address to proceed",
		CHANGE: "CHANGE"
	},
	OpenSidebar,
	isExpandable,
	shippingAddress,
}: any) => {
	return (
		<>
			{isExpandable === "Shipping Address" ? (
				<LeftOuterMost>
					<LeftRoot>
						<HeaderLeft
							background={isExpandable === "Shipping Address" ? true : false}
						>
							<HeaderText>
								{dataLocale.HEADER}
							</HeaderText>
						</HeaderLeft>
					</LeftRoot>
					{isExpandable === "Shipping Address" ? (
						<Body>
							{shippingAddress ? (
								<SubHead>
									<div>{dataLocale.SHIPPING_ADDRESS}</div>
									<ChangeButton onClick={() => OpenSidebar()}>
										{dataLocale.CHANGE}
									</ChangeButton>
								</SubHead>
							) : (
								<SubHead>
									<DarkDiv>
										{dataLocale.SUBHEADER}
									</DarkDiv>
									<PrimaryButton
										primaryText={dataLocale.SELECT_ADDRESS}
										font={TypographyENUM.serif}
										componentSize={ComponentSizeENUM.medium}
										onBtnClick={() => OpenSidebar()}
										id="btn-primary-cl"
										width={"17%"}
										height="35px"
										theme={ThemeENUM.primary}
									/>
								</SubHead>
							)}
							{shippingAddress ? (
								<SummaryAdd>
									<DivMar>
										{shippingAddress.firstName} {shippingAddress.lastName}
									</DivMar>
									<AddressColumnFlex>
										<AddressData font={TypographyENUM.lkSansRegular}>
											{shippingAddress.addressline1}
										</AddressData>
										<AddressData font={TypographyENUM.lkSansRegular}>
											{shippingAddress.addressline2}
										</AddressData>
										<AddressData font={TypographyENUM.lkSansRegular}>
											{shippingAddress.region}
										</AddressData>
										<AddressData font={TypographyENUM.lkSansRegular}>
											{shippingAddress.city}, {shippingAddress.state},{" "}
											{shippingAddress.postcode}
										</AddressData>
									</AddressColumnFlex>
								</SummaryAdd>
							) : null}
						</Body>
					) : (
						<></>
					)}
				</LeftOuterMost>
			) : null}
		</>
	);
};

export { Shipping };
