// @lk/ui-library/src/Component/V1/PDP/Review/ReviewForm/ReviewForm.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const ReviewFormWrapper = styled.div<{styleFont:TypographyENUM}>`
	font-family:${props => props.styleFont};
	padding: 35px 30px 30px;
	border-top: 1px solid var(--border);
	textarea{
		height:115px;
		user-select: initial;
		&::before, &::after {
    		user-select: initial;
		}
	}
	@media screen and (max-width:767px){
		padding: 22px 15px 15px;
	}
`;

export const FormTitle = styled.p`
	font-weight: var(--fw-medium);
	font-size:var(--fs-28);
	color:var(--blackish-gray);
	margin:0;
	padding-top:10px;
	text-align: center;
	font-family: ${TypographyENUM.lkSerifNormal};
	@media screen and (max-width:767px){
		text-align: left;
		font-size:var(--fs-16);
	}
`;

export const Form=styled.form`
	display:flex;
	flex-direction:column;
	gap:var(--spacing-lg);
	padding: var(--pd-20) 0;
`;

export const RatingWrapper = styled.div`
	display:flex;
	flex-direction:column;
	gap:var(--spacing-sm);
`;

export const RatingLabel = styled.p<{isRTL: boolean}>`
	font-size:var(--fs-16);
	font-family: ${TypographyENUM.lkSansRegular};
	margin:0;
	text-align: ${props => props.isRTL ? "end": ""};
`;

export const InputWrapper = styled.div`
	display:flex;
	gap:var(--spacing-md);
	input,textarea{
		border-top:1px solid #ccc;
		border-left:1px solid #ccc;
		border-right:1px solid #ccc;
		padding:6px 12px;
		border-radius: 4px;
		font-family: ${TypographyENUM.lkSansRegular};
		user-select: initial;
		transition:0.3s;
		&::before, &::after {
    		user-select: initial;
		}
	}
	textarea + label{
         background:none !important;
	   }
	label{
		padding:0px 12px;
		font-family: ${TypographyENUM.lkSansRegular};
	}
	input{
		height:34px;
	}
	input:focus ~ label,textarea:focus ~ label{
		padding: 0px 5px;
		top: 16px;
		background: #fff;
		left: 5px !important;
		width: fit-content;
		right: 5px;
	}
	textarea:focus + label{
		left: 11px !important;
	}
	input::not(:empty) ~ label,textarea::not(:empty) ~ label{
		padding: 0px 5px;
		top: 16px;
		background: #fff;
		left: 5px !important;
		width: fit-content;
		right: 5px;
	}
	input:not([value=""]) ~ label{
		padding: 0px 5px;
		top: 14px;
		background: #fff;
		left: 5px !important;
		width: fit-content;
		right: 5px;
	}
`;

export const InputField = styled.input<{width:string}>`
	width: ${props => props.width};
`;

export const TextAreaField = styled.textarea<{width:string}>`
	width: ${props => props.width};
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const FormActionWrapper = styled.div<{hide?: boolean}>`
	display:flex;
	align-items:center;
	justify-content:center;
	button{
		margin-right:inherit;
	}
	${props => props?.hide && "visibility: hidden;" }
`;

export const FormInputError = styled.small`
	color:var(--error);
`;

export const SuccessMessage = styled.div`
	background-color: #dff0d8;
	padding: var(--pd-15);
	margin: 15px;
	border-radius: 4px;
	color: #3c773e;
`;

export const ErrorMessage = styled.div`
	background-color: #f2dede;
	padding: var(--pd-15);
	border-radius: 4px;
	color: #ac4b49;
	margin: 10px 0;
	font-size: 14px;
`;
export const Error = styled.span`
	color: #445862;
	font-size: 14px;
`;
export const ReviewLoader = styled.span`
	position:relative;
	// height: 70px;
	display:block;
	img{
		max-width: 100%;
		position: absolute;
		left: 0;
		bottom: -100px;
		right: 0;
		@media screen and (max-width:767px){
			left: 94px;
			bottom: -16px;
			height: 112px;
		}
	 }
	}
`;
