// @lk/ui-library/src/Component/V1/PDP/Review/ReviewForm/ReviewForm.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import { ReviewForm } from "./ReviewForm.component";

export default {
	title: "Components/V1/PDP/Review/ReviewForm",
	component: ReviewForm,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "select",
			options: TypographyENUM,
		}),
		dataLocale: generateArgTypes({
			type: "object",
			required: true,
			description: "static data",
			category: "Component",
			defaultValue: {
				HOW_WOULD_YOU_RATE_PRODUCT: "How would you rate the product?",
				WRITE_A_REVIEW: "Write A Review",
				ENTER_EMAIL_ID: "Enter Email Id",
				YOUR_NAME: "Your Name",
				TITLE: "Title",
				REVIEW: "Review",
				POST_REVIEW: "POST REVIEW",
				THIS_FIELD_IS_REQUIRED: "This Field is Required",

			},
			control:"object"
		}),
		onSubmit: generateArgTypes({
			type: "function",
			required: true,
			description: "event on form submission",
			category: "Event",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show right to left",
			category: "Status",
		}),
	}
} as ComponentMeta<typeof ReviewForm>;

const Template: ComponentStory<typeof ReviewForm> = args => <ReviewForm {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "review-form",
	dataLocale: {
		HOW_WOULD_YOU_RATE_PRODUCT: "How would you rate the product?",
		WRITE_A_REVIEW: "Write A Review",
		ENTER_EMAIL_ID: "Enter Email Id",
		YOUR_NAME: "Your Name",
		TITLE: "Title",
		REVIEW: "Review",
		POST_REVIEW: "POST REVIEW",
		THIS_FIELD_IS_REQUIRED: "This Field is Required",

	},
	font:TypographyENUM.defaultBook,
	onSubmit: () => console.log("submit"),
};
