// @lk/ui-library/src/Component/V1/PrescriptionModal/DropDown/DropDown.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const Root = styled.div`
  box-shadow: 0px 0px 12px rgb(0 0 66 / 6%);
`;

export const Header = styled.div<{ selected: boolean }>`
  display: flex;
  align-items: center;
  height: 80px;
  background-color: #ffffff;
  border-radius: 12px;
  font-weight: 700;
  font-size: 16px;
  line-height: 24px;
  letter-spacing: -0.02em;
  position: relative;
  color: #333368;

  ${(props) =>
		props.selected &&
    "border-bottom-left-radius: 0;border-bottom-right-radius: 0;"}
`;

export const Icon = styled.div<{ selected: boolean, isRTL?: boolean }>`
  position: absolute;
  cursor: pointer;
  right: ${(props) => props.isRTL ? "" : "10px"};
  left: ${(props) => props.isRTL ? "16px" : ""};
  transform: ${(props) =>
		props.selected ? "rotate(-180deg)" : "rotate(0deg)"};
  transition: transform 500ms ease;
`;

export const Content = styled.div<{ selected: boolean }>`
  max-height: 0;
  color: #8b7f75;
  overflow: hidden;
  transition: all 0.5s cubic-bezier(0, 1, 0, 1);
  ${(props) =>
		props.selected &&
    "height: auto;max-height: 9999px;transition: all 0.5s cubic-bezier(1, 0, 1, 0);"}
`;

export const Options = styled.div<{ selected: boolean }>`
  ${(props) => !props.selected && "display: none;"}
  background-color: white;
  padding: 10px 20px;
  border-radius: 12px;
  border-top-left-radius: 0;
  border-top-right-radius: 0;

  font-weight: 400;
  font-size: 14px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #000042;
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
`;

export const Border = styled.div<{ selected: boolean }>`
  ${(props) => !props.selected && "display: none;"}
  border: 1px dashed #E2E2EE;
`;

export const Img = styled.img`
  padding: 16px;
`;
export const Title = styled.span`
  font-family: ${TypographyENUM.lkSansBold};
  font-style: normal;
  /* font-weight: 700; */
  font-size: 16px;
  line-height: 24px;
`;
