// @lk/ui-library/src/Component/V1/PrimerCard/PrimerCard.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import PrimerCard from "./index";

export default {
	title:"Components/V1/PrimerCard",
	component: PrimerCard,
	argTypes: {
		...generateGeneralArgs(),
		onSignIn: generateArgTypes({
			type: "function",
			required: false,
			description: "onSignIn handler for sign in form",
			category: "Events",
		}),
		onProceed: generateArgTypes({
			type: "function",
			required: false,
			description: "onProceed handler for sign in form",
			category: "Events",
		}),
		countryCode: generateArgTypes({
			type: "string",
			required: true,
			description: "country code",
			category: "Component",
			defaultValue: "+91",
		}),
		showOTPView: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show OTP view",
			category: "Component",
			defaultValue: false,
		}),
		isLoginSuccess: generateArgTypes({
			type: "boolean",
			required: true,
			description: "if login is successful",
			category: "Component",
			defaultValue: false,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show to right to left",
			category: "Status",
			defaultValue: false,
		}),
	}
} as ComponentMeta<typeof PrimerCard>;

const Template: ComponentStory<typeof PrimerCard> = (args) => <PrimerCard {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	isContactLensConsentEnabled: false,
	dataLocale: {
		PAY_NOW: "Pay Now"
	}
};
