// @lk/ui-library/src/Component/V1/PrimerCard/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const PrimerCardWrapper = styled.div`
.inputPrimer { display: block;
    width: 100%;
    height: 34px;
    // padding: 8px 12px;
    font-size: var(--fs-14);
    line-height: 1.42857143;
    color: #555;
    background-color: #fff;
    background-image: none;
    border: 1px solid #ccc;
    border-radius: 4px;
    margin-bottom: 25px;
 };
 .validateError {
    margin-top: 24px;
    display: block;
    margin-top: 5px;
    font-size: var(--fs-10);
    color: #e34934;
 };
 .cardHolderInput {
    color: #242a2f !important;
    font-family: sans-serif !important;
    font-weight: 400 !important;
    font-size: 16px !important;
    padding: 0 16px !important;
    margin-bottom: 0px !important;
		user-select: auto !important;
 }
`;

export const PayLaterButtonWrapper = styled.div`
    display: flex;
	justify-content: right;
	gap: 15px;
	button{
		margin-top:0px;
	}
`;

export const CardNumberRequire = styled.div`
    margin-top: 24px;
    display: block;
    margin-top: 5px;
    font-size: var(--fs-10);
    color: #e34934;
`;


export const PayNowBtn = styled.button<{font: TypographyENUM, checked: boolean, isContactLensConsentEnabled?: boolean }>`
    margin-top: 20px;
	padding: 12px 20px;
	width: 250px;
	height: 48px;
	border-radius: 100px;
	background:  ${props => props.isContactLensConsentEnabled ? props.checked ? "var(--vivid-green-100)" : "#E4E4E4" : "#11DAAC" };
	color: #000042;
	cursor: pointer;
	font-family: ${props => props.font};
	font-style: normal;
	font-weight: 700;
	font-size: 16px;
	line-height: 24px;
	display: flex;
	align-items: center;
	justify-content: center;
	letter-spacing: -0.02em;
	outline: none;
	border: none;
	&:disabled,
	&.disabled {
		color: #9999B3;
		background: #DBDBEA;
		pointer-events: none;
		cursor: disabled;
	};
	span {
		display: inline-flex;
		margin-right: 10px;
		svg {
			width: 6px;
    		height: 12px;
			path{
				stroke: #000042;
			}
		}
	};
	.disable {
		pointer-events: none;
	}
`;

// export const PayLaterModal = styled.div`
// 	position: relative;
//     padding: 15px;
//     height: 100%;
// 	max-height: 506px;
// 	box-shadow: 0 5px 15px rgba(0,0,0,.5);
// 	background-color: #fff;
// 	width: 900px;
// 	margin: auto;
// 	top: 15%;
// `;

// export const CloseButton = styled.div`
// 	position: absolute;
//     top: 19px;
//     right: 34px;
// `;

// export const PayLaterContainer = styled.div`
// 	display: flex;
//     align-items: center;
//     justify-content: center;
//     flex-direction: column;
//     margin: 20px 0px;
// `;

// export const PayLaterHeader = styled.div`
// 	font-weight: 500;
//     font-size: 24px;
//     line-height: 160%;
//     letter-spacing: 0.15px;
// `;

// export const PayLaterDescription = styled.div`
// 	font-weight: 400;
//     font-size: 16px;
//     line-height: 143%;
//     text-align: center;
//     letter-spacing: 0.15px;
//     margin: 10px 20px;
//     width: 640px;
//     height: 46px;
// `;

// export const PayLaterSubText = styled.div`
// 	font-weight: 400;
//     font-size: 16px;
//     letter-spacing: 0.15px;
//     margin-top: 30px;
// `;

// export const PayLaterButtonContainer = styled.div`
// 	display: flex;
//     align-items: center;
//     justify-content: center;
//     flex-direction: column;
//     margin: 20px 0px;
// `;

// export const PayLaterButtonPrimary = styled.div`
// 	color: #fff;
//     background: #00BBC6;
//     font-size: 14px;
//     display: block;
//     border: none;
//     padding: 12px 20px;
//     letter-spacing: 2px;
//     border-radius: 5px;
//     min-width: 290px;
//     font-weight: 700;
//     text-align: center;
//     cursor: pointer;
// `;

// export const PayLaterButtonSecondary = styled.div`
// 	font-weight: 600;
//     font-size: 16px;
//     line-height: 26px;
//     letter-spacing: 0.47px;
//     text-transform: uppercase;
//     color: #00B0BF;
//     margin-top: 10px;
//     cursor: pointer;
// `;
// export const ModalWrapper = styled.div`
// 	position: fixed;
//     margin: 0 auto;
//     top: 0;
//     left: 0;
//     padding-top: 10px;
//     z-index: 1200 !important;
//     overflow-x: hidden;
//     overflow-y: hidden;
//     height: 100vh;
// 	width: 100vw;
// `;
