// @lk/ui-library/src/Component/V1/Cart/PowerTable/PowerTable.styles.tsx
import styled from "styled-components";

export const Container = styled.div`
	cursor: pointer;
`;

export const Header = styled.div<{ showTable?: boolean }>`
	display: flex;
	justify-content: space-between;
	align-items: center;
	margin: 10px 0;
	font-size: var(--fs-12);
	font-family: var(--font-lksans-regular);
	color: var(--dark-blue-100);
	span {
		u {
			font-family: var(--font-lksans-bold);
		}
		&:first-child {
			display: flex;
			align-items: center;
			gap: 5px;
			svg {
				transition: transform ease-in-out 0.2s;
				transform: ${props => props.showTable ? "rotate(180deg)" : ""};
			}
		}
		&:last-child {
			font-size: var(--fs-10);
			color: var(--text);
		}
	}
	z-index: 99;
`;

export const TableContainer = styled.div<{ showTable?: boolean }>`
	overflow: hidden;
	transition: height ease-in-out 0.2s;
	height: ${props => props.showTable ? "130px" : "0"};
`;

export const Table = styled.div<{ fullWidth?: boolean }>`
	width: ${props => props.fullWidth ? "100%" : "296px"};
	box-sizing: border-box;
	border: 1px solid var(--purple-haze);
	border-radius: 8px;
	letter-spacing: -0.02em;
	line-height: 18px;
	overflow: hidden;
	font-size: 12px;
	background-color: var(--white);
	display: table;
	/* display: flex; */
	/* gap: 3px; */
	/* flex-direction: column; */
`;

export const Head = styled.div`
	/* display: flex; */
	/* align-items: center; */
	/* flex-flow: row wrap; */
	/* justify-content: space-evenly; */
	background-color: #F5F5FF;
	font-family: var(--font-lksans-bold);
	color: var(--dark-blue-100);
	text-align: center;
	display: table-row;
`;

export const Row = styled.div`
	/* display: flex; */
	/* align-items: center; */
	/* justify-content: space-evenly; */
	font-family: var(--font-lksans-regular);
	color: var(--text);
	align-items: center;
	display: table-row;
`;

export const Col = styled.div`
	text-align: center;
	padding: 8px;
	display: table-cell;
`;

export const HeadCol = styled.div`
	padding: 12px 8px;
	display: table-cell;
	text-transform: uppercase;
`;

export const Box = styled.div<{ boxMax?: boolean }>`
	width: ${props => props.boxMax ? "33%" : ""};
	padding: 12px 8px;
	text-align: center;
	color: var(--dark-blue-100);
	font-family: var(--font-lksans-bold);
	display: table-cell;
`;
