// @lk/ui-library/src/Component/V1/StudioFlow/OtherStores/OtherStores.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../../StorybookHelper";
import { OtherStores } from "./OtherStores.component";
import { action } from "@storybook/addon-actions";
import { Store } from "./OtherStores.types";

export default {
	title: "Components/v1/StudioFlow/OtherStores",
	component: OtherStores,
	argTypes: {
		selected: generateArgTypes({
			type: "object",
			category: "Component",
			required: true,
			description: "Tells which store is selected",
			defaultValue: false
		}),
		stores: generateArgTypes({
			type: "array",
			category: "Component",
			required: true,
			description: "Array of store data",
			defaultValue: [{
				"code": "LKST401",
				"storeType": "LENSKART",
				"name": "Funan",
				"address": "Basement 1, #B1-08/09, 107, North Bridge Road, Funan, Funan, Opposite Supreme Court, Downtown, Singapore, 179105",
				"openingTime": "12:00 PM",
				"closingTime": "09:00 PM",
				"area": "Funan",
				"landmark": "Opposite Supreme Court",
				"city": "Downtown",
				"state": "Singapore",
				"country": "Singapore",
				"telephone": "69701335",
				"phoneCode": "+65",
				"longitude": "103.850301",
				"latitude": "1.291066",
				"url": "https://stores.lenskart.com/lenskart-com-Funan-Funan-Singapore-391092/Home#book",
				"shortUrl": "http://lnskt.co/LX65uE4E",
				"googleUrl": "https://maps.google.com/maps?cid=10076176020890546085",
				"pincode": "179105",
				"email": "funan@lenskart.sg"
			}]
		}),
		onClick: generateArgTypes({
			type: "function",
			category: "Action",
			required: true,
			description: "Handler for for mouse event on store row",
		})
	},
} as ComponentMeta<typeof OtherStores>;

const Template: ComponentStory<typeof OtherStores> = (args) => (
	<OtherStores {...args} />
);
export const Overview = Template.bind({});

Overview.args = {
	selected: {
		"code": "LKST401",
		"storeType": "LENSKART",
		"name": "Funan",
		"address": "Basement 1, #B1-08/09, 107, North Bridge Road, Funan, Funan, Opposite Supreme Court, Downtown, Singapore, 179105",
		"openingTime": "12:00 PM",
		"closingTime": "09:00 PM",
		"area": "Funan",
		"landmark": "Opposite Supreme Court",
		"city": "Downtown",
		"state": "Singapore",
		"country": "Singapore",
		"telephone": "69701335",
		"phoneCode": "+65",
		"longitude": "103.850301",
		"latitude": "1.291066",
		"url": "https://stores.lenskart.com/lenskart-com-Funan-Funan-Singapore-391092/Home#book",
		"shortUrl": "http://lnskt.co/LX65uE4E",
		"googleUrl": "https://maps.google.com/maps?cid=10076176020890546085",
		"pincode": "179105",
		"email": "funan@lenskart.sg"
	},
	stores: [
		{
			"code": "LKST401",
			"storeType": "LENSKART",
			"name": "Funan",
			"address": "Basement 1, #B1-08/09, 107, North Bridge Road, Funan, Funan, Opposite Supreme Court, Downtown, Singapore, 179105",
			"openingTime": "12:00 PM",
			"closingTime": "09:00 PM",
			"area": "Funan",
			"landmark": "Opposite Supreme Court",
			"city": "Downtown",
			"state": "Singapore",
			"country": "Singapore",
			"telephone": "69701335",
			"phoneCode": "+65",
			"longitude": "103.850301",
			"latitude": "1.291066",
			"url": "https://stores.lenskart.com/lenskart-com-Funan-Funan-Singapore-391092/Home#book",
			"shortUrl": "http://lnskt.co/LX65uE4E",
			"googleUrl": "https://maps.google.com/maps?cid=10076176020890546085",
			"pincode": "179105",
			"email": "funan@lenskart.sg"
		},
		{
			"code": "LKST402",
			"storeType": "LENSKART",
			"name": "Funan",
			"address": "Basement 1, #B1-08/09, 107, North Bridge Road, Funan, Funan, Opposite Supreme Court, Downtown, Singapore, 179105",
			"openingTime": "12:00 PM",
			"closingTime": "09:00 PM",
			"area": "Funan",
			"landmark": "Opposite Supreme Court",
			"city": "Downtown",
			"state": "Singapore",
			"country": "Singapore",
			"telephone": "69701335",
			"phoneCode": "+65",
			"longitude": "103.850301",
			"latitude": "1.291066",
			"url": "https://stores.lenskart.com/lenskart-com-Funan-Funan-Singapore-391092/Home#book",
			"shortUrl": "http://lnskt.co/LX65uE4E",
			"googleUrl": "https://maps.google.com/maps?cid=10076176020890546085",
			"pincode": "179105",
			"email": "funan@lenskart.sg"
		},
		{
			"code": "LKST403",
			"storeType": "LENSKART",
			"name": "Funan",
			"address": "Basement 1, #B1-08/09, 107, North Bridge Road, Funan, Funan, Opposite Supreme Court, Downtown, Singapore, 179105",
			"openingTime": "12:00 PM",
			"closingTime": "09:00 PM",
			"area": "Funan",
			"landmark": "Opposite Supreme Court",
			"city": "Downtown",
			"state": "Singapore",
			"country": "Singapore",
			"telephone": "69701335",
			"phoneCode": "+65",
			"longitude": "103.850301",
			"latitude": "1.291066",
			"url": "https://stores.lenskart.com/lenskart-com-Funan-Funan-Singapore-391092/Home#book",
			"shortUrl": "http://lnskt.co/LX65uE4E",
			"googleUrl": "https://maps.google.com/maps?cid=10076176020890546085",
			"pincode": "179105",
			"email": "funan@lenskart.sg"
		}
	],
	onClick: (store: Store) => action("onClickHandler: " + store.code)("Fired"),
};
