// @lk/ui-library/src/Component/V1/StudioFlow/OtherStores/OtherStores.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const OtherStoresWrapper = styled.div`
  box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
  border-radius: 12px;
  /* padding: var(--pd-16); */
  background-color: var(--white);
  overflow: hidden;
`;

export const StoreRow = styled.div<{ selected: boolean; }>`
  display: flex;
  align-items: center;
  gap: 16px;
  padding: var(--pd-16);
  background: ${({ selected }) => selected ? "var(--ghost-white)" : "var(--white)"};
  border-bottom: 1px dashed #E2E2EE;
  cursor: pointer;
  &:last-child {
    border-bottom: none;
  }
`;

export const ImageContainer = styled.div``;

export const Image = styled.img``;

export const StoreDetailsAndCheckbox = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
`;

export const StoreDetails = styled.div``;

export const Address = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-weight: 700;
  font-size: var(--fs-16);
  line-height: 24px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-100);

`;

export const Details = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: 400;
  font-size: var(--fs-14);
  line-height: 20px;
  letter-spacing: -0.02em;
  color: var(--dark-blue-75);
  padding-top: var(--pd-4);
`;
