// @lk/ui-library/src/Component/V1/SequenceTab/SequenceTab.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import SequenceTab from "./index";

export default {
	title: "Components/v1/SequenceTab",
	component: SequenceTab,
	argTypes: {
		...generateGeneralArgs(),
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
		btText: generateArgTypes({
			type: "string",
			required: true,
			description: "Main text",
			category: "General",
		}),
		topText: generateArgTypes({
			type: "string",
			required: false,
			description: "Top text",
			category: "General",
		}),
		currentlySelected: generateArgTypes({
			type: "boolean",
			required: true,
			description: "is the tab selected",
			category: "General",
		}),
		showDividerLine: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Whether to show the dashed line",
			category: "General",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show to right to left",
			category: "Staus",
		}),
	},
} as ComponentMeta<typeof SequenceTab>;

const Template: ComponentStory<typeof SequenceTab> = args => <SequenceTab {...args} />;
export const Overview = Template.bind({});
Overview.args = {
	topText: "Power Type",
	btText: "Bifocal/Progressive",
	currentlySelected: true,
	enabled: true,
	id: "abc1",
	onClick: (id: any) => console.log(`${id} is selected`),
	showDividerLine: true
};
