// @lk/ui-library/src/Component/V1/SequenceTab/styles.tsx
import styled from "styled-components";

export const SequenceContainer = styled.div`
	-webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
	display: flex;
	text-align: -webkit-center;
	margin-top: 10px;
	margin: 0;
    padding: 0;
`;
export const SqHolder = styled.div<{isRTL: boolean}>`
	text-align: left;
	display: flex;
	font-family: rajdhani-regular;
    font-style: normal;
    font-weight: var(--fw-bold);
    font-size: 16px;
    line-height: 20px;
    color: #c5c3c3;
    cursor: var(--pointer);
    -ms-flex-item-align: end;
    align-self: flex-end;
	margin: 0;
    padding: 0;
	flex-direction: ${ props => props.isRTL ? "row-reverse" : ""};
`;
export const SqTextContainer = styled.div<{enabled: boolean;currentlySelected: boolean;}>`
	padding: 10px;
	-webkit-box-orient: vertical;
	-webkit-box-direction: normal;
	-ms-flex-direction: column;
	flex-direction: column;
	margin-top: 3px;
	display: flex;
	cursor: ${props => props.enabled ? "pointer" : "not-allowed"};
	${props => props.currentlySelected && "border-bottom: 2px solid #00bac6"};
	${props => props.currentlySelected && "color: #00bac6"};
`;
export const SqTopText = styled.div`
	color: #8b9a98;
	font-style: normal;
	font-weight: 700;
	font-size: 12px;
	line-height: 14px;
	margin: 0;
    padding: 0;
`;
export const SqBottomText = styled.div<{currentlySelected: boolean;}>`
	${props => props.currentlySelected && "color: #00bac6"};
`;
export const SqDividerLine = styled.div<{hasTopText: boolean;}>`
	width: 90px;
	border-bottom: 2px dashed #999;
	display: -webkit-box;
	display: -ms-flexbox;
	display: flex;
	-ms-flex-item-align: center;
	align-self: center;
	padding: 0;
	margin: 0;
	${props => props.hasTopText && "margin-top: 13px"};
`;
