// @lk/ui-library/src/Component/V1/Auth/SignUp/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const ErrorField = styled.div`
  color: #d90502;
  font-size: 12px;
  padding: 0px 0px 10px 0;
  font-weight: 700;
  line-height: 16px;
  letter-spacing: -.02em;
  font-family: ${TypographyENUM.lkSansBold};
`;
export const InfoText = styled.div< {isMobileView?: boolean} >`
	margin-bottom: 24px;
	font-family: ${TypographyENUM.lkSansRegular};
	font-style: normal;
	font-weight: 400;
	font-size: ${props => props.isMobileView ? "var(--fs-12)" : "var(--fs-14)"};
	line-height: 20px;
	letter-spacing: -0.03em;
	color: var(--text);
	display: flex;
	&.checkoutSignUp{
		margin-top:20px;
		button{
			margin-left:5px;
		}
	}
	@media screen and (min-width:768px) and (max-width:1630px){
		font-size: 12px;
		margin-bottom:18px;
		margin-top:5px;
	}
	@media screen and (min-width:768px) and (max-width:1439px){
		font-size: 10px;
		margin-bottom:10px;
		margin-top:14px;
		button{
			font-size:10px;
		}
	}
	@media screen and (max-width: 1300px) and (max-height: 900px) and (orientation: landscape) {
		margin-top:0px;
		font-size: 10px;
		margin-bottom: 10px;
		button{
			font-size:10px;
		}
    }
	@media screen and (max-width:767px){
		font-size:var(--fs-12);
	}
`;

export const CheckoutSignUpContainer = styled.div<{font: TypographyENUM; rightMargin?: boolean; isRTL?: boolean }>`
	max-width: 400px;
	width: 400px;
	display: flex;
	flex-direction: column;
	// flex-wrap: wrap;
	// max-height: 370px;
	align-items: flex-start;
	font-family: ${props => props.font};
	${props => props.rightMargin && `
		margin-right: 32px;
	`}
	${props => props.rightMargin && props.isRTL && `
		margin-right: 0;
		margin-left: 32px;
	`}
`;
export const FlexFullCenter = styled.div`
	width: 100%;
	display: flex;
	justify-content: space-around;
`;

export const PolicyAnchor = styled.a`
	text-decoration: underline !important;
	font-weight: 700;
	cursor: pointer;
	color: var(--text);
	font-family: ${TypographyENUM.lkSansBold};
	padding-left: 3px;
	font-size:14px;
	@media screen and (min-width:768px) and (max-width:1439px){
		font-size: 0.9vw;
	}
	@media screen and (max-width: 1300px) and (max-height: 900px) and (orientation: landscape) {
		font-size: 0.9vw;
	}
	@media screen and (max-width:767px){
		font-size:12px;
	}
`;

export const RefSpan = styled.span`
font-family: ${TypographyENUM.lkSansBold};
font-style: normal;
/* font-weight: 700; */
font-size: 14px;
line-height: 20px;
letter-spacing: -0.02em;
color: #333368;
text-decoration: underline;
cursor:pointer;
margin-right:4px;
@media screen and (min-width:768px) and (max-width:1439px){
	font-size: 0.9vw;
}
`;

export const ButtonWrapper = styled.div`
	width: 100%;
	margin-bottom: 10px;
`;
