// @lk/ui-library/src/Component/V1/PDP/ContactLens/ContactLens.styles.tsx
import styled from "styled-components";

export const ContactLensOptionsWrapper = styled.div`
background-color: #fff;
font-family: roboto;
`;

export const CLInfo = styled.div`
font-family: roboto;
padding: 10px;
font-weight: 400;
font-size: 11px;
line-height: 13px;
color: rgba(60,60,60,.54);
`;

export const CLOptionCard = styled.div<{isSelected:boolean}>`
    display: flex;
    flex-direction: column;
    align-items: center;
    flex-shrink: 0;
    object-fit: cover;
    width: 76px;
    position: relative;
    margin: 0 12px;
    overflow: hidden;
    transition: transform 350ms ease-in-out,-webkit-transform 350ms ease-in-out;
    border: ${props => props.isSelected ? "1px solid #00b9c6" : "1px solid #a9a9a9"};
    border-radius: 4px;
    padding: 8px;
`;

export const CLOptionImg = styled.img`
display: block;
    margin: 0 auto;
    width: 52px;
    height: 52px;
    margin: 0;`;


export const CLOptionLabel = styled.label`
    width: 100%;
    margin: 8px 0;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    text-align: center;
    vertical-align: middle;
    font-size: 14px;
    color: gray;
  `;

export const CLOptionCardWrapper = styled.div`
    display:flex;
    overflow-x:scroll;
    ::-webkit-scrollbar {
        display: none;
      }  
    `;
