// @lk/ui-library/src/Component/V1/Cart/CartCard/PowerDetailMobile/PowerDetailMobile.styles.tsx
import styled from "styled-components";

export const Container = styled.div`
	background-color: #EEEEEE;
	border-radius: 5px;
	padding: 10px 0;
	font-family: var(--font-lksans-regular);
	color: #333333;
`;

export const DropdownContainer = styled.div`
	padding: 0px 15px;
`;

export const Dropdown = styled.div`
	display: flex;
	justify-content: space-between;
	align-items: center;
	font-size: 13px;
`;

export const Dropper = styled.span< {displayTable: boolean} >`
	transform: ${props => props.displayTable ? "rotate(180deg)" : "rotate(0deg)"};
`;

export const TableContainer = styled.div<{ displayTable: boolean }>`
	border-top: 1px solid #cecece;
	margin-top: 10px;
	padding-top: 10px;
	font-size: 11px;
	display: ${props => props.displayTable ? "": "none"};
	width: 100%;
`;

export const Table = styled.div`
	display: table;
	width: 100%;
`;

export const HeadRow = styled.div`
	text-transform: uppercase;
	width: 100%;
	display: table-row;
`;

export const Row = styled.div`
	display: table-row;
	width: 100%;
`;

export const PowerCell = styled.div`
	display: table-cell;
	width: 70%;
	padding-bottom: 8px;
`;

export const Cell = styled.div`
	display: table-cell;
	width: 15%;
	text-align: center;
`;

export const BoldCell = styled.div`
	font-family: var(--font-lksans-bold);
	text-align: center;
	display: table-cell;
`;

export const EditPowerContainer = styled.div`
	width: 100%;
	margin-top: 8px;
`;

export const EditPowerButton = styled.button`
	width: 100%;
	color: #00BAC6;
    border: 2px solid #00BAC6;
	border-radius: 5px;
	padding: 10px;
    text-transform: uppercase;
    margin-bottom: 10px;
	background-color: #ffffff;
`;
