// @lk/ui-library/src/Component/V1/SignUpForm/SignUpForm.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { SignUpForm } from "./SignUpForm.component";
import { generateArgTypes, generateGeneralArgs } from "../../../StorybookHelper";
import { TypographyENUM } from "../../../Types/general";

export default {
	title:"Components/V1/SignUpForm",
	component: SignUpForm,
	argTypes: {
		...generateGeneralArgs(),
		onSignUp: generateArgTypes({
			type: "function",
			required: false,
			description: "onSignUp handler for sign up form",
			category: "Events",
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left signupForm",
			category: "Staus",
		}),
		formError: generateArgTypes({
			type: "boolean",
			required: false,
			description: "onSignUp handler for sign up form",
			category: "Components",
		}),
	}
} as ComponentMeta<typeof SignUpForm>;

const Template: ComponentStory<typeof SignUpForm> = args => <SignUpForm {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"signup-form",
	dataLocale:{
		"REGISTER": "Register",
		"FIRST_NAME": "First Name",
		"PASSWORD": "Password",
		"LAST_NAME": "Last Name",
		"EMAIL_ADDRESS": "Email Address",
		"MOBILE": "Mobile",
		"REFERRAL_CODE": "Referral code",
		"T_C_APPLY": "T&C Apply",
		"OPTIONAL": "Optional",
		"HAVE_REFERRAL": "Have a referral code",
		"SPECIAL_CHARACTERS_NOT_ALLOWED": "Special characters are not allowed.",
		"THIS_IS_REQUIRED_FIELD": "This is a required field.",
		"PLEASE_ENTER_VALID_EMAIL_ADDRESS_EG": "Please enter a valid email address e.g. johndoe@domain.com.",
		"PASSWORD_SHOULD_BE_MORE_THAN": "Password should be more than 6 characters.",
	},
	font:TypographyENUM.defaultBook,
	onSignUp: () => console.log("api called and form submitted"),
};

export const signUpFormWithError = Template.bind({});

signUpFormWithError.args = {
	...Overview.args,
	formError:"EmailId already exists:sample.user@email.com"
};
