// @lk/ui-library/src/Component/V1/SignUpForm/SignUpForm.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const SignUpFormContainer = styled.form<{styleFont:TypographyENUM}>`
	padding:var(--pd-10);
	display:flex;
	flex-direction:column;
	gap:var(--spacing-md);
	font-size:var(--fs-13);
	font-family:${props => props.styleFont};
	width:100%;
`;

export const UpdateOnWhatsappWrapper = styled.div`
	display:flex;
	align-items:center;
	gap:var(--spacing-sm);
	justify-content:center;
`;

export const CheckboxInput = styled.input`
	accent-color:var(--blashish-gray);
	width:1.25em;
	height:1.25em;
	cursor: var(--pointer);
	user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const ImageWhatsapp = styled.img`
	max-height: 2em;
`;

export const CheckboxLabel = styled.label`
	display:flex;
	align-items:center;
	justify-content:center;
	gap:var(--spacing-sm);
	cursor: var(--pointer);
`;

export const AnchorTag = styled.a`
	font-size:var(--fs-12);
	color:var(--dark-grey);
	text-align:right;
	&:hover{
		text-decoration:underline;
	}
`;

export const ReferralText = styled.p`
	cursor: var(--pointer);
	margin:0;
	color:var(--blashish-gray);
	&:hover{
		text-decoration:underline;
	}
`;

export const ReferralInputWrapper = styled.div`
	display:flex;
	flex-wrap:wrap;
	align-items:center;
	gap:var(--spacing-sm);
	font-size:var(--fs-12);
	color:var(--dark-grey);
	width:50%;

	*{
		flex-grow:1;
	}

`;

export const InputWrapper = styled.div`
	position:relative;
	width:100%
`;

export const ShowPasswordImage = styled.img<{showPassword:boolean, hasError:boolean}>`
	width:1.78em;
	object-fit: cover;
	object-position:${props => props.showPassword ? "100%" : "0"};
	height:1.25em;
	cursor: var(--pointer);
	position:absolute;
	bottom:${props => props.hasError ? "2em" : "0.5em"};
	right:0.2em;
`;

export const ErrorContainer = styled.div`
	color:var(--error);
    background-color: var(--error-background);
	padding: var(--pd-15);
    border: 1px solid var(--error-border);
    border-radius: var(--border-radius-xxxs);
`;

export const ButtonErrorWrapper = styled.div`

`;
