// @lk/ui-library/src/Component/Common/Grid/Grid.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { generateArgTypes } from "../../../StorybookHelper";
import Grid from "./index";
import ProductImage from "../../V1/ProductImage";

export default {
	title: "Components/Common/Grid",
	component: Grid,
	argTypes: {
		containerCss: generateArgTypes({
			type: "object",
			category: "Component",
			required: false,
			description: "Container CSS",
			defaultValue: ""
		}),
		itemCss: generateArgTypes({
			type: "object",
			category: "Component",
			required: false,
			description: "Item CSS",
			defaultValue: ""
		}),
		children: generateArgTypes({
			type: "node",
			category: "Component",
			required: true,
			description: "Grid items",
			defaultValue: ""
		}),
	}
} as ComponentMeta<typeof Grid>;

const Template: ComponentStory<typeof Grid> = (args) => <Grid {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	containerClassName: "grid-container-class",
	containerId: "grid-container-id",
	containerCss: { gap: "25px" },
	itemClassName:"grid-item-class",
	itemId:"grid-item-id",
	itemCss: { width: "calc( 100% - 25px )", display: "flex", justifyContent: "center", alignItems: "center" },
	children: <Grid.Item><ProductImage imageSrc="https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/480x480/9df78eab33525d08d6e5fb8d27136e95//v/i/vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_vincent-chase-vc-e13445-fm-recgle-act-meal-sen-c1-eyegl0sses_G_5142.jpg" /></Grid.Item>
};
