// @lk/ui-library/src/Component/V1/MyAccount/MyOrder/MyOrder.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import MyOrder from ".";

export default {
	title: "Components/V1/MyAccount/MyOrder",
	component: MyOrder
} as ComponentMeta<typeof MyOrder>;

const Template: ComponentStory<typeof MyOrder> = (args) => <MyOrder {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		LOGOUT: "LogOut",
		GET_ORDER_UPDATES_ON_WHATSAPP: "Get Order Updates on WhatsApp",
		ORDER_PAGE: "Orders Page",
		ORDER_ID: "Order ID",
		DATE: "Date",
		TOTAL_PRICE: "Total Price",
		NEED_HELP: "Need Help",
		ORDER_DETAIL: "Order_Detail",
		PAYMENT_PENDING: "Payment Pending",
		COMPLETE_YPUR_PAYMENT_TO_PROCESS_ORDER: "Complete your payment to process order",
		CURRENCY_SYMBOL: "₹",
		PAY_NOW: "Pay Now",
		CONFIRM_ORDER: "Confirm Order",
		EARLY: "Early",
		ENTER_PRESCRIPTION_MANAULLY: "Enter Prescription Manually",
		USE_SAVED_PRESCRIPTION: "Use Saved Prescription",
		UPLOAD_PRESCRIPTION: "Upload Prescription",
		REQUEST_CALL_BACK: "Request A Call Back",
		SEND_VIA_EMAIL: "Send It Via Email"
	},
	orderStatusRedis: {},
	handleManuallprescription: () => console.log(""),
	onClickWhatsappToggle: () => console.log("whatsapp toggle"),
	whatsAppActive: false,
	clickOnPage: () => console.log("pagination"),
	//formatDate: (createdAt: string, pattern:string, year:string) => console.log(createdAt, pattern, year),
	orderList: [
		{
			id: 1930491166,
			legacyEntityId: 51210568,
			cartId: 501585879,
			storeId: 1,
			customerId: 40720011,
			customerEmail: "lenskart.test52@gmail.com",
			customerWalletId: "3050000014272841",
			type: "NORMAL",
			childOrderIds: [],
			quantity: 1,
			itemCount: 1,
			status: {
				status: "PENDING",
				state: "NEW",
				returnable: false,
				cancellable: false,
				comment: "Order placed using cc",
				timestamp: 1669365502798,
				trackingStatus: "PAYMENT_NOT_INITIATED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "ORDER_PLACED"
			},
			statusHistory: [
				{
					status: "PENDING",
					state: "NEW",
					comment: "Order placed using cc",
					isVisibleOnFront: false,
					timestamp: 1669365502000
				}
			],
			items: [
				{
					id: 570068,
					productId: 150437,
					productType: "Eyeglasses",
					relatedItems: [],
					sku: "eye:vincent-chase-vc-e14333-c1-eyeglasses",
					quantity: 1,
					availableQuantity: 1165,
					quantityCancelled: 0,
					quantityInvoiced: 0,
					quantityRefunded: 0,
					quantityShipped: 0,
					quantityReturnable: 0,
					powerRequired: "POWER_NOT_REQUIRED",
					lensCategory: "EYEGLASS",
					lensType: "FRAME_ONLY",
					prescription: {
						showPd: false
					},
					options: [],
					catalogPrices: [
						{
							currencyCode: "INR",
							value: 1700,
							name: "Market Price"
						},
						{
							currencyCode: "INR",
							value: 1700,
							name: "Lenskart Price"
						}
					],
					price: {
						currencyCode: "INR",
						value: 1700,
						name: ""
					},
					framePrice: {
						currencyCode: "INR",
						value: 1700
					},
					amountRefunded: {
						currencyCode: "INR",
						value: 0,
						name: ""
					},
					freeFrame: false,
					gender: "Unisex",
					productTypeValue: "Eyeglasses",
					amount: {
						currencyCode: "INR",
						discounts: [],
						totalDiscount: 0,
						taxes: [
							{
								name: "VAT",
								amount: 204
							}
						],
						totalTax: 204,
						shipping: 0,
						subTotal: 1700,
						total: 1904
					},
					appliedRules: [],
					coupons: [],
					inventory: {
						deliveryStoreId: 0,
						posStoreInventoryId: 0
					},
					brandName: "Vincent Chase",
					name: "Blue Black Rimless Aviator Vincent Chase SLEEK STEEL VC E12421-C4 Eyeglasses",
					modelName: "VC E12421",
					productUrl: "https://www.lenskart.com/vincent-chase-vc-e14333-c1-eyeglasses.html",
					thumbnail:
						"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//v/i/blue-black-rimless-aviator-vincent-chase-sleek-steel-vc-e12421-c4-eyeglasses_g_774728_02_2022.jpg",
					classificationId: "11355",
					frameColour: "Blue",
					status: {
						state: "NEW",
						status: "PENDING",
						returnableStatus: "NON_RETURNABLE",
						timestamp: 1669365502798,
						returnable: false
					},
					statusHistory: [
						{
							state: "NEW",
							status: "PENDING",
							returnableStatus: "NON_RETURNABLE",
							timestamp: 1669365502798,
							returnable: false
						}
					],
					createdAt: 1669365502000,
					updatedAt: 1669365502000,
					frameType: "rimless",
					classification: "eyeframe",
					frameShape: "Aviator",
					frameSize: "Medium",
					frameWidth: "134 mm",
					subcategory: "Eyeglasses-Brands",
					shipToStoreRequired: false,
					isLocalFittingRequired: false,
					productDeliveryType: "B2B",
					reviews: [],
					itemTracking: {
						itemTrackingStatus: "ORDER_PLACED",
						itemTrackingStatusCheckpoint: "ORDER_PLACED",
						status: "ORDER_PLACED",
						statusLabel: "Order Created",
						histories: [
							{
								status: "ORDER_PLACED",
								statusLabel: "Order Created",
								createdAt: 1669365502798,
								updatedAt: 1669365502798,
								current: true,
								isVisible: true,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/OrderCreated.png",
								subTitle: "Payment Pending",
								isEarly: false
							},
							{
								status: "IN_WAREHOUSE",
								statusLabel: "Preparing for Dispatch",
								current: false,
								isVisible: true,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Processing.png",
								description: "Check out how your product is prepared",
								descriptionLink: "MR-ZnDoZx1A",
								descriptionImageLink:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Group_17.png",
								isEarly: false
							},
							{
								status: "IN_TRANSIT",
								statusLabel: "On the Way",
								current: false,
								isVisible: true,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/OnTheWay.png",
								isEarly: false
							},
							{
								status: "DELIVERED_TO_CUSTOMER",
								statusLabel: "Delivered",
								current: false,
								isVisible: true,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/DeliveredToStore.png",
								isEarly: false
							},
							{
								status: "CANCELLED",
								statusLabel: "Cancelled",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Cancelled.png",
								isEarly: false
							},
							{
								status: "RETURN_TO_ORIGIN",
								statusLabel: "Returned to Warehouse",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png",
								isEarly: false
							},
							{
								status: "RETURN_REQUESTED",
								statusLabel: "Return Requested",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png",
								isEarly: false
							},
							{
								status: "RETURN_RESHIPPED",
								statusLabel: "Returned Product Shipped Back",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static1.lenskart.com/media/desktop/img/Sep20/ProductShippedBacktoCustomer.png",
								isEarly: false
							},
							{
								status: "RETURNED",
								statusLabel: "Returned",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png",
								isEarly: false
							},
							{
								status: "RETURN_CANCELLED",
								statusLabel: "Return Cancelled ",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/RetuenCancelled.png",
								isEarly: false
							},
							{
								status: "RETURN_REFUNDED",
								statusLabel: "Refund Completed",
								current: false,
								isVisible: false,
								iconUrl: "https://static1.lenskart.com/media/desktop/img/Sep20/Refund.png",
								isEarly: false
							},
							{
								status: "RETURN_RECEIVED",
								statusLabel: "Return Received",
								current: false,
								isVisible: false,
								iconUrl: "https://static1.lenskart.com/media/desktop/img/Sep20/ReturnReceived.png",
								isEarly: false
							},
							{
								status: "EXCHANGE_CREATED",
								statusLabel: "Exchange Order Created",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static1.lenskart.com/media/desktop/img/Sep20/ExchangeOrderCreated.png",
								isEarly: false
							}
						]
					},
					image:
						"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//v/i/blue-black-rimless-aviator-vincent-chase-sleek-steel-vc-e12421-c4-eyeglasses_g_774628_02_2022.jpg",
					isSplitOrderItem: true,
					isBogoItem: false,
					merchantId: "DEALSKART",
					sellerLabel: "Sold by Dealskart Online Services Pvt Ltd.",
					jit: true,
					flags: {
						showTrackingHistory: true,
						canUpdatePrescription: false
					},
					deliveryType: "FR0_WAREHOUSE_TO_CUSTOMER",
					eligibleForReorder: false,
					prescriptionView: {
						showPd: false,
						pdConfigAvailable: false
					},
					hubCode: "LKH03",
					hsnCode: "90031100",
					activeInventories: [
						{
							productId: 150437,
							storeId: "LKH03",
							facilityCode: "LKH03",
							quantity: 1076,
							priority: 3
						}
					],
					isExpressDelivery: false
				}
			],
			appliedRules: [],
			amount: {
				currencyCode: "INR",
				discounts: [
					{
						type: "implicit",
						amount: 0
					}
				],
				totalDiscount: 0,
				taxes: [
					{
						name: "VAT",
						amount: 204
					}
				],
				totalTax: 204,
				shipping: 0,
				subTotal: 1700,
				total: 1904
			},
			payments: {
				paymentList: [
					{
						method: "cc",
						gateway: "PU",
						amount: 1904
					}
				],
				totalPrepaid: 1904,
				totalCOD: 0
			},
			billingAddress: {
				id: 40983506,
				firstName: "ccc",
				lastName: "vvv",
				phone: "9999999999",
				phoneCode: "+91",
				email: "test@gmail.com",
				addressType: "billing",
				addressline1: "koekod",
				addressline2: "dkmks",
				city: "JAIPUR",
				state: "RAJASTHAN",
				postcode: "302020",
				country: "IN",
				gender: "male"
			},
			shippingAddress: {
				id: 40983505,
				firstName: "ccc",
				lastName: "vvv",
				phone: "9999999999",
				phoneCode: "+91",
				email: "test@gmail.com",
				addressType: "shipping",
				addressline1: "koekod",
				addressline2: "dkmks",
				city: "JAIPUR",
				state: "RAJASTHAN",
				postcode: "302020",
				country: "IN",
				gender: "male"
			},
			otpState: "OTP_NOT_REQUIRED",
			guest: false,
			offer3OrFree: "00000000",
			createdAt: 1669365502798,
			updatedAt: 1669440233000,
			deliveryDate: 1669852800000,
			dispatchDate: 1669420800000,
			etaGenerated: true,
			trackingDetails: [
				{
					createdTime: 1669365502798,
					updatedTime: 1669365502798,
					trackStatus: "ORDER_PLACED",
					status: "ORDER_PLACED",
					details: [
						{
							time: "25 Nov 2:08 PM",
							trackStatus: "ORDER_PLACED"
						}
					]
				}
			],
			deliveryOption: "STANDARD",
			version: 2,
			isLoyalty: false,
			facilityCode: "0",
			isDualCompanyEnabled: true,
			isExchangeEnabled: false,
			isBulkOrder: false,
			shipToStoreRequired: false,
			isLocalFittingRequired: false,
			isPaymentCaptured: false,
			transactionId: "",
			merchantId: "DEALSKART",
			isWhatsappConsented: true,
			lkCountry: "IN",
			metadata: {
				apiClient: "DESKTOP",
				saleSource: "google"
			},
			isMarketPlace: false,
			customerPhone: "6000000077",
			phoneCode: "+91",
			isVirtualOrder: false,
			customerTierName: "gold_premium",
			flags: {
				ORDER_PLACED_NOTIFICATION_SENT: true,
				canUpdateAddress: true,
				itemGroupingEnabled: true
			},
			studioFlow: false,
			isDigitalCart: false,
			gokwikRtoDetails: {
				requestId: "f04d9d0b-f08d-4878-8da0-bd6181d7a58a",
				score: 0
			},
			payLater: false
		},
		{
			id: 1930491072,
			legacyEntityId: 51210474,
			cartId: 501585777,
			storeId: 1,
			customerId: 40720011,
			customerEmail: "lenskart.test52@gmail.com",
			customerWalletId: "3050000014272841",
			type: "NORMAL_SG",
			childOrderIds: [],
			quantity: 1,
			itemCount: 1,
			status: {
				status: "PROCESSING",
				state: "PROCESSING",
				returnable: false,
				cancellable: false,
				comment: "Successful Payment with juspay",
				timestamp: 1669305835645,
				trackingStatus: "ORDER_PLACED",
				cancellableStatus: "NON_CANCELLABLE",
				returnableStatus: "NON_RETURNABLE",
				orderTrackingStatusCheckpoint: "ORDER_PLACED"
			},
			statusHistory: [
				{
					status: "PENDING",
					state: "NEW",
					comment: "Order placed using juspay",
					isVisibleOnFront: false,
					timestamp: 1669305765000
				},
				{
					status: "PROCESSING",
					state: "PROCESSING",
					comment: "Successful Payment with juspay",
					isVisibleOnFront: false,
					timestamp: 1669305835000
				}
			],
			items: [
				{
					id: 569941,
					productId: 152186,
					productType: "Eyeglasses",
					relatedItems: [],
					sku: "eye:john-jacobs-jj-e14652af-c1-eyeglasses",
					quantity: 1,
					availableQuantity: 0,
					quantityCancelled: 0,
					quantityInvoiced: 0,
					quantityRefunded: 0,
					quantityShipped: 0,
					quantityReturnable: 0,
					powerRequired: "POWER_NOT_REQUIRED",
					lensCategory: "EYEGLASS",
					lensType: "FRAME_ONLY",
					prescription: {
						showPd: false
					},
					options: [],
					catalogPrices: [
						{
							currencyCode: "SGD",
							value: 198,
							name: "Market Price"
						},
						{
							currencyCode: "SGD",
							value: 198,
							name: "Lenskart Price"
						}
					],
					price: {
						currencyCode: "SGD",
						value: 198,
						name: ""
					},
					framePrice: {
						currencyCode: "SGD",
						value: 198
					},
					amountRefunded: {
						currencyCode: "SGD",
						value: 0,
						name: ""
					},
					freeFrame: false,
					gender: "Women",
					productTypeValue: "Eyeglasses",
					amount: {
						currencyCode: "SGD",
						discounts: [
							{
								code: "SGTEST98",
								type: "gv",
								amount: 194.04
							}
						],
						totalDiscount: 194.04,
						taxes: [
							{
								name: "VAT",
								amount: 0
							}
						],
						totalTax: 0,
						shipping: 0,
						subTotal: 198,
						total: 3.96
					},
					appliedRules: [],
					coupons: [],
					inventory: {
						deliveryStoreId: 0,
						posStoreInventoryId: 0
					},
					brandName: "John Jacobs",
					name:
						"Silver Pink Transparent Tortoise Full Rim Round John Jacobs Supreme Steel JJ E14652AF-C1 Eyeglasses",
					modelName: "JJ E14652AF",
					productUrl: "https://www.lenskart.com/john-jacobs-jj-e14652af-c1-eyeglasses.html",
					thumbnail:
						"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//j/i/silver-pink-transparent-tortoise-full-rim-round-john-jacobs-supreme-steel-jj-e14652af-c1-eyeglasses_g_8562_04_03_2022.jpg",
					classificationId: "11355",
					frameColour: "Silver",
					status: {
						state: "PROCESSING",
						status: "PROCESSING",
						returnableStatus: "NON_RETURNABLE",
						timestamp: 1669305835645,
						returnable: false
					},
					statusHistory: [
						{
							state: "NEW",
							status: "PENDING",
							returnableStatus: "NON_RETURNABLE",
							timestamp: 1669305765311,
							returnable: false
						},
						{
							state: "PROCESSING",
							status: "PROCESSING",
							returnableStatus: "NON_RETURNABLE",
							timestamp: 1669305835645,
							returnable: false
						}
					],
					createdAt: 1669305765000,
					updatedAt: 1669305765000,
					frameType: "full_rim",
					classification: "eyeframe",
					frameShape: "Round",
					frameSize: "Medium",
					frameWidth: "135 mm",
					subcategory: "Eyeglasses-Brands",
					shipToStoreRequired: false,
					isLocalFittingRequired: false,
					productDeliveryType: "B2B",
					reviews: [],
					uwItemId: 319836645,
					itemTracking: {
						itemTrackingStatus: "ORDER_PLACED",
						itemTrackingStatusCheckpoint: "ORDER_PLACED",
						expectedDeliveryDate: 1669852800000,
						status: "IN_WAREHOUSE",
						statusLabel: "Preparing for Dispatch",
						histories: [
							{
								status: "ORDER_PLACED",
								statusLabel: "Order Created",
								createdAt: 1669305765311,
								updatedAt: 1669305765311,
								current: false,
								isVisible: true,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/OrderCreated.png",
								isEarly: false
							},
							{
								status: "IN_WAREHOUSE",
								statusLabel: "Preparing for Dispatch",
								createdAt: 1669325637000,
								updatedAt: 1669325637000,
								current: true,
								isVisible: true,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Processing.png",
								description: "Check out how your product is prepared",
								descriptionLink: "MR-ZnDoZx1A",
								descriptionImageLink:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Group_17.png",
								isEarly: false
							},
							{
								status: "IN_TRANSIT",
								statusLabel: "On the Way",
								current: false,
								isVisible: true,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/OnTheWay.png",
								isEarly: false
							},
							{
								status: "DELIVERED_TO_CUSTOMER",
								statusLabel: "Delivered",
								current: false,
								isVisible: true,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/DeliveredToStore.png",
								isEarly: false
							},
							{
								status: "CANCELLED",
								statusLabel: "Cancelled",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Cancelled.png",
								isEarly: false
							},
							{
								status: "RETURN_TO_ORIGIN",
								statusLabel: "Returned to Warehouse",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png",
								isEarly: false
							},
							{
								status: "RETURN_REQUESTED",
								statusLabel: "Return Requested",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png",
								isEarly: false
							},
							{
								status: "RETURN_RESHIPPED",
								statusLabel: "Returned Product Shipped Back",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static1.lenskart.com/media/desktop/img/Sep20/ProductShippedBacktoCustomer.png",
								isEarly: false
							},
							{
								status: "RETURNED",
								statusLabel: "Returned",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/Returned.png",
								isEarly: false
							},
							{
								status: "RETURN_CANCELLED",
								statusLabel: "Return Cancelled ",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static5.lenskart.com/images/cust_mailer/23rd-jun-icons/Icons/RetuenCancelled.png",
								isEarly: false
							},
							{
								status: "RETURN_REFUNDED",
								statusLabel: "Refund Completed",
								current: false,
								isVisible: false,
								iconUrl: "https://static1.lenskart.com/media/desktop/img/Sep20/Refund.png",
								isEarly: false
							},
							{
								status: "RETURN_RECEIVED",
								statusLabel: "Return Received",
								current: false,
								isVisible: false,
								iconUrl: "https://static1.lenskart.com/media/desktop/img/Sep20/ReturnReceived.png",
								isEarly: false
							},
							{
								status: "EXCHANGE_CREATED",
								statusLabel: "Exchange Order Created",
								current: false,
								isVisible: false,
								iconUrl:
									"https://static1.lenskart.com/media/desktop/img/Sep20/ExchangeOrderCreated.png",
								isEarly: false
							}
						]
					},
					image:
						"https://static5.lenskart.com/media/catalog/product/pro/1/thumbnail/628x301/9df78eab33525d08d6e5fb8d27136e95//j/i/john-jacobs-jj-e14652af-c1-eyeglasses_152186-204_03_2022.jpg",
					isSplitOrderItem: true,
					isBogoItem: false,
					merchantId: "DEALSKART",
					sellerLabel: "Sold by Dealskart Online Services Pvt Ltd.",
					jit: true,
					flags: {
						showTrackingHistory: true,
						canUpdatePrescription: false
					},
					deliveryDate: 1669852800000,
					dispatchDate: 1669334400000,
					deliveryType: "FR0_WAREHOUSE_TO_CUSTOMER",
					eligibleForReorder: false,
					prescriptionView: {
						showPd: false,
						pdConfigAvailable: false
					},
					hubCode: "LKH03",
					hsnCode: "90031100",
					activeInventories: [
						{
							productId: 152186,
							storeId: "LKH03",
							facilityCode: "LKH03",
							quantity: 935
						}
					],
					isExpressDelivery: false
				}
			],
			appliedRules: [],
			amount: {
				currencyCode: "SGD",
				discounts: [
					{
						type: "implicit",
						amount: 0
					},
					{
						code: "SGTEST98",
						type: "gv",
						amount: 194.04
					}
				],
				totalDiscount: 194.04,
				taxes: [
					{
						name: "VAT",
						amount: 0
					}
				],
				totalTax: 0,
				shipping: 0,
				subTotal: 198,
				total: 3.96
			},
			payments: {
				paymentList: [
					{
						method: "juspay",
						gateway: "JUSPAY",
						amount: 3.96,
						mode: "STRIPE",
						transactionId: "lenskart-70426055-1"
					}
				],
				totalPrepaid: 3.96,
				totalCOD: 0
			},
			billingAddress: {
				id: 40983328,
				firstName: "Test",
				lastName: "Test",
				phone: "96565537",
				phoneCode: "+65",
				email: "test@lenskart.in",
				addressType: "billing",
				addressline1: "Test Address",
				addressline2: "0",
				city: "Singapore",
				state: "Singapore",
				postcode: "388134",
				country: "SG",
				gender: "female"
			},
			shippingAddress: {
				id: 40983327,
				firstName: "Test",
				lastName: "Test",
				phone: "96565537",
				phoneCode: "+65",
				email: "test@lenskart.in",
				addressType: "shipping",
				addressline1: "Test Address",
				addressline2: "0",
				city: "Singapore",
				state: "Singapore",
				postcode: "388134",
				country: "SG",
				gender: "female"
			},
			otpState: "OTP_NOT_REQUIRED",
			guest: false,
			offer3OrFree: "00000000",
			createdAt: 1669305765311,
			updatedAt: 1669305836000,
			deliveryDate: 1669852800000,
			dispatchDate: 1669334400000,
			etaGenerated: true,
			trackingDetails: [
				{
					createdTime: 1669305765311,
					updatedTime: 1669305765311,
					trackStatus: "ORDER_PLACED",
					status: "ORDER_PLACED",
					details: [
						{
							time: "24 Nov 9:32 PM",
							trackStatus: "ORDER_PLACED"
						}
					]
				}
			],
			deliveryOption: "STANDARD",
			version: 13,
			isLoyalty: false,
			facilityCode: "0",
			isDualCompanyEnabled: true,
			isExchangeEnabled: false,
			isBulkOrder: false,
			shipToStoreRequired: false,
			isLocalFittingRequired: false,
			isPaymentCaptured: true,
			transactionId: "",
			merchantId: "DEALSKART",
			isWhatsappConsented: true,
			lkCountry: "SG",
			metadata: {
				apiClient: "DESKTOP"
			},
			isMarketPlace: false,
			customerPhone: "6000000077",
			phoneCode: "+91",
			isVirtualOrder: false,
			customerTierName: "gold_premium",
			flags: {
				ORDER_PLACED_NOTIFICATION_SENT: true,
				isSmsSent: false,
				isEmailSent: false,
				canUpdateAddress: true,
				itemGroupingEnabled: false
			},
			studioFlow: false,
			isDigitalCart: false,
			payLater: false
		}
	]
};
