// @lk/ui-library/src/Component/V1/MyAccount/MyOrder/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const MyOrderContainer = styled.div`
	width: 100%;
`;

export const WhatsAppAndLogoutWrapper = styled.div`
	display: flex;
	justify-content: var(--space-between);
	margin-top: 10px;
	margin-bottom: 30px;
`;

export const WhatsAppLabel = styled.div`
	margin-top: 6px;
	align-items: center;
	display: flex;
	margin-bottom: 12px;
	max-width: 100%;
`;

export const WhatsAppLogo = styled.div`
	width: 44px;
	margin-right: 3px;
	display: inline-block;
`;

export const Image = styled.img`
	max-width: 100%;
	vertical-align: middle;
`;

export const WhatsAppText = styled.span`
	font-weight: 500;
	font-family: ${TypographyENUM.lkSansHairline};
	font-size: var(--fs-13);
`;

export const SwitchContainer = styled.div`padding-left: 0px;`;

export const CmnToggleLabelWrapper =
	styled.label <{checked: boolean}>`
	display: block;
	position: relative;
	cursor: var(--pointer);
	outline: none;
	user-select: none;
	padding: 2px;
	width: 55px;
	height: 25px;
	background-color: var(--white);
	border-radius: var(--border-radius-sm);
	&:before {
		display: block;
		position: absolute;
		top: 1px;
		left: 1px;
		bottom: 1px;
		content: "";
		right: 1px;
		background-color: ${(props) => (!props.checked ? "#8e9bad" : "#45C476")};
		border-radius: var(--border-radius-sm);
		border-radius: white;
		transition: background 0.4s;
		&:checked {
			background-color: #8e9bad;
		}
	}
	&:after {
		display: block;
		position: absolute;
		top: 1px;
		left: ${(props) => (!props.checked ? "1px" : "inherit")};
		right: ${(props) => (props.checked ? "1px" : "inherit")};
		bottom: 1px;
		content: "";
		width: 24px;
		background-color: var(--white);
		border-radius: 100%;
		box-shadow: 0 2px 5px var(--light-shadow);
		transition: margin 0.4s;
		&:checked {
			margin-left: 44px;
		}
	}
`;

export const LogOutContainer = styled.div`margin-top: 20px;`;

export const LogOutButton = styled.button`
	background-color: rgb(61, 176, 166);
	letter-spacing: 1.3px;
	padding: 13px 26px;
	text-transform: var(--uppercase);
	color: var(--white);
	font-size: var(--fs-12);
	border-color: #4cae4c;
	border: 1px solid transparent;
	border-radius: 4px;
	cursor: var(--pointer);
`;

export const OrderPageWithPagination = styled.div`
	display: flex;
	justify-content: var(--space-between);
`;

export const OrderPage = styled.div`
	color: rgb(109, 111, 113);
	text-transform: var(--uppercase);
	margin: 0 0 10px;
	letter-spacing: 1px;
`;

export const OrderListWrapper = styled.div`
	padding-top: 1px;
`;

export const OrderWrapper = styled.div<{showNeedHelp?: boolean}>`
	margin-bottom: ${props => props.showNeedHelp ? "30px" : "80px" };
`;

export const OrderBox = styled.div`
	border: 1px solid #DDDDDD;
	border-radius: 8px;
`;

export const Orderheader = styled.div`
	display: flex;
	font-size: 14px;
	padding: 10px 30px;
	margin: 10px 0px 8px 0px;
	justify-content: var(--space-between);
`;

export const OrderIdDate = styled.div`
	flex-basis: 40%;
	justify-content: var(--space-between);
	display: flex;
`;

export const OrderPriceSpan = styled.div<{mobileView?: boolean;}>`
    display: flex;
    justify-content: flex-end;
    align-items: center;
    gap: 5px;
`;

export const OrderIdSpan = styled.div<{mobileView?: boolean;}>`
    display: flex;
    align-items: center;
    gap: 5px;
`;
export const OrderDateSpan = styled.div<{mobileView?: boolean;}>`
    display: flex;
    align-items: center;
    gap: 5px;
`;

export const Span = styled.span<{mobileView?: boolean;}>`
	${props => !props.mobileView ? `
	color: #999999;
` : `
color: rgba(60, 60, 60, 0.54);
	font-size: 14px;
	font-family: Roboto, sans-serif, Arial, Helvetica, sans-serif;
`}
`;

export const Strong = styled.strong<{mobileView?: boolean;}>`
	font-family: var(--font-default-heavy);
	font-weight: 700;
${props => props.mobileView ? `
	color: rgb(51, 51, 51);
` : `
	color: #333;
	color: rgb(51, 51, 51);
	`}
`;

export const OrderTotal = styled.div<{mobileView?: boolean;}>`
${props => props.mobileView ? `
border-top: 1px solid rgb(221, 221, 221);
	padding-top: 10px;
` : `
text-align: end;
	flex-basis: 60%;
	font-size: 16px;
`}
`;

export const CompltePaymentWrapper = styled.div<{mobileView?: boolean}>`
	box-shadow: 0px 0.5px 1.5px rgb(0 0 0 / 25%);
	display: flex;
	justify-content: var(--space-between);
	border-radius: 8px;
	align-items: center;
	background-color: #ffffff;
	margin: ${props => props.mobileView ? "10px 0 0 0" : "10px 20px 15px 20px"};
	font-family: ${props => props.mobileView ? "var(--font-lksans-regular)" : "FuturaStd-Book, helvetica neue, Helvetica, Arial, sans-serif"};
	font-synthesis: none;
	padding: 12px;
`;

export const CompletePaymentheader = styled.div`
	flex-basis: 78%;
`;

export const CompletePaymentText = styled.div<{ mobileView?: boolean }>`
	font-family: var(--font-lksans-bold);
	color: #333;
	font-size: ${props => props.mobileView ? "11px" : "16px"};
	text-transform:capitalize;
`;

export const CompltePaymentSubText = styled.div<{ mobileView?: boolean }>`
	color: #333;
	font-size: ${props => props.mobileView ? "11px" : "14px"};
`;

export const PayNowButton = styled.a<{ mobileView?: boolean }>`
	text-transform: uppercase;
	flex-basis: ${props => props.mobileView ? "24%" : "20%"};
	border-radius: 4px;
	text-decoration: none;
	font-family: var(--font-lksans-bold);
	background-color: #00bac6;
	color: #ffffff;
	text-align: center;
	cursor: pointer;
	padding: ${props => props.mobileView ? "8px 0px" : "10px 0px"};
	font-size: ${props => props.mobileView ? "10px" : "14px"};
`;

export const OrderItems = styled.div``;

export const Div = styled.div<{ mobileView?: boolean }>`
	${props => props.mobileView ? "width: 100%" : ""};
`;

export const ButtonNeedHelpOrderDetails = styled.div<{mobileView?: boolean;}>`
${props => props.mobileView ? `
	display: flex;
	justify-content: var(--space-between);
	margin-top: 15px;
	width: 100%;
	font-family: var(--font-lksans-regular);
	color: #424242;
` : `
display: flex;
	justify-content: var(--space-between);
	float: right;
	margin-top: 15px;
	gap: 15px;
	`}

`;

export const ButtonNeedhelp = styled.button`
	min-width: 245px;
	letter-spacing: 0.4px;
	border: 1px solid #ddd;
	flex: 1;
	border-radius: 4px;
	color: #666;
	background-color: #ffffff;
	font-size: var(--fs-14);
	padding: 12px;
	cursor: var(--pointer);
`;

export const ButtonOrderDetails = styled.button`
	min-width: 245px;
	letter-spacing: 0.4px;
	border: 1px solid #ddd;
	flex: 1;
	border-radius: 4px;
	color: #666;
	background-color: #ffffff;
	font-size: var(--fs-14);
	padding: 12px;
	// margin-left: 15px;
	cursor: var(--pointer);
`;

export const ReturnExchangeWrapper = styled.div`
	margin: 20px 15px 10px 15px;
	text-align: right;
	display: flex;
	align-items: center;
`;

export const ReturnExchangeButton = styled.button`
	border: 1px solid rgb(0, 186, 198);
	color: rgb(0, 186, 198);
	border-radius: 4px;
	background-color: #ffffff;
	font-size: var(--fs-14);
	padding: 12px;
	margin-bottom: 5px;
	text-transform: var(--uppercase);
	cursor: var(--pointer);
`;

export const BackButtonAndlogOutWrapper = styled.div`
	display: flex;
	justify-content: var(--space-between);
	margin-top: 10px;
	margin-bottom: 20px;
	font-family: "FuturaStd-Medium";
`;

export const BackButton = styled.div`
	text-decoration: none;
	font-weight: 700;
	text-transform: uppercase;
	cursor: pointer;
	font-family: "FuturaStd-Medium";
	color: #00bac6;
	margin-top: 30px;
	font-size: 14px;
	svg path{
		stroke: #00bac6;
	}
`;

export const Icon = styled.i`
    margin-right: 5px;
	color: #00bac6;
	font-size: 12px;
`;

export const MyOrderWapper = styled.div<{mobileView?: boolean;}>`
	${props => props.mobileView ? "padding: 0 10px;" : "margin-top: 10px; display: flex; flex-direction: column"};
`;

export const PriceSpan = styled.span`
	font-family: var(--font-default-heavy);
	font-weight: 700;
	color: #333;

`;

export const ReturnExchangeSection = styled.section`
border-top: 1px solid rgb(234, 234, 234);
padding: 15px;
margin-top: 20px;
`;
export const ReturnExchangeContainer = styled.div`
justify-content: space-between;
align-items: center;
display: flex;
`;
export const DetailTextContainer = styled.div`
margin-right: 20px;
flex: 1;
`;
export const MobileCta = styled.button`
border: 1px solid rgb(0, 186, 198);
text-transform: uppercase;
border-radius: 4px;
color: #00bac6;
/* font-weight: 700; */
font-family: var(--font-lksans-bold);
font-size: 10px;
align-self: flex-start;
padding: 8px;
display: block;
background: #fff;
`;

export const DetailInfoText = styled.span`
	/* font-weight: 700; */
	display: block;
	user-select: none;
	font-size: 11px;
    -webkit-tap-highlight-color: transparent;
	color: #27394e;
	font-family: ${TypographyENUM.lkSansBold};
`;
export const DetailDateText = styled.span`
	margin-top: 5px;
	display: block;
	font-size: 11px;
	color: #27394e;
	font-family: ${TypographyENUM.lkSansRegular};
`;
