// @lk/ui-library/src/Component/V1/PDP/Mobile/TechnicalDetails/TechnicalDetails.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../../Types/general";

export const ProductInfoWrapper = styled.div<{styleFont:TypographyENUM, readMore:boolean}>`
	flex-grow:1;
	display:flex;
	flex-direction:column;
	gap:var(--pd-10);
	font-family:${props => props.styleFont};
	background:#fff;
	/* padding:var(--pd-10); */
	position:relative;
	&.tech-bot-info{
    .pdp-inner-info{
		max-height:${props => props.readMore ? "1000px" : "72px"};
	}
	}
`;

export const ProductDetailsContentInner = styled.div<{readMore:boolean}>`
    transition: all ease 0.5s;
	max-height:${props => props.readMore ? "1000px" : "275px"};
	padding: 2px;
	overflow:hidden;
	svg{
		fill:#18cfa8;
		width:18px;
		height:18px;
		margin-bottom: -4.5px;
		path{
		  stroke:#fff;
		}
	}
`;

export const AboutHeader = styled.div`
	color: #18cfa8;
	font-size:var(--pd-15);
`;

export const BtnWrapper = styled.div<{isRTL?:boolean}>`
	text-align:${props=>props?.isRTL?"left":"right"};
	margin-bottom:-4px;
	margin-top: 5px;
	span{
		color: #18cfa8;
		font-size:var(--pd-12);
	}
`;
export const ProductDetailsContent = styled.div<{styleFont:TypographyENUM}>`
    color: #27394e;
	font-size:var(--pd-13);
	border: 1px solid #eaeff4;
	margin-top:4px;
	border-radius: 4px;
	padding:12px 10px 15px 10px;
	h3{
		color:var(--dark-gray);
		font-size:var(--pd-15);
		margin-bottom:4px;
		font-size:15px;
		font-weight: 500;
		text-transform: uppercase;
		font-family:${props => props.styleFont};
	}
	p{
		color:var(--dark-gray);
		font-size:var(--pd-12);
	}
	 ul{
		padding-left:23px;
		margin-top:12px;
		li{

			font-size: 13px;
    		line-height: 20px;
			strong{
				position:relative;
				a{
					position: absolute;
					right: -21px;
					top: 3px;
				}
			}
		}
	}
`;
