// @lk/ui-library/src/Component/V1/PDP/Mobile/TechnicalDetails/TechnicalDetails.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../../../StorybookHelper";
import { TypographyENUM } from "../../../../../Types/general";
import TechnicalDetails from "./TechnicalDetails";

const mock = [
	{
		"name": "Product Id",
		"value": "147507"
	},
	{
		"name": "Model No.",
		"value": "VC E13037"
	},
	{
		"name": "Weight",
		"value": "24 gm",
		"additionalInfoUrl": {
			url: "https://www.lenskart.com/what_does_eye"
		}
		,
	},
	{
		"name": "Frame Material",
		"value": "Cellulose Acetate",
		"additionalInfoUrl": {
			url: "https://www.lenskart.com/what_does_eye"
		}
	},
	{
		"name": "Temple Material",
		"value": "Cellulose Acetate",
		"additionalInfoUrl": {
			url: "https://www.lenskart.com/what_does_eye"
		}
	},
	{
		"name": "Collection",
		"value": "Classic Acetate",
		"additionalInfoUrl": {
			url: "https://www.lenskart.com/what_does_eye"
		}
	},
	{
		"name": "Frame Size",
		"value": "Medium"
	},
	{
		"name": "Frame Width",
		"value": "136 mm"
	},
	{
		"name": "Frame Dimensions",
		"value": "52-19-140"
	},
	{
		"name": "Frame colour",
		"value": "Transparent Blue"
	},
	{
		"name": "Weight",
		"value": "24 gm"
	},
	{
		"name": "Model No.",
		"value": "VC E13037"
	},
	{
		"name": "Weight",
		"value": "24 gm"
	},
	{
		"name": "Model No.",
		"value": "VC E13037"
	},
	{
		"name": "Weight",
		"value": "24 gm"
	},
	{
		"name": "Model No.",
		"value": "VC E13037"
	},
	{
		"name": "Weight",
		"value": "24 gm"
	},
	{
		"name": "Model No.",
		"value": "VC E13037"
	}
];

export default {
	title:"Components/V1/PDP/mobile/TechnicalDetails",
	component: TechnicalDetails,
	argTypes:{
		font: generateArgTypes({
			type: "string",
			required: true,
			description: "Typography of the component",
			category: "General",
			defaultValue: TypographyENUM.serif,
			control: "radio",
			options: TypographyENUM,
		}),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left ",
			category: "Status",
			defaultValue: "false",
		}),
		aboutText: generateArgTypes({
			type: "string",
			required: true,
			description: "About the product ",
			category: "Component",
			defaultValue: "About the product",
		}),
		productDetailTitle: generateArgTypes({
			type: "string",
			required: true,
			description: "About the product ",
			category: "Component",
			defaultValue: "PRODUCT DETAILS",
		}),
		productDetailText: generateArgTypes({
			type: "string",
			required: true,
			description: "About the product ",
			category: "Component",
			defaultValue: "Sold by Lenskart Solutions Pvt Ltd.",
		}),
		specifications: generateArgTypes({
			type: "array",
			required: true,
			description: "array of product specifications",
			category: "Component"
		})

	}
} as ComponentMeta<typeof TechnicalDetails>;

const Template: ComponentStory<typeof TechnicalDetails> = args => <TechnicalDetails {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	id:"product-details-info",
	font:TypographyENUM.defaultBook,
	aboutText:"About the product",
	productDetailTitle:"PRODUCT DETAILS",
	productDetailText:"Sold by Lenskart Solutions Pvt Ltd.",
	specifications: mock,
	onClickHandler:(url) => {
		console.log(url);
	},
};
