// @lk/ui-library/src/Component/V1/CheckoutMobile/BillDetails/BillDetails.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";
import { BillDetails } from "./BillDetails.component";
import { CartAmountType } from "./BillDetails.types";

export default {
	title: "Components/v1/CheckoutMobile/BillDetails",
	component: BillDetails,
	argTypes: {
		...generateGeneralArgs(),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		})
	},
} as ComponentMeta<typeof BillDetails>;

const Template: ComponentStory<typeof BillDetails> = (args) => <BillDetails {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "price-breakup",
	width: "100",
	priceData: [
		{ label: "Item Total", amount: 999, type: CartAmountType.POSITIVE },
		{ label: "LK Cash", amount: 120, type: CartAmountType.LKCASH },
		{ label: "Total before Tax", amount: 999, type: CartAmountType.TOTAL_BEFORE_TAX },
		{ label: "Taxes", amount: 119.88, type: CartAmountType.POSITIVE },
		{ label: "Convenience Fees", amount: 0, type: CartAmountType.FREE ,shipping: 20.2341 },
		{ label: "Convenience Fees", amount: 0, type: CartAmountType.FREE ,shipping: 0 },
		{ amount: 1118.88, label: "Total payable", type: CartAmountType.TOTAL }
	],
	dataLocale: {
		ITEM_TOTAL: "Item Total",
		LK_CASH: "LK Cash",
		TOTAL_BEFORE_TAX: "Total before tax",
		TAXES: "Taxes",
		CONVENICNCE_FEE: "Convenience Fee",
		TOTAL_PAYABLE: "Total Payable"
	},
	currencyCode: "₹",
	font: TypographyENUM.defaultBook,
	// cartAmount: cartAmount
};
