// @lk/ui-library/src/Component/V1/CheckoutMobile/BillDetails/BillDetails.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const PriceBreakupWrapper = styled.div <{ width: any } >`
  padding: 16px;
  background: #FFFFFF;
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
  border-radius: 12px;

  display:flex;
  flex-direction:column;
`;

export const FlexWrapper = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  justify-content: var(--space-between);
`;
export const InfoIcon = styled.img`
	width: var(--grid-space-desktop);
`;

export const BreakUp = styled.div`
	display: flex;
	flex-direction: row;
  align-items:center;
	justify-content: var(--space-between);
`;

export const HR = styled.hr`
	width: 100%;
	margin-bottom: 15px;
	border-bottom: 1px dashed var(--light-purple-10);
	border-top: none;
	border-right: none;
	border-left: none;
	overflow: hidden;
`;


// export const Text = styled.div<{isDiscount?: boolean, isBold?: boolean, isTotal?: boolean, styleFont: TypographyENUM, isTax?:boolean}>`
// 	font-family:  ${(props) => (props.isBold ? "var(--font-lksans-bold)" : "var(--font-lksans-regular)")};
// 	font-size: ${(props) => (props.isTotal ? props.isTax ? "var(--fs-12)" : "var(--fs-14)" : props.isDiscount ? "var(--fs-14)" : "var(--fs-12)")};
// 	color:  ${(props) => (props.isDiscount ? "#0FBD95;" : "#333368;")};
//   margin-bottom:15px;
// `;

export const Text = styled.div<{isGreen?: boolean, isBold?: boolean, weight?:boolean, smallText?: boolean, styleFont: TypographyENUM}>`
	font-family:  ${(props) => (props.isBold ? "var(--font-lksans-bold)" : "var(--font-lksans-regular)")};
	font-size: ${props =>props.smallText ? "var(--fs-12)" : "var(--fs-14)"};
	color:  ${(props) => (props.isGreen ? "#0FBD95;" : "#333368;")};
  margin-bottom:15px;
	/* font-weight: ${props => props.weight ? "700" : "400"}; */
`;
