// @lk/ui-library/src/Component/V1/ProductCard/ProductCard.types.ts
export interface ProductCardTypes{
	productData:ProductTypeBasic
	dataLocale:any
	isGridView: boolean
}

export interface ProductTypeBasic {
	id: number,
	size: string,
	productName: string,
	showProductRating: boolean,
	productRating: number,
	showPurchaseCount: boolean,
	purchaseCount: number,
	showWishlistCount: boolean,
	wishListCount: number,
	tags: string,
	isColorOptionExtra: boolean,
	colorOptionExtraCount: number,
	classification: string,
	productURL: string,
	productURLWithoutDomain: string,
	// prescriptionType: PrescriptionType[],
	productImage: imageHoverType,
	price: PriceType,
	colorOptions: ColorOptionType[]
	isRTL: boolean
}

export interface imageHoverType {
	url: string,
	hoverURL: string
}

// export interface PrescriptionType {
//     id: string;
//     title: string;
//     isDefault: boolean;
//     isPackageAvailable: boolean;
// }

export interface ColorOptionType {
	id: number,
	color: string,
	colorID: number,
	productURL: string,
	price: PriceType,
	arModel: ARModelType,
	productImage: imageHoverType,
}

export interface ARModelType {
	android: string | null,
	ios: string | null
}


export enum CurrencySymbol {
	INR="₹",
	SGD="$",
	AED = "AED"
}

export enum CurrencyType {
	INR="INR",
	SGD="SGD",
	AED="AED"
}

export interface PriceType {
	symbol: CurrencySymbol,
	currency: CurrencyType,
	basePrice: number,
	lkPrice: number
}
