// @lk/ui-library/src/Component/V1/ProductCard/styles.tsx
import styled from "styled-components";

export const WishlistContainer = styled.div`
position:absolute;
right:var(--spacing-md);
display:flex;
flex-direction:column;
align-items:center;
justify-content:center;
gap:2px;
&:hover .wishlist-icon{
fill: var(--wishlist);
transform: scale(1.2);
transition: all 0.3s;
cursor: var(--pointer);
}
`;

export const WishlistCount = styled.p`
margin:0;
color:var(--darker-grey);
`;

export const ProductImage = styled.img`
width:100%;
padding:15px 15px 0;
margin-bottom:40px;
`;

export const ProductTitle = styled.h4<{isRTL: boolean}>`
white-space: nowrap;
text-overflow: ellipsis;
overflow: hidden;
max-width: 100%;
padding:var(--pd-10) 0 4px;
text-align: ${ props => props.isRTL ? "end" : ""};
`;

export const ProductDetails = styled.div<{isRTL: boolean}>`
display:flex;
gap:var(--spacing-sm);
align-items:center;
justify-content:var(--space-between);
flex-direction: ${ props => props.isRTL ? "row-reverse" : ""};
`;

export const ProductStats = styled.div`
display:flex;
flex-wrap:wrap;
gap:var(--spacing-sm);
align-items:center;
`;

export const Rating = styled.div`
border-radius:var(--border-radius-xs);
background-color:var(--vivid-green-100);
padding:1px var(--pd-5);
color:var(--white);
font-size:var(--fs-12);
`;

export const ProductSubDetails = styled.p`
margin:0;
color: var(--whitish-gray);
font-weight:var(--fw-light);
font-size:var(--fs-14);
`;

export const VerticalPipe = styled.span`
border: 0.5px solid var(--border);
height: 1em;
`;

export const ProductLabel = styled.div`
color:var(--gray);
background-color:var(--warm-grey-50);
border-radius:2px;
font-size:var(--fs-14);
padding:2px 4px;
`;

export const ProductSizeContainer = styled.div<{isRTL: boolean}>`
display:none;
flex-direction: ${ props => props.isRTL ? "row-reverse" : ""};
`;

export const ProductSize = styled.span`
color:var(--darker-grey);
`;

export const ProductDetailsContainer = styled.a`
cursor: var(--pointer);
display:flex;
flex-direction:column;
gap:var(--spacing-sm);
`;

export const ProductContainer = styled.div`
display:flex;
flex-direction:column;
width:100%;
background-color:white;
padding:var(--pd-20);
gap:var(--spacing-sm);
border: 1px solid var(--border);
position:relative;
&:hover{
	padding-bottom:0;
	box-shadow:
	transition: box-shadow 0.2s ease-in-out, -webkit-box-shadow 0.2s ease-in-out;
	box-shadow: 0 2px 5px var(--border);
}
&:hover ${ProductSizeContainer} {
display:flex;
align-items:center;
font-size:var(--fs-16);
margin-bottom:var(--spacing-sm);
display:flex;
flex-wrap:wrap;
align-items:center;
justify-content:var(--space-between);
}
transition: box-shadow 0.2s ease-in-out;

&:hover ${ProductSizeContainer}{
opacity: 1;
visibility: visible;
position: relative;
left: auto;
}

&:hover ${ProductDetailsContainer} {
margin-top: -20px;
transition: margin 0.35s ease-in-out;
}
`;

export const AnchorTag = styled.a`
`;

export const ColorOptionsContainer = styled.div`
display:flex;
flex-wrap:wrap;
gap:2px;
font-size:var(--fs-12);
align-items:center;
`;
