// @lk/ui-library/src/Component/V1/Cart/NewPriceBreakup/NewPriceBreakup.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import NewPriceBreakup from "./index";
import { TypographyENUM } from "../../../../Types/general";
import { CartAmountType } from "./PriceBreakup.types";
// import { cartAmount } from "./mockValue";

export default {
	title: "Components/v1/Cart/NewPriceBreakup",
	component: NewPriceBreakup,
	argTypes: {
		...generateGeneralArgs(),
		width: generateArgTypes({
			type: "string",
			required: false,
			description: "To set the specific width of input",
			category: "Component",
			defaultValue: "50",
		})
	},
} as ComponentMeta<typeof NewPriceBreakup>;

const Template: ComponentStory<typeof NewPriceBreakup> = (args) => <NewPriceBreakup {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "price-breakup",
	width: "100",
	priceData: [
		{ label: "Total Gold VIP price", amount: 999, type: CartAmountType.POSITIVE },
		{ label: "Tax", amount: 119.88, type: CartAmountType.POSITIVE },
		{ label: "Convenience Fees", amount: 0, type: CartAmountType.FREE },
		{ amount: 1118.88, label: "Total payable", type: CartAmountType.TOTAL }
	],
	dataLocale: {
		ITEM_TOTAL: "Item Total",
		LK_CASH: "LK Cash",
		TOTAL_BEFORE_TAX: "Total before tax",
		TAXES: "Taxes",
		CONVENICNCE_FEE: "Convenience Fee",
		TOTAL_PAYABLE: "Total Payable"
	},
	showPolicy: true,
	policyLinks: {
		termsOfUse: "https://www.lenskart.com/",
		privacyPolicy: "https://www.lenskart.com/",
	},
	showCart: true,
	font: TypographyENUM.defaultBook,
	infoText: ["Tax on Eyeglasses 12%", "Second"],
	// infoText: "Tax"
	// cartAmount: cartAmount
};
