// @lk/ui-library/src/Component/V1/Cart/NewPriceBreakup/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const HeadWrapper = styled.div`
	display: flex;
	justify-content: space-between;
	font-size: var(--fs-12);
	font-family: var(--font-lksans-regular);
	color: var(--dark-blue-100);
`;

export const Cart = styled.div`

`;

export const B = styled.span`
	font-family	: var(--font-lksans-bold);
`;

export const ShowCartBtn = styled.button`
	background-color: transparent;
	cursor: pointer;
	border: none;
	text-decoration: underline;
	color: var(--dark-blue-100);
`;

export const PriceBreakupWrapper = styled.div <{ width?: string } >`
	box-sizing: border-box;
	padding: var(--pd-16);
	box-shadow: rgba(99, 99, 99, 0.2) 0px 2px 8px 0px;
	/* border: 1px solid var(--light-purple-10); */
	background-color: var(--white);
	border-radius: var(--border-radius-xs);
	display: flex;
	flex-flow: column;
	gap: 6px;
	@media screen and (max-width:767px){
	  margin-top:20px;
	}
	width: ${props => props.width ? `${props.width}%` : "100%"};
`;

export const FlexWrapper = styled.div`
  display: flex;
  align-items: center;
  width: 100%;
  justify-content: var(--space-between);
`;
export const InfoIcon = styled.img`
	width: var(--grid-space-desktop);
`;

export const BreakUp = styled.div<{marginTop?: boolean}>`
	display: flex;
	flex-direction: row;
	justify-content: var(--space-between);
	margin: 4px 0;

	${props => props.marginTop ? "margin-top: 12px;" : ""};
`;

export const HR = styled.hr`
	width: 100%;
	margin: 0;
	border-bottom: 1px dashed var(--light-purple-10);
	border-top: none;
	border-right: none;
	border-left: none;
	overflow: hidden;
	height: 5px;
	margin-top: 10px;
`;


export const Text = styled.div<{isDiscount?: boolean, isBold?: boolean, isTotal?: boolean, styleFont: TypographyENUM, infoTooltip?: boolean}>`
	font-family:  ${(props) => (props.isBold ? "var(--font-lksans-bold)" : "var(--font-lksans-regular)")};
	font-size: ${(props) => (props.isTotal ? "var(--fs-16)" : "var(--fs-14)")};
	color:  ${(props) => (props.isDiscount ? "#0FBD95" : "var(--dark-blue-100)")};
	${props => props.infoTooltip ? "display: flex; align-items: center; gap: 5px" : ""};
`;

export const Policy = styled.div`
	height: 40px;
	margin-top: -14px;
	font-family: ${TypographyENUM.lkSansBold};
	font-size: var(--fs-14);
	letter-spacing: -0.02em;
	color: var(--dark-blue-75);
	line-height: 20px;
	font-style: normal;
`;

export const A = styled.a`
	cursor: pointer;
	text-decoration: none;
	color: var(--vivid-green-100);
` ;

export const Flex = styled.div`
	display: flex;
	justify-content: space-between;
	width: 100%;
	align-items: center;
`;

export const InfoText = styled.div<{ showInfo: boolean, isMobileView?: boolean }>`
	position: fixed;
	background-color: ${props => props.isMobileView ? "#0b0b35" : "#fff"};
	color: ${props => props.isMobileView ? "white" : "var(--dark-blue-100)"};
	box-shadow: ${props => props.isMobileView ? "" : "0 1px 20px rgb(0 0 0 / 50%);"};
	display: inline;
	bottom: ${props => props.showInfo ? (props.isMobileView ? "10px" : "50px") : "-300px"};
	transition: bottom ease-in-out 1s;
	text-align: center;
	margin: auto;
	left: 50%;
	transform: translateX(-50%);
	padding: ${props => props.isMobileView ? "15px" : "18px 48px"};
	border-radius: ${props => props.isMobileView ? "5px" : "100px"};
	font-family: ${props => props.isMobileView ? TypographyENUM.lkSansRegular : TypographyENUM.lkSansBold};
	font-size: ${props => props.isMobileView ? "10px" : "14px"};
	z-index: 999999;
	/* transform: translateY(10px); */
	display: flex;
	flex-direction: column;
	align-items: flex-start;
`;

export const InfoSpan = styled.div`
	display: flex;
	cursor: pointer;
	img {
		width: 100%;
	}
`;
