// @lk/ui-library/src/Component/Common/SideModal/SideModal.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import SideModal from "./SideModal";

export default{
	title: "Components/Common/SideModal",
	component: SideModal,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: false,
			description: "String used as unique identifier",
			category: "General"
		}),
		show: generateArgTypes({
			type: "boolean",
			required: true,
			description: "show or hide spinner",
			category: "Component",
			defaultValue: true
		}),
		fullPage: generateArgTypes({
			type: "boolean",
			required: false,
			description: "show spinner full page or not",
			category: "Component",
			defaultValue: false
		})
	}
} as ComponentMeta<typeof SideModal>;

const Template: ComponentStory<typeof SideModal> = (args) => <SideModal {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	children: "Hello world",
	isRTL: true,
	title: "Select Lenses",
	centerHeading: true,
	isVisible: false
};
