// @lk/ui-library/src/Component/V1/MyAccount/MyDitto/Myditto.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import MyDitto from "./index";

export default {
	title: "Components/V1/MyAccount/MyDitto",
	component: MyDitto,
	argTypes: {
		...generateGeneralArgs(),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left onKeepMEPosted",
			category: "Staus",
		}),
	}
} as ComponentMeta<typeof MyDitto>;

const Template: ComponentStory<typeof MyDitto> = (args) => <MyDitto {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		MANAGE_YOUR_DITTO: "Manage Your Ditto",
		OR_SELECT_FROM_EXISTING_DITTO: "Or Select From Existing Ditto",
	},
	dittoPrfileData: [
		{ dittoID: "4866d3c5bf4132db5d0fd36ed9ff709c2c410f04", image: "blob:http://localhost:6006/5f00d01b-d9e7-4556-8769-a511941eb01c", created: 1667900312504 },
		{ dittoID: "e68c210badb1dd1e2723b65e145e01c14df024a1", image: "blob:http://localhost:6006/a6a2f385-d202-4c8a-b55c-cfbb8b400615", created: 1667900312504 },
		{ dittoID: "5fdb6cecec372f492ccb56128a846e2c631141cf", image: "blob:http://localhost:6006/57485257-4e02-43c2-af0a-2fb357eba676", created: 1667900312504 },
		{ dittoID: "c722995e27ccf010af9e6145751908cc6bc3e5e9", image: "blob:http://localhost:6006/69d6cdb3-6b86-4137-8291-05975334b4da", created: 1667900312504 },
		{ dittoID: "21adb9acfa0eb4a18312bbf1dd5e458a9eb26884", image: "blob:http://localhost:6006/3faba0c9-761f-4355-84a2-0a1460f6cb23", created: 1667900312504 },
		{ dittoID: "0c3077984ed6b718bf2d16fbfd86e57d3570e49e", image: "blob:http://localhost:6006/45dee6a8-a3c0-4f78-ab0e-d7b0ed4c3ad0", created: 1667900312504 },
		{ dittoID: "1e320a314f0fdc6402be2077a959ebfe30c3de4a", image: "blob:http://localhost:6006/cc09f3b9-d6f5-41d6-84b9-c574b0ebd1ab", created: 1667900312504 }
	],
	deleteDitto: (id:string, count:number) => console.log("ditto delete", id, count),
	defaultDittoId: "1e320a314f0fdc6402be2077a959ebfe30c3de4a",
	setDefaultDitto: (id: string) => console.log("setDefault", id),
	createNewDitto: () => console.log("create new ditto"),
	onSaveDittoName: () => console.log("on save Ditto Name")
};
