// @lk/ui-library/src/Component/V1/MyAccount/MyDitto/styles.tsx
import styled from "styled-components";

export const DittoWrapper = styled.div`margin: 10px;`;

export const DittoProfiles = styled.div``;

export const DittoHeadingWrapper = styled.div`
	margin-top: 20px;
	margin-bottom: 10px;
	display: flex;
	justify-content: space-between;
`;

export const DittoHeading = styled.h2`
	font-family: 'LKFuturaStd-Medium';
	font-weight: 400 !important;
`;

export const DittoProfilesSelect = styled.div`
	padding: 2px 0 10px;
	position: relative;
`;

export const DittoProfilesSelectyInfo = styled.span`
	transform: translateX(-50%);
	left: 50%;
	position: absolute;
	top: 10px;
	background: #fff;
	padding: 0 5px;
`;

export const HorizontalLine = styled.hr`
	margin-top: 20px;
	margin-bottom: 20px;
	border: 0;
	border-top: 1px solid #eee;
`;

export const DittoProfileList = styled.div`
	margin: 0;
	padding: 0;
	width: 100%;
	overflow: hidden;
	display: flex;
	justify-content: left;
	flex-wrap: wrap;
`;

export const DittoProfileListItem = styled.div`
	flex-basis: 30%;
	margin: 0 1% 35px;
	border: 1px solid #ccc;
	background: #efefef;
	padding: 10px;
	box-sizing: border-box;
	display: block;
	position: relative;
	box-shadow: 0 5px 5px #ccc;
	text-align: center;
	transition: 0.8s all;
	transform: scale(1);
	min-height: 250px;
	overflow: hidden;
	:hover .dittoProfiles_list_item_nav {
		display: block;
		bottom: 70px;
	}
`;

export const DittoImage = styled.img`
	max-width: 100%;
	vertical-align: middle;
	border: 0;
`;

export const DittoSubHeading = styled.h4`
	font-size: 16px;
	text-transform: capitalize;
	line-height: normal;
	padding: 0;
	margin: 10px 0 5px;
`;

export const DittoProfileListItemCreated = styled.span`
	font-size: 12px;
	display: block;
	padding-top: 5px;
	color: green;
	letter-spacing: 1px;
`;

export const DittoProfileListItemNav = styled.div`
	display: none;
	position: absolute;
	bottom: 0;
	z-index: 2;
	text-align: center;
	width: 92%;
	color: #fff;
	font-size: 12px;
	text-transform: uppercase;
	letter-spacing: 2px;
`;

export const DittoSetDefault = styled.span`
	background: #f18132;
	border: 1px solid #ef721a;
	padding: 5px 10px;
	margin: 0 5px;
	border-radius: 5px;
	display: inline-block;
	cursor: pointer;
`;

export const DittoDelete = styled.span`
	background: #f18132;
	border: 1px solid #ef721a;
	padding: 5px 10px;
	margin: 0 5px;
	border-radius: 5px;
	display: inline-block;
	cursor: pointer;
`;

export const CreateNewDitto = styled.button`
	color: #fff;
	background-color: #f0ad4e;
	border-color: #eea236;
	display: inline-block;
	margin-bottom: 0;
	font-weight: 400;
	text-align: center;
	white-space: nowrap;
	vertical-align: middle;
	touch-action: manipulation;
	cursor: pointer;
	background-image: none;
	border: 1px solid transparent;
	padding: 6px 12px;
	font-size: 14px;
	line-height: 1.42857143;
	border-radius: 4px;
	user-select: none;
`;

export const Input = styled.input`
border: 0;
border-bottom: 1px dashed #000;
padding: 3px 5px;
margin: 0;
background: 0 0;
text-transform: capitalize;
user-select: initial;
	&::before, &::after {
    	user-select: initial;
	}
`;

export const DittoSaveNameBox = styled.div`
background: #efefef;
position: absolute;
bottom: 30px;
width: 92%;
overflow: hidden;
`;
