// @lk/ui-library/src/Component/V1/Cart/MobileEmptyCart/MobileEmptyCart.styles.tsx
import styled from "styled-components";

export const Container = styled.div`
	width: 100%;
	padding-top: 20px;
`;

export const NeedHelpContainer = styled.div`
display:flex;
justify-content: space-between;
align-items: center;
border-bottom: 1px solid #00000022;
margin-top: -8px;
width: 100vw;
margin-left: -20px;
margin-bottom: 0;
padding: 0 15px;
}
`;

export const CartHeading = styled.span`
font-size: 20px;
`;

export const Flex = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	color: #27394E;
	margin-bottom: 10px;
	h3 {
		font-family: var(--font-lksans-bold);
		font-size: 16px;
		margin-bottom: 10px;
	}
	p {
		font-family: var(--font-lksans-regular);
		font-size: 12px;
		margin-bottom: 15px;
	}
`;

export const ButtonContainer = styled.div<{marginBottom?:string}>`
	width: 100%;
	margin-bottom: ${props=> props.marginBottom ? `${props.marginBottom}` :""}
`;

export const Home = styled.a`
	text-align: center;
	width: 100%;
	cursor: pointer;
	background-color: #18CFA8;
	color: #FFFFFF;
	border-radius: 4px;
	text-transform: uppercase;
	letter-spacing: 1px;
	padding: 10px;
	font-family: var(--font-lkserif-normal);
`;

export const FooterWrapper = styled.div`
	margin: auto -20px;
	width:100vw;
`;
export const EmptyCartImage = styled.div`
font-size: 190px;
margin-bottom: -2rem;
margin-top: -2rem;
`;
