// @lk/ui-library/src/Component/V1/MyAccount/MyPrescription/MyPrescription.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes, generateGeneralArgs } from "../../../../StorybookHelper";
import MyPrescription from "./index";

export default {
	title: "Components/V1/MyAccount/MyPrescription",
	component: MyPrescription,
	argTypes: {
		...generateGeneralArgs(),
		isRTL: generateArgTypes({
			type: "boolean",
			required: true,
			description: "Show right to left onKeepMEPosted",
			category: "Staus",
		}),
	}
} as ComponentMeta<typeof MyPrescription>;

const Template: ComponentStory<typeof MyPrescription> = (args) => <MyPrescription {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale: {
		MY_SAVED_PRESCRIPTION: "My Saved Prescriptions",
		NAME: "Name",
		POWER_TYPE: "Power Type",
		CREATED: "Created",
		EYE: "Eye",
		RIGHT_EYE:"Right EYE",
		LEFT_EYE:"Left EYE",
		SPHERICAL: "Spherical",
		CYLINDRICAL: "Cylindrical",
		AXIS: "Axis",
		AP: "Ap"
	},
	prescriptionList: [{
		id: 4550807,
		left: { sph: "-0.25" },
		powerType: "SINGLE_VISION",
		recordedAt: 1667541209326,
		right: { sph: "+6.00", cyl: "-5.00", axis: "180", ap: "+4.00" },
		showPd: false,
		userName: "tesr test"
	},
	{
		id: 4550807,
		left: { sph: "-0.25" },
		powerType: "SINGLE_VISION",
		recordedAt: 1667541209326,
		right: { sph: "+6.00", cyl: "-5.00", axis: "180", ap: "+4.00" },
		showPd: false,
		userName: "tesr test"
	},
	{
		id: 4550807,
		left: { sph: "+6.00", cyl: "-5.00", axis: "180", ap: "+4.00" },
		powerType: "SINGLE_VISION",
		recordedAt: 1667541209326,
		right: { sph: "+6.00", cyl: "-5.00", axis: "180", ap: "+4.00" },
		showPd: false,
		userName: "tesr test"
	}]
};
