// @lk/ui-library/src/Component/V1/MyAccount/MyPrescription/styles.tsx
import styled from "styled-components";

export const MyPrescriptionWrapper = styled.div`
    flex: 0 49%;
    flex-direction: column;
    margin: 1% 1% 1px 0;
    padding: 20px;
    border: 1px solid #e0e0e0;
    background: #f3f8fa;
`;

export const HeaderWrapper = styled.div``;

export const Heading = styled.h2`
    margin-top: 20px;
    margin-bottom: 10px;
    font-size: 14px;
    font-family: 'Roboto', 'Helvetica', 'Arial', sans-serif;
    line-height: 143%;
    letter-spacing: .15px;
	font-weight:400;
`;

export const PresCriptionWrapper = styled.div`
    display: flex;
    flex-wrap: wrap;
    margin-top: 10px;
`;

export const UserPowerType = styled.div``;

export const Subheading = styled.h3`
    font-size: 14px;
    line-height: 20px;
    font-family: FuturaStd-Heavy;
    display: block;
    text-transform: uppercase;
    letter-spacing: 1.3px;
    margin-top: 20px;
    margin-bottom: 10px;
`;

export const Span = styled.span``;

export const DateSpan = styled.span`
    font-size: 13px;
    font-family: 'FuturaStd-Book';
	font-synthesis: none;
`;

export const P = styled.p`
    font-size: 14px;
    line-height: 20px;
    font-family: FuturaStd-Heavy;
    display: block;
    text-transform: uppercase;
    letter-spacing: 1.3px;
    margin-bottom: 10px;
`;

export const PowerDetailsWrapper = styled.div`

`;

export const AddPowerTable = styled.div`

`;

export const PowerHead = styled.div`
    background: #eceeef;
    margin: 0;
    padding: 0;
    display: table;
    width: 100%;
    text-transform: uppercase;
`;

export const PowerHeading = styled.div`
    background: #384246;
    color: #fff;
    box-shadow: 2px 0 0 0 #fff, 0 2px 0 0 #fff, 2px 2px 0 0 #fff, inset 2px 0 0 0 #fff, inset 0 2px 0 0 #fff;
    font-family: 'LKFuturaStd-Medium';
    font-size: 11px;
    letter-spacing: 1.5px;
    position: relative;
    cursor: pointer;
    vertical-align: middle;
    display: table-cell;
    width: 33.33%;
    padding: 12px 0 9px;
    text-align: center;
`;

export const PowerTableBlock = styled.div`
    background: #eceeef;
    margin: 0;
    padding: 0;
    display: table;
    width: 100%;
`;

export const PowerTableList = styled.div`
    margin: 0;
    padding: 0;
    display: table;
    width: 100%;
`;

export const PowerItemList = styled.div`
    color: #384246;
    text-align: center;
    list-style: none;
    box-shadow: 2px 0 0 0 #fff, 0 2px 0 0 #fff, 2px 2px 0 0 #fff, inset 2px 0 0 0 #fff, inset 0 2px 0 0 #fff;
    font-family: 'LKFuturaStd-Medium';
    font-size: 11px;
    letter-spacing: 1.5px;
    position: relative;
    cursor: pointer;
    display: table-cell;
    width: 33.33%;
    padding: 10px 0 7px;
`;

export const ButtonConatiner = styled.div`
    position: static;
    width: 50%;
    margin: 0 0 10px;
`;

export const SelectPrescriptionButton = styled.div`
    margin: 0;
    padding: 8px 64px;
    width: 88%;
    background: #1cbbb4;
    border-radius: 5px;
    font-size: 16px;
    font-family: FuturaStd-Heavy;
    letter-spacing: 1.2px;
    text-transform: uppercase;
    line-height: 20px;
    color: #fff;
    text-decoration: none;
    box-shadow: 0 3px 0 #19a8a2;
    border-color: transparent;
    font-weight: 500;
    overflow: visible;
    cursor: pointer;
    overflow: visible;
`;
