// @lk/ui-library/src/Component/V1/Header/Header.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { generateArgTypes } from "../../../StorybookHelper";
import Header from "./index";

export default {
	title: "Components/v1/Header",
	component: Header,
	argTypes: {
		id: generateArgTypes({
			type: "string",
			required: true,
			description: "String used as unique identifier",
			category: "General",
		}),
	},
} as ComponentMeta<typeof Header>;

const Template: ComponentStory<typeof Header> = args => <Header {...args} />;
export const Overview = Template.bind({});
Overview.args = {
	id: "Header",
	isLoading: false,
	onWishlistClick: () => null,
	onSignInClick: () => null,
	onTrackOrderClick: () => null,
	onCartClick: () => {console.log("Cart clicked");},
	signedInData: {
		isSignedIn: true,
		username: "test"
	},
	wishlistCount: 1,
	cartCount: 2,
	trendingMenus: [
		{ text: "Ray ban", link: "https://www.lenskart.com/sunglasses/brands/ray-ban-sunglasses.html" },
		{ text: "Eyeglasses", link: "https://www.lenskart.com/eyeglasses.html" },
		{ text: "Sunglasses", link: "https://www.lenskart.com/sunglasses.html" },
		{ text: "Contact Lenses", link: "https://www.lenskart.com/contact-lenses.html" },
		{ text: "Acuvue", link: "https://www.lenskart.com/contact-lenses/most-popular-contact-lenses/acuvue-contact-lenses.html" },
		{ text: "Eyewear Accessories", link: "https://www.lenskart.com/eyewear-accessories.html" },
		{ text: "Oakley", link: "https://www.lenskart.com/sunglasses/brands/oakley-sunglasses.html" },
		{ text: "Mens Sunglasses", link: "https://www.lenskart.com/sunglasses/find-eyewear/mens-sunglasses.html" },
		{ text: "Women Sunglasses", link: "https://www.lenskart.com/sunglasses/find-eyewear/womens-sunglasses.html" },
		{ text: "Aviator", link: "https://www.lenskart.com/sunglasses/frame-shape/aviator-sunglasses.html" },
		{ text: "Purevision", link: "https://www.lenskart.com/contact-lenses/most-popular-contact-lenses/purevision-contact-lenses.html" },
		{ text: "Sunpocket - Folding Sunglasses", link: "https://www.lenskart.com/sunglasses/brands/sunpocket.html" },
		{ text: "Eye Checkup", link: "https://www.lenskart.com/HTO/" },
	],
};
