// @lk/ui-library/src/Component/V1/Header/mockValues3.ts
export const subHeaderData = {
	defaultGender: 1,
	genders: [
		{ id: 1, text: "Man", url: "/", img: "https://static.lenskart.com/media/desktop/img/men_pic.png", submenuIds: ["c1", "c3"], defaultCategoryId: "c1" },
		{ id: 2, text: "Woman", url: "/", img: "https://static.lenskart.com/media/desktop/img/women_pic.png", submenuIds: ["c1"], defaultCategoryId: "c1" },
		{ id: 3, text: "Kids", url: "/", img: "https://static.lenskart.com/media/desktop/img/kid_pic.png", submenuIds: ["c2", "c3"], defaultCategoryId: "c2" },
	],
	categories: [
		{ id: "c1", text: "Classic", subtext: "starting from <strong>₹999</strong>", submenuIds: ["sc1", "sc2", "sc3"] },
		{ id: "c2", text: "Classic", subtext: "starting from <strong>₹1999</strong>", submenuIds: ["sc1", "sc3", "sc4"] },
		{ id: "c3", text: "Premium", subtext: "starting from <strong>₹1999</strong>", submenuIds: ["sc1", "sc2", "sc3", "sc4"] },
	],
	subCategories: [
		{ id: "sc1", text: "Our top picks", submenuIds: ["lk1", "lk3"] },
		{ id: "sc2", text: "Frame type", submenuIds: ["lk2"] },
		{ id: "sc3", text: "Collection", submenuIds: ["lk1", "lk3"] },
		{ id: "sc4", text: "Brand", submenuIds: ["lk1", "lk3", "lk4"] },
	],
	links: {
		lk1: { id: "lk1", text: "New arrivals", url: "/" },
		lk3: { id: "lk3", text: "Best sellers", url: "/" },
		lk4: { id: "lk4", text: "Brand", url: "/" },
		lk2: { id: "lk2", text: "Round Eyeglasses", url: "/" },
		lk5: { id: "lk5", text: "Wayfarer Eyeglasses", url: "/" },
	}
};

export const topLinks = {
	left: [
		{
			label: "Try in 3D",
			url: "https://www.lenskart.com/compare-looks",
		},
		{
			label: "Store Locator",
			url: "https://www.lenskart.com/optical-store/",
		},
		{
			label: "Quality",
			url: "covid-advisory-cms",
		},
		{
			label: "USA",
			url: "https://lenskart.us/?utm_source=lenskart&utm_medium=india&utm_campaign=website",
		},
		{
			label: "Singapore",
			url: "https://lenskart.sg/?utm_source=lenskart&utm_medium=india&utm_campaign=website",
		},
		{
			label: "UAE",
			url: "https://ae.lenskart.com/?utm_source=lenskart&utm_medium=india&utm_campaign=website",
		},
		{
			label: "John Jacobs",
			url: "https://www.johnjacobseyewear.com/",
		},
		{
			label: "Aqualens",
			url: "https://aqualens.in/",
		},
		{
			label: "Franchise Enquiry",
			url: "https://partners.lenskart.com/",
		},
		{
			label: "Cobrowsing",
			url: "https://www.lenskart.com/cobrowsing",
		},
		{
			label: "Engineering Blog",
			url: "https://medium.com/lenskart-engineering",
		}
	],
	right: [
		{
			label: "Contact Us",
			url: "https://www.lenskart.com",
		}
	]
};

export const lastRowData = {
	menu: [
		{
			label: "Eyeglasses",
			url: "",
			id: "lrd1",
			type: "three-col-layout",
			data: {
				defaultGender: 1,
				genders: [
					{ id: 1, text: "Man", url: "/", img: "https://static.lenskart.com/media/desktop/img/men_pic.png", submenuIds: ["c1", "c3"], defaultCategoryId: "c1" },
					{ id: 2, text: "Woman", url: "/", img: "https://static.lenskart.com/media/desktop/img/women_pic.png", submenuIds: ["c1"], defaultCategoryId: "c1" },
					{ id: 3, text: "Kids", url: "/", img: "https://static.lenskart.com/media/desktop/img/kid_pic.png", submenuIds: ["c2", "c3"], defaultCategoryId: "c2" },
				],
				categories: [
					{ id: "c1", text: "Classic", subtext: "starting from <strong>₹999</strong>", submenuIds: ["sc1", "sc2", "sc3"] },
					{ id: "c2", text: "Classic", subtext: "starting from <strong>₹1999</strong>", submenuIds: ["sc1", "sc3", "sc4"] },
					{ id: "c3", text: "Premium", subtext: "starting from <strong>₹1999</strong>", submenuIds: ["sc1", "sc2", "sc3", "sc4"] },
				],
				subCategories: [
					{ id: "sc1", text: "Our top picks", submenuIds: ["lk1", "lk3"] },
					{ id: "sc2", text: "Frame type", submenuIds: ["lk2"] },
					{ id: "sc3", text: "Collection", submenuIds: ["lk1", "lk3"] },
					{ id: "sc4", text: "Brand", submenuIds: ["lk1", "lk3", "lk4"] },
				],
				links: {
					lk1: { id: "lk1", text: "New arrivals", url: "/" },
					lk3: { id: "lk3", text: "Best sellers", url: "/" },
					lk4: { id: "lk4", text: "Brand", url: "/" },
					lk2: { id: "lk2", text: "Round Eyeglasses", url: "/" },
					lk5: { id: "lk5", text: "Wayfarer Eyeglasses", url: "/" },
				}
			}
		},
		{
			label: "Kids Glasses",
			url: "https://www.lenskart.com/optical-store/",
			id: "lrd2",
			type: "img-layout",
			data: [
				{ id: "clD1", text: "Eyeglasses", img: "https://static1.lenskart.com/media/desktop/img/May22/glasses.jpg", url: "/eyeglasses/promotions/all-kids-eyeglasses.html" },
				{ id: "clD2", text: "Zero Power Computer Glasses", img: "https://static1.lenskart.com/media/desktop/img/May22/computer-glasses.jpg", url: "/eyeglasses/promotions/all-kids-computer-glasses.html" },
				{ id: "clD3", text: "Sunglasses", img: "https://static1.lenskart.com/media/desktop/img/May22/Sunnies.jpg", url: "/sunglasses/collections/kids-sunglasses.html" },
			]
		},
		{
			label: "Contact Lenses",
			url: "covid-advisory-cms",
			id: "lrd3",
			type: "list-grid",
			data: [
				{ id: "sc1", text: "Our top picks", links: [{ id: "lk1", text: "New arrivals", url: "/" }, { id: "lk3", text: "Best sellers", url: "/" }] },
				{ id: "sc2", text: "Frame type", links: [{ id: "lk2", text: "Round Eyeglasses", url: "/" }] },
				{ id: "sc3", text: "Collection", links: [{ id: "lk1", text: "New arrivals", url: "/" }, { id: "lk3", text: "Best sellers", url: "/" }] },
				{ id: "sc4", text: "Brand", links: [{ id: "lk1", text: "New arrivals", url: "/" }, { id: "lk3", text: "Best sellers", url: "/" }, { id: "lk4", text: "Brand", url: "/" }] },
			]
		},
		{
			label: "Home Eye-test & trial",
			url: "https://lenskart.us/?utm_source=lenskart&utm_medium=india&utm_campaign=website",
			id: "lrd4",
			type: "two-col-layout-2",
			data: {
				singleImg: true,
				img: {
					url: "https://www.lenskart.com/HTO/",
					img: "https://static1.lenskart.com/media/desktop/img/HomeTryOut.png"
				},
				head: "Get your eyes checked at home",
				content: ["A certified refractionist will visit you with", " latest eye testing machines & 100 trial frames"],
				button: {
					text: "Book appointment",
					link: "https://www.lenskart.com/HTO/"
				}
			}
		},
		{
			label: "Store Locator",
			url: "https://lenskart.sg/?utm_source=lenskart&utm_medium=india&utm_campaign=website",
			id: "lrd5",
			type: "two-col-layout-1",
			data: {
				singleImg: false,
				imgs: [
					{
						url: "https://www.lenskart.com/HTO/",
						img: "https://static.lenskart.com/media/desktop/img/Delhi1.png",
						city: "Delhi"
					},
					{
						url: "https://www.lenskart.com/HTO/",
						img: "https://static.lenskart.com/media/desktop/img/Banglore1.png",
						city: "Bangalore"
					},
					{
						url: "https://www.lenskart.com/HTO/",
						img: "https://static.lenskart.com/media/desktop/img/Mumbai1.png",
						city: "Mumbai"
					},
					{
						url: "https://www.lenskart.com/HTO/",
						img: "https://static.lenskart.com/media/desktop/img/Ahmedabad1.png",
						city: "Ahmedabad"
					},
					{
						url: "https://www.lenskart.com/HTO/",
						img: "https://static.lenskart.com/media/desktop/img/Chennai1.png",
						city: "Chennai"
					},
					{
						url: "https://www.lenskart.com/HTO/",
						img: "https://static.lenskart.com/media/desktop/img/Hyderabad1.png",
						city: "Hyderabad"
					},
					{
						url: "https://www.lenskart.com/HTO/",
						img: "https://static.lenskart.com/media/desktop/img/Cities1.png",
						city: "+ 100 Cities"
					}
				],
				head: "Get your eyes checked at home",
				content: ["A certified refractionist will visit you with", " latest eye testing machines & 100 trial frames"],
				button: {
					text: "Book appointment",
					link: "https://www.lenskart.com/HTO/"
				}
			}
		}
	],
	imageLink: [
		{
			imgLink: "https://static1.lenskart.com/media/desktop/img/May22/3dtryon1.png",
			url: "https://www.lenskart.com/compare-looks",
			altText: "3D try on"
		},
		{
			imgLink: "https://static1.lenskart.com/media/desktop/img/Mar22/13-Mar/blulogo.png",
			url: "https://www.lenskart.com/eyewear/blue-cut-lenses",
			altText: "Blu"
		},
		{
			imgLink: "https://static.lenskart.com/media/desktop/img/Feb22/18-Feb/goldlogo.jpg",
			url: "https://www.lenskart.com/loyalty",
			altText: "Gold"
		}
	]
};

export interface LastRowType {
	menu: {
		label: string,
		url: string,
		id: string,
		type: string,
		data: {
			defaultGender: number,
			genders: { id: number, text: string, url: string, img: string, submenuIds: string[], defaultCategoryId: string }[],
			categories: { id: string, text: string, subtext: string, submenuIds: string[] }[],
			subCategories: { id: string, text: string, submenuIds: string[] }[],
			links: {
				[key: string]: { id: string, text: string, url: string }
			}
		} | { id: string, text: string, img: string, url: string }[] |
		{ id: string, text: string, links: { id: string, text: string, url: string }[] } |
		{
			singleImg: true,
			img: {
				url: string,
				img: string
			},
			head: string,
			content: string[],
			button: {
				text: string,
				link: string
			}
		} |
		{
			singleImg: false,
			imgs: {
				url: string,
				img: string,
				city: string
			}[],
			head: string,
			content: string[],
			button: {
				text: string,
				link: string
			}
		}
	}[],
	imageLink: {
		imgLink: string,
		url: string,
		altText: string
	}[]
}

export const clData = [
	{ id: "sc1", text: "Our top picks", links: [{ id: "lk1", text: "New arrivals", url: "/" }, { id: "lk3", text: "Best sellers", url: "/" }] },
	{ id: "sc2", text: "Frame type", links: [{ id: "lk2", text: "Round Eyeglasses", url: "/" }] },
	{ id: "sc3", text: "Collection", links: [{ id: "lk1", text: "New arrivals", url: "/" }, { id: "lk3", text: "Best sellers", url: "/" }] },
	{ id: "sc4", text: "Brand", links: [{ id: "lk1", text: "New arrivals", url: "/" }, { id: "lk3", text: "Best sellers", url: "/" }, { id: "lk4", text: "Brand", url: "/" }] },
];

export const clDataImg = [
	{ id: "clD1", text: "Eyeglasses", img: "https://static1.lenskart.com/media/desktop/img/May22/glasses.jpg", url: "/eyeglasses/promotions/all-kids-eyeglasses.html" },
	{ id: "clD2", text: "Zero Power Computer Glasses", img: "https://static1.lenskart.com/media/desktop/img/May22/computer-glasses.jpg", url: "/eyeglasses/promotions/all-kids-computer-glasses.html" },
	{ id: "clD3", text: "Sunglasses", img: "https://static1.lenskart.com/media/desktop/img/May22/Sunnies.jpg", url: "/sunglasses/collections/kids-sunglasses.html" },
];

export const homeTest = {
	singleImg: true,
	img: {
		url: "https://www.lenskart.com/HTO/",
		img: "https://static1.lenskart.com/media/desktop/img/HomeTryOut.png"
	},
	head: "Get your eyes checked at home",
	content: ["A certified refractionist will visit you with", " latest eye testing machines & 100 trial frames"],
	button: {
		text: "Book appointment",
		link: "https://www.lenskart.com/HTO/"
	}
};

export const storesData = {
	singleImg: false,
	imgs: [
		{
			url: "https://www.lenskart.com/HTO/",
			img: "https://static.lenskart.com/media/desktop/img/Delhi1.png",
			city: "Delhi"
		},
		{
			url: "https://www.lenskart.com/HTO/",
			img: "https://static.lenskart.com/media/desktop/img/Banglore1.png",
			city: "Bangalore"
		},
		{
			url: "https://www.lenskart.com/HTO/",
			img: "https://static.lenskart.com/media/desktop/img/Mumbai1.png",
			city: "Mumbai"
		},
		{
			url: "https://www.lenskart.com/HTO/",
			img: "https://static.lenskart.com/media/desktop/img/Ahmedabad1.png",
			city: "Ahmedabad"
		},
		{
			url: "https://www.lenskart.com/HTO/",
			img: "https://static.lenskart.com/media/desktop/img/Chennai1.png",
			city: "Chennai"
		},
		{
			url: "https://www.lenskart.com/HTO/",
			img: "https://static.lenskart.com/media/desktop/img/Hyderabad1.png",
			city: "Hyderabad"
		},
		{
			url: "https://www.lenskart.com/HTO/",
			img: "https://static.lenskart.com/media/desktop/img/Cities1.png",
			city: "+ 100 Cities"
		}
	],
	head: "Get your eyes checked at home",
	content: ["A certified refractionist will visit you with", " latest eye testing machines & 100 trial frames"],
	button: {
		text: "Book appointment",
		link: "https://www.lenskart.com/HTO/"
	}
};
