// @lk/ui-library/src/Component/V1/Header/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../Types/general";

export const HeaderWrapper = styled.div`
  font-family: "FuturaStd-Book";
  font-synthesis: none;
  background-color: #fff;
  display: flex;
  flex-direction: column;
  justify-content: var(--space-between);
  width: 100%;
  @media screen and (min-width: 1920px){
	.big-screen-wrapper {
		max-width: 1920px;
		margin: 0 auto !important;
	}
   }
`;
export const TopNavBarWrapper = styled.div`
  width: 100%;
  background: white;
  z-index: 999;
  left: 0;
  transition: 0.7s all ease-in-out;
`;

export const PaddingContainer = styled.div`
	padding: 0 25px;
    margin: 4px 0 0;
	@media screen and (min-width: 1920px){
		padding: 0px;
	}
`;

export const TopLinkContainer = styled.div`
	display: flex;
	flex-direction: row;
	align-items: center;
	justify-content: space-between;
`;

export const ContactUsContainer = styled.div`
	color: var(--dark-blue-100);
	font-size: 10px;
	padding: 4px 35px;
    font-family: ${TypographyENUM.lkSansRegular};
	cursor: pointer;
`;

export const HeaderContainer = styled.div`
  /* min-width: 978px;
  max-width: 1980px; */
  /* display: flex; */
  /* justify-content: center; */
  /* width: 100%; */
  /* flex-flow: column; */
	background-color: #fff;
	position: fixed;
  top: 0;
	left: 0;
  right:0px;
	z-index: 99;
    display: block!important;
	// -webkit-box-shadow: 0 1px 1px 0 rgb(148 150 159 / 30%);
    // box-shadow: 0 1px 1px 0 rgb(148 150 159 / 30%);
	margin: 0 auto;
    /* min-width: 978px; */
    max-width: 1920px;
`;

export const CustomFlex = styled.div<{column?: boolean; borderColor?: string; icons?: boolean}>`
	display: flex;
	flex-direction: ${props => props.column ? "column" : "row"};
	${props => props.borderColor && `border-top: 1px solid ${props.borderColor}`}
	${props => props.icons && `img {
		height: 40px;
		cursor: pointer;
		@media screen and (max-width:1460px){
			height: 27px;
		}
		@media screen and (max-width:1300px){
			height: 22px;
		}
		@media screen and (max-width: 1236px){
			height: 14px;
		}

		@media screen and (max-width:1180px){
			height: 16px;
			width:20px;
		}
		@media screen and (max-width:1099px){
			height: 13px;
            width: 14px;
		}
	}`}
`;
