// @lk/ui-library/src/Component/V1/Cart/AddDeleteGroup/AddDeleteGroup.stories.tsx
import React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import {
	generateArgTypes,
	generateGeneralArgs,
} from "../../../../StorybookHelper";
import AddDeleteGroup from "./index";

export default {
	title: "Components/v1/Cart/AddDeleteGroup",
	component: AddDeleteGroup,
	argTypes: {
		...generateGeneralArgs(),
		quantity: generateArgTypes({
			type: "number",
			required: true,
			description: "show items quantity",
			category: "Component",
		}),
		onDeleteClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for removing items",
			category: "Events",
		}),
		onAddClick: generateArgTypes({
			type: "function",
			required: false,
			description: "onClick handler for adding items",
			category: "Events",
		}),
	},
} as ComponentMeta<typeof AddDeleteGroup>;

const Template: ComponentStory<typeof AddDeleteGroup> = (args) => <AddDeleteGroup {...args} />;

export const Overview = Template.bind({});
Overview.args = {
	id: "AddDeleteGroup",
	quantity: 1,
	onAddClick: () => {console.log("Item Readded Clicked");},
	onDeleteClick: ()=>{console.log("Item Removed from Cart");}
};
