// @lk/ui-library/src/Component/V1/CheckoutMobile/SavedAddressCard/SavedAddressCard.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import { SavedAddressCard } from "./SavedAddressCard.component";
import { generateArgTypes } from "../../../../StorybookHelper";
import { TypographyENUM } from "../../../../Types/general";

export default {
	title: "Components/v1/CheckoutMobile/SavedAddressCard",
	component: SavedAddressCard,
	argTypes: {
		editAddress: generateArgTypes({
			type: "function",
			required: true,
			description: "editAddress handler to edit existing address",
			category: "Events",
		}),
		deleteAddress: generateArgTypes({
			type: "function",
			required: true,
			description: "deleteAddress handler to delete existing address",
			category: "Events",
		}),
		isSelected: generateArgTypes({
			type: "boolean",
			required: true,
			description: "iSelected state to check if Address is selected or not",
			category: "state",
		}),
	}
} as ComponentMeta<typeof SavedAddressCard>;

const Template: ComponentStory<typeof SavedAddressCard> = (args) => <SavedAddressCard {...args} />;

export const Overview = Template.bind({});

Overview.args = {
	dataLocale:{
		EDIT: "Edit",
		DELETE: "Delete",
		PH: "Ph:",
		EXPECTED_DELIVERY_BY: "Expected delivery by"
	},
	address:{
		addressline1: "test",
		addressline2: "test",
		city: "Noida",
		country: "IN",
		firstName: "test",
		lastName: "test",
		phone: "9999999999",
		postcode: "878787",
		state: "CHANDIGARH",
		addressLabel: "Home",
		id: 6
	},
	font:TypographyENUM.lkSansRegular,
	isSelected: true,
};
