// @lk/ui-library/src/Component/V1/CheckoutMobile/SavedAddressCard/styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const SavedAddressContainer = styled.div<{styleFont:TypographyENUM, isSelected: boolean}>`
	font-family:${props => props.styleFont};
	display: flex;
	flex-direction: column;
	align-items: flex-start;
	padding: var(--fs-16);
	gap: 12px;
	/* background: ${props => props.isSelected ? "var(--ghost-white)" : "var(--white)"};
	border: 1px solid ${({ isSelected }) => isSelected ? "var(--dark-blue-50)" : "var(--white)"}; */
	box-shadow: 0px 0px 8px rgba(0, 0, 0, 0.1);
	border-radius: 12px;
`;

export const NameAndCheckboxContainer = styled.div`
	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
	padding: 0px;
	width: 100%;
`;

export const AddressLabel = styled.span`
	font-weight: var(--fw-bold);
	font-size: var(--fs-12);
	line-height: 20px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-100);
	font-family: ${TypographyENUM.lkSansBold};
`;

export const CheckboxContainer = styled.span<{isSelected: boolean}>`
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	padding: 0px;
	width: 16px;
	height: 16px;
	background: ${props => props.isSelected ? "var(--dark-blue-100)" : ""};
	border-radius: 32px;
	color: var(--white);
  padding: 5px;
	border: ${props => props.isSelected ? "" : "1px solid #A2A2B8"};
	transition: 0.2s ease all;
`;

export const Address = styled.div`
	font-weight: var(--fw-regular);
	font-size: var(--fs-12);
	line-height: 20px;
	letter-spacing: -0.02em;
	color: var(--dark-blue-75);
`;

export const ActionButtonWrapper = styled.div`
	display: flex;
	flex-direction: row;
	align-items: flex-start;
	padding: 0px;
	gap: 40px;
`;
export const ExpectedDeliveryWrapper = styled.div`
	border-top: 1px dashed var(--light-purple-10);
	padding-top:12px;
	font-weight: 400;
	font-size: var(--fs-12);
	line-height: 20px;
	letter-spacing: -0.02em;
	color: #333368;
	width: 100%;
`;

export const ActionButton = styled.a`
	font-weight: var(--fw-bold);
	font-size: var(--fs-12);
	line-height: 18px;
	display: flex;
	align-items: center;
	letter-spacing: -0.02em;
	text-decoration-line: underline;
	color: #2E2E64;
	font-family: ${TypographyENUM.lkSansBold};
`;
