// @lk/ui-library/src/Component/V1/CheckoutMobile/MorePaymentOptions/MorePaymentOptions.styles.tsx
import styled from "styled-components";
import { TypographyENUM } from "../../../../Types/general";

export const MorePayOptions = styled.div`
  padding: var(--fs-16);
  background: #FFFFFF;
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
  border-radius: var(--fs-12);
  .sectionPayWrapper {
    margin-bottom: 0px;
}
`;

export const Padding=styled.div`
  padding-top: 15px;
`;

export const NetBankingSection = styled.div`
  padding: 40px 16px 16px;
  background: #FBF9F7;
  height:100%;
  overflow-y: scroll;
`;

export const BackArrow = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  padding: 6px 12px;
  width: 44px;
  height: 32px;
  background: #FFFFFF;
  border-radius: 100px;
`;

export const SectionHeader = styled.div`
  font-family: ${TypographyENUM.lkSerifNormal};
  font-style: normal;
  font-weight: var(--fw-regular);
  font-size: 18px;
  line-height: 28px;
  letter-spacing: -0.02em;
  color: #000042;
  margin-right:auto;
  margin-left:20px;
`;

export const NetBankingHeader = styled.div`
  display:flex;
  justify-content:space-between;
  align-items:center;
  margin-bottom:15px;
`;

export const SectionHelp = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-bold);
  font-size: var(--fs-12);
  line-height: 20px;
  display: flex;
  align-items: center;
  letter-spacing: -0.02em;
  text-decoration-line: underline;
  color: #000042;
`;

export const Img = styled.img`
  max-width: 100%;
  height: auto;
  vertical-align: middle;
  padding: 0 4px;
`;

export const NetBankingContent = styled.div`
  margin-top:135px;

  div{
    font-family: ${TypographyENUM.lkSansRegular};
    font-style: normal;
    font-weight: var(--fw-regular);
    font-size: 14px;
    line-height: 24px;
    letter-spacing: -0.02em;
    color: #000042;
  }
`;

export const NetBankingHeaderOuter = styled.div`
  position:fixed;
  background: #FBF9F7;
  width:100%;
  top:0;
  padding: 40px 20px 20px 20px;
  left: 0;
  z-index:99;
`;

export const PopularBanks = styled.div`
  margin-top:20px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 24px 16px 16px;
  width: 100%;
  background: #FFFFFF;
  border-radius: var(--fs-12);
  img{
    width: 58px;
    // height: 40px;
  }

  ul::-webkit-scrollbar {
    display: none;
  }

  ul{
    display: flex;
    align-items: center;
    justify-content: space-between;
    overflow-x: scroll;
    white-space: nowrap;
    li{
      display: block;
      padding: 0;
      text-align: center;
      border: none;
      margin: 0 6px;
      label{
        padding-bottom: 0px;
        justify-content: center;
      }
      span{
        font-weight: var(--fw-regular);;
        font-size: 10px;
        line-height: var(--fs-16)
        letter-spacing: -0.02em;
        color: #66668E;
        margin: 0;
        display: BLOCK;
        text-align: center;
        padding: 0;
      }
      i{
        margin: 0 auto 7px;
        max-width: 58px;
        min-width: 58px;
        height: auto;
      }
    }
`;

export const AllBanks = styled.div`
  margin-top:20px;

  div{
    margin-bottom:20px;
  }
`;


export const Banks = styled.div`
  background: #FFFFFF;
  box-shadow: 0px 0px 12px rgba(0, 0, 66, 0.06);
  border-radius: var(--fs-12);
  padding:16px 16px 4px 16px;

  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);;
  font-size: 14px;
  line-height: 24px;
  letter-spacing: -0.02em;
  color: #333368;
`;

export const BanksSection=styled.div`
  display:flex;
  justify-content:space-between;
  align-items:center;

  img:first-child{
    width: 58px;
    object-fit: contain;
    height: 40px;
  }

  span{
    margin-right:auto;
    margin-left:20px;
  }
`;

export const InputWrapper = styled.div`
  position:relative;
`;

export const SearchIcon = styled.span`
	position:absolute;
	top: 18px;
  left: 18px;
`;

export const CrossIcon = styled.span`
	position:absolute;
	right:18px;
	top:18px;
`;

export const NoRelatedBanks = styled.div`
  font-family: ${TypographyENUM.lkSansRegular};
  font-style: normal;
  font-weight: var(--fw-regular);;
  font-size: 14px;
  line-height: 24px;
  text-align: center;
  letter-spacing: -0.02em;
  color: #333368;
  padding-top:30px;
`;
