// @lk/ui-library/src/Component/V1/CheckoutMobile/MorePaymentOptions/MorePaymentOptions.stories.tsx
import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";
import MorePaymentOptions from "./MorePaymentOptions.component";

export default{
	title: "Components/v1/CheckoutMobile/MorePaymentOptions",
	component: MorePaymentOptions,
	
} as ComponentMeta<typeof MorePaymentOptions>;

const Template: ComponentStory<typeof MorePaymentOptions> = (args) => <MorePaymentOptions {...args}/>;

export const Overview = Template.bind({});

Overview.args = {
	popularBanks : [
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Allahabad Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Bank of Baroda" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Bank of India" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Bank of Maharashtra" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Canara Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Catholic Syrian Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Google Pay" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Syndicate Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Catholic Syrian Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Syndicate Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Allahabad Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
	],

	allBanks: [
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Allahabad Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Bank of Baroda" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Bank of India" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Bank of Maharashtra" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Canara Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Catholic Syrian Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Google Pay" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Syndicate Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Catholic Syrian Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Syndicate Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "Allahabad Bank" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
		{ img: "https://static5.lenskart.com/media/uploads/Size=lg,_Payment_method=GooglePay.png" , name: "HDFC" },
	],

	dataLocale: {
		"NETBANKING": "Netbanking",
		"WALLETS": "Wallets",
		"BANKS_SUPPORTED": "40+ banks supported",
		"PAYTM_AND_MORE": "Paytm, PhonePe, & more",
		"SELECT_BANK": "Select Bank",
		"SEARCH_FOR_A_BANK" :"Search for a Bank",
		"POPULAR_BANKS": "Popular Banks",
		"HELP": "Help",
		"ALL_BANKS":"All Banks"
	}
};
